// kaltura-node-typescript-client/api/types/AdminUserLoginAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdminUserLoginAction = void 0;
var tslib_1 = require("tslib");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'adminUser' action 'login'.
 *
 * Usage: Get an admin session using admin email and password (Used for login to the KMC application)
 *
 * Server response type:         string
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var AdminUserLoginAction = /** @class */ (function (_super) {
    tslib_1.__extends(AdminUserLoginAction, _super);
    function AdminUserLoginAction(data) {
        return _super.call(this, data, { responseType: 's', responseSubType: '', responseConstructor: null }) || this;
    }
    AdminUserLoginAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'adminuser' },
            action: { type: 'c', default: 'login' },
            email: { type: 's' },
            password: { type: 's' },
            partnerId: { type: 'n' }
        });
        return result;
    };
    return AdminUserLoginAction;
}(kaltura_request_1.KalturaRequest));
exports.AdminUserLoginAction = AdminUserLoginAction;
