// kaltura-node-typescript-client/api/types/AdminUserUpdatePasswordAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AdminUserUpdatePasswordAction = void 0;
var tslib_1 = require("tslib");
var KalturaAdminUser_1 = require("./KalturaAdminUser");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'adminUser' action 'updatePassword'.
 *
 * Usage: Update admin user password and email
 *
 * Server response type:         KalturaAdminUser
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var AdminUserUpdatePasswordAction = /** @class */ (function (_super) {
    tslib_1.__extends(AdminUserUpdatePasswordAction, _super);
    function AdminUserUpdatePasswordAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaAdminUser', responseConstructor: KalturaAdminUser_1.KalturaAdminUser }) || this;
    }
    AdminUserUpdatePasswordAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'adminuser' },
            action: { type: 'c', default: 'updatePassword' },
            email: { type: 's' },
            password: { type: 's' },
            newEmail: { type: 's' },
            newPassword: { type: 's' },
            otp: { type: 's' }
        });
        return result;
    };
    return AdminUserUpdatePasswordAction;
}(kaltura_request_1.KalturaRequest));
exports.AdminUserUpdatePasswordAction = AdminUserUpdatePasswordAction;
