// kaltura-node-typescript-client/api/types/AnnotationCloneAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnnotationCloneAction = void 0;
var tslib_1 = require("tslib");
var KalturaAnnotation_1 = require("./KalturaAnnotation");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'annotation' action 'clone'.
 *
 * Usage: Clone cuePoint with id to given entry
 *
 * Server response type:         KalturaAnnotation
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var AnnotationCloneAction = /** @class */ (function (_super) {
    tslib_1.__extends(AnnotationCloneAction, _super);
    function AnnotationCloneAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaAnnotation', responseConstructor: KalturaAnnotation_1.KalturaAnnotation }) || this;
    }
    AnnotationCloneAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'annotation_annotation' },
            action: { type: 'c', default: 'clone' },
            id: { type: 's' },
            entryId: { type: 's' },
            parentId: { type: 's' }
        });
        return result;
    };
    return AnnotationCloneAction;
}(kaltura_request_1.KalturaRequest));
exports.AnnotationCloneAction = AnnotationCloneAction;
