// kaltura-node-typescript-client/api/types/AnnotationUpdateCuePointsTimesAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AnnotationUpdateCuePointsTimesAction = void 0;
var tslib_1 = require("tslib");
var KalturaCuePoint_1 = require("./KalturaCuePoint");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'annotation' action 'updateCuePointsTimes'.
 *
 *
 *
 * Server response type:         KalturaCuePoint
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var AnnotationUpdateCuePointsTimesAction = /** @class */ (function (_super) {
    tslib_1.__extends(AnnotationUpdateCuePointsTimesAction, _super);
    function AnnotationUpdateCuePointsTimesAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaCuePoint', responseConstructor: KalturaCuePoint_1.KalturaCuePoint }) || this;
    }
    AnnotationUpdateCuePointsTimesAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'annotation_annotation' },
            action: { type: 'c', default: 'updateCuePointsTimes' },
            id: { type: 's' },
            startTime: { type: 'n' },
            endTime: { type: 'n' }
        });
        return result;
    };
    return AnnotationUpdateCuePointsTimesAction;
}(kaltura_request_1.KalturaRequest));
exports.AnnotationUpdateCuePointsTimesAction = AnnotationUpdateCuePointsTimesAction;
