// kaltura-node-typescript-client/api/types/AppTokenStartSessionAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.AppTokenStartSessionAction = void 0;
var tslib_1 = require("tslib");
var KalturaSessionInfo_1 = require("./KalturaSessionInfo");
var KalturaSessionType_1 = require("./KalturaSessionType");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'appToken' action 'startSession'.
 *
 * Usage: Starts a new KS (kaltura Session) based on an application authentication token ID
 *
 * Server response type:         KalturaSessionInfo
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var AppTokenStartSessionAction = /** @class */ (function (_super) {
    tslib_1.__extends(AppTokenStartSessionAction, _super);
    function AppTokenStartSessionAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaSessionInfo', responseConstructor: KalturaSessionInfo_1.KalturaSessionInfo }) || this;
    }
    AppTokenStartSessionAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'apptoken' },
            action: { type: 'c', default: 'startSession' },
            id: { type: 's' },
            tokenHash: { type: 's' },
            userId: { type: 's' },
            type: { type: 'en', subTypeConstructor: KalturaSessionType_1.KalturaSessionType, subType: 'KalturaSessionType' },
            expiry: { type: 'n' },
            sessionPrivileges: { type: 's' }
        });
        return result;
    };
    return AppTokenStartSessionAction;
}(kaltura_request_1.KalturaRequest));
exports.AppTokenStartSessionAction = AppTokenStartSessionAction;
