// kaltura-node-typescript-client/api/types/BaseEntryUpdateContentAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BaseEntryUpdateContentAction = void 0;
var tslib_1 = require("tslib");
var KalturaBaseEntry_1 = require("./KalturaBaseEntry");
var KalturaResource_1 = require("./KalturaResource");
var KalturaEntryReplacementOptions_1 = require("./KalturaEntryReplacementOptions");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'baseEntry' action 'updateContent'.
 *
 * Usage: Update the content resource associated with the entry
 *
 * Server response type:         KalturaBaseEntry
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var BaseEntryUpdateContentAction = /** @class */ (function (_super) {
    tslib_1.__extends(BaseEntryUpdateContentAction, _super);
    function BaseEntryUpdateContentAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaBaseEntry', responseConstructor: KalturaBaseEntry_1.KalturaBaseEntry }) || this;
    }
    BaseEntryUpdateContentAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'baseentry' },
            action: { type: 'c', default: 'updateContent' },
            entryId: { type: 's' },
            resource: { type: 'o', subTypeConstructor: KalturaResource_1.KalturaResource, subType: 'KalturaResource' },
            conversionProfileId: { type: 'n' },
            advancedOptions: { type: 'o', subTypeConstructor: KalturaEntryReplacementOptions_1.KalturaEntryReplacementOptions, subType: 'KalturaEntryReplacementOptions' }
        });
        return result;
    };
    return BaseEntryUpdateContentAction;
}(kaltura_request_1.KalturaRequest));
exports.BaseEntryUpdateContentAction = BaseEntryUpdateContentAction;
