// kaltura-node-typescript-client/api/types/BaseEntryAddFromUploadedFileAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BaseEntryAddFromUploadedFileAction = void 0;
var tslib_1 = require("tslib");
var KalturaBaseEntry_1 = require("./KalturaBaseEntry");
var KalturaEntryType_1 = require("./KalturaEntryType");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'baseEntry' action 'addFromUploadedFile'.
 *
 * Usage: Generic add entry using an uploaded file, should be used when the uploaded entry type is not known
 *
 * Server response type:         KalturaBaseEntry
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var BaseEntryAddFromUploadedFileAction = /** @class */ (function (_super) {
    tslib_1.__extends(BaseEntryAddFromUploadedFileAction, _super);
    function BaseEntryAddFromUploadedFileAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaBaseEntry', responseConstructor: KalturaBaseEntry_1.KalturaBaseEntry }) || this;
    }
    BaseEntryAddFromUploadedFileAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'baseentry' },
            action: { type: 'c', default: 'addFromUploadedFile' },
            entry: { type: 'o', subTypeConstructor: KalturaBaseEntry_1.KalturaBaseEntry, subType: 'KalturaBaseEntry' },
            uploadTokenId: { type: 's' },
            type: { type: 'es', subTypeConstructor: KalturaEntryType_1.KalturaEntryType, subType: 'KalturaEntryType' }
        });
        return result;
    };
    return BaseEntryAddFromUploadedFileAction;
}(kaltura_request_1.KalturaRequest));
exports.BaseEntryAddFromUploadedFileAction = BaseEntryAddFromUploadedFileAction;
