// kaltura-node-typescript-client/api/types/BaseEntryUpdateThumbnailFromSourceEntryAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BaseEntryUpdateThumbnailFromSourceEntryAction = void 0;
var tslib_1 = require("tslib");
var KalturaBaseEntry_1 = require("./KalturaBaseEntry");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'baseEntry' action 'updateThumbnailFromSourceEntry'.
 *
 * Usage: Update entry thumbnail from a different entry by a specified time offset (in seconds)
 *
 * Server response type:         KalturaBaseEntry
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var BaseEntryUpdateThumbnailFromSourceEntryAction = /** @class */ (function (_super) {
    tslib_1.__extends(BaseEntryUpdateThumbnailFromSourceEntryAction, _super);
    function BaseEntryUpdateThumbnailFromSourceEntryAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaBaseEntry', responseConstructor: KalturaBaseEntry_1.KalturaBaseEntry }) || this;
    }
    BaseEntryUpdateThumbnailFromSourceEntryAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'baseentry' },
            action: { type: 'c', default: 'updateThumbnailFromSourceEntry' },
            entryId: { type: 's' },
            sourceEntryId: { type: 's' },
            timeOffset: { type: 'n' }
        });
        return result;
    };
    return BaseEntryUpdateThumbnailFromSourceEntryAction;
}(kaltura_request_1.KalturaRequest));
exports.BaseEntryUpdateThumbnailFromSourceEntryAction = BaseEntryUpdateThumbnailFromSourceEntryAction;
