// kaltura-node-typescript-client/api/types/BulkUploadAddAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BulkUploadAddAction = void 0;
var tslib_1 = require("tslib");
var KalturaBulkUpload_1 = require("./KalturaBulkUpload");
var KalturaBulkUploadType_1 = require("./KalturaBulkUploadType");
var kaltura_upload_request_1 = require("../kaltura-upload-request");
/**
 * Build request payload for service 'bulkUpload' action 'add'.
 *
 * Usage: Add new bulk upload batch job
 * Conversion profile id can be specified in the API or in the CSV file, the one in the CSV file will be stronger.
 * If no conversion profile was specified, partner's default will be used
 *
 * Server response type:         KalturaBulkUpload
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaUploadRequest
 */
var BulkUploadAddAction = /** @class */ (function (_super) {
    tslib_1.__extends(BulkUploadAddAction, _super);
    function BulkUploadAddAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaBulkUpload', responseConstructor: KalturaBulkUpload_1.KalturaBulkUpload }) || this;
    }
    BulkUploadAddAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'bulkupload' },
            action: { type: 'c', default: 'add' },
            conversionProfileId: { type: 'n' },
            csvFileData: { type: 'f' },
            bulkUploadType: { type: 'es', subTypeConstructor: KalturaBulkUploadType_1.KalturaBulkUploadType, subType: 'KalturaBulkUploadType' },
            uploadedBy: { type: 's' },
            fileName: { type: 's' }
        });
        return result;
    };
    return BulkUploadAddAction;
}(kaltura_upload_request_1.KalturaUploadRequest));
exports.BulkUploadAddAction = BulkUploadAddAction;
