// kaltura-node-typescript-client/api/types/BusinessProcessCaseAbortAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BusinessProcessCaseAbortAction = void 0;
var tslib_1 = require("tslib");
var KalturaEventNotificationEventObjectType_1 = require("./KalturaEventNotificationEventObjectType");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'businessProcessCase' action 'abort'.
 *
 * Usage: Abort business-process case
 *
 * Server response type:         void
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var BusinessProcessCaseAbortAction = /** @class */ (function (_super) {
    tslib_1.__extends(BusinessProcessCaseAbortAction, _super);
    function BusinessProcessCaseAbortAction(data) {
        return _super.call(this, data, { responseType: 'v', responseSubType: '', responseConstructor: null }) || this;
    }
    BusinessProcessCaseAbortAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'businessprocessnotification_businessprocesscase' },
            action: { type: 'c', default: 'abort' },
            objectType: { type: 'es', subTypeConstructor: KalturaEventNotificationEventObjectType_1.KalturaEventNotificationEventObjectType, subType: 'KalturaEventNotificationEventObjectType' },
            objectId: { type: 's' },
            businessProcessStartNotificationTemplateId: { type: 'n' }
        });
        return result;
    };
    return BusinessProcessCaseAbortAction;
}(kaltura_request_1.KalturaRequest));
exports.BusinessProcessCaseAbortAction = BusinessProcessCaseAbortAction;
