// kaltura-node-typescript-client/api/types/BusinessProcessCaseServeDiagramAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.BusinessProcessCaseServeDiagramAction = void 0;
var tslib_1 = require("tslib");
var KalturaEventNotificationEventObjectType_1 = require("./KalturaEventNotificationEventObjectType");
var kaltura_file_request_1 = require("../kaltura-file-request");
/**
 * Build request payload for service 'businessProcessCase' action 'serveDiagram'.
 *
 * Usage: Server business-process case diagram
 *
 * Server response type:         { url: string }
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaFileRequest
 */
var BusinessProcessCaseServeDiagramAction = /** @class */ (function (_super) {
    tslib_1.__extends(BusinessProcessCaseServeDiagramAction, _super);
    function BusinessProcessCaseServeDiagramAction(data) {
        return _super.call(this, data) || this;
    }
    BusinessProcessCaseServeDiagramAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'businessprocessnotification_businessprocesscase' },
            action: { type: 'c', default: 'serveDiagram' },
            objectType: { type: 'es', subTypeConstructor: KalturaEventNotificationEventObjectType_1.KalturaEventNotificationEventObjectType, subType: 'KalturaEventNotificationEventObjectType' },
            objectId: { type: 's' },
            businessProcessStartNotificationTemplateId: { type: 'n' }
        });
        return result;
    };
    return BusinessProcessCaseServeDiagramAction;
}(kaltura_file_request_1.KalturaFileRequest));
exports.BusinessProcessCaseServeDiagramAction = BusinessProcessCaseServeDiagramAction;
