// kaltura-node-typescript-client/api/types/CaptionAssetItemSearchAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaptionAssetItemSearchAction = void 0;
var tslib_1 = require("tslib");
var KalturaCaptionAssetItemListResponse_1 = require("./KalturaCaptionAssetItemListResponse");
var KalturaBaseEntryFilter_1 = require("./KalturaBaseEntryFilter");
var KalturaCaptionAssetItemFilter_1 = require("./KalturaCaptionAssetItemFilter");
var KalturaFilterPager_1 = require("./KalturaFilterPager");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'captionAssetItem' action 'search'.
 *
 * Usage: Search caption asset items by filter, pager and free text
 *
 * Server response type:         KalturaCaptionAssetItemListResponse
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var CaptionAssetItemSearchAction = /** @class */ (function (_super) {
    tslib_1.__extends(CaptionAssetItemSearchAction, _super);
    function CaptionAssetItemSearchAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaCaptionAssetItemListResponse', responseConstructor: KalturaCaptionAssetItemListResponse_1.KalturaCaptionAssetItemListResponse }) || this;
    }
    CaptionAssetItemSearchAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'captionsearch_captionassetitem' },
            action: { type: 'c', default: 'search' },
            entryFilter: { type: 'o', subTypeConstructor: KalturaBaseEntryFilter_1.KalturaBaseEntryFilter, subType: 'KalturaBaseEntryFilter' },
            captionAssetItemFilter: { type: 'o', subTypeConstructor: KalturaCaptionAssetItemFilter_1.KalturaCaptionAssetItemFilter, subType: 'KalturaCaptionAssetItemFilter' },
            captionAssetItemPager: { type: 'o', subTypeConstructor: KalturaFilterPager_1.KalturaFilterPager, subType: 'KalturaFilterPager' }
        });
        return result;
    };
    return CaptionAssetItemSearchAction;
}(kaltura_request_1.KalturaRequest));
exports.CaptionAssetItemSearchAction = CaptionAssetItemSearchAction;
