// kaltura-node-typescript-client/api/types/CaptionAssetServeWebVTTAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaptionAssetServeWebVTTAction = void 0;
var tslib_1 = require("tslib");
var kaltura_file_request_1 = require("../kaltura-file-request");
/**
 * Build request payload for service 'captionAsset' action 'serveWebVTT'.
 *
 * Usage: Serves caption by its id converting it to segmented WebVTT
 *
 * Server response type:         { url: string }
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaFileRequest
 */
var CaptionAssetServeWebVTTAction = /** @class */ (function (_super) {
    tslib_1.__extends(CaptionAssetServeWebVTTAction, _super);
    function CaptionAssetServeWebVTTAction(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.segmentDuration === 'undefined')
            _this.segmentDuration = 30;
        if (typeof _this.localTimestamp === 'undefined')
            _this.localTimestamp = 10000;
        return _this;
    }
    CaptionAssetServeWebVTTAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'caption_captionasset' },
            action: { type: 'c', default: 'serveWebVTT' },
            captionAssetId: { type: 's' },
            segmentDuration: { type: 'n' },
            segmentIndex: { type: 'n' },
            localTimestamp: { type: 'n' }
        });
        return result;
    };
    return CaptionAssetServeWebVTTAction;
}(kaltura_file_request_1.KalturaFileRequest));
exports.CaptionAssetServeWebVTTAction = CaptionAssetServeWebVTTAction;
