// kaltura-node-typescript-client/api/types/CaptionAssetItemSearchEntriesAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaptionAssetItemSearchEntriesAction = void 0;
var tslib_1 = require("tslib");
var KalturaBaseEntryListResponse_1 = require("./KalturaBaseEntryListResponse");
var KalturaBaseEntryFilter_1 = require("./KalturaBaseEntryFilter");
var KalturaCaptionAssetItemFilter_1 = require("./KalturaCaptionAssetItemFilter");
var KalturaFilterPager_1 = require("./KalturaFilterPager");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'captionAssetItem' action 'searchEntries'.
 *
 * Usage: Search caption asset items by filter, pager and free text
 *
 * Server response type:         KalturaBaseEntryListResponse
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var CaptionAssetItemSearchEntriesAction = /** @class */ (function (_super) {
    tslib_1.__extends(CaptionAssetItemSearchEntriesAction, _super);
    function CaptionAssetItemSearchEntriesAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaBaseEntryListResponse', responseConstructor: KalturaBaseEntryListResponse_1.KalturaBaseEntryListResponse }) || this;
    }
    CaptionAssetItemSearchEntriesAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'captionsearch_captionassetitem' },
            action: { type: 'c', default: 'searchEntries' },
            entryFilter: { type: 'o', subTypeConstructor: KalturaBaseEntryFilter_1.KalturaBaseEntryFilter, subType: 'KalturaBaseEntryFilter' },
            captionAssetItemFilter: { type: 'o', subTypeConstructor: KalturaCaptionAssetItemFilter_1.KalturaCaptionAssetItemFilter, subType: 'KalturaCaptionAssetItemFilter' },
            captionAssetItemPager: { type: 'o', subTypeConstructor: KalturaFilterPager_1.KalturaFilterPager, subType: 'KalturaFilterPager' }
        });
        return result;
    };
    return CaptionAssetItemSearchEntriesAction;
}(kaltura_request_1.KalturaRequest));
exports.CaptionAssetItemSearchEntriesAction = CaptionAssetItemSearchEntriesAction;
