// kaltura-node-typescript-client/api/types/CaptionAssetItemListAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CaptionAssetItemListAction = void 0;
var tslib_1 = require("tslib");
var KalturaCaptionAssetItemListResponse_1 = require("./KalturaCaptionAssetItemListResponse");
var KalturaCaptionAssetItemFilter_1 = require("./KalturaCaptionAssetItemFilter");
var KalturaFilterPager_1 = require("./KalturaFilterPager");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'captionAssetItem' action 'list'.
 *
 * Usage: List caption asset items by filter and pager
 *
 * Server response type:         KalturaCaptionAssetItemListResponse
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var CaptionAssetItemListAction = /** @class */ (function (_super) {
    tslib_1.__extends(CaptionAssetItemListAction, _super);
    function CaptionAssetItemListAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaCaptionAssetItemListResponse', responseConstructor: KalturaCaptionAssetItemListResponse_1.KalturaCaptionAssetItemListResponse }) || this;
    }
    CaptionAssetItemListAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'captionsearch_captionassetitem' },
            action: { type: 'c', default: 'list' },
            captionAssetId: { type: 's' },
            captionAssetItemFilter: { type: 'o', subTypeConstructor: KalturaCaptionAssetItemFilter_1.KalturaCaptionAssetItemFilter, subType: 'KalturaCaptionAssetItemFilter' },
            captionAssetItemPager: { type: 'o', subTypeConstructor: KalturaFilterPager_1.KalturaFilterPager, subType: 'KalturaFilterPager' }
        });
        return result;
    };
    return CaptionAssetItemListAction;
}(kaltura_request_1.KalturaRequest));
exports.CaptionAssetItemListAction = CaptionAssetItemListAction;
