// kaltura-node-typescript-client/api/types/CategoryEntryIndexAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CategoryEntryIndexAction = void 0;
var tslib_1 = require("tslib");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'categoryEntry' action 'index'.
 *
 * Usage: Index CategoryEntry by Id
 *
 * Server response type:         number
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var CategoryEntryIndexAction = /** @class */ (function (_super) {
    tslib_1.__extends(CategoryEntryIndexAction, _super);
    function CategoryEntryIndexAction(data) {
        var _this = _super.call(this, data, { responseType: 'n', responseSubType: '', responseConstructor: null }) || this;
        if (typeof _this.shouldUpdate === 'undefined')
            _this.shouldUpdate = true;
        return _this;
    }
    CategoryEntryIndexAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'categoryentry' },
            action: { type: 'c', default: 'index' },
            entryId: { type: 's' },
            categoryId: { type: 'n' },
            shouldUpdate: { type: 'b' }
        });
        return result;
    };
    return CategoryEntryIndexAction;
}(kaltura_request_1.KalturaRequest));
exports.CategoryEntryIndexAction = CategoryEntryIndexAction;
