// kaltura-node-typescript-client/api/types/CategoryAddFromBulkUploadAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CategoryAddFromBulkUploadAction = void 0;
var tslib_1 = require("tslib");
var KalturaBulkUpload_1 = require("./KalturaBulkUpload");
var KalturaBulkUploadJobData_1 = require("./KalturaBulkUploadJobData");
var KalturaBulkUploadCategoryData_1 = require("./KalturaBulkUploadCategoryData");
var kaltura_upload_request_1 = require("../kaltura-upload-request");
/**
 * Build request payload for service 'category' action 'addFromBulkUpload'.
 *
 *
 *
 * Server response type:         KalturaBulkUpload
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaUploadRequest
 */
var CategoryAddFromBulkUploadAction = /** @class */ (function (_super) {
    tslib_1.__extends(CategoryAddFromBulkUploadAction, _super);
    function CategoryAddFromBulkUploadAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaBulkUpload', responseConstructor: KalturaBulkUpload_1.KalturaBulkUpload }) || this;
    }
    CategoryAddFromBulkUploadAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'category' },
            action: { type: 'c', default: 'addFromBulkUpload' },
            fileData: { type: 'f' },
            bulkUploadData: { type: 'o', subTypeConstructor: KalturaBulkUploadJobData_1.KalturaBulkUploadJobData, subType: 'KalturaBulkUploadJobData' },
            bulkUploadCategoryData: { type: 'o', subTypeConstructor: KalturaBulkUploadCategoryData_1.KalturaBulkUploadCategoryData, subType: 'KalturaBulkUploadCategoryData' }
        });
        return result;
    };
    return CategoryAddFromBulkUploadAction;
}(kaltura_upload_request_1.KalturaUploadRequest));
exports.CategoryAddFromBulkUploadAction = CategoryAddFromBulkUploadAction;
