// kaltura-node-typescript-client/api/types/CategoryExportToCsvAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CategoryExportToCsvAction = void 0;
var tslib_1 = require("tslib");
var KalturaCategoryFilter_1 = require("./KalturaCategoryFilter");
var KalturaCsvAdditionalFieldInfo_1 = require("./KalturaCsvAdditionalFieldInfo");
var KalturaKeyValue_1 = require("./KalturaKeyValue");
var KalturaExportToCsvOptions_1 = require("./KalturaExportToCsvOptions");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'category' action 'exportToCsv'.
 *
 * Usage: Creates a batch job that sends an email with a link to download a CSV containing a list of categories
 *
 * Server response type:         string
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var CategoryExportToCsvAction = /** @class */ (function (_super) {
    tslib_1.__extends(CategoryExportToCsvAction, _super);
    function CategoryExportToCsvAction(data) {
        var _this = _super.call(this, data, { responseType: 's', responseSubType: '', responseConstructor: null }) || this;
        if (typeof _this.additionalFields === 'undefined')
            _this.additionalFields = [];
        if (typeof _this.mappedFields === 'undefined')
            _this.mappedFields = [];
        return _this;
    }
    CategoryExportToCsvAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'category' },
            action: { type: 'c', default: 'exportToCsv' },
            filter: { type: 'o', subTypeConstructor: KalturaCategoryFilter_1.KalturaCategoryFilter, subType: 'KalturaCategoryFilter' },
            metadataProfileId: { type: 'n' },
            additionalFields: { type: 'a', subTypeConstructor: KalturaCsvAdditionalFieldInfo_1.KalturaCsvAdditionalFieldInfo, subType: 'KalturaCsvAdditionalFieldInfo' },
            mappedFields: { type: 'a', subTypeConstructor: KalturaKeyValue_1.KalturaKeyValue, subType: 'KalturaKeyValue' },
            options: { type: 'o', subTypeConstructor: KalturaExportToCsvOptions_1.KalturaExportToCsvOptions, subType: 'KalturaExportToCsvOptions' }
        });
        return result;
    };
    return CategoryExportToCsvAction;
}(kaltura_request_1.KalturaRequest));
exports.CategoryExportToCsvAction = CategoryExportToCsvAction;
