// kaltura-node-typescript-client/api/types/CategoryEntryUpdateStatusFromBulkAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CategoryEntryUpdateStatusFromBulkAction = void 0;
var tslib_1 = require("tslib");
var KalturaBulkUpload_1 = require("./KalturaBulkUpload");
var KalturaBulkUploadJobData_1 = require("./KalturaBulkUploadJobData");
var KalturaBulkUploadCategoryEntryData_1 = require("./KalturaBulkUploadCategoryEntryData");
var kaltura_upload_request_1 = require("../kaltura-upload-request");
/**
 * Build request payload for service 'categoryEntry' action 'updateStatusFromBulk'.
 *
 *
 *
 * Server response type:         KalturaBulkUpload
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaUploadRequest
 */
var CategoryEntryUpdateStatusFromBulkAction = /** @class */ (function (_super) {
    tslib_1.__extends(CategoryEntryUpdateStatusFromBulkAction, _super);
    function CategoryEntryUpdateStatusFromBulkAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaBulkUpload', responseConstructor: KalturaBulkUpload_1.KalturaBulkUpload }) || this;
    }
    CategoryEntryUpdateStatusFromBulkAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'categoryentry' },
            action: { type: 'c', default: 'updateStatusFromBulk' },
            fileData: { type: 'f' },
            bulkUploadData: { type: 'o', subTypeConstructor: KalturaBulkUploadJobData_1.KalturaBulkUploadJobData, subType: 'KalturaBulkUploadJobData' },
            bulkUploadCategoryEntryData: { type: 'o', subTypeConstructor: KalturaBulkUploadCategoryEntryData_1.KalturaBulkUploadCategoryEntryData, subType: 'KalturaBulkUploadCategoryEntryData' }
        });
        return result;
    };
    return CategoryEntryUpdateStatusFromBulkAction;
}(kaltura_upload_request_1.KalturaUploadRequest));
exports.CategoryEntryUpdateStatusFromBulkAction = CategoryEntryUpdateStatusFromBulkAction;
