// kaltura-node-typescript-client/api/types/CategoryUserUpdateAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CategoryUserUpdateAction = void 0;
var tslib_1 = require("tslib");
var KalturaCategoryUser_1 = require("./KalturaCategoryUser");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'categoryUser' action 'update'.
 *
 * Usage: Update CategoryUser by id
 *
 * Server response type:         KalturaCategoryUser
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var CategoryUserUpdateAction = /** @class */ (function (_super) {
    tslib_1.__extends(CategoryUserUpdateAction, _super);
    function CategoryUserUpdateAction(data) {
        var _this = _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaCategoryUser', responseConstructor: KalturaCategoryUser_1.KalturaCategoryUser }) || this;
        if (typeof _this.override === 'undefined')
            _this.override = false;
        return _this;
    }
    CategoryUserUpdateAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'categoryuser' },
            action: { type: 'c', default: 'update' },
            categoryId: { type: 'n' },
            userId: { type: 's' },
            categoryUser: { type: 'o', subTypeConstructor: KalturaCategoryUser_1.KalturaCategoryUser, subType: 'KalturaCategoryUser' },
            override: { type: 'b' }
        });
        return result;
    };
    return CategoryUserUpdateAction;
}(kaltura_request_1.KalturaRequest));
exports.CategoryUserUpdateAction = CategoryUserUpdateAction;
