// kaltura-node-typescript-client/api/types/CategoryCloneAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CategoryCloneAction = void 0;
var tslib_1 = require("tslib");
var KalturaCategory_1 = require("./KalturaCategory");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'category' action 'clone'.
 *
 * Usage: Clone Category
 *
 * Server response type:         KalturaCategory
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var CategoryCloneAction = /** @class */ (function (_super) {
    tslib_1.__extends(CategoryCloneAction, _super);
    function CategoryCloneAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaCategory', responseConstructor: KalturaCategory_1.KalturaCategory }) || this;
    }
    CategoryCloneAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'category' },
            action: { type: 'c', default: 'clone' },
            categoryId: { type: 'n' },
            fromPartnerId: { type: 'n' },
            parentCategoryId: { type: 'n' }
        });
        return result;
    };
    return CategoryCloneAction;
}(kaltura_request_1.KalturaRequest));
exports.CategoryCloneAction = CategoryCloneAction;
