// kaltura-node-typescript-client/api/types/CategoryUserAddFromBulkUploadAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CategoryUserAddFromBulkUploadAction = void 0;
var tslib_1 = require("tslib");
var KalturaBulkUpload_1 = require("./KalturaBulkUpload");
var KalturaBulkUploadJobData_1 = require("./KalturaBulkUploadJobData");
var KalturaBulkUploadCategoryUserData_1 = require("./KalturaBulkUploadCategoryUserData");
var kaltura_upload_request_1 = require("../kaltura-upload-request");
/**
 * Build request payload for service 'categoryUser' action 'addFromBulkUpload'.
 *
 *
 *
 * Server response type:         KalturaBulkUpload
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaUploadRequest
 */
var CategoryUserAddFromBulkUploadAction = /** @class */ (function (_super) {
    tslib_1.__extends(CategoryUserAddFromBulkUploadAction, _super);
    function CategoryUserAddFromBulkUploadAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaBulkUpload', responseConstructor: KalturaBulkUpload_1.KalturaBulkUpload }) || this;
    }
    CategoryUserAddFromBulkUploadAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'categoryuser' },
            action: { type: 'c', default: 'addFromBulkUpload' },
            fileData: { type: 'f' },
            bulkUploadData: { type: 'o', subTypeConstructor: KalturaBulkUploadJobData_1.KalturaBulkUploadJobData, subType: 'KalturaBulkUploadJobData' },
            bulkUploadCategoryUserData: { type: 'o', subTypeConstructor: KalturaBulkUploadCategoryUserData_1.KalturaBulkUploadCategoryUserData, subType: 'KalturaBulkUploadCategoryUserData' }
        });
        return result;
    };
    return CategoryUserAddFromBulkUploadAction;
}(kaltura_upload_request_1.KalturaUploadRequest));
exports.CategoryUserAddFromBulkUploadAction = CategoryUserAddFromBulkUploadAction;
