// kaltura-node-typescript-client/api/types/ConversionProfileAssetParamsUpdateAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConversionProfileAssetParamsUpdateAction = void 0;
var tslib_1 = require("tslib");
var KalturaConversionProfileAssetParams_1 = require("./KalturaConversionProfileAssetParams");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'conversionProfileAssetParams' action 'update'.
 *
 * Usage: Update asset parmas of conversion profile by ID
 *
 * Server response type:         KalturaConversionProfileAssetParams
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var ConversionProfileAssetParamsUpdateAction = /** @class */ (function (_super) {
    tslib_1.__extends(ConversionProfileAssetParamsUpdateAction, _super);
    function ConversionProfileAssetParamsUpdateAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaConversionProfileAssetParams', responseConstructor: KalturaConversionProfileAssetParams_1.KalturaConversionProfileAssetParams }) || this;
    }
    ConversionProfileAssetParamsUpdateAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'conversionprofileassetparams' },
            action: { type: 'c', default: 'update' },
            conversionProfileId: { type: 'n' },
            assetParamsId: { type: 'n' },
            conversionProfileAssetParams: { type: 'o', subTypeConstructor: KalturaConversionProfileAssetParams_1.KalturaConversionProfileAssetParams, subType: 'KalturaConversionProfileAssetParams' }
        });
        return result;
    };
    return ConversionProfileAssetParamsUpdateAction;
}(kaltura_request_1.KalturaRequest));
exports.ConversionProfileAssetParamsUpdateAction = ConversionProfileAssetParamsUpdateAction;
