// kaltura-node-typescript-client/api/types/CuePointUpdateCuePointsTimesAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.CuePointUpdateCuePointsTimesAction = void 0;
var tslib_1 = require("tslib");
var KalturaCuePoint_1 = require("./KalturaCuePoint");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'cuePoint' action 'updateCuePointsTimes'.
 *
 *
 *
 * Server response type:         KalturaCuePoint
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var CuePointUpdateCuePointsTimesAction = /** @class */ (function (_super) {
    tslib_1.__extends(CuePointUpdateCuePointsTimesAction, _super);
    function CuePointUpdateCuePointsTimesAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaCuePoint', responseConstructor: KalturaCuePoint_1.KalturaCuePoint }) || this;
    }
    CuePointUpdateCuePointsTimesAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'cuepoint_cuepoint' },
            action: { type: 'c', default: 'updateCuePointsTimes' },
            id: { type: 's' },
            startTime: { type: 'n' },
            endTime: { type: 'n' }
        });
        return result;
    };
    return CuePointUpdateCuePointsTimesAction;
}(kaltura_request_1.KalturaRequest));
exports.CuePointUpdateCuePointsTimesAction = CuePointUpdateCuePointsTimesAction;
