// kaltura-node-typescript-client/api/types/DataServeAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DataServeAction = void 0;
var tslib_1 = require("tslib");
var kaltura_file_request_1 = require("../kaltura-file-request");
/**
 * Build request payload for service 'data' action 'serve'.
 *
 * Usage: return the file from dataContent field
 *
 * Server response type:         { url: string }
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaFileRequest
 */
var DataServeAction = /** @class */ (function (_super) {
    tslib_1.__extends(DataServeAction, _super);
    function DataServeAction(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.version === 'undefined')
            _this.version = -1;
        if (typeof _this.forceProxy === 'undefined')
            _this.forceProxy = false;
        return _this;
    }
    DataServeAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'data' },
            action: { type: 'c', default: 'serve' },
            entryId: { type: 's' },
            version: { type: 'n' },
            forceProxy: { type: 'b' }
        });
        return result;
    };
    return DataServeAction;
}(kaltura_file_request_1.KalturaFileRequest));
exports.DataServeAction = DataServeAction;
