// kaltura-node-typescript-client/api/types/DocumentsUpdateContentAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentsUpdateContentAction = void 0;
var tslib_1 = require("tslib");
var KalturaDocumentEntry_1 = require("./KalturaDocumentEntry");
var KalturaResource_1 = require("./KalturaResource");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'documents' action 'updateContent'.
 *
 * Usage: Replace content associated with the given document entry
 *
 * Server response type:         KalturaDocumentEntry
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var DocumentsUpdateContentAction = /** @class */ (function (_super) {
    tslib_1.__extends(DocumentsUpdateContentAction, _super);
    function DocumentsUpdateContentAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaDocumentEntry', responseConstructor: KalturaDocumentEntry_1.KalturaDocumentEntry }) || this;
    }
    DocumentsUpdateContentAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'document_documents' },
            action: { type: 'c', default: 'updateContent' },
            entryId: { type: 's' },
            resource: { type: 'o', subTypeConstructor: KalturaResource_1.KalturaResource, subType: 'KalturaResource' },
            conversionProfileId: { type: 'n' }
        });
        return result;
    };
    return DocumentsUpdateContentAction;
}(kaltura_request_1.KalturaRequest));
exports.DocumentsUpdateContentAction = DocumentsUpdateContentAction;
