// kaltura-node-typescript-client/api/types/DocumentsAddFromEntryAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentsAddFromEntryAction = void 0;
var tslib_1 = require("tslib");
var KalturaDocumentEntry_1 = require("./KalturaDocumentEntry");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'documents' action 'addFromEntry'.
 *
 * Usage: Copy entry into new entry
 *
 * Server response type:         KalturaDocumentEntry
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var DocumentsAddFromEntryAction = /** @class */ (function (_super) {
    tslib_1.__extends(DocumentsAddFromEntryAction, _super);
    function DocumentsAddFromEntryAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaDocumentEntry', responseConstructor: KalturaDocumentEntry_1.KalturaDocumentEntry }) || this;
    }
    DocumentsAddFromEntryAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'document_documents' },
            action: { type: 'c', default: 'addFromEntry' },
            sourceEntryId: { type: 's' },
            documentEntry: { type: 'o', subTypeConstructor: KalturaDocumentEntry_1.KalturaDocumentEntry, subType: 'KalturaDocumentEntry' },
            sourceFlavorParamsId: { type: 'n' }
        });
        return result;
    };
    return DocumentsAddFromEntryAction;
}(kaltura_request_1.KalturaRequest));
exports.DocumentsAddFromEntryAction = DocumentsAddFromEntryAction;
