// kaltura-node-typescript-client/api/types/DocumentsServeAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentsServeAction = void 0;
var tslib_1 = require("tslib");
var kaltura_file_request_1 = require("../kaltura-file-request");
/**
 * Build request payload for service 'documents' action 'serve'.
 *
 * Usage: Serves the file content
 *
 * Server response type:         { url: string }
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaFileRequest
 */
var DocumentsServeAction = /** @class */ (function (_super) {
    tslib_1.__extends(DocumentsServeAction, _super);
    function DocumentsServeAction(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.forceProxy === 'undefined')
            _this.forceProxy = false;
        return _this;
    }
    DocumentsServeAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'document_documents' },
            action: { type: 'c', default: 'serve' },
            entryId: { type: 's' },
            flavorAssetId: { type: 's' },
            forceProxy: { type: 'b' }
        });
        return result;
    };
    return DocumentsServeAction;
}(kaltura_file_request_1.KalturaFileRequest));
exports.DocumentsServeAction = DocumentsServeAction;
