// kaltura-node-typescript-client/api/types/DocumentsServeByFlavorParamsIdAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DocumentsServeByFlavorParamsIdAction = void 0;
var tslib_1 = require("tslib");
var kaltura_file_request_1 = require("../kaltura-file-request");
/**
 * Build request payload for service 'documents' action 'serveByFlavorParamsId'.
 *
 * Usage: Serves the file content
 *
 * Server response type:         { url: string }
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaFileRequest
 */
var DocumentsServeByFlavorParamsIdAction = /** @class */ (function (_super) {
    tslib_1.__extends(DocumentsServeByFlavorParamsIdAction, _super);
    function DocumentsServeByFlavorParamsIdAction(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.forceProxy === 'undefined')
            _this.forceProxy = false;
        return _this;
    }
    DocumentsServeByFlavorParamsIdAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'document_documents' },
            action: { type: 'c', default: 'serveByFlavorParamsId' },
            entryId: { type: 's' },
            flavorParamsId: { type: 's' },
            forceProxy: { type: 'b' }
        });
        return result;
    };
    return DocumentsServeByFlavorParamsIdAction;
}(kaltura_file_request_1.KalturaFileRequest));
exports.DocumentsServeByFlavorParamsIdAction = DocumentsServeByFlavorParamsIdAction;
