// kaltura-node-typescript-client/api/types/DrmLicenseAccessGetAccessAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DrmLicenseAccessGetAccessAction = void 0;
var tslib_1 = require("tslib");
var KalturaDrmLicenseAccessDetails_1 = require("./KalturaDrmLicenseAccessDetails");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'drmLicenseAccess' action 'getAccess'.
 *
 * Usage: getAccessAction
 * input: flavor ids, drmProvider
 * Get Access Action
 *
 * Server response type:         KalturaDrmLicenseAccessDetails
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var DrmLicenseAccessGetAccessAction = /** @class */ (function (_super) {
    tslib_1.__extends(DrmLicenseAccessGetAccessAction, _super);
    function DrmLicenseAccessGetAccessAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaDrmLicenseAccessDetails', responseConstructor: KalturaDrmLicenseAccessDetails_1.KalturaDrmLicenseAccessDetails }) || this;
    }
    DrmLicenseAccessGetAccessAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'drm_drmlicenseaccess' },
            action: { type: 'c', default: 'getAccess' },
            entryId: { type: 's' },
            flavorIds: { type: 's' },
            referrer: { type: 's' }
        });
        return result;
    };
    return DrmLicenseAccessGetAccessAction;
}(kaltura_request_1.KalturaRequest));
exports.DrmLicenseAccessGetAccessAction = DrmLicenseAccessGetAccessAction;
