// kaltura-node-typescript-client/api/types/DropFolderFreeExclusiveDropFolderAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DropFolderFreeExclusiveDropFolderAction = void 0;
var tslib_1 = require("tslib");
var KalturaDropFolder_1 = require("./KalturaDropFolder");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'dropFolder' action 'freeExclusiveDropFolder'.
 *
 * Usage: freeExclusive KalturaDropFolder object
 *
 * Server response type:         KalturaDropFolder
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var DropFolderFreeExclusiveDropFolderAction = /** @class */ (function (_super) {
    tslib_1.__extends(DropFolderFreeExclusiveDropFolderAction, _super);
    function DropFolderFreeExclusiveDropFolderAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaDropFolder', responseConstructor: KalturaDropFolder_1.KalturaDropFolder }) || this;
    }
    DropFolderFreeExclusiveDropFolderAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'dropfolder_dropfolder' },
            action: { type: 'c', default: 'freeExclusiveDropFolder' },
            dropFolderId: { type: 'n' },
            errorCode: { type: 's' },
            errorDescription: { type: 's' }
        });
        return result;
    };
    return DropFolderFreeExclusiveDropFolderAction;
}(kaltura_request_1.KalturaRequest));
exports.DropFolderFreeExclusiveDropFolderAction = DropFolderFreeExclusiveDropFolderAction;
