// kaltura-node-typescript-client/api/types/EmailIngestionProfileAddMediaEntryAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailIngestionProfileAddMediaEntryAction = void 0;
var tslib_1 = require("tslib");
var KalturaMediaEntry_1 = require("./KalturaMediaEntry");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'EmailIngestionProfile' action 'addMediaEntry'.
 *
 * Usage: add KalturaMediaEntry from email ingestion
 *
 * Server response type:         KalturaMediaEntry
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var EmailIngestionProfileAddMediaEntryAction = /** @class */ (function (_super) {
    tslib_1.__extends(EmailIngestionProfileAddMediaEntryAction, _super);
    function EmailIngestionProfileAddMediaEntryAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaMediaEntry', responseConstructor: KalturaMediaEntry_1.KalturaMediaEntry }) || this;
    }
    EmailIngestionProfileAddMediaEntryAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'emailingestionprofile' },
            action: { type: 'c', default: 'addMediaEntry' },
            mediaEntry: { type: 'o', subTypeConstructor: KalturaMediaEntry_1.KalturaMediaEntry, subType: 'KalturaMediaEntry' },
            uploadTokenId: { type: 's' },
            emailProfId: { type: 'n' },
            fromAddress: { type: 's' },
            emailMsgId: { type: 's' }
        });
        return result;
    };
    return EmailIngestionProfileAddMediaEntryAction;
}(kaltura_request_1.KalturaRequest));
exports.EmailIngestionProfileAddMediaEntryAction = EmailIngestionProfileAddMediaEntryAction;
