// kaltura-node-typescript-client/api/types/EntryVendorTaskServeAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntryVendorTaskServeAction = void 0;
var tslib_1 = require("tslib");
var kaltura_file_request_1 = require("../kaltura-file-request");
/**
 * Build request payload for service 'entryVendorTask' action 'serve'.
 *
 *
 *
 * Server response type:         { url: string }
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaFileRequest
 */
var EntryVendorTaskServeAction = /** @class */ (function (_super) {
    tslib_1.__extends(EntryVendorTaskServeAction, _super);
    function EntryVendorTaskServeAction(data) {
        return _super.call(this, data) || this;
    }
    EntryVendorTaskServeAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'reach_entryvendortask' },
            action: { type: 'c', default: 'serve' },
            vendorPartnerId: { type: 'n' },
            partnerId: { type: 'n' },
            status: { type: 'n' },
            dueDate: { type: 's' }
        });
        return result;
    };
    return EntryVendorTaskServeAction;
}(kaltura_file_request_1.KalturaFileRequest));
exports.EntryVendorTaskServeAction = EntryVendorTaskServeAction;
