// kaltura-node-typescript-client/api/types/EntryVendorTaskGetServeUrlAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EntryVendorTaskGetServeUrlAction = void 0;
var tslib_1 = require("tslib");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'entryVendorTask' action 'getServeUrl'.
 *
 *
 *
 * Server response type:         string
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var EntryVendorTaskGetServeUrlAction = /** @class */ (function (_super) {
    tslib_1.__extends(EntryVendorTaskGetServeUrlAction, _super);
    function EntryVendorTaskGetServeUrlAction(data) {
        return _super.call(this, data, { responseType: 's', responseSubType: '', responseConstructor: null }) || this;
    }
    EntryVendorTaskGetServeUrlAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'reach_entryvendortask' },
            action: { type: 'c', default: 'getServeUrl' },
            filterType: { type: 's' },
            filterInput: { type: 'n' },
            status: { type: 'n' },
            dueDate: { type: 's' }
        });
        return result;
    };
    return EntryVendorTaskGetServeUrlAction;
}(kaltura_request_1.KalturaRequest));
exports.EntryVendorTaskGetServeUrlAction = EntryVendorTaskGetServeUrlAction;
