// kaltura-node-typescript-client/api/types/EventNotificationTemplateSendCommandAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.EventNotificationTemplateSendCommandAction = void 0;
var tslib_1 = require("tslib");
var KalturaPushNotificationParams_1 = require("./KalturaPushNotificationParams");
var KalturaPushNotificationCommandType_1 = require("./KalturaPushNotificationCommandType");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'eventNotificationTemplate' action 'sendCommand'.
 *
 * Usage: Clear queue messages
 *
 * Server response type:         void
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var EventNotificationTemplateSendCommandAction = /** @class */ (function (_super) {
    tslib_1.__extends(EventNotificationTemplateSendCommandAction, _super);
    function EventNotificationTemplateSendCommandAction(data) {
        return _super.call(this, data, { responseType: 'v', responseSubType: '', responseConstructor: null }) || this;
    }
    EventNotificationTemplateSendCommandAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'eventnotification_eventnotificationtemplate' },
            action: { type: 'c', default: 'sendCommand' },
            notificationTemplateSystemName: { type: 's' },
            pushNotificationParams: { type: 'o', subTypeConstructor: KalturaPushNotificationParams_1.KalturaPushNotificationParams, subType: 'KalturaPushNotificationParams' },
            command: { type: 'es', subTypeConstructor: KalturaPushNotificationCommandType_1.KalturaPushNotificationCommandType, subType: 'KalturaPushNotificationCommandType' }
        });
        return result;
    };
    return EventNotificationTemplateSendCommandAction;
}(kaltura_request_1.KalturaRequest));
exports.EventNotificationTemplateSendCommandAction = EventNotificationTemplateSendCommandAction;
