// kaltura-node-typescript-client/api/types/FlavorAssetConvertAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlavorAssetConvertAction = void 0;
var tslib_1 = require("tslib");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'flavorAsset' action 'convert'.
 *
 * Usage: Add and convert new Flavor Asset for Entry with specific Flavor Params
 *
 * Server response type:         void
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var FlavorAssetConvertAction = /** @class */ (function (_super) {
    tslib_1.__extends(FlavorAssetConvertAction, _super);
    function FlavorAssetConvertAction(data) {
        return _super.call(this, data, { responseType: 'v', responseSubType: '', responseConstructor: null }) || this;
    }
    FlavorAssetConvertAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'flavorasset' },
            action: { type: 'c', default: 'convert' },
            entryId: { type: 's' },
            flavorParamsId: { type: 'n' },
            priority: { type: 'n' }
        });
        return result;
    };
    return FlavorAssetConvertAction;
}(kaltura_request_1.KalturaRequest));
exports.FlavorAssetConvertAction = FlavorAssetConvertAction;
