// kaltura-node-typescript-client/api/types/FlavorAssetGetUrlAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlavorAssetGetUrlAction = void 0;
var tslib_1 = require("tslib");
var KalturaFlavorAssetUrlOptions_1 = require("./KalturaFlavorAssetUrlOptions");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'flavorAsset' action 'getUrl'.
 *
 * Usage: Get download URL for the asset
 *
 * Server response type:         string
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var FlavorAssetGetUrlAction = /** @class */ (function (_super) {
    tslib_1.__extends(FlavorAssetGetUrlAction, _super);
    function FlavorAssetGetUrlAction(data) {
        var _this = _super.call(this, data, { responseType: 's', responseSubType: '', responseConstructor: null }) || this;
        if (typeof _this.forceProxy === 'undefined')
            _this.forceProxy = false;
        return _this;
    }
    FlavorAssetGetUrlAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'flavorasset' },
            action: { type: 'c', default: 'getUrl' },
            id: { type: 's' },
            storageId: { type: 'n' },
            forceProxy: { type: 'b' },
            options: { type: 'o', subTypeConstructor: KalturaFlavorAssetUrlOptions_1.KalturaFlavorAssetUrlOptions, subType: 'KalturaFlavorAssetUrlOptions' }
        });
        return result;
    };
    return FlavorAssetGetUrlAction;
}(kaltura_request_1.KalturaRequest));
exports.FlavorAssetGetUrlAction = FlavorAssetGetUrlAction;
