// kaltura-node-typescript-client/api/types/FlavorAssetServeAdStitchCmdAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.FlavorAssetServeAdStitchCmdAction = void 0;
var tslib_1 = require("tslib");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'flavorAsset' action 'serveAdStitchCmd'.
 *
 * Usage: serve cmd line to transcode the ad
 *
 * Server response type:         string
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var FlavorAssetServeAdStitchCmdAction = /** @class */ (function (_super) {
    tslib_1.__extends(FlavorAssetServeAdStitchCmdAction, _super);
    function FlavorAssetServeAdStitchCmdAction(data) {
        return _super.call(this, data, { responseType: 's', responseSubType: '', responseConstructor: null }) || this;
    }
    FlavorAssetServeAdStitchCmdAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'flavorasset' },
            action: { type: 'c', default: 'serveAdStitchCmd' },
            assetId: { type: 's' },
            ffprobeJson: { type: 's' },
            duration: { type: 's' }
        });
        return result;
    };
    return FlavorAssetServeAdStitchCmdAction;
}(kaltura_request_1.KalturaRequest));
exports.FlavorAssetServeAdStitchCmdAction = FlavorAssetServeAdStitchCmdAction;
