// kaltura-node-typescript-client/api/types/GroupCloneAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GroupCloneAction = void 0;
var tslib_1 = require("tslib");
var KalturaGroup_1 = require("./KalturaGroup");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'group' action 'clone'.
 *
 * Usage: clone the group (groupId), and set group id with the neeGroupName
 *
 * Server response type:         KalturaGroup
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var GroupCloneAction = /** @class */ (function (_super) {
    tslib_1.__extends(GroupCloneAction, _super);
    function GroupCloneAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaGroup', responseConstructor: KalturaGroup_1.KalturaGroup }) || this;
    }
    GroupCloneAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'group_group' },
            action: { type: 'c', default: 'clone' },
            originalGroupId: { type: 's' },
            newGroupId: { type: 's' },
            newGroupName: { type: 's' }
        });
        return result;
    };
    return GroupCloneAction;
}(kaltura_request_1.KalturaRequest));
exports.GroupCloneAction = GroupCloneAction;
