// kaltura-node-typescript-client/api/types/GroupUserSyncAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.GroupUserSyncAction = void 0;
var tslib_1 = require("tslib");
var KalturaBulkUpload_1 = require("./KalturaBulkUpload");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'groupUser' action 'sync'.
 *
 * Usage: sync by userId and groupIds
 *
 * Server response type:         KalturaBulkUpload
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var GroupUserSyncAction = /** @class */ (function (_super) {
    tslib_1.__extends(GroupUserSyncAction, _super);
    function GroupUserSyncAction(data) {
        var _this = _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaBulkUpload', responseConstructor: KalturaBulkUpload_1.KalturaBulkUpload }) || this;
        if (typeof _this.removeFromExistingGroups === 'undefined')
            _this.removeFromExistingGroups = true;
        if (typeof _this.createNewGroups === 'undefined')
            _this.createNewGroups = true;
        return _this;
    }
    GroupUserSyncAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'groupuser' },
            action: { type: 'c', default: 'sync' },
            userId: { type: 's' },
            groupIds: { type: 's' },
            removeFromExistingGroups: { type: 'b' },
            createNewGroups: { type: 'b' }
        });
        return result;
    };
    return GroupUserSyncAction;
}(kaltura_request_1.KalturaRequest));
exports.GroupUserSyncAction = GroupUserSyncAction;
