// kaltura-node-typescript-client/api/types/IntegrationDispatchAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.IntegrationDispatchAction = void 0;
var tslib_1 = require("tslib");
var KalturaIntegrationJobData_1 = require("./KalturaIntegrationJobData");
var KalturaBatchJobObjectType_1 = require("./KalturaBatchJobObjectType");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'integration' action 'dispatch'.
 *
 * Usage: Dispatch integration task
 *
 * Server response type:         number
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var IntegrationDispatchAction = /** @class */ (function (_super) {
    tslib_1.__extends(IntegrationDispatchAction, _super);
    function IntegrationDispatchAction(data) {
        return _super.call(this, data, { responseType: 'n', responseSubType: '', responseConstructor: null }) || this;
    }
    IntegrationDispatchAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'integration_integration' },
            action: { type: 'c', default: 'dispatch' },
            data: { type: 'o', subTypeConstructor: KalturaIntegrationJobData_1.KalturaIntegrationJobData, subType: 'KalturaIntegrationJobData' },
            objectType: { type: 'es', subTypeConstructor: KalturaBatchJobObjectType_1.KalturaBatchJobObjectType, subType: 'KalturaBatchJobObjectType' },
            objectId: { type: 's' }
        });
        return result;
    };
    return IntegrationDispatchAction;
}(kaltura_request_1.KalturaRequest));
exports.IntegrationDispatchAction = IntegrationDispatchAction;
