// kaltura-node-typescript-client/api/types/InteractivityUpdateAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.InteractivityUpdateAction = void 0;
var tslib_1 = require("tslib");
var KalturaInteractivity_1 = require("./KalturaInteractivity");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'interactivity' action 'update'.
 *
 * Usage: Update an existing interactivity object
 *
 * Server response type:         KalturaInteractivity
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var InteractivityUpdateAction = /** @class */ (function (_super) {
    tslib_1.__extends(InteractivityUpdateAction, _super);
    function InteractivityUpdateAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaInteractivity', responseConstructor: KalturaInteractivity_1.KalturaInteractivity }) || this;
    }
    InteractivityUpdateAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'interactivity_interactivity' },
            action: { type: 'c', default: 'update' },
            entryId: { type: 's' },
            version: { type: 'n' },
            kalturaInteractivity: { type: 'o', subTypeConstructor: KalturaInteractivity_1.KalturaInteractivity, subType: 'KalturaInteractivity' }
        });
        return result;
    };
    return InteractivityUpdateAction;
}(kaltura_request_1.KalturaRequest));
exports.InteractivityUpdateAction = InteractivityUpdateAction;
