// kaltura-node-typescript-client/api/types/KalturaFtpDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFtpDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDistributionProtocol_1 = require("./KalturaDistributionProtocol");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaFtpDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaFtpDistributionProfile, _super);
    function KalturaFtpDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaFtpDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaFtpDistributionProfile' },
            protocol: { type: 'en', subTypeConstructor: KalturaDistributionProtocol_1.KalturaDistributionProtocol, subType: 'KalturaDistributionProtocol' },
            host: { type: 's' },
            port: { type: 'n' },
            basePath: { type: 's' },
            username: { type: 's' },
            password: { type: 's' },
            passphrase: { type: 's' },
            sftpPublicKey: { type: 's' },
            sftpPrivateKey: { type: 's' },
            disableMetadata: { type: 'b' },
            metadataXslt: { type: 's' },
            metadataFilenameXslt: { type: 's' },
            flavorAssetFilenameXslt: { type: 's' },
            thumbnailAssetFilenameXslt: { type: 's' },
            assetFilenameXslt: { type: 's' },
            asperaPublicKey: { type: 's' },
            asperaPrivateKey: { type: 's' },
            sendMetadataAfterAssets: { type: 'b' }
        });
        return result;
    };
    return KalturaFtpDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaFtpDistributionProfile = KalturaFtpDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaFtpDistributionProfile', KalturaFtpDistributionProfile);
