// kaltura-node-typescript-client/api/types/KalturaEntryServerNode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEntryServerNode = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaEntryServerNodeStatus_1 = require("./KalturaEntryServerNodeStatus");
var KalturaEntryServerNodeType_1 = require("./KalturaEntryServerNodeType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaEntryServerNode = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaEntryServerNode, _super);
    function KalturaEntryServerNode(data) {
        return _super.call(this, data) || this;
    }
    KalturaEntryServerNode.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaEntryServerNode' },
            id: { type: 'n', readOnly: true },
            entryId: { type: 's', readOnly: true },
            serverNodeId: { type: 'n', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaEntryServerNodeStatus_1.KalturaEntryServerNodeStatus, subType: 'KalturaEntryServerNodeStatus' },
            serverType: { type: 'es', readOnly: true, subTypeConstructor: KalturaEntryServerNodeType_1.KalturaEntryServerNodeType, subType: 'KalturaEntryServerNodeType' }
        });
        return result;
    };
    return KalturaEntryServerNode;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaEntryServerNode = KalturaEntryServerNode;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaEntryServerNode', KalturaEntryServerNode);
