// kaltura-node-typescript-client/api/types/KalturaAssetParamsBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAssetParamsBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaAssetParamsBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAssetParamsBaseFilter, _super);
    function KalturaAssetParamsBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaAssetParamsBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAssetParamsBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            systemNameEqual: { type: 's' },
            systemNameIn: { type: 's' },
            isSystemDefaultEqual: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            tagsEqual: { type: 's' }
        });
        return result;
    };
    return KalturaAssetParamsBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaAssetParamsBaseFilter = KalturaAssetParamsBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAssetParamsBaseFilter', KalturaAssetParamsBaseFilter);
