// kaltura-node-typescript-client/api/types/KalturaCrossKalturaDistributionJobProviderData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCrossKalturaDistributionJobProviderData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionJobProviderData_1 = require("./KalturaConfigurableDistributionJobProviderData");
var KalturaCrossKalturaDistributionJobProviderData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCrossKalturaDistributionJobProviderData, _super);
    function KalturaCrossKalturaDistributionJobProviderData(data) {
        return _super.call(this, data) || this;
    }
    KalturaCrossKalturaDistributionJobProviderData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCrossKalturaDistributionJobProviderData' },
            distributedFlavorAssets: { type: 's' },
            distributedThumbAssets: { type: 's' },
            distributedMetadata: { type: 's' },
            distributedCaptionAssets: { type: 's' },
            distributedCuePoints: { type: 's' },
            distributedThumbCuePoints: { type: 's' },
            distributedTimedThumbAssets: { type: 's' }
        });
        return result;
    };
    return KalturaCrossKalturaDistributionJobProviderData;
}(KalturaConfigurableDistributionJobProviderData_1.KalturaConfigurableDistributionJobProviderData));
exports.KalturaCrossKalturaDistributionJobProviderData = KalturaCrossKalturaDistributionJobProviderData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCrossKalturaDistributionJobProviderData', KalturaCrossKalturaDistributionJobProviderData);
