// kaltura-node-typescript-client/api/types/KalturaExternalMediaEntryBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaExternalMediaEntryBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaExternalMediaSourceType_1 = require("./KalturaExternalMediaSourceType");
var KalturaMediaEntryFilter_1 = require("./KalturaMediaEntryFilter");
var KalturaExternalMediaEntryBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaExternalMediaEntryBaseFilter, _super);
    function KalturaExternalMediaEntryBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaExternalMediaEntryBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaExternalMediaEntryBaseFilter' },
            externalSourceTypeEqual: { type: 'es', subTypeConstructor: KalturaExternalMediaSourceType_1.KalturaExternalMediaSourceType, subType: 'KalturaExternalMediaSourceType' },
            externalSourceTypeIn: { type: 's' },
            assetParamsIdsMatchOr: { type: 's' },
            assetParamsIdsMatchAnd: { type: 's' }
        });
        return result;
    };
    return KalturaExternalMediaEntryBaseFilter;
}(KalturaMediaEntryFilter_1.KalturaMediaEntryFilter));
exports.KalturaExternalMediaEntryBaseFilter = KalturaExternalMediaEntryBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaExternalMediaEntryBaseFilter', KalturaExternalMediaEntryBaseFilter);
