// kaltura-node-typescript-client/api/types/KalturaLicenseType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLicenseType = void 0;
var KalturaLicenseType;
(function (KalturaLicenseType) {
    KalturaLicenseType[KalturaLicenseType["afferoGpl"] = 11] = "afferoGpl";
    KalturaLicenseType[KalturaLicenseType["apache"] = 14] = "apache";
    KalturaLicenseType[KalturaLicenseType["bsd"] = 13] = "bsd";
    KalturaLicenseType[KalturaLicenseType["copyrighted"] = 1] = "copyrighted";
    KalturaLicenseType[KalturaLicenseType["creativecommonsAttribution"] = 3] = "creativecommonsAttribution";
    KalturaLicenseType[KalturaLicenseType["creativecommonsAttributionNonCommercial"] = 6] = "creativecommonsAttributionNonCommercial";
    KalturaLicenseType[KalturaLicenseType["creativecommonsAttributionNonCommercialNoDerivatives"] = 8] = "creativecommonsAttributionNonCommercialNoDerivatives";
    KalturaLicenseType[KalturaLicenseType["creativecommonsAttributionNonCommercialShareAlike"] = 7] = "creativecommonsAttributionNonCommercialShareAlike";
    KalturaLicenseType[KalturaLicenseType["creativecommonsAttributionNoDerivatives"] = 5] = "creativecommonsAttributionNoDerivatives";
    KalturaLicenseType[KalturaLicenseType["creativecommonsAttributionShareAlike"] = 4] = "creativecommonsAttributionShareAlike";
    KalturaLicenseType[KalturaLicenseType["gfdl"] = 9] = "gfdl";
    KalturaLicenseType[KalturaLicenseType["gpl"] = 10] = "gpl";
    KalturaLicenseType[KalturaLicenseType["lgpl"] = 12] = "lgpl";
    KalturaLicenseType[KalturaLicenseType["mozilla"] = 15] = "mozilla";
    KalturaLicenseType[KalturaLicenseType["none"] = 0] = "none";
    KalturaLicenseType[KalturaLicenseType["publicDomain"] = 2] = "publicDomain";
    KalturaLicenseType[KalturaLicenseType["unknown"] = -1] = "unknown";
})(KalturaLicenseType = exports.KalturaLicenseType || (exports.KalturaLicenseType = {}));
