// kaltura-node-typescript-client/api/types/KalturaEventNotificationEventObjectType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEventNotificationEventObjectType = void 0;
var KalturaEventNotificationEventObjectType;
(function (KalturaEventNotificationEventObjectType) {
    KalturaEventNotificationEventObjectType["accesscontrol"] = "9";
    KalturaEventNotificationEventObjectType["adCuePoint"] = "adCuePointEventNotifications.AdCuePoint";
    KalturaEventNotificationEventObjectType["annotation"] = "annotationEventNotifications.Annotation";
    KalturaEventNotificationEventObjectType["asset"] = "3";
    KalturaEventNotificationEventObjectType["attachmentAsset"] = "attachmentAssetEventNotifications.AttachmentAsset";
    KalturaEventNotificationEventObjectType["batchjob"] = "10";
    KalturaEventNotificationEventObjectType["bulkuploadresult"] = "11";
    KalturaEventNotificationEventObjectType["captionAsset"] = "captionAssetEventNotifications.CaptionAsset";
    KalturaEventNotificationEventObjectType["category"] = "2";
    KalturaEventNotificationEventObjectType["categoryentry"] = "37";
    KalturaEventNotificationEventObjectType["categorykuser"] = "12";
    KalturaEventNotificationEventObjectType["codeCuePoint"] = "codeCuePointEventNotifications.CodeCuePoint";
    KalturaEventNotificationEventObjectType["conversionprofile2"] = "14";
    KalturaEventNotificationEventObjectType["cuePoint"] = "cuePointEventNotifications.CuePoint";
    KalturaEventNotificationEventObjectType["distributionProfile"] = "contentDistributionEventNotifications.DistributionProfile";
    KalturaEventNotificationEventObjectType["dropFolder"] = "dropFolderEventNotifications.DropFolder";
    KalturaEventNotificationEventObjectType["dropFolderFile"] = "dropFolderEventNotifications.DropFolderFile";
    KalturaEventNotificationEventObjectType["entry"] = "1";
    KalturaEventNotificationEventObjectType["entryDistribution"] = "contentDistributionEventNotifications.EntryDistribution";
    KalturaEventNotificationEventObjectType["entryServerNode"] = "40";
    KalturaEventNotificationEventObjectType["entryVendorTask"] = "42";
    KalturaEventNotificationEventObjectType["flavorasset"] = "4";
    KalturaEventNotificationEventObjectType["flavorparams"] = "15";
    KalturaEventNotificationEventObjectType["flavorparamsconversionprofile"] = "16";
    KalturaEventNotificationEventObjectType["flavorparamsoutput"] = "17";
    KalturaEventNotificationEventObjectType["genericsyndicationfeed"] = "18";
    KalturaEventNotificationEventObjectType["kuser"] = "8";
    KalturaEventNotificationEventObjectType["kusertouserrole"] = "19";
    KalturaEventNotificationEventObjectType["liveStream"] = "38";
    KalturaEventNotificationEventObjectType["metadata"] = "metadataEventNotifications.Metadata";
    KalturaEventNotificationEventObjectType["partner"] = "20";
    KalturaEventNotificationEventObjectType["permission"] = "21";
    KalturaEventNotificationEventObjectType["permissionitem"] = "22";
    KalturaEventNotificationEventObjectType["permissiontopermissionitem"] = "23";
    KalturaEventNotificationEventObjectType["reachProfile"] = "41";
    KalturaEventNotificationEventObjectType["scheduler"] = "24";
    KalturaEventNotificationEventObjectType["schedulerconfig"] = "25";
    KalturaEventNotificationEventObjectType["schedulerstatus"] = "26";
    KalturaEventNotificationEventObjectType["schedulerworker"] = "27";
    KalturaEventNotificationEventObjectType["serverNode"] = "39";
    KalturaEventNotificationEventObjectType["storageprofile"] = "28";
    KalturaEventNotificationEventObjectType["syndicationfeed"] = "29";
    KalturaEventNotificationEventObjectType["thumbasset"] = "5";
    KalturaEventNotificationEventObjectType["thumbparams"] = "31";
    KalturaEventNotificationEventObjectType["thumbparamsoutput"] = "32";
    KalturaEventNotificationEventObjectType["transcriptAsset"] = "transcriptAssetEventNotifications.TranscriptAsset";
    KalturaEventNotificationEventObjectType["uploadtoken"] = "33";
    KalturaEventNotificationEventObjectType["userlogindata"] = "34";
    KalturaEventNotificationEventObjectType["userrole"] = "35";
    KalturaEventNotificationEventObjectType["virtualEvent"] = "virtualEventEventNotifications.VirtualEvent";
    KalturaEventNotificationEventObjectType["widget"] = "36";
})(KalturaEventNotificationEventObjectType = exports.KalturaEventNotificationEventObjectType || (exports.KalturaEventNotificationEventObjectType = {}));
