// kaltura-node-typescript-client/api/types/KalturaImportJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaImportJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaString_1 = require("./KalturaString");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaImportJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaImportJobData, _super);
    function KalturaImportJobData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.urlHeaders === 'undefined')
            _this.urlHeaders = [];
        return _this;
    }
    KalturaImportJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaImportJobData' },
            srcFileUrl: { type: 's' },
            destFileLocalPath: { type: 's' },
            flavorAssetId: { type: 's' },
            fileSize: { type: 'n' },
            destFileSharedPath: { type: 's' },
            urlHeaders: { type: 'a', subTypeConstructor: KalturaString_1.KalturaString, subType: 'KalturaString' },
            shouldRedirect: { type: 'b' }
        });
        return result;
    };
    return KalturaImportJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaImportJobData = KalturaImportJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaImportJobData', KalturaImportJobData);
