// kaltura-node-typescript-client/api/types/KalturaLiveCaptionFeature.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveCaptionFeature = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaLiveFeature_1 = require("./KalturaLiveFeature");
var KalturaLiveCaptionFeature = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveCaptionFeature, _super);
    function KalturaLiveCaptionFeature(data) {
        return _super.call(this, data) || this;
    }
    KalturaLiveCaptionFeature.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveCaptionFeature' },
            mediaUrl: { type: 's' },
            mediaKey: { type: 's' },
            captionUrl: { type: 's' },
            captionToken: { type: 's' },
            inputDelay: { type: 'n' },
            captionFormat: { type: 's' },
            language: { type: 's' }
        });
        return result;
    };
    return KalturaLiveCaptionFeature;
}(KalturaLiveFeature_1.KalturaLiveFeature));
exports.KalturaLiveCaptionFeature = KalturaLiveCaptionFeature;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveCaptionFeature', KalturaLiveCaptionFeature);
