// kaltura-node-typescript-client/api/types/KalturaModerationFlag.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaModerationFlag = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaModerationObjectType_1 = require("./KalturaModerationObjectType");
var KalturaModerationFlagStatus_1 = require("./KalturaModerationFlagStatus");
var KalturaModerationFlagType_1 = require("./KalturaModerationFlagType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaModerationFlag = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaModerationFlag, _super);
    function KalturaModerationFlag(data) {
        return _super.call(this, data) || this;
    }
    KalturaModerationFlag.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaModerationFlag' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            userId: { type: 's', readOnly: true },
            moderationObjectType: { type: 'es', readOnly: true, subTypeConstructor: KalturaModerationObjectType_1.KalturaModerationObjectType, subType: 'KalturaModerationObjectType' },
            flaggedEntryId: { type: 's' },
            flaggedUserId: { type: 's' },
            status: { type: 'es', readOnly: true, subTypeConstructor: KalturaModerationFlagStatus_1.KalturaModerationFlagStatus, subType: 'KalturaModerationFlagStatus' },
            comments: { type: 's' },
            flagType: { type: 'en', subTypeConstructor: KalturaModerationFlagType_1.KalturaModerationFlagType, subType: 'KalturaModerationFlagType' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true }
        });
        return result;
    };
    return KalturaModerationFlag;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaModerationFlag = KalturaModerationFlag;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaModerationFlag', KalturaModerationFlag);
