// kaltura-node-typescript-client/api/types/KalturaPartnerBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPartnerBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaPartnerStatus_1 = require("./KalturaPartnerStatus");
var KalturaPartnerGroupType_1 = require("./KalturaPartnerGroupType");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaPartnerBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPartnerBaseFilter, _super);
    function KalturaPartnerBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaPartnerBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPartnerBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            idNotIn: { type: 's' },
            nameLike: { type: 's' },
            nameMultiLikeOr: { type: 's' },
            nameMultiLikeAnd: { type: 's' },
            nameEqual: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaPartnerStatus_1.KalturaPartnerStatus, subType: 'KalturaPartnerStatus' },
            statusIn: { type: 's' },
            partnerPackageEqual: { type: 'n' },
            partnerPackageGreaterThanOrEqual: { type: 'n' },
            partnerPackageLessThanOrEqual: { type: 'n' },
            partnerPackageIn: { type: 's' },
            partnerGroupTypeEqual: { type: 'en', subTypeConstructor: KalturaPartnerGroupType_1.KalturaPartnerGroupType, subType: 'KalturaPartnerGroupType' },
            partnerNameDescriptionWebsiteAdminNameAdminEmailLike: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            idGreaterThan: { type: 'n' },
            monitorUsageEqual: { type: 'n' }
        });
        return result;
    };
    return KalturaPartnerBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaPartnerBaseFilter = KalturaPartnerBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPartnerBaseFilter', KalturaPartnerBaseFilter);
