// kaltura-node-typescript-client/api/types/KalturaCaptionParams.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCaptionParams = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaLanguage_1 = require("./KalturaLanguage");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaCaptionType_1 = require("./KalturaCaptionType");
var KalturaAssetParams_1 = require("./KalturaAssetParams");
var KalturaCaptionParams = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCaptionParams, _super);
    function KalturaCaptionParams(data) {
        return _super.call(this, data) || this;
    }
    KalturaCaptionParams.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCaptionParams' },
            language: { type: 'es', subTypeConstructor: KalturaLanguage_1.KalturaLanguage, subType: 'KalturaLanguage' },
            isDefault: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            label: { type: 's' },
            format: { type: 'es', subTypeConstructor: KalturaCaptionType_1.KalturaCaptionType, subType: 'KalturaCaptionType' },
            sourceParamsId: { type: 'n' }
        });
        return result;
    };
    return KalturaCaptionParams;
}(KalturaAssetParams_1.KalturaAssetParams));
exports.KalturaCaptionParams = KalturaCaptionParams;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCaptionParams', KalturaCaptionParams);
