// kaltura-node-typescript-client/api/types/KalturaThumbCuePoint.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaThumbCuePoint = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaThumbCuePointSubType_1 = require("./KalturaThumbCuePointSubType");
var KalturaCuePoint_1 = require("./KalturaCuePoint");
var KalturaThumbCuePoint = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaThumbCuePoint, _super);
    function KalturaThumbCuePoint(data) {
        return _super.call(this, data) || this;
    }
    KalturaThumbCuePoint.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaThumbCuePoint' },
            assetId: { type: 's' },
            description: { type: 's' },
            title: { type: 's' },
            subType: { type: 'en', subTypeConstructor: KalturaThumbCuePointSubType_1.KalturaThumbCuePointSubType, subType: 'KalturaThumbCuePointSubType' }
        });
        return result;
    };
    return KalturaThumbCuePoint;
}(KalturaCuePoint_1.KalturaCuePoint));
exports.KalturaThumbCuePoint = KalturaThumbCuePoint;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaThumbCuePoint', KalturaThumbCuePoint);
