// kaltura-node-typescript-client/api/types/KalturaDropFolderFileErrorCode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDropFolderFileErrorCode = void 0;
var KalturaDropFolderFileErrorCode;
(function (KalturaDropFolderFileErrorCode) {
    KalturaDropFolderFileErrorCode["errorAddingBulkUpload"] = "dropFolderXmlBulkUpload.ERROR_ADDING_BULK_UPLOAD";
    KalturaDropFolderFileErrorCode["errorAddingContentProcessor"] = "10";
    KalturaDropFolderFileErrorCode["errorAddContentResource"] = "dropFolderXmlBulkUpload.ERROR_ADD_CONTENT_RESOURCE";
    KalturaDropFolderFileErrorCode["errorAddEntry"] = "2";
    KalturaDropFolderFileErrorCode["errorDeletingFile"] = "12";
    KalturaDropFolderFileErrorCode["errorDownloadingFile"] = "7";
    KalturaDropFolderFileErrorCode["errorInBulkUpload"] = "dropFolderXmlBulkUpload.ERROR_IN_BULK_UPLOAD";
    KalturaDropFolderFileErrorCode["errorInContentProcessor"] = "11";
    KalturaDropFolderFileErrorCode["errorReadingFile"] = "6";
    KalturaDropFolderFileErrorCode["errorUpdateEntry"] = "1";
    KalturaDropFolderFileErrorCode["errorUpdateFile"] = "8";
    KalturaDropFolderFileErrorCode["errorWritingTempFile"] = "dropFolderXmlBulkUpload.ERROR_WRITING_TEMP_FILE";
    KalturaDropFolderFileErrorCode["fileNoMatch"] = "13";
    KalturaDropFolderFileErrorCode["flavorMissingInFileName"] = "4";
    KalturaDropFolderFileErrorCode["flavorNotFound"] = "3";
    KalturaDropFolderFileErrorCode["localFileWrongChecksum"] = "dropFolderXmlBulkUpload.LOCAL_FILE_WRONG_CHECKSUM";
    KalturaDropFolderFileErrorCode["localFileWrongSize"] = "dropFolderXmlBulkUpload.LOCAL_FILE_WRONG_SIZE";
    KalturaDropFolderFileErrorCode["malformedXmlFile"] = "dropFolderXmlBulkUpload.MALFORMED_XML_FILE";
    KalturaDropFolderFileErrorCode["slugRegexNoMatch"] = "5";
    KalturaDropFolderFileErrorCode["xmlFileSizeExceedLimit"] = "dropFolderXmlBulkUpload.XML_FILE_SIZE_EXCEED_LIMIT";
})(KalturaDropFolderFileErrorCode = exports.KalturaDropFolderFileErrorCode || (exports.KalturaDropFolderFileErrorCode = {}));
