// kaltura-node-typescript-client/api/types/KalturaBulkUploadResultScheduleEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBulkUploadResultScheduleEvent = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBulkUploadResult_1 = require("./KalturaBulkUploadResult");
var KalturaBulkUploadResultScheduleEvent = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBulkUploadResultScheduleEvent, _super);
    function KalturaBulkUploadResultScheduleEvent(data) {
        return _super.call(this, data) || this;
    }
    KalturaBulkUploadResultScheduleEvent.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBulkUploadResultScheduleEvent' },
            referenceId: { type: 's' },
            templateEntryId: { type: 's' },
            eventType: { type: 'n' },
            title: { type: 's' },
            description: { type: 's' },
            tags: { type: 's' },
            categoryIds: { type: 's' },
            resourceId: { type: 's' },
            startTime: { type: 'd' },
            duration: { type: 'n' },
            endTime: { type: 'd' },
            recurrence: { type: 's' },
            coEditors: { type: 's' },
            coPublishers: { type: 's' },
            eventOrganizerId: { type: 's' },
            contentOwnerId: { type: 's' },
            templateEntryType: { type: 's' }
        });
        return result;
    };
    return KalturaBulkUploadResultScheduleEvent;
}(KalturaBulkUploadResult_1.KalturaBulkUploadResult));
exports.KalturaBulkUploadResultScheduleEvent = KalturaBulkUploadResultScheduleEvent;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBulkUploadResultScheduleEvent', KalturaBulkUploadResultScheduleEvent);
