// kaltura-node-typescript-client/api/types/KalturaUverseDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUverseDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaUverseDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUverseDistributionProfile, _super);
    function KalturaUverseDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaUverseDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUverseDistributionProfile' },
            feedUrl: { type: 's', readOnly: true },
            channelTitle: { type: 's' },
            channelLink: { type: 's' },
            channelDescription: { type: 's' },
            channelLanguage: { type: 's' },
            channelCopyright: { type: 's' },
            channelImageTitle: { type: 's' },
            channelImageUrl: { type: 's' },
            channelImageLink: { type: 's' },
            ftpHost: { type: 's' },
            ftpLogin: { type: 's' },
            ftpPassword: { type: 's' }
        });
        return result;
    };
    return KalturaUverseDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaUverseDistributionProfile = KalturaUverseDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUverseDistributionProfile', KalturaUverseDistributionProfile);
