// kaltura-node-typescript-client/api/types/KalturaScheduleResource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaScheduleResource = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaScheduleResourceStatus_1 = require("./KalturaScheduleResourceStatus");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaScheduleResource = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaScheduleResource, _super);
    function KalturaScheduleResource(data) {
        return _super.call(this, data) || this;
    }
    KalturaScheduleResource.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaScheduleResource' },
            id: { type: 'n', readOnly: true },
            parentId: { type: 'n' },
            partnerId: { type: 'n', readOnly: true },
            name: { type: 's' },
            systemName: { type: 's' },
            description: { type: 's' },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaScheduleResourceStatus_1.KalturaScheduleResourceStatus, subType: 'KalturaScheduleResourceStatus' },
            tags: { type: 's' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true }
        });
        return result;
    };
    return KalturaScheduleResource;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaScheduleResource = KalturaScheduleResource;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaScheduleResource', KalturaScheduleResource);
