// kaltura-node-typescript-client/api/types/KalturaZoomDropFolderFile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaZoomDropFolderFile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaZoomMeetingMetadata_1 = require("./KalturaZoomMeetingMetadata");
var KalturaZoomRecordingFile_1 = require("./KalturaZoomRecordingFile");
var KalturaDropFolderFile_1 = require("./KalturaDropFolderFile");
var KalturaZoomDropFolderFile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaZoomDropFolderFile, _super);
    function KalturaZoomDropFolderFile(data) {
        return _super.call(this, data) || this;
    }
    KalturaZoomDropFolderFile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaZoomDropFolderFile' },
            meetingMetadata: { type: 'o', subTypeConstructor: KalturaZoomMeetingMetadata_1.KalturaZoomMeetingMetadata, subType: 'KalturaZoomMeetingMetadata' },
            recordingFile: { type: 'o', subTypeConstructor: KalturaZoomRecordingFile_1.KalturaZoomRecordingFile, subType: 'KalturaZoomRecordingFile' },
            parentEntryId: { type: 's' },
            isParentEntry: { type: 'b' }
        });
        return result;
    };
    return KalturaZoomDropFolderFile;
}(KalturaDropFolderFile_1.KalturaDropFolderFile));
exports.KalturaZoomDropFolderFile = KalturaZoomDropFolderFile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaZoomDropFolderFile', KalturaZoomDropFolderFile);
