// kaltura-node-typescript-client/api/types/KalturaBatchJob.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBatchJob = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBatchJobType_1 = require("./KalturaBatchJobType");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaBatchJobStatus_1 = require("./KalturaBatchJobStatus");
var KalturaBatchHistoryData_1 = require("./KalturaBatchHistoryData");
var KalturaBatchJobErrorTypes_1 = require("./KalturaBatchJobErrorTypes");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaBatchJob = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBatchJob, _super);
    function KalturaBatchJob(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.history === 'undefined')
            _this.history = [];
        return _this;
    }
    KalturaBatchJob.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBatchJob' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            deletedAt: { type: 'd', readOnly: true },
            lockExpiration: { type: 'n', readOnly: true },
            executionAttempts: { type: 'n', readOnly: true },
            lockVersion: { type: 'n', readOnly: true },
            entryId: { type: 's' },
            entryName: { type: 's' },
            jobType: { type: 'es', readOnly: true, subTypeConstructor: KalturaBatchJobType_1.KalturaBatchJobType, subType: 'KalturaBatchJobType' },
            jobSubType: { type: 'n' },
            data: { type: 'o', subTypeConstructor: KalturaJobData_1.KalturaJobData, subType: 'KalturaJobData' },
            status: { type: 'en', subTypeConstructor: KalturaBatchJobStatus_1.KalturaBatchJobStatus, subType: 'KalturaBatchJobStatus' },
            abort: { type: 'n' },
            checkAgainTimeout: { type: 'n' },
            message: { type: 's' },
            description: { type: 's' },
            priority: { type: 'n' },
            history: { type: 'a', subTypeConstructor: KalturaBatchHistoryData_1.KalturaBatchHistoryData, subType: 'KalturaBatchHistoryData' },
            bulkJobId: { type: 'n' },
            batchVersion: { type: 'n' },
            parentJobId: { type: 'n' },
            rootJobId: { type: 'n' },
            queueTime: { type: 'n' },
            finishTime: { type: 'n' },
            errType: { type: 'en', subTypeConstructor: KalturaBatchJobErrorTypes_1.KalturaBatchJobErrorTypes, subType: 'KalturaBatchJobErrorTypes' },
            errNumber: { type: 'n' },
            estimatedEffort: { type: 'n' },
            urgency: { type: 'n' },
            schedulerId: { type: 'n' },
            workerId: { type: 'n' },
            batchIndex: { type: 'n' },
            lastSchedulerId: { type: 'n' },
            lastWorkerId: { type: 'n' },
            dc: { type: 'n' },
            jobObjectId: { type: 's' },
            jobObjectType: { type: 'n' }
        });
        return result;
    };
    return KalturaBatchJob;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaBatchJob = KalturaBatchJob;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBatchJob', KalturaBatchJob);
