// kaltura-node-typescript-client/api/types/KalturaWebexAPIIntegrationSetting.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaWebexAPIIntegrationSetting = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaWebexAPIUsersMatching_1 = require("./KalturaWebexAPIUsersMatching");
var KalturaWebexAPIGroupParticipationType_1 = require("./KalturaWebexAPIGroupParticipationType");
var KalturaIntegrationSetting_1 = require("./KalturaIntegrationSetting");
var KalturaWebexAPIIntegrationSetting = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaWebexAPIIntegrationSetting, _super);
    function KalturaWebexAPIIntegrationSetting(data) {
        return _super.call(this, data) || this;
    }
    KalturaWebexAPIIntegrationSetting.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaWebexAPIIntegrationSetting' },
            webexCategory: { type: 's' },
            enableRecordingUpload: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            enableTranscription: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            userMatchingMode: { type: 'en', subTypeConstructor: KalturaWebexAPIUsersMatching_1.KalturaWebexAPIUsersMatching, subType: 'KalturaWebexAPIUsersMatching' },
            userPostfix: { type: 's' },
            webexAccountDescription: { type: 's' },
            groupParticipationType: { type: 'en', subTypeConstructor: KalturaWebexAPIGroupParticipationType_1.KalturaWebexAPIGroupParticipationType, subType: 'KalturaWebexAPIGroupParticipationType' },
            optOutGroupNames: { type: 's' },
            optInGroupNames: { type: 's' },
            siteUrl: { type: 's' }
        });
        return result;
    };
    return KalturaWebexAPIIntegrationSetting;
}(KalturaIntegrationSetting_1.KalturaIntegrationSetting));
exports.KalturaWebexAPIIntegrationSetting = KalturaWebexAPIIntegrationSetting;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaWebexAPIIntegrationSetting', KalturaWebexAPIIntegrationSetting);
