// kaltura-node-typescript-client/api/types/KalturaSystemPartnerUsageItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSystemPartnerUsageItem = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaPartnerStatus_1 = require("./KalturaPartnerStatus");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaSystemPartnerUsageItem = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaSystemPartnerUsageItem, _super);
    function KalturaSystemPartnerUsageItem(data) {
        return _super.call(this, data) || this;
    }
    KalturaSystemPartnerUsageItem.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaSystemPartnerUsageItem' },
            partnerId: { type: 'n' },
            partnerName: { type: 's' },
            partnerStatus: { type: 'en', subTypeConstructor: KalturaPartnerStatus_1.KalturaPartnerStatus, subType: 'KalturaPartnerStatus' },
            partnerPackage: { type: 'n' },
            partnerCreatedAt: { type: 'n' },
            views: { type: 'n' },
            plays: { type: 'n' },
            entriesCount: { type: 'n' },
            totalEntriesCount: { type: 'n' },
            videoEntriesCount: { type: 'n' },
            imageEntriesCount: { type: 'n' },
            audioEntriesCount: { type: 'n' },
            mixEntriesCount: { type: 'n' },
            bandwidth: { type: 'n' },
            totalStorage: { type: 'n' },
            storage: { type: 'n' },
            peakStorage: { type: 'n' },
            avgStorage: { type: 'n' },
            combinedBandwidthStorage: { type: 'n' },
            deletedStorage: { type: 'n' },
            transcodingUsage: { type: 'n' }
        });
        return result;
    };
    return KalturaSystemPartnerUsageItem;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaSystemPartnerUsageItem = KalturaSystemPartnerUsageItem;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaSystemPartnerUsageItem', KalturaSystemPartnerUsageItem);
