// kaltura-node-typescript-client/api/types/KalturaPlayerDeliveryType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPlayerDeliveryType = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaKeyValue_1 = require("./KalturaKeyValue");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaPlayerDeliveryType = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPlayerDeliveryType, _super);
    function KalturaPlayerDeliveryType(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.flashvars === 'undefined')
            _this.flashvars = [];
        return _this;
    }
    KalturaPlayerDeliveryType.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPlayerDeliveryType' },
            id: { type: 's' },
            label: { type: 's' },
            flashvars: { type: 'a', subTypeConstructor: KalturaKeyValue_1.KalturaKeyValue, subType: 'KalturaKeyValue' },
            minVersion: { type: 's' },
            enabledByDefault: { type: 'b' }
        });
        return result;
    };
    return KalturaPlayerDeliveryType;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaPlayerDeliveryType = KalturaPlayerDeliveryType;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPlayerDeliveryType', KalturaPlayerDeliveryType);
