// kaltura-node-typescript-client/api/types/KalturaEntryContextDataParams.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEntryContextDataParams = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaAccessControlScope_1 = require("./KalturaAccessControlScope");
var KalturaEntryContextDataParams = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaEntryContextDataParams, _super);
    function KalturaEntryContextDataParams(data) {
        return _super.call(this, data) || this;
    }
    KalturaEntryContextDataParams.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaEntryContextDataParams' },
            flavorAssetId: { type: 's' },
            flavorTags: { type: 's' },
            streamerType: { type: 's' },
            mediaProtocol: { type: 's' }
        });
        return result;
    };
    return KalturaEntryContextDataParams;
}(KalturaAccessControlScope_1.KalturaAccessControlScope));
exports.KalturaEntryContextDataParams = KalturaEntryContextDataParams;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaEntryContextDataParams', KalturaEntryContextDataParams);
