// kaltura-node-typescript-client/api/types/KalturaLiveEntryRecordingOptions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveEntryRecordingOptions = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaLiveEntryRecordingOptions = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveEntryRecordingOptions, _super);
    function KalturaLiveEntryRecordingOptions(data) {
        return _super.call(this, data) || this;
    }
    KalturaLiveEntryRecordingOptions.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveEntryRecordingOptions' },
            shouldCopyEntitlement: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            shouldCopyScheduling: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            shouldCopyThumbnail: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            shouldMakeHidden: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            shouldAutoArchive: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            nonDeletedCuePointsTags: { type: 's' },
            archiveVodSuffixTimezone: { type: 's' }
        });
        return result;
    };
    return KalturaLiveEntryRecordingOptions;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaLiveEntryRecordingOptions = KalturaLiveEntryRecordingOptions;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveEntryRecordingOptions', KalturaLiveEntryRecordingOptions);
