// kaltura-node-typescript-client/api/types/KalturaAppToken.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAppToken = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaAppTokenStatus_1 = require("./KalturaAppTokenStatus");
var KalturaSessionType_1 = require("./KalturaSessionType");
var KalturaAppTokenHashType_1 = require("./KalturaAppTokenHashType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaAppToken = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAppToken, _super);
    function KalturaAppToken(data) {
        return _super.call(this, data) || this;
    }
    KalturaAppToken.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAppToken' },
            id: { type: 's', readOnly: true },
            token: { type: 's', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaAppTokenStatus_1.KalturaAppTokenStatus, subType: 'KalturaAppTokenStatus' },
            expiry: { type: 'd' },
            sessionType: { type: 'en', subTypeConstructor: KalturaSessionType_1.KalturaSessionType, subType: 'KalturaSessionType' },
            sessionUserId: { type: 's' },
            sessionDuration: { type: 'n' },
            sessionPrivileges: { type: 's' },
            hashType: { type: 'es', subTypeConstructor: KalturaAppTokenHashType_1.KalturaAppTokenHashType, subType: 'KalturaAppTokenHashType' },
            description: { type: 's' }
        });
        return result;
    };
    return KalturaAppToken;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaAppToken = KalturaAppToken;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAppToken', KalturaAppToken);
