// kaltura-node-typescript-client/api/types/KalturaESearchRange.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaESearchRange = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaESearchRange = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaESearchRange, _super);
    function KalturaESearchRange(data) {
        return _super.call(this, data) || this;
    }
    KalturaESearchRange.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaESearchRange' },
            greaterThanOrEqual: { type: 'n' },
            lessThanOrEqual: { type: 'n' },
            greaterThan: { type: 'n' },
            lessThan: { type: 'n' }
        });
        return result;
    };
    return KalturaESearchRange;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaESearchRange = KalturaESearchRange;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaESearchRange', KalturaESearchRange);
