// kaltura-node-typescript-client/api/types/KalturaConvertCollectionJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaConvertCollectionJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConvertCollectionFlavorData_1 = require("./KalturaConvertCollectionFlavorData");
var KalturaConvartableJobData_1 = require("./KalturaConvartableJobData");
var KalturaConvertCollectionJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaConvertCollectionJobData, _super);
    function KalturaConvertCollectionJobData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.flavors === 'undefined')
            _this.flavors = [];
        return _this;
    }
    KalturaConvertCollectionJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaConvertCollectionJobData' },
            destDirLocalPath: { type: 's' },
            destDirRemoteUrl: { type: 's' },
            destFileName: { type: 's' },
            inputXmlLocalPath: { type: 's' },
            inputXmlRemoteUrl: { type: 's' },
            commandLinesStr: { type: 's' },
            flavors: { type: 'a', subTypeConstructor: KalturaConvertCollectionFlavorData_1.KalturaConvertCollectionFlavorData, subType: 'KalturaConvertCollectionFlavorData' }
        });
        return result;
    };
    return KalturaConvertCollectionJobData;
}(KalturaConvartableJobData_1.KalturaConvartableJobData));
exports.KalturaConvertCollectionJobData = KalturaConvertCollectionJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaConvertCollectionJobData', KalturaConvertCollectionJobData);
