// kaltura-node-typescript-client/api/types/KalturaVendorCredit.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaVendorCredit = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBaseVendorCredit_1 = require("./KalturaBaseVendorCredit");
var KalturaVendorCredit = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaVendorCredit, _super);
    function KalturaVendorCredit(data) {
        return _super.call(this, data) || this;
    }
    KalturaVendorCredit.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaVendorCredit' },
            credit: { type: 'n' },
            fromDate: { type: 'd' },
            overageCredit: { type: 'n' },
            addOn: { type: 'n' }
        });
        return result;
    };
    return KalturaVendorCredit;
}(KalturaBaseVendorCredit_1.KalturaBaseVendorCredit));
exports.KalturaVendorCredit = KalturaVendorCredit;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaVendorCredit', KalturaVendorCredit);
