// kaltura-node-typescript-client/api/types/KalturaBulkUploadResultEntry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBulkUploadResultEntry = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBulkUploadResult_1 = require("./KalturaBulkUploadResult");
var KalturaBulkUploadResultEntry = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBulkUploadResultEntry, _super);
    function KalturaBulkUploadResultEntry(data) {
        return _super.call(this, data) || this;
    }
    KalturaBulkUploadResultEntry.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBulkUploadResultEntry' },
            entryId: { type: 's' },
            title: { type: 's' },
            description: { type: 's' },
            tags: { type: 's' },
            url: { type: 's' },
            contentType: { type: 's' },
            conversionProfileId: { type: 'n' },
            accessControlProfileId: { type: 'n' },
            category: { type: 's' },
            scheduleStartDate: { type: 'n' },
            scheduleEndDate: { type: 'n' },
            entryStatus: { type: 'n' },
            thumbnailUrl: { type: 's' },
            thumbnailSaved: { type: 'b' },
            sshPrivateKey: { type: 's' },
            sshPublicKey: { type: 's' },
            sshKeyPassphrase: { type: 's' },
            creatorId: { type: 's' },
            entitledUsersEdit: { type: 's' },
            entitledUsersPublish: { type: 's' },
            entitledUsersView: { type: 's' },
            ownerId: { type: 's' },
            referenceId: { type: 's' },
            templateEntryId: { type: 's' }
        });
        return result;
    };
    return KalturaBulkUploadResultEntry;
}(KalturaBulkUploadResult_1.KalturaBulkUploadResult));
exports.KalturaBulkUploadResultEntry = KalturaBulkUploadResultEntry;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBulkUploadResultEntry', KalturaBulkUploadResultEntry);
