// kaltura-node-typescript-client/api/types/KalturaLiveStreamEntryMatchAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveStreamEntryMatchAttribute = void 0;
var KalturaLiveStreamEntryMatchAttribute;
(function (KalturaLiveStreamEntryMatchAttribute) {
    KalturaLiveStreamEntryMatchAttribute["adminTags"] = "adminTags";
    KalturaLiveStreamEntryMatchAttribute["categoriesIds"] = "categoriesIds";
    KalturaLiveStreamEntryMatchAttribute["creatorId"] = "creatorId";
    KalturaLiveStreamEntryMatchAttribute["description"] = "description";
    KalturaLiveStreamEntryMatchAttribute["durationType"] = "durationType";
    KalturaLiveStreamEntryMatchAttribute["flavorParamsIds"] = "flavorParamsIds";
    KalturaLiveStreamEntryMatchAttribute["groupId"] = "groupId";
    KalturaLiveStreamEntryMatchAttribute["id"] = "id";
    KalturaLiveStreamEntryMatchAttribute["_name"] = "name";
    KalturaLiveStreamEntryMatchAttribute["referenceId"] = "referenceId";
    KalturaLiveStreamEntryMatchAttribute["replacedEntryId"] = "replacedEntryId";
    KalturaLiveStreamEntryMatchAttribute["replacingEntryId"] = "replacingEntryId";
    KalturaLiveStreamEntryMatchAttribute["searchText"] = "searchText";
    KalturaLiveStreamEntryMatchAttribute["tags"] = "tags";
    KalturaLiveStreamEntryMatchAttribute["userId"] = "userId";
})(KalturaLiveStreamEntryMatchAttribute = exports.KalturaLiveStreamEntryMatchAttribute || (exports.KalturaLiveStreamEntryMatchAttribute = {}));
