// kaltura-node-typescript-client/api/types/KalturaBatchJobType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBatchJobType = void 0;
var KalturaBatchJobType;
(function (KalturaBatchJobType) {
    KalturaBatchJobType["bulkdownload"] = "19";
    KalturaBatchJobType["bulkupload"] = "4";
    KalturaBatchJobType["captureThumb"] = "30";
    KalturaBatchJobType["chunkedEncodeJobScheduler"] = "44";
    KalturaBatchJobType["cleanup"] = "17";
    KalturaBatchJobType["clipConcat"] = "47";
    KalturaBatchJobType["concat"] = "35";
    KalturaBatchJobType["convert"] = "0";
    KalturaBatchJobType["convertCaptionAsset"] = "caption.convertcaptionasset";
    KalturaBatchJobType["convertCollection"] = "22";
    KalturaBatchJobType["convertLiveSegment"] = "36";
    KalturaBatchJobType["convertProfile"] = "10";
    KalturaBatchJobType["copy"] = "34";
    KalturaBatchJobType["copyCaptions"] = "43";
    KalturaBatchJobType["copyCuePoints"] = "48";
    KalturaBatchJobType["copyPartner"] = "37";
    KalturaBatchJobType["dbCleanup"] = "20";
    KalturaBatchJobType["delete"] = "2";
    KalturaBatchJobType["deleteFile"] = "31";
    KalturaBatchJobType["distributionDelete"] = "contentDistribution.DistributionDelete";
    KalturaBatchJobType["distributionDisable"] = "contentDistribution.DistributionDisable";
    KalturaBatchJobType["distributionEnable"] = "contentDistribution.DistributionEnable";
    KalturaBatchJobType["distributionFetchReport"] = "contentDistribution.DistributionFetchReport";
    KalturaBatchJobType["distributionSubmit"] = "contentDistribution.DistributionSubmit";
    KalturaBatchJobType["distributionSync"] = "contentDistribution.DistributionSync";
    KalturaBatchJobType["distributionUpdate"] = "contentDistribution.DistributionUpdate";
    KalturaBatchJobType["download"] = "6";
    KalturaBatchJobType["dropFolderContentProcessor"] = "dropFolder.DropFolderContentProcessor";
    KalturaBatchJobType["dropFolderWatcher"] = "dropFolder.DropFolderWatcher";
    KalturaBatchJobType["dvdcreator"] = "5";
    KalturaBatchJobType["emailIngestion"] = "26";
    KalturaBatchJobType["entryVendorTaskCsv"] = "reach.EntryVendorTasksCsv";
    KalturaBatchJobType["eventNotificationHandler"] = "eventNotification.EventNotificationHandler";
    KalturaBatchJobType["exportCsv"] = "49";
    KalturaBatchJobType["extractMedia"] = "14";
    KalturaBatchJobType["filesyncImport"] = "29";
    KalturaBatchJobType["flatten"] = "3";
    KalturaBatchJobType["import"] = "1";
    KalturaBatchJobType["index"] = "32";
    KalturaBatchJobType["indexTags"] = "tagSearch.IndexTagsByPrivacyContext";
    KalturaBatchJobType["integration"] = "integration.Integration";
    KalturaBatchJobType["liveEntryArchive"] = "51";
    KalturaBatchJobType["liveReportExport"] = "40";
    KalturaBatchJobType["liveToVod"] = "42";
    KalturaBatchJobType["mail"] = "15";
    KalturaBatchJobType["metadataImport"] = "27";
    KalturaBatchJobType["metadataTransform"] = "28";
    KalturaBatchJobType["moveCategoryEntries"] = "33";
    KalturaBatchJobType["multiClipConcat"] = "57";
    KalturaBatchJobType["notification"] = "16";
    KalturaBatchJobType["ooconvert"] = "7";
    KalturaBatchJobType["parseCaptionAsset"] = "captionSearch.parseCaptionAsset";
    KalturaBatchJobType["parseMultiLanguageCaptionAsset"] = "caption.parsemultilanguagecaptionasset";
    KalturaBatchJobType["postconvert"] = "11";
    KalturaBatchJobType["provisionDelete"] = "24";
    KalturaBatchJobType["provisionProvide"] = "21";
    KalturaBatchJobType["reachInternalQueueHandler"] = "ReachInternal.ReachInternalQueueHandler";
    KalturaBatchJobType["reachJobCleaner"] = "56";
    KalturaBatchJobType["recalculateCache"] = "41";
    KalturaBatchJobType["reportExport"] = "50";
    KalturaBatchJobType["scheduledTask"] = "scheduledTask.ScheduledTask";
    KalturaBatchJobType["schedulerHelper"] = "18";
    KalturaBatchJobType["searchTermCsv"] = "searchHistory.searchTermsCsv";
    KalturaBatchJobType["serverNodeMonitor"] = "45";
    KalturaBatchJobType["storageDelete"] = "25";
    KalturaBatchJobType["storageExport"] = "23";
    KalturaBatchJobType["storagePeriodicDeleteLocal"] = "55";
    KalturaBatchJobType["storagePeriodicExport"] = "53";
    KalturaBatchJobType["storagePeriodicPurge"] = "54";
    KalturaBatchJobType["storageUpdate"] = "52";
    KalturaBatchJobType["syncCategoryPrivacyContext"] = "39";
    KalturaBatchJobType["syncReachCreditTask"] = "reach.SyncReachCreditTask";
    KalturaBatchJobType["tagResolve"] = "tagSearch.TagResolve";
    KalturaBatchJobType["usersCsv"] = "46";
    KalturaBatchJobType["validateLiveMediaServers"] = "38";
    KalturaBatchJobType["virusScan"] = "virusScan.VirusScan";
    KalturaBatchJobType["widevineRepositorySync"] = "widevine.WidevineRepositorySync";
})(KalturaBatchJobType = exports.KalturaBatchJobType || (exports.KalturaBatchJobType = {}));
