// kaltura-node-typescript-client/api/types/KalturaQuizVendorTaskData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaQuizVendorTaskData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaVendorTaskData_1 = require("./KalturaVendorTaskData");
var KalturaQuizVendorTaskData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaQuizVendorTaskData, _super);
    function KalturaQuizVendorTaskData(data) {
        return _super.call(this, data) || this;
    }
    KalturaQuizVendorTaskData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaQuizVendorTaskData' },
            numberOfQuestions: { type: 'n' },
            questionsType: { type: 's' },
            context: { type: 's' },
            quizOutput: { type: 's' }
        });
        return result;
    };
    return KalturaQuizVendorTaskData;
}(KalturaVendorTaskData_1.KalturaVendorTaskData));
exports.KalturaQuizVendorTaskData = KalturaQuizVendorTaskData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaQuizVendorTaskData', KalturaQuizVendorTaskData);
