// kaltura-node-typescript-client/api/types/KalturaFacebookCaptionDistributionInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFacebookCaptionDistributionInfo = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaFacebookCaptionDistributionInfo = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaFacebookCaptionDistributionInfo, _super);
    function KalturaFacebookCaptionDistributionInfo(data) {
        return _super.call(this, data) || this;
    }
    KalturaFacebookCaptionDistributionInfo.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaFacebookCaptionDistributionInfo' },
            language: { type: 's' },
            label: { type: 's' },
            filePath: { type: 's' },
            remoteId: { type: 's' },
            version: { type: 's' },
            assetId: { type: 's' }
        });
        return result;
    };
    return KalturaFacebookCaptionDistributionInfo;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaFacebookCaptionDistributionInfo = KalturaFacebookCaptionDistributionInfo;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaFacebookCaptionDistributionInfo', KalturaFacebookCaptionDistributionInfo);
