// kaltura-node-typescript-client/api/types/KalturaGoogleVideoSyndicationFeedOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaGoogleVideoSyndicationFeedOrderBy = void 0;
var KalturaGoogleVideoSyndicationFeedOrderBy;
(function (KalturaGoogleVideoSyndicationFeedOrderBy) {
    KalturaGoogleVideoSyndicationFeedOrderBy["createdAtAsc"] = "+createdAt";
    KalturaGoogleVideoSyndicationFeedOrderBy["createdAtDesc"] = "-createdAt";
    KalturaGoogleVideoSyndicationFeedOrderBy["nameAsc"] = "+name";
    KalturaGoogleVideoSyndicationFeedOrderBy["nameDesc"] = "-name";
    KalturaGoogleVideoSyndicationFeedOrderBy["playlistIdAsc"] = "+playlistId";
    KalturaGoogleVideoSyndicationFeedOrderBy["playlistIdDesc"] = "-playlistId";
    KalturaGoogleVideoSyndicationFeedOrderBy["typeAsc"] = "+type";
    KalturaGoogleVideoSyndicationFeedOrderBy["typeDesc"] = "-type";
    KalturaGoogleVideoSyndicationFeedOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaGoogleVideoSyndicationFeedOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaGoogleVideoSyndicationFeedOrderBy = exports.KalturaGoogleVideoSyndicationFeedOrderBy || (exports.KalturaGoogleVideoSyndicationFeedOrderBy = {}));
