// kaltura-node-typescript-client/api/types/KalturaRoomDetails.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaRoomDetails = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaRoomDetails = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaRoomDetails, _super);
    function KalturaRoomDetails(data) {
        return _super.call(this, data) || this;
    }
    KalturaRoomDetails.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaRoomDetails' },
            serverUrl: { type: 's' },
            entryId: { type: 's' },
            token: { type: 's' },
            expiry: { type: 'n' },
            serverName: { type: 's' }
        });
        return result;
    };
    return KalturaRoomDetails;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaRoomDetails = KalturaRoomDetails;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaRoomDetails', KalturaRoomDetails);
