// kaltura-node-typescript-client/api/types/KalturaQuickPlayDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaQuickPlayDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaQuickPlayDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaQuickPlayDistributionProfile, _super);
    function KalturaQuickPlayDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaQuickPlayDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaQuickPlayDistributionProfile' },
            sftpHost: { type: 's' },
            sftpLogin: { type: 's' },
            sftpPass: { type: 's' },
            sftpBasePath: { type: 's' },
            channelTitle: { type: 's' },
            channelLink: { type: 's' },
            channelDescription: { type: 's' },
            channelManagingEditor: { type: 's' },
            channelLanguage: { type: 's' },
            channelImageTitle: { type: 's' },
            channelImageWidth: { type: 's' },
            channelImageHeight: { type: 's' },
            channelImageLink: { type: 's' },
            channelImageUrl: { type: 's' },
            channelCopyright: { type: 's' },
            channelGenerator: { type: 's' },
            channelRating: { type: 's' }
        });
        return result;
    };
    return KalturaQuickPlayDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaQuickPlayDistributionProfile = KalturaQuickPlayDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaQuickPlayDistributionProfile', KalturaQuickPlayDistributionProfile);
