// kaltura-node-typescript-client/api/types/KalturaIntegrationJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaIntegrationJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaIntegrationProviderType_1 = require("./KalturaIntegrationProviderType");
var KalturaIntegrationJobProviderData_1 = require("./KalturaIntegrationJobProviderData");
var KalturaIntegrationTriggerType_1 = require("./KalturaIntegrationTriggerType");
var KalturaIntegrationJobTriggerData_1 = require("./KalturaIntegrationJobTriggerData");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaIntegrationJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaIntegrationJobData, _super);
    function KalturaIntegrationJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaIntegrationJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaIntegrationJobData' },
            callbackNotificationUrl: { type: 's', readOnly: true },
            providerType: { type: 'es', subTypeConstructor: KalturaIntegrationProviderType_1.KalturaIntegrationProviderType, subType: 'KalturaIntegrationProviderType' },
            providerData: { type: 'o', subTypeConstructor: KalturaIntegrationJobProviderData_1.KalturaIntegrationJobProviderData, subType: 'KalturaIntegrationJobProviderData' },
            triggerType: { type: 'es', subTypeConstructor: KalturaIntegrationTriggerType_1.KalturaIntegrationTriggerType, subType: 'KalturaIntegrationTriggerType' },
            triggerData: { type: 'o', subTypeConstructor: KalturaIntegrationJobTriggerData_1.KalturaIntegrationJobTriggerData, subType: 'KalturaIntegrationJobTriggerData' }
        });
        return result;
    };
    return KalturaIntegrationJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaIntegrationJobData = KalturaIntegrationJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaIntegrationJobData', KalturaIntegrationJobData);
