// kaltura-node-typescript-client/api/types/KalturaLiveReportExportJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveReportExportJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaLiveReportExportJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveReportExportJobData, _super);
    function KalturaLiveReportExportJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaLiveReportExportJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveReportExportJobData' },
            timeReference: { type: 'd' },
            timeZoneOffset: { type: 'n' },
            entryIds: { type: 's' },
            outputPath: { type: 's' },
            recipientEmail: { type: 's' }
        });
        return result;
    };
    return KalturaLiveReportExportJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaLiveReportExportJobData = KalturaLiveReportExportJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveReportExportJobData', KalturaLiveReportExportJobData);
