// kaltura-node-typescript-client/api/types/KalturaExternalMediaEntryOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaExternalMediaEntryOrderBy = void 0;
var KalturaExternalMediaEntryOrderBy;
(function (KalturaExternalMediaEntryOrderBy) {
    KalturaExternalMediaEntryOrderBy["createdAtAsc"] = "+createdAt";
    KalturaExternalMediaEntryOrderBy["createdAtDesc"] = "-createdAt";
    KalturaExternalMediaEntryOrderBy["durationAsc"] = "+duration";
    KalturaExternalMediaEntryOrderBy["durationDesc"] = "-duration";
    KalturaExternalMediaEntryOrderBy["endDateAsc"] = "+endDate";
    KalturaExternalMediaEntryOrderBy["endDateDesc"] = "-endDate";
    KalturaExternalMediaEntryOrderBy["lastPlayedAtAsc"] = "+lastPlayedAt";
    KalturaExternalMediaEntryOrderBy["lastPlayedAtDesc"] = "-lastPlayedAt";
    KalturaExternalMediaEntryOrderBy["mediaTypeAsc"] = "+mediaType";
    KalturaExternalMediaEntryOrderBy["mediaTypeDesc"] = "-mediaType";
    KalturaExternalMediaEntryOrderBy["moderationCountAsc"] = "+moderationCount";
    KalturaExternalMediaEntryOrderBy["moderationCountDesc"] = "-moderationCount";
    KalturaExternalMediaEntryOrderBy["nameAsc"] = "+name";
    KalturaExternalMediaEntryOrderBy["nameDesc"] = "-name";
    KalturaExternalMediaEntryOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaExternalMediaEntryOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaExternalMediaEntryOrderBy["playsAsc"] = "+plays";
    KalturaExternalMediaEntryOrderBy["playsDesc"] = "-plays";
    KalturaExternalMediaEntryOrderBy["rankAsc"] = "+rank";
    KalturaExternalMediaEntryOrderBy["rankDesc"] = "-rank";
    KalturaExternalMediaEntryOrderBy["recentAsc"] = "+recent";
    KalturaExternalMediaEntryOrderBy["recentDesc"] = "-recent";
    KalturaExternalMediaEntryOrderBy["startDateAsc"] = "+startDate";
    KalturaExternalMediaEntryOrderBy["startDateDesc"] = "-startDate";
    KalturaExternalMediaEntryOrderBy["totalRankAsc"] = "+totalRank";
    KalturaExternalMediaEntryOrderBy["totalRankDesc"] = "-totalRank";
    KalturaExternalMediaEntryOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaExternalMediaEntryOrderBy["updatedAtDesc"] = "-updatedAt";
    KalturaExternalMediaEntryOrderBy["viewsAsc"] = "+views";
    KalturaExternalMediaEntryOrderBy["viewsDesc"] = "-views";
    KalturaExternalMediaEntryOrderBy["weightAsc"] = "+weight";
    KalturaExternalMediaEntryOrderBy["weightDesc"] = "-weight";
})(KalturaExternalMediaEntryOrderBy = exports.KalturaExternalMediaEntryOrderBy || (exports.KalturaExternalMediaEntryOrderBy = {}));
