// kaltura-node-typescript-client/api/types/KalturaAssetBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAssetBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaAssetBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAssetBaseFilter, _super);
    function KalturaAssetBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaAssetBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAssetBaseFilter' },
            idEqual: { type: 's' },
            idIn: { type: 's' },
            entryIdEqual: { type: 's' },
            entryIdIn: { type: 's' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            sizeGreaterThanOrEqual: { type: 'n' },
            sizeLessThanOrEqual: { type: 'n' },
            tagsLike: { type: 's' },
            tagsMultiLikeOr: { type: 's' },
            tagsMultiLikeAnd: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            deletedAtGreaterThanOrEqual: { type: 'd' },
            deletedAtLessThanOrEqual: { type: 'd' }
        });
        return result;
    };
    return KalturaAssetBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaAssetBaseFilter = KalturaAssetBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAssetBaseFilter', KalturaAssetBaseFilter);
