// kaltura-node-typescript-client/api/types/KalturaScheduleEventResource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaScheduleEventResource = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaScheduleEventResource = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaScheduleEventResource, _super);
    function KalturaScheduleEventResource(data) {
        return _super.call(this, data) || this;
    }
    KalturaScheduleEventResource.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaScheduleEventResource' },
            eventId: { type: 'n' },
            resourceId: { type: 'n' },
            partnerId: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true }
        });
        return result;
    };
    return KalturaScheduleEventResource;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaScheduleEventResource = KalturaScheduleEventResource;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaScheduleEventResource', KalturaScheduleEventResource);
