// kaltura-node-typescript-client/api/types/KalturaUrlResource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUrlResource = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaString_1 = require("./KalturaString");
var KalturaContentResource_1 = require("./KalturaContentResource");
var KalturaUrlResource = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUrlResource, _super);
    function KalturaUrlResource(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.urlHeaders === 'undefined')
            _this.urlHeaders = [];
        return _this;
    }
    KalturaUrlResource.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUrlResource' },
            url: { type: 's' },
            forceAsyncDownload: { type: 'b' },
            urlHeaders: { type: 'a', subTypeConstructor: KalturaString_1.KalturaString, subType: 'KalturaString' },
            shouldRedirect: { type: 'b' }
        });
        return result;
    };
    return KalturaUrlResource;
}(KalturaContentResource_1.KalturaContentResource));
exports.KalturaUrlResource = KalturaUrlResource;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUrlResource', KalturaUrlResource);
