// kaltura-node-typescript-client/api/types/KalturaEntryDistributionBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEntryDistributionBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaEntryDistributionStatus_1 = require("./KalturaEntryDistributionStatus");
var KalturaEntryDistributionFlag_1 = require("./KalturaEntryDistributionFlag");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaEntryDistributionBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaEntryDistributionBaseFilter, _super);
    function KalturaEntryDistributionBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaEntryDistributionBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaEntryDistributionBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            submittedAtGreaterThanOrEqual: { type: 'd' },
            submittedAtLessThanOrEqual: { type: 'd' },
            entryIdEqual: { type: 's' },
            entryIdIn: { type: 's' },
            distributionProfileIdEqual: { type: 'n' },
            distributionProfileIdIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaEntryDistributionStatus_1.KalturaEntryDistributionStatus, subType: 'KalturaEntryDistributionStatus' },
            statusIn: { type: 's' },
            dirtyStatusEqual: { type: 'en', subTypeConstructor: KalturaEntryDistributionFlag_1.KalturaEntryDistributionFlag, subType: 'KalturaEntryDistributionFlag' },
            dirtyStatusIn: { type: 's' },
            sunriseGreaterThanOrEqual: { type: 'd' },
            sunriseLessThanOrEqual: { type: 'd' },
            sunsetGreaterThanOrEqual: { type: 'd' },
            sunsetLessThanOrEqual: { type: 'd' }
        });
        return result;
    };
    return KalturaEntryDistributionBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaEntryDistributionBaseFilter = KalturaEntryDistributionBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaEntryDistributionBaseFilter', KalturaEntryDistributionBaseFilter);
