// kaltura-node-typescript-client/api/types/KalturaCategoryEntryCondition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCategoryEntryCondition = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCategoryUserPermissionLevel_1 = require("./KalturaCategoryUserPermissionLevel");
var KalturaSearchConditionComparison_1 = require("./KalturaSearchConditionComparison");
var KalturaCondition_1 = require("./KalturaCondition");
var KalturaCategoryEntryCondition = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCategoryEntryCondition, _super);
    function KalturaCategoryEntryCondition(data) {
        return _super.call(this, data) || this;
    }
    KalturaCategoryEntryCondition.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCategoryEntryCondition' },
            categoryId: { type: 'n' },
            categoryIds: { type: 's' },
            categoryUserPermission: { type: 'en', subTypeConstructor: KalturaCategoryUserPermissionLevel_1.KalturaCategoryUserPermissionLevel, subType: 'KalturaCategoryUserPermissionLevel' },
            comparison: { type: 'es', subTypeConstructor: KalturaSearchConditionComparison_1.KalturaSearchConditionComparison, subType: 'KalturaSearchConditionComparison' }
        });
        return result;
    };
    return KalturaCategoryEntryCondition;
}(KalturaCondition_1.KalturaCondition));
exports.KalturaCategoryEntryCondition = KalturaCategoryEntryCondition;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCategoryEntryCondition', KalturaCategoryEntryCondition);
