// kaltura-node-typescript-client/api/types/KalturaSchedulerConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSchedulerConfig = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaSchedulerConfig = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaSchedulerConfig, _super);
    function KalturaSchedulerConfig(data) {
        return _super.call(this, data) || this;
    }
    KalturaSchedulerConfig.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaSchedulerConfig' },
            id: { type: 'n', readOnly: true },
            createdBy: { type: 's' },
            updatedBy: { type: 's' },
            commandId: { type: 's' },
            commandStatus: { type: 's' },
            schedulerId: { type: 'n' },
            schedulerConfiguredId: { type: 'n' },
            schedulerName: { type: 's' },
            workerId: { type: 'n' },
            workerConfiguredId: { type: 'n' },
            workerName: { type: 's' },
            variable: { type: 's' },
            variablePart: { type: 's' },
            value: { type: 's' }
        });
        return result;
    };
    return KalturaSchedulerConfig;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaSchedulerConfig = KalturaSchedulerConfig;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaSchedulerConfig', KalturaSchedulerConfig);
