// kaltura-node-typescript-client/api/types/KalturaGenericDistributionProviderActionBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaGenericDistributionProviderActionBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDistributionAction_1 = require("./KalturaDistributionAction");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaGenericDistributionProviderActionBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaGenericDistributionProviderActionBaseFilter, _super);
    function KalturaGenericDistributionProviderActionBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaGenericDistributionProviderActionBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaGenericDistributionProviderActionBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            genericDistributionProviderIdEqual: { type: 'n' },
            genericDistributionProviderIdIn: { type: 's' },
            actionEqual: { type: 'en', subTypeConstructor: KalturaDistributionAction_1.KalturaDistributionAction, subType: 'KalturaDistributionAction' },
            actionIn: { type: 's' }
        });
        return result;
    };
    return KalturaGenericDistributionProviderActionBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaGenericDistributionProviderActionBaseFilter = KalturaGenericDistributionProviderActionBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaGenericDistributionProviderActionBaseFilter', KalturaGenericDistributionProviderActionBaseFilter);
