// kaltura-node-typescript-client/api/types/KalturaESearchEntryOrderByFieldName.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaESearchEntryOrderByFieldName = void 0;
var KalturaESearchEntryOrderByFieldName;
(function (KalturaESearchEntryOrderByFieldName) {
    KalturaESearchEntryOrderByFieldName["createdAt"] = "created_at";
    KalturaESearchEntryOrderByFieldName["endDate"] = "end_date";
    KalturaESearchEntryOrderByFieldName["lastPlayedAt"] = "last_played_at";
    KalturaESearchEntryOrderByFieldName["_name"] = "name";
    KalturaESearchEntryOrderByFieldName["plays"] = "plays";
    KalturaESearchEntryOrderByFieldName["playsLast1Day"] = "plays_last_1_day";
    KalturaESearchEntryOrderByFieldName["playsLast30Days"] = "plays_last_30_days";
    KalturaESearchEntryOrderByFieldName["playsLast7Days"] = "plays_last_7_days";
    KalturaESearchEntryOrderByFieldName["rank"] = "rank";
    KalturaESearchEntryOrderByFieldName["recycledAt"] = "recycled_at";
    KalturaESearchEntryOrderByFieldName["startDate"] = "start_date";
    KalturaESearchEntryOrderByFieldName["updatedAt"] = "updated_at";
    KalturaESearchEntryOrderByFieldName["views"] = "views";
    KalturaESearchEntryOrderByFieldName["viewsLast1Day"] = "views_last_1_day";
    KalturaESearchEntryOrderByFieldName["viewsLast30Days"] = "views_last_30_days";
    KalturaESearchEntryOrderByFieldName["viewsLast7Days"] = "views_last_7_days";
    KalturaESearchEntryOrderByFieldName["votes"] = "votes";
})(KalturaESearchEntryOrderByFieldName = exports.KalturaESearchEntryOrderByFieldName || (exports.KalturaESearchEntryOrderByFieldName = {}));
