// kaltura-node-typescript-client/api/types/KalturaControlPanelCommand.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaControlPanelCommand = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaControlPanelCommandType_1 = require("./KalturaControlPanelCommandType");
var KalturaControlPanelCommandTargetType_1 = require("./KalturaControlPanelCommandTargetType");
var KalturaControlPanelCommandStatus_1 = require("./KalturaControlPanelCommandStatus");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaControlPanelCommand = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaControlPanelCommand, _super);
    function KalturaControlPanelCommand(data) {
        return _super.call(this, data) || this;
    }
    KalturaControlPanelCommand.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaControlPanelCommand' },
            id: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            createdBy: { type: 's' },
            updatedAt: { type: 'd', readOnly: true },
            updatedBy: { type: 's' },
            createdById: { type: 'n' },
            schedulerId: { type: 'n' },
            workerId: { type: 'n' },
            workerConfiguredId: { type: 'n' },
            workerName: { type: 'n' },
            batchIndex: { type: 'n' },
            type: { type: 'en', subTypeConstructor: KalturaControlPanelCommandType_1.KalturaControlPanelCommandType, subType: 'KalturaControlPanelCommandType' },
            targetType: { type: 'en', subTypeConstructor: KalturaControlPanelCommandTargetType_1.KalturaControlPanelCommandTargetType, subType: 'KalturaControlPanelCommandTargetType' },
            status: { type: 'en', subTypeConstructor: KalturaControlPanelCommandStatus_1.KalturaControlPanelCommandStatus, subType: 'KalturaControlPanelCommandStatus' },
            cause: { type: 's' },
            description: { type: 's' },
            errorDescription: { type: 's' }
        });
        return result;
    };
    return KalturaControlPanelCommand;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaControlPanelCommand = KalturaControlPanelCommand;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaControlPanelCommand', KalturaControlPanelCommand);
