// kaltura-node-typescript-client/api/types/KalturaVirusScanProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaVirusScanProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaVirusScanProfileStatus_1 = require("./KalturaVirusScanProfileStatus");
var KalturaVirusScanEngineType_1 = require("./KalturaVirusScanEngineType");
var KalturaBaseEntryFilter_1 = require("./KalturaBaseEntryFilter");
var KalturaVirusFoundAction_1 = require("./KalturaVirusFoundAction");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaVirusScanProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaVirusScanProfile, _super);
    function KalturaVirusScanProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaVirusScanProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaVirusScanProfile' },
            id: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            name: { type: 's' },
            status: { type: 'en', subTypeConstructor: KalturaVirusScanProfileStatus_1.KalturaVirusScanProfileStatus, subType: 'KalturaVirusScanProfileStatus' },
            engineType: { type: 'es', subTypeConstructor: KalturaVirusScanEngineType_1.KalturaVirusScanEngineType, subType: 'KalturaVirusScanEngineType' },
            entryFilter: { type: 'o', subTypeConstructor: KalturaBaseEntryFilter_1.KalturaBaseEntryFilter, subType: 'KalturaBaseEntryFilter' },
            actionIfInfected: { type: 'en', subTypeConstructor: KalturaVirusFoundAction_1.KalturaVirusFoundAction, subType: 'KalturaVirusFoundAction' }
        });
        return result;
    };
    return KalturaVirusScanProfile;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaVirusScanProfile = KalturaVirusScanProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaVirusScanProfile', KalturaVirusScanProfile);
