// kaltura-node-typescript-client/api/types/KalturaReportBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaReportBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaReportBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaReportBaseFilter, _super);
    function KalturaReportBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaReportBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaReportBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            systemNameEqual: { type: 's' },
            systemNameIn: { type: 's' }
        });
        return result;
    };
    return KalturaReportBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaReportBaseFilter = KalturaReportBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaReportBaseFilter', KalturaReportBaseFilter);
