// kaltura-node-typescript-client/api/types/KalturaESearchParams.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaESearchParams = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaESearchOrderBy_1 = require("./KalturaESearchOrderBy");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaESearchParams = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaESearchParams, _super);
    function KalturaESearchParams(data) {
        return _super.call(this, data) || this;
    }
    KalturaESearchParams.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaESearchParams' },
            objectStatuses: { type: 's' },
            objectId: { type: 's' },
            orderBy: { type: 'o', subTypeConstructor: KalturaESearchOrderBy_1.KalturaESearchOrderBy, subType: 'KalturaESearchOrderBy' },
            ignoreSynonym: { type: 'b' }
        });
        return result;
    };
    return KalturaESearchParams;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaESearchParams = KalturaESearchParams;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaESearchParams', KalturaESearchParams);
