// kaltura-node-typescript-client/api/types/KalturaEntryVendorTaskStatus.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEntryVendorTaskStatus = void 0;
var KalturaEntryVendorTaskStatus;
(function (KalturaEntryVendorTaskStatus) {
    KalturaEntryVendorTaskStatus[KalturaEntryVendorTaskStatus["aborted"] = 7] = "aborted";
    KalturaEntryVendorTaskStatus[KalturaEntryVendorTaskStatus["error"] = 6] = "error";
    KalturaEntryVendorTaskStatus[KalturaEntryVendorTaskStatus["pending"] = 1] = "pending";
    KalturaEntryVendorTaskStatus[KalturaEntryVendorTaskStatus["pendingEntryReady"] = 8] = "pendingEntryReady";
    KalturaEntryVendorTaskStatus[KalturaEntryVendorTaskStatus["pendingModeration"] = 4] = "pendingModeration";
    KalturaEntryVendorTaskStatus[KalturaEntryVendorTaskStatus["processing"] = 3] = "processing";
    KalturaEntryVendorTaskStatus[KalturaEntryVendorTaskStatus["ready"] = 2] = "ready";
    KalturaEntryVendorTaskStatus[KalturaEntryVendorTaskStatus["rejected"] = 5] = "rejected";
    KalturaEntryVendorTaskStatus[KalturaEntryVendorTaskStatus["scheduled"] = 9] = "scheduled";
})(KalturaEntryVendorTaskStatus = exports.KalturaEntryVendorTaskStatus || (exports.KalturaEntryVendorTaskStatus = {}));
