// kaltura-node-typescript-client/api/types/KalturaEventCuePointOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEventCuePointOrderBy = void 0;
var KalturaEventCuePointOrderBy;
(function (KalturaEventCuePointOrderBy) {
    KalturaEventCuePointOrderBy["createdAtAsc"] = "+createdAt";
    KalturaEventCuePointOrderBy["createdAtDesc"] = "-createdAt";
    KalturaEventCuePointOrderBy["intIdAsc"] = "+intId";
    KalturaEventCuePointOrderBy["intIdDesc"] = "-intId";
    KalturaEventCuePointOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaEventCuePointOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaEventCuePointOrderBy["startTimeAsc"] = "+startTime";
    KalturaEventCuePointOrderBy["startTimeDesc"] = "-startTime";
    KalturaEventCuePointOrderBy["triggeredAtAsc"] = "+triggeredAt";
    KalturaEventCuePointOrderBy["triggeredAtDesc"] = "-triggeredAt";
    KalturaEventCuePointOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaEventCuePointOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaEventCuePointOrderBy = exports.KalturaEventCuePointOrderBy || (exports.KalturaEventCuePointOrderBy = {}));
