// kaltura-node-typescript-client/api/types/KalturaControlPanelCommandBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaControlPanelCommandBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaControlPanelCommandType_1 = require("./KalturaControlPanelCommandType");
var KalturaControlPanelCommandTargetType_1 = require("./KalturaControlPanelCommandTargetType");
var KalturaControlPanelCommandStatus_1 = require("./KalturaControlPanelCommandStatus");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaControlPanelCommandBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaControlPanelCommandBaseFilter, _super);
    function KalturaControlPanelCommandBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaControlPanelCommandBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaControlPanelCommandBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            createdByIdEqual: { type: 'n' },
            typeEqual: { type: 'en', subTypeConstructor: KalturaControlPanelCommandType_1.KalturaControlPanelCommandType, subType: 'KalturaControlPanelCommandType' },
            typeIn: { type: 's' },
            targetTypeEqual: { type: 'en', subTypeConstructor: KalturaControlPanelCommandTargetType_1.KalturaControlPanelCommandTargetType, subType: 'KalturaControlPanelCommandTargetType' },
            targetTypeIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaControlPanelCommandStatus_1.KalturaControlPanelCommandStatus, subType: 'KalturaControlPanelCommandStatus' },
            statusIn: { type: 's' }
        });
        return result;
    };
    return KalturaControlPanelCommandBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaControlPanelCommandBaseFilter = KalturaControlPanelCommandBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaControlPanelCommandBaseFilter', KalturaControlPanelCommandBaseFilter);
