// kaltura-node-typescript-client/api/types/KalturaSessionCuePointOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSessionCuePointOrderBy = void 0;
var KalturaSessionCuePointOrderBy;
(function (KalturaSessionCuePointOrderBy) {
    KalturaSessionCuePointOrderBy["createdAtAsc"] = "+createdAt";
    KalturaSessionCuePointOrderBy["createdAtDesc"] = "-createdAt";
    KalturaSessionCuePointOrderBy["endTimeAsc"] = "+endTime";
    KalturaSessionCuePointOrderBy["endTimeDesc"] = "-endTime";
    KalturaSessionCuePointOrderBy["intIdAsc"] = "+intId";
    KalturaSessionCuePointOrderBy["intIdDesc"] = "-intId";
    KalturaSessionCuePointOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaSessionCuePointOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaSessionCuePointOrderBy["startTimeAsc"] = "+startTime";
    KalturaSessionCuePointOrderBy["startTimeDesc"] = "-startTime";
    KalturaSessionCuePointOrderBy["triggeredAtAsc"] = "+triggeredAt";
    KalturaSessionCuePointOrderBy["triggeredAtDesc"] = "-triggeredAt";
    KalturaSessionCuePointOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaSessionCuePointOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaSessionCuePointOrderBy = exports.KalturaSessionCuePointOrderBy || (exports.KalturaSessionCuePointOrderBy = {}));
