// kaltura-node-typescript-client/api/types/KalturaMetadataBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMetadataBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaMetadataObjectType_1 = require("./KalturaMetadataObjectType");
var KalturaMetadataStatus_1 = require("./KalturaMetadataStatus");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaMetadataBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMetadataBaseFilter, _super);
    function KalturaMetadataBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaMetadataBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMetadataBaseFilter' },
            partnerIdEqual: { type: 'n' },
            metadataProfileIdEqual: { type: 'n' },
            metadataProfileIdIn: { type: 's' },
            metadataProfileVersionEqual: { type: 'n' },
            metadataProfileVersionGreaterThanOrEqual: { type: 'n' },
            metadataProfileVersionLessThanOrEqual: { type: 'n' },
            metadataObjectTypeEqual: { type: 'es', subTypeConstructor: KalturaMetadataObjectType_1.KalturaMetadataObjectType, subType: 'KalturaMetadataObjectType' },
            objectIdEqual: { type: 's' },
            objectIdIn: { type: 's' },
            versionEqual: { type: 'n' },
            versionGreaterThanOrEqual: { type: 'n' },
            versionLessThanOrEqual: { type: 'n' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaMetadataStatus_1.KalturaMetadataStatus, subType: 'KalturaMetadataStatus' },
            statusIn: { type: 's' }
        });
        return result;
    };
    return KalturaMetadataBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaMetadataBaseFilter = KalturaMetadataBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMetadataBaseFilter', KalturaMetadataBaseFilter);
