// kaltura-node-typescript-client/api/types/KalturaAccessControlProfileBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAccessControlProfileBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaAccessControlProfileBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAccessControlProfileBaseFilter, _super);
    function KalturaAccessControlProfileBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaAccessControlProfileBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAccessControlProfileBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            systemNameEqual: { type: 's' },
            systemNameIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' }
        });
        return result;
    };
    return KalturaAccessControlProfileBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaAccessControlProfileBaseFilter = KalturaAccessControlProfileBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAccessControlProfileBaseFilter', KalturaAccessControlProfileBaseFilter);
