// kaltura-node-typescript-client/api/types/KalturaScheduleEventOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaScheduleEventOrderBy = void 0;
var KalturaScheduleEventOrderBy;
(function (KalturaScheduleEventOrderBy) {
    KalturaScheduleEventOrderBy["createdAtAsc"] = "+createdAt";
    KalturaScheduleEventOrderBy["createdAtDesc"] = "-createdAt";
    KalturaScheduleEventOrderBy["endDateAsc"] = "+endDate";
    KalturaScheduleEventOrderBy["endDateDesc"] = "-endDate";
    KalturaScheduleEventOrderBy["priorityAsc"] = "+priority";
    KalturaScheduleEventOrderBy["priorityDesc"] = "-priority";
    KalturaScheduleEventOrderBy["startDateAsc"] = "+startDate";
    KalturaScheduleEventOrderBy["startDateDesc"] = "-startDate";
    KalturaScheduleEventOrderBy["summaryAsc"] = "+summary";
    KalturaScheduleEventOrderBy["summaryDesc"] = "-summary";
    KalturaScheduleEventOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaScheduleEventOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaScheduleEventOrderBy = exports.KalturaScheduleEventOrderBy || (exports.KalturaScheduleEventOrderBy = {}));
