// kaltura-node-typescript-client/api/types/KalturaTranscriptAsset.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaTranscriptAsset = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaLanguage_1 = require("./KalturaLanguage");
var KalturaTranscriptProviderType_1 = require("./KalturaTranscriptProviderType");
var KalturaAttachmentAsset_1 = require("./KalturaAttachmentAsset");
var KalturaTranscriptAsset = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaTranscriptAsset, _super);
    function KalturaTranscriptAsset(data) {
        return _super.call(this, data) || this;
    }
    KalturaTranscriptAsset.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaTranscriptAsset' },
            accuracy: { type: 'n' },
            humanVerified: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            language: { type: 'es', subTypeConstructor: KalturaLanguage_1.KalturaLanguage, subType: 'KalturaLanguage' },
            providerType: { type: 'es', subTypeConstructor: KalturaTranscriptProviderType_1.KalturaTranscriptProviderType, subType: 'KalturaTranscriptProviderType' }
        });
        return result;
    };
    return KalturaTranscriptAsset;
}(KalturaAttachmentAsset_1.KalturaAttachmentAsset));
exports.KalturaTranscriptAsset = KalturaTranscriptAsset;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaTranscriptAsset', KalturaTranscriptAsset);
