// kaltura-node-typescript-client/api/types/KalturaAnswerCuePointOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAnswerCuePointOrderBy = void 0;
var KalturaAnswerCuePointOrderBy;
(function (KalturaAnswerCuePointOrderBy) {
    KalturaAnswerCuePointOrderBy["createdAtAsc"] = "+createdAt";
    KalturaAnswerCuePointOrderBy["createdAtDesc"] = "-createdAt";
    KalturaAnswerCuePointOrderBy["intIdAsc"] = "+intId";
    KalturaAnswerCuePointOrderBy["intIdDesc"] = "-intId";
    KalturaAnswerCuePointOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaAnswerCuePointOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaAnswerCuePointOrderBy["startTimeAsc"] = "+startTime";
    KalturaAnswerCuePointOrderBy["startTimeDesc"] = "-startTime";
    KalturaAnswerCuePointOrderBy["triggeredAtAsc"] = "+triggeredAt";
    KalturaAnswerCuePointOrderBy["triggeredAtDesc"] = "-triggeredAt";
    KalturaAnswerCuePointOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaAnswerCuePointOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaAnswerCuePointOrderBy = exports.KalturaAnswerCuePointOrderBy || (exports.KalturaAnswerCuePointOrderBy = {}));
