// kaltura-node-typescript-client/api/types/KalturaDexterIntegrationJobProviderData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDexterIntegrationJobProviderData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaIntegrationJobProviderData_1 = require("./KalturaIntegrationJobProviderData");
var KalturaDexterIntegrationJobProviderData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDexterIntegrationJobProviderData, _super);
    function KalturaDexterIntegrationJobProviderData(data) {
        return _super.call(this, data) || this;
    }
    KalturaDexterIntegrationJobProviderData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDexterIntegrationJobProviderData' },
            metadataProfileId: { type: 'n' },
            transcriptAssetId: { type: 's' },
            entryId: { type: 's' },
            voicebaseApiKey: { type: 's' },
            voicebaseApiPassword: { type: 's' }
        });
        return result;
    };
    return KalturaDexterIntegrationJobProviderData;
}(KalturaIntegrationJobProviderData_1.KalturaIntegrationJobProviderData));
exports.KalturaDexterIntegrationJobProviderData = KalturaDexterIntegrationJobProviderData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDexterIntegrationJobProviderData', KalturaDexterIntegrationJobProviderData);
