// kaltura-node-typescript-client/api/types/KalturaSsoBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSsoBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaSsoStatus_1 = require("./KalturaSsoStatus");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaSsoBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaSsoBaseFilter, _super);
    function KalturaSsoBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaSsoBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaSsoBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            applicationTypeEqual: { type: 's' },
            partnerIdEqual: { type: 'n' },
            domainEqual: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaSsoStatus_1.KalturaSsoStatus, subType: 'KalturaSsoStatus' },
            statusIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            redirectUrlEqual: { type: 's' }
        });
        return result;
    };
    return KalturaSsoBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaSsoBaseFilter = KalturaSsoBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaSsoBaseFilter', KalturaSsoBaseFilter);
