// kaltura-node-typescript-client/api/types/KalturaPlayableEntry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPlayableEntry = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDurationType_1 = require("./KalturaDurationType");
var KalturaBaseEntry_1 = require("./KalturaBaseEntry");
var KalturaPlayableEntry = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPlayableEntry, _super);
    function KalturaPlayableEntry(data) {
        return _super.call(this, data) || this;
    }
    KalturaPlayableEntry.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPlayableEntry' },
            plays: { type: 'n', readOnly: true },
            views: { type: 'n', readOnly: true },
            lastPlayedAt: { type: 'd', readOnly: true },
            width: { type: 'n', readOnly: true },
            height: { type: 'n', readOnly: true },
            duration: { type: 'n', readOnly: true },
            msDuration: { type: 'n' },
            durationType: { type: 'es', readOnly: true, subTypeConstructor: KalturaDurationType_1.KalturaDurationType, subType: 'KalturaDurationType' }
        });
        return result;
    };
    return KalturaPlayableEntry;
}(KalturaBaseEntry_1.KalturaBaseEntry));
exports.KalturaPlayableEntry = KalturaPlayableEntry;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPlayableEntry', KalturaPlayableEntry);
