// kaltura-node-typescript-client/api/types/KalturaSchedulerWorker.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSchedulerWorker = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBatchJobType_1 = require("./KalturaBatchJobType");
var KalturaSchedulerStatus_1 = require("./KalturaSchedulerStatus");
var KalturaSchedulerConfig_1 = require("./KalturaSchedulerConfig");
var KalturaBatchJob_1 = require("./KalturaBatchJob");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaSchedulerWorker = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaSchedulerWorker, _super);
    function KalturaSchedulerWorker(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.statuses === 'undefined')
            _this.statuses = [];
        if (typeof _this.configs === 'undefined')
            _this.configs = [];
        if (typeof _this.lockedJobs === 'undefined')
            _this.lockedJobs = [];
        return _this;
    }
    KalturaSchedulerWorker.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaSchedulerWorker' },
            id: { type: 'n', readOnly: true },
            configuredId: { type: 'n' },
            schedulerId: { type: 'n' },
            schedulerConfiguredId: { type: 'n' },
            type: { type: 'es', subTypeConstructor: KalturaBatchJobType_1.KalturaBatchJobType, subType: 'KalturaBatchJobType' },
            typeName: { type: 's' },
            name: { type: 's' },
            statuses: { type: 'a', subTypeConstructor: KalturaSchedulerStatus_1.KalturaSchedulerStatus, subType: 'KalturaSchedulerStatus' },
            configs: { type: 'a', subTypeConstructor: KalturaSchedulerConfig_1.KalturaSchedulerConfig, subType: 'KalturaSchedulerConfig' },
            lockedJobs: { type: 'a', subTypeConstructor: KalturaBatchJob_1.KalturaBatchJob, subType: 'KalturaBatchJob' },
            avgWait: { type: 'n' },
            avgWork: { type: 'n' },
            lastStatus: { type: 'n' },
            lastStatusStr: { type: 's' }
        });
        return result;
    };
    return KalturaSchedulerWorker;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaSchedulerWorker = KalturaSchedulerWorker;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaSchedulerWorker', KalturaSchedulerWorker);
