// kaltura-node-typescript-client/api/types/KalturaPlayableEntryCompareAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPlayableEntryCompareAttribute = void 0;
var KalturaPlayableEntryCompareAttribute;
(function (KalturaPlayableEntryCompareAttribute) {
    KalturaPlayableEntryCompareAttribute["accessControlId"] = "accessControlId";
    KalturaPlayableEntryCompareAttribute["createdAt"] = "createdAt";
    KalturaPlayableEntryCompareAttribute["endDate"] = "endDate";
    KalturaPlayableEntryCompareAttribute["lastPlayedAt"] = "lastPlayedAt";
    KalturaPlayableEntryCompareAttribute["moderationCount"] = "moderationCount";
    KalturaPlayableEntryCompareAttribute["moderationStatus"] = "moderationStatus";
    KalturaPlayableEntryCompareAttribute["msDuration"] = "msDuration";
    KalturaPlayableEntryCompareAttribute["partnerId"] = "partnerId";
    KalturaPlayableEntryCompareAttribute["partnerSortValue"] = "partnerSortValue";
    KalturaPlayableEntryCompareAttribute["plays"] = "plays";
    KalturaPlayableEntryCompareAttribute["rank"] = "rank";
    KalturaPlayableEntryCompareAttribute["replacementStatus"] = "replacementStatus";
    KalturaPlayableEntryCompareAttribute["startDate"] = "startDate";
    KalturaPlayableEntryCompareAttribute["status"] = "status";
    KalturaPlayableEntryCompareAttribute["totalRank"] = "totalRank";
    KalturaPlayableEntryCompareAttribute["type"] = "type";
    KalturaPlayableEntryCompareAttribute["updatedAt"] = "updatedAt";
    KalturaPlayableEntryCompareAttribute["views"] = "views";
})(KalturaPlayableEntryCompareAttribute = exports.KalturaPlayableEntryCompareAttribute || (exports.KalturaPlayableEntryCompareAttribute = {}));
