// kaltura-node-typescript-client/api/types/KalturaConvertLiveSegmentJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaConvertLiveSegmentJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaEntryServerNodeType_1 = require("./KalturaEntryServerNodeType");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaConvertLiveSegmentJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaConvertLiveSegmentJobData, _super);
    function KalturaConvertLiveSegmentJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaConvertLiveSegmentJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaConvertLiveSegmentJobData' },
            entryId: { type: 's' },
            assetId: { type: 's' },
            mediaServerIndex: { type: 'es', subTypeConstructor: KalturaEntryServerNodeType_1.KalturaEntryServerNodeType, subType: 'KalturaEntryServerNodeType' },
            fileIndex: { type: 'n' },
            srcFilePath: { type: 's' },
            destFilePath: { type: 's' },
            endTime: { type: 'n' },
            destDataFilePath: { type: 's' }
        });
        return result;
    };
    return KalturaConvertLiveSegmentJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaConvertLiveSegmentJobData = KalturaConvertLiveSegmentJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaConvertLiveSegmentJobData', KalturaConvertLiveSegmentJobData);
