// kaltura-node-typescript-client/api/types/KalturaIndexJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaIndexJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaIndexJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaIndexJobData, _super);
    function KalturaIndexJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaIndexJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaIndexJobData' },
            filter: { type: 'o', subTypeConstructor: KalturaFilter_1.KalturaFilter, subType: 'KalturaFilter' },
            lastIndexId: { type: 'n' },
            lastIndexDepth: { type: 'n' },
            shouldUpdate: { type: 'b' }
        });
        return result;
    };
    return KalturaIndexJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaIndexJobData = KalturaIndexJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaIndexJobData', KalturaIndexJobData);
