// kaltura-node-typescript-client/api/types/KalturaAdCuePointOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAdCuePointOrderBy = void 0;
var KalturaAdCuePointOrderBy;
(function (KalturaAdCuePointOrderBy) {
    KalturaAdCuePointOrderBy["createdAtAsc"] = "+createdAt";
    KalturaAdCuePointOrderBy["createdAtDesc"] = "-createdAt";
    KalturaAdCuePointOrderBy["durationAsc"] = "+duration";
    KalturaAdCuePointOrderBy["durationDesc"] = "-duration";
    KalturaAdCuePointOrderBy["endTimeAsc"] = "+endTime";
    KalturaAdCuePointOrderBy["endTimeDesc"] = "-endTime";
    KalturaAdCuePointOrderBy["intIdAsc"] = "+intId";
    KalturaAdCuePointOrderBy["intIdDesc"] = "-intId";
    KalturaAdCuePointOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaAdCuePointOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaAdCuePointOrderBy["startTimeAsc"] = "+startTime";
    KalturaAdCuePointOrderBy["startTimeDesc"] = "-startTime";
    KalturaAdCuePointOrderBy["triggeredAtAsc"] = "+triggeredAt";
    KalturaAdCuePointOrderBy["triggeredAtDesc"] = "-triggeredAt";
    KalturaAdCuePointOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaAdCuePointOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaAdCuePointOrderBy = exports.KalturaAdCuePointOrderBy || (exports.KalturaAdCuePointOrderBy = {}));
