// kaltura-node-typescript-client/api/types/KalturaUserEntryFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUserEntryFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaUserEntryBaseFilter_1 = require("./KalturaUserEntryBaseFilter");
var KalturaUserEntryFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUserEntryFilter, _super);
    function KalturaUserEntryFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaUserEntryFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUserEntryFilter' },
            userIdEqualCurrent: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            isAnonymous: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            privacyContextEqual: { type: 's' },
            privacyContextIn: { type: 's' },
            partnerId: { type: 'n' }
        });
        return result;
    };
    return KalturaUserEntryFilter;
}(KalturaUserEntryBaseFilter_1.KalturaUserEntryBaseFilter));
exports.KalturaUserEntryFilter = KalturaUserEntryFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUserEntryFilter', KalturaUserEntryFilter);
