// kaltura-node-typescript-client/api/types/KalturaConvartableJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaConvartableJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaSourceFileSyncDescriptor_1 = require("./KalturaSourceFileSyncDescriptor");
var KalturaFlavorParamsOutput_1 = require("./KalturaFlavorParamsOutput");
var KalturaKeyValue_1 = require("./KalturaKeyValue");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaConvartableJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaConvartableJobData, _super);
    function KalturaConvartableJobData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.srcFileSyncs === 'undefined')
            _this.srcFileSyncs = [];
        if (typeof _this.pluginData === 'undefined')
            _this.pluginData = [];
        return _this;
    }
    KalturaConvartableJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaConvartableJobData' },
            srcFileSyncLocalPath: { type: 's' },
            actualSrcFileSyncLocalPath: { type: 's' },
            srcFileSyncRemoteUrl: { type: 's' },
            srcFileSyncs: { type: 'a', subTypeConstructor: KalturaSourceFileSyncDescriptor_1.KalturaSourceFileSyncDescriptor, subType: 'KalturaSourceFileSyncDescriptor' },
            engineVersion: { type: 'n' },
            flavorParamsOutputId: { type: 'n' },
            flavorParamsOutput: { type: 'o', subTypeConstructor: KalturaFlavorParamsOutput_1.KalturaFlavorParamsOutput, subType: 'KalturaFlavorParamsOutput' },
            mediaInfoId: { type: 'n' },
            currentOperationSet: { type: 'n' },
            currentOperationIndex: { type: 'n' },
            pluginData: { type: 'a', subTypeConstructor: KalturaKeyValue_1.KalturaKeyValue, subType: 'KalturaKeyValue' }
        });
        return result;
    };
    return KalturaConvartableJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaConvartableJobData = KalturaConvartableJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaConvartableJobData', KalturaConvartableJobData);
