// kaltura-node-typescript-client/api/types/KalturaAnnotation.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAnnotation = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaCuePoint_1 = require("./KalturaCuePoint");
var KalturaAnnotation = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAnnotation, _super);
    function KalturaAnnotation(data) {
        return _super.call(this, data) || this;
    }
    KalturaAnnotation.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAnnotation' },
            parentId: { type: 's' },
            text: { type: 's' },
            endTime: { type: 'n' },
            duration: { type: 'n', readOnly: true },
            depth: { type: 'n', readOnly: true },
            childrenCount: { type: 'n', readOnly: true },
            directChildrenCount: { type: 'n', readOnly: true },
            isPublic: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            searchableOnEntry: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' }
        });
        return result;
    };
    return KalturaAnnotation;
}(KalturaCuePoint_1.KalturaCuePoint));
exports.KalturaAnnotation = KalturaAnnotation;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAnnotation', KalturaAnnotation);
