// kaltura-node-typescript-client/api/types/KalturaBatchJobOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBatchJobOrderBy = void 0;
var KalturaBatchJobOrderBy;
(function (KalturaBatchJobOrderBy) {
    KalturaBatchJobOrderBy["createdAtAsc"] = "+createdAt";
    KalturaBatchJobOrderBy["createdAtDesc"] = "-createdAt";
    KalturaBatchJobOrderBy["estimatedEffortAsc"] = "+estimatedEffort";
    KalturaBatchJobOrderBy["estimatedEffortDesc"] = "-estimatedEffort";
    KalturaBatchJobOrderBy["executionAttemptsAsc"] = "+executionAttempts";
    KalturaBatchJobOrderBy["executionAttemptsDesc"] = "-executionAttempts";
    KalturaBatchJobOrderBy["finishTimeAsc"] = "+finishTime";
    KalturaBatchJobOrderBy["finishTimeDesc"] = "-finishTime";
    KalturaBatchJobOrderBy["lockVersionAsc"] = "+lockVersion";
    KalturaBatchJobOrderBy["lockVersionDesc"] = "-lockVersion";
    KalturaBatchJobOrderBy["priorityAsc"] = "+priority";
    KalturaBatchJobOrderBy["priorityDesc"] = "-priority";
    KalturaBatchJobOrderBy["queueTimeAsc"] = "+queueTime";
    KalturaBatchJobOrderBy["queueTimeDesc"] = "-queueTime";
    KalturaBatchJobOrderBy["statusAsc"] = "+status";
    KalturaBatchJobOrderBy["statusDesc"] = "-status";
    KalturaBatchJobOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaBatchJobOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaBatchJobOrderBy = exports.KalturaBatchJobOrderBy || (exports.KalturaBatchJobOrderBy = {}));
