// kaltura-node-typescript-client/api/types/KalturaFlavorParamsOutputBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFlavorParamsOutputBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaFlavorParamsFilter_1 = require("./KalturaFlavorParamsFilter");
var KalturaFlavorParamsOutputBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaFlavorParamsOutputBaseFilter, _super);
    function KalturaFlavorParamsOutputBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaFlavorParamsOutputBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaFlavorParamsOutputBaseFilter' },
            flavorParamsIdEqual: { type: 'n' },
            flavorParamsVersionEqual: { type: 's' },
            flavorAssetIdEqual: { type: 's' },
            flavorAssetVersionEqual: { type: 's' }
        });
        return result;
    };
    return KalturaFlavorParamsOutputBaseFilter;
}(KalturaFlavorParamsFilter_1.KalturaFlavorParamsFilter));
exports.KalturaFlavorParamsOutputBaseFilter = KalturaFlavorParamsOutputBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaFlavorParamsOutputBaseFilter', KalturaFlavorParamsOutputBaseFilter);
