// kaltura-node-typescript-client/api/types/KalturaPermission.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPermission = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaPermissionType_1 = require("./KalturaPermissionType");
var KalturaPermissionStatus_1 = require("./KalturaPermissionStatus");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaPermission = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPermission, _super);
    function KalturaPermission(data) {
        return _super.call(this, data) || this;
    }
    KalturaPermission.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPermission' },
            id: { type: 'n', readOnly: true },
            type: { type: 'en', readOnly: true, subTypeConstructor: KalturaPermissionType_1.KalturaPermissionType, subType: 'KalturaPermissionType' },
            name: { type: 's' },
            friendlyName: { type: 's' },
            description: { type: 's' },
            status: { type: 'en', subTypeConstructor: KalturaPermissionStatus_1.KalturaPermissionStatus, subType: 'KalturaPermissionStatus' },
            partnerId: { type: 'n', readOnly: true },
            dependsOnPermissionNames: { type: 's' },
            tags: { type: 's' },
            permissionItemsIds: { type: 's' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            partnerGroup: { type: 's' }
        });
        return result;
    };
    return KalturaPermission;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaPermission = KalturaPermission;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPermission', KalturaPermission);
