// kaltura-node-typescript-client/api/types/KalturaCuePoint.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCuePoint = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCuePointType_1 = require("./KalturaCuePointType");
var KalturaCuePointStatus_1 = require("./KalturaCuePointStatus");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaCuePoint = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCuePoint, _super);
    function KalturaCuePoint(data) {
        return _super.call(this, data) || this;
    }
    KalturaCuePoint.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCuePoint' },
            id: { type: 's', readOnly: true },
            intId: { type: 'n', readOnly: true },
            cuePointType: { type: 'es', readOnly: true, subTypeConstructor: KalturaCuePointType_1.KalturaCuePointType, subType: 'KalturaCuePointType' },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaCuePointStatus_1.KalturaCuePointStatus, subType: 'KalturaCuePointStatus' },
            entryId: { type: 's' },
            partnerId: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            triggeredAt: { type: 'd' },
            tags: { type: 's' },
            startTime: { type: 'n' },
            userId: { type: 's' },
            partnerData: { type: 's' },
            partnerSortValue: { type: 'n' },
            forceStop: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            thumbOffset: { type: 'n' },
            systemName: { type: 's' },
            isMomentary: { type: 'b', readOnly: true },
            copiedFrom: { type: 's', readOnly: true }
        });
        return result;
    };
    return KalturaCuePoint;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaCuePoint = KalturaCuePoint;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCuePoint', KalturaCuePoint);
