// kaltura-node-typescript-client/api/types/KalturaThumbAsset.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaThumbAsset = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaThumbAssetStatus_1 = require("./KalturaThumbAssetStatus");
var KalturaAsset_1 = require("./KalturaAsset");
var KalturaThumbAsset = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaThumbAsset, _super);
    function KalturaThumbAsset(data) {
        return _super.call(this, data) || this;
    }
    KalturaThumbAsset.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaThumbAsset' },
            thumbParamsId: { type: 'n' },
            width: { type: 'n', readOnly: true },
            height: { type: 'n', readOnly: true },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaThumbAssetStatus_1.KalturaThumbAssetStatus, subType: 'KalturaThumbAssetStatus' }
        });
        return result;
    };
    return KalturaThumbAsset;
}(KalturaAsset_1.KalturaAsset));
exports.KalturaThumbAsset = KalturaThumbAsset;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaThumbAsset', KalturaThumbAsset);
