// kaltura-node-typescript-client/api/types/KalturaTubeMogulSyndicationFeedOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaTubeMogulSyndicationFeedOrderBy = void 0;
var KalturaTubeMogulSyndicationFeedOrderBy;
(function (KalturaTubeMogulSyndicationFeedOrderBy) {
    KalturaTubeMogulSyndicationFeedOrderBy["createdAtAsc"] = "+createdAt";
    KalturaTubeMogulSyndicationFeedOrderBy["createdAtDesc"] = "-createdAt";
    KalturaTubeMogulSyndicationFeedOrderBy["nameAsc"] = "+name";
    KalturaTubeMogulSyndicationFeedOrderBy["nameDesc"] = "-name";
    KalturaTubeMogulSyndicationFeedOrderBy["playlistIdAsc"] = "+playlistId";
    KalturaTubeMogulSyndicationFeedOrderBy["playlistIdDesc"] = "-playlistId";
    KalturaTubeMogulSyndicationFeedOrderBy["typeAsc"] = "+type";
    KalturaTubeMogulSyndicationFeedOrderBy["typeDesc"] = "-type";
    KalturaTubeMogulSyndicationFeedOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaTubeMogulSyndicationFeedOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaTubeMogulSyndicationFeedOrderBy = exports.KalturaTubeMogulSyndicationFeedOrderBy || (exports.KalturaTubeMogulSyndicationFeedOrderBy = {}));
