// kaltura-node-typescript-client/api/types/KalturaPlayReadyLicenseDetails.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPlayReadyLicenseDetails = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaPlayReadyPolicy_1 = require("./KalturaPlayReadyPolicy");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaPlayReadyLicenseDetails = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPlayReadyLicenseDetails, _super);
    function KalturaPlayReadyLicenseDetails(data) {
        return _super.call(this, data) || this;
    }
    KalturaPlayReadyLicenseDetails.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPlayReadyLicenseDetails' },
            policy: { type: 'o', subTypeConstructor: KalturaPlayReadyPolicy_1.KalturaPlayReadyPolicy, subType: 'KalturaPlayReadyPolicy' },
            beginDate: { type: 'n' },
            expirationDate: { type: 'n' },
            removalDate: { type: 'n' }
        });
        return result;
    };
    return KalturaPlayReadyLicenseDetails;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaPlayReadyLicenseDetails = KalturaPlayReadyLicenseDetails;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPlayReadyLicenseDetails', KalturaPlayReadyLicenseDetails);
