// kaltura-node-typescript-client/api/types/KalturaDropFolderFileStatus.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDropFolderFileStatus = void 0;
var KalturaDropFolderFileStatus;
(function (KalturaDropFolderFileStatus) {
    KalturaDropFolderFileStatus[KalturaDropFolderFileStatus["deleted"] = 6] = "deleted";
    KalturaDropFolderFileStatus[KalturaDropFolderFileStatus["detected"] = 15] = "detected";
    KalturaDropFolderFileStatus[KalturaDropFolderFileStatus["downloading"] = 11] = "downloading";
    KalturaDropFolderFileStatus[KalturaDropFolderFileStatus["errorDeleting"] = 10] = "errorDeleting";
    KalturaDropFolderFileStatus[KalturaDropFolderFileStatus["errorDownloading"] = 12] = "errorDownloading";
    KalturaDropFolderFileStatus[KalturaDropFolderFileStatus["errorHandling"] = 9] = "errorHandling";
    KalturaDropFolderFileStatus[KalturaDropFolderFileStatus["handled"] = 4] = "handled";
    KalturaDropFolderFileStatus[KalturaDropFolderFileStatus["ignore"] = 5] = "ignore";
    KalturaDropFolderFileStatus[KalturaDropFolderFileStatus["noMatch"] = 8] = "noMatch";
    KalturaDropFolderFileStatus[KalturaDropFolderFileStatus["parsed"] = 14] = "parsed";
    KalturaDropFolderFileStatus[KalturaDropFolderFileStatus["pending"] = 2] = "pending";
    KalturaDropFolderFileStatus[KalturaDropFolderFileStatus["processing"] = 13] = "processing";
    KalturaDropFolderFileStatus[KalturaDropFolderFileStatus["purged"] = 7] = "purged";
    KalturaDropFolderFileStatus[KalturaDropFolderFileStatus["uploading"] = 1] = "uploading";
    KalturaDropFolderFileStatus[KalturaDropFolderFileStatus["waiting"] = 3] = "waiting";
})(KalturaDropFolderFileStatus = exports.KalturaDropFolderFileStatus || (exports.KalturaDropFolderFileStatus = {}));
