// kaltura-node-typescript-client/api/types/KalturaBaseInteractivity.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBaseInteractivity = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaBaseInteractivity = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBaseInteractivity, _super);
    function KalturaBaseInteractivity(data) {
        return _super.call(this, data) || this;
    }
    KalturaBaseInteractivity.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBaseInteractivity' },
            data: { type: 's' },
            version: { type: 'n', readOnly: true },
            entryId: { type: 's', readOnly: true },
            updatedAt: { type: 'd', readOnly: true }
        });
        return result;
    };
    return KalturaBaseInteractivity;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaBaseInteractivity = KalturaBaseInteractivity;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBaseInteractivity', KalturaBaseInteractivity);
