// kaltura-node-typescript-client/api/types/KalturaHttpNotification.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaHttpNotification = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaEventNotificationEventObjectType_1 = require("./KalturaEventNotificationEventObjectType");
var KalturaEventNotificationEventType_1 = require("./KalturaEventNotificationEventType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaHttpNotification = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaHttpNotification, _super);
    function KalturaHttpNotification(data) {
        return _super.call(this, data) || this;
    }
    KalturaHttpNotification.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaHttpNotification' },
            object: { type: 'o', subTypeConstructor: kaltura_object_base_1.KalturaObjectBase, subType: 'KalturaObjectBase' },
            eventObjectType: { type: 'es', subTypeConstructor: KalturaEventNotificationEventObjectType_1.KalturaEventNotificationEventObjectType, subType: 'KalturaEventNotificationEventObjectType' },
            eventNotificationJobId: { type: 'n' },
            templateId: { type: 'n' },
            templateName: { type: 's' },
            templateSystemName: { type: 's' },
            eventType: { type: 'es', subTypeConstructor: KalturaEventNotificationEventType_1.KalturaEventNotificationEventType, subType: 'KalturaEventNotificationEventType' }
        });
        return result;
    };
    return KalturaHttpNotification;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaHttpNotification = KalturaHttpNotification;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaHttpNotification', KalturaHttpNotification);
