// kaltura-node-typescript-client/api/types/KalturaDropFolderType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDropFolderType = void 0;
var KalturaDropFolderType;
(function (KalturaDropFolderType) {
    KalturaDropFolderType["apFeed"] = "ApFeedDropFolder.AP_FEED";
    KalturaDropFolderType["feed"] = "FeedDropFolder.FEED";
    KalturaDropFolderType["ftp"] = "2";
    KalturaDropFolderType["local"] = "1";
    KalturaDropFolderType["msTeams"] = "MicrosoftTeamsDropFolder.MS_TEAMS";
    KalturaDropFolderType["s3"] = "6";
    KalturaDropFolderType["s3dropfolder"] = "S3DropFolder.S3DROPFOLDER";
    KalturaDropFolderType["scp"] = "3";
    KalturaDropFolderType["sftp"] = "4";
    KalturaDropFolderType["trRdsCompany"] = "TrRdsSyncDropFolder.TR_RDS_COMPANY";
    KalturaDropFolderType["trRdsTmcterm"] = "TrRdsSyncDropFolder.TR_RDS_TMCTERM";
    KalturaDropFolderType["webex"] = "WebexDropFolder.WEBEX";
    KalturaDropFolderType["webexApi"] = "WebexAPIDropFolder.WEBEX_API";
    KalturaDropFolderType["zoom"] = "ZoomDropFolder.ZOOM";
})(KalturaDropFolderType = exports.KalturaDropFolderType || (exports.KalturaDropFolderType = {}));
