// kaltura-node-typescript-client/api/types/KalturaAssetParams.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAssetParams = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaString_1 = require("./KalturaString");
var KalturaMediaParserType_1 = require("./KalturaMediaParserType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaAssetParams = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAssetParams, _super);
    function KalturaAssetParams(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.requiredPermissions === 'undefined')
            _this.requiredPermissions = [];
        return _this;
    }
    KalturaAssetParams.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAssetParams' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n' },
            name: { type: 's' },
            systemName: { type: 's' },
            description: { type: 's' },
            createdAt: { type: 'd', readOnly: true },
            isSystemDefault: { type: 'en', readOnly: true, subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            tags: { type: 's' },
            requiredPermissions: { type: 'a', subTypeConstructor: KalturaString_1.KalturaString, subType: 'KalturaString' },
            sourceRemoteStorageProfileId: { type: 'n' },
            remoteStorageProfileIds: { type: 'n' },
            mediaParserType: { type: 'es', subTypeConstructor: KalturaMediaParserType_1.KalturaMediaParserType, subType: 'KalturaMediaParserType' },
            sourceAssetParamsIds: { type: 's' }
        });
        return result;
    };
    return KalturaAssetParams;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaAssetParams = KalturaAssetParams;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAssetParams', KalturaAssetParams);
