// kaltura-node-typescript-client/api/types/KalturaGenericDistributionProvider.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaGenericDistributionProvider = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaGenericDistributionProviderStatus_1 = require("./KalturaGenericDistributionProviderStatus");
var KalturaDistributionThumbDimensions_1 = require("./KalturaDistributionThumbDimensions");
var KalturaDistributionProvider_1 = require("./KalturaDistributionProvider");
var KalturaGenericDistributionProvider = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaGenericDistributionProvider, _super);
    function KalturaGenericDistributionProvider(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.optionalThumbDimensions === 'undefined')
            _this.optionalThumbDimensions = [];
        if (typeof _this.requiredThumbDimensions === 'undefined')
            _this.requiredThumbDimensions = [];
        return _this;
    }
    KalturaGenericDistributionProvider.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaGenericDistributionProvider' },
            id: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            isDefault: { type: 'b' },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaGenericDistributionProviderStatus_1.KalturaGenericDistributionProviderStatus, subType: 'KalturaGenericDistributionProviderStatus' },
            optionalFlavorParamsIds: { type: 's' },
            requiredFlavorParamsIds: { type: 's' },
            optionalThumbDimensions: { type: 'a', subTypeConstructor: KalturaDistributionThumbDimensions_1.KalturaDistributionThumbDimensions, subType: 'KalturaDistributionThumbDimensions' },
            requiredThumbDimensions: { type: 'a', subTypeConstructor: KalturaDistributionThumbDimensions_1.KalturaDistributionThumbDimensions, subType: 'KalturaDistributionThumbDimensions' },
            editableFields: { type: 's' },
            mandatoryFields: { type: 's' }
        });
        return result;
    };
    return KalturaGenericDistributionProvider;
}(KalturaDistributionProvider_1.KalturaDistributionProvider));
exports.KalturaGenericDistributionProvider = KalturaGenericDistributionProvider;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaGenericDistributionProvider', KalturaGenericDistributionProvider);
