// kaltura-node-typescript-client/api/types/KalturaPlaybackProtocol.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPlaybackProtocol = void 0;
var KalturaPlaybackProtocol;
(function (KalturaPlaybackProtocol) {
    KalturaPlaybackProtocol["akamaiHd"] = "hdnetwork";
    KalturaPlaybackProtocol["akamaiHds"] = "hdnetworkmanifest";
    KalturaPlaybackProtocol["appleHttp"] = "applehttp";
    KalturaPlaybackProtocol["appleHttpToMc"] = "applehttp_to_mc";
    KalturaPlaybackProtocol["auto"] = "auto";
    KalturaPlaybackProtocol["download"] = "download";
    KalturaPlaybackProtocol["hds"] = "hds";
    KalturaPlaybackProtocol["hls"] = "hls";
    KalturaPlaybackProtocol["http"] = "http";
    KalturaPlaybackProtocol["mpegDash"] = "mpegdash";
    KalturaPlaybackProtocol["multicastSl"] = "multicast_silverlight";
    KalturaPlaybackProtocol["rtmp"] = "rtmp";
    KalturaPlaybackProtocol["rtsp"] = "rtsp";
    KalturaPlaybackProtocol["silverLight"] = "sl";
    KalturaPlaybackProtocol["url"] = "url";
})(KalturaPlaybackProtocol = exports.KalturaPlaybackProtocol || (exports.KalturaPlaybackProtocol = {}));
