// kaltura-node-typescript-client/api/types/KalturaAsset.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAsset = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaAsset = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAsset, _super);
    function KalturaAsset(data) {
        return _super.call(this, data) || this;
    }
    KalturaAsset.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAsset' },
            id: { type: 's', readOnly: true },
            entryId: { type: 's', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            version: { type: 'n', readOnly: true },
            size: { type: 'n', readOnly: true },
            tags: { type: 's' },
            fileExt: { type: 's' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            deletedAt: { type: 'd', readOnly: true },
            description: { type: 's', readOnly: true },
            partnerData: { type: 's' },
            partnerDescription: { type: 's' },
            actualSourceAssetParamsIds: { type: 's' },
            sizeInBytes: { type: 'n', readOnly: true }
        });
        return result;
    };
    return KalturaAsset;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaAsset = KalturaAsset;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAsset', KalturaAsset);
