// kaltura-node-typescript-client/api/types/KalturaPlayableEntryOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPlayableEntryOrderBy = void 0;
var KalturaPlayableEntryOrderBy;
(function (KalturaPlayableEntryOrderBy) {
    KalturaPlayableEntryOrderBy["createdAtAsc"] = "+createdAt";
    KalturaPlayableEntryOrderBy["createdAtDesc"] = "-createdAt";
    KalturaPlayableEntryOrderBy["durationAsc"] = "+duration";
    KalturaPlayableEntryOrderBy["durationDesc"] = "-duration";
    KalturaPlayableEntryOrderBy["endDateAsc"] = "+endDate";
    KalturaPlayableEntryOrderBy["endDateDesc"] = "-endDate";
    KalturaPlayableEntryOrderBy["lastPlayedAtAsc"] = "+lastPlayedAt";
    KalturaPlayableEntryOrderBy["lastPlayedAtDesc"] = "-lastPlayedAt";
    KalturaPlayableEntryOrderBy["moderationCountAsc"] = "+moderationCount";
    KalturaPlayableEntryOrderBy["moderationCountDesc"] = "-moderationCount";
    KalturaPlayableEntryOrderBy["nameAsc"] = "+name";
    KalturaPlayableEntryOrderBy["nameDesc"] = "-name";
    KalturaPlayableEntryOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaPlayableEntryOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaPlayableEntryOrderBy["playsAsc"] = "+plays";
    KalturaPlayableEntryOrderBy["playsDesc"] = "-plays";
    KalturaPlayableEntryOrderBy["rankAsc"] = "+rank";
    KalturaPlayableEntryOrderBy["rankDesc"] = "-rank";
    KalturaPlayableEntryOrderBy["recentAsc"] = "+recent";
    KalturaPlayableEntryOrderBy["recentDesc"] = "-recent";
    KalturaPlayableEntryOrderBy["startDateAsc"] = "+startDate";
    KalturaPlayableEntryOrderBy["startDateDesc"] = "-startDate";
    KalturaPlayableEntryOrderBy["totalRankAsc"] = "+totalRank";
    KalturaPlayableEntryOrderBy["totalRankDesc"] = "-totalRank";
    KalturaPlayableEntryOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaPlayableEntryOrderBy["updatedAtDesc"] = "-updatedAt";
    KalturaPlayableEntryOrderBy["viewsAsc"] = "+views";
    KalturaPlayableEntryOrderBy["viewsDesc"] = "-views";
    KalturaPlayableEntryOrderBy["weightAsc"] = "+weight";
    KalturaPlayableEntryOrderBy["weightDesc"] = "-weight";
})(KalturaPlayableEntryOrderBy = exports.KalturaPlayableEntryOrderBy || (exports.KalturaPlayableEntryOrderBy = {}));
