// kaltura-node-typescript-client/api/types/KalturaUrlTokenizerAkamaiRtmp.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUrlTokenizerAkamaiRtmp = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUrlTokenizer_1 = require("./KalturaUrlTokenizer");
var KalturaUrlTokenizerAkamaiRtmp = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUrlTokenizerAkamaiRtmp, _super);
    function KalturaUrlTokenizerAkamaiRtmp(data) {
        return _super.call(this, data) || this;
    }
    KalturaUrlTokenizerAkamaiRtmp.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUrlTokenizerAkamaiRtmp' },
            profile: { type: 's' },
            type: { type: 's' },
            aifp: { type: 's' },
            usePrefix: { type: 'b' }
        });
        return result;
    };
    return KalturaUrlTokenizerAkamaiRtmp;
}(KalturaUrlTokenizer_1.KalturaUrlTokenizer));
exports.KalturaUrlTokenizerAkamaiRtmp = KalturaUrlTokenizerAkamaiRtmp;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUrlTokenizerAkamaiRtmp', KalturaUrlTokenizerAkamaiRtmp);
