// kaltura-node-typescript-client/api/types/KalturaReportExportParams.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaReportExportParams = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaReportExportItem_1 = require("./KalturaReportExportItem");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaReportExportParams = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaReportExportParams, _super);
    function KalturaReportExportParams(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.reportItems === 'undefined')
            _this.reportItems = [];
        return _this;
    }
    KalturaReportExportParams.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaReportExportParams' },
            recipientEmail: { type: 's' },
            timeZoneOffset: { type: 'n' },
            reportItems: { type: 'a', subTypeConstructor: KalturaReportExportItem_1.KalturaReportExportItem, subType: 'KalturaReportExportItem' },
            reportsItemsGroup: { type: 's' },
            baseUrl: { type: 's' }
        });
        return result;
    };
    return KalturaReportExportParams;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaReportExportParams = KalturaReportExportParams;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaReportExportParams', KalturaReportExportParams);
