// kaltura-node-typescript-client/api/types/KalturaBaseSyndicationFeedOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBaseSyndicationFeedOrderBy = void 0;
var KalturaBaseSyndicationFeedOrderBy;
(function (KalturaBaseSyndicationFeedOrderBy) {
    KalturaBaseSyndicationFeedOrderBy["createdAtAsc"] = "+createdAt";
    KalturaBaseSyndicationFeedOrderBy["createdAtDesc"] = "-createdAt";
    KalturaBaseSyndicationFeedOrderBy["nameAsc"] = "+name";
    KalturaBaseSyndicationFeedOrderBy["nameDesc"] = "-name";
    KalturaBaseSyndicationFeedOrderBy["playlistIdAsc"] = "+playlistId";
    KalturaBaseSyndicationFeedOrderBy["playlistIdDesc"] = "-playlistId";
    KalturaBaseSyndicationFeedOrderBy["typeAsc"] = "+type";
    KalturaBaseSyndicationFeedOrderBy["typeDesc"] = "-type";
    KalturaBaseSyndicationFeedOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaBaseSyndicationFeedOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaBaseSyndicationFeedOrderBy = exports.KalturaBaseSyndicationFeedOrderBy || (exports.KalturaBaseSyndicationFeedOrderBy = {}));
