// kaltura-node-typescript-client/api/types/KalturaClipDescription.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaClipDescription = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaClipDescription = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaClipDescription, _super);
    function KalturaClipDescription(data) {
        return _super.call(this, data) || this;
    }
    KalturaClipDescription.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaClipDescription' },
            sourceEntryId: { type: 's' },
            startTime: { type: 'n' },
            duration: { type: 'n' },
            offsetInDestination: { type: 'n' }
        });
        return result;
    };
    return KalturaClipDescription;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaClipDescription = KalturaClipDescription;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaClipDescription', KalturaClipDescription);
