// kaltura-node-typescript-client/api/types/KalturaMediaEntryBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMediaEntryBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaMediaType_1 = require("./KalturaMediaType");
var KalturaSourceType_1 = require("./KalturaSourceType");
var KalturaPlayableEntryFilter_1 = require("./KalturaPlayableEntryFilter");
var KalturaMediaEntryBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMediaEntryBaseFilter, _super);
    function KalturaMediaEntryBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaMediaEntryBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMediaEntryBaseFilter' },
            mediaTypeEqual: { type: 'en', subTypeConstructor: KalturaMediaType_1.KalturaMediaType, subType: 'KalturaMediaType' },
            mediaTypeIn: { type: 's' },
            sourceTypeEqual: { type: 'es', subTypeConstructor: KalturaSourceType_1.KalturaSourceType, subType: 'KalturaSourceType' },
            sourceTypeNotEqual: { type: 'es', subTypeConstructor: KalturaSourceType_1.KalturaSourceType, subType: 'KalturaSourceType' },
            sourceTypeIn: { type: 's' },
            sourceTypeNotIn: { type: 's' },
            mediaDateGreaterThanOrEqual: { type: 'd' },
            mediaDateLessThanOrEqual: { type: 'd' },
            flavorParamsIdsMatchOr: { type: 's' },
            flavorParamsIdsMatchAnd: { type: 's' }
        });
        return result;
    };
    return KalturaMediaEntryBaseFilter;
}(KalturaPlayableEntryFilter_1.KalturaPlayableEntryFilter));
exports.KalturaMediaEntryBaseFilter = KalturaMediaEntryBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMediaEntryBaseFilter', KalturaMediaEntryBaseFilter);
