// kaltura-node-typescript-client/api/types/KalturaUserEntry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUserEntry = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUserEntryStatus_1 = require("./KalturaUserEntryStatus");
var KalturaUserEntryType_1 = require("./KalturaUserEntryType");
var KalturaUserEntryExtendedStatus_1 = require("./KalturaUserEntryExtendedStatus");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaUserEntry = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUserEntry, _super);
    function KalturaUserEntry(data) {
        return _super.call(this, data) || this;
    }
    KalturaUserEntry.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUserEntry' },
            id: { type: 'n', readOnly: true },
            entryId: { type: 's' },
            userId: { type: 's' },
            partnerId: { type: 'n', readOnly: true },
            status: { type: 'es', readOnly: true, subTypeConstructor: KalturaUserEntryStatus_1.KalturaUserEntryStatus, subType: 'KalturaUserEntryStatus' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            type: { type: 'es', readOnly: true, subTypeConstructor: KalturaUserEntryType_1.KalturaUserEntryType, subType: 'KalturaUserEntryType' },
            extendedStatus: { type: 'es', subTypeConstructor: KalturaUserEntryExtendedStatus_1.KalturaUserEntryExtendedStatus, subType: 'KalturaUserEntryExtendedStatus' }
        });
        return result;
    };
    return KalturaUserEntry;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaUserEntry = KalturaUserEntry;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUserEntry', KalturaUserEntry);
