// kaltura-node-typescript-client/api/types/KalturaCaptionPlaybackPluginData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCaptionPlaybackPluginData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaCaptionPlaybackPluginData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCaptionPlaybackPluginData, _super);
    function KalturaCaptionPlaybackPluginData(data) {
        return _super.call(this, data) || this;
    }
    KalturaCaptionPlaybackPluginData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCaptionPlaybackPluginData' },
            label: { type: 's' },
            format: { type: 's' },
            language: { type: 's' },
            webVttUrl: { type: 's' },
            url: { type: 's' },
            isDefault: { type: 'b' },
            languageCode: { type: 's' }
        });
        return result;
    };
    return KalturaCaptionPlaybackPluginData;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaCaptionPlaybackPluginData = KalturaCaptionPlaybackPluginData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCaptionPlaybackPluginData', KalturaCaptionPlaybackPluginData);
