// kaltura-node-typescript-client/api/types/KalturaEventNotificationTemplateBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEventNotificationTemplateBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaEventNotificationTemplateType_1 = require("./KalturaEventNotificationTemplateType");
var KalturaEventNotificationTemplateStatus_1 = require("./KalturaEventNotificationTemplateStatus");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaEventNotificationTemplateBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaEventNotificationTemplateBaseFilter, _super);
    function KalturaEventNotificationTemplateBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaEventNotificationTemplateBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaEventNotificationTemplateBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            systemNameEqual: { type: 's' },
            systemNameIn: { type: 's' },
            typeEqual: { type: 'es', subTypeConstructor: KalturaEventNotificationTemplateType_1.KalturaEventNotificationTemplateType, subType: 'KalturaEventNotificationTemplateType' },
            typeIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaEventNotificationTemplateStatus_1.KalturaEventNotificationTemplateStatus, subType: 'KalturaEventNotificationTemplateStatus' },
            statusIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' }
        });
        return result;
    };
    return KalturaEventNotificationTemplateBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaEventNotificationTemplateBaseFilter = KalturaEventNotificationTemplateBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaEventNotificationTemplateBaseFilter', KalturaEventNotificationTemplateBaseFilter);
