// kaltura-node-typescript-client/api/types/KalturaExternalMediaEntryCompareAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaExternalMediaEntryCompareAttribute = void 0;
var KalturaExternalMediaEntryCompareAttribute;
(function (KalturaExternalMediaEntryCompareAttribute) {
    KalturaExternalMediaEntryCompareAttribute["accessControlId"] = "accessControlId";
    KalturaExternalMediaEntryCompareAttribute["createdAt"] = "createdAt";
    KalturaExternalMediaEntryCompareAttribute["endDate"] = "endDate";
    KalturaExternalMediaEntryCompareAttribute["lastPlayedAt"] = "lastPlayedAt";
    KalturaExternalMediaEntryCompareAttribute["mediaDate"] = "mediaDate";
    KalturaExternalMediaEntryCompareAttribute["mediaType"] = "mediaType";
    KalturaExternalMediaEntryCompareAttribute["moderationCount"] = "moderationCount";
    KalturaExternalMediaEntryCompareAttribute["moderationStatus"] = "moderationStatus";
    KalturaExternalMediaEntryCompareAttribute["msDuration"] = "msDuration";
    KalturaExternalMediaEntryCompareAttribute["partnerId"] = "partnerId";
    KalturaExternalMediaEntryCompareAttribute["partnerSortValue"] = "partnerSortValue";
    KalturaExternalMediaEntryCompareAttribute["plays"] = "plays";
    KalturaExternalMediaEntryCompareAttribute["rank"] = "rank";
    KalturaExternalMediaEntryCompareAttribute["replacementStatus"] = "replacementStatus";
    KalturaExternalMediaEntryCompareAttribute["startDate"] = "startDate";
    KalturaExternalMediaEntryCompareAttribute["status"] = "status";
    KalturaExternalMediaEntryCompareAttribute["totalRank"] = "totalRank";
    KalturaExternalMediaEntryCompareAttribute["type"] = "type";
    KalturaExternalMediaEntryCompareAttribute["updatedAt"] = "updatedAt";
    KalturaExternalMediaEntryCompareAttribute["views"] = "views";
})(KalturaExternalMediaEntryCompareAttribute = exports.KalturaExternalMediaEntryCompareAttribute || (exports.KalturaExternalMediaEntryCompareAttribute = {}));
