// kaltura-node-typescript-client/api/types/KalturaDeliveryProfileType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDeliveryProfileType = void 0;
var KalturaDeliveryProfileType;
(function (KalturaDeliveryProfileType) {
    KalturaDeliveryProfileType["akamaiHd"] = "12";
    KalturaDeliveryProfileType["akamaiHds"] = "13";
    KalturaDeliveryProfileType["akamaiHlsDirect"] = "10";
    KalturaDeliveryProfileType["akamaiHlsManifest"] = "11";
    KalturaDeliveryProfileType["akamaiHttp"] = "14";
    KalturaDeliveryProfileType["akamaiRtmp"] = "15";
    KalturaDeliveryProfileType["akamaiRtsp"] = "16";
    KalturaDeliveryProfileType["akamaiSs"] = "17";
    KalturaDeliveryProfileType["appleHttp"] = "1";
    KalturaDeliveryProfileType["edgeCastHttp"] = "edgeCast.EDGE_CAST_HTTP";
    KalturaDeliveryProfileType["edgeCastRtmp"] = "edgeCast.EDGE_CAST_RTMP";
    KalturaDeliveryProfileType["forensicWatermarkAppleHttp"] = "forensicWatermark.FORENSIC_WATERMARK_APPLE_HTTP";
    KalturaDeliveryProfileType["forensicWatermarkDash"] = "forensicWatermark.FORENSIC_WATERMARK_DASH";
    KalturaDeliveryProfileType["genericHds"] = "23";
    KalturaDeliveryProfileType["genericHdsManifest"] = "26";
    KalturaDeliveryProfileType["genericHls"] = "21";
    KalturaDeliveryProfileType["genericHlsManifest"] = "25";
    KalturaDeliveryProfileType["genericHttp"] = "24";
    KalturaDeliveryProfileType["genericRtmp"] = "28";
    KalturaDeliveryProfileType["genericSs"] = "27";
    KalturaDeliveryProfileType["hds"] = "3";
    KalturaDeliveryProfileType["http"] = "4";
    KalturaDeliveryProfileType["kontikiHttp"] = "kontiki.KONTIKI_HTTP";
    KalturaDeliveryProfileType["level3Hls"] = "31";
    KalturaDeliveryProfileType["level3Http"] = "34";
    KalturaDeliveryProfileType["level3Rtmp"] = "35";
    KalturaDeliveryProfileType["limelightHttp"] = "44";
    KalturaDeliveryProfileType["limelightRtmp"] = "45";
    KalturaDeliveryProfileType["liveAkamaiHds"] = "1013";
    KalturaDeliveryProfileType["liveDash"] = "1003";
    KalturaDeliveryProfileType["liveHds"] = "1002";
    KalturaDeliveryProfileType["liveHls"] = "1001";
    KalturaDeliveryProfileType["liveHlsToMulticast"] = "1006";
    KalturaDeliveryProfileType["livePackagerDash"] = "1009";
    KalturaDeliveryProfileType["livePackagerHds"] = "1008";
    KalturaDeliveryProfileType["livePackagerHls"] = "1007";
    KalturaDeliveryProfileType["livePackagerMss"] = "1010";
    KalturaDeliveryProfileType["liveRtmp"] = "1005";
    KalturaDeliveryProfileType["localPathAppleHttp"] = "51";
    KalturaDeliveryProfileType["localPathHds"] = "53";
    KalturaDeliveryProfileType["localPathHttp"] = "54";
    KalturaDeliveryProfileType["localPathRtmp"] = "55";
    KalturaDeliveryProfileType["rtmp"] = "5";
    KalturaDeliveryProfileType["rtsp"] = "6";
    KalturaDeliveryProfileType["silverLight"] = "7";
    KalturaDeliveryProfileType["uplynkHttp"] = "uplynk.UPLYNK_HTTP";
    KalturaDeliveryProfileType["uplynkRtmp"] = "uplynk.UPLYNK_RTMP";
    KalturaDeliveryProfileType["velocixHds"] = "velocix.VELOCIX_HDS";
    KalturaDeliveryProfileType["velocixHls"] = "velocix.VELOCIX_HLS";
    KalturaDeliveryProfileType["vodPackagerDash"] = "68";
    KalturaDeliveryProfileType["vodPackagerHds"] = "63";
    KalturaDeliveryProfileType["vodPackagerHls"] = "61";
    KalturaDeliveryProfileType["vodPackagerHlsManifest"] = "69";
    KalturaDeliveryProfileType["vodPackagerMss"] = "67";
})(KalturaDeliveryProfileType = exports.KalturaDeliveryProfileType || (exports.KalturaDeliveryProfileType = {}));
