// kaltura-node-typescript-client/api/types/KalturaFileSyncObjectType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFileSyncObjectType = void 0;
var KalturaFileSyncObjectType;
(function (KalturaFileSyncObjectType) {
    KalturaFileSyncObjectType["asset"] = "4";
    KalturaFileSyncObjectType["batchjob"] = "3";
    KalturaFileSyncObjectType["conversionProfile"] = "8";
    KalturaFileSyncObjectType["distributionProfile"] = "contentDistribution.DistributionProfile";
    KalturaFileSyncObjectType["emailNotificationTemplate"] = "emailNotification.EmailNotificationTemplate";
    KalturaFileSyncObjectType["entry"] = "1";
    KalturaFileSyncObjectType["entryDistribution"] = "contentDistribution.EntryDistribution";
    KalturaFileSyncObjectType["fileAsset"] = "9";
    KalturaFileSyncObjectType["flavorAsset"] = "4";
    KalturaFileSyncObjectType["genericDistributionAction"] = "contentDistribution.GenericDistributionAction";
    KalturaFileSyncObjectType["httpNotificationTemplate"] = "httpNotification.HttpNotificationTemplate";
    KalturaFileSyncObjectType["metadata"] = "5";
    KalturaFileSyncObjectType["metadataProfile"] = "6";
    KalturaFileSyncObjectType["syndicationFeed"] = "7";
    KalturaFileSyncObjectType["uiconf"] = "2";
})(KalturaFileSyncObjectType = exports.KalturaFileSyncObjectType || (exports.KalturaFileSyncObjectType = {}));
