// kaltura-node-typescript-client/api/types/KalturaThumbAssetBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaThumbAssetBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaThumbAssetStatus_1 = require("./KalturaThumbAssetStatus");
var KalturaAssetFilter_1 = require("./KalturaAssetFilter");
var KalturaThumbAssetBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaThumbAssetBaseFilter, _super);
    function KalturaThumbAssetBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaThumbAssetBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaThumbAssetBaseFilter' },
            thumbParamsIdEqual: { type: 'n' },
            thumbParamsIdIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaThumbAssetStatus_1.KalturaThumbAssetStatus, subType: 'KalturaThumbAssetStatus' },
            statusIn: { type: 's' },
            statusNotIn: { type: 's' }
        });
        return result;
    };
    return KalturaThumbAssetBaseFilter;
}(KalturaAssetFilter_1.KalturaAssetFilter));
exports.KalturaThumbAssetBaseFilter = KalturaThumbAssetBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaThumbAssetBaseFilter', KalturaThumbAssetBaseFilter);
