// kaltura-node-typescript-client/api/types/KalturaDistributionFieldConfig.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDistributionFieldConfig = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDistributionFieldRequiredStatus_1 = require("./KalturaDistributionFieldRequiredStatus");
var KalturaString_1 = require("./KalturaString");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaDistributionFieldConfig = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDistributionFieldConfig, _super);
    function KalturaDistributionFieldConfig(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.updateParams === 'undefined')
            _this.updateParams = [];
        return _this;
    }
    KalturaDistributionFieldConfig.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDistributionFieldConfig' },
            fieldName: { type: 's' },
            userFriendlyFieldName: { type: 's' },
            entryMrssXslt: { type: 's' },
            isRequired: { type: 'en', subTypeConstructor: KalturaDistributionFieldRequiredStatus_1.KalturaDistributionFieldRequiredStatus, subType: 'KalturaDistributionFieldRequiredStatus' },
            type: { type: 's' },
            updateOnChange: { type: 'b' },
            updateParams: { type: 'a', subTypeConstructor: KalturaString_1.KalturaString, subType: 'KalturaString' },
            isDefault: { type: 'b', readOnly: true },
            triggerDeleteOnError: { type: 'b' }
        });
        return result;
    };
    return KalturaDistributionFieldConfig;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaDistributionFieldConfig = KalturaDistributionFieldConfig;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDistributionFieldConfig', KalturaDistributionFieldConfig);
