// kaltura-node-typescript-client/api/types/KalturaBeacon.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBeacon = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBeaconObjectTypes_1 = require("./KalturaBeaconObjectTypes");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaBeacon = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBeacon, _super);
    function KalturaBeacon(data) {
        return _super.call(this, data) || this;
    }
    KalturaBeacon.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBeacon' },
            id: { type: 's', readOnly: true },
            indexType: { type: 's', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            relatedObjectType: { type: 'es', subTypeConstructor: KalturaBeaconObjectTypes_1.KalturaBeaconObjectTypes, subType: 'KalturaBeaconObjectTypes' },
            eventType: { type: 's' },
            objectId: { type: 's' },
            privateData: { type: 's' },
            rawData: { type: 's' }
        });
        return result;
    };
    return KalturaBeacon;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaBeacon = KalturaBeacon;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBeacon', KalturaBeacon);
