// kaltura-node-typescript-client/api/types/KalturaLiveReportInputFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveReportInputFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaLiveReportOrderBy_1 = require("./KalturaLiveReportOrderBy");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaLiveReportInputFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveReportInputFilter, _super);
    function KalturaLiveReportInputFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaLiveReportInputFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveReportInputFilter' },
            entryIds: { type: 's' },
            fromTime: { type: 'd' },
            toTime: { type: 'd' },
            live: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            orderBy: { type: 'es', subTypeConstructor: KalturaLiveReportOrderBy_1.KalturaLiveReportOrderBy, subType: 'KalturaLiveReportOrderBy' }
        });
        return result;
    };
    return KalturaLiveReportInputFilter;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaLiveReportInputFilter = KalturaLiveReportInputFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveReportInputFilter', KalturaLiveReportInputFilter);
