// kaltura-node-typescript-client/api/types/KalturaDistributionProviderType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDistributionProviderType = void 0;
var KalturaDistributionProviderType;
(function (KalturaDistributionProviderType) {
    KalturaDistributionProviderType["attUverse"] = "attUverseDistribution.ATT_UVERSE";
    KalturaDistributionProviderType["avn"] = "avnDistribution.AVN";
    KalturaDistributionProviderType["comcastMrss"] = "comcastMrssDistribution.COMCAST_MRSS";
    KalturaDistributionProviderType["cortexApi"] = "cortexApiDistribution.CORTEX_API";
    KalturaDistributionProviderType["crossKaltura"] = "crossKalturaDistribution.CROSS_KALTURA";
    KalturaDistributionProviderType["dailymotion"] = "dailymotionDistribution.DAILYMOTION";
    KalturaDistributionProviderType["doubleclick"] = "doubleClickDistribution.DOUBLECLICK";
    KalturaDistributionProviderType["facebook"] = "facebookDistribution.FACEBOOK";
    KalturaDistributionProviderType["freewheel"] = "freewheelDistribution.FREEWHEEL";
    KalturaDistributionProviderType["freewheelGeneric"] = "freewheelGenericDistribution.FREEWHEEL_GENERIC";
    KalturaDistributionProviderType["ftp"] = "ftpDistribution.FTP";
    KalturaDistributionProviderType["ftpScheduled"] = "ftpDistribution.FTP_SCHEDULED";
    KalturaDistributionProviderType["generic"] = "1";
    KalturaDistributionProviderType["hulu"] = "huluDistribution.HULU";
    KalturaDistributionProviderType["idetic"] = "ideticDistribution.IDETIC";
    KalturaDistributionProviderType["metroPcs"] = "metroPcsDistribution.METRO_PCS";
    KalturaDistributionProviderType["msn"] = "msnDistribution.MSN";
    KalturaDistributionProviderType["ndn"] = "ndnDistribution.NDN";
    KalturaDistributionProviderType["podcast"] = "podcastDistribution.PODCAST";
    KalturaDistributionProviderType["pushToNews"] = "pushToNewsDistribution.PUSH_TO_NEWS";
    KalturaDistributionProviderType["quickplay"] = "quickPlayDistribution.QUICKPLAY";
    KalturaDistributionProviderType["synacorHbo"] = "synacorHboDistribution.SYNACOR_HBO";
    KalturaDistributionProviderType["syndication"] = "2";
    KalturaDistributionProviderType["timeWarner"] = "timeWarnerDistribution.TIME_WARNER";
    KalturaDistributionProviderType["tvcom"] = "tvComDistribution.TVCOM";
    KalturaDistributionProviderType["tvinci"] = "tvinciDistribution.TVINCI";
    KalturaDistributionProviderType["unicorn"] = "unicornDistribution.UNICORN";
    KalturaDistributionProviderType["uverse"] = "uverseDistribution.UVERSE";
    KalturaDistributionProviderType["uverseClickToOrder"] = "uverseClickToOrderDistribution.UVERSE_CLICK_TO_ORDER";
    KalturaDistributionProviderType["verizonVcast"] = "verizonVcastDistribution.VERIZON_VCAST";
    KalturaDistributionProviderType["yahoo"] = "yahooDistribution.YAHOO";
    KalturaDistributionProviderType["youtube"] = "youTubeDistribution.YOUTUBE";
    KalturaDistributionProviderType["youtubeApi"] = "youtubeApiDistribution.YOUTUBE_API";
})(KalturaDistributionProviderType = exports.KalturaDistributionProviderType || (exports.KalturaDistributionProviderType = {}));
