// kaltura-node-typescript-client/api/types/KalturaTubeMogulSyndicationFeedCategories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaTubeMogulSyndicationFeedCategories = void 0;
var KalturaTubeMogulSyndicationFeedCategories;
(function (KalturaTubeMogulSyndicationFeedCategories) {
    KalturaTubeMogulSyndicationFeedCategories["animalsAndPets"] = "Animals &amp; Pets";
    KalturaTubeMogulSyndicationFeedCategories["artsAndAnimation"] = "Arts &amp; Animation";
    KalturaTubeMogulSyndicationFeedCategories["autos"] = "Autos";
    KalturaTubeMogulSyndicationFeedCategories["comedy"] = "Comedy";
    KalturaTubeMogulSyndicationFeedCategories["commercialsPromotional"] = "Commercials/Promotional";
    KalturaTubeMogulSyndicationFeedCategories["entertainment"] = "Entertainment";
    KalturaTubeMogulSyndicationFeedCategories["familyAndKids"] = "Family &amp; Kids";
    KalturaTubeMogulSyndicationFeedCategories["howToInstructionalDiy"] = "How To/Instructional/DIY";
    KalturaTubeMogulSyndicationFeedCategories["music"] = "Music";
    KalturaTubeMogulSyndicationFeedCategories["newsAndBlogs"] = "News &amp; Blogs";
    KalturaTubeMogulSyndicationFeedCategories["scienceAndTechnology"] = "Science &amp; Technology";
    KalturaTubeMogulSyndicationFeedCategories["sports"] = "Sports";
    KalturaTubeMogulSyndicationFeedCategories["travelAndPlaces"] = "Travel &amp; Places";
    KalturaTubeMogulSyndicationFeedCategories["videoGames"] = "Video Games";
    KalturaTubeMogulSyndicationFeedCategories["vlogsPeople"] = "Vlogs &amp; People";
})(KalturaTubeMogulSyndicationFeedCategories = exports.KalturaTubeMogulSyndicationFeedCategories || (exports.KalturaTubeMogulSyndicationFeedCategories = {}));
