// kaltura-node-typescript-client/api/types/KalturaReachProfileBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaReachProfileBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaReachProfileStatus_1 = require("./KalturaReachProfileStatus");
var KalturaReachProfileType_1 = require("./KalturaReachProfileType");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaReachProfileBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaReachProfileBaseFilter, _super);
    function KalturaReachProfileBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaReachProfileBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaReachProfileBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaReachProfileStatus_1.KalturaReachProfileStatus, subType: 'KalturaReachProfileStatus' },
            statusIn: { type: 's' },
            profileTypeEqual: { type: 'en', subTypeConstructor: KalturaReachProfileType_1.KalturaReachProfileType, subType: 'KalturaReachProfileType' },
            profileTypeIn: { type: 's' }
        });
        return result;
    };
    return KalturaReachProfileBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaReachProfileBaseFilter = KalturaReachProfileBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaReachProfileBaseFilter', KalturaReachProfileBaseFilter);
