// kaltura-node-typescript-client/api/types/KalturaAkamaiUniversalProvisionJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAkamaiUniversalProvisionJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDVRStatus_1 = require("./KalturaDVRStatus");
var KalturaAkamaiUniversalStreamType_1 = require("./KalturaAkamaiUniversalStreamType");
var KalturaProvisionJobData_1 = require("./KalturaProvisionJobData");
var KalturaAkamaiUniversalProvisionJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAkamaiUniversalProvisionJobData, _super);
    function KalturaAkamaiUniversalProvisionJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaAkamaiUniversalProvisionJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAkamaiUniversalProvisionJobData' },
            streamId: { type: 'n' },
            systemUserName: { type: 's' },
            systemPassword: { type: 's' },
            domainName: { type: 's' },
            dvrEnabled: { type: 'en', subTypeConstructor: KalturaDVRStatus_1.KalturaDVRStatus, subType: 'KalturaDVRStatus' },
            dvrWindow: { type: 'n' },
            primaryContact: { type: 's' },
            secondaryContact: { type: 's' },
            streamType: { type: 'es', subTypeConstructor: KalturaAkamaiUniversalStreamType_1.KalturaAkamaiUniversalStreamType, subType: 'KalturaAkamaiUniversalStreamType' },
            notificationEmail: { type: 's' }
        });
        return result;
    };
    return KalturaAkamaiUniversalProvisionJobData;
}(KalturaProvisionJobData_1.KalturaProvisionJobData));
exports.KalturaAkamaiUniversalProvisionJobData = KalturaAkamaiUniversalProvisionJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAkamaiUniversalProvisionJobData', KalturaAkamaiUniversalProvisionJobData);
