// kaltura-node-typescript-client/api/types/KalturaCategoryUserBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCategoryUserBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCategoryUserPermissionLevel_1 = require("./KalturaCategoryUserPermissionLevel");
var KalturaCategoryUserStatus_1 = require("./KalturaCategoryUserStatus");
var KalturaUpdateMethodType_1 = require("./KalturaUpdateMethodType");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaCategoryUserBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCategoryUserBaseFilter, _super);
    function KalturaCategoryUserBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaCategoryUserBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCategoryUserBaseFilter' },
            categoryIdEqual: { type: 'n' },
            categoryIdIn: { type: 's' },
            userIdEqual: { type: 's' },
            userIdIn: { type: 's' },
            permissionLevelEqual: { type: 'en', subTypeConstructor: KalturaCategoryUserPermissionLevel_1.KalturaCategoryUserPermissionLevel, subType: 'KalturaCategoryUserPermissionLevel' },
            permissionLevelIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaCategoryUserStatus_1.KalturaCategoryUserStatus, subType: 'KalturaCategoryUserStatus' },
            statusIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            updateMethodEqual: { type: 'en', subTypeConstructor: KalturaUpdateMethodType_1.KalturaUpdateMethodType, subType: 'KalturaUpdateMethodType' },
            updateMethodIn: { type: 's' },
            categoryFullIdsStartsWith: { type: 's' },
            categoryFullIdsEqual: { type: 's' },
            permissionNamesMatchAnd: { type: 's' },
            permissionNamesMatchOr: { type: 's' },
            permissionNamesNotContains: { type: 's' }
        });
        return result;
    };
    return KalturaCategoryUserBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaCategoryUserBaseFilter = KalturaCategoryUserBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCategoryUserBaseFilter', KalturaCategoryUserBaseFilter);
