// kaltura-node-typescript-client/api/types/KalturaScheduleResourceBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaScheduleResourceBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaScheduleResourceStatus_1 = require("./KalturaScheduleResourceStatus");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaScheduleResourceBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaScheduleResourceBaseFilter, _super);
    function KalturaScheduleResourceBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaScheduleResourceBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaScheduleResourceBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            idNotIn: { type: 's' },
            parentIdEqual: { type: 'n' },
            parentIdIn: { type: 's' },
            nameEqual: { type: 's' },
            systemNameEqual: { type: 's' },
            systemNameIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaScheduleResourceStatus_1.KalturaScheduleResourceStatus, subType: 'KalturaScheduleResourceStatus' },
            statusIn: { type: 's' },
            tagsLike: { type: 's' },
            tagsMultiLikeOr: { type: 's' },
            tagsMultiLikeAnd: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' }
        });
        return result;
    };
    return KalturaScheduleResourceBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaScheduleResourceBaseFilter = KalturaScheduleResourceBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaScheduleResourceBaseFilter', KalturaScheduleResourceBaseFilter);
