// kaltura-node-typescript-client/api/types/KalturaAdCuePointBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAdCuePointBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaAdProtocolType_1 = require("./KalturaAdProtocolType");
var KalturaCuePointFilter_1 = require("./KalturaCuePointFilter");
var KalturaAdCuePointBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAdCuePointBaseFilter, _super);
    function KalturaAdCuePointBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaAdCuePointBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAdCuePointBaseFilter' },
            protocolTypeEqual: { type: 'es', subTypeConstructor: KalturaAdProtocolType_1.KalturaAdProtocolType, subType: 'KalturaAdProtocolType' },
            protocolTypeIn: { type: 's' },
            titleLike: { type: 's' },
            titleMultiLikeOr: { type: 's' },
            titleMultiLikeAnd: { type: 's' },
            endTimeGreaterThanOrEqual: { type: 'n' },
            endTimeLessThanOrEqual: { type: 'n' },
            durationGreaterThanOrEqual: { type: 'n' },
            durationLessThanOrEqual: { type: 'n' }
        });
        return result;
    };
    return KalturaAdCuePointBaseFilter;
}(KalturaCuePointFilter_1.KalturaCuePointFilter));
exports.KalturaAdCuePointBaseFilter = KalturaAdCuePointBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAdCuePointBaseFilter', KalturaAdCuePointBaseFilter);
