// kaltura-node-typescript-client/api/types/KalturaTag.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaTag = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaTaggedObjectType_1 = require("./KalturaTaggedObjectType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaTag = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaTag, _super);
    function KalturaTag(data) {
        return _super.call(this, data) || this;
    }
    KalturaTag.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaTag' },
            id: { type: 'n', readOnly: true },
            tag: { type: 's', readOnly: true },
            taggedObjectType: { type: 'es', readOnly: true, subTypeConstructor: KalturaTaggedObjectType_1.KalturaTaggedObjectType, subType: 'KalturaTaggedObjectType' },
            partnerId: { type: 'n', readOnly: true },
            instanceCount: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true }
        });
        return result;
    };
    return KalturaTag;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaTag = KalturaTag;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaTag', KalturaTag);
