// kaltura-node-typescript-client/api/types/KalturaNdnDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaNdnDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaNdnDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaNdnDistributionProfile, _super);
    function KalturaNdnDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaNdnDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaNdnDistributionProfile' },
            feedUrl: { type: 's', readOnly: true },
            channelTitle: { type: 's' },
            channelLink: { type: 's' },
            channelDescription: { type: 's' },
            channelLanguage: { type: 's' },
            channelCopyright: { type: 's' },
            channelImageTitle: { type: 's' },
            channelImageUrl: { type: 's' },
            channelImageLink: { type: 's' },
            itemMediaRating: { type: 's' }
        });
        return result;
    };
    return KalturaNdnDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaNdnDistributionProfile = KalturaNdnDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaNdnDistributionProfile', KalturaNdnDistributionProfile);
