// kaltura-node-typescript-client/api/types/KalturaDrmPolicy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDrmPolicy = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDrmProviderType_1 = require("./KalturaDrmProviderType");
var KalturaDrmPolicyStatus_1 = require("./KalturaDrmPolicyStatus");
var KalturaDrmLicenseScenario_1 = require("./KalturaDrmLicenseScenario");
var KalturaDrmLicenseType_1 = require("./KalturaDrmLicenseType");
var KalturaDrmLicenseExpirationPolicy_1 = require("./KalturaDrmLicenseExpirationPolicy");
var KalturaKeyValue_1 = require("./KalturaKeyValue");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaDrmPolicy = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDrmPolicy, _super);
    function KalturaDrmPolicy(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.licenseParams === 'undefined')
            _this.licenseParams = [];
        return _this;
    }
    KalturaDrmPolicy.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDrmPolicy' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n' },
            name: { type: 's' },
            systemName: { type: 's' },
            description: { type: 's' },
            provider: { type: 'es', subTypeConstructor: KalturaDrmProviderType_1.KalturaDrmProviderType, subType: 'KalturaDrmProviderType' },
            status: { type: 'en', subTypeConstructor: KalturaDrmPolicyStatus_1.KalturaDrmPolicyStatus, subType: 'KalturaDrmPolicyStatus' },
            scenario: { type: 'es', subTypeConstructor: KalturaDrmLicenseScenario_1.KalturaDrmLicenseScenario, subType: 'KalturaDrmLicenseScenario' },
            licenseType: { type: 'es', subTypeConstructor: KalturaDrmLicenseType_1.KalturaDrmLicenseType, subType: 'KalturaDrmLicenseType' },
            licenseExpirationPolicy: { type: 'en', subTypeConstructor: KalturaDrmLicenseExpirationPolicy_1.KalturaDrmLicenseExpirationPolicy, subType: 'KalturaDrmLicenseExpirationPolicy' },
            duration: { type: 'n' },
            createdAt: { type: 'n', readOnly: true },
            updatedAt: { type: 'n', readOnly: true },
            licenseParams: { type: 'a', subTypeConstructor: KalturaKeyValue_1.KalturaKeyValue, subType: 'KalturaKeyValue' }
        });
        return result;
    };
    return KalturaDrmPolicy;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaDrmPolicy = KalturaDrmPolicy;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDrmPolicy', KalturaDrmPolicy);
