// kaltura-node-typescript-client/api/types/KalturaRequestConfiguration.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaRequestConfiguration = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBaseResponseProfile_1 = require("./KalturaBaseResponseProfile");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaRequestConfiguration = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaRequestConfiguration, _super);
    function KalturaRequestConfiguration(data) {
        return _super.call(this, data) || this;
    }
    KalturaRequestConfiguration.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaRequestConfiguration' },
            partnerId: { type: 'n' },
            ks: { type: 's' },
            language: { type: 's' },
            responseProfile: { type: 'o', subTypeConstructor: KalturaBaseResponseProfile_1.KalturaBaseResponseProfile, subType: 'KalturaBaseResponseProfile' }
        });
        return result;
    };
    return KalturaRequestConfiguration;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaRequestConfiguration = KalturaRequestConfiguration;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaRequestConfiguration', KalturaRequestConfiguration);
