// kaltura-node-typescript-client/api/types/KalturaZoomRecordingFile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaZoomRecordingFile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaRecordingFileType_1 = require("./KalturaRecordingFileType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaZoomRecordingFile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaZoomRecordingFile, _super);
    function KalturaZoomRecordingFile(data) {
        return _super.call(this, data) || this;
    }
    KalturaZoomRecordingFile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaZoomRecordingFile' },
            id: { type: 's' },
            recordingStart: { type: 's' },
            fileType: { type: 'en', subTypeConstructor: KalturaRecordingFileType_1.KalturaRecordingFileType, subType: 'KalturaRecordingFileType' },
            downloadUrl: { type: 's' },
            fileExtension: { type: 's' },
            downloadToken: { type: 's' }
        });
        return result;
    };
    return KalturaZoomRecordingFile;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaZoomRecordingFile = KalturaZoomRecordingFile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaZoomRecordingFile', KalturaZoomRecordingFile);
