// kaltura-node-typescript-client/api/types/KalturaVideoCodec.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaVideoCodec = void 0;
var KalturaVideoCodec;
(function (KalturaVideoCodec) {
    KalturaVideoCodec["apch"] = "apch";
    KalturaVideoCodec["apcn"] = "apcn";
    KalturaVideoCodec["apco"] = "apco";
    KalturaVideoCodec["apcs"] = "apcs";
    KalturaVideoCodec["av1"] = "av1";
    KalturaVideoCodec["copy"] = "copy";
    KalturaVideoCodec["dnxhd"] = "dnxhd";
    KalturaVideoCodec["dv"] = "dv";
    KalturaVideoCodec["flv"] = "flv";
    KalturaVideoCodec["h263"] = "h263";
    KalturaVideoCodec["h264"] = "h264";
    KalturaVideoCodec["h264b"] = "h264b";
    KalturaVideoCodec["h264h"] = "h264h";
    KalturaVideoCodec["h264m"] = "h264m";
    KalturaVideoCodec["h265"] = "h265";
    KalturaVideoCodec["mpeg2"] = "mpeg2";
    KalturaVideoCodec["mpeg4"] = "mpeg4";
    KalturaVideoCodec["none"] = "";
    KalturaVideoCodec["theora"] = "theora";
    KalturaVideoCodec["vp6"] = "vp6";
    KalturaVideoCodec["vp8"] = "vp8";
    KalturaVideoCodec["vp9"] = "vp9";
    KalturaVideoCodec["wmv2"] = "wmv2";
    KalturaVideoCodec["wmv3"] = "wmv3";
    KalturaVideoCodec["wvc1a"] = "wvc1a";
})(KalturaVideoCodec = exports.KalturaVideoCodec || (exports.KalturaVideoCodec = {}));
