// kaltura-node-typescript-client/api/types/KalturaPlaylistMatchAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPlaylistMatchAttribute = void 0;
var KalturaPlaylistMatchAttribute;
(function (KalturaPlaylistMatchAttribute) {
    KalturaPlaylistMatchAttribute["adminTags"] = "adminTags";
    KalturaPlaylistMatchAttribute["categoriesIds"] = "categoriesIds";
    KalturaPlaylistMatchAttribute["creatorId"] = "creatorId";
    KalturaPlaylistMatchAttribute["description"] = "description";
    KalturaPlaylistMatchAttribute["groupId"] = "groupId";
    KalturaPlaylistMatchAttribute["id"] = "id";
    KalturaPlaylistMatchAttribute["_name"] = "name";
    KalturaPlaylistMatchAttribute["referenceId"] = "referenceId";
    KalturaPlaylistMatchAttribute["replacedEntryId"] = "replacedEntryId";
    KalturaPlaylistMatchAttribute["replacingEntryId"] = "replacingEntryId";
    KalturaPlaylistMatchAttribute["searchText"] = "searchText";
    KalturaPlaylistMatchAttribute["tags"] = "tags";
    KalturaPlaylistMatchAttribute["userId"] = "userId";
})(KalturaPlaylistMatchAttribute = exports.KalturaPlaylistMatchAttribute || (exports.KalturaPlaylistMatchAttribute = {}));
