// kaltura-node-typescript-client/api/types/KalturaEmailNotificationDispatchJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEmailNotificationDispatchJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaEmailNotificationRecipientJobData_1 = require("./KalturaEmailNotificationRecipientJobData");
var KalturaEmailNotificationTemplatePriority_1 = require("./KalturaEmailNotificationTemplatePriority");
var KalturaKeyValue_1 = require("./KalturaKeyValue");
var KalturaEventNotificationDispatchJobData_1 = require("./KalturaEventNotificationDispatchJobData");
var KalturaEmailNotificationDispatchJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaEmailNotificationDispatchJobData, _super);
    function KalturaEmailNotificationDispatchJobData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.customHeaders === 'undefined')
            _this.customHeaders = [];
        return _this;
    }
    KalturaEmailNotificationDispatchJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaEmailNotificationDispatchJobData' },
            fromEmail: { type: 's' },
            fromName: { type: 's' },
            to: { type: 'o', subTypeConstructor: KalturaEmailNotificationRecipientJobData_1.KalturaEmailNotificationRecipientJobData, subType: 'KalturaEmailNotificationRecipientJobData' },
            cc: { type: 'o', subTypeConstructor: KalturaEmailNotificationRecipientJobData_1.KalturaEmailNotificationRecipientJobData, subType: 'KalturaEmailNotificationRecipientJobData' },
            bcc: { type: 'o', subTypeConstructor: KalturaEmailNotificationRecipientJobData_1.KalturaEmailNotificationRecipientJobData, subType: 'KalturaEmailNotificationRecipientJobData' },
            replyTo: { type: 'o', subTypeConstructor: KalturaEmailNotificationRecipientJobData_1.KalturaEmailNotificationRecipientJobData, subType: 'KalturaEmailNotificationRecipientJobData' },
            priority: { type: 'en', subTypeConstructor: KalturaEmailNotificationTemplatePriority_1.KalturaEmailNotificationTemplatePriority, subType: 'KalturaEmailNotificationTemplatePriority' },
            confirmReadingTo: { type: 's' },
            hostname: { type: 's' },
            messageID: { type: 's' },
            customHeaders: { type: 'a', subTypeConstructor: KalturaKeyValue_1.KalturaKeyValue, subType: 'KalturaKeyValue' }
        });
        return result;
    };
    return KalturaEmailNotificationDispatchJobData;
}(KalturaEventNotificationDispatchJobData_1.KalturaEventNotificationDispatchJobData));
exports.KalturaEmailNotificationDispatchJobData = KalturaEmailNotificationDispatchJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaEmailNotificationDispatchJobData', KalturaEmailNotificationDispatchJobData);
