// kaltura-node-typescript-client/api/types/KalturaFileAssetBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFileAssetBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaFileAssetObjectType_1 = require("./KalturaFileAssetObjectType");
var KalturaFileAssetStatus_1 = require("./KalturaFileAssetStatus");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaFileAssetBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaFileAssetBaseFilter, _super);
    function KalturaFileAssetBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaFileAssetBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaFileAssetBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            partnerIdEqual: { type: 'n' },
            fileAssetObjectTypeEqual: { type: 'es', subTypeConstructor: KalturaFileAssetObjectType_1.KalturaFileAssetObjectType, subType: 'KalturaFileAssetObjectType' },
            objectIdEqual: { type: 's' },
            objectIdIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'n' },
            createdAtLessThanOrEqual: { type: 'n' },
            updatedAtGreaterThanOrEqual: { type: 'n' },
            updatedAtLessThanOrEqual: { type: 'n' },
            statusEqual: { type: 'es', subTypeConstructor: KalturaFileAssetStatus_1.KalturaFileAssetStatus, subType: 'KalturaFileAssetStatus' },
            statusIn: { type: 's' }
        });
        return result;
    };
    return KalturaFileAssetBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaFileAssetBaseFilter = KalturaFileAssetBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaFileAssetBaseFilter', KalturaFileAssetBaseFilter);
