// kaltura-node-typescript-client/api/types/KalturaSynacorHboDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSynacorHboDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaSynacorHboDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaSynacorHboDistributionProfile, _super);
    function KalturaSynacorHboDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaSynacorHboDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaSynacorHboDistributionProfile' },
            feedUrl: { type: 's', readOnly: true },
            feedTitle: { type: 's' },
            feedSubtitle: { type: 's' },
            feedLink: { type: 's' },
            feedAuthorName: { type: 's' }
        });
        return result;
    };
    return KalturaSynacorHboDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaSynacorHboDistributionProfile = KalturaSynacorHboDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaSynacorHboDistributionProfile', KalturaSynacorHboDistributionProfile);
