// kaltura-node-typescript-client/api/types/KalturaStatsEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaStatsEvent = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaStatsEventType_1 = require("./KalturaStatsEventType");
var KalturaStatsFeatureType_1 = require("./KalturaStatsFeatureType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaStatsEvent = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaStatsEvent, _super);
    function KalturaStatsEvent(data) {
        return _super.call(this, data) || this;
    }
    KalturaStatsEvent.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaStatsEvent' },
            clientVer: { type: 's' },
            eventType: { type: 'en', subTypeConstructor: KalturaStatsEventType_1.KalturaStatsEventType, subType: 'KalturaStatsEventType' },
            eventTimestamp: { type: 'n' },
            sessionId: { type: 's' },
            partnerId: { type: 'n' },
            entryId: { type: 's' },
            uniqueViewer: { type: 's' },
            widgetId: { type: 's' },
            uiconfId: { type: 'n' },
            userId: { type: 's' },
            currentPoint: { type: 'n' },
            duration: { type: 'n' },
            userIp: { type: 's', readOnly: true },
            processDuration: { type: 'n' },
            controlId: { type: 's' },
            seek: { type: 'b' },
            newPoint: { type: 'n' },
            referrer: { type: 's' },
            isFirstInSession: { type: 'b' },
            applicationId: { type: 's' },
            contextId: { type: 'n' },
            featureType: { type: 'en', subTypeConstructor: KalturaStatsFeatureType_1.KalturaStatsFeatureType, subType: 'KalturaStatsFeatureType' }
        });
        return result;
    };
    return KalturaStatsEvent;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaStatsEvent = KalturaStatsEvent;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaStatsEvent', KalturaStatsEvent);
