// kaltura-node-typescript-client/api/types/KalturaPostConvertJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPostConvertJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConvartableJobData_1 = require("./KalturaConvartableJobData");
var KalturaPostConvertJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPostConvertJobData, _super);
    function KalturaPostConvertJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaPostConvertJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPostConvertJobData' },
            flavorAssetId: { type: 's' },
            flavorAssetEncryptionKey: { type: 's' },
            createThumb: { type: 'b' },
            thumbPath: { type: 's' },
            thumbOffset: { type: 'n' },
            thumbHeight: { type: 'n' },
            thumbBitrate: { type: 'n' },
            customData: { type: 's' }
        });
        return result;
    };
    return KalturaPostConvertJobData;
}(KalturaConvartableJobData_1.KalturaConvartableJobData));
exports.KalturaPostConvertJobData = KalturaPostConvertJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPostConvertJobData', KalturaPostConvertJobData);
