// kaltura-node-typescript-client/api/types/KalturaEntryDistributionStatus.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEntryDistributionStatus = void 0;
var KalturaEntryDistributionStatus;
(function (KalturaEntryDistributionStatus) {
    KalturaEntryDistributionStatus[KalturaEntryDistributionStatus["deleted"] = 3] = "deleted";
    KalturaEntryDistributionStatus[KalturaEntryDistributionStatus["deleting"] = 6] = "deleting";
    KalturaEntryDistributionStatus[KalturaEntryDistributionStatus["errorDeleting"] = 9] = "errorDeleting";
    KalturaEntryDistributionStatus[KalturaEntryDistributionStatus["errorSubmitting"] = 7] = "errorSubmitting";
    KalturaEntryDistributionStatus[KalturaEntryDistributionStatus["errorUpdating"] = 8] = "errorUpdating";
    KalturaEntryDistributionStatus[KalturaEntryDistributionStatus["importSubmitting"] = 11] = "importSubmitting";
    KalturaEntryDistributionStatus[KalturaEntryDistributionStatus["importUpdating"] = 12] = "importUpdating";
    KalturaEntryDistributionStatus[KalturaEntryDistributionStatus["pending"] = 0] = "pending";
    KalturaEntryDistributionStatus[KalturaEntryDistributionStatus["queued"] = 1] = "queued";
    KalturaEntryDistributionStatus[KalturaEntryDistributionStatus["ready"] = 2] = "ready";
    KalturaEntryDistributionStatus[KalturaEntryDistributionStatus["removed"] = 10] = "removed";
    KalturaEntryDistributionStatus[KalturaEntryDistributionStatus["submitting"] = 4] = "submitting";
    KalturaEntryDistributionStatus[KalturaEntryDistributionStatus["updating"] = 5] = "updating";
})(KalturaEntryDistributionStatus = exports.KalturaEntryDistributionStatus || (exports.KalturaEntryDistributionStatus = {}));
