// kaltura-node-typescript-client/api/types/KalturaShortLinkBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaShortLinkBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaShortLinkStatus_1 = require("./KalturaShortLinkStatus");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaShortLinkBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaShortLinkBaseFilter, _super);
    function KalturaShortLinkBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaShortLinkBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaShortLinkBaseFilter' },
            idEqual: { type: 's' },
            idIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            expiresAtGreaterThanOrEqual: { type: 'n' },
            expiresAtLessThanOrEqual: { type: 'n' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            userIdEqual: { type: 's' },
            userIdIn: { type: 's' },
            systemNameEqual: { type: 's' },
            systemNameIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaShortLinkStatus_1.KalturaShortLinkStatus, subType: 'KalturaShortLinkStatus' },
            statusIn: { type: 's' }
        });
        return result;
    };
    return KalturaShortLinkBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaShortLinkBaseFilter = KalturaShortLinkBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaShortLinkBaseFilter', KalturaShortLinkBaseFilter);
