// kaltura-node-typescript-client/api/types/KalturaConditionType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaConditionType = void 0;
var KalturaConditionType;
(function (KalturaConditionType) {
    KalturaConditionType["abcWatermark"] = "abcScreenersWatermarkAccessControl.abcWatermark";
    KalturaConditionType["actionName"] = "20";
    KalturaConditionType["activeEdgeValidate"] = "14";
    KalturaConditionType["anonymousIp"] = "15";
    KalturaConditionType["assetPropertiesCompare"] = "8";
    KalturaConditionType["assetType"] = "16";
    KalturaConditionType["authenticated"] = "1";
    KalturaConditionType["boolean"] = "17";
    KalturaConditionType["country"] = "2";
    KalturaConditionType["deliveryProfile"] = "13";
    KalturaConditionType["entryScheduled"] = "19";
    KalturaConditionType["eventCategoryEntry"] = "reach.CategoryEntry";
    KalturaConditionType["eventNotificationField"] = "eventNotification.BooleanField";
    KalturaConditionType["eventNotificationObjectChanged"] = "eventNotification.ObjectChanged";
    KalturaConditionType["fieldCompare"] = "7";
    KalturaConditionType["fieldMatch"] = "6";
    KalturaConditionType["geoDistance"] = "10";
    KalturaConditionType["hash"] = "12";
    KalturaConditionType["httpHeader"] = "18";
    KalturaConditionType["ipAddress"] = "3";
    KalturaConditionType["metadataFieldChanged"] = "metadata.FieldChanged";
    KalturaConditionType["metadataFieldCompare"] = "metadata.FieldCompare";
    KalturaConditionType["metadataFieldMatch"] = "metadata.FieldMatch";
    KalturaConditionType["orOperator"] = "11";
    KalturaConditionType["site"] = "4";
    KalturaConditionType["urlAuthParams"] = "21";
    KalturaConditionType["userAgent"] = "5";
    KalturaConditionType["userRole"] = "9";
})(KalturaConditionType = exports.KalturaConditionType || (exports.KalturaConditionType = {}));
