// kaltura-node-typescript-client/api/types/KalturaQuizUserEntry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaQuizUserEntry = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUserEntry_1 = require("./KalturaUserEntry");
var KalturaQuizUserEntry = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaQuizUserEntry, _super);
    function KalturaQuizUserEntry(data) {
        return _super.call(this, data) || this;
    }
    KalturaQuizUserEntry.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaQuizUserEntry' },
            score: { type: 'n', readOnly: true },
            calculatedScore: { type: 'n', readOnly: true },
            feedback: { type: 's' },
            version: { type: 'n', readOnly: true }
        });
        return result;
    };
    return KalturaQuizUserEntry;
}(KalturaUserEntry_1.KalturaUserEntry));
exports.KalturaQuizUserEntry = KalturaQuizUserEntry;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaQuizUserEntry', KalturaQuizUserEntry);
