// kaltura-node-typescript-client/api/types/KalturaCaptureThumbJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCaptureThumbJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaFileContainer_1 = require("./KalturaFileContainer");
var KalturaAssetType_1 = require("./KalturaAssetType");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaCaptureThumbJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCaptureThumbJobData, _super);
    function KalturaCaptureThumbJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaCaptureThumbJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCaptureThumbJobData' },
            fileContainer: { type: 'o', subTypeConstructor: KalturaFileContainer_1.KalturaFileContainer, subType: 'KalturaFileContainer' },
            actualSrcFileSyncLocalPath: { type: 's' },
            srcFileSyncRemoteUrl: { type: 's' },
            thumbParamsOutputId: { type: 'n' },
            thumbAssetId: { type: 's' },
            srcAssetId: { type: 's' },
            srcAssetEncryptionKey: { type: 's' },
            srcAssetType: { type: 'es', subTypeConstructor: KalturaAssetType_1.KalturaAssetType, subType: 'KalturaAssetType' },
            thumbPath: { type: 's' }
        });
        return result;
    };
    return KalturaCaptureThumbJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaCaptureThumbJobData = KalturaCaptureThumbJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCaptureThumbJobData', KalturaCaptureThumbJobData);
