// kaltura-node-typescript-client/api/types/KalturaLiveEntryMatchAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveEntryMatchAttribute = void 0;
var KalturaLiveEntryMatchAttribute;
(function (KalturaLiveEntryMatchAttribute) {
    KalturaLiveEntryMatchAttribute["adminTags"] = "adminTags";
    KalturaLiveEntryMatchAttribute["categoriesIds"] = "categoriesIds";
    KalturaLiveEntryMatchAttribute["creatorId"] = "creatorId";
    KalturaLiveEntryMatchAttribute["description"] = "description";
    KalturaLiveEntryMatchAttribute["durationType"] = "durationType";
    KalturaLiveEntryMatchAttribute["flavorParamsIds"] = "flavorParamsIds";
    KalturaLiveEntryMatchAttribute["groupId"] = "groupId";
    KalturaLiveEntryMatchAttribute["id"] = "id";
    KalturaLiveEntryMatchAttribute["_name"] = "name";
    KalturaLiveEntryMatchAttribute["referenceId"] = "referenceId";
    KalturaLiveEntryMatchAttribute["replacedEntryId"] = "replacedEntryId";
    KalturaLiveEntryMatchAttribute["replacingEntryId"] = "replacingEntryId";
    KalturaLiveEntryMatchAttribute["searchText"] = "searchText";
    KalturaLiveEntryMatchAttribute["tags"] = "tags";
    KalturaLiveEntryMatchAttribute["userId"] = "userId";
})(KalturaLiveEntryMatchAttribute = exports.KalturaLiveEntryMatchAttribute || (exports.KalturaLiveEntryMatchAttribute = {}));
