// kaltura-node-typescript-client/api/types/KalturaLiveChannelCompareAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveChannelCompareAttribute = void 0;
var KalturaLiveChannelCompareAttribute;
(function (KalturaLiveChannelCompareAttribute) {
    KalturaLiveChannelCompareAttribute["accessControlId"] = "accessControlId";
    KalturaLiveChannelCompareAttribute["createdAt"] = "createdAt";
    KalturaLiveChannelCompareAttribute["endDate"] = "endDate";
    KalturaLiveChannelCompareAttribute["lastPlayedAt"] = "lastPlayedAt";
    KalturaLiveChannelCompareAttribute["mediaDate"] = "mediaDate";
    KalturaLiveChannelCompareAttribute["mediaType"] = "mediaType";
    KalturaLiveChannelCompareAttribute["moderationCount"] = "moderationCount";
    KalturaLiveChannelCompareAttribute["moderationStatus"] = "moderationStatus";
    KalturaLiveChannelCompareAttribute["msDuration"] = "msDuration";
    KalturaLiveChannelCompareAttribute["partnerId"] = "partnerId";
    KalturaLiveChannelCompareAttribute["partnerSortValue"] = "partnerSortValue";
    KalturaLiveChannelCompareAttribute["plays"] = "plays";
    KalturaLiveChannelCompareAttribute["rank"] = "rank";
    KalturaLiveChannelCompareAttribute["replacementStatus"] = "replacementStatus";
    KalturaLiveChannelCompareAttribute["startDate"] = "startDate";
    KalturaLiveChannelCompareAttribute["status"] = "status";
    KalturaLiveChannelCompareAttribute["totalRank"] = "totalRank";
    KalturaLiveChannelCompareAttribute["type"] = "type";
    KalturaLiveChannelCompareAttribute["updatedAt"] = "updatedAt";
    KalturaLiveChannelCompareAttribute["views"] = "views";
})(KalturaLiveChannelCompareAttribute = exports.KalturaLiveChannelCompareAttribute || (exports.KalturaLiveChannelCompareAttribute = {}));
