// kaltura-node-typescript-client/api/types/KalturaESearchAbstractUserItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaESearchAbstractUserItem = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaESearchItemType_1 = require("./KalturaESearchItemType");
var KalturaESearchRange_1 = require("./KalturaESearchRange");
var KalturaESearchUserBaseItem_1 = require("./KalturaESearchUserBaseItem");
var KalturaESearchAbstractUserItem = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaESearchAbstractUserItem, _super);
    function KalturaESearchAbstractUserItem(data) {
        return _super.call(this, data) || this;
    }
    KalturaESearchAbstractUserItem.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaESearchAbstractUserItem' },
            searchTerm: { type: 's' },
            itemType: { type: 'en', subTypeConstructor: KalturaESearchItemType_1.KalturaESearchItemType, subType: 'KalturaESearchItemType' },
            range: { type: 'o', subTypeConstructor: KalturaESearchRange_1.KalturaESearchRange, subType: 'KalturaESearchRange' },
            addHighlight: { type: 'b' }
        });
        return result;
    };
    return KalturaESearchAbstractUserItem;
}(KalturaESearchUserBaseItem_1.KalturaESearchUserBaseItem));
exports.KalturaESearchAbstractUserItem = KalturaESearchAbstractUserItem;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaESearchAbstractUserItem', KalturaESearchAbstractUserItem);
