// kaltura-node-typescript-client/api/types/KalturaPlayableEntryBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPlayableEntryBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBaseEntryFilter_1 = require("./KalturaBaseEntryFilter");
var KalturaPlayableEntryBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPlayableEntryBaseFilter, _super);
    function KalturaPlayableEntryBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaPlayableEntryBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPlayableEntryBaseFilter' },
            lastPlayedAtGreaterThanOrEqual: { type: 'd' },
            lastPlayedAtLessThanOrEqual: { type: 'd' },
            lastPlayedAtLessThanOrEqualOrNull: { type: 'd' },
            durationLessThan: { type: 'n' },
            durationGreaterThan: { type: 'n' },
            durationLessThanOrEqual: { type: 'n' },
            durationGreaterThanOrEqual: { type: 'n' },
            durationTypeMatchOr: { type: 's' }
        });
        return result;
    };
    return KalturaPlayableEntryBaseFilter;
}(KalturaBaseEntryFilter_1.KalturaBaseEntryFilter));
exports.KalturaPlayableEntryBaseFilter = KalturaPlayableEntryBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPlayableEntryBaseFilter', KalturaPlayableEntryBaseFilter);
