// kaltura-node-typescript-client/api/types/KalturaESearchCategoryFieldName.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaESearchCategoryFieldName = void 0;
var KalturaESearchCategoryFieldName;
(function (KalturaESearchCategoryFieldName) {
    KalturaESearchCategoryFieldName["contributionPolicy"] = "contribution_policy";
    KalturaESearchCategoryFieldName["createdAt"] = "created_at";
    KalturaESearchCategoryFieldName["depth"] = "depth";
    KalturaESearchCategoryFieldName["description"] = "description";
    KalturaESearchCategoryFieldName["directEntriesCount"] = "direct_entries_count";
    KalturaESearchCategoryFieldName["directSubCategoriesCount"] = "direct_sub_categories_count";
    KalturaESearchCategoryFieldName["displayInSearch"] = "display_in_search";
    KalturaESearchCategoryFieldName["entriesCount"] = "entries_count";
    KalturaESearchCategoryFieldName["fullIds"] = "full_ids";
    KalturaESearchCategoryFieldName["fullName"] = "full_name";
    KalturaESearchCategoryFieldName["id"] = "id";
    KalturaESearchCategoryFieldName["inheritanceType"] = "inheritance_type";
    KalturaESearchCategoryFieldName["inheritedParentId"] = "inherited_parent_id";
    KalturaESearchCategoryFieldName["membersCount"] = "members_count";
    KalturaESearchCategoryFieldName["moderation"] = "moderation";
    KalturaESearchCategoryFieldName["_name"] = "name";
    KalturaESearchCategoryFieldName["parentId"] = "parent_id";
    KalturaESearchCategoryFieldName["pendingEntriesCount"] = "pending_entries_count";
    KalturaESearchCategoryFieldName["pendingMembersCount"] = "pending_members_count";
    KalturaESearchCategoryFieldName["privacy"] = "privacy";
    KalturaESearchCategoryFieldName["privacyContext"] = "privacy_context";
    KalturaESearchCategoryFieldName["privacyContexts"] = "privacy_contexts";
    KalturaESearchCategoryFieldName["referenceId"] = "reference_id";
    KalturaESearchCategoryFieldName["tags"] = "tags";
    KalturaESearchCategoryFieldName["updatedAt"] = "updated_at";
    KalturaESearchCategoryFieldName["userId"] = "user_id";
})(KalturaESearchCategoryFieldName = exports.KalturaESearchCategoryFieldName || (exports.KalturaESearchCategoryFieldName = {}));
