// kaltura-node-typescript-client/api/types/KalturaReportType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaReportType = void 0;
var KalturaReportType;
(function (KalturaReportType) {
    KalturaReportType["applications"] = "16";
    KalturaReportType["appDomainUniqueActiveUsers"] = "40";
    KalturaReportType["browsers"] = "23";
    KalturaReportType["browsersFamilies"] = "33";
    KalturaReportType["browsersFamiliesVpaas"] = "20009";
    KalturaReportType["browsersVpaas"] = "20007";
    KalturaReportType["categoryHighlights"] = "52";
    KalturaReportType["cdnBandwidthUsage"] = "64";
    KalturaReportType["cncParticipation"] = "70001";
    KalturaReportType["contentContributions"] = "7";
    KalturaReportType["contentDropoff"] = "2";
    KalturaReportType["contentDropoffVpaas"] = "20001";
    KalturaReportType["contentInteractions"] = "3";
    KalturaReportType["contentRealtime"] = "10012";
    KalturaReportType["contentReportReasons"] = "44";
    KalturaReportType["contentReportReasonsVpaas"] = "20018";
    KalturaReportType["discoveryRealtime"] = "10007";
    KalturaReportType["discoveryViewRealtime"] = "10013";
    KalturaReportType["engagementBreakdownWebcast"] = "40010";
    KalturaReportType["engagementToolsWebcast"] = "40012";
    KalturaReportType["engagementWebcast"] = "40002";
    KalturaReportType["engagmentTimelineWebcast"] = "40011";
    KalturaReportType["entryLevelUsersDiscoveryRealtime"] = "10008";
    KalturaReportType["entryLevelUsersStatusRealtime"] = "10009";
    KalturaReportType["entryUsage"] = "27";
    KalturaReportType["epAttendees"] = "60014";
    KalturaReportType["epTopMoments"] = "60016";
    KalturaReportType["epTopSessions"] = "60017";
    KalturaReportType["epViewtime"] = "60015";
    KalturaReportType["epWebcastEngagement"] = "60003";
    KalturaReportType["epWebcastEngagementOverTime"] = "60018";
    KalturaReportType["epWebcastEngagementTimeline"] = "60004";
    KalturaReportType["epWebcastHighlights"] = "60001";
    KalturaReportType["epWebcastLiveUserEngagement"] = "60010";
    KalturaReportType["epWebcastLiveUserEngagementLevel"] = "60011";
    KalturaReportType["epWebcastMapOverlayCity"] = "60009";
    KalturaReportType["epWebcastMapOverlayCountry"] = "60007";
    KalturaReportType["epWebcastMapOverlayRegion"] = "60008";
    KalturaReportType["epWebcastTopPlatforms"] = "60006";
    KalturaReportType["epWebcastTopRecording"] = "60005";
    KalturaReportType["epWebcastUniqueUsers"] = "60002";
    KalturaReportType["epWebcastVodLiveUsersEngagement"] = "60013";
    KalturaReportType["epWebcastVodUserTopContent"] = "60012";
    KalturaReportType["flavorParamsTranscodingUsage"] = "61";
    KalturaReportType["highlightsWebcast"] = "40001";
    KalturaReportType["interactiveVideoHotspotClickedPercentiles"] = "56";
    KalturaReportType["interactiveVideoNodeSwitchHotspotClickedPercentiles"] = "57";
    KalturaReportType["interactiveVideoNodeTopHotspots"] = "54";
    KalturaReportType["interactiveVideoTopNodes"] = "50";
    KalturaReportType["intercativeVideoNodeSwitchTopHotspots"] = "55";
    KalturaReportType["latestPlayedEntries"] = "51";
    KalturaReportType["live"] = "24";
    KalturaReportType["liveMeetingUsersOverviewRealtime"] = "10016";
    KalturaReportType["mapOverlay"] = "4";
    KalturaReportType["mapOverlayCity"] = "30";
    KalturaReportType["mapOverlayCityRealtime"] = "10003";
    KalturaReportType["mapOverlayCityVpaas"] = "20014";
    KalturaReportType["mapOverlayCityWebcast"] = "40006";
    KalturaReportType["mapOverlayCountry"] = "36";
    KalturaReportType["mapOverlayCountryRealtime"] = "10001";
    KalturaReportType["mapOverlayCountryVpaas"] = "20012";
    KalturaReportType["mapOverlayCountryWebcast"] = "40004";
    KalturaReportType["mapOverlayRegion"] = "37";
    KalturaReportType["mapOverlayRegionRealtime"] = "10002";
    KalturaReportType["mapOverlayRegionVpaas"] = "20013";
    KalturaReportType["mapOverlayRegionWebcast"] = "40005";
    KalturaReportType["operatingSystem"] = "22";
    KalturaReportType["operatingSystemFamilies"] = "32";
    KalturaReportType["operatingSystemFamiliesVpaas"] = "20008";
    KalturaReportType["operatingSystemVpaas"] = "20006";
    KalturaReportType["partnerUsage"] = "201";
    KalturaReportType["partnerUsageHighlights"] = "63";
    KalturaReportType["partnerUsageVpaas"] = "20021";
    KalturaReportType["percentiles"] = "43";
    KalturaReportType["platforms"] = "21";
    KalturaReportType["platformsDiscoveryRealtime"] = "10010";
    KalturaReportType["platformsRealtime"] = "10004";
    KalturaReportType["platformsVpaas"] = "20005";
    KalturaReportType["platformsWebcast"] = "40007";
    KalturaReportType["playbackRate"] = "46";
    KalturaReportType["playbackRateVpaas"] = "20020";
    KalturaReportType["playbackTypeRealtime"] = "10011";
    KalturaReportType["playerHighlights"] = "62";
    KalturaReportType["playerRelatedInteractions"] = "45";
    KalturaReportType["playerRelatedInteractionsVpaas"] = "20019";
    KalturaReportType["qoeEngagement"] = "30014";
    KalturaReportType["qoeEngagementApplicationVersion"] = "30062";
    KalturaReportType["qoeEngagementBrowsers"] = "30020";
    KalturaReportType["qoeEngagementBrowsersFamilies"] = "30019";
    KalturaReportType["qoeEngagementCity"] = "30018";
    KalturaReportType["qoeEngagementCountry"] = "30016";
    KalturaReportType["qoeEngagementCustomVar1"] = "30052";
    KalturaReportType["qoeEngagementCustomVar2"] = "30053";
    KalturaReportType["qoeEngagementCustomVar3"] = "30054";
    KalturaReportType["qoeEngagementEntry"] = "30024";
    KalturaReportType["qoeEngagementIsp"] = "30025";
    KalturaReportType["qoeEngagementOperatingSystem"] = "30022";
    KalturaReportType["qoeEngagementOperatingSystemFamilies"] = "30021";
    KalturaReportType["qoeEngagementPlatforms"] = "30015";
    KalturaReportType["qoeEngagementPlayerVersion"] = "30023";
    KalturaReportType["qoeEngagementRegion"] = "30017";
    KalturaReportType["qoeErrorTracking"] = "30038";
    KalturaReportType["qoeErrorTrackingApplicationVersion"] = "30064";
    KalturaReportType["qoeErrorTrackingBrowsers"] = "30042";
    KalturaReportType["qoeErrorTrackingBrowsersFamilies"] = "30041";
    KalturaReportType["qoeErrorTrackingCodes"] = "30039";
    KalturaReportType["qoeErrorTrackingCustomVar1"] = "30058";
    KalturaReportType["qoeErrorTrackingCustomVar2"] = "30059";
    KalturaReportType["qoeErrorTrackingCustomVar3"] = "30060";
    KalturaReportType["qoeErrorTrackingEntry"] = "30046";
    KalturaReportType["qoeErrorTrackingOperatingSystem"] = "30044";
    KalturaReportType["qoeErrorTrackingOperatingSystemFamilies"] = "30043";
    KalturaReportType["qoeErrorTrackingPlatforms"] = "30040";
    KalturaReportType["qoeErrorTrackingPlayerVersion"] = "30045";
    KalturaReportType["qoeExperience"] = "30002";
    KalturaReportType["qoeExperienceApplicationVersion"] = "30061";
    KalturaReportType["qoeExperienceBrowsers"] = "30008";
    KalturaReportType["qoeExperienceBrowsersFamilies"] = "30007";
    KalturaReportType["qoeExperienceCity"] = "30006";
    KalturaReportType["qoeExperienceCountry"] = "30004";
    KalturaReportType["qoeExperienceCustomVar1"] = "30049";
    KalturaReportType["qoeExperienceCustomVar2"] = "30050";
    KalturaReportType["qoeExperienceCustomVar3"] = "30051";
    KalturaReportType["qoeExperienceEntry"] = "30012";
    KalturaReportType["qoeExperienceIsp"] = "30013";
    KalturaReportType["qoeExperienceOperatingSystem"] = "30010";
    KalturaReportType["qoeExperienceOperatingSystemFamilies"] = "30009";
    KalturaReportType["qoeExperiencePlatforms"] = "30003";
    KalturaReportType["qoeExperiencePlayerVersion"] = "30011";
    KalturaReportType["qoeExperienceRegion"] = "30005";
    KalturaReportType["qoeLiveSessionFlow"] = "30048";
    KalturaReportType["qoeOverview"] = "30001";
    KalturaReportType["qoeStreamQuality"] = "30026";
    KalturaReportType["qoeStreamQualityApplicationVersion"] = "30063";
    KalturaReportType["qoeStreamQualityBrowsers"] = "30032";
    KalturaReportType["qoeStreamQualityBrowsersFamilies"] = "30031";
    KalturaReportType["qoeStreamQualityCity"] = "30030";
    KalturaReportType["qoeStreamQualityCountry"] = "30028";
    KalturaReportType["qoeStreamQualityCustomVar1"] = "30055";
    KalturaReportType["qoeStreamQualityCustomVar2"] = "30056";
    KalturaReportType["qoeStreamQualityCustomVar3"] = "30057";
    KalturaReportType["qoeStreamQualityEntry"] = "30036";
    KalturaReportType["qoeStreamQualityIsp"] = "30037";
    KalturaReportType["qoeStreamQualityOperatingSystem"] = "30034";
    KalturaReportType["qoeStreamQualityOperatingSystemFamilies"] = "30033";
    KalturaReportType["qoeStreamQualityPlatforms"] = "30027";
    KalturaReportType["qoeStreamQualityPlayerVersion"] = "30035";
    KalturaReportType["qoeStreamQualityRegion"] = "30029";
    KalturaReportType["qoeVodSessionFlow"] = "30047";
    KalturaReportType["qosOverviewRealtime"] = "10006";
    KalturaReportType["qualityWebcast"] = "40003";
    KalturaReportType["quiz"] = "quiz.QUIZ";
    KalturaReportType["quizAggregateByQuestion"] = "quiz.QUIZ_AGGREGATE_BY_QUESTION";
    KalturaReportType["quizUserAggregateByQuestion"] = "quiz.QUIZ_USER_AGGREGATE_BY_QUESTION";
    KalturaReportType["quizUserPercentage"] = "quiz.QUIZ_USER_PERCENTAGE";
    KalturaReportType["reachCatalogUsage"] = "65";
    KalturaReportType["reachProfileUsage"] = "66";
    KalturaReportType["reachUsage"] = "28";
    KalturaReportType["reactionsBreakdownWebcast"] = "40013";
    KalturaReportType["selfServeBandwidth"] = "67";
    KalturaReportType["selfServeUsage"] = "60";
    KalturaReportType["selfServeUsageVpaas"] = "20023";
    KalturaReportType["specificUserEngagement"] = "12";
    KalturaReportType["specificUserUsage"] = "18";
    KalturaReportType["subCategories"] = "53";
    KalturaReportType["topContent"] = "1";
    KalturaReportType["topContentContributors"] = "39";
    KalturaReportType["topContentContributorsVpaas"] = "20016";
    KalturaReportType["topContentCreator"] = "38";
    KalturaReportType["topContentCreatorVpaas"] = "20015";
    KalturaReportType["topContributors"] = "5";
    KalturaReportType["topCreators"] = "20";
    KalturaReportType["topCustomVar1"] = "29";
    KalturaReportType["topCustomVar2"] = "58";
    KalturaReportType["topCustomVar3"] = "59";
    KalturaReportType["topDomainsWebcast"] = "40008";
    KalturaReportType["topEndedBroadcastEntries"] = "10014";
    KalturaReportType["topLiveNowEntries"] = "10015";
    KalturaReportType["topPlaybackContext"] = "25";
    KalturaReportType["topPlaybackContextVpaas"] = "20022";
    KalturaReportType["topSources"] = "41";
    KalturaReportType["topSourcesVpaas"] = "20017";
    KalturaReportType["topSyndication"] = "6";
    KalturaReportType["topSyndicationVpaas"] = "20002";
    KalturaReportType["topUsersWebcast"] = "40009";
    KalturaReportType["topUserContent"] = "47";
    KalturaReportType["uniqueUsersPlay"] = "35";
    KalturaReportType["uniqueUsersPlayVpaas"] = "20011";
    KalturaReportType["usersOverviewRealtime"] = "10005";
    KalturaReportType["userContentDropoff"] = "14";
    KalturaReportType["userContentInteractions"] = "15";
    KalturaReportType["userEngagement"] = "11";
    KalturaReportType["userEngagementTimeline"] = "34";
    KalturaReportType["userEngagementTimelineVpaas"] = "20010";
    KalturaReportType["userHighlights"] = "48";
    KalturaReportType["userInteractiveVideo"] = "49";
    KalturaReportType["userTopContent"] = "13";
    KalturaReportType["userTopContentVpaas"] = "20003";
    KalturaReportType["userUsage"] = "17";
    KalturaReportType["userUsageVpaas"] = "20004";
    KalturaReportType["varUsage"] = "19";
    KalturaReportType["veAttendanceHighlights"] = "50007";
    KalturaReportType["veHighlights"] = "50000";
    KalturaReportType["veRegisteredCompanies"] = "50008";
    KalturaReportType["veRegisteredCountries"] = "50004";
    KalturaReportType["veRegisteredIndustry"] = "50002";
    KalturaReportType["veRegisteredPlatforms"] = "50001";
    KalturaReportType["veRegisteredRoles"] = "50003";
    KalturaReportType["veRegisteredWorldRegions"] = "50005";
    KalturaReportType["veUserHighlights"] = "50006";
    KalturaReportType["vpaasUsage"] = "26";
    KalturaReportType["vpaasUsageMulti"] = "42";
})(KalturaReportType = exports.KalturaReportType || (exports.KalturaReportType = {}));
