// kaltura-node-typescript-client/api/types/KalturaAccessControlModifyRequestHostRegexAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAccessControlModifyRequestHostRegexAction = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaRuleAction_1 = require("./KalturaRuleAction");
var KalturaAccessControlModifyRequestHostRegexAction = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAccessControlModifyRequestHostRegexAction, _super);
    function KalturaAccessControlModifyRequestHostRegexAction(data) {
        return _super.call(this, data) || this;
    }
    KalturaAccessControlModifyRequestHostRegexAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAccessControlModifyRequestHostRegexAction' },
            pattern: { type: 's' },
            replacement: { type: 's' },
            replacmenServerNodeId: { type: 'n' },
            checkAliveTimeoutMs: { type: 'n' }
        });
        return result;
    };
    return KalturaAccessControlModifyRequestHostRegexAction;
}(KalturaRuleAction_1.KalturaRuleAction));
exports.KalturaAccessControlModifyRequestHostRegexAction = KalturaAccessControlModifyRequestHostRegexAction;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAccessControlModifyRequestHostRegexAction', KalturaAccessControlModifyRequestHostRegexAction);
