// kaltura-node-typescript-client/api/types/KalturaExtractMediaJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaExtractMediaJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConvartableJobData_1 = require("./KalturaConvartableJobData");
var KalturaExtractMediaJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaExtractMediaJobData, _super);
    function KalturaExtractMediaJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaExtractMediaJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaExtractMediaJobData' },
            flavorAssetId: { type: 's' },
            calculateComplexity: { type: 'b' },
            extractId3Tags: { type: 'b' },
            destDataFilePath: { type: 's' },
            detectGOP: { type: 'n' }
        });
        return result;
    };
    return KalturaExtractMediaJobData;
}(KalturaConvartableJobData_1.KalturaConvartableJobData));
exports.KalturaExtractMediaJobData = KalturaExtractMediaJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaExtractMediaJobData', KalturaExtractMediaJobData);
