// kaltura-node-typescript-client/api/types/KalturaWowzaMediaServerNode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaWowzaMediaServerNode = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaMediaServerNode_1 = require("./KalturaMediaServerNode");
var KalturaWowzaMediaServerNode = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaWowzaMediaServerNode, _super);
    function KalturaWowzaMediaServerNode(data) {
        return _super.call(this, data) || this;
    }
    KalturaWowzaMediaServerNode.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaWowzaMediaServerNode' },
            appPrefix: { type: 's' },
            transcoder: { type: 's' },
            GPUID: { type: 'n' },
            liveServicePort: { type: 'n' },
            liveServiceProtocol: { type: 's' },
            liveServiceInternalDomain: { type: 's' }
        });
        return result;
    };
    return KalturaWowzaMediaServerNode;
}(KalturaMediaServerNode_1.KalturaMediaServerNode));
exports.KalturaWowzaMediaServerNode = KalturaWowzaMediaServerNode;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaWowzaMediaServerNode', KalturaWowzaMediaServerNode);
