// kaltura-node-typescript-client/api/types/KalturaYahooDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaYahooDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaYahooDistributionProcessFeedActionStatus_1 = require("./KalturaYahooDistributionProcessFeedActionStatus");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaYahooDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaYahooDistributionProfile, _super);
    function KalturaYahooDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaYahooDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaYahooDistributionProfile' },
            ftpPath: { type: 's' },
            ftpUsername: { type: 's' },
            ftpPassword: { type: 's' },
            ftpHost: { type: 's' },
            contactTelephone: { type: 's' },
            contactEmail: { type: 's' },
            processFeed: { type: 'en', subTypeConstructor: KalturaYahooDistributionProcessFeedActionStatus_1.KalturaYahooDistributionProcessFeedActionStatus, subType: 'KalturaYahooDistributionProcessFeedActionStatus' }
        });
        return result;
    };
    return KalturaYahooDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaYahooDistributionProfile = KalturaYahooDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaYahooDistributionProfile', KalturaYahooDistributionProfile);
