// kaltura-node-typescript-client/api/types/KalturaLiveChannelSegmentBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveChannelSegmentBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaLiveChannelSegmentStatus_1 = require("./KalturaLiveChannelSegmentStatus");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaLiveChannelSegmentBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveChannelSegmentBaseFilter, _super);
    function KalturaLiveChannelSegmentBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaLiveChannelSegmentBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveChannelSegmentBaseFilter' },
            createdAtGreaterThanOrEqual: { type: 'n' },
            createdAtLessThanOrEqual: { type: 'n' },
            updatedAtGreaterThanOrEqual: { type: 'n' },
            updatedAtLessThanOrEqual: { type: 'n' },
            statusEqual: { type: 'es', subTypeConstructor: KalturaLiveChannelSegmentStatus_1.KalturaLiveChannelSegmentStatus, subType: 'KalturaLiveChannelSegmentStatus' },
            statusIn: { type: 's' },
            channelIdEqual: { type: 's' },
            channelIdIn: { type: 's' },
            startTimeGreaterThanOrEqual: { type: 'n' },
            startTimeLessThanOrEqual: { type: 'n' }
        });
        return result;
    };
    return KalturaLiveChannelSegmentBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaLiveChannelSegmentBaseFilter = KalturaLiveChannelSegmentBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveChannelSegmentBaseFilter', KalturaLiveChannelSegmentBaseFilter);
