// kaltura-node-typescript-client/api/types/KalturaCuePointOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCuePointOrderBy = void 0;
var KalturaCuePointOrderBy;
(function (KalturaCuePointOrderBy) {
    KalturaCuePointOrderBy["createdAtAsc"] = "+createdAt";
    KalturaCuePointOrderBy["createdAtDesc"] = "-createdAt";
    KalturaCuePointOrderBy["intIdAsc"] = "+intId";
    KalturaCuePointOrderBy["intIdDesc"] = "-intId";
    KalturaCuePointOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaCuePointOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaCuePointOrderBy["startTimeAsc"] = "+startTime";
    KalturaCuePointOrderBy["startTimeDesc"] = "-startTime";
    KalturaCuePointOrderBy["triggeredAtAsc"] = "+triggeredAt";
    KalturaCuePointOrderBy["triggeredAtDesc"] = "-triggeredAt";
    KalturaCuePointOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaCuePointOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaCuePointOrderBy = exports.KalturaCuePointOrderBy || (exports.KalturaCuePointOrderBy = {}));
