// kaltura-node-typescript-client/api/types/KalturaSso.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSso = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaSsoStatus_1 = require("./KalturaSsoStatus");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaSso = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaSso, _super);
    function KalturaSso(data) {
        return _super.call(this, data) || this;
    }
    KalturaSso.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaSso' },
            id: { type: 'n', readOnly: true },
            applicationType: { type: 's' },
            partnerId: { type: 'n' },
            domain: { type: 's' },
            status: { type: 'en', subTypeConstructor: KalturaSsoStatus_1.KalturaSsoStatus, subType: 'KalturaSsoStatus' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            redirectUrl: { type: 's' },
            data: { type: 's' }
        });
        return result;
    };
    return KalturaSso;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaSso = KalturaSso;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaSso', KalturaSso);
