// kaltura-node-typescript-client/api/types/KalturaMediaEntryOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMediaEntryOrderBy = void 0;
var KalturaMediaEntryOrderBy;
(function (KalturaMediaEntryOrderBy) {
    KalturaMediaEntryOrderBy["createdAtAsc"] = "+createdAt";
    KalturaMediaEntryOrderBy["createdAtDesc"] = "-createdAt";
    KalturaMediaEntryOrderBy["durationAsc"] = "+duration";
    KalturaMediaEntryOrderBy["durationDesc"] = "-duration";
    KalturaMediaEntryOrderBy["endDateAsc"] = "+endDate";
    KalturaMediaEntryOrderBy["endDateDesc"] = "-endDate";
    KalturaMediaEntryOrderBy["lastPlayedAtAsc"] = "+lastPlayedAt";
    KalturaMediaEntryOrderBy["lastPlayedAtDesc"] = "-lastPlayedAt";
    KalturaMediaEntryOrderBy["mediaTypeAsc"] = "+mediaType";
    KalturaMediaEntryOrderBy["mediaTypeDesc"] = "-mediaType";
    KalturaMediaEntryOrderBy["moderationCountAsc"] = "+moderationCount";
    KalturaMediaEntryOrderBy["moderationCountDesc"] = "-moderationCount";
    KalturaMediaEntryOrderBy["nameAsc"] = "+name";
    KalturaMediaEntryOrderBy["nameDesc"] = "-name";
    KalturaMediaEntryOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaMediaEntryOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaMediaEntryOrderBy["playsAsc"] = "+plays";
    KalturaMediaEntryOrderBy["playsDesc"] = "-plays";
    KalturaMediaEntryOrderBy["rankAsc"] = "+rank";
    KalturaMediaEntryOrderBy["rankDesc"] = "-rank";
    KalturaMediaEntryOrderBy["recentAsc"] = "+recent";
    KalturaMediaEntryOrderBy["recentDesc"] = "-recent";
    KalturaMediaEntryOrderBy["startDateAsc"] = "+startDate";
    KalturaMediaEntryOrderBy["startDateDesc"] = "-startDate";
    KalturaMediaEntryOrderBy["totalRankAsc"] = "+totalRank";
    KalturaMediaEntryOrderBy["totalRankDesc"] = "-totalRank";
    KalturaMediaEntryOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaMediaEntryOrderBy["updatedAtDesc"] = "-updatedAt";
    KalturaMediaEntryOrderBy["viewsAsc"] = "+views";
    KalturaMediaEntryOrderBy["viewsDesc"] = "-views";
    KalturaMediaEntryOrderBy["weightAsc"] = "+weight";
    KalturaMediaEntryOrderBy["weightDesc"] = "-weight";
})(KalturaMediaEntryOrderBy = exports.KalturaMediaEntryOrderBy || (exports.KalturaMediaEntryOrderBy = {}));
