// kaltura-node-typescript-client/api/types/KalturaProvisionJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaProvisionJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaProvisionJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaProvisionJobData, _super);
    function KalturaProvisionJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaProvisionJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaProvisionJobData' },
            streamID: { type: 's' },
            backupStreamID: { type: 's' },
            rtmp: { type: 's' },
            encoderIP: { type: 's' },
            backupEncoderIP: { type: 's' },
            encoderPassword: { type: 's' },
            encoderUsername: { type: 's' },
            endDate: { type: 'n' },
            returnVal: { type: 's' },
            mediaType: { type: 'n' },
            primaryBroadcastingUrl: { type: 's' },
            secondaryBroadcastingUrl: { type: 's' },
            streamName: { type: 's' }
        });
        return result;
    };
    return KalturaProvisionJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaProvisionJobData = KalturaProvisionJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaProvisionJobData', KalturaProvisionJobData);
