// kaltura-node-typescript-client/api/types/KalturaFileAsset.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFileAsset = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaFileAssetObjectType_1 = require("./KalturaFileAssetObjectType");
var KalturaFileAssetStatus_1 = require("./KalturaFileAssetStatus");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaFileAsset = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaFileAsset, _super);
    function KalturaFileAsset(data) {
        return _super.call(this, data) || this;
    }
    KalturaFileAsset.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaFileAsset' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            fileAssetObjectType: { type: 'es', subTypeConstructor: KalturaFileAssetObjectType_1.KalturaFileAssetObjectType, subType: 'KalturaFileAssetObjectType' },
            objectId: { type: 's' },
            name: { type: 's' },
            systemName: { type: 's' },
            fileExt: { type: 's' },
            version: { type: 'n', readOnly: true },
            createdAt: { type: 'n', readOnly: true },
            updatedAt: { type: 'n', readOnly: true },
            status: { type: 'es', readOnly: true, subTypeConstructor: KalturaFileAssetStatus_1.KalturaFileAssetStatus, subType: 'KalturaFileAssetStatus' }
        });
        return result;
    };
    return KalturaFileAsset;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaFileAsset = KalturaFileAsset;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaFileAsset', KalturaFileAsset);
