// kaltura-node-typescript-client/api/types/KalturaUserEntryBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUserEntryBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUserEntryStatus_1 = require("./KalturaUserEntryStatus");
var KalturaUserEntryType_1 = require("./KalturaUserEntryType");
var KalturaUserEntryExtendedStatus_1 = require("./KalturaUserEntryExtendedStatus");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaUserEntryBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUserEntryBaseFilter, _super);
    function KalturaUserEntryBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaUserEntryBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUserEntryBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            idNotIn: { type: 's' },
            entryIdEqual: { type: 's' },
            entryIdIn: { type: 's' },
            entryIdNotIn: { type: 's' },
            userIdEqual: { type: 's' },
            userIdIn: { type: 's' },
            userIdNotIn: { type: 's' },
            statusEqual: { type: 'es', subTypeConstructor: KalturaUserEntryStatus_1.KalturaUserEntryStatus, subType: 'KalturaUserEntryStatus' },
            createdAtLessThanOrEqual: { type: 'd' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            typeEqual: { type: 'es', subTypeConstructor: KalturaUserEntryType_1.KalturaUserEntryType, subType: 'KalturaUserEntryType' },
            extendedStatusEqual: { type: 'es', subTypeConstructor: KalturaUserEntryExtendedStatus_1.KalturaUserEntryExtendedStatus, subType: 'KalturaUserEntryExtendedStatus' },
            extendedStatusIn: { type: 's' },
            extendedStatusNotIn: { type: 's' }
        });
        return result;
    };
    return KalturaUserEntryBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaUserEntryBaseFilter = KalturaUserEntryBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUserEntryBaseFilter', KalturaUserEntryBaseFilter);
