// kaltura-node-typescript-client/api/types/KalturaUserAppRoleBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUserAppRoleBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaUserAppRoleBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUserAppRoleBaseFilter, _super);
    function KalturaUserAppRoleBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaUserAppRoleBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUserAppRoleBaseFilter' },
            appGuidEqual: { type: 's' },
            appGuidIn: { type: 's' },
            userRoleIdEqual: { type: 's' },
            userRoleIdIn: { type: 's' },
            createdAtLessThanOrEqual: { type: 'd' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' }
        });
        return result;
    };
    return KalturaUserAppRoleBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaUserAppRoleBaseFilter = KalturaUserAppRoleBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUserAppRoleBaseFilter', KalturaUserAppRoleBaseFilter);
