// kaltura-node-typescript-client/api/types/KalturaAccessControl.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAccessControl = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaBaseRestriction_1 = require("./KalturaBaseRestriction");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaAccessControl = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAccessControl, _super);
    function KalturaAccessControl(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.restrictions === 'undefined')
            _this.restrictions = [];
        return _this;
    }
    KalturaAccessControl.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAccessControl' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            name: { type: 's' },
            systemName: { type: 's' },
            description: { type: 's' },
            createdAt: { type: 'd', readOnly: true },
            isDefault: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            restrictions: { type: 'a', subTypeConstructor: KalturaBaseRestriction_1.KalturaBaseRestriction, subType: 'KalturaBaseRestriction' },
            containsUnsuportedRestrictions: { type: 'b', readOnly: true }
        });
        return result;
    };
    return KalturaAccessControl;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaAccessControl = KalturaAccessControl;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAccessControl', KalturaAccessControl);
