// kaltura-node-typescript-client/api/types/KalturaFlavorAsset.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFlavorAsset = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaFlavorAssetStatus_1 = require("./KalturaFlavorAssetStatus");
var KalturaLanguage_1 = require("./KalturaLanguage");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaAsset_1 = require("./KalturaAsset");
var KalturaFlavorAsset = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaFlavorAsset, _super);
    function KalturaFlavorAsset(data) {
        return _super.call(this, data) || this;
    }
    KalturaFlavorAsset.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaFlavorAsset' },
            flavorParamsId: { type: 'n' },
            width: { type: 'n', readOnly: true },
            height: { type: 'n', readOnly: true },
            bitrate: { type: 'n', readOnly: true },
            frameRate: { type: 'n', readOnly: true },
            isOriginal: { type: 'b', readOnly: true },
            isWeb: { type: 'b', readOnly: true },
            containerFormat: { type: 's', readOnly: true },
            videoCodecId: { type: 's', readOnly: true },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaFlavorAssetStatus_1.KalturaFlavorAssetStatus, subType: 'KalturaFlavorAssetStatus' },
            language: { type: 'es', subTypeConstructor: KalturaLanguage_1.KalturaLanguage, subType: 'KalturaLanguage' },
            label: { type: 's' },
            isDefault: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' }
        });
        return result;
    };
    return KalturaFlavorAsset;
}(KalturaAsset_1.KalturaAsset));
exports.KalturaFlavorAsset = KalturaFlavorAsset;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaFlavorAsset', KalturaFlavorAsset);
