// kaltura-node-typescript-client/api/types/KalturaObjectTaskType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaObjectTaskType = void 0;
var KalturaObjectTaskType;
(function (KalturaObjectTaskType) {
    KalturaObjectTaskType["convertEntryFlavors"] = "4";
    KalturaObjectTaskType["deleteEntry"] = "1";
    KalturaObjectTaskType["deleteEntryFlavors"] = "3";
    KalturaObjectTaskType["deleteLocalContent"] = "5";
    KalturaObjectTaskType["dispatchEventNotification"] = "scheduledTaskEventNotification.DispatchEventNotification";
    KalturaObjectTaskType["distribute"] = "scheduledTaskContentDistribution.Distribute";
    KalturaObjectTaskType["executeMetadataXslt"] = "scheduledTaskMetadata.ExecuteMetadataXslt";
    KalturaObjectTaskType["mailNotification"] = "8";
    KalturaObjectTaskType["modifyCategories"] = "2";
    KalturaObjectTaskType["modifyEntry"] = "7";
    KalturaObjectTaskType["recycleBinCleanup"] = "9";
    KalturaObjectTaskType["storageExport"] = "6";
})(KalturaObjectTaskType = exports.KalturaObjectTaskType || (exports.KalturaObjectTaskType = {}));
