// kaltura-node-typescript-client/api/types/KalturaGenericDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaGenericDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaGenericDistributionProfileAction_1 = require("./KalturaGenericDistributionProfileAction");
var KalturaDistributionProfile_1 = require("./KalturaDistributionProfile");
var KalturaGenericDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaGenericDistributionProfile, _super);
    function KalturaGenericDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaGenericDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaGenericDistributionProfile' },
            genericProviderId: { type: 'n' },
            submitAction: { type: 'o', subTypeConstructor: KalturaGenericDistributionProfileAction_1.KalturaGenericDistributionProfileAction, subType: 'KalturaGenericDistributionProfileAction' },
            updateAction: { type: 'o', subTypeConstructor: KalturaGenericDistributionProfileAction_1.KalturaGenericDistributionProfileAction, subType: 'KalturaGenericDistributionProfileAction' },
            deleteAction: { type: 'o', subTypeConstructor: KalturaGenericDistributionProfileAction_1.KalturaGenericDistributionProfileAction, subType: 'KalturaGenericDistributionProfileAction' },
            fetchReportAction: { type: 'o', subTypeConstructor: KalturaGenericDistributionProfileAction_1.KalturaGenericDistributionProfileAction, subType: 'KalturaGenericDistributionProfileAction' },
            updateRequiredEntryFields: { type: 's' },
            updateRequiredMetadataXPaths: { type: 's' }
        });
        return result;
    };
    return KalturaGenericDistributionProfile;
}(KalturaDistributionProfile_1.KalturaDistributionProfile));
exports.KalturaGenericDistributionProfile = KalturaGenericDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaGenericDistributionProfile', KalturaGenericDistributionProfile);
