// kaltura-node-typescript-client/api/types/KalturaDocumentEntryMatchAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDocumentEntryMatchAttribute = void 0;
var KalturaDocumentEntryMatchAttribute;
(function (KalturaDocumentEntryMatchAttribute) {
    KalturaDocumentEntryMatchAttribute["adminTags"] = "adminTags";
    KalturaDocumentEntryMatchAttribute["categoriesIds"] = "categoriesIds";
    KalturaDocumentEntryMatchAttribute["creatorId"] = "creatorId";
    KalturaDocumentEntryMatchAttribute["description"] = "description";
    KalturaDocumentEntryMatchAttribute["groupId"] = "groupId";
    KalturaDocumentEntryMatchAttribute["id"] = "id";
    KalturaDocumentEntryMatchAttribute["_name"] = "name";
    KalturaDocumentEntryMatchAttribute["referenceId"] = "referenceId";
    KalturaDocumentEntryMatchAttribute["replacedEntryId"] = "replacedEntryId";
    KalturaDocumentEntryMatchAttribute["replacingEntryId"] = "replacingEntryId";
    KalturaDocumentEntryMatchAttribute["searchText"] = "searchText";
    KalturaDocumentEntryMatchAttribute["tags"] = "tags";
    KalturaDocumentEntryMatchAttribute["userId"] = "userId";
})(KalturaDocumentEntryMatchAttribute = exports.KalturaDocumentEntryMatchAttribute || (exports.KalturaDocumentEntryMatchAttribute = {}));
