// kaltura-node-typescript-client/api/types/KalturaYouTubeApiCaptionDistributionInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaYouTubeApiCaptionDistributionInfo = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaYouTubeApiDistributionCaptionAction_1 = require("./KalturaYouTubeApiDistributionCaptionAction");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaYouTubeApiCaptionDistributionInfo = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaYouTubeApiCaptionDistributionInfo, _super);
    function KalturaYouTubeApiCaptionDistributionInfo(data) {
        return _super.call(this, data) || this;
    }
    KalturaYouTubeApiCaptionDistributionInfo.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaYouTubeApiCaptionDistributionInfo' },
            language: { type: 's' },
            label: { type: 's' },
            filePath: { type: 's' },
            remoteId: { type: 's' },
            action: { type: 'en', subTypeConstructor: KalturaYouTubeApiDistributionCaptionAction_1.KalturaYouTubeApiDistributionCaptionAction, subType: 'KalturaYouTubeApiDistributionCaptionAction' },
            version: { type: 's' },
            assetId: { type: 's' }
        });
        return result;
    };
    return KalturaYouTubeApiCaptionDistributionInfo;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaYouTubeApiCaptionDistributionInfo = KalturaYouTubeApiCaptionDistributionInfo;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaYouTubeApiCaptionDistributionInfo', KalturaYouTubeApiCaptionDistributionInfo);
