// kaltura-node-typescript-client/api/types/KalturaYahooSyndicationFeedOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaYahooSyndicationFeedOrderBy = void 0;
var KalturaYahooSyndicationFeedOrderBy;
(function (KalturaYahooSyndicationFeedOrderBy) {
    KalturaYahooSyndicationFeedOrderBy["createdAtAsc"] = "+createdAt";
    KalturaYahooSyndicationFeedOrderBy["createdAtDesc"] = "-createdAt";
    KalturaYahooSyndicationFeedOrderBy["nameAsc"] = "+name";
    KalturaYahooSyndicationFeedOrderBy["nameDesc"] = "-name";
    KalturaYahooSyndicationFeedOrderBy["playlistIdAsc"] = "+playlistId";
    KalturaYahooSyndicationFeedOrderBy["playlistIdDesc"] = "-playlistId";
    KalturaYahooSyndicationFeedOrderBy["typeAsc"] = "+type";
    KalturaYahooSyndicationFeedOrderBy["typeDesc"] = "-type";
    KalturaYahooSyndicationFeedOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaYahooSyndicationFeedOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaYahooSyndicationFeedOrderBy = exports.KalturaYahooSyndicationFeedOrderBy || (exports.KalturaYahooSyndicationFeedOrderBy = {}));
