// kaltura-node-typescript-client/api/types/KalturaConversionProfileAssetParamsBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaConversionProfileAssetParamsBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaFlavorReadyBehaviorType_1 = require("./KalturaFlavorReadyBehaviorType");
var KalturaAssetParamsOrigin_1 = require("./KalturaAssetParamsOrigin");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaConversionProfileAssetParamsBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaConversionProfileAssetParamsBaseFilter, _super);
    function KalturaConversionProfileAssetParamsBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaConversionProfileAssetParamsBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaConversionProfileAssetParamsBaseFilter' },
            conversionProfileIdEqual: { type: 'n' },
            conversionProfileIdIn: { type: 's' },
            assetParamsIdEqual: { type: 'n' },
            assetParamsIdIn: { type: 's' },
            readyBehaviorEqual: { type: 'en', subTypeConstructor: KalturaFlavorReadyBehaviorType_1.KalturaFlavorReadyBehaviorType, subType: 'KalturaFlavorReadyBehaviorType' },
            readyBehaviorIn: { type: 's' },
            originEqual: { type: 'en', subTypeConstructor: KalturaAssetParamsOrigin_1.KalturaAssetParamsOrigin, subType: 'KalturaAssetParamsOrigin' },
            originIn: { type: 's' },
            systemNameEqual: { type: 's' },
            systemNameIn: { type: 's' }
        });
        return result;
    };
    return KalturaConversionProfileAssetParamsBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaConversionProfileAssetParamsBaseFilter = KalturaConversionProfileAssetParamsBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaConversionProfileAssetParamsBaseFilter', KalturaConversionProfileAssetParamsBaseFilter);
