// kaltura-node-typescript-client/api/types/KalturaReportExportItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaReportExportItem = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaReportExportItemType_1 = require("./KalturaReportExportItemType");
var KalturaReportType_1 = require("./KalturaReportType");
var KalturaReportInputFilter_1 = require("./KalturaReportInputFilter");
var KalturaReportResponseOptions_1 = require("./KalturaReportResponseOptions");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaReportExportItem = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaReportExportItem, _super);
    function KalturaReportExportItem(data) {
        return _super.call(this, data) || this;
    }
    KalturaReportExportItem.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaReportExportItem' },
            reportTitle: { type: 's' },
            action: { type: 'en', subTypeConstructor: KalturaReportExportItemType_1.KalturaReportExportItemType, subType: 'KalturaReportExportItemType' },
            reportType: { type: 'es', subTypeConstructor: KalturaReportType_1.KalturaReportType, subType: 'KalturaReportType' },
            filter: { type: 'o', subTypeConstructor: KalturaReportInputFilter_1.KalturaReportInputFilter, subType: 'KalturaReportInputFilter' },
            order: { type: 's' },
            objectIds: { type: 's' },
            responseOptions: { type: 'o', subTypeConstructor: KalturaReportResponseOptions_1.KalturaReportResponseOptions, subType: 'KalturaReportResponseOptions' }
        });
        return result;
    };
    return KalturaReportExportItem;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaReportExportItem = KalturaReportExportItem;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaReportExportItem', KalturaReportExportItem);
