// kaltura-node-typescript-client/api/types/KalturaMoveCategoryEntriesJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMoveCategoryEntriesJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaMoveCategoryEntriesJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMoveCategoryEntriesJobData, _super);
    function KalturaMoveCategoryEntriesJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaMoveCategoryEntriesJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMoveCategoryEntriesJobData' },
            srcCategoryId: { type: 'n' },
            destCategoryId: { type: 'n' },
            lastMovedCategoryId: { type: 'n' },
            lastMovedCategoryPageIndex: { type: 'n' },
            lastMovedCategoryEntryPageIndex: { type: 'n' },
            moveFromChildren: { type: 'b' },
            destCategoryFullIds: { type: 's' }
        });
        return result;
    };
    return KalturaMoveCategoryEntriesJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaMoveCategoryEntriesJobData = KalturaMoveCategoryEntriesJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMoveCategoryEntriesJobData', KalturaMoveCategoryEntriesJobData);
