// kaltura-node-typescript-client/api/types/KalturaRecalculateResponseProfileCacheJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaRecalculateResponseProfileCacheJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaSessionType_1 = require("./KalturaSessionType");
var KalturaIntegerValue_1 = require("./KalturaIntegerValue");
var KalturaRecalculateCacheJobData_1 = require("./KalturaRecalculateCacheJobData");
var KalturaRecalculateResponseProfileCacheJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaRecalculateResponseProfileCacheJobData, _super);
    function KalturaRecalculateResponseProfileCacheJobData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.userRoles === 'undefined')
            _this.userRoles = [];
        return _this;
    }
    KalturaRecalculateResponseProfileCacheJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaRecalculateResponseProfileCacheJobData' },
            protocol: { type: 's' },
            ksType: { type: 'en', subTypeConstructor: KalturaSessionType_1.KalturaSessionType, subType: 'KalturaSessionType' },
            userRoles: { type: 'a', subTypeConstructor: KalturaIntegerValue_1.KalturaIntegerValue, subType: 'KalturaIntegerValue' },
            cachedObjectType: { type: 's' },
            objectId: { type: 's' },
            startObjectKey: { type: 's' },
            endObjectKey: { type: 's' }
        });
        return result;
    };
    return KalturaRecalculateResponseProfileCacheJobData;
}(KalturaRecalculateCacheJobData_1.KalturaRecalculateCacheJobData));
exports.KalturaRecalculateResponseProfileCacheJobData = KalturaRecalculateResponseProfileCacheJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaRecalculateResponseProfileCacheJobData', KalturaRecalculateResponseProfileCacheJobData);
