// kaltura-node-typescript-client/api/types/KalturaDataEntryCompareAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDataEntryCompareAttribute = void 0;
var KalturaDataEntryCompareAttribute;
(function (KalturaDataEntryCompareAttribute) {
    KalturaDataEntryCompareAttribute["accessControlId"] = "accessControlId";
    KalturaDataEntryCompareAttribute["createdAt"] = "createdAt";
    KalturaDataEntryCompareAttribute["endDate"] = "endDate";
    KalturaDataEntryCompareAttribute["moderationCount"] = "moderationCount";
    KalturaDataEntryCompareAttribute["moderationStatus"] = "moderationStatus";
    KalturaDataEntryCompareAttribute["partnerId"] = "partnerId";
    KalturaDataEntryCompareAttribute["partnerSortValue"] = "partnerSortValue";
    KalturaDataEntryCompareAttribute["rank"] = "rank";
    KalturaDataEntryCompareAttribute["replacementStatus"] = "replacementStatus";
    KalturaDataEntryCompareAttribute["startDate"] = "startDate";
    KalturaDataEntryCompareAttribute["status"] = "status";
    KalturaDataEntryCompareAttribute["totalRank"] = "totalRank";
    KalturaDataEntryCompareAttribute["type"] = "type";
    KalturaDataEntryCompareAttribute["updatedAt"] = "updatedAt";
})(KalturaDataEntryCompareAttribute = exports.KalturaDataEntryCompareAttribute || (exports.KalturaDataEntryCompareAttribute = {}));
