// kaltura-node-typescript-client/api/types/KalturaLiveReportExportType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveReportExportType = void 0;
var KalturaLiveReportExportType;
(function (KalturaLiveReportExportType) {
    KalturaLiveReportExportType[KalturaLiveReportExportType["entryTimeLineAll"] = 11] = "entryTimeLineAll";
    KalturaLiveReportExportType[KalturaLiveReportExportType["entryTimeLineLive"] = 12] = "entryTimeLineLive";
    KalturaLiveReportExportType[KalturaLiveReportExportType["locationAll"] = 21] = "locationAll";
    KalturaLiveReportExportType[KalturaLiveReportExportType["locationLive"] = 22] = "locationLive";
    KalturaLiveReportExportType[KalturaLiveReportExportType["partnerTotalAll"] = 1] = "partnerTotalAll";
    KalturaLiveReportExportType[KalturaLiveReportExportType["partnerTotalLive"] = 2] = "partnerTotalLive";
    KalturaLiveReportExportType[KalturaLiveReportExportType["syndicationAll"] = 31] = "syndicationAll";
    KalturaLiveReportExportType[KalturaLiveReportExportType["syndicationLive"] = 32] = "syndicationLive";
})(KalturaLiveReportExportType = exports.KalturaLiveReportExportType || (exports.KalturaLiveReportExportType = {}));
