// kaltura-node-typescript-client/api/types/KalturaESearchCaptionItemData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaESearchCaptionItemData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaESearchItemData_1 = require("./KalturaESearchItemData");
var KalturaESearchCaptionItemData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaESearchCaptionItemData, _super);
    function KalturaESearchCaptionItemData(data) {
        return _super.call(this, data) || this;
    }
    KalturaESearchCaptionItemData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaESearchCaptionItemData' },
            line: { type: 's' },
            startsAt: { type: 'n' },
            endsAt: { type: 'n' },
            language: { type: 's' },
            captionAssetId: { type: 's' },
            label: { type: 's' }
        });
        return result;
    };
    return KalturaESearchCaptionItemData;
}(KalturaESearchItemData_1.KalturaESearchItemData));
exports.KalturaESearchCaptionItemData = KalturaESearchCaptionItemData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaESearchCaptionItemData', KalturaESearchCaptionItemData);
