// kaltura-node-typescript-client/api/types/KalturaThumbParams.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaThumbParams = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaThumbCropType_1 = require("./KalturaThumbCropType");
var KalturaContainerFormat_1 = require("./KalturaContainerFormat");
var KalturaAssetParams_1 = require("./KalturaAssetParams");
var KalturaThumbParams = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaThumbParams, _super);
    function KalturaThumbParams(data) {
        return _super.call(this, data) || this;
    }
    KalturaThumbParams.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaThumbParams' },
            cropType: { type: 'en', subTypeConstructor: KalturaThumbCropType_1.KalturaThumbCropType, subType: 'KalturaThumbCropType' },
            quality: { type: 'n' },
            cropX: { type: 'n' },
            cropY: { type: 'n' },
            cropWidth: { type: 'n' },
            cropHeight: { type: 'n' },
            videoOffset: { type: 'n' },
            width: { type: 'n' },
            height: { type: 'n' },
            scaleWidth: { type: 'n' },
            scaleHeight: { type: 'n' },
            backgroundColor: { type: 's' },
            sourceParamsId: { type: 'n' },
            format: { type: 'es', subTypeConstructor: KalturaContainerFormat_1.KalturaContainerFormat, subType: 'KalturaContainerFormat' },
            density: { type: 'n' },
            stripProfiles: { type: 'b' },
            videoOffsetInPercentage: { type: 'n' },
            interval: { type: 'n' }
        });
        return result;
    };
    return KalturaThumbParams;
}(KalturaAssetParams_1.KalturaAssetParams));
exports.KalturaThumbParams = KalturaThumbParams;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaThumbParams', KalturaThumbParams);
