// kaltura-node-typescript-client/api/types/KalturaStatsEventType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaStatsEventType = void 0;
var KalturaStatsEventType;
(function (KalturaStatsEventType) {
    KalturaStatsEventType[KalturaStatsEventType["bufferEnd"] = 13] = "bufferEnd";
    KalturaStatsEventType[KalturaStatsEventType["bufferStart"] = 12] = "bufferStart";
    KalturaStatsEventType[KalturaStatsEventType["bumperClicked"] = 23] = "bumperClicked";
    KalturaStatsEventType[KalturaStatsEventType["closeEditor"] = 20] = "closeEditor";
    KalturaStatsEventType[KalturaStatsEventType["closeFullScreen"] = 15] = "closeFullScreen";
    KalturaStatsEventType[KalturaStatsEventType["mediaLoaded"] = 2] = "mediaLoaded";
    KalturaStatsEventType[KalturaStatsEventType["midroll25"] = 35] = "midroll25";
    KalturaStatsEventType[KalturaStatsEventType["midroll50"] = 36] = "midroll50";
    KalturaStatsEventType[KalturaStatsEventType["midroll75"] = 37] = "midroll75";
    KalturaStatsEventType[KalturaStatsEventType["midrollClicked"] = 29] = "midrollClicked";
    KalturaStatsEventType[KalturaStatsEventType["midrollStarted"] = 25] = "midrollStarted";
    KalturaStatsEventType[KalturaStatsEventType["openDownload"] = 10] = "openDownload";
    KalturaStatsEventType[KalturaStatsEventType["openEdit"] = 8] = "openEdit";
    KalturaStatsEventType[KalturaStatsEventType["openFullScreen"] = 14] = "openFullScreen";
    KalturaStatsEventType[KalturaStatsEventType["openReport"] = 11] = "openReport";
    KalturaStatsEventType[KalturaStatsEventType["openUpload"] = 18] = "openUpload";
    KalturaStatsEventType[KalturaStatsEventType["openViral"] = 9] = "openViral";
    KalturaStatsEventType[KalturaStatsEventType["overlayClicked"] = 31] = "overlayClicked";
    KalturaStatsEventType[KalturaStatsEventType["overlayStarted"] = 27] = "overlayStarted";
    KalturaStatsEventType[KalturaStatsEventType["play"] = 3] = "play";
    KalturaStatsEventType[KalturaStatsEventType["playReached100"] = 7] = "playReached100";
    KalturaStatsEventType[KalturaStatsEventType["playReached25"] = 4] = "playReached25";
    KalturaStatsEventType[KalturaStatsEventType["playReached50"] = 5] = "playReached50";
    KalturaStatsEventType[KalturaStatsEventType["playReached75"] = 6] = "playReached75";
    KalturaStatsEventType[KalturaStatsEventType["postroll25"] = 38] = "postroll25";
    KalturaStatsEventType[KalturaStatsEventType["postroll50"] = 39] = "postroll50";
    KalturaStatsEventType[KalturaStatsEventType["postroll75"] = 40] = "postroll75";
    KalturaStatsEventType[KalturaStatsEventType["postrollClicked"] = 30] = "postrollClicked";
    KalturaStatsEventType[KalturaStatsEventType["postrollStarted"] = 26] = "postrollStarted";
    KalturaStatsEventType[KalturaStatsEventType["postBumperPlayed"] = 22] = "postBumperPlayed";
    KalturaStatsEventType[KalturaStatsEventType["preroll25"] = 32] = "preroll25";
    KalturaStatsEventType[KalturaStatsEventType["preroll50"] = 33] = "preroll50";
    KalturaStatsEventType[KalturaStatsEventType["preroll75"] = 34] = "preroll75";
    KalturaStatsEventType[KalturaStatsEventType["prerollClicked"] = 28] = "prerollClicked";
    KalturaStatsEventType[KalturaStatsEventType["prerollStarted"] = 24] = "prerollStarted";
    KalturaStatsEventType[KalturaStatsEventType["preBumperPlayed"] = 21] = "preBumperPlayed";
    KalturaStatsEventType[KalturaStatsEventType["replay"] = 16] = "replay";
    KalturaStatsEventType[KalturaStatsEventType["savePublish"] = 19] = "savePublish";
    KalturaStatsEventType[KalturaStatsEventType["seek"] = 17] = "seek";
    KalturaStatsEventType[KalturaStatsEventType["widgetLoaded"] = 1] = "widgetLoaded";
})(KalturaStatsEventType = exports.KalturaStatsEventType || (exports.KalturaStatsEventType = {}));
