// kaltura-node-typescript-client/api/types/KalturaConversionProfileBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaConversionProfileBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConversionProfileStatus_1 = require("./KalturaConversionProfileStatus");
var KalturaConversionProfileType_1 = require("./KalturaConversionProfileType");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaConversionProfileBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaConversionProfileBaseFilter, _super);
    function KalturaConversionProfileBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaConversionProfileBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaConversionProfileBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            statusEqual: { type: 'es', subTypeConstructor: KalturaConversionProfileStatus_1.KalturaConversionProfileStatus, subType: 'KalturaConversionProfileStatus' },
            statusIn: { type: 's' },
            typeEqual: { type: 'es', subTypeConstructor: KalturaConversionProfileType_1.KalturaConversionProfileType, subType: 'KalturaConversionProfileType' },
            typeIn: { type: 's' },
            nameEqual: { type: 's' },
            systemNameEqual: { type: 's' },
            systemNameIn: { type: 's' },
            tagsMultiLikeOr: { type: 's' },
            tagsMultiLikeAnd: { type: 's' },
            defaultEntryIdEqual: { type: 's' },
            defaultEntryIdIn: { type: 's' }
        });
        return result;
    };
    return KalturaConversionProfileBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaConversionProfileBaseFilter = KalturaConversionProfileBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaConversionProfileBaseFilter', KalturaConversionProfileBaseFilter);
