// kaltura-node-typescript-client/api/types/KalturaBaseEntryBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBaseEntryBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaEntryStatus_1 = require("./KalturaEntryStatus");
var KalturaEntryModerationStatus_1 = require("./KalturaEntryModerationStatus");
var KalturaEntryType_1 = require("./KalturaEntryType");
var KalturaEntryReplacementStatus_1 = require("./KalturaEntryReplacementStatus");
var KalturaEntryDisplayInSearchType_1 = require("./KalturaEntryDisplayInSearchType");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaBaseEntryBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBaseEntryBaseFilter, _super);
    function KalturaBaseEntryBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaBaseEntryBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBaseEntryBaseFilter' },
            idEqual: { type: 's' },
            idIn: { type: 's' },
            idNotIn: { type: 's' },
            nameLike: { type: 's' },
            nameMultiLikeOr: { type: 's' },
            nameMultiLikeAnd: { type: 's' },
            nameEqual: { type: 's' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            userIdEqual: { type: 's' },
            userIdIn: { type: 's' },
            userIdNotIn: { type: 's' },
            creatorIdEqual: { type: 's' },
            tagsLike: { type: 's' },
            tagsMultiLikeOr: { type: 's' },
            tagsMultiLikeAnd: { type: 's' },
            adminTagsLike: { type: 's' },
            adminTagsMultiLikeOr: { type: 's' },
            adminTagsMultiLikeAnd: { type: 's' },
            categoriesMatchAnd: { type: 's' },
            categoriesMatchOr: { type: 's' },
            categoriesNotContains: { type: 's' },
            categoriesIdsMatchAnd: { type: 's' },
            categoriesIdsMatchOr: { type: 's' },
            categoriesIdsNotContains: { type: 's' },
            categoriesIdsEmpty: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            statusEqual: { type: 'es', subTypeConstructor: KalturaEntryStatus_1.KalturaEntryStatus, subType: 'KalturaEntryStatus' },
            statusNotEqual: { type: 'es', subTypeConstructor: KalturaEntryStatus_1.KalturaEntryStatus, subType: 'KalturaEntryStatus' },
            statusIn: { type: 's' },
            statusNotIn: { type: 's' },
            moderationStatusEqual: { type: 'en', subTypeConstructor: KalturaEntryModerationStatus_1.KalturaEntryModerationStatus, subType: 'KalturaEntryModerationStatus' },
            moderationStatusNotEqual: { type: 'en', subTypeConstructor: KalturaEntryModerationStatus_1.KalturaEntryModerationStatus, subType: 'KalturaEntryModerationStatus' },
            moderationStatusIn: { type: 's' },
            moderationStatusNotIn: { type: 's' },
            typeEqual: { type: 'es', subTypeConstructor: KalturaEntryType_1.KalturaEntryType, subType: 'KalturaEntryType' },
            typeIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            rankLessThanOrEqual: { type: 'n' },
            rankGreaterThanOrEqual: { type: 'n' },
            totalRankLessThanOrEqual: { type: 'n' },
            totalRankGreaterThanOrEqual: { type: 'n' },
            groupIdEqual: { type: 'n' },
            searchTextMatchAnd: { type: 's' },
            searchTextMatchOr: { type: 's' },
            accessControlIdEqual: { type: 'n' },
            accessControlIdIn: { type: 's' },
            startDateGreaterThanOrEqual: { type: 'd' },
            startDateLessThanOrEqual: { type: 'd' },
            startDateGreaterThanOrEqualOrNull: { type: 'd' },
            startDateLessThanOrEqualOrNull: { type: 'd' },
            endDateGreaterThanOrEqual: { type: 'd' },
            endDateLessThanOrEqual: { type: 'd' },
            endDateGreaterThanOrEqualOrNull: { type: 'd' },
            endDateLessThanOrEqualOrNull: { type: 'd' },
            referenceIdEqual: { type: 's' },
            referenceIdIn: { type: 's' },
            replacingEntryIdEqual: { type: 's' },
            replacingEntryIdIn: { type: 's' },
            replacedEntryIdEqual: { type: 's' },
            replacedEntryIdIn: { type: 's' },
            replacementStatusEqual: { type: 'es', subTypeConstructor: KalturaEntryReplacementStatus_1.KalturaEntryReplacementStatus, subType: 'KalturaEntryReplacementStatus' },
            replacementStatusIn: { type: 's' },
            partnerSortValueGreaterThanOrEqual: { type: 'n' },
            partnerSortValueLessThanOrEqual: { type: 'n' },
            rootEntryIdEqual: { type: 's' },
            rootEntryIdIn: { type: 's' },
            parentEntryIdEqual: { type: 's' },
            entitledUsersEditMatchAnd: { type: 's' },
            entitledUsersEditMatchOr: { type: 's' },
            entitledUsersPublishMatchAnd: { type: 's' },
            entitledUsersPublishMatchOr: { type: 's' },
            entitledUsersViewMatchAnd: { type: 's' },
            entitledUsersViewMatchOr: { type: 's' },
            tagsNameMultiLikeOr: { type: 's' },
            tagsAdminTagsMultiLikeOr: { type: 's' },
            tagsAdminTagsNameMultiLikeOr: { type: 's' },
            tagsNameMultiLikeAnd: { type: 's' },
            tagsAdminTagsMultiLikeAnd: { type: 's' },
            tagsAdminTagsNameMultiLikeAnd: { type: 's' },
            displayInSearchEqual: { type: 'en', subTypeConstructor: KalturaEntryDisplayInSearchType_1.KalturaEntryDisplayInSearchType, subType: 'KalturaEntryDisplayInSearchType' },
            displayInSearchIn: { type: 's' }
        });
        return result;
    };
    return KalturaBaseEntryBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaBaseEntryBaseFilter = KalturaBaseEntryBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBaseEntryBaseFilter', KalturaBaseEntryBaseFilter);
