// kaltura-node-typescript-client/api/types/KalturaExportCsvJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaExportCsvJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaExportCsvJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaExportCsvJobData, _super);
    function KalturaExportCsvJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaExportCsvJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaExportCsvJobData' },
            userName: { type: 's' },
            userMail: { type: 's' },
            outputPath: { type: 's' },
            sharedOutputPath: { type: 's' }
        });
        return result;
    };
    return KalturaExportCsvJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaExportCsvJobData = KalturaExportCsvJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaExportCsvJobData', KalturaExportCsvJobData);
