// kaltura-node-typescript-client/api/types/KalturaScheduler.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaScheduler = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaSchedulerStatus_1 = require("./KalturaSchedulerStatus");
var KalturaSchedulerConfig_1 = require("./KalturaSchedulerConfig");
var KalturaSchedulerWorker_1 = require("./KalturaSchedulerWorker");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaScheduler = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaScheduler, _super);
    function KalturaScheduler(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.statuses === 'undefined')
            _this.statuses = [];
        if (typeof _this.configs === 'undefined')
            _this.configs = [];
        if (typeof _this.workers === 'undefined')
            _this.workers = [];
        return _this;
    }
    KalturaScheduler.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaScheduler' },
            id: { type: 'n', readOnly: true },
            configuredId: { type: 'n' },
            name: { type: 's' },
            host: { type: 's' },
            statuses: { type: 'a', readOnly: true, subTypeConstructor: KalturaSchedulerStatus_1.KalturaSchedulerStatus, subType: 'KalturaSchedulerStatus' },
            configs: { type: 'a', readOnly: true, subTypeConstructor: KalturaSchedulerConfig_1.KalturaSchedulerConfig, subType: 'KalturaSchedulerConfig' },
            workers: { type: 'a', readOnly: true, subTypeConstructor: KalturaSchedulerWorker_1.KalturaSchedulerWorker, subType: 'KalturaSchedulerWorker' },
            createdAt: { type: 'd', readOnly: true },
            lastStatus: { type: 'n', readOnly: true },
            lastStatusStr: { type: 's', readOnly: true }
        });
        return result;
    };
    return KalturaScheduler;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaScheduler = KalturaScheduler;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaScheduler', KalturaScheduler);
