// kaltura-node-typescript-client/api/types/KalturaWebexDropFolderFile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaWebexDropFolderFile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDropFolderFile_1 = require("./KalturaDropFolderFile");
var KalturaWebexDropFolderFile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaWebexDropFolderFile, _super);
    function KalturaWebexDropFolderFile(data) {
        return _super.call(this, data) || this;
    }
    KalturaWebexDropFolderFile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaWebexDropFolderFile' },
            recordingId: { type: 'n' },
            webexHostId: { type: 's' },
            description: { type: 's' },
            confId: { type: 's' },
            contentUrl: { type: 's' }
        });
        return result;
    };
    return KalturaWebexDropFolderFile;
}(KalturaDropFolderFile_1.KalturaDropFolderFile));
exports.KalturaWebexDropFolderFile = KalturaWebexDropFolderFile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaWebexDropFolderFile', KalturaWebexDropFolderFile);
