// kaltura-node-typescript-client/api/types/KalturaEntryVendorTaskOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEntryVendorTaskOrderBy = void 0;
var KalturaEntryVendorTaskOrderBy;
(function (KalturaEntryVendorTaskOrderBy) {
    KalturaEntryVendorTaskOrderBy["createdAtAsc"] = "+createdAt";
    KalturaEntryVendorTaskOrderBy["createdAtDesc"] = "-createdAt";
    KalturaEntryVendorTaskOrderBy["expectedFinishTimeAsc"] = "+expectedFinishTime";
    KalturaEntryVendorTaskOrderBy["expectedFinishTimeDesc"] = "-expectedFinishTime";
    KalturaEntryVendorTaskOrderBy["finishTimeAsc"] = "+finishTime";
    KalturaEntryVendorTaskOrderBy["finishTimeDesc"] = "-finishTime";
    KalturaEntryVendorTaskOrderBy["idAsc"] = "+id";
    KalturaEntryVendorTaskOrderBy["idDesc"] = "-id";
    KalturaEntryVendorTaskOrderBy["priceAsc"] = "+price";
    KalturaEntryVendorTaskOrderBy["priceDesc"] = "-price";
    KalturaEntryVendorTaskOrderBy["queueTimeAsc"] = "+queueTime";
    KalturaEntryVendorTaskOrderBy["queueTimeDesc"] = "-queueTime";
    KalturaEntryVendorTaskOrderBy["statusAsc"] = "+status";
    KalturaEntryVendorTaskOrderBy["statusDesc"] = "-status";
    KalturaEntryVendorTaskOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaEntryVendorTaskOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaEntryVendorTaskOrderBy = exports.KalturaEntryVendorTaskOrderBy || (exports.KalturaEntryVendorTaskOrderBy = {}));
