// kaltura-node-typescript-client/api/types/KalturaAmazonS3StorageExportJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAmazonS3StorageExportJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaAmazonS3StorageProfileFilesPermissionLevel_1 = require("./KalturaAmazonS3StorageProfileFilesPermissionLevel");
var KalturaStorageExportJobData_1 = require("./KalturaStorageExportJobData");
var KalturaAmazonS3StorageExportJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAmazonS3StorageExportJobData, _super);
    function KalturaAmazonS3StorageExportJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaAmazonS3StorageExportJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAmazonS3StorageExportJobData' },
            filesPermissionInS3: { type: 'es', subTypeConstructor: KalturaAmazonS3StorageProfileFilesPermissionLevel_1.KalturaAmazonS3StorageProfileFilesPermissionLevel, subType: 'KalturaAmazonS3StorageProfileFilesPermissionLevel' },
            s3Region: { type: 's' },
            sseType: { type: 's' },
            sseKmsKeyId: { type: 's' },
            signatureType: { type: 's' },
            endPoint: { type: 's' },
            storageClass: { type: 's' }
        });
        return result;
    };
    return KalturaAmazonS3StorageExportJobData;
}(KalturaStorageExportJobData_1.KalturaStorageExportJobData));
exports.KalturaAmazonS3StorageExportJobData = KalturaAmazonS3StorageExportJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAmazonS3StorageExportJobData', KalturaAmazonS3StorageExportJobData);
