// kaltura-node-typescript-client/api/types/KalturaSchedulerStatusType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSchedulerStatusType = void 0;
var KalturaSchedulerStatusType;
(function (KalturaSchedulerStatusType) {
    KalturaSchedulerStatusType[KalturaSchedulerStatusType["runningBatchesCount"] = 1] = "runningBatchesCount";
    KalturaSchedulerStatusType[KalturaSchedulerStatusType["runningBatchesCpu"] = 2] = "runningBatchesCpu";
    KalturaSchedulerStatusType[KalturaSchedulerStatusType["runningBatchesDiscIo"] = 5] = "runningBatchesDiscIo";
    KalturaSchedulerStatusType[KalturaSchedulerStatusType["runningBatchesDiscSpace"] = 6] = "runningBatchesDiscSpace";
    KalturaSchedulerStatusType[KalturaSchedulerStatusType["runningBatchesIsRunning"] = 7] = "runningBatchesIsRunning";
    KalturaSchedulerStatusType[KalturaSchedulerStatusType["runningBatchesLastExecutionTime"] = 8] = "runningBatchesLastExecutionTime";
    KalturaSchedulerStatusType[KalturaSchedulerStatusType["runningBatchesMemory"] = 3] = "runningBatchesMemory";
    KalturaSchedulerStatusType[KalturaSchedulerStatusType["runningBatchesNetwork"] = 4] = "runningBatchesNetwork";
})(KalturaSchedulerStatusType = exports.KalturaSchedulerStatusType || (exports.KalturaSchedulerStatusType = {}));
