// kaltura-node-typescript-client/api/types/KalturaAppRole.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAppRole = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaAppRole = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAppRole, _super);
    function KalturaAppRole(data) {
        return _super.call(this, data) || this;
    }
    KalturaAppRole.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAppRole' },
            appGuid: { type: 's' },
            userRoleId: { type: 'n' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true }
        });
        return result;
    };
    return KalturaAppRole;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaAppRole = KalturaAppRole;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAppRole', KalturaAppRole);
