// kaltura-node-typescript-client/api/types/KalturaCuePointBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCuePointBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCuePointType_1 = require("./KalturaCuePointType");
var KalturaCuePointStatus_1 = require("./KalturaCuePointStatus");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaCuePointBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCuePointBaseFilter, _super);
    function KalturaCuePointBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaCuePointBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCuePointBaseFilter' },
            idEqual: { type: 's' },
            idIn: { type: 's' },
            cuePointTypeEqual: { type: 'es', subTypeConstructor: KalturaCuePointType_1.KalturaCuePointType, subType: 'KalturaCuePointType' },
            cuePointTypeIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaCuePointStatus_1.KalturaCuePointStatus, subType: 'KalturaCuePointStatus' },
            statusIn: { type: 's' },
            entryIdEqual: { type: 's' },
            entryIdIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            triggeredAtGreaterThanOrEqual: { type: 'd' },
            triggeredAtLessThanOrEqual: { type: 'd' },
            tagsLike: { type: 's' },
            tagsMultiLikeOr: { type: 's' },
            tagsMultiLikeAnd: { type: 's' },
            startTimeGreaterThanOrEqual: { type: 'n' },
            startTimeLessThanOrEqual: { type: 'n' },
            userIdEqual: { type: 's' },
            userIdIn: { type: 's' },
            partnerSortValueEqual: { type: 'n' },
            partnerSortValueIn: { type: 's' },
            partnerSortValueGreaterThanOrEqual: { type: 'n' },
            partnerSortValueLessThanOrEqual: { type: 'n' },
            forceStopEqual: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            systemNameEqual: { type: 's' },
            systemNameIn: { type: 's' }
        });
        return result;
    };
    return KalturaCuePointBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaCuePointBaseFilter = KalturaCuePointBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCuePointBaseFilter', KalturaCuePointBaseFilter);
