// kaltura-node-typescript-client/api/types/KalturaResponseProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaResponseProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaResponseProfileStatus_1 = require("./KalturaResponseProfileStatus");
var KalturaDetachedResponseProfile_1 = require("./KalturaDetachedResponseProfile");
var KalturaResponseProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaResponseProfile, _super);
    function KalturaResponseProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaResponseProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaResponseProfile' },
            id: { type: 'n', readOnly: true },
            systemName: { type: 's' },
            partnerId: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaResponseProfileStatus_1.KalturaResponseProfileStatus, subType: 'KalturaResponseProfileStatus' },
            version: { type: 'n', readOnly: true }
        });
        return result;
    };
    return KalturaResponseProfile;
}(KalturaDetachedResponseProfile_1.KalturaDetachedResponseProfile));
exports.KalturaResponseProfile = KalturaResponseProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaResponseProfile', KalturaResponseProfile);
