// kaltura-node-typescript-client/api/types/KalturaFacebookDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFacebookDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaFacebookDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaFacebookDistributionProfile, _super);
    function KalturaFacebookDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaFacebookDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaFacebookDistributionProfile' },
            apiAuthorizeUrl: { type: 's' },
            pageId: { type: 's' },
            pageAccessToken: { type: 's' },
            userAccessToken: { type: 's' },
            state: { type: 's' },
            permissions: { type: 's' },
            reRequestPermissions: { type: 'n' }
        });
        return result;
    };
    return KalturaFacebookDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaFacebookDistributionProfile = KalturaFacebookDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaFacebookDistributionProfile', KalturaFacebookDistributionProfile);
