// kaltura-node-typescript-client/api/types/KalturaZoomIntegrationSetting.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaZoomIntegrationSetting = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaZoomUsersMatching_1 = require("./KalturaZoomUsersMatching");
var KalturaZoomGroupParticipationType_1 = require("./KalturaZoomGroupParticipationType");
var KalturaHandleParticipantsMode_1 = require("./KalturaHandleParticipantsMode");
var KalturaIntegrationSetting_1 = require("./KalturaIntegrationSetting");
var KalturaZoomIntegrationSetting = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaZoomIntegrationSetting, _super);
    function KalturaZoomIntegrationSetting(data) {
        return _super.call(this, data) || this;
    }
    KalturaZoomIntegrationSetting.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaZoomIntegrationSetting' },
            zoomCategory: { type: 's' },
            enableRecordingUpload: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            zoomUserMatchingMode: { type: 'en', subTypeConstructor: KalturaZoomUsersMatching_1.KalturaZoomUsersMatching, subType: 'KalturaZoomUsersMatching' },
            zoomUserPostfix: { type: 's' },
            zoomWebinarCategory: { type: 's' },
            enableWebinarUploads: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            enableZoomTranscription: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            zoomAccountDescription: { type: 's' },
            optOutGroupNames: { type: 's' },
            optInGroupNames: { type: 's' },
            groupParticipationType: { type: 'en', subTypeConstructor: KalturaZoomGroupParticipationType_1.KalturaZoomGroupParticipationType, subType: 'KalturaZoomGroupParticipationType' },
            handleCohostsMode: { type: 'en', subTypeConstructor: KalturaHandleParticipantsMode_1.KalturaHandleParticipantsMode, subType: 'KalturaHandleParticipantsMode' },
            handleAlternativeHostsMode: { type: 'en', subTypeConstructor: KalturaHandleParticipantsMode_1.KalturaHandleParticipantsMode, subType: 'KalturaHandleParticipantsMode' }
        });
        return result;
    };
    return KalturaZoomIntegrationSetting;
}(KalturaIntegrationSetting_1.KalturaIntegrationSetting));
exports.KalturaZoomIntegrationSetting = KalturaZoomIntegrationSetting;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaZoomIntegrationSetting', KalturaZoomIntegrationSetting);
