// kaltura-node-typescript-client/api/types/KalturaAttUverseDistributionJobProviderData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAttUverseDistributionJobProviderData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaAttUverseDistributionFile_1 = require("./KalturaAttUverseDistributionFile");
var KalturaConfigurableDistributionJobProviderData_1 = require("./KalturaConfigurableDistributionJobProviderData");
var KalturaAttUverseDistributionJobProviderData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAttUverseDistributionJobProviderData, _super);
    function KalturaAttUverseDistributionJobProviderData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.filesForDistribution === 'undefined')
            _this.filesForDistribution = [];
        return _this;
    }
    KalturaAttUverseDistributionJobProviderData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAttUverseDistributionJobProviderData' },
            filesForDistribution: { type: 'a', subTypeConstructor: KalturaAttUverseDistributionFile_1.KalturaAttUverseDistributionFile, subType: 'KalturaAttUverseDistributionFile' },
            remoteAssetFileUrls: { type: 's' },
            remoteThumbnailFileUrls: { type: 's' },
            remoteCaptionFileUrls: { type: 's' }
        });
        return result;
    };
    return KalturaAttUverseDistributionJobProviderData;
}(KalturaConfigurableDistributionJobProviderData_1.KalturaConfigurableDistributionJobProviderData));
exports.KalturaAttUverseDistributionJobProviderData = KalturaAttUverseDistributionJobProviderData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAttUverseDistributionJobProviderData', KalturaAttUverseDistributionJobProviderData);
