// kaltura-node-typescript-client/api/types/KalturaUserFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUserFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaUserBaseFilter_1 = require("./KalturaUserBaseFilter");
var KalturaUserFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUserFilter, _super);
    function KalturaUserFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaUserFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUserFilter' },
            idOrScreenNameStartsWith: { type: 's' },
            loginEnabledEqual: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            roleIdEqual: { type: 's' },
            roleIdsEqual: { type: 's' },
            roleIdsIn: { type: 's' },
            firstNameOrLastNameStartsWith: { type: 's' },
            permissionNamesMultiLikeOr: { type: 's' },
            permissionNamesMultiLikeAnd: { type: 's' }
        });
        return result;
    };
    return KalturaUserFilter;
}(KalturaUserBaseFilter_1.KalturaUserBaseFilter));
exports.KalturaUserFilter = KalturaUserFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUserFilter', KalturaUserFilter);
