// kaltura-node-typescript-client/api/types/KalturaEntryDistributionOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEntryDistributionOrderBy = void 0;
var KalturaEntryDistributionOrderBy;
(function (KalturaEntryDistributionOrderBy) {
    KalturaEntryDistributionOrderBy["createdAtAsc"] = "+createdAt";
    KalturaEntryDistributionOrderBy["createdAtDesc"] = "-createdAt";
    KalturaEntryDistributionOrderBy["submittedAtAsc"] = "+submittedAt";
    KalturaEntryDistributionOrderBy["submittedAtDesc"] = "-submittedAt";
    KalturaEntryDistributionOrderBy["sunriseAsc"] = "+sunrise";
    KalturaEntryDistributionOrderBy["sunriseDesc"] = "-sunrise";
    KalturaEntryDistributionOrderBy["sunsetAsc"] = "+sunset";
    KalturaEntryDistributionOrderBy["sunsetDesc"] = "-sunset";
    KalturaEntryDistributionOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaEntryDistributionOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaEntryDistributionOrderBy = exports.KalturaEntryDistributionOrderBy || (exports.KalturaEntryDistributionOrderBy = {}));
