// kaltura-node-typescript-client/api/types/KalturaServerNode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaServerNode = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaServerNodeStatus_1 = require("./KalturaServerNodeStatus");
var KalturaServerNodeType_1 = require("./KalturaServerNodeType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaServerNode = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaServerNode, _super);
    function KalturaServerNode(data) {
        return _super.call(this, data) || this;
    }
    KalturaServerNode.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaServerNode' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            heartbeatTime: { type: 'd', readOnly: true },
            name: { type: 's' },
            systemName: { type: 's' },
            description: { type: 's' },
            hostName: { type: 's' },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaServerNodeStatus_1.KalturaServerNodeStatus, subType: 'KalturaServerNodeStatus' },
            type: { type: 'es', readOnly: true, subTypeConstructor: KalturaServerNodeType_1.KalturaServerNodeType, subType: 'KalturaServerNodeType' },
            tags: { type: 's' },
            dc: { type: 'n', readOnly: true },
            parentId: { type: 's' },
            environment: { type: 's' }
        });
        return result;
    };
    return KalturaServerNode;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaServerNode = KalturaServerNode;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaServerNode', KalturaServerNode);
