// kaltura-node-typescript-client/api/types/KalturaContentDistributionSearchItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaContentDistributionSearchItem = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaEntryDistributionSunStatus_1 = require("./KalturaEntryDistributionSunStatus");
var KalturaEntryDistributionFlag_1 = require("./KalturaEntryDistributionFlag");
var KalturaEntryDistributionStatus_1 = require("./KalturaEntryDistributionStatus");
var KalturaSearchItem_1 = require("./KalturaSearchItem");
var KalturaContentDistributionSearchItem = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaContentDistributionSearchItem, _super);
    function KalturaContentDistributionSearchItem(data) {
        return _super.call(this, data) || this;
    }
    KalturaContentDistributionSearchItem.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaContentDistributionSearchItem' },
            noDistributionProfiles: { type: 'b' },
            distributionProfileId: { type: 'n' },
            distributionSunStatus: { type: 'en', subTypeConstructor: KalturaEntryDistributionSunStatus_1.KalturaEntryDistributionSunStatus, subType: 'KalturaEntryDistributionSunStatus' },
            entryDistributionFlag: { type: 'en', subTypeConstructor: KalturaEntryDistributionFlag_1.KalturaEntryDistributionFlag, subType: 'KalturaEntryDistributionFlag' },
            entryDistributionStatus: { type: 'en', subTypeConstructor: KalturaEntryDistributionStatus_1.KalturaEntryDistributionStatus, subType: 'KalturaEntryDistributionStatus' },
            hasEntryDistributionValidationErrors: { type: 'b' },
            entryDistributionValidationErrors: { type: 's' }
        });
        return result;
    };
    return KalturaContentDistributionSearchItem;
}(KalturaSearchItem_1.KalturaSearchItem));
exports.KalturaContentDistributionSearchItem = KalturaContentDistributionSearchItem;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaContentDistributionSearchItem', KalturaContentDistributionSearchItem);
