// kaltura-node-typescript-client/api/types/KalturaDrmProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDrmProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDrmProviderType_1 = require("./KalturaDrmProviderType");
var KalturaDrmProfileStatus_1 = require("./KalturaDrmProfileStatus");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaDrmProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDrmProfile, _super);
    function KalturaDrmProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaDrmProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDrmProfile' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n' },
            name: { type: 's' },
            description: { type: 's' },
            provider: { type: 'es', subTypeConstructor: KalturaDrmProviderType_1.KalturaDrmProviderType, subType: 'KalturaDrmProviderType' },
            status: { type: 'en', subTypeConstructor: KalturaDrmProfileStatus_1.KalturaDrmProfileStatus, subType: 'KalturaDrmProfileStatus' },
            licenseServerUrl: { type: 's' },
            defaultPolicy: { type: 's' },
            createdAt: { type: 'n', readOnly: true },
            updatedAt: { type: 'n', readOnly: true },
            signingKey: { type: 's' }
        });
        return result;
    };
    return KalturaDrmProfile;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaDrmProfile = KalturaDrmProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDrmProfile', KalturaDrmProfile);
