// kaltura-node-typescript-client/api/types/KalturaReportExportJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaReportExportJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaReportExportItem_1 = require("./KalturaReportExportItem");
var KalturaReportExportFile_1 = require("./KalturaReportExportFile");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaReportExportJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaReportExportJobData, _super);
    function KalturaReportExportJobData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.reportItems === 'undefined')
            _this.reportItems = [];
        if (typeof _this.files === 'undefined')
            _this.files = [];
        return _this;
    }
    KalturaReportExportJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaReportExportJobData' },
            recipientEmail: { type: 's' },
            reportItems: { type: 'a', subTypeConstructor: KalturaReportExportItem_1.KalturaReportExportItem, subType: 'KalturaReportExportItem' },
            filePaths: { type: 's' },
            reportsGroup: { type: 's' },
            files: { type: 'a', subTypeConstructor: KalturaReportExportFile_1.KalturaReportExportFile, subType: 'KalturaReportExportFile' },
            baseUrl: { type: 's' }
        });
        return result;
    };
    return KalturaReportExportJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaReportExportJobData = KalturaReportExportJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaReportExportJobData', KalturaReportExportJobData);
