// kaltura-node-typescript-client/api/types/KalturaMultiClipConcatJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMultiClipConcatJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaChapterNamePolicy_1 = require("./KalturaChapterNamePolicy");
var KalturaCropAspectRatio_1 = require("./KalturaCropAspectRatio");
var KalturaOperationResource_1 = require("./KalturaOperationResource");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaMultiClipConcatJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMultiClipConcatJobData, _super);
    function KalturaMultiClipConcatJobData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.operationResources === 'undefined')
            _this.operationResources = [];
        return _this;
    }
    KalturaMultiClipConcatJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMultiClipConcatJobData' },
            destEntryId: { type: 's' },
            multiTempEntryId: { type: 's' },
            partnerId: { type: 'n' },
            priority: { type: 'n' },
            chapterNamePolicy: { type: 'en', subTypeConstructor: KalturaChapterNamePolicy_1.KalturaChapterNamePolicy, subType: 'KalturaChapterNamePolicy' },
            aspectRatio: { type: 'o', subTypeConstructor: KalturaCropAspectRatio_1.KalturaCropAspectRatio, subType: 'KalturaCropAspectRatio' },
            operationResources: { type: 'a', subTypeConstructor: KalturaOperationResource_1.KalturaOperationResource, subType: 'KalturaOperationResource' }
        });
        return result;
    };
    return KalturaMultiClipConcatJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaMultiClipConcatJobData = KalturaMultiClipConcatJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMultiClipConcatJobData', KalturaMultiClipConcatJobData);
