// kaltura-node-typescript-client/api/types/KalturaLiveEntryOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveEntryOrderBy = void 0;
var KalturaLiveEntryOrderBy;
(function (KalturaLiveEntryOrderBy) {
    KalturaLiveEntryOrderBy["createdAtAsc"] = "+createdAt";
    KalturaLiveEntryOrderBy["createdAtDesc"] = "-createdAt";
    KalturaLiveEntryOrderBy["durationAsc"] = "+duration";
    KalturaLiveEntryOrderBy["durationDesc"] = "-duration";
    KalturaLiveEntryOrderBy["endDateAsc"] = "+endDate";
    KalturaLiveEntryOrderBy["endDateDesc"] = "-endDate";
    KalturaLiveEntryOrderBy["firstBroadcastAsc"] = "+firstBroadcast";
    KalturaLiveEntryOrderBy["firstBroadcastDesc"] = "-firstBroadcast";
    KalturaLiveEntryOrderBy["lastBroadcastAsc"] = "+lastBroadcast";
    KalturaLiveEntryOrderBy["lastBroadcastDesc"] = "-lastBroadcast";
    KalturaLiveEntryOrderBy["lastPlayedAtAsc"] = "+lastPlayedAt";
    KalturaLiveEntryOrderBy["lastPlayedAtDesc"] = "-lastPlayedAt";
    KalturaLiveEntryOrderBy["mediaTypeAsc"] = "+mediaType";
    KalturaLiveEntryOrderBy["mediaTypeDesc"] = "-mediaType";
    KalturaLiveEntryOrderBy["moderationCountAsc"] = "+moderationCount";
    KalturaLiveEntryOrderBy["moderationCountDesc"] = "-moderationCount";
    KalturaLiveEntryOrderBy["nameAsc"] = "+name";
    KalturaLiveEntryOrderBy["nameDesc"] = "-name";
    KalturaLiveEntryOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaLiveEntryOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaLiveEntryOrderBy["playsAsc"] = "+plays";
    KalturaLiveEntryOrderBy["playsDesc"] = "-plays";
    KalturaLiveEntryOrderBy["rankAsc"] = "+rank";
    KalturaLiveEntryOrderBy["rankDesc"] = "-rank";
    KalturaLiveEntryOrderBy["recentAsc"] = "+recent";
    KalturaLiveEntryOrderBy["recentDesc"] = "-recent";
    KalturaLiveEntryOrderBy["startDateAsc"] = "+startDate";
    KalturaLiveEntryOrderBy["startDateDesc"] = "-startDate";
    KalturaLiveEntryOrderBy["totalRankAsc"] = "+totalRank";
    KalturaLiveEntryOrderBy["totalRankDesc"] = "-totalRank";
    KalturaLiveEntryOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaLiveEntryOrderBy["updatedAtDesc"] = "-updatedAt";
    KalturaLiveEntryOrderBy["viewsAsc"] = "+views";
    KalturaLiveEntryOrderBy["viewsDesc"] = "-views";
    KalturaLiveEntryOrderBy["weightAsc"] = "+weight";
    KalturaLiveEntryOrderBy["weightDesc"] = "-weight";
})(KalturaLiveEntryOrderBy = exports.KalturaLiveEntryOrderBy || (exports.KalturaLiveEntryOrderBy = {}));
