// kaltura-node-typescript-client/api/types/KalturaPartner.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPartner = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCommercialUseType_1 = require("./KalturaCommercialUseType");
var KalturaPartnerType_1 = require("./KalturaPartnerType");
var KalturaPartnerStatus_1 = require("./KalturaPartnerStatus");
var KalturaKeyValue_1 = require("./KalturaKeyValue");
var KalturaPartnerGroupType_1 = require("./KalturaPartnerGroupType");
var KalturaPlayerDeliveryType_1 = require("./KalturaPlayerDeliveryType");
var KalturaPlayerEmbedCodeType_1 = require("./KalturaPlayerEmbedCodeType");
var KalturaESearchLanguageItem_1 = require("./KalturaESearchLanguageItem");
var KalturaPartnerAuthenticationType_1 = require("./KalturaPartnerAuthenticationType");
var KalturaRegexItem_1 = require("./KalturaRegexItem");
var KalturaTwoFactorAuthenticationMode_1 = require("./KalturaTwoFactorAuthenticationMode");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaPartner = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPartner, _super);
    function KalturaPartner(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.additionalParams === 'undefined')
            _this.additionalParams = [];
        if (typeof _this.deliveryTypes === 'undefined')
            _this.deliveryTypes = [];
        if (typeof _this.embedCodeTypes === 'undefined')
            _this.embedCodeTypes = [];
        if (typeof _this.eSearchLanguages === 'undefined')
            _this.eSearchLanguages = [];
        if (typeof _this.passwordStructureValidations === 'undefined')
            _this.passwordStructureValidations = [];
        return _this;
    }
    KalturaPartner.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPartner' },
            id: { type: 'n', readOnly: true },
            name: { type: 's' },
            website: { type: 's' },
            notificationUrl: { type: 's' },
            appearInSearch: { type: 'n' },
            createdAt: { type: 'd', readOnly: true },
            adminName: { type: 's' },
            adminEmail: { type: 's' },
            description: { type: 's' },
            commercialUse: { type: 'en', subTypeConstructor: KalturaCommercialUseType_1.KalturaCommercialUseType, subType: 'KalturaCommercialUseType' },
            landingPage: { type: 's' },
            userLandingPage: { type: 's' },
            contentCategories: { type: 's' },
            type: { type: 'en', subTypeConstructor: KalturaPartnerType_1.KalturaPartnerType, subType: 'KalturaPartnerType' },
            phone: { type: 's' },
            describeYourself: { type: 's' },
            adultContent: { type: 'b' },
            defConversionProfileType: { type: 's' },
            notify: { type: 'n' },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaPartnerStatus_1.KalturaPartnerStatus, subType: 'KalturaPartnerStatus' },
            allowQuickEdit: { type: 'n' },
            mergeEntryLists: { type: 'n' },
            notificationsConfig: { type: 's' },
            allowedFromEmailWhiteList: { type: 's' },
            maxUploadSize: { type: 'n' },
            partnerPackage: { type: 'n' },
            secret: { type: 's', readOnly: true },
            adminSecret: { type: 's', readOnly: true },
            cmsPassword: { type: 's', readOnly: true },
            allowMultiNotification: { type: 'n' },
            adminLoginUsersQuota: { type: 'n', readOnly: true },
            adminUserId: { type: 's' },
            firstName: { type: 's' },
            lastName: { type: 's' },
            country: { type: 's' },
            state: { type: 's' },
            additionalParams: { type: 'a', subTypeConstructor: KalturaKeyValue_1.KalturaKeyValue, subType: 'KalturaKeyValue' },
            publishersQuota: { type: 'n', readOnly: true },
            partnerGroupType: { type: 'en', readOnly: true, subTypeConstructor: KalturaPartnerGroupType_1.KalturaPartnerGroupType, subType: 'KalturaPartnerGroupType' },
            defaultEntitlementEnforcement: { type: 'b', readOnly: true },
            defaultDeliveryType: { type: 's', readOnly: true },
            defaultEmbedCodeType: { type: 's', readOnly: true },
            deliveryTypes: { type: 'a', readOnly: true, subTypeConstructor: KalturaPlayerDeliveryType_1.KalturaPlayerDeliveryType, subType: 'KalturaPlayerDeliveryType' },
            embedCodeTypes: { type: 'a', readOnly: true, subTypeConstructor: KalturaPlayerEmbedCodeType_1.KalturaPlayerEmbedCodeType, subType: 'KalturaPlayerEmbedCodeType' },
            templatePartnerId: { type: 'n', readOnly: true },
            ignoreSeoLinks: { type: 'b', readOnly: true },
            useTwoFactorAuthentication: { type: 'b', readOnly: true },
            useSso: { type: 'b' },
            blockDirectLogin: { type: 'b' },
            host: { type: 's', readOnly: true },
            cdnHost: { type: 's', readOnly: true },
            isFirstLogin: { type: 'b', readOnly: true },
            logoutUrl: { type: 's', readOnly: true },
            partnerParentId: { type: 'n' },
            crmId: { type: 's', readOnly: true },
            referenceId: { type: 's' },
            timeAlignedRenditions: { type: 'b', readOnly: true },
            publisherEnvironmentType: { type: 'n', readOnly: true },
            ovpEnvironmentUrl: { type: 's', readOnly: true },
            ottEnvironmentUrl: { type: 's', readOnly: true },
            eSearchLanguages: { type: 'a', subTypeConstructor: KalturaESearchLanguageItem_1.KalturaESearchLanguageItem, subType: 'KalturaESearchLanguageItem' },
            authenticationType: { type: 'en', readOnly: true, subTypeConstructor: KalturaPartnerAuthenticationType_1.KalturaPartnerAuthenticationType, subType: 'KalturaPartnerAuthenticationType' },
            extendedFreeTrailExpiryReason: { type: 's', readOnly: true },
            extendedFreeTrailExpiryDate: { type: 'd', readOnly: true },
            extendedFreeTrail: { type: 'n', readOnly: true },
            extendedFreeTrailEndsWarning: { type: 'b', readOnly: true },
            eightyPercentWarning: { type: 'n', readOnly: true },
            usageLimitWarning: { type: 'n', readOnly: true },
            lastFreeTrialNotificationDay: { type: 'n', readOnly: true },
            monitorUsage: { type: 'n', readOnly: true },
            passwordStructureValidations: { type: 'a', subTypeConstructor: KalturaRegexItem_1.KalturaRegexItem, subType: 'KalturaRegexItem' },
            passwordStructureValidationsDescription: { type: 's' },
            passReplaceFreq: { type: 'n' },
            maxLoginAttempts: { type: 'n' },
            loginBlockPeriod: { type: 'n' },
            numPrevPassToKeep: { type: 'n' },
            allowDefaultPasswordRestrictions: { type: 'b', readOnly: true },
            twoFactorAuthenticationMode: { type: 'en', readOnly: true, subTypeConstructor: KalturaTwoFactorAuthenticationMode_1.KalturaTwoFactorAuthenticationMode, subType: 'KalturaTwoFactorAuthenticationMode' },
            isSelfServe: { type: 'b' },
            allowedDomains: { type: 's', readOnly: true },
            excludedAdminRoleName: { type: 's', readOnly: true },
            eventPlatformAllowedTemplates: { type: 's' },
            verticalClassificationId: { type: 'n', readOnly: true },
            recycleBinRetentionPeriod: { type: 'n', readOnly: true },
            customAnalyticsDomain: { type: 's' },
            allowedEmailDomainsForAdmins: { type: 's' }
        });
        return result;
    };
    return KalturaPartner;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaPartner = KalturaPartner;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPartner', KalturaPartner);
