// kaltura-node-typescript-client/api/types/KalturaBatchJobStatus.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBatchJobStatus = void 0;
var KalturaBatchJobStatus;
(function (KalturaBatchJobStatus) {
    KalturaBatchJobStatus[KalturaBatchJobStatus["aborted"] = 7] = "aborted";
    KalturaBatchJobStatus[KalturaBatchJobStatus["almostDone"] = 8] = "almostDone";
    KalturaBatchJobStatus[KalturaBatchJobStatus["dontProcess"] = 11] = "dontProcess";
    KalturaBatchJobStatus[KalturaBatchJobStatus["failed"] = 6] = "failed";
    KalturaBatchJobStatus[KalturaBatchJobStatus["fatal"] = 10] = "fatal";
    KalturaBatchJobStatus[KalturaBatchJobStatus["finished"] = 5] = "finished";
    KalturaBatchJobStatus[KalturaBatchJobStatus["finishedPartially"] = 12] = "finishedPartially";
    KalturaBatchJobStatus[KalturaBatchJobStatus["movefile"] = 4] = "movefile";
    KalturaBatchJobStatus[KalturaBatchJobStatus["pending"] = 0] = "pending";
    KalturaBatchJobStatus[KalturaBatchJobStatus["processed"] = 3] = "processed";
    KalturaBatchJobStatus[KalturaBatchJobStatus["processing"] = 2] = "processing";
    KalturaBatchJobStatus[KalturaBatchJobStatus["queued"] = 1] = "queued";
    KalturaBatchJobStatus[KalturaBatchJobStatus["retry"] = 9] = "retry";
    KalturaBatchJobStatus[KalturaBatchJobStatus["suspend"] = 13] = "suspend";
    KalturaBatchJobStatus[KalturaBatchJobStatus["suspendAlmostDone"] = 14] = "suspendAlmostDone";
})(KalturaBatchJobStatus = exports.KalturaBatchJobStatus || (exports.KalturaBatchJobStatus = {}));
