// kaltura-node-typescript-client/api/types/KalturaAuditTrailObjectType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAuditTrailObjectType = void 0;
var KalturaAuditTrailObjectType;
(function (KalturaAuditTrailObjectType) {
    KalturaAuditTrailObjectType["accessControl"] = "accessControl";
    KalturaAuditTrailObjectType["batchJob"] = "BatchJob";
    KalturaAuditTrailObjectType["category"] = "category";
    KalturaAuditTrailObjectType["categoryEntry"] = "categoryEntry";
    KalturaAuditTrailObjectType["conversionProfile2"] = "conversionProfile2";
    KalturaAuditTrailObjectType["emailIngestionProfile"] = "EmailIngestionProfile";
    KalturaAuditTrailObjectType["entry"] = "entry";
    KalturaAuditTrailObjectType["fileSync"] = "FileSync";
    KalturaAuditTrailObjectType["flavorAsset"] = "flavorAsset";
    KalturaAuditTrailObjectType["flavorParams"] = "flavorParams";
    KalturaAuditTrailObjectType["flavorParamsConversionProfile"] = "flavorParamsConversionProfile";
    KalturaAuditTrailObjectType["flavorParamsOutput"] = "flavorParamsOutput";
    KalturaAuditTrailObjectType["kshow"] = "kshow";
    KalturaAuditTrailObjectType["kshowKuser"] = "KshowKuser";
    KalturaAuditTrailObjectType["kuser"] = "kuser";
    KalturaAuditTrailObjectType["mediaInfo"] = "mediaInfo";
    KalturaAuditTrailObjectType["metadata"] = "Metadata";
    KalturaAuditTrailObjectType["metadataProfile"] = "MetadataProfile";
    KalturaAuditTrailObjectType["moderation"] = "moderation";
    KalturaAuditTrailObjectType["partner"] = "Partner";
    KalturaAuditTrailObjectType["permission"] = "Permission";
    KalturaAuditTrailObjectType["reachProfile"] = "ReachProfile";
    KalturaAuditTrailObjectType["roughcut"] = "roughcutEntry";
    KalturaAuditTrailObjectType["syndication"] = "syndicationFeed";
    KalturaAuditTrailObjectType["thumbnailAsset"] = "thumbAsset";
    KalturaAuditTrailObjectType["thumbnailParams"] = "thumbParams";
    KalturaAuditTrailObjectType["thumbnailParamsOutput"] = "thumbParamsOutput";
    KalturaAuditTrailObjectType["uiConf"] = "uiConf";
    KalturaAuditTrailObjectType["uploadToken"] = "UploadToken";
    KalturaAuditTrailObjectType["userLoginData"] = "UserLoginData";
    KalturaAuditTrailObjectType["userRole"] = "UserRole";
    KalturaAuditTrailObjectType["widget"] = "widget";
})(KalturaAuditTrailObjectType = exports.KalturaAuditTrailObjectType || (exports.KalturaAuditTrailObjectType = {}));
