// kaltura-node-typescript-client/api/types/KalturaIntegrationSetting.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaIntegrationSetting = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaVendorIntegrationStatus_1 = require("./KalturaVendorIntegrationStatus");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaHandleParticipantsMode_1 = require("./KalturaHandleParticipantsMode");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaIntegrationSetting = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaIntegrationSetting, _super);
    function KalturaIntegrationSetting(data) {
        return _super.call(this, data) || this;
    }
    KalturaIntegrationSetting.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaIntegrationSetting' },
            id: { type: 'n', readOnly: true },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaVendorIntegrationStatus_1.KalturaVendorIntegrationStatus, subType: 'KalturaVendorIntegrationStatus' },
            defaultUserId: { type: 's' },
            accountId: { type: 's', readOnly: true },
            createUserIfNotExist: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            conversionProfileId: { type: 'n' },
            handleParticipantsMode: { type: 'en', subTypeConstructor: KalturaHandleParticipantsMode_1.KalturaHandleParticipantsMode, subType: 'KalturaHandleParticipantsMode' },
            deletionPolicy: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            createdAt: { type: 's', readOnly: true },
            updatedAt: { type: 's', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            enableMeetingUpload: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            enableMeetingChat: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' }
        });
        return result;
    };
    return KalturaIntegrationSetting;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaIntegrationSetting = KalturaIntegrationSetting;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaIntegrationSetting', KalturaIntegrationSetting);
