// kaltura-node-typescript-client/api/types/KalturaSyndicationFeedType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSyndicationFeedType = void 0;
var KalturaSyndicationFeedType;
(function (KalturaSyndicationFeedType) {
    KalturaSyndicationFeedType[KalturaSyndicationFeedType["googleVideo"] = 1] = "googleVideo";
    KalturaSyndicationFeedType[KalturaSyndicationFeedType["itunes"] = 3] = "itunes";
    KalturaSyndicationFeedType[KalturaSyndicationFeedType["kaltura"] = 5] = "kaltura";
    KalturaSyndicationFeedType[KalturaSyndicationFeedType["kalturaXslt"] = 6] = "kalturaXslt";
    KalturaSyndicationFeedType[KalturaSyndicationFeedType["operaTvSnap"] = 8] = "operaTvSnap";
    KalturaSyndicationFeedType[KalturaSyndicationFeedType["rokuDirectPublisher"] = 7] = "rokuDirectPublisher";
    KalturaSyndicationFeedType[KalturaSyndicationFeedType["tubeMogul"] = 4] = "tubeMogul";
    KalturaSyndicationFeedType[KalturaSyndicationFeedType["yahoo"] = 2] = "yahoo";
})(KalturaSyndicationFeedType = exports.KalturaSyndicationFeedType || (exports.KalturaSyndicationFeedType = {}));
