// kaltura-node-typescript-client/api/types/KalturaNotificationJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaNotificationJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNotificationType_1 = require("./KalturaNotificationType");
var KalturaNotificationStatus_1 = require("./KalturaNotificationStatus");
var KalturaNotificationObjectType_1 = require("./KalturaNotificationObjectType");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaNotificationJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaNotificationJobData, _super);
    function KalturaNotificationJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaNotificationJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaNotificationJobData' },
            userId: { type: 's' },
            type: { type: 'en', subTypeConstructor: KalturaNotificationType_1.KalturaNotificationType, subType: 'KalturaNotificationType' },
            typeAsString: { type: 's' },
            objectId: { type: 's' },
            status: { type: 'en', subTypeConstructor: KalturaNotificationStatus_1.KalturaNotificationStatus, subType: 'KalturaNotificationStatus' },
            data: { type: 's' },
            numberOfAttempts: { type: 'n' },
            notificationResult: { type: 's' },
            objType: { type: 'en', subTypeConstructor: KalturaNotificationObjectType_1.KalturaNotificationObjectType, subType: 'KalturaNotificationObjectType' }
        });
        return result;
    };
    return KalturaNotificationJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaNotificationJobData = KalturaNotificationJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaNotificationJobData', KalturaNotificationJobData);
