// kaltura-node-typescript-client/api/types/KalturaPushNotificationTemplate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPushNotificationTemplate = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaPushEventNotificationParameter_1 = require("./KalturaPushEventNotificationParameter");
var KalturaResponseType_1 = require("./KalturaResponseType");
var KalturaEventNotificationTemplate_1 = require("./KalturaEventNotificationTemplate");
var KalturaPushNotificationTemplate = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPushNotificationTemplate, _super);
    function KalturaPushNotificationTemplate(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.queueNameParameters === 'undefined')
            _this.queueNameParameters = [];
        if (typeof _this.queueKeyParameters === 'undefined')
            _this.queueKeyParameters = [];
        return _this;
    }
    KalturaPushNotificationTemplate.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPushNotificationTemplate' },
            queueNameParameters: { type: 'a', subTypeConstructor: KalturaPushEventNotificationParameter_1.KalturaPushEventNotificationParameter, subType: 'KalturaPushEventNotificationParameter' },
            queueKeyParameters: { type: 'a', subTypeConstructor: KalturaPushEventNotificationParameter_1.KalturaPushEventNotificationParameter, subType: 'KalturaPushEventNotificationParameter' },
            apiObjectType: { type: 's' },
            objectFormat: { type: 'en', subTypeConstructor: KalturaResponseType_1.KalturaResponseType, subType: 'KalturaResponseType' },
            responseProfileId: { type: 'n' }
        });
        return result;
    };
    return KalturaPushNotificationTemplate;
}(KalturaEventNotificationTemplate_1.KalturaEventNotificationTemplate));
exports.KalturaPushNotificationTemplate = KalturaPushNotificationTemplate;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPushNotificationTemplate', KalturaPushNotificationTemplate);
