// kaltura-node-typescript-client/api/types/KalturaBulkUploadResultCategoryUser.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBulkUploadResultCategoryUser = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBulkUploadResult_1 = require("./KalturaBulkUploadResult");
var KalturaBulkUploadResultCategoryUser = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBulkUploadResultCategoryUser, _super);
    function KalturaBulkUploadResultCategoryUser(data) {
        return _super.call(this, data) || this;
    }
    KalturaBulkUploadResultCategoryUser.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBulkUploadResultCategoryUser' },
            categoryId: { type: 'n' },
            categoryReferenceId: { type: 's' },
            userId: { type: 's' },
            permissionLevel: { type: 'n' },
            updateMethod: { type: 'n' },
            requiredObjectStatus: { type: 'n' }
        });
        return result;
    };
    return KalturaBulkUploadResultCategoryUser;
}(KalturaBulkUploadResult_1.KalturaBulkUploadResult));
exports.KalturaBulkUploadResultCategoryUser = KalturaBulkUploadResultCategoryUser;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBulkUploadResultCategoryUser', KalturaBulkUploadResultCategoryUser);
