// kaltura-node-typescript-client/api/types/KalturaUserRole.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUserRole = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUserRoleStatus_1 = require("./KalturaUserRoleStatus");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaUserRole = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUserRole, _super);
    function KalturaUserRole(data) {
        return _super.call(this, data) || this;
    }
    KalturaUserRole.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUserRole' },
            id: { type: 'n', readOnly: true },
            name: { type: 's' },
            systemName: { type: 's' },
            description: { type: 's' },
            status: { type: 'en', subTypeConstructor: KalturaUserRoleStatus_1.KalturaUserRoleStatus, subType: 'KalturaUserRoleStatus' },
            partnerId: { type: 'n', readOnly: true },
            permissionNames: { type: 's' },
            tags: { type: 's' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true }
        });
        return result;
    };
    return KalturaUserRole;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaUserRole = KalturaUserRole;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUserRole', KalturaUserRole);
