// kaltura-node-typescript-client/api/types/KalturaBaseUserBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBaseUserBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUserStatus_1 = require("./KalturaUserStatus");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaBaseUserBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBaseUserBaseFilter, _super);
    function KalturaBaseUserBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaBaseUserBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBaseUserBaseFilter' },
            partnerIdEqual: { type: 'n' },
            screenNameLike: { type: 's' },
            screenNameStartsWith: { type: 's' },
            emailLike: { type: 's' },
            emailStartsWith: { type: 's' },
            tagsMultiLikeOr: { type: 's' },
            tagsMultiLikeAnd: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaUserStatus_1.KalturaUserStatus, subType: 'KalturaUserStatus' },
            statusIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' }
        });
        return result;
    };
    return KalturaBaseUserBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaBaseUserBaseFilter = KalturaBaseUserBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBaseUserBaseFilter', KalturaBaseUserBaseFilter);
