// kaltura-node-typescript-client/api/types/KalturaScheduledTaskProfileBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaScheduledTaskProfileBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaScheduledTaskProfileStatus_1 = require("./KalturaScheduledTaskProfileStatus");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaScheduledTaskProfileBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaScheduledTaskProfileBaseFilter, _super);
    function KalturaScheduledTaskProfileBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaScheduledTaskProfileBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaScheduledTaskProfileBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            systemNameEqual: { type: 's' },
            systemNameIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaScheduledTaskProfileStatus_1.KalturaScheduledTaskProfileStatus, subType: 'KalturaScheduledTaskProfileStatus' },
            statusIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            lastExecutionStartedAtGreaterThanOrEqual: { type: 'd' },
            lastExecutionStartedAtLessThanOrEqual: { type: 'd' },
            lastExecutionStartedAtLessThanOrEqualOrNull: { type: 'd' },
            objectFilterEngineTypeIn: { type: 's' }
        });
        return result;
    };
    return KalturaScheduledTaskProfileBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaScheduledTaskProfileBaseFilter = KalturaScheduledTaskProfileBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaScheduledTaskProfileBaseFilter', KalturaScheduledTaskProfileBaseFilter);
