// kaltura-node-typescript-client/api/types/KalturaLiveStreamConfiguration.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveStreamConfiguration = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaPlaybackProtocol_1 = require("./KalturaPlaybackProtocol");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaLiveStreamConfiguration = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveStreamConfiguration, _super);
    function KalturaLiveStreamConfiguration(data) {
        return _super.call(this, data) || this;
    }
    KalturaLiveStreamConfiguration.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveStreamConfiguration' },
            protocol: { type: 'es', subTypeConstructor: KalturaPlaybackProtocol_1.KalturaPlaybackProtocol, subType: 'KalturaPlaybackProtocol' },
            url: { type: 's' },
            publishUrl: { type: 's' },
            backupUrl: { type: 's' },
            streamName: { type: 's' }
        });
        return result;
    };
    return KalturaLiveStreamConfiguration;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaLiveStreamConfiguration = KalturaLiveStreamConfiguration;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveStreamConfiguration', KalturaLiveStreamConfiguration);
