// kaltura-node-typescript-client/api/types/KalturaMailType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMailType = void 0;
var KalturaMailType;
(function (KalturaMailType) {
    KalturaMailType["mailTypeAccountReactiveConfirm"] = "86";
    KalturaMailType["mailTypeAccountUpgradeConfirm"] = "80";
    KalturaMailType["mailTypeAddedKalturaToHisFavorites"] = "32";
    KalturaMailType["mailTypeAddedKalturaToHisFavoritesSubscribed"] = "42";
    KalturaMailType["mailTypeAddedKalturaToYourFavorites"] = "23";
    KalturaMailType["mailTypeAddedToClipFavorites"] = "12";
    KalturaMailType["mailTypeAddedToFavorites"] = "11";
    KalturaMailType["mailTypeBulkuploadAborted"] = "66";
    KalturaMailType["mailTypeBulkuploadFailed"] = "65";
    KalturaMailType["mailTypeBulkuploadFinished"] = "64";
    KalturaMailType["mailTypeBulkDownloadReady"] = "63";
    KalturaMailType["mailTypeClipAdded"] = "30";
    KalturaMailType["mailTypeClipAddedYourKaltura"] = "20";
    KalturaMailType["mailTypeClipContributed"] = "40";
    KalturaMailType["mailTypeExtendedFreeTrialEndsWarning"] = "87";
    KalturaMailType["mailTypeKalturaNewsletter"] = "10";
    KalturaMailType["mailTypeLiveReportExportAbort"] = "132";
    KalturaMailType["mailTypeLiveReportExportFailure"] = "131";
    KalturaMailType["mailTypeLiveReportExportSuccess"] = "130";
    KalturaMailType["mailTypeLoginMailReset"] = "52";
    KalturaMailType["mailTypeNewCommentInKaltura"] = "24";
    KalturaMailType["mailTypeNewCommentInKalturaYouContributed"] = "33";
    KalturaMailType["mailTypeNewCommentInKalturaYouSubscribed"] = "43";
    KalturaMailType["mailTypeNewCommentInProfile"] = "13";
    KalturaMailType["mailTypeNotifyErr"] = "70";
    KalturaMailType["mailTypeObjectsCsv"] = "135";
    KalturaMailType["mailTypePasswordReset"] = "51";
    KalturaMailType["mailTypeRegisterConfirm"] = "50";
    KalturaMailType["mailTypeRegisterConfirmVideoService"] = "54";
    KalturaMailType["mailTypeReportExportAbort"] = "138";
    KalturaMailType["mailTypeReportExportFailure"] = "137";
    KalturaMailType["mailTypeReportExportSuccess"] = "136";
    KalturaMailType["mailTypeRoughcutCreated"] = "22";
    KalturaMailType["mailTypeRoughcutCreatedSubscribed"] = "41";
    KalturaMailType["mailTypeSipFailure"] = "139";
    KalturaMailType["mailTypeSystemUserCredentialsSaved"] = "113";
    KalturaMailType["mailTypeSystemUserNewPassword"] = "112";
    KalturaMailType["mailTypeSystemUserResetPassword"] = "110";
    KalturaMailType["mailTypeSystemUserResetPasswordSuccess"] = "111";
    KalturaMailType["mailTypeUsersCsv"] = "133";
    KalturaMailType["mailTypeVideoAdded"] = "21";
    KalturaMailType["mailTypeVideoCreated"] = "31";
    KalturaMailType["mailTypeVideoIsReady"] = "62";
    KalturaMailType["mailTypeVideoReady"] = "60";
    KalturaMailType["mailTypeVideoServiceNotice"] = "81";
    KalturaMailType["mailTypeVideoServiceNoticeAccountDeleted"] = "84";
    KalturaMailType["mailTypeVideoServiceNoticeAccountLocked"] = "83";
    KalturaMailType["mailTypeVideoServiceNoticeLimitReached"] = "82";
    KalturaMailType["mailTypeVideoServiceNoticeUpgradeOffer"] = "85";
})(KalturaMailType = exports.KalturaMailType || (exports.KalturaMailType = {}));
