// kaltura-node-typescript-client/api/types/KalturaCEError.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCEError = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaCEError = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCEError, _super);
    function KalturaCEError(data) {
        return _super.call(this, data) || this;
    }
    KalturaCEError.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCEError' },
            id: { type: 's', readOnly: true },
            partnerId: { type: 'n' },
            browser: { type: 's' },
            serverIp: { type: 's' },
            serverOs: { type: 's' },
            phpVersion: { type: 's' },
            ceAdminEmail: { type: 's' },
            type: { type: 's' },
            description: { type: 's' },
            data: { type: 's' }
        });
        return result;
    };
    return KalturaCEError;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaCEError = KalturaCEError;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCEError', KalturaCEError);
