// kaltura-node-typescript-client/api/types/KalturaDropFolderFileOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDropFolderFileOrderBy = void 0;
var KalturaDropFolderFileOrderBy;
(function (KalturaDropFolderFileOrderBy) {
    KalturaDropFolderFileOrderBy["createdAtAsc"] = "+createdAt";
    KalturaDropFolderFileOrderBy["createdAtDesc"] = "-createdAt";
    KalturaDropFolderFileOrderBy["fileNameAsc"] = "+fileName";
    KalturaDropFolderFileOrderBy["fileNameDesc"] = "-fileName";
    KalturaDropFolderFileOrderBy["fileSizeAsc"] = "+fileSize";
    KalturaDropFolderFileOrderBy["fileSizeDesc"] = "-fileSize";
    KalturaDropFolderFileOrderBy["fileSizeLastSetAtAsc"] = "+fileSizeLastSetAt";
    KalturaDropFolderFileOrderBy["fileSizeLastSetAtDesc"] = "-fileSizeLastSetAt";
    KalturaDropFolderFileOrderBy["idAsc"] = "+id";
    KalturaDropFolderFileOrderBy["idDesc"] = "-id";
    KalturaDropFolderFileOrderBy["parsedFlavorAsc"] = "+parsedFlavor";
    KalturaDropFolderFileOrderBy["parsedFlavorDesc"] = "-parsedFlavor";
    KalturaDropFolderFileOrderBy["parsedSlugAsc"] = "+parsedSlug";
    KalturaDropFolderFileOrderBy["parsedSlugDesc"] = "-parsedSlug";
    KalturaDropFolderFileOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaDropFolderFileOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaDropFolderFileOrderBy = exports.KalturaDropFolderFileOrderBy || (exports.KalturaDropFolderFileOrderBy = {}));
