// kaltura-node-typescript-client/api/types/KalturaCategoryEntry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCategoryEntry = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCategoryEntryStatus_1 = require("./KalturaCategoryEntryStatus");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaCategoryEntry = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCategoryEntry, _super);
    function KalturaCategoryEntry(data) {
        return _super.call(this, data) || this;
    }
    KalturaCategoryEntry.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCategoryEntry' },
            categoryId: { type: 'n' },
            entryId: { type: 's' },
            createdAt: { type: 'd', readOnly: true },
            categoryFullIds: { type: 's', readOnly: true },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaCategoryEntryStatus_1.KalturaCategoryEntryStatus, subType: 'KalturaCategoryEntryStatus' },
            creatorUserId: { type: 's', readOnly: true }
        });
        return result;
    };
    return KalturaCategoryEntry;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaCategoryEntry = KalturaCategoryEntry;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCategoryEntry', KalturaCategoryEntry);
