// kaltura-node-typescript-client/api/types/KalturaConvertJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaConvertJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDestFileSyncDescriptor_1 = require("./KalturaDestFileSyncDescriptor");
var KalturaConvartableJobData_1 = require("./KalturaConvartableJobData");
var KalturaConvertJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaConvertJobData, _super);
    function KalturaConvertJobData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.extraDestFileSyncs === 'undefined')
            _this.extraDestFileSyncs = [];
        return _this;
    }
    KalturaConvertJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaConvertJobData' },
            destFileSyncLocalPath: { type: 's' },
            destFileSyncRemoteUrl: { type: 's' },
            logFileSyncLocalPath: { type: 's' },
            logFileSyncRemoteUrl: { type: 's' },
            flavorAssetId: { type: 's' },
            remoteMediaId: { type: 's' },
            customData: { type: 's' },
            extraDestFileSyncs: { type: 'a', subTypeConstructor: KalturaDestFileSyncDescriptor_1.KalturaDestFileSyncDescriptor, subType: 'KalturaDestFileSyncDescriptor' },
            engineMessage: { type: 's' },
            destFileSyncSharedPath: { type: 's' },
            userCpu: { type: 'n' },
            estimatedEffort: { type: 'n' }
        });
        return result;
    };
    return KalturaConvertJobData;
}(KalturaConvartableJobData_1.KalturaConvartableJobData));
exports.KalturaConvertJobData = KalturaConvertJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaConvertJobData', KalturaConvertJobData);
