// kaltura-node-typescript-client/api/types/KalturaVendorCatalogItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaVendorCatalogItem = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaVendorCatalogItemStatus_1 = require("./KalturaVendorCatalogItemStatus");
var KalturaVendorServiceType_1 = require("./KalturaVendorServiceType");
var KalturaVendorServiceFeature_1 = require("./KalturaVendorServiceFeature");
var KalturaVendorServiceTurnAroundTime_1 = require("./KalturaVendorServiceTurnAroundTime");
var KalturaVendorCatalogItemPricing_1 = require("./KalturaVendorCatalogItemPricing");
var KalturaReachVendorEngineType_1 = require("./KalturaReachVendorEngineType");
var KalturaCatalogItemLanguage_1 = require("./KalturaCatalogItemLanguage");
var KalturaVendorCatalogItemStage_1 = require("./KalturaVendorCatalogItemStage");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaVendorCatalogItem = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaVendorCatalogItem, _super);
    function KalturaVendorCatalogItem(data) {
        return _super.call(this, data) || this;
    }
    KalturaVendorCatalogItem.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaVendorCatalogItem' },
            id: { type: 'n', readOnly: true },
            vendorPartnerId: { type: 'n' },
            name: { type: 's' },
            systemName: { type: 's' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaVendorCatalogItemStatus_1.KalturaVendorCatalogItemStatus, subType: 'KalturaVendorCatalogItemStatus' },
            serviceType: { type: 'en', subTypeConstructor: KalturaVendorServiceType_1.KalturaVendorServiceType, subType: 'KalturaVendorServiceType' },
            serviceFeature: { type: 'en', readOnly: true, subTypeConstructor: KalturaVendorServiceFeature_1.KalturaVendorServiceFeature, subType: 'KalturaVendorServiceFeature' },
            turnAroundTime: { type: 'en', subTypeConstructor: KalturaVendorServiceTurnAroundTime_1.KalturaVendorServiceTurnAroundTime, subType: 'KalturaVendorServiceTurnAroundTime' },
            pricing: { type: 'o', subTypeConstructor: KalturaVendorCatalogItemPricing_1.KalturaVendorCatalogItemPricing, subType: 'KalturaVendorCatalogItemPricing' },
            engineType: { type: 'es', subTypeConstructor: KalturaReachVendorEngineType_1.KalturaReachVendorEngineType, subType: 'KalturaReachVendorEngineType' },
            sourceLanguage: { type: 'es', subTypeConstructor: KalturaCatalogItemLanguage_1.KalturaCatalogItemLanguage, subType: 'KalturaCatalogItemLanguage' },
            allowResubmission: { type: 'b' },
            vendorData: { type: 's' },
            stage: { type: 'en', subTypeConstructor: KalturaVendorCatalogItemStage_1.KalturaVendorCatalogItemStage, subType: 'KalturaVendorCatalogItemStage' },
            lastBulkUpdateId: { type: 'n' },
            contract: { type: 's' },
            createdBy: { type: 's' },
            notes: { type: 's' },
            partnerId: { type: 'n' }
        });
        return result;
    };
    return KalturaVendorCatalogItem;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaVendorCatalogItem = KalturaVendorCatalogItem;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaVendorCatalogItem', KalturaVendorCatalogItem);
