// kaltura-node-typescript-client/api/types/KalturaBulkUploadCsvJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBulkUploadCsvJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBulkUploadCsvVersion_1 = require("./KalturaBulkUploadCsvVersion");
var KalturaString_1 = require("./KalturaString");
var KalturaBulkUploadJobData_1 = require("./KalturaBulkUploadJobData");
var KalturaBulkUploadCsvJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBulkUploadCsvJobData, _super);
    function KalturaBulkUploadCsvJobData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.columns === 'undefined')
            _this.columns = [];
        return _this;
    }
    KalturaBulkUploadCsvJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBulkUploadCsvJobData' },
            csvVersion: { type: 'en', readOnly: true, subTypeConstructor: KalturaBulkUploadCsvVersion_1.KalturaBulkUploadCsvVersion, subType: 'KalturaBulkUploadCsvVersion' },
            columns: { type: 'a', subTypeConstructor: KalturaString_1.KalturaString, subType: 'KalturaString' },
            processObjectId: { type: 's' },
            processObjectType: { type: 's' }
        });
        return result;
    };
    return KalturaBulkUploadCsvJobData;
}(KalturaBulkUploadJobData_1.KalturaBulkUploadJobData));
exports.KalturaBulkUploadCsvJobData = KalturaBulkUploadCsvJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBulkUploadCsvJobData', KalturaBulkUploadCsvJobData);
