// kaltura-node-typescript-client/api/types/KalturaCortexApiDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCortexApiDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaCortexApiDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCortexApiDistributionProfile, _super);
    function KalturaCortexApiDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaCortexApiDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCortexApiDistributionProfile' },
            username: { type: 's' },
            host: { type: 's' },
            password: { type: 's' },
            folderrecordid: { type: 's' },
            metadataprofileid: { type: 's' },
            metadataprofileidpushing: { type: 's' }
        });
        return result;
    };
    return KalturaCortexApiDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaCortexApiDistributionProfile = KalturaCortexApiDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCortexApiDistributionProfile', KalturaCortexApiDistributionProfile);
