// kaltura-node-typescript-client/api/types/KalturaSearchProviderType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSearchProviderType = void 0;
var KalturaSearchProviderType;
(function (KalturaSearchProviderType) {
    KalturaSearchProviderType[KalturaSearchProviderType["archiveOrg"] = 22] = "archiveOrg";
    KalturaSearchProviderType[KalturaSearchProviderType["ccmixter"] = 10] = "ccmixter";
    KalturaSearchProviderType[KalturaSearchProviderType["current"] = 12] = "current";
    KalturaSearchProviderType[KalturaSearchProviderType["flickr"] = 3] = "flickr";
    KalturaSearchProviderType[KalturaSearchProviderType["jamendo"] = 9] = "jamendo";
    KalturaSearchProviderType[KalturaSearchProviderType["kaltura"] = 20] = "kaltura";
    KalturaSearchProviderType[KalturaSearchProviderType["kalturaPartner"] = 23] = "kalturaPartner";
    KalturaSearchProviderType[KalturaSearchProviderType["kalturaUserClips"] = 21] = "kalturaUserClips";
    KalturaSearchProviderType[KalturaSearchProviderType["mediaCommons"] = 13] = "mediaCommons";
    KalturaSearchProviderType[KalturaSearchProviderType["metacafe"] = 24] = "metacafe";
    KalturaSearchProviderType[KalturaSearchProviderType["myspace"] = 7] = "myspace";
    KalturaSearchProviderType[KalturaSearchProviderType["nypl"] = 11] = "nypl";
    KalturaSearchProviderType[KalturaSearchProviderType["partnerSpecific"] = 100] = "partnerSpecific";
    KalturaSearchProviderType[KalturaSearchProviderType["photobucket"] = 8] = "photobucket";
    KalturaSearchProviderType[KalturaSearchProviderType["searchProxy"] = 28] = "searchProxy";
    KalturaSearchProviderType[KalturaSearchProviderType["youtube"] = 4] = "youtube";
})(KalturaSearchProviderType = exports.KalturaSearchProviderType || (exports.KalturaSearchProviderType = {}));
