// kaltura-node-typescript-client/api/types/KalturaLiveChannelOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveChannelOrderBy = void 0;
var KalturaLiveChannelOrderBy;
(function (KalturaLiveChannelOrderBy) {
    KalturaLiveChannelOrderBy["createdAtAsc"] = "+createdAt";
    KalturaLiveChannelOrderBy["createdAtDesc"] = "-createdAt";
    KalturaLiveChannelOrderBy["durationAsc"] = "+duration";
    KalturaLiveChannelOrderBy["durationDesc"] = "-duration";
    KalturaLiveChannelOrderBy["endDateAsc"] = "+endDate";
    KalturaLiveChannelOrderBy["endDateDesc"] = "-endDate";
    KalturaLiveChannelOrderBy["firstBroadcastAsc"] = "+firstBroadcast";
    KalturaLiveChannelOrderBy["firstBroadcastDesc"] = "-firstBroadcast";
    KalturaLiveChannelOrderBy["lastBroadcastAsc"] = "+lastBroadcast";
    KalturaLiveChannelOrderBy["lastBroadcastDesc"] = "-lastBroadcast";
    KalturaLiveChannelOrderBy["lastPlayedAtAsc"] = "+lastPlayedAt";
    KalturaLiveChannelOrderBy["lastPlayedAtDesc"] = "-lastPlayedAt";
    KalturaLiveChannelOrderBy["mediaTypeAsc"] = "+mediaType";
    KalturaLiveChannelOrderBy["mediaTypeDesc"] = "-mediaType";
    KalturaLiveChannelOrderBy["moderationCountAsc"] = "+moderationCount";
    KalturaLiveChannelOrderBy["moderationCountDesc"] = "-moderationCount";
    KalturaLiveChannelOrderBy["nameAsc"] = "+name";
    KalturaLiveChannelOrderBy["nameDesc"] = "-name";
    KalturaLiveChannelOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaLiveChannelOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaLiveChannelOrderBy["playsAsc"] = "+plays";
    KalturaLiveChannelOrderBy["playsDesc"] = "-plays";
    KalturaLiveChannelOrderBy["rankAsc"] = "+rank";
    KalturaLiveChannelOrderBy["rankDesc"] = "-rank";
    KalturaLiveChannelOrderBy["recentAsc"] = "+recent";
    KalturaLiveChannelOrderBy["recentDesc"] = "-recent";
    KalturaLiveChannelOrderBy["startDateAsc"] = "+startDate";
    KalturaLiveChannelOrderBy["startDateDesc"] = "-startDate";
    KalturaLiveChannelOrderBy["totalRankAsc"] = "+totalRank";
    KalturaLiveChannelOrderBy["totalRankDesc"] = "-totalRank";
    KalturaLiveChannelOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaLiveChannelOrderBy["updatedAtDesc"] = "-updatedAt";
    KalturaLiveChannelOrderBy["viewsAsc"] = "+views";
    KalturaLiveChannelOrderBy["viewsDesc"] = "-views";
    KalturaLiveChannelOrderBy["weightAsc"] = "+weight";
    KalturaLiveChannelOrderBy["weightDesc"] = "-weight";
})(KalturaLiveChannelOrderBy = exports.KalturaLiveChannelOrderBy || (exports.KalturaLiveChannelOrderBy = {}));
