// kaltura-node-typescript-client/api/types/KalturaUnicornDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUnicornDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaUnicornDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUnicornDistributionProfile, _super);
    function KalturaUnicornDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaUnicornDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUnicornDistributionProfile' },
            username: { type: 's' },
            password: { type: 's' },
            domainName: { type: 's' },
            channelGuid: { type: 's' },
            apiHostUrl: { type: 's' },
            domainGuid: { type: 's' },
            adFreeApplicationGuid: { type: 's' },
            remoteAssetParamsId: { type: 'n' },
            storageProfileId: { type: 's' }
        });
        return result;
    };
    return KalturaUnicornDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaUnicornDistributionProfile = KalturaUnicornDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUnicornDistributionProfile', KalturaUnicornDistributionProfile);
