// kaltura-node-typescript-client/api/types/KalturaMailNotificationObjectTask.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMailNotificationObjectTask = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaObjectTask_1 = require("./KalturaObjectTask");
var KalturaMailNotificationObjectTask = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMailNotificationObjectTask, _super);
    function KalturaMailNotificationObjectTask(data) {
        return _super.call(this, data) || this;
    }
    KalturaMailNotificationObjectTask.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMailNotificationObjectTask' },
            mailTo: { type: 's' },
            sender: { type: 's' },
            subject: { type: 's' },
            message: { type: 's' },
            footer: { type: 's' },
            link: { type: 's' },
            sendToUsers: { type: 'b' }
        });
        return result;
    };
    return KalturaMailNotificationObjectTask;
}(KalturaObjectTask_1.KalturaObjectTask));
exports.KalturaMailNotificationObjectTask = KalturaMailNotificationObjectTask;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMailNotificationObjectTask', KalturaMailNotificationObjectTask);
