// kaltura-node-typescript-client/api/types/KalturaVirtualEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaVirtualEvent = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaVirtualEventStatus_1 = require("./KalturaVirtualEventStatus");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaVirtualEvent = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaVirtualEvent, _super);
    function KalturaVirtualEvent(data) {
        return _super.call(this, data) || this;
    }
    KalturaVirtualEvent.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaVirtualEvent' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            name: { type: 's' },
            description: { type: 's' },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaVirtualEventStatus_1.KalturaVirtualEventStatus, subType: 'KalturaVirtualEventStatus' },
            tags: { type: 's' },
            attendeesGroupIds: { type: 's' },
            adminsGroupIds: { type: 's' },
            registrationScheduleEventId: { type: 'n' },
            agendaScheduleEventId: { type: 'n' },
            mainEventScheduleEventId: { type: 'n' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            deletionDueDate: { type: 'd' },
            registrationFormSchema: { type: 's' },
            eventUrl: { type: 's' },
            webhookRegistrationToken: { type: 's' }
        });
        return result;
    };
    return KalturaVirtualEvent;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaVirtualEvent = KalturaVirtualEvent;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaVirtualEvent', KalturaVirtualEvent);
