// kaltura-node-typescript-client/api/types/KalturaCodeCuePointOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCodeCuePointOrderBy = void 0;
var KalturaCodeCuePointOrderBy;
(function (KalturaCodeCuePointOrderBy) {
    KalturaCodeCuePointOrderBy["createdAtAsc"] = "+createdAt";
    KalturaCodeCuePointOrderBy["createdAtDesc"] = "-createdAt";
    KalturaCodeCuePointOrderBy["durationAsc"] = "+duration";
    KalturaCodeCuePointOrderBy["durationDesc"] = "-duration";
    KalturaCodeCuePointOrderBy["endTimeAsc"] = "+endTime";
    KalturaCodeCuePointOrderBy["endTimeDesc"] = "-endTime";
    KalturaCodeCuePointOrderBy["intIdAsc"] = "+intId";
    KalturaCodeCuePointOrderBy["intIdDesc"] = "-intId";
    KalturaCodeCuePointOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaCodeCuePointOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaCodeCuePointOrderBy["startTimeAsc"] = "+startTime";
    KalturaCodeCuePointOrderBy["startTimeDesc"] = "-startTime";
    KalturaCodeCuePointOrderBy["triggeredAtAsc"] = "+triggeredAt";
    KalturaCodeCuePointOrderBy["triggeredAtDesc"] = "-triggeredAt";
    KalturaCodeCuePointOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaCodeCuePointOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaCodeCuePointOrderBy = exports.KalturaCodeCuePointOrderBy || (exports.KalturaCodeCuePointOrderBy = {}));
