// kaltura-node-typescript-client/api/types/KalturaCortexApiCaptionDistributionInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCortexApiCaptionDistributionInfo = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCortexApiDistributionCaptionAction_1 = require("./KalturaCortexApiDistributionCaptionAction");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaCortexApiCaptionDistributionInfo = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCortexApiCaptionDistributionInfo, _super);
    function KalturaCortexApiCaptionDistributionInfo(data) {
        return _super.call(this, data) || this;
    }
    KalturaCortexApiCaptionDistributionInfo.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCortexApiCaptionDistributionInfo' },
            language: { type: 's' },
            label: { type: 's' },
            filePath: { type: 's' },
            remoteId: { type: 's' },
            action: { type: 'en', subTypeConstructor: KalturaCortexApiDistributionCaptionAction_1.KalturaCortexApiDistributionCaptionAction, subType: 'KalturaCortexApiDistributionCaptionAction' },
            version: { type: 's' },
            assetId: { type: 's' },
            fileExt: { type: 's' }
        });
        return result;
    };
    return KalturaCortexApiCaptionDistributionInfo;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaCortexApiCaptionDistributionInfo = KalturaCortexApiCaptionDistributionInfo;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCortexApiCaptionDistributionInfo', KalturaCortexApiCaptionDistributionInfo);
