// kaltura-node-typescript-client/api/types/KalturaVirtualEventBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaVirtualEventBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaVirtualEventStatus_1 = require("./KalturaVirtualEventStatus");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaVirtualEventBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaVirtualEventBaseFilter, _super);
    function KalturaVirtualEventBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaVirtualEventBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaVirtualEventBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            idNotIn: { type: 's' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaVirtualEventStatus_1.KalturaVirtualEventStatus, subType: 'KalturaVirtualEventStatus' },
            statusIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' }
        });
        return result;
    };
    return KalturaVirtualEventBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaVirtualEventBaseFilter = KalturaVirtualEventBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaVirtualEventBaseFilter', KalturaVirtualEventBaseFilter);
