// kaltura-node-typescript-client/api/types/KalturaTVComDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaTVComDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaTVComDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaTVComDistributionProfile, _super);
    function KalturaTVComDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaTVComDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaTVComDistributionProfile' },
            metadataProfileId: { type: 'n' },
            feedUrl: { type: 's', readOnly: true },
            feedTitle: { type: 's' },
            feedLink: { type: 's' },
            feedDescription: { type: 's' },
            feedLanguage: { type: 's' },
            feedCopyright: { type: 's' },
            feedImageTitle: { type: 's' },
            feedImageUrl: { type: 's' },
            feedImageLink: { type: 's' },
            feedImageWidth: { type: 'n' },
            feedImageHeight: { type: 'n' }
        });
        return result;
    };
    return KalturaTVComDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaTVComDistributionProfile = KalturaTVComDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaTVComDistributionProfile', KalturaTVComDistributionProfile);
