// kaltura-node-typescript-client/api/types/KalturaLiveStatsEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveStatsEvent = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaLiveStatsEventType_1 = require("./KalturaLiveStatsEventType");
var KalturaPlaybackProtocol_1 = require("./KalturaPlaybackProtocol");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaLiveStatsEvent = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveStatsEvent, _super);
    function KalturaLiveStatsEvent(data) {
        return _super.call(this, data) || this;
    }
    KalturaLiveStatsEvent.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveStatsEvent' },
            partnerId: { type: 'n' },
            entryId: { type: 's' },
            eventType: { type: 'en', subTypeConstructor: KalturaLiveStatsEventType_1.KalturaLiveStatsEventType, subType: 'KalturaLiveStatsEventType' },
            sessionId: { type: 's' },
            eventIndex: { type: 'n' },
            bufferTime: { type: 'n' },
            bitrate: { type: 'n' },
            referrer: { type: 's' },
            isLive: { type: 'b' },
            startTime: { type: 's' },
            deliveryType: { type: 'es', subTypeConstructor: KalturaPlaybackProtocol_1.KalturaPlaybackProtocol, subType: 'KalturaPlaybackProtocol' }
        });
        return result;
    };
    return KalturaLiveStatsEvent;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaLiveStatsEvent = KalturaLiveStatsEvent;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveStatsEvent', KalturaLiveStatsEvent);
