// kaltura-node-typescript-client/api/types/KalturaLiveStreamAdminEntryCompareAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveStreamAdminEntryCompareAttribute = void 0;
var KalturaLiveStreamAdminEntryCompareAttribute;
(function (KalturaLiveStreamAdminEntryCompareAttribute) {
    KalturaLiveStreamAdminEntryCompareAttribute["accessControlId"] = "accessControlId";
    KalturaLiveStreamAdminEntryCompareAttribute["createdAt"] = "createdAt";
    KalturaLiveStreamAdminEntryCompareAttribute["endDate"] = "endDate";
    KalturaLiveStreamAdminEntryCompareAttribute["lastPlayedAt"] = "lastPlayedAt";
    KalturaLiveStreamAdminEntryCompareAttribute["mediaDate"] = "mediaDate";
    KalturaLiveStreamAdminEntryCompareAttribute["mediaType"] = "mediaType";
    KalturaLiveStreamAdminEntryCompareAttribute["moderationCount"] = "moderationCount";
    KalturaLiveStreamAdminEntryCompareAttribute["moderationStatus"] = "moderationStatus";
    KalturaLiveStreamAdminEntryCompareAttribute["msDuration"] = "msDuration";
    KalturaLiveStreamAdminEntryCompareAttribute["partnerId"] = "partnerId";
    KalturaLiveStreamAdminEntryCompareAttribute["partnerSortValue"] = "partnerSortValue";
    KalturaLiveStreamAdminEntryCompareAttribute["plays"] = "plays";
    KalturaLiveStreamAdminEntryCompareAttribute["rank"] = "rank";
    KalturaLiveStreamAdminEntryCompareAttribute["replacementStatus"] = "replacementStatus";
    KalturaLiveStreamAdminEntryCompareAttribute["startDate"] = "startDate";
    KalturaLiveStreamAdminEntryCompareAttribute["status"] = "status";
    KalturaLiveStreamAdminEntryCompareAttribute["totalRank"] = "totalRank";
    KalturaLiveStreamAdminEntryCompareAttribute["type"] = "type";
    KalturaLiveStreamAdminEntryCompareAttribute["updatedAt"] = "updatedAt";
    KalturaLiveStreamAdminEntryCompareAttribute["views"] = "views";
})(KalturaLiveStreamAdminEntryCompareAttribute = exports.KalturaLiveStreamAdminEntryCompareAttribute || (exports.KalturaLiveStreamAdminEntryCompareAttribute = {}));
