// kaltura-node-typescript-client/api/types/KalturaThumbCuePointBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaThumbCuePointBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaThumbCuePointSubType_1 = require("./KalturaThumbCuePointSubType");
var KalturaCuePointFilter_1 = require("./KalturaCuePointFilter");
var KalturaThumbCuePointBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaThumbCuePointBaseFilter, _super);
    function KalturaThumbCuePointBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaThumbCuePointBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaThumbCuePointBaseFilter' },
            descriptionLike: { type: 's' },
            descriptionMultiLikeOr: { type: 's' },
            descriptionMultiLikeAnd: { type: 's' },
            titleLike: { type: 's' },
            titleMultiLikeOr: { type: 's' },
            titleMultiLikeAnd: { type: 's' },
            subTypeEqual: { type: 'en', subTypeConstructor: KalturaThumbCuePointSubType_1.KalturaThumbCuePointSubType, subType: 'KalturaThumbCuePointSubType' },
            subTypeIn: { type: 's' }
        });
        return result;
    };
    return KalturaThumbCuePointBaseFilter;
}(KalturaCuePointFilter_1.KalturaCuePointFilter));
exports.KalturaThumbCuePointBaseFilter = KalturaThumbCuePointBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaThumbCuePointBaseFilter', KalturaThumbCuePointBaseFilter);
