// kaltura-node-typescript-client/api/types/KalturaWidget.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaWidget = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaWidgetSecurityType_1 = require("./KalturaWidgetSecurityType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaWidget = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaWidget, _super);
    function KalturaWidget(data) {
        return _super.call(this, data) || this;
    }
    KalturaWidget.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaWidget' },
            id: { type: 's', readOnly: true },
            sourceWidgetId: { type: 's' },
            rootWidgetId: { type: 's', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            entryId: { type: 's' },
            uiConfId: { type: 'n' },
            securityType: { type: 'en', subTypeConstructor: KalturaWidgetSecurityType_1.KalturaWidgetSecurityType, subType: 'KalturaWidgetSecurityType' },
            securityPolicy: { type: 'n' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            partnerData: { type: 's' },
            widgetHTML: { type: 's', readOnly: true },
            enforceEntitlement: { type: 'b' },
            privacyContext: { type: 's' },
            addEmbedHtml5Support: { type: 'b' },
            roles: { type: 's' },
            privileges: { type: 's' }
        });
        return result;
    };
    return KalturaWidget;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaWidget = KalturaWidget;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaWidget', KalturaWidget);
