// kaltura-node-typescript-client/api/types/KalturaCaptionAsset.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCaptionAsset = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaLanguage_1 = require("./KalturaLanguage");
var KalturaLanguageCode_1 = require("./KalturaLanguageCode");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaCaptionType_1 = require("./KalturaCaptionType");
var KalturaCaptionSource_1 = require("./KalturaCaptionSource");
var KalturaCaptionAssetStatus_1 = require("./KalturaCaptionAssetStatus");
var KalturaAsset_1 = require("./KalturaAsset");
var KalturaCaptionAsset = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCaptionAsset, _super);
    function KalturaCaptionAsset(data) {
        return _super.call(this, data) || this;
    }
    KalturaCaptionAsset.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCaptionAsset' },
            captionParamsId: { type: 'n' },
            language: { type: 'es', subTypeConstructor: KalturaLanguage_1.KalturaLanguage, subType: 'KalturaLanguage' },
            languageCode: { type: 'es', readOnly: true, subTypeConstructor: KalturaLanguageCode_1.KalturaLanguageCode, subType: 'KalturaLanguageCode' },
            isDefault: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            label: { type: 's' },
            format: { type: 'es', subTypeConstructor: KalturaCaptionType_1.KalturaCaptionType, subType: 'KalturaCaptionType' },
            source: { type: 'es', subTypeConstructor: KalturaCaptionSource_1.KalturaCaptionSource, subType: 'KalturaCaptionSource' },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaCaptionAssetStatus_1.KalturaCaptionAssetStatus, subType: 'KalturaCaptionAssetStatus' },
            parentId: { type: 's' },
            accuracy: { type: 'n' },
            displayOnPlayer: { type: 'b' },
            associatedTranscriptIds: { type: 's' }
        });
        return result;
    };
    return KalturaCaptionAsset;
}(KalturaAsset_1.KalturaAsset));
exports.KalturaCaptionAsset = KalturaCaptionAsset;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCaptionAsset', KalturaCaptionAsset);
