// kaltura-node-typescript-client/api/types/KalturaUserAppRoleFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUserAppRoleFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUserStatus_1 = require("./KalturaUserStatus");
var KalturaUserAppRoleBaseFilter_1 = require("./KalturaUserAppRoleBaseFilter");
var KalturaUserAppRoleFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUserAppRoleFilter, _super);
    function KalturaUserAppRoleFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaUserAppRoleFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUserAppRoleFilter' },
            userIdEqual: { type: 's' },
            userIdIn: { type: 's' },
            userStatusEqual: { type: 'en', subTypeConstructor: KalturaUserStatus_1.KalturaUserStatus, subType: 'KalturaUserStatus' },
            userStatusIn: { type: 's' }
        });
        return result;
    };
    return KalturaUserAppRoleFilter;
}(KalturaUserAppRoleBaseFilter_1.KalturaUserAppRoleBaseFilter));
exports.KalturaUserAppRoleFilter = KalturaUserAppRoleFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUserAppRoleFilter', KalturaUserAppRoleFilter);
