// kaltura-node-typescript-client/api/types/KalturaVirusScanJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaVirusScanJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaFileContainer_1 = require("./KalturaFileContainer");
var KalturaVirusScanJobResult_1 = require("./KalturaVirusScanJobResult");
var KalturaVirusFoundAction_1 = require("./KalturaVirusFoundAction");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaVirusScanJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaVirusScanJobData, _super);
    function KalturaVirusScanJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaVirusScanJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaVirusScanJobData' },
            fileContainer: { type: 'o', subTypeConstructor: KalturaFileContainer_1.KalturaFileContainer, subType: 'KalturaFileContainer' },
            flavorAssetId: { type: 's' },
            scanResult: { type: 'en', subTypeConstructor: KalturaVirusScanJobResult_1.KalturaVirusScanJobResult, subType: 'KalturaVirusScanJobResult' },
            virusFoundAction: { type: 'en', subTypeConstructor: KalturaVirusFoundAction_1.KalturaVirusFoundAction, subType: 'KalturaVirusFoundAction' }
        });
        return result;
    };
    return KalturaVirusScanJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaVirusScanJobData = KalturaVirusScanJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaVirusScanJobData', KalturaVirusScanJobData);
