// kaltura-node-typescript-client/api/types/KalturaYahooSyndicationFeed.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaYahooSyndicationFeed = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaYahooSyndicationFeedCategories_1 = require("./KalturaYahooSyndicationFeedCategories");
var KalturaYahooSyndicationFeedAdultValues_1 = require("./KalturaYahooSyndicationFeedAdultValues");
var KalturaBaseSyndicationFeed_1 = require("./KalturaBaseSyndicationFeed");
var KalturaYahooSyndicationFeed = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaYahooSyndicationFeed, _super);
    function KalturaYahooSyndicationFeed(data) {
        return _super.call(this, data) || this;
    }
    KalturaYahooSyndicationFeed.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaYahooSyndicationFeed' },
            category: { type: 'es', readOnly: true, subTypeConstructor: KalturaYahooSyndicationFeedCategories_1.KalturaYahooSyndicationFeedCategories, subType: 'KalturaYahooSyndicationFeedCategories' },
            adultContent: { type: 'es', subTypeConstructor: KalturaYahooSyndicationFeedAdultValues_1.KalturaYahooSyndicationFeedAdultValues, subType: 'KalturaYahooSyndicationFeedAdultValues' },
            feedDescription: { type: 's' },
            feedLandingPage: { type: 's' }
        });
        return result;
    };
    return KalturaYahooSyndicationFeed;
}(KalturaBaseSyndicationFeed_1.KalturaBaseSyndicationFeed));
exports.KalturaYahooSyndicationFeed = KalturaYahooSyndicationFeed;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaYahooSyndicationFeed', KalturaYahooSyndicationFeed);
