// kaltura-node-typescript-client/api/types/KalturaUrlRecognizerAkamaiG2O.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUrlRecognizerAkamaiG2O = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUrlRecognizer_1 = require("./KalturaUrlRecognizer");
var KalturaUrlRecognizerAkamaiG2O = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUrlRecognizerAkamaiG2O, _super);
    function KalturaUrlRecognizerAkamaiG2O(data) {
        return _super.call(this, data) || this;
    }
    KalturaUrlRecognizerAkamaiG2O.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUrlRecognizerAkamaiG2O' },
            headerData: { type: 's' },
            headerSign: { type: 's' },
            timeout: { type: 'n' },
            salt: { type: 's' }
        });
        return result;
    };
    return KalturaUrlRecognizerAkamaiG2O;
}(KalturaUrlRecognizer_1.KalturaUrlRecognizer));
exports.KalturaUrlRecognizerAkamaiG2O = KalturaUrlRecognizerAkamaiG2O;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUrlRecognizerAkamaiG2O', KalturaUrlRecognizerAkamaiG2O);
