// kaltura-node-typescript-client/api/types/KalturaLikeBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLikeBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaLikeBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLikeBaseFilter, _super);
    function KalturaLikeBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaLikeBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLikeBaseFilter' },
            entryIdEqual: { type: 's' },
            entryIdIn: { type: 's' },
            userIdEqual: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' }
        });
        return result;
    };
    return KalturaLikeBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaLikeBaseFilter = KalturaLikeBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLikeBaseFilter', KalturaLikeBaseFilter);
