// kaltura-node-typescript-client/api/types/KalturaAkamaiProvisionJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAkamaiProvisionJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaProvisionJobData_1 = require("./KalturaProvisionJobData");
var KalturaAkamaiProvisionJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAkamaiProvisionJobData, _super);
    function KalturaAkamaiProvisionJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaAkamaiProvisionJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAkamaiProvisionJobData' },
            wsdlUsername: { type: 's' },
            wsdlPassword: { type: 's' },
            cpcode: { type: 's' },
            emailId: { type: 's' },
            primaryContact: { type: 's' },
            secondaryContact: { type: 's' }
        });
        return result;
    };
    return KalturaAkamaiProvisionJobData;
}(KalturaProvisionJobData_1.KalturaProvisionJobData));
exports.KalturaAkamaiProvisionJobData = KalturaAkamaiProvisionJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAkamaiProvisionJobData', KalturaAkamaiProvisionJobData);
