// kaltura-node-typescript-client/api/types/KalturaPlayableEntryMatchAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPlayableEntryMatchAttribute = void 0;
var KalturaPlayableEntryMatchAttribute;
(function (KalturaPlayableEntryMatchAttribute) {
    KalturaPlayableEntryMatchAttribute["adminTags"] = "adminTags";
    KalturaPlayableEntryMatchAttribute["categoriesIds"] = "categoriesIds";
    KalturaPlayableEntryMatchAttribute["creatorId"] = "creatorId";
    KalturaPlayableEntryMatchAttribute["description"] = "description";
    KalturaPlayableEntryMatchAttribute["durationType"] = "durationType";
    KalturaPlayableEntryMatchAttribute["groupId"] = "groupId";
    KalturaPlayableEntryMatchAttribute["id"] = "id";
    KalturaPlayableEntryMatchAttribute["_name"] = "name";
    KalturaPlayableEntryMatchAttribute["referenceId"] = "referenceId";
    KalturaPlayableEntryMatchAttribute["replacedEntryId"] = "replacedEntryId";
    KalturaPlayableEntryMatchAttribute["replacingEntryId"] = "replacingEntryId";
    KalturaPlayableEntryMatchAttribute["searchText"] = "searchText";
    KalturaPlayableEntryMatchAttribute["tags"] = "tags";
    KalturaPlayableEntryMatchAttribute["userId"] = "userId";
})(KalturaPlayableEntryMatchAttribute = exports.KalturaPlayableEntryMatchAttribute || (exports.KalturaPlayableEntryMatchAttribute = {}));
