// kaltura-node-typescript-client/api/types/KalturaTvinciDistributionTag.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaTvinciDistributionTag = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaTvinciDistributionTag = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaTvinciDistributionTag, _super);
    function KalturaTvinciDistributionTag(data) {
        return _super.call(this, data) || this;
    }
    KalturaTvinciDistributionTag.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaTvinciDistributionTag' },
            tagname: { type: 's' },
            extension: { type: 's' },
            protocol: { type: 's' },
            format: { type: 's' },
            filename: { type: 's' },
            ppvmodule: { type: 's' }
        });
        return result;
    };
    return KalturaTvinciDistributionTag;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaTvinciDistributionTag = KalturaTvinciDistributionTag;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaTvinciDistributionTag', KalturaTvinciDistributionTag);
