// kaltura-node-typescript-client/api/types/KalturaUrlTokenizerAkamaiSecureHd.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUrlTokenizerAkamaiSecureHd = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUrlTokenizer_1 = require("./KalturaUrlTokenizer");
var KalturaUrlTokenizerAkamaiSecureHd = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUrlTokenizerAkamaiSecureHd, _super);
    function KalturaUrlTokenizerAkamaiSecureHd(data) {
        return _super.call(this, data) || this;
    }
    KalturaUrlTokenizerAkamaiSecureHd.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUrlTokenizerAkamaiSecureHd' },
            paramName: { type: 's' },
            aclPostfix: { type: 's' },
            customPostfixes: { type: 's' },
            useCookieHosts: { type: 's' },
            rootDir: { type: 's' }
        });
        return result;
    };
    return KalturaUrlTokenizerAkamaiSecureHd;
}(KalturaUrlTokenizer_1.KalturaUrlTokenizer));
exports.KalturaUrlTokenizerAkamaiSecureHd = KalturaUrlTokenizerAkamaiSecureHd;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUrlTokenizerAkamaiSecureHd', KalturaUrlTokenizerAkamaiSecureHd);
