// kaltura-node-typescript-client/api/types/KalturaVoicebaseJobProviderData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaVoicebaseJobProviderData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaLanguage_1 = require("./KalturaLanguage");
var KalturaIntegrationJobProviderData_1 = require("./KalturaIntegrationJobProviderData");
var KalturaVoicebaseJobProviderData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaVoicebaseJobProviderData, _super);
    function KalturaVoicebaseJobProviderData(data) {
        return _super.call(this, data) || this;
    }
    KalturaVoicebaseJobProviderData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaVoicebaseJobProviderData' },
            entryId: { type: 's' },
            flavorAssetId: { type: 's' },
            transcriptId: { type: 's' },
            captionAssetFormats: { type: 's' },
            apiKey: { type: 's', readOnly: true },
            apiPassword: { type: 's', readOnly: true },
            spokenLanguage: { type: 'es', subTypeConstructor: KalturaLanguage_1.KalturaLanguage, subType: 'KalturaLanguage' },
            fileLocation: { type: 's', readOnly: true },
            replaceMediaContent: { type: 'b' },
            additionalParameters: { type: 's', readOnly: true }
        });
        return result;
    };
    return KalturaVoicebaseJobProviderData;
}(KalturaIntegrationJobProviderData_1.KalturaIntegrationJobProviderData));
exports.KalturaVoicebaseJobProviderData = KalturaVoicebaseJobProviderData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaVoicebaseJobProviderData', KalturaVoicebaseJobProviderData);
