// kaltura-node-typescript-client/api/types/KalturaEntryContextDataResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEntryContextDataResult = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaString_1 = require("./KalturaString");
var KalturaRuleAction_1 = require("./KalturaRuleAction");
var KalturaFlavorAsset_1 = require("./KalturaFlavorAsset");
var KalturaPluginData_1 = require("./KalturaPluginData");
var KalturaContextDataResult_1 = require("./KalturaContextDataResult");
var KalturaEntryContextDataResult = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaEntryContextDataResult, _super);
    function KalturaEntryContextDataResult(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.accessControlMessages === 'undefined')
            _this.accessControlMessages = [];
        if (typeof _this.accessControlActions === 'undefined')
            _this.accessControlActions = [];
        if (typeof _this.flavorAssets === 'undefined')
            _this.flavorAssets = [];
        return _this;
    }
    KalturaEntryContextDataResult.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaEntryContextDataResult' },
            isSiteRestricted: { type: 'b' },
            isCountryRestricted: { type: 'b' },
            isSessionRestricted: { type: 'b' },
            isIpAddressRestricted: { type: 'b' },
            isUserAgentRestricted: { type: 'b' },
            previewLength: { type: 'n' },
            isScheduledNow: { type: 'b' },
            isAdmin: { type: 'b' },
            streamerType: { type: 's' },
            mediaProtocol: { type: 's' },
            storageProfilesXML: { type: 's' },
            accessControlMessages: { type: 'a', subTypeConstructor: KalturaString_1.KalturaString, subType: 'KalturaString' },
            accessControlActions: { type: 'a', subTypeConstructor: KalturaRuleAction_1.KalturaRuleAction, subType: 'KalturaRuleAction' },
            flavorAssets: { type: 'a', subTypeConstructor: KalturaFlavorAsset_1.KalturaFlavorAsset, subType: 'KalturaFlavorAsset' },
            msDuration: { type: 'n' },
            pluginData: { type: 'm', subTypeConstructor: KalturaPluginData_1.KalturaPluginData, subType: 'KalturaPluginData' }
        });
        return result;
    };
    return KalturaEntryContextDataResult;
}(KalturaContextDataResult_1.KalturaContextDataResult));
exports.KalturaEntryContextDataResult = KalturaEntryContextDataResult;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaEntryContextDataResult', KalturaEntryContextDataResult);
