// kaltura-node-typescript-client/api/types/KalturaCategoryEntryBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCategoryEntryBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCategoryEntryStatus_1 = require("./KalturaCategoryEntryStatus");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaCategoryEntryBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCategoryEntryBaseFilter, _super);
    function KalturaCategoryEntryBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaCategoryEntryBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCategoryEntryBaseFilter' },
            categoryIdEqual: { type: 'n' },
            categoryIdIn: { type: 's' },
            entryIdEqual: { type: 's' },
            entryIdIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            categoryFullIdsStartsWith: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaCategoryEntryStatus_1.KalturaCategoryEntryStatus, subType: 'KalturaCategoryEntryStatus' },
            statusIn: { type: 's' },
            creatorUserIdEqual: { type: 's' },
            creatorUserIdIn: { type: 's' }
        });
        return result;
    };
    return KalturaCategoryEntryBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaCategoryEntryBaseFilter = KalturaCategoryEntryBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCategoryEntryBaseFilter', KalturaCategoryEntryBaseFilter);
