// kaltura-node-typescript-client/api/types/KalturaGenericSyndicationFeed.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaGenericSyndicationFeed = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBaseEntryFilter_1 = require("./KalturaBaseEntryFilter");
var KalturaBaseSyndicationFeed_1 = require("./KalturaBaseSyndicationFeed");
var KalturaGenericSyndicationFeed = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaGenericSyndicationFeed, _super);
    function KalturaGenericSyndicationFeed(data) {
        return _super.call(this, data) || this;
    }
    KalturaGenericSyndicationFeed.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaGenericSyndicationFeed' },
            feedDescription: { type: 's' },
            feedLandingPage: { type: 's' },
            entryFilter: { type: 'o', subTypeConstructor: KalturaBaseEntryFilter_1.KalturaBaseEntryFilter, subType: 'KalturaBaseEntryFilter' },
            pageSize: { type: 'n' }
        });
        return result;
    };
    return KalturaGenericSyndicationFeed;
}(KalturaBaseSyndicationFeed_1.KalturaBaseSyndicationFeed));
exports.KalturaGenericSyndicationFeed = KalturaGenericSyndicationFeed;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaGenericSyndicationFeed', KalturaGenericSyndicationFeed);
