// kaltura-node-typescript-client/api/types/KalturaBeaconBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBeaconBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBeaconObjectTypes_1 = require("./KalturaBeaconObjectTypes");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaBeaconBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBeaconBaseFilter, _super);
    function KalturaBeaconBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaBeaconBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBeaconBaseFilter' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            relatedObjectTypeIn: { type: 's' },
            relatedObjectTypeEqual: { type: 'es', subTypeConstructor: KalturaBeaconObjectTypes_1.KalturaBeaconObjectTypes, subType: 'KalturaBeaconObjectTypes' },
            eventTypeIn: { type: 's' },
            objectIdIn: { type: 's' }
        });
        return result;
    };
    return KalturaBeaconBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaBeaconBaseFilter = KalturaBeaconBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBeaconBaseFilter', KalturaBeaconBaseFilter);
