// kaltura-node-typescript-client/api/types/KalturaBulkUploadResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBulkUploadResult = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBulkUploadResultStatus_1 = require("./KalturaBulkUploadResultStatus");
var KalturaBulkUploadAction_1 = require("./KalturaBulkUploadAction");
var KalturaBulkUploadObjectType_1 = require("./KalturaBulkUploadObjectType");
var KalturaBulkUploadPluginData_1 = require("./KalturaBulkUploadPluginData");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaBulkUploadResult = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBulkUploadResult, _super);
    function KalturaBulkUploadResult(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.pluginsData === 'undefined')
            _this.pluginsData = [];
        return _this;
    }
    KalturaBulkUploadResult.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBulkUploadResult' },
            id: { type: 'n', readOnly: true },
            bulkUploadJobId: { type: 'n' },
            lineIndex: { type: 'n' },
            partnerId: { type: 'n' },
            status: { type: 'es', subTypeConstructor: KalturaBulkUploadResultStatus_1.KalturaBulkUploadResultStatus, subType: 'KalturaBulkUploadResultStatus' },
            action: { type: 'es', subTypeConstructor: KalturaBulkUploadAction_1.KalturaBulkUploadAction, subType: 'KalturaBulkUploadAction' },
            objectId: { type: 's' },
            objectStatus: { type: 'n' },
            bulkUploadResultObjectType: { type: 'es', subTypeConstructor: KalturaBulkUploadObjectType_1.KalturaBulkUploadObjectType, subType: 'KalturaBulkUploadObjectType' },
            rowData: { type: 's' },
            partnerData: { type: 's' },
            objectErrorDescription: { type: 's' },
            pluginsData: { type: 'a', subTypeConstructor: KalturaBulkUploadPluginData_1.KalturaBulkUploadPluginData, subType: 'KalturaBulkUploadPluginData' },
            errorDescription: { type: 's' },
            errorCode: { type: 's' },
            errorType: { type: 'n' }
        });
        return result;
    };
    return KalturaBulkUploadResult;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaBulkUploadResult = KalturaBulkUploadResult;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBulkUploadResult', KalturaBulkUploadResult);
