// kaltura-node-typescript-client/api/types/KalturaWebexDropFolderFileOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaWebexDropFolderFileOrderBy = void 0;
var KalturaWebexDropFolderFileOrderBy;
(function (KalturaWebexDropFolderFileOrderBy) {
    KalturaWebexDropFolderFileOrderBy["createdAtAsc"] = "+createdAt";
    KalturaWebexDropFolderFileOrderBy["createdAtDesc"] = "-createdAt";
    KalturaWebexDropFolderFileOrderBy["fileNameAsc"] = "+fileName";
    KalturaWebexDropFolderFileOrderBy["fileNameDesc"] = "-fileName";
    KalturaWebexDropFolderFileOrderBy["fileSizeAsc"] = "+fileSize";
    KalturaWebexDropFolderFileOrderBy["fileSizeDesc"] = "-fileSize";
    KalturaWebexDropFolderFileOrderBy["fileSizeLastSetAtAsc"] = "+fileSizeLastSetAt";
    KalturaWebexDropFolderFileOrderBy["fileSizeLastSetAtDesc"] = "-fileSizeLastSetAt";
    KalturaWebexDropFolderFileOrderBy["idAsc"] = "+id";
    KalturaWebexDropFolderFileOrderBy["idDesc"] = "-id";
    KalturaWebexDropFolderFileOrderBy["parsedFlavorAsc"] = "+parsedFlavor";
    KalturaWebexDropFolderFileOrderBy["parsedFlavorDesc"] = "-parsedFlavor";
    KalturaWebexDropFolderFileOrderBy["parsedSlugAsc"] = "+parsedSlug";
    KalturaWebexDropFolderFileOrderBy["parsedSlugDesc"] = "-parsedSlug";
    KalturaWebexDropFolderFileOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaWebexDropFolderFileOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaWebexDropFolderFileOrderBy = exports.KalturaWebexDropFolderFileOrderBy || (exports.KalturaWebexDropFolderFileOrderBy = {}));
