// kaltura-node-typescript-client/api/types/KalturaFreewheelDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFreewheelDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDistributionProfile_1 = require("./KalturaDistributionProfile");
var KalturaFreewheelDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaFreewheelDistributionProfile, _super);
    function KalturaFreewheelDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaFreewheelDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaFreewheelDistributionProfile' },
            apikey: { type: 's' },
            email: { type: 's' },
            sftpPass: { type: 's' },
            sftpLogin: { type: 's' },
            accountId: { type: 's' },
            metadataProfileId: { type: 'n' }
        });
        return result;
    };
    return KalturaFreewheelDistributionProfile;
}(KalturaDistributionProfile_1.KalturaDistributionProfile));
exports.KalturaFreewheelDistributionProfile = KalturaFreewheelDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaFreewheelDistributionProfile', KalturaFreewheelDistributionProfile);
