// kaltura-node-typescript-client/api/types/KalturaWebexAPIDropFolderFile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaWebexAPIDropFolderFile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDropFolderFile_1 = require("./KalturaDropFolderFile");
var KalturaWebexAPIDropFolderFile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaWebexAPIDropFolderFile, _super);
    function KalturaWebexAPIDropFolderFile(data) {
        return _super.call(this, data) || this;
    }
    KalturaWebexAPIDropFolderFile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaWebexAPIDropFolderFile' },
            recordingId: { type: 's' },
            description: { type: 's' },
            contentUrl: { type: 's' },
            urlExpiry: { type: 'n' },
            fileExtension: { type: 's' },
            meetingId: { type: 's' },
            recordingStartTime: { type: 'n' },
            hostEmail: { type: 's' }
        });
        return result;
    };
    return KalturaWebexAPIDropFolderFile;
}(KalturaDropFolderFile_1.KalturaDropFolderFile));
exports.KalturaWebexAPIDropFolderFile = KalturaWebexAPIDropFolderFile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaWebexAPIDropFolderFile', KalturaWebexAPIDropFolderFile);
