// kaltura-node-typescript-client/api/types/KalturaQuestionCuePoint.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaQuestionCuePoint = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaOptionalAnswer_1 = require("./KalturaOptionalAnswer");
var KalturaQuestionType_1 = require("./KalturaQuestionType");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaCuePoint_1 = require("./KalturaCuePoint");
var KalturaQuestionCuePoint = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaQuestionCuePoint, _super);
    function KalturaQuestionCuePoint(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.optionalAnswers === 'undefined')
            _this.optionalAnswers = [];
        return _this;
    }
    KalturaQuestionCuePoint.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaQuestionCuePoint' },
            optionalAnswers: { type: 'a', subTypeConstructor: KalturaOptionalAnswer_1.KalturaOptionalAnswer, subType: 'KalturaOptionalAnswer' },
            hint: { type: 's' },
            question: { type: 's' },
            explanation: { type: 's' },
            questionType: { type: 'en', subTypeConstructor: KalturaQuestionType_1.KalturaQuestionType, subType: 'KalturaQuestionType' },
            presentationOrder: { type: 'n' },
            excludeFromScore: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' }
        });
        return result;
    };
    return KalturaQuestionCuePoint;
}(KalturaCuePoint_1.KalturaCuePoint));
exports.KalturaQuestionCuePoint = KalturaQuestionCuePoint;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaQuestionCuePoint', KalturaQuestionCuePoint);
