// kaltura-node-typescript-client/api/types/KalturaUserScorePropertiesFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUserScorePropertiesFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaGameObjectType_1 = require("./KalturaGameObjectType");
var KalturaUserScorePropertiesBaseFilter_1 = require("./KalturaUserScorePropertiesBaseFilter");
var KalturaUserScorePropertiesFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUserScorePropertiesFilter, _super);
    function KalturaUserScorePropertiesFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaUserScorePropertiesFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUserScorePropertiesFilter' },
            gameObjectId: { type: 's' },
            gameObjectType: { type: 'es', subTypeConstructor: KalturaGameObjectType_1.KalturaGameObjectType, subType: 'KalturaGameObjectType' },
            userIdEqual: { type: 's' },
            placesAboveUser: { type: 'n' },
            placesBelowUser: { type: 'n' }
        });
        return result;
    };
    return KalturaUserScorePropertiesFilter;
}(KalturaUserScorePropertiesBaseFilter_1.KalturaUserScorePropertiesBaseFilter));
exports.KalturaUserScorePropertiesFilter = KalturaUserScorePropertiesFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUserScorePropertiesFilter', KalturaUserScorePropertiesFilter);
