// kaltura-node-typescript-client/api/types/KalturaFileSyncResource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFileSyncResource = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaContentResource_1 = require("./KalturaContentResource");
var KalturaFileSyncResource = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaFileSyncResource, _super);
    function KalturaFileSyncResource(data) {
        return _super.call(this, data) || this;
    }
    KalturaFileSyncResource.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaFileSyncResource' },
            fileSyncObjectType: { type: 'n' },
            objectSubType: { type: 'n' },
            objectId: { type: 's' },
            version: { type: 's' },
            originEntryId: { type: 's' }
        });
        return result;
    };
    return KalturaFileSyncResource;
}(KalturaContentResource_1.KalturaContentResource));
exports.KalturaFileSyncResource = KalturaFileSyncResource;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaFileSyncResource', KalturaFileSyncResource);
