// kaltura-node-typescript-client/api/types/KalturaBulkUploadResultCategory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBulkUploadResultCategory = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBulkUploadResult_1 = require("./KalturaBulkUploadResult");
var KalturaBulkUploadResultCategory = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBulkUploadResultCategory, _super);
    function KalturaBulkUploadResultCategory(data) {
        return _super.call(this, data) || this;
    }
    KalturaBulkUploadResultCategory.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBulkUploadResultCategory' },
            relativePath: { type: 's' },
            name: { type: 's' },
            referenceId: { type: 's' },
            description: { type: 's' },
            tags: { type: 's' },
            appearInList: { type: 'n' },
            privacy: { type: 'n' },
            inheritanceType: { type: 'n' },
            userJoinPolicy: { type: 'n' },
            defaultPermissionLevel: { type: 'n' },
            owner: { type: 's' },
            contributionPolicy: { type: 'n' },
            partnerSortValue: { type: 'n' },
            moderation: { type: 'b' }
        });
        return result;
    };
    return KalturaBulkUploadResultCategory;
}(KalturaBulkUploadResult_1.KalturaBulkUploadResult));
exports.KalturaBulkUploadResultCategory = KalturaBulkUploadResultCategory;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBulkUploadResultCategory', KalturaBulkUploadResultCategory);
