// kaltura-node-typescript-client/api/types/KalturaDrmProfileBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDrmProfileBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDrmProviderType_1 = require("./KalturaDrmProviderType");
var KalturaDrmProfileStatus_1 = require("./KalturaDrmProfileStatus");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaDrmProfileBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDrmProfileBaseFilter, _super);
    function KalturaDrmProfileBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaDrmProfileBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDrmProfileBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            nameLike: { type: 's' },
            providerEqual: { type: 'es', subTypeConstructor: KalturaDrmProviderType_1.KalturaDrmProviderType, subType: 'KalturaDrmProviderType' },
            providerIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaDrmProfileStatus_1.KalturaDrmProfileStatus, subType: 'KalturaDrmProfileStatus' },
            statusIn: { type: 's' }
        });
        return result;
    };
    return KalturaDrmProfileBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaDrmProfileBaseFilter = KalturaDrmProfileBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDrmProfileBaseFilter', KalturaDrmProfileBaseFilter);
