// kaltura-node-typescript-client/api/types/KalturaClipConcatJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaClipConcatJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaOperationAttributes_1 = require("./KalturaOperationAttributes");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaClipConcatJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaClipConcatJobData, _super);
    function KalturaClipConcatJobData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.operationAttributes === 'undefined')
            _this.operationAttributes = [];
        return _this;
    }
    KalturaClipConcatJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaClipConcatJobData' },
            destEntryId: { type: 's' },
            tempEntryId: { type: 's' },
            sourceEntryId: { type: 's' },
            importUrl: { type: 's' },
            partnerId: { type: 'n' },
            priority: { type: 'n' },
            operationAttributes: { type: 'a', subTypeConstructor: KalturaOperationAttributes_1.KalturaOperationAttributes, subType: 'KalturaOperationAttributes' },
            resourceOrder: { type: 'n' },
            conversionParams: { type: 's' }
        });
        return result;
    };
    return KalturaClipConcatJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaClipConcatJobData = KalturaClipConcatJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaClipConcatJobData', KalturaClipConcatJobData);
