// kaltura-node-typescript-client/api/types/KalturaESearchLanguage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaESearchLanguage = void 0;
var KalturaESearchLanguage;
(function (KalturaESearchLanguage) {
    KalturaESearchLanguage["arabic"] = "Arabic";
    KalturaESearchLanguage["basque"] = "Basque";
    KalturaESearchLanguage["brazilian"] = "Brazilian";
    KalturaESearchLanguage["bulgarian"] = "Bulgarian";
    KalturaESearchLanguage["catalan"] = "Catalan";
    KalturaESearchLanguage["chinese"] = "Chinese";
    KalturaESearchLanguage["czech"] = "Czech";
    KalturaESearchLanguage["danish"] = "Danish";
    KalturaESearchLanguage["dutch"] = "Dutch";
    KalturaESearchLanguage["english"] = "English";
    KalturaESearchLanguage["finnish"] = "Finnish";
    KalturaESearchLanguage["french"] = "French";
    KalturaESearchLanguage["galician"] = "Galician";
    KalturaESearchLanguage["german"] = "German";
    KalturaESearchLanguage["greek"] = "Greek";
    KalturaESearchLanguage["hindi"] = "Hindi";
    KalturaESearchLanguage["hungrian"] = "Hungarian";
    KalturaESearchLanguage["indonesian"] = "Indonesian";
    KalturaESearchLanguage["italian"] = "Italian";
    KalturaESearchLanguage["japanese"] = "Japanese";
    KalturaESearchLanguage["korean"] = "Korean";
    KalturaESearchLanguage["latvian"] = "Latvian";
    KalturaESearchLanguage["lithuanian"] = "Lithuanian";
    KalturaESearchLanguage["norwegian"] = "Norwegian";
    KalturaESearchLanguage["persian"] = "Persian";
    KalturaESearchLanguage["portuguese"] = "Prtuguese";
    KalturaESearchLanguage["romanian"] = "Romanian";
    KalturaESearchLanguage["russian"] = "Russian";
    KalturaESearchLanguage["sorani"] = "Sorani";
    KalturaESearchLanguage["spanish"] = "Spanish";
    KalturaESearchLanguage["swedish"] = "Swedish";
    KalturaESearchLanguage["thai"] = "Thai";
    KalturaESearchLanguage["turkish"] = "Turkish";
})(KalturaESearchLanguage = exports.KalturaESearchLanguage || (exports.KalturaESearchLanguage = {}));
