// kaltura-node-typescript-client/api/types/KalturaScheduledTaskProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaScheduledTaskProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaScheduledTaskProfileStatus_1 = require("./KalturaScheduledTaskProfileStatus");
var KalturaObjectFilterEngineType_1 = require("./KalturaObjectFilterEngineType");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaObjectTask_1 = require("./KalturaObjectTask");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaScheduledTaskProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaScheduledTaskProfile, _super);
    function KalturaScheduledTaskProfile(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.objectTasks === 'undefined')
            _this.objectTasks = [];
        return _this;
    }
    KalturaScheduledTaskProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaScheduledTaskProfile' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            name: { type: 's' },
            systemName: { type: 's' },
            description: { type: 's' },
            status: { type: 'en', subTypeConstructor: KalturaScheduledTaskProfileStatus_1.KalturaScheduledTaskProfileStatus, subType: 'KalturaScheduledTaskProfileStatus' },
            objectFilterEngineType: { type: 'es', subTypeConstructor: KalturaObjectFilterEngineType_1.KalturaObjectFilterEngineType, subType: 'KalturaObjectFilterEngineType' },
            objectFilter: { type: 'o', subTypeConstructor: KalturaFilter_1.KalturaFilter, subType: 'KalturaFilter' },
            objectTasks: { type: 'a', subTypeConstructor: KalturaObjectTask_1.KalturaObjectTask, subType: 'KalturaObjectTask' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            lastExecutionStartedAt: { type: 'd' },
            maxTotalCountAllowed: { type: 'n' }
        });
        return result;
    };
    return KalturaScheduledTaskProfile;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaScheduledTaskProfile = KalturaScheduledTaskProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaScheduledTaskProfile', KalturaScheduledTaskProfile);
