// kaltura-node-typescript-client/api/types/KalturaUnicornDistributionJobProviderData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUnicornDistributionJobProviderData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionJobProviderData_1 = require("./KalturaConfigurableDistributionJobProviderData");
var KalturaUnicornDistributionJobProviderData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUnicornDistributionJobProviderData, _super);
    function KalturaUnicornDistributionJobProviderData(data) {
        return _super.call(this, data) || this;
    }
    KalturaUnicornDistributionJobProviderData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUnicornDistributionJobProviderData' },
            catalogGuid: { type: 's' },
            title: { type: 's' },
            mediaChanged: { type: 'b' },
            flavorAssetVersion: { type: 's' },
            notificationBaseUrl: { type: 's' }
        });
        return result;
    };
    return KalturaUnicornDistributionJobProviderData;
}(KalturaConfigurableDistributionJobProviderData_1.KalturaConfigurableDistributionJobProviderData));
exports.KalturaUnicornDistributionJobProviderData = KalturaUnicornDistributionJobProviderData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUnicornDistributionJobProviderData', KalturaUnicornDistributionJobProviderData);
