// kaltura-node-typescript-client/api/types/KalturaLiveStreamEntry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveStreamEntry = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaLiveStreamBitrate_1 = require("./KalturaLiveStreamBitrate");
var KalturaSipSourceType_1 = require("./KalturaSipSourceType");
var KalturaLiveEntry_1 = require("./KalturaLiveEntry");
var KalturaLiveStreamEntry = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveStreamEntry, _super);
    function KalturaLiveStreamEntry(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.bitrates === 'undefined')
            _this.bitrates = [];
        return _this;
    }
    KalturaLiveStreamEntry.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveStreamEntry' },
            streamRemoteId: { type: 's', readOnly: true },
            streamRemoteBackupId: { type: 's', readOnly: true },
            bitrates: { type: 'a', subTypeConstructor: KalturaLiveStreamBitrate_1.KalturaLiveStreamBitrate, subType: 'KalturaLiveStreamBitrate' },
            primaryBroadcastingUrl: { type: 's' },
            secondaryBroadcastingUrl: { type: 's' },
            primarySecuredBroadcastingUrl: { type: 's' },
            secondarySecuredBroadcastingUrl: { type: 's' },
            primaryRtspBroadcastingUrl: { type: 's' },
            secondaryRtspBroadcastingUrl: { type: 's' },
            primarySrtBroadcastingUrl: { type: 's' },
            primarySrtStreamId: { type: 's' },
            secondarySrtBroadcastingUrl: { type: 's' },
            secondarySrtStreamId: { type: 's' },
            streamName: { type: 's' },
            streamUrl: { type: 's' },
            hlsStreamUrl: { type: 's' },
            urlManager: { type: 's' },
            encodingIP1: { type: 's' },
            encodingIP2: { type: 's' },
            streamPassword: { type: 's' },
            streamUsername: { type: 's', readOnly: true },
            srtPass: { type: 's' },
            primaryServerNodeId: { type: 'n', readOnly: true },
            sipToken: { type: 's', readOnly: true },
            sipSourceType: { type: 'en', readOnly: true, subTypeConstructor: KalturaSipSourceType_1.KalturaSipSourceType, subType: 'KalturaSipSourceType' }
        });
        return result;
    };
    return KalturaLiveStreamEntry;
}(KalturaLiveEntry_1.KalturaLiveEntry));
exports.KalturaLiveStreamEntry = KalturaLiveStreamEntry;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveStreamEntry', KalturaLiveStreamEntry);
