// kaltura-node-typescript-client/api/types/KalturaEntryServerNodeStatus.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEntryServerNodeStatus = void 0;
var KalturaEntryServerNodeStatus;
(function (KalturaEntryServerNodeStatus) {
    KalturaEntryServerNodeStatus[KalturaEntryServerNodeStatus["authenticated"] = 3] = "authenticated";
    KalturaEntryServerNodeStatus[KalturaEntryServerNodeStatus["broadcasting"] = 2] = "broadcasting";
    KalturaEntryServerNodeStatus[KalturaEntryServerNodeStatus["error"] = -1] = "error";
    KalturaEntryServerNodeStatus[KalturaEntryServerNodeStatus["markedForDeletion"] = 4] = "markedForDeletion";
    KalturaEntryServerNodeStatus[KalturaEntryServerNodeStatus["playable"] = 1] = "playable";
    KalturaEntryServerNodeStatus[KalturaEntryServerNodeStatus["stopped"] = 0] = "stopped";
    KalturaEntryServerNodeStatus[KalturaEntryServerNodeStatus["taskFinished"] = 9] = "taskFinished";
    KalturaEntryServerNodeStatus[KalturaEntryServerNodeStatus["taskPending"] = 5] = "taskPending";
    KalturaEntryServerNodeStatus[KalturaEntryServerNodeStatus["taskProcessing"] = 7] = "taskProcessing";
    KalturaEntryServerNodeStatus[KalturaEntryServerNodeStatus["taskQueued"] = 6] = "taskQueued";
    KalturaEntryServerNodeStatus[KalturaEntryServerNodeStatus["taskUploading"] = 8] = "taskUploading";
})(KalturaEntryServerNodeStatus = exports.KalturaEntryServerNodeStatus || (exports.KalturaEntryServerNodeStatus = {}));
