// kaltura-node-typescript-client/api/types/KalturaLiveStreamPushPublishRTMPConfiguration.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveStreamPushPublishRTMPConfiguration = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaLiveStreamPushPublishConfiguration_1 = require("./KalturaLiveStreamPushPublishConfiguration");
var KalturaLiveStreamPushPublishRTMPConfiguration = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveStreamPushPublishRTMPConfiguration, _super);
    function KalturaLiveStreamPushPublishRTMPConfiguration(data) {
        return _super.call(this, data) || this;
    }
    KalturaLiveStreamPushPublishRTMPConfiguration.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveStreamPushPublishRTMPConfiguration' },
            userId: { type: 's' },
            password: { type: 's' },
            streamName: { type: 's' },
            applicationName: { type: 's' }
        });
        return result;
    };
    return KalturaLiveStreamPushPublishRTMPConfiguration;
}(KalturaLiveStreamPushPublishConfiguration_1.KalturaLiveStreamPushPublishConfiguration));
exports.KalturaLiveStreamPushPublishRTMPConfiguration = KalturaLiveStreamPushPublishRTMPConfiguration;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveStreamPushPublishRTMPConfiguration', KalturaLiveStreamPushPublishRTMPConfiguration);
