// kaltura-node-typescript-client/api/types/KalturaSessionInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSessionInfo = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaSessionType_1 = require("./KalturaSessionType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaSessionInfo = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaSessionInfo, _super);
    function KalturaSessionInfo(data) {
        return _super.call(this, data) || this;
    }
    KalturaSessionInfo.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaSessionInfo' },
            ks: { type: 's', readOnly: true },
            sessionType: { type: 'en', readOnly: true, subTypeConstructor: KalturaSessionType_1.KalturaSessionType, subType: 'KalturaSessionType' },
            partnerId: { type: 'n', readOnly: true },
            userId: { type: 's', readOnly: true },
            expiry: { type: 'd', readOnly: true },
            privileges: { type: 's', readOnly: true }
        });
        return result;
    };
    return KalturaSessionInfo;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaSessionInfo = KalturaSessionInfo;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaSessionInfo', KalturaSessionInfo);
