// kaltura-node-typescript-client/api/types/KalturaConvertCaptionAssetJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaConvertCaptionAssetJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaConvertCaptionAssetJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaConvertCaptionAssetJobData, _super);
    function KalturaConvertCaptionAssetJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaConvertCaptionAssetJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaConvertCaptionAssetJobData' },
            captionAssetId: { type: 's' },
            fileLocation: { type: 's' },
            fileEncryptionKey: { type: 's' },
            fromType: { type: 's' },
            toType: { type: 's' }
        });
        return result;
    };
    return KalturaConvertCaptionAssetJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaConvertCaptionAssetJobData = KalturaConvertCaptionAssetJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaConvertCaptionAssetJobData', KalturaConvertCaptionAssetJobData);
