// kaltura-node-typescript-client/api/types/KalturaVirusScanProfileBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaVirusScanProfileBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaVirusScanProfileStatus_1 = require("./KalturaVirusScanProfileStatus");
var KalturaVirusScanEngineType_1 = require("./KalturaVirusScanEngineType");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaVirusScanProfileBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaVirusScanProfileBaseFilter, _super);
    function KalturaVirusScanProfileBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaVirusScanProfileBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaVirusScanProfileBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            nameEqual: { type: 's' },
            nameLike: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaVirusScanProfileStatus_1.KalturaVirusScanProfileStatus, subType: 'KalturaVirusScanProfileStatus' },
            statusIn: { type: 's' },
            engineTypeEqual: { type: 'es', subTypeConstructor: KalturaVirusScanEngineType_1.KalturaVirusScanEngineType, subType: 'KalturaVirusScanEngineType' },
            engineTypeIn: { type: 's' }
        });
        return result;
    };
    return KalturaVirusScanProfileBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaVirusScanProfileBaseFilter = KalturaVirusScanProfileBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaVirusScanProfileBaseFilter', KalturaVirusScanProfileBaseFilter);
