// kaltura-node-typescript-client/api/types/KalturaLiveStats.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveStats = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaLiveStats = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveStats, _super);
    function KalturaLiveStats(data) {
        return _super.call(this, data) || this;
    }
    KalturaLiveStats.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveStats' },
            audience: { type: 'n' },
            dvrAudience: { type: 'n' },
            avgBitrate: { type: 'n' },
            bufferTime: { type: 'n' },
            plays: { type: 'n' },
            secondsViewed: { type: 'n' },
            startEvent: { type: 'n' },
            timestamp: { type: 'd' }
        });
        return result;
    };
    return KalturaLiveStats;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaLiveStats = KalturaLiveStats;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveStats', KalturaLiveStats);
