// kaltura-node-typescript-client/api/types/KalturaPartnerType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPartnerType = void 0;
var KalturaPartnerType;
(function (KalturaPartnerType) {
    KalturaPartnerType[KalturaPartnerType["adminConsole"] = 109] = "adminConsole";
    KalturaPartnerType[KalturaPartnerType["blackboard"] = 107] = "blackboard";
    KalturaPartnerType[KalturaPartnerType["communityEdition"] = 105] = "communityEdition";
    KalturaPartnerType[KalturaPartnerType["dekiwiki"] = 103] = "dekiwiki";
    KalturaPartnerType[KalturaPartnerType["drupal"] = 102] = "drupal";
    KalturaPartnerType[KalturaPartnerType["joomla"] = 106] = "joomla";
    KalturaPartnerType[KalturaPartnerType["kmc"] = 1] = "kmc";
    KalturaPartnerType[KalturaPartnerType["moodle"] = 104] = "moodle";
    KalturaPartnerType[KalturaPartnerType["sakai"] = 108] = "sakai";
    KalturaPartnerType[KalturaPartnerType["wiki"] = 100] = "wiki";
    KalturaPartnerType[KalturaPartnerType["wordpress"] = 101] = "wordpress";
})(KalturaPartnerType = exports.KalturaPartnerType || (exports.KalturaPartnerType = {}));
