// kaltura-node-typescript-client/api/types/KalturaESearchEntryFieldName.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaESearchEntryFieldName = void 0;
var KalturaESearchEntryFieldName;
(function (KalturaESearchEntryFieldName) {
    KalturaESearchEntryFieldName["accessControlId"] = "access_control_id";
    KalturaESearchEntryFieldName["adminTags"] = "admin_tags";
    KalturaESearchEntryFieldName["broadcastEntryId"] = "broadcast_entry_id";
    KalturaESearchEntryFieldName["captionsContent"] = "captions_content";
    KalturaESearchEntryFieldName["conversionProfileId"] = "conversion_profile_id";
    KalturaESearchEntryFieldName["createdAt"] = "created_at";
    KalturaESearchEntryFieldName["creatorId"] = "creator_kuser_id";
    KalturaESearchEntryFieldName["credit"] = "credit";
    KalturaESearchEntryFieldName["description"] = "description";
    KalturaESearchEntryFieldName["displayInSearch"] = "display_in_search";
    KalturaESearchEntryFieldName["endDate"] = "end_date";
    KalturaESearchEntryFieldName["entitledUserEdit"] = "entitled_kusers_edit";
    KalturaESearchEntryFieldName["entitledUserPublish"] = "entitled_kusers_publish";
    KalturaESearchEntryFieldName["entitledUserView"] = "entitled_kusers_view";
    KalturaESearchEntryFieldName["entryType"] = "entry_type";
    KalturaESearchEntryFieldName["externalSourceType"] = "external_source_type";
    KalturaESearchEntryFieldName["id"] = "id";
    KalturaESearchEntryFieldName["isLive"] = "is_live";
    KalturaESearchEntryFieldName["isQuiz"] = "is_quiz";
    KalturaESearchEntryFieldName["lastPlayedAt"] = "last_played_at";
    KalturaESearchEntryFieldName["lengthInMsecs"] = "length_in_msecs";
    KalturaESearchEntryFieldName["mediaType"] = "media_type";
    KalturaESearchEntryFieldName["moderationStatus"] = "moderation_status";
    KalturaESearchEntryFieldName["_name"] = "name";
    KalturaESearchEntryFieldName["parentEntryId"] = "parent_id";
    KalturaESearchEntryFieldName["partnerSortValue"] = "partner_sort_value";
    KalturaESearchEntryFieldName["plays"] = "plays";
    KalturaESearchEntryFieldName["pushPublish"] = "push_publish";
    KalturaESearchEntryFieldName["rank"] = "rank";
    KalturaESearchEntryFieldName["recordedEntryId"] = "recorded_entry_id";
    KalturaESearchEntryFieldName["recycledAt"] = "recycled_at";
    KalturaESearchEntryFieldName["redirectEntryId"] = "redirect_entry_id";
    KalturaESearchEntryFieldName["referenceId"] = "reference_id";
    KalturaESearchEntryFieldName["roomType"] = "room_type";
    KalturaESearchEntryFieldName["rootId"] = "root_id";
    KalturaESearchEntryFieldName["siteUrl"] = "site_url";
    KalturaESearchEntryFieldName["sourceType"] = "source_type";
    KalturaESearchEntryFieldName["startDate"] = "start_date";
    KalturaESearchEntryFieldName["tags"] = "tags";
    KalturaESearchEntryFieldName["templateEntryId"] = "template_entry_id";
    KalturaESearchEntryFieldName["updatedAt"] = "updated_at";
    KalturaESearchEntryFieldName["userId"] = "kuser_id";
    KalturaESearchEntryFieldName["userNames"] = "user_names";
    KalturaESearchEntryFieldName["votes"] = "votes";
})(KalturaESearchEntryFieldName = exports.KalturaESearchEntryFieldName || (exports.KalturaESearchEntryFieldName = {}));
