// kaltura-node-typescript-client/api/types/KalturaEventNotificationEventType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEventNotificationEventType = void 0;
var KalturaEventNotificationEventType;
(function (KalturaEventNotificationEventType) {
    KalturaEventNotificationEventType["batchJobStatus"] = "1";
    KalturaEventNotificationEventType["integrationJobClosed"] = "integrationEventNotifications.INTEGRATION_JOB_CLOSED";
    KalturaEventNotificationEventType["objectAdded"] = "2";
    KalturaEventNotificationEventType["objectChanged"] = "3";
    KalturaEventNotificationEventType["objectCopied"] = "4";
    KalturaEventNotificationEventType["objectCreated"] = "5";
    KalturaEventNotificationEventType["objectDataChanged"] = "6";
    KalturaEventNotificationEventType["objectDeleted"] = "7";
    KalturaEventNotificationEventType["objectErased"] = "8";
    KalturaEventNotificationEventType["objectReadyForIndex"] = "13";
    KalturaEventNotificationEventType["objectReadyForReplacment"] = "9";
    KalturaEventNotificationEventType["objectReplaced"] = "12";
    KalturaEventNotificationEventType["objectSaved"] = "10";
    KalturaEventNotificationEventType["objectUpdated"] = "11";
})(KalturaEventNotificationEventType = exports.KalturaEventNotificationEventType || (exports.KalturaEventNotificationEventType = {}));
