// kaltura-node-typescript-client/api/types/KalturaMetadata.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMetadata = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaMetadataObjectType_1 = require("./KalturaMetadataObjectType");
var KalturaMetadataStatus_1 = require("./KalturaMetadataStatus");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaMetadata = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMetadata, _super);
    function KalturaMetadata(data) {
        return _super.call(this, data) || this;
    }
    KalturaMetadata.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMetadata' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            metadataProfileId: { type: 'n', readOnly: true },
            metadataProfileVersion: { type: 'n', readOnly: true },
            metadataObjectType: { type: 'es', readOnly: true, subTypeConstructor: KalturaMetadataObjectType_1.KalturaMetadataObjectType, subType: 'KalturaMetadataObjectType' },
            objectId: { type: 's', readOnly: true },
            version: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaMetadataStatus_1.KalturaMetadataStatus, subType: 'KalturaMetadataStatus' },
            xml: { type: 's', readOnly: true }
        });
        return result;
    };
    return KalturaMetadata;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaMetadata = KalturaMetadata;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMetadata', KalturaMetadata);
