// kaltura-node-typescript-client/api/types/KalturaAssetParamsOutput.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAssetParamsOutput = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaContainerFormat_1 = require("./KalturaContainerFormat");
var KalturaAssetParams_1 = require("./KalturaAssetParams");
var KalturaAssetParamsOutput = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAssetParamsOutput, _super);
    function KalturaAssetParamsOutput(data) {
        return _super.call(this, data) || this;
    }
    KalturaAssetParamsOutput.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAssetParamsOutput' },
            assetParamsId: { type: 'n' },
            assetParamsVersion: { type: 's' },
            assetId: { type: 's' },
            assetVersion: { type: 's' },
            readyBehavior: { type: 'n' },
            format: { type: 'es', subTypeConstructor: KalturaContainerFormat_1.KalturaContainerFormat, subType: 'KalturaContainerFormat' }
        });
        return result;
    };
    return KalturaAssetParamsOutput;
}(KalturaAssetParams_1.KalturaAssetParams));
exports.KalturaAssetParamsOutput = KalturaAssetParamsOutput;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAssetParamsOutput', KalturaAssetParamsOutput);
