// kaltura-node-typescript-client/api/types/KalturaShortLink.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaShortLink = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaShortLinkStatus_1 = require("./KalturaShortLinkStatus");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaShortLink = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaShortLink, _super);
    function KalturaShortLink(data) {
        return _super.call(this, data) || this;
    }
    KalturaShortLink.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaShortLink' },
            id: { type: 's', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            expiresAt: { type: 'n' },
            partnerId: { type: 'n', readOnly: true },
            userId: { type: 's' },
            name: { type: 's' },
            systemName: { type: 's' },
            fullUrl: { type: 's' },
            status: { type: 'en', subTypeConstructor: KalturaShortLinkStatus_1.KalturaShortLinkStatus, subType: 'KalturaShortLinkStatus' }
        });
        return result;
    };
    return KalturaShortLink;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaShortLink = KalturaShortLink;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaShortLink', KalturaShortLink);
