// kaltura-node-typescript-client/api/types/KalturaReport.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaReport = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaReport = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaReport, _super);
    function KalturaReport(data) {
        return _super.call(this, data) || this;
    }
    KalturaReport.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaReport' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n' },
            name: { type: 's' },
            systemName: { type: 's' },
            description: { type: 's' },
            query: { type: 's' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true }
        });
        return result;
    };
    return KalturaReport;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaReport = KalturaReport;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaReport', KalturaReport);
