// kaltura-node-typescript-client/api/types/KalturaMetadataProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMetadataProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaMetadataObjectType_1 = require("./KalturaMetadataObjectType");
var KalturaMetadataProfileStatus_1 = require("./KalturaMetadataProfileStatus");
var KalturaMetadataProfileCreateMode_1 = require("./KalturaMetadataProfileCreateMode");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaMetadataProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMetadataProfile, _super);
    function KalturaMetadataProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaMetadataProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMetadataProfile' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            metadataObjectType: { type: 'es', subTypeConstructor: KalturaMetadataObjectType_1.KalturaMetadataObjectType, subType: 'KalturaMetadataObjectType' },
            version: { type: 'n', readOnly: true },
            name: { type: 's' },
            systemName: { type: 's' },
            description: { type: 's' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaMetadataProfileStatus_1.KalturaMetadataProfileStatus, subType: 'KalturaMetadataProfileStatus' },
            xsd: { type: 's', readOnly: true },
            views: { type: 's', readOnly: true },
            xslt: { type: 's', readOnly: true },
            createMode: { type: 'en', subTypeConstructor: KalturaMetadataProfileCreateMode_1.KalturaMetadataProfileCreateMode, subType: 'KalturaMetadataProfileCreateMode' },
            disableReIndexing: { type: 'b' }
        });
        return result;
    };
    return KalturaMetadataProfile;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaMetadataProfile = KalturaMetadataProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMetadataProfile', KalturaMetadataProfile);
