// kaltura-node-typescript-client/api/types/KalturaPlaylistOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPlaylistOrderBy = void 0;
var KalturaPlaylistOrderBy;
(function (KalturaPlaylistOrderBy) {
    KalturaPlaylistOrderBy["createdAtAsc"] = "+createdAt";
    KalturaPlaylistOrderBy["createdAtDesc"] = "-createdAt";
    KalturaPlaylistOrderBy["endDateAsc"] = "+endDate";
    KalturaPlaylistOrderBy["endDateDesc"] = "-endDate";
    KalturaPlaylistOrderBy["moderationCountAsc"] = "+moderationCount";
    KalturaPlaylistOrderBy["moderationCountDesc"] = "-moderationCount";
    KalturaPlaylistOrderBy["nameAsc"] = "+name";
    KalturaPlaylistOrderBy["nameDesc"] = "-name";
    KalturaPlaylistOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaPlaylistOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaPlaylistOrderBy["rankAsc"] = "+rank";
    KalturaPlaylistOrderBy["rankDesc"] = "-rank";
    KalturaPlaylistOrderBy["recentAsc"] = "+recent";
    KalturaPlaylistOrderBy["recentDesc"] = "-recent";
    KalturaPlaylistOrderBy["startDateAsc"] = "+startDate";
    KalturaPlaylistOrderBy["startDateDesc"] = "-startDate";
    KalturaPlaylistOrderBy["totalRankAsc"] = "+totalRank";
    KalturaPlaylistOrderBy["totalRankDesc"] = "-totalRank";
    KalturaPlaylistOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaPlaylistOrderBy["updatedAtDesc"] = "-updatedAt";
    KalturaPlaylistOrderBy["weightAsc"] = "+weight";
    KalturaPlaylistOrderBy["weightDesc"] = "-weight";
})(KalturaPlaylistOrderBy = exports.KalturaPlaylistOrderBy || (exports.KalturaPlaylistOrderBy = {}));
