// kaltura-node-typescript-client/api/types/KalturaReachProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaReachProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaReachProfileStatus_1 = require("./KalturaReachProfileStatus");
var KalturaReachProfileType_1 = require("./KalturaReachProfileType");
var KalturaVendorCatalogItemOutputFormat_1 = require("./KalturaVendorCatalogItemOutputFormat");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaReachProfileContentDeletionPolicy_1 = require("./KalturaReachProfileContentDeletionPolicy");
var KalturaRule_1 = require("./KalturaRule");
var KalturaBaseVendorCredit_1 = require("./KalturaBaseVendorCredit");
var KalturaDictionary_1 = require("./KalturaDictionary");
var KalturaVendorTaskProcessingRegion_1 = require("./KalturaVendorTaskProcessingRegion");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaReachProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaReachProfile, _super);
    function KalturaReachProfile(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.rules === 'undefined')
            _this.rules = [];
        if (typeof _this.dictionaries === 'undefined')
            _this.dictionaries = [];
        return _this;
    }
    KalturaReachProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaReachProfile' },
            id: { type: 'n', readOnly: true },
            name: { type: 's' },
            partnerId: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaReachProfileStatus_1.KalturaReachProfileStatus, subType: 'KalturaReachProfileStatus' },
            profileType: { type: 'en', subTypeConstructor: KalturaReachProfileType_1.KalturaReachProfileType, subType: 'KalturaReachProfileType' },
            defaultOutputFormat: { type: 'en', subTypeConstructor: KalturaVendorCatalogItemOutputFormat_1.KalturaVendorCatalogItemOutputFormat, subType: 'KalturaVendorCatalogItemOutputFormat' },
            enableMachineModeration: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            enableHumanModeration: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            autoDisplayMachineCaptionsOnPlayer: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            autoDisplayHumanCaptionsOnPlayer: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            enableMetadataExtraction: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            enableSpeakerChangeIndication: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            enableAudioTags: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            enableProfanityRemoval: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            maxCharactersPerCaptionLine: { type: 'n' },
            labelAdditionForMachineServiceType: { type: 's' },
            labelAdditionForHumanServiceType: { type: 's' },
            contentDeletionPolicy: { type: 'en', subTypeConstructor: KalturaReachProfileContentDeletionPolicy_1.KalturaReachProfileContentDeletionPolicy, subType: 'KalturaReachProfileContentDeletionPolicy' },
            rules: { type: 'a', subTypeConstructor: KalturaRule_1.KalturaRule, subType: 'KalturaRule' },
            credit: { type: 'o', subTypeConstructor: KalturaBaseVendorCredit_1.KalturaBaseVendorCredit, subType: 'KalturaBaseVendorCredit' },
            usedCredit: { type: 'n', readOnly: true },
            dictionaries: { type: 'a', subTypeConstructor: KalturaDictionary_1.KalturaDictionary, subType: 'KalturaDictionary' },
            flavorParamsIds: { type: 's' },
            vendorTaskProcessingRegion: { type: 'en', subTypeConstructor: KalturaVendorTaskProcessingRegion_1.KalturaVendorTaskProcessingRegion, subType: 'KalturaVendorTaskProcessingRegion' }
        });
        return result;
    };
    return KalturaReachProfile;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaReachProfile = KalturaReachProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaReachProfile', KalturaReachProfile);
