// kaltura-node-typescript-client/api/types/KalturaRenderCaptionAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaRenderCaptionAttributes = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBorderStyle_1 = require("./KalturaBorderStyle");
var KalturaCaptionsAlignment_1 = require("./KalturaCaptionsAlignment");
var KalturaCaptionAttributes_1 = require("./KalturaCaptionAttributes");
var KalturaRenderCaptionAttributes = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaRenderCaptionAttributes, _super);
    function KalturaRenderCaptionAttributes(data) {
        return _super.call(this, data) || this;
    }
    KalturaRenderCaptionAttributes.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaRenderCaptionAttributes' },
            fontName: { type: 's' },
            fontSize: { type: 'n' },
            primaryColour: { type: 's' },
            borderStyle: { type: 'en', subTypeConstructor: KalturaBorderStyle_1.KalturaBorderStyle, subType: 'KalturaBorderStyle' },
            backColour: { type: 's' },
            outlineColour: { type: 's' },
            shadow: { type: 'n' },
            bold: { type: 'b' },
            italic: { type: 'b' },
            underline: { type: 'b' },
            alignment: { type: 'en', subTypeConstructor: KalturaCaptionsAlignment_1.KalturaCaptionsAlignment, subType: 'KalturaCaptionsAlignment' },
            captionAssetId: { type: 's' }
        });
        return result;
    };
    return KalturaRenderCaptionAttributes;
}(KalturaCaptionAttributes_1.KalturaCaptionAttributes));
exports.KalturaRenderCaptionAttributes = KalturaRenderCaptionAttributes;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaRenderCaptionAttributes', KalturaRenderCaptionAttributes);
