// kaltura-node-typescript-client/api/types/KalturaGenericDistributionProviderAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaGenericDistributionProviderAction = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDistributionAction_1 = require("./KalturaDistributionAction");
var KalturaGenericDistributionProviderStatus_1 = require("./KalturaGenericDistributionProviderStatus");
var KalturaGenericDistributionProviderParser_1 = require("./KalturaGenericDistributionProviderParser");
var KalturaDistributionProtocol_1 = require("./KalturaDistributionProtocol");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaGenericDistributionProviderAction = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaGenericDistributionProviderAction, _super);
    function KalturaGenericDistributionProviderAction(data) {
        return _super.call(this, data) || this;
    }
    KalturaGenericDistributionProviderAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaGenericDistributionProviderAction' },
            id: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            genericDistributionProviderId: { type: 'n' },
            action: { type: 'en', subTypeConstructor: KalturaDistributionAction_1.KalturaDistributionAction, subType: 'KalturaDistributionAction' },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaGenericDistributionProviderStatus_1.KalturaGenericDistributionProviderStatus, subType: 'KalturaGenericDistributionProviderStatus' },
            resultsParser: { type: 'en', subTypeConstructor: KalturaGenericDistributionProviderParser_1.KalturaGenericDistributionProviderParser, subType: 'KalturaGenericDistributionProviderParser' },
            protocol: { type: 'en', subTypeConstructor: KalturaDistributionProtocol_1.KalturaDistributionProtocol, subType: 'KalturaDistributionProtocol' },
            serverAddress: { type: 's' },
            remotePath: { type: 's' },
            remoteUsername: { type: 's' },
            remotePassword: { type: 's' },
            editableFields: { type: 's' },
            mandatoryFields: { type: 's' },
            mrssTransformer: { type: 's', readOnly: true },
            mrssValidator: { type: 's', readOnly: true },
            resultsTransformer: { type: 's', readOnly: true }
        });
        return result;
    };
    return KalturaGenericDistributionProviderAction;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaGenericDistributionProviderAction = KalturaGenericDistributionProviderAction;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaGenericDistributionProviderAction', KalturaGenericDistributionProviderAction);
