// kaltura-node-typescript-client/api/types/KalturaPartnerOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPartnerOrderBy = void 0;
var KalturaPartnerOrderBy;
(function (KalturaPartnerOrderBy) {
    KalturaPartnerOrderBy["adminEmailAsc"] = "+adminEmail";
    KalturaPartnerOrderBy["adminEmailDesc"] = "-adminEmail";
    KalturaPartnerOrderBy["adminNameAsc"] = "+adminName";
    KalturaPartnerOrderBy["adminNameDesc"] = "-adminName";
    KalturaPartnerOrderBy["createdAtAsc"] = "+createdAt";
    KalturaPartnerOrderBy["createdAtDesc"] = "-createdAt";
    KalturaPartnerOrderBy["idAsc"] = "+id";
    KalturaPartnerOrderBy["idDesc"] = "-id";
    KalturaPartnerOrderBy["nameAsc"] = "+name";
    KalturaPartnerOrderBy["nameDesc"] = "-name";
    KalturaPartnerOrderBy["statusAsc"] = "+status";
    KalturaPartnerOrderBy["statusDesc"] = "-status";
    KalturaPartnerOrderBy["websiteAsc"] = "+website";
    KalturaPartnerOrderBy["websiteDesc"] = "-website";
})(KalturaPartnerOrderBy = exports.KalturaPartnerOrderBy || (exports.KalturaPartnerOrderBy = {}));
