// kaltura-node-typescript-client/api/types/KalturaClipAttributes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaClipAttributes = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaEffect_1 = require("./KalturaEffect");
var KalturaCaptionAttributes_1 = require("./KalturaCaptionAttributes");
var KalturaOperationAttributes_1 = require("./KalturaOperationAttributes");
var KalturaClipAttributes = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaClipAttributes, _super);
    function KalturaClipAttributes(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.effectArray === 'undefined')
            _this.effectArray = [];
        if (typeof _this.captionAttributes === 'undefined')
            _this.captionAttributes = [];
        return _this;
    }
    KalturaClipAttributes.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaClipAttributes' },
            offset: { type: 'n' },
            duration: { type: 'n' },
            globalOffsetInDestination: { type: 'n' },
            effectArray: { type: 'a', subTypeConstructor: KalturaEffect_1.KalturaEffect, subType: 'KalturaEffect' },
            cropAlignment: { type: 'n' },
            captionAttributes: { type: 'a', subTypeConstructor: KalturaCaptionAttributes_1.KalturaCaptionAttributes, subType: 'KalturaCaptionAttributes' }
        });
        return result;
    };
    return KalturaClipAttributes;
}(KalturaOperationAttributes_1.KalturaOperationAttributes));
exports.KalturaClipAttributes = KalturaClipAttributes;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaClipAttributes', KalturaClipAttributes);
