// kaltura-node-typescript-client/api/types/KalturaScheduledTaskJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaScheduledTaskJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDryRunFileType_1 = require("./KalturaDryRunFileType");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaScheduledTaskJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaScheduledTaskJobData, _super);
    function KalturaScheduledTaskJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaScheduledTaskJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaScheduledTaskJobData' },
            maxResults: { type: 'n' },
            totalCount: { type: 'n' },
            fileFormat: { type: 'en', subTypeConstructor: KalturaDryRunFileType_1.KalturaDryRunFileType, subType: 'KalturaDryRunFileType' },
            resultsFilePath: { type: 's' },
            referenceTime: { type: 'd' }
        });
        return result;
    };
    return KalturaScheduledTaskJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaScheduledTaskJobData = KalturaScheduledTaskJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaScheduledTaskJobData', KalturaScheduledTaskJobData);
