// kaltura-node-typescript-client/api/types/KalturaUser.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUser = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUserType_1 = require("./KalturaUserType");
var KalturaGender_1 = require("./KalturaGender");
var KalturaBaseUser_1 = require("./KalturaBaseUser");
var KalturaUser = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUser, _super);
    function KalturaUser(data) {
        return _super.call(this, data) || this;
    }
    KalturaUser.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUser' },
            type: { type: 'en', subTypeConstructor: KalturaUserType_1.KalturaUserType, subType: 'KalturaUserType' },
            dateOfBirth: { type: 'n' },
            gender: { type: 'en', subTypeConstructor: KalturaGender_1.KalturaGender, subType: 'KalturaGender' },
            isAdmin: { type: 'b' },
            isGuest: { type: 'b' },
            roleIds: { type: 's' },
            roleNames: { type: 's', readOnly: true },
            isAccountOwner: { type: 'b' },
            password: { type: 's' },
            firstName: { type: 's' },
            lastName: { type: 's' },
            loginEnabled: { type: 'b' },
            registrationInfo: { type: 's' },
            attendanceInfo: { type: 's' },
            title: { type: 's' },
            company: { type: 's' },
            ksPrivileges: { type: 's' },
            encryptedSeed: { type: 's', readOnly: true },
            isSsoExcluded: { type: 'b' },
            externalId: { type: 's' }
        });
        return result;
    };
    return KalturaUser;
}(KalturaBaseUser_1.KalturaBaseUser));
exports.KalturaUser = KalturaUser;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUser', KalturaUser);
