// kaltura-node-typescript-client/api/types/KalturaTransformMetadataJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaTransformMetadataJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaFileContainer_1 = require("./KalturaFileContainer");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaTransformMetadataJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaTransformMetadataJobData, _super);
    function KalturaTransformMetadataJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaTransformMetadataJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaTransformMetadataJobData' },
            srcXsl: { type: 'o', subTypeConstructor: KalturaFileContainer_1.KalturaFileContainer, subType: 'KalturaFileContainer' },
            srcVersion: { type: 'n' },
            destVersion: { type: 'n' },
            destXsd: { type: 'o', subTypeConstructor: KalturaFileContainer_1.KalturaFileContainer, subType: 'KalturaFileContainer' },
            metadataProfileId: { type: 'n' }
        });
        return result;
    };
    return KalturaTransformMetadataJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaTransformMetadataJobData = KalturaTransformMetadataJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaTransformMetadataJobData', KalturaTransformMetadataJobData);
