// kaltura-node-typescript-client/api/types/KalturaDeliveryProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDeliveryProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDeliveryProfileType_1 = require("./KalturaDeliveryProfileType");
var KalturaPlaybackProtocol_1 = require("./KalturaPlaybackProtocol");
var KalturaDeliveryStatus_1 = require("./KalturaDeliveryStatus");
var KalturaUrlRecognizer_1 = require("./KalturaUrlRecognizer");
var KalturaUrlTokenizer_1 = require("./KalturaUrlTokenizer");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaAssetFilter_1 = require("./KalturaAssetFilter");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaDeliveryProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDeliveryProfile, _super);
    function KalturaDeliveryProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaDeliveryProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDeliveryProfile' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            name: { type: 's' },
            type: { type: 'es', subTypeConstructor: KalturaDeliveryProfileType_1.KalturaDeliveryProfileType, subType: 'KalturaDeliveryProfileType' },
            systemName: { type: 's' },
            description: { type: 's' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            streamerType: { type: 'es', subTypeConstructor: KalturaPlaybackProtocol_1.KalturaPlaybackProtocol, subType: 'KalturaPlaybackProtocol' },
            url: { type: 's' },
            hostName: { type: 's', readOnly: true },
            status: { type: 'en', subTypeConstructor: KalturaDeliveryStatus_1.KalturaDeliveryStatus, subType: 'KalturaDeliveryStatus' },
            recognizer: { type: 'o', subTypeConstructor: KalturaUrlRecognizer_1.KalturaUrlRecognizer, subType: 'KalturaUrlRecognizer' },
            tokenizer: { type: 'o', subTypeConstructor: KalturaUrlTokenizer_1.KalturaUrlTokenizer, subType: 'KalturaUrlTokenizer' },
            isDefault: { type: 'en', readOnly: true, subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            parentId: { type: 'n', readOnly: true },
            mediaProtocols: { type: 's' },
            priority: { type: 'n' },
            extraParams: { type: 's' },
            supplementaryAssetsFilter: { type: 'o', subTypeConstructor: KalturaAssetFilter_1.KalturaAssetFilter, subType: 'KalturaAssetFilter' },
            enforceDeliveriesSupport: { type: 's' }
        });
        return result;
    };
    return KalturaDeliveryProfile;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaDeliveryProfile = KalturaDeliveryProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDeliveryProfile', KalturaDeliveryProfile);
