// kaltura-node-typescript-client/api/types/KalturaITunesSyndicationFeed.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaITunesSyndicationFeed = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaITunesSyndicationFeedCategories_1 = require("./KalturaITunesSyndicationFeedCategories");
var KalturaITunesSyndicationFeedAdultValues_1 = require("./KalturaITunesSyndicationFeedAdultValues");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaBaseSyndicationFeed_1 = require("./KalturaBaseSyndicationFeed");
var KalturaITunesSyndicationFeed = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaITunesSyndicationFeed, _super);
    function KalturaITunesSyndicationFeed(data) {
        return _super.call(this, data) || this;
    }
    KalturaITunesSyndicationFeed.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaITunesSyndicationFeed' },
            feedDescription: { type: 's' },
            language: { type: 's' },
            feedLandingPage: { type: 's' },
            ownerName: { type: 's' },
            ownerEmail: { type: 's' },
            feedImageUrl: { type: 's' },
            category: { type: 'es', readOnly: true, subTypeConstructor: KalturaITunesSyndicationFeedCategories_1.KalturaITunesSyndicationFeedCategories, subType: 'KalturaITunesSyndicationFeedCategories' },
            adultContent: { type: 'es', subTypeConstructor: KalturaITunesSyndicationFeedAdultValues_1.KalturaITunesSyndicationFeedAdultValues, subType: 'KalturaITunesSyndicationFeedAdultValues' },
            feedAuthor: { type: 's' },
            enforceFeedAuthor: { type: 'b' },
            enforceOrder: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' }
        });
        return result;
    };
    return KalturaITunesSyndicationFeed;
}(KalturaBaseSyndicationFeed_1.KalturaBaseSyndicationFeed));
exports.KalturaITunesSyndicationFeed = KalturaITunesSyndicationFeed;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaITunesSyndicationFeed', KalturaITunesSyndicationFeed);
