// kaltura-node-typescript-client/api/types/KalturaRoomEntryOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaRoomEntryOrderBy = void 0;
var KalturaRoomEntryOrderBy;
(function (KalturaRoomEntryOrderBy) {
    KalturaRoomEntryOrderBy["createdAtAsc"] = "+createdAt";
    KalturaRoomEntryOrderBy["createdAtDesc"] = "-createdAt";
    KalturaRoomEntryOrderBy["endDateAsc"] = "+endDate";
    KalturaRoomEntryOrderBy["endDateDesc"] = "-endDate";
    KalturaRoomEntryOrderBy["moderationCountAsc"] = "+moderationCount";
    KalturaRoomEntryOrderBy["moderationCountDesc"] = "-moderationCount";
    KalturaRoomEntryOrderBy["nameAsc"] = "+name";
    KalturaRoomEntryOrderBy["nameDesc"] = "-name";
    KalturaRoomEntryOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaRoomEntryOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaRoomEntryOrderBy["rankAsc"] = "+rank";
    KalturaRoomEntryOrderBy["rankDesc"] = "-rank";
    KalturaRoomEntryOrderBy["recentAsc"] = "+recent";
    KalturaRoomEntryOrderBy["recentDesc"] = "-recent";
    KalturaRoomEntryOrderBy["startDateAsc"] = "+startDate";
    KalturaRoomEntryOrderBy["startDateDesc"] = "-startDate";
    KalturaRoomEntryOrderBy["totalRankAsc"] = "+totalRank";
    KalturaRoomEntryOrderBy["totalRankDesc"] = "-totalRank";
    KalturaRoomEntryOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaRoomEntryOrderBy["updatedAtDesc"] = "-updatedAt";
    KalturaRoomEntryOrderBy["weightAsc"] = "+weight";
    KalturaRoomEntryOrderBy["weightDesc"] = "-weight";
})(KalturaRoomEntryOrderBy = exports.KalturaRoomEntryOrderBy || (exports.KalturaRoomEntryOrderBy = {}));
