// kaltura-node-typescript-client/api/types/KalturaConversionProfileAssetParams.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaConversionProfileAssetParams = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaFlavorReadyBehaviorType_1 = require("./KalturaFlavorReadyBehaviorType");
var KalturaAssetParamsOrigin_1 = require("./KalturaAssetParamsOrigin");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaAssetParamsDeletePolicy_1 = require("./KalturaAssetParamsDeletePolicy");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaConversionProfileAssetParams = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaConversionProfileAssetParams, _super);
    function KalturaConversionProfileAssetParams(data) {
        return _super.call(this, data) || this;
    }
    KalturaConversionProfileAssetParams.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaConversionProfileAssetParams' },
            conversionProfileId: { type: 'n', readOnly: true },
            assetParamsId: { type: 'n', readOnly: true },
            readyBehavior: { type: 'en', subTypeConstructor: KalturaFlavorReadyBehaviorType_1.KalturaFlavorReadyBehaviorType, subType: 'KalturaFlavorReadyBehaviorType' },
            origin: { type: 'en', subTypeConstructor: KalturaAssetParamsOrigin_1.KalturaAssetParamsOrigin, subType: 'KalturaAssetParamsOrigin' },
            systemName: { type: 's' },
            forceNoneComplied: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            deletePolicy: { type: 'en', subTypeConstructor: KalturaAssetParamsDeletePolicy_1.KalturaAssetParamsDeletePolicy, subType: 'KalturaAssetParamsDeletePolicy' },
            isEncrypted: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            contentAwareness: { type: 'n' },
            chunkedEncodeMode: { type: 'n' },
            twoPass: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            tags: { type: 's' },
            overloadParams: { type: 's' }
        });
        return result;
    };
    return KalturaConversionProfileAssetParams;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaConversionProfileAssetParams = KalturaConversionProfileAssetParams;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaConversionProfileAssetParams', KalturaConversionProfileAssetParams);
