// kaltura-node-typescript-client/api/types/KalturaCategoryUserProviderFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCategoryUserProviderFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCategoryUserStatus_1 = require("./KalturaCategoryUserStatus");
var KalturaUpdateMethodType_1 = require("./KalturaUpdateMethodType");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaCategoryUserProviderFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCategoryUserProviderFilter, _super);
    function KalturaCategoryUserProviderFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaCategoryUserProviderFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCategoryUserProviderFilter' },
            userIdEqual: { type: 's' },
            userIdIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaCategoryUserStatus_1.KalturaCategoryUserStatus, subType: 'KalturaCategoryUserStatus' },
            statusIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'n' },
            createdAtLessThanOrEqual: { type: 'n' },
            updatedAtGreaterThanOrEqual: { type: 'n' },
            updatedAtLessThanOrEqual: { type: 'n' },
            updateMethodEqual: { type: 'en', subTypeConstructor: KalturaUpdateMethodType_1.KalturaUpdateMethodType, subType: 'KalturaUpdateMethodType' },
            updateMethodIn: { type: 's' },
            permissionNamesMatchAnd: { type: 's' },
            permissionNamesMatchOr: { type: 's' }
        });
        return result;
    };
    return KalturaCategoryUserProviderFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaCategoryUserProviderFilter = KalturaCategoryUserProviderFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCategoryUserProviderFilter', KalturaCategoryUserProviderFilter);
