// kaltura-node-typescript-client/api/types/KalturaPartnerStatistics.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPartnerStatistics = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaPartnerStatistics = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPartnerStatistics, _super);
    function KalturaPartnerStatistics(data) {
        return _super.call(this, data) || this;
    }
    KalturaPartnerStatistics.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPartnerStatistics' },
            packageBandwidthAndStorage: { type: 'n', readOnly: true },
            hosting: { type: 'n', readOnly: true },
            bandwidth: { type: 'n', readOnly: true },
            usage: { type: 'n', readOnly: true },
            usagePercent: { type: 'n', readOnly: true },
            reachedLimitDate: { type: 'n', readOnly: true }
        });
        return result;
    };
    return KalturaPartnerStatistics;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaPartnerStatistics = KalturaPartnerStatistics;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPartnerStatistics', KalturaPartnerStatistics);
