// kaltura-node-typescript-client/api/types/KalturaESearchHistoryFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaESearchHistoryFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaESearchRange_1 = require("./KalturaESearchRange");
var KalturaESearchHistoryAggregationItem_1 = require("./KalturaESearchHistoryAggregationItem");
var KalturaESearchBaseFilter_1 = require("./KalturaESearchBaseFilter");
var KalturaESearchHistoryFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaESearchHistoryFilter, _super);
    function KalturaESearchHistoryFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaESearchHistoryFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaESearchHistoryFilter' },
            searchTermStartsWith: { type: 's' },
            searchedObjectIn: { type: 's' },
            timestampRange: { type: 'o', subTypeConstructor: KalturaESearchRange_1.KalturaESearchRange, subType: 'KalturaESearchRange' },
            aggregation: { type: 'o', subTypeConstructor: KalturaESearchHistoryAggregationItem_1.KalturaESearchHistoryAggregationItem, subType: 'KalturaESearchHistoryAggregationItem' }
        });
        return result;
    };
    return KalturaESearchHistoryFilter;
}(KalturaESearchBaseFilter_1.KalturaESearchBaseFilter));
exports.KalturaESearchHistoryFilter = KalturaESearchHistoryFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaESearchHistoryFilter', KalturaESearchHistoryFilter);
