// kaltura-node-typescript-client/api/types/KalturaYoutubeApiDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaYoutubeApiDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaYoutubeApiDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaYoutubeApiDistributionProfile, _super);
    function KalturaYoutubeApiDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaYoutubeApiDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaYoutubeApiDistributionProfile' },
            username: { type: 's' },
            defaultCategory: { type: 'n' },
            allowComments: { type: 's' },
            allowEmbedding: { type: 's' },
            allowRatings: { type: 's' },
            allowResponses: { type: 's' },
            apiAuthorizeUrl: { type: 's' },
            googleClientId: { type: 's' },
            googleClientSecret: { type: 's' },
            googleTokenData: { type: 's' },
            assumeSuccess: { type: 'b' },
            privacyStatus: { type: 's' }
        });
        return result;
    };
    return KalturaYoutubeApiDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaYoutubeApiDistributionProfile = KalturaYoutubeApiDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaYoutubeApiDistributionProfile', KalturaYoutubeApiDistributionProfile);
