// kaltura-node-typescript-client/api/types/KalturaGenericSyndicationFeedOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaGenericSyndicationFeedOrderBy = void 0;
var KalturaGenericSyndicationFeedOrderBy;
(function (KalturaGenericSyndicationFeedOrderBy) {
    KalturaGenericSyndicationFeedOrderBy["createdAtAsc"] = "+createdAt";
    KalturaGenericSyndicationFeedOrderBy["createdAtDesc"] = "-createdAt";
    KalturaGenericSyndicationFeedOrderBy["nameAsc"] = "+name";
    KalturaGenericSyndicationFeedOrderBy["nameDesc"] = "-name";
    KalturaGenericSyndicationFeedOrderBy["playlistIdAsc"] = "+playlistId";
    KalturaGenericSyndicationFeedOrderBy["playlistIdDesc"] = "-playlistId";
    KalturaGenericSyndicationFeedOrderBy["typeAsc"] = "+type";
    KalturaGenericSyndicationFeedOrderBy["typeDesc"] = "-type";
    KalturaGenericSyndicationFeedOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaGenericSyndicationFeedOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaGenericSyndicationFeedOrderBy = exports.KalturaGenericSyndicationFeedOrderBy || (exports.KalturaGenericSyndicationFeedOrderBy = {}));
