// kaltura-node-typescript-client/api/types/KalturaDailymotionDistributionJobProviderData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDailymotionDistributionJobProviderData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDailymotionDistributionCaptionInfo_1 = require("./KalturaDailymotionDistributionCaptionInfo");
var KalturaConfigurableDistributionJobProviderData_1 = require("./KalturaConfigurableDistributionJobProviderData");
var KalturaDailymotionDistributionJobProviderData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDailymotionDistributionJobProviderData, _super);
    function KalturaDailymotionDistributionJobProviderData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.captionsInfo === 'undefined')
            _this.captionsInfo = [];
        return _this;
    }
    KalturaDailymotionDistributionJobProviderData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDailymotionDistributionJobProviderData' },
            videoAssetFilePath: { type: 's' },
            accessControlGeoBlockingOperation: { type: 's' },
            accessControlGeoBlockingCountryList: { type: 's' },
            captionsInfo: { type: 'a', subTypeConstructor: KalturaDailymotionDistributionCaptionInfo_1.KalturaDailymotionDistributionCaptionInfo, subType: 'KalturaDailymotionDistributionCaptionInfo' }
        });
        return result;
    };
    return KalturaDailymotionDistributionJobProviderData;
}(KalturaConfigurableDistributionJobProviderData_1.KalturaConfigurableDistributionJobProviderData));
exports.KalturaDailymotionDistributionJobProviderData = KalturaDailymotionDistributionJobProviderData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDailymotionDistributionJobProviderData', KalturaDailymotionDistributionJobProviderData);
