// kaltura-node-typescript-client/api/types/KalturaEmailNotificationTemplate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEmailNotificationTemplate = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaEmailNotificationFormat_1 = require("./KalturaEmailNotificationFormat");
var KalturaEmailNotificationRecipientProvider_1 = require("./KalturaEmailNotificationRecipientProvider");
var KalturaEmailNotificationTemplatePriority_1 = require("./KalturaEmailNotificationTemplatePriority");
var KalturaKeyValue_1 = require("./KalturaKeyValue");
var KalturaEventNotificationTemplate_1 = require("./KalturaEventNotificationTemplate");
var KalturaEmailNotificationTemplate = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaEmailNotificationTemplate, _super);
    function KalturaEmailNotificationTemplate(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.customHeaders === 'undefined')
            _this.customHeaders = [];
        return _this;
    }
    KalturaEmailNotificationTemplate.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaEmailNotificationTemplate' },
            format: { type: 'es', subTypeConstructor: KalturaEmailNotificationFormat_1.KalturaEmailNotificationFormat, subType: 'KalturaEmailNotificationFormat' },
            subject: { type: 's' },
            body: { type: 's' },
            fromEmail: { type: 's' },
            fromName: { type: 's' },
            to: { type: 'o', subTypeConstructor: KalturaEmailNotificationRecipientProvider_1.KalturaEmailNotificationRecipientProvider, subType: 'KalturaEmailNotificationRecipientProvider' },
            cc: { type: 'o', subTypeConstructor: KalturaEmailNotificationRecipientProvider_1.KalturaEmailNotificationRecipientProvider, subType: 'KalturaEmailNotificationRecipientProvider' },
            bcc: { type: 'o', subTypeConstructor: KalturaEmailNotificationRecipientProvider_1.KalturaEmailNotificationRecipientProvider, subType: 'KalturaEmailNotificationRecipientProvider' },
            replyTo: { type: 'o', subTypeConstructor: KalturaEmailNotificationRecipientProvider_1.KalturaEmailNotificationRecipientProvider, subType: 'KalturaEmailNotificationRecipientProvider' },
            priority: { type: 'en', subTypeConstructor: KalturaEmailNotificationTemplatePriority_1.KalturaEmailNotificationTemplatePriority, subType: 'KalturaEmailNotificationTemplatePriority' },
            confirmReadingTo: { type: 's' },
            hostname: { type: 's' },
            messageID: { type: 's' },
            customHeaders: { type: 'a', subTypeConstructor: KalturaKeyValue_1.KalturaKeyValue, subType: 'KalturaKeyValue' }
        });
        return result;
    };
    return KalturaEmailNotificationTemplate;
}(KalturaEventNotificationTemplate_1.KalturaEventNotificationTemplate));
exports.KalturaEmailNotificationTemplate = KalturaEmailNotificationTemplate;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaEmailNotificationTemplate', KalturaEmailNotificationTemplate);
