// kaltura-node-typescript-client/api/types/KalturaAttachmentAssetBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAttachmentAssetBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaAttachmentType_1 = require("./KalturaAttachmentType");
var KalturaAttachmentAssetStatus_1 = require("./KalturaAttachmentAssetStatus");
var KalturaAssetFilter_1 = require("./KalturaAssetFilter");
var KalturaAttachmentAssetBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAttachmentAssetBaseFilter, _super);
    function KalturaAttachmentAssetBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaAttachmentAssetBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAttachmentAssetBaseFilter' },
            formatEqual: { type: 'es', subTypeConstructor: KalturaAttachmentType_1.KalturaAttachmentType, subType: 'KalturaAttachmentType' },
            formatIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaAttachmentAssetStatus_1.KalturaAttachmentAssetStatus, subType: 'KalturaAttachmentAssetStatus' },
            statusIn: { type: 's' },
            statusNotIn: { type: 's' }
        });
        return result;
    };
    return KalturaAttachmentAssetBaseFilter;
}(KalturaAssetFilter_1.KalturaAssetFilter));
exports.KalturaAttachmentAssetBaseFilter = KalturaAttachmentAssetBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAttachmentAssetBaseFilter', KalturaAttachmentAssetBaseFilter);
