// kaltura-node-typescript-client/api/types/KalturaPlayReadyPolicy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPlayReadyPolicy = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaPlayReadyLicenseRemovalPolicy_1 = require("./KalturaPlayReadyLicenseRemovalPolicy");
var KalturaPlayReadyMinimumLicenseSecurityLevel_1 = require("./KalturaPlayReadyMinimumLicenseSecurityLevel");
var KalturaPlayReadyRight_1 = require("./KalturaPlayReadyRight");
var KalturaDrmPolicy_1 = require("./KalturaDrmPolicy");
var KalturaPlayReadyPolicy = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPlayReadyPolicy, _super);
    function KalturaPlayReadyPolicy(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.rights === 'undefined')
            _this.rights = [];
        return _this;
    }
    KalturaPlayReadyPolicy.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPlayReadyPolicy' },
            gracePeriod: { type: 'n' },
            licenseRemovalPolicy: { type: 'en', subTypeConstructor: KalturaPlayReadyLicenseRemovalPolicy_1.KalturaPlayReadyLicenseRemovalPolicy, subType: 'KalturaPlayReadyLicenseRemovalPolicy' },
            licenseRemovalDuration: { type: 'n' },
            minSecurityLevel: { type: 'en', subTypeConstructor: KalturaPlayReadyMinimumLicenseSecurityLevel_1.KalturaPlayReadyMinimumLicenseSecurityLevel, subType: 'KalturaPlayReadyMinimumLicenseSecurityLevel' },
            rights: { type: 'a', subTypeConstructor: KalturaPlayReadyRight_1.KalturaPlayReadyRight, subType: 'KalturaPlayReadyRight' }
        });
        return result;
    };
    return KalturaPlayReadyPolicy;
}(KalturaDrmPolicy_1.KalturaDrmPolicy));
exports.KalturaPlayReadyPolicy = KalturaPlayReadyPolicy;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPlayReadyPolicy', KalturaPlayReadyPolicy);
