// kaltura-node-typescript-client/api/types/KalturaCaptionsAlignment.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCaptionsAlignment = void 0;
var KalturaCaptionsAlignment;
(function (KalturaCaptionsAlignment) {
    KalturaCaptionsAlignment[KalturaCaptionsAlignment["bottomCenter"] = 2] = "bottomCenter";
    KalturaCaptionsAlignment[KalturaCaptionsAlignment["bottomLeft"] = 1] = "bottomLeft";
    KalturaCaptionsAlignment[KalturaCaptionsAlignment["bottomRight"] = 3] = "bottomRight";
    KalturaCaptionsAlignment[KalturaCaptionsAlignment["centerCenter"] = 10] = "centerCenter";
    KalturaCaptionsAlignment[KalturaCaptionsAlignment["centerLeft"] = 8] = "centerLeft";
    KalturaCaptionsAlignment[KalturaCaptionsAlignment["centerRight"] = 11] = "centerRight";
    KalturaCaptionsAlignment[KalturaCaptionsAlignment["topCenter"] = 6] = "topCenter";
    KalturaCaptionsAlignment[KalturaCaptionsAlignment["topLeft"] = 4] = "topLeft";
    KalturaCaptionsAlignment[KalturaCaptionsAlignment["topRight"] = 7] = "topRight";
})(KalturaCaptionsAlignment = exports.KalturaCaptionsAlignment || (exports.KalturaCaptionsAlignment = {}));
