// kaltura-node-typescript-client/api/types/KalturaCategory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCategory = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaAppearInListType_1 = require("./KalturaAppearInListType");
var KalturaPrivacyType_1 = require("./KalturaPrivacyType");
var KalturaInheritanceType_1 = require("./KalturaInheritanceType");
var KalturaUserJoinPolicyType_1 = require("./KalturaUserJoinPolicyType");
var KalturaCategoryUserPermissionLevel_1 = require("./KalturaCategoryUserPermissionLevel");
var KalturaContributionPolicyType_1 = require("./KalturaContributionPolicyType");
var KalturaCategoryStatus_1 = require("./KalturaCategoryStatus");
var KalturaCategoryOrderBy_1 = require("./KalturaCategoryOrderBy");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaCategory = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCategory, _super);
    function KalturaCategory(data) {
        return _super.call(this, data) || this;
    }
    KalturaCategory.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCategory' },
            id: { type: 'n', readOnly: true },
            parentId: { type: 'n' },
            depth: { type: 'n', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            name: { type: 's' },
            fullName: { type: 's', readOnly: true },
            fullIds: { type: 's', readOnly: true },
            entriesCount: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            description: { type: 's' },
            tags: { type: 's' },
            appearInList: { type: 'en', subTypeConstructor: KalturaAppearInListType_1.KalturaAppearInListType, subType: 'KalturaAppearInListType' },
            privacy: { type: 'en', subTypeConstructor: KalturaPrivacyType_1.KalturaPrivacyType, subType: 'KalturaPrivacyType' },
            inheritanceType: { type: 'en', subTypeConstructor: KalturaInheritanceType_1.KalturaInheritanceType, subType: 'KalturaInheritanceType' },
            userJoinPolicy: { type: 'en', readOnly: true, subTypeConstructor: KalturaUserJoinPolicyType_1.KalturaUserJoinPolicyType, subType: 'KalturaUserJoinPolicyType' },
            defaultPermissionLevel: { type: 'en', subTypeConstructor: KalturaCategoryUserPermissionLevel_1.KalturaCategoryUserPermissionLevel, subType: 'KalturaCategoryUserPermissionLevel' },
            owner: { type: 's' },
            directEntriesCount: { type: 'n', readOnly: true },
            referenceId: { type: 's' },
            contributionPolicy: { type: 'en', subTypeConstructor: KalturaContributionPolicyType_1.KalturaContributionPolicyType, subType: 'KalturaContributionPolicyType' },
            membersCount: { type: 'n', readOnly: true },
            pendingMembersCount: { type: 'n', readOnly: true },
            privacyContext: { type: 's' },
            privacyContexts: { type: 's', readOnly: true },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaCategoryStatus_1.KalturaCategoryStatus, subType: 'KalturaCategoryStatus' },
            inheritedParentId: { type: 'n', readOnly: true },
            partnerSortValue: { type: 'n' },
            partnerData: { type: 's' },
            defaultOrderBy: { type: 'es', subTypeConstructor: KalturaCategoryOrderBy_1.KalturaCategoryOrderBy, subType: 'KalturaCategoryOrderBy' },
            directSubCategoriesCount: { type: 'n', readOnly: true },
            moderation: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            pendingEntriesCount: { type: 'n', readOnly: true },
            isAggregationCategory: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            aggregationCategories: { type: 's' },
            adminTags: { type: 's' }
        });
        return result;
    };
    return KalturaCategory;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaCategory = KalturaCategory;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCategory', KalturaCategory);
