// kaltura-node-typescript-client/api/types/KalturaViewHistoryUserEntryAdvancedFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaViewHistoryUserEntryAdvancedFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUserEntryExtendedStatus_1 = require("./KalturaUserEntryExtendedStatus");
var KalturaSearchItem_1 = require("./KalturaSearchItem");
var KalturaViewHistoryUserEntryAdvancedFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaViewHistoryUserEntryAdvancedFilter, _super);
    function KalturaViewHistoryUserEntryAdvancedFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaViewHistoryUserEntryAdvancedFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaViewHistoryUserEntryAdvancedFilter' },
            idEqual: { type: 's' },
            idIn: { type: 's' },
            userIdEqual: { type: 's' },
            userIdIn: { type: 's' },
            updatedAtGreaterThanOrEqual: { type: 's' },
            updatedAtLessThanOrEqual: { type: 's' },
            extendedStatusEqual: { type: 'es', subTypeConstructor: KalturaUserEntryExtendedStatus_1.KalturaUserEntryExtendedStatus, subType: 'KalturaUserEntryExtendedStatus' },
            extendedStatusIn: { type: 's' }
        });
        return result;
    };
    return KalturaViewHistoryUserEntryAdvancedFilter;
}(KalturaSearchItem_1.KalturaSearchItem));
exports.KalturaViewHistoryUserEntryAdvancedFilter = KalturaViewHistoryUserEntryAdvancedFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaViewHistoryUserEntryAdvancedFilter', KalturaViewHistoryUserEntryAdvancedFilter);
