// kaltura-node-typescript-client/api/types/KalturaResponseProfileBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaResponseProfileBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaResponseProfileStatus_1 = require("./KalturaResponseProfileStatus");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaResponseProfileBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaResponseProfileBaseFilter, _super);
    function KalturaResponseProfileBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaResponseProfileBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaResponseProfileBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            systemNameEqual: { type: 's' },
            systemNameIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaResponseProfileStatus_1.KalturaResponseProfileStatus, subType: 'KalturaResponseProfileStatus' },
            statusIn: { type: 's' }
        });
        return result;
    };
    return KalturaResponseProfileBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaResponseProfileBaseFilter = KalturaResponseProfileBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaResponseProfileBaseFilter', KalturaResponseProfileBaseFilter);
