// kaltura-node-typescript-client/api/types/KalturaDeliveryProfileBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDeliveryProfileBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaPlaybackProtocol_1 = require("./KalturaPlaybackProtocol");
var KalturaDeliveryStatus_1 = require("./KalturaDeliveryStatus");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaDeliveryProfileBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDeliveryProfileBaseFilter, _super);
    function KalturaDeliveryProfileBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaDeliveryProfileBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDeliveryProfileBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            systemNameEqual: { type: 's' },
            systemNameIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            streamerTypeEqual: { type: 'es', subTypeConstructor: KalturaPlaybackProtocol_1.KalturaPlaybackProtocol, subType: 'KalturaPlaybackProtocol' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaDeliveryStatus_1.KalturaDeliveryStatus, subType: 'KalturaDeliveryStatus' },
            statusIn: { type: 's' }
        });
        return result;
    };
    return KalturaDeliveryProfileBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaDeliveryProfileBaseFilter = KalturaDeliveryProfileBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDeliveryProfileBaseFilter', KalturaDeliveryProfileBaseFilter);
