// kaltura-node-typescript-client/api/types/KalturaPermissionItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPermissionItem = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaPermissionItemType_1 = require("./KalturaPermissionItemType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaPermissionItem = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPermissionItem, _super);
    function KalturaPermissionItem(data) {
        return _super.call(this, data) || this;
    }
    KalturaPermissionItem.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPermissionItem' },
            id: { type: 'n', readOnly: true },
            type: { type: 'es', readOnly: true, subTypeConstructor: KalturaPermissionItemType_1.KalturaPermissionItemType, subType: 'KalturaPermissionItemType' },
            partnerId: { type: 'n', readOnly: true },
            tags: { type: 's' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true }
        });
        return result;
    };
    return KalturaPermissionItem;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaPermissionItem = KalturaPermissionItem;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPermissionItem', KalturaPermissionItem);
