// kaltura-node-typescript-client/api/types/KalturaDrmLicenseAccessDetails.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDrmLicenseAccessDetails = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaKeyValue_1 = require("./KalturaKeyValue");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaDrmLicenseAccessDetails = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDrmLicenseAccessDetails, _super);
    function KalturaDrmLicenseAccessDetails(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.licenseParams === 'undefined')
            _this.licenseParams = [];
        return _this;
    }
    KalturaDrmLicenseAccessDetails.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDrmLicenseAccessDetails' },
            policy: { type: 's' },
            duration: { type: 'n' },
            absolute_duration: { type: 'n' },
            licenseParams: { type: 'a', subTypeConstructor: KalturaKeyValue_1.KalturaKeyValue, subType: 'KalturaKeyValue' }
        });
        return result;
    };
    return KalturaDrmLicenseAccessDetails;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaDrmLicenseAccessDetails = KalturaDrmLicenseAccessDetails;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDrmLicenseAccessDetails', KalturaDrmLicenseAccessDetails);
