// kaltura-node-typescript-client/api/types/KalturaHttpNotificationObjectData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaHttpNotificationObjectData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaResponseType_1 = require("./KalturaResponseType");
var KalturaKeyValue_1 = require("./KalturaKeyValue");
var KalturaHttpNotificationData_1 = require("./KalturaHttpNotificationData");
var KalturaHttpNotificationObjectData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaHttpNotificationObjectData, _super);
    function KalturaHttpNotificationObjectData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.dataStringReplacements === 'undefined')
            _this.dataStringReplacements = [];
        return _this;
    }
    KalturaHttpNotificationObjectData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaHttpNotificationObjectData' },
            apiObjectType: { type: 's' },
            format: { type: 'en', subTypeConstructor: KalturaResponseType_1.KalturaResponseType, subType: 'KalturaResponseType' },
            ignoreNull: { type: 'b' },
            code: { type: 's' },
            dataStringReplacements: { type: 'a', subTypeConstructor: KalturaKeyValue_1.KalturaKeyValue, subType: 'KalturaKeyValue' }
        });
        return result;
    };
    return KalturaHttpNotificationObjectData;
}(KalturaHttpNotificationData_1.KalturaHttpNotificationData));
exports.KalturaHttpNotificationObjectData = KalturaHttpNotificationObjectData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaHttpNotificationObjectData', KalturaHttpNotificationObjectData);
