// kaltura-node-typescript-client/api/types/KalturaVendorCatalogItemBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaVendorCatalogItemBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaVendorCatalogItemStatus_1 = require("./KalturaVendorCatalogItemStatus");
var KalturaVendorServiceType_1 = require("./KalturaVendorServiceType");
var KalturaVendorServiceFeature_1 = require("./KalturaVendorServiceFeature");
var KalturaVendorServiceTurnAroundTime_1 = require("./KalturaVendorServiceTurnAroundTime");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaVendorCatalogItemBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaVendorCatalogItemBaseFilter, _super);
    function KalturaVendorCatalogItemBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaVendorCatalogItemBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaVendorCatalogItemBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            idNotIn: { type: 's' },
            vendorPartnerIdEqual: { type: 'n' },
            vendorPartnerIdIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaVendorCatalogItemStatus_1.KalturaVendorCatalogItemStatus, subType: 'KalturaVendorCatalogItemStatus' },
            statusIn: { type: 's' },
            serviceTypeEqual: { type: 'en', subTypeConstructor: KalturaVendorServiceType_1.KalturaVendorServiceType, subType: 'KalturaVendorServiceType' },
            serviceTypeIn: { type: 's' },
            serviceFeatureEqual: { type: 'en', subTypeConstructor: KalturaVendorServiceFeature_1.KalturaVendorServiceFeature, subType: 'KalturaVendorServiceFeature' },
            serviceFeatureIn: { type: 's' },
            turnAroundTimeEqual: { type: 'en', subTypeConstructor: KalturaVendorServiceTurnAroundTime_1.KalturaVendorServiceTurnAroundTime, subType: 'KalturaVendorServiceTurnAroundTime' },
            turnAroundTimeIn: { type: 's' }
        });
        return result;
    };
    return KalturaVendorCatalogItemBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaVendorCatalogItemBaseFilter = KalturaVendorCatalogItemBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaVendorCatalogItemBaseFilter', KalturaVendorCatalogItemBaseFilter);
