// kaltura-node-typescript-client/api/types/KalturaPlaybackContext.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPlaybackContext = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaPlaybackSource_1 = require("./KalturaPlaybackSource");
var KalturaCaptionPlaybackPluginData_1 = require("./KalturaCaptionPlaybackPluginData");
var KalturaFlavorAsset_1 = require("./KalturaFlavorAsset");
var KalturaRuleAction_1 = require("./KalturaRuleAction");
var KalturaAccessControlMessage_1 = require("./KalturaAccessControlMessage");
var KalturaObject_1 = require("./KalturaObject");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaPlaybackContext = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPlaybackContext, _super);
    function KalturaPlaybackContext(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.sources === 'undefined')
            _this.sources = [];
        if (typeof _this.playbackCaptions === 'undefined')
            _this.playbackCaptions = [];
        if (typeof _this.flavorAssets === 'undefined')
            _this.flavorAssets = [];
        if (typeof _this.actions === 'undefined')
            _this.actions = [];
        if (typeof _this.messages === 'undefined')
            _this.messages = [];
        if (typeof _this.bumperData === 'undefined')
            _this.bumperData = [];
        return _this;
    }
    KalturaPlaybackContext.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPlaybackContext' },
            sources: { type: 'a', subTypeConstructor: KalturaPlaybackSource_1.KalturaPlaybackSource, subType: 'KalturaPlaybackSource' },
            playbackCaptions: { type: 'a', subTypeConstructor: KalturaCaptionPlaybackPluginData_1.KalturaCaptionPlaybackPluginData, subType: 'KalturaCaptionPlaybackPluginData' },
            flavorAssets: { type: 'a', subTypeConstructor: KalturaFlavorAsset_1.KalturaFlavorAsset, subType: 'KalturaFlavorAsset' },
            actions: { type: 'a', subTypeConstructor: KalturaRuleAction_1.KalturaRuleAction, subType: 'KalturaRuleAction' },
            messages: { type: 'a', subTypeConstructor: KalturaAccessControlMessage_1.KalturaAccessControlMessage, subType: 'KalturaAccessControlMessage' },
            bumperData: { type: 'a', subTypeConstructor: KalturaObject_1.KalturaObject, subType: 'KalturaObject' }
        });
        return result;
    };
    return KalturaPlaybackContext;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaPlaybackContext = KalturaPlaybackContext;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPlaybackContext', KalturaPlaybackContext);
