// kaltura-node-typescript-client/api/types/KalturaCaptionAssetItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCaptionAssetItem = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCaptionAsset_1 = require("./KalturaCaptionAsset");
var KalturaBaseEntry_1 = require("./KalturaBaseEntry");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaCaptionAssetItem = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCaptionAssetItem, _super);
    function KalturaCaptionAssetItem(data) {
        return _super.call(this, data) || this;
    }
    KalturaCaptionAssetItem.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCaptionAssetItem' },
            asset: { type: 'o', subTypeConstructor: KalturaCaptionAsset_1.KalturaCaptionAsset, subType: 'KalturaCaptionAsset' },
            entry: { type: 'o', subTypeConstructor: KalturaBaseEntry_1.KalturaBaseEntry, subType: 'KalturaBaseEntry' },
            startTime: { type: 'n' },
            endTime: { type: 'n' },
            content: { type: 's' }
        });
        return result;
    };
    return KalturaCaptionAssetItem;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaCaptionAssetItem = KalturaCaptionAssetItem;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCaptionAssetItem', KalturaCaptionAssetItem);
