// kaltura-node-typescript-client/api/types/KalturaKafkaNotificationTemplate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaKafkaNotificationTemplate = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaKafkaNotificationFormat_1 = require("./KalturaKafkaNotificationFormat");
var KalturaEventNotificationTemplate_1 = require("./KalturaEventNotificationTemplate");
var KalturaKafkaNotificationTemplate = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaKafkaNotificationTemplate, _super);
    function KalturaKafkaNotificationTemplate(data) {
        return _super.call(this, data) || this;
    }
    KalturaKafkaNotificationTemplate.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaKafkaNotificationTemplate' },
            topicName: { type: 's' },
            partitionKey: { type: 's' },
            messageFormat: { type: 'en', subTypeConstructor: KalturaKafkaNotificationFormat_1.KalturaKafkaNotificationFormat, subType: 'KalturaKafkaNotificationFormat' },
            apiObjectType: { type: 's' },
            responseProfileSystemName: { type: 's' },
            requiresPermissions: { type: 's' }
        });
        return result;
    };
    return KalturaKafkaNotificationTemplate;
}(KalturaEventNotificationTemplate_1.KalturaEventNotificationTemplate));
exports.KalturaKafkaNotificationTemplate = KalturaKafkaNotificationTemplate;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaKafkaNotificationTemplate', KalturaKafkaNotificationTemplate);
