// kaltura-node-typescript-client/api/types/KalturaPermissionBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPermissionBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaPermissionType_1 = require("./KalturaPermissionType");
var KalturaPermissionStatus_1 = require("./KalturaPermissionStatus");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaPermissionBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPermissionBaseFilter, _super);
    function KalturaPermissionBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaPermissionBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPermissionBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            typeEqual: { type: 'en', subTypeConstructor: KalturaPermissionType_1.KalturaPermissionType, subType: 'KalturaPermissionType' },
            typeIn: { type: 's' },
            nameEqual: { type: 's' },
            nameIn: { type: 's' },
            friendlyNameLike: { type: 's' },
            descriptionLike: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaPermissionStatus_1.KalturaPermissionStatus, subType: 'KalturaPermissionStatus' },
            statusIn: { type: 's' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            dependsOnPermissionNamesMultiLikeOr: { type: 's' },
            dependsOnPermissionNamesMultiLikeAnd: { type: 's' },
            tagsMultiLikeOr: { type: 's' },
            tagsMultiLikeAnd: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' }
        });
        return result;
    };
    return KalturaPermissionBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaPermissionBaseFilter = KalturaPermissionBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPermissionBaseFilter', KalturaPermissionBaseFilter);
