// kaltura-node-typescript-client/api/types/KalturaLiveEntryServerNode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveEntryServerNode = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaLiveStreamParams_1 = require("./KalturaLiveStreamParams");
var KalturaLiveEntryServerNodeRecordingInfo_1 = require("./KalturaLiveEntryServerNodeRecordingInfo");
var KalturaViewMode_1 = require("./KalturaViewMode");
var KalturaEntryServerNode_1 = require("./KalturaEntryServerNode");
var KalturaLiveEntryServerNode = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveEntryServerNode, _super);
    function KalturaLiveEntryServerNode(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.streams === 'undefined')
            _this.streams = [];
        if (typeof _this.recordingInfo === 'undefined')
            _this.recordingInfo = [];
        return _this;
    }
    KalturaLiveEntryServerNode.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveEntryServerNode' },
            streams: { type: 'a', subTypeConstructor: KalturaLiveStreamParams_1.KalturaLiveStreamParams, subType: 'KalturaLiveStreamParams' },
            recordingInfo: { type: 'a', subTypeConstructor: KalturaLiveEntryServerNodeRecordingInfo_1.KalturaLiveEntryServerNodeRecordingInfo, subType: 'KalturaLiveEntryServerNodeRecordingInfo' },
            isPlayableUser: { type: 'b' },
            viewMode: { type: 'en', subTypeConstructor: KalturaViewMode_1.KalturaViewMode, subType: 'KalturaViewMode' },
            featuresUpdatedAt: { type: 'd' },
            viewModeUpdatedAt: { type: 'd' }
        });
        return result;
    };
    return KalturaLiveEntryServerNode;
}(KalturaEntryServerNode_1.KalturaEntryServerNode));
exports.KalturaLiveEntryServerNode = KalturaLiveEntryServerNode;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveEntryServerNode', KalturaLiveEntryServerNode);
