// kaltura-node-typescript-client/api/types/KalturaBusinessProcessServer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBusinessProcessServer = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBusinessProcessServerStatus_1 = require("./KalturaBusinessProcessServerStatus");
var KalturaBusinessProcessProvider_1 = require("./KalturaBusinessProcessProvider");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaBusinessProcessServer = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBusinessProcessServer, _super);
    function KalturaBusinessProcessServer(data) {
        return _super.call(this, data) || this;
    }
    KalturaBusinessProcessServer.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBusinessProcessServer' },
            id: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            name: { type: 's' },
            systemName: { type: 's' },
            description: { type: 's' },
            status: { type: 'es', readOnly: true, subTypeConstructor: KalturaBusinessProcessServerStatus_1.KalturaBusinessProcessServerStatus, subType: 'KalturaBusinessProcessServerStatus' },
            type: { type: 'es', readOnly: true, subTypeConstructor: KalturaBusinessProcessProvider_1.KalturaBusinessProcessProvider, subType: 'KalturaBusinessProcessProvider' },
            dc: { type: 'n' }
        });
        return result;
    };
    return KalturaBusinessProcessServer;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaBusinessProcessServer = KalturaBusinessProcessServer;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBusinessProcessServer', KalturaBusinessProcessServer);
