// kaltura-node-typescript-client/api/types/KalturaLiveStreamScheduleEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveStreamScheduleEvent = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaLiveFeature_1 = require("./KalturaLiveFeature");
var KalturaBaseLiveScheduleEvent_1 = require("./KalturaBaseLiveScheduleEvent");
var KalturaLiveStreamScheduleEvent = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveStreamScheduleEvent, _super);
    function KalturaLiveStreamScheduleEvent(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.liveFeatures === 'undefined')
            _this.liveFeatures = [];
        return _this;
    }
    KalturaLiveStreamScheduleEvent.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveStreamScheduleEvent' },
            sourceEntryId: { type: 's' },
            projectedAudience: { type: 'n' },
            preStartTime: { type: 'n' },
            postEndTime: { type: 'n' },
            preStartEntryId: { type: 's' },
            postEndEntryId: { type: 's' },
            isContentInterruptible: { type: 'b' },
            liveFeatures: { type: 'a', subTypeConstructor: KalturaLiveFeature_1.KalturaLiveFeature, subType: 'KalturaLiveFeature' }
        });
        return result;
    };
    return KalturaLiveStreamScheduleEvent;
}(KalturaBaseLiveScheduleEvent_1.KalturaBaseLiveScheduleEvent));
exports.KalturaLiveStreamScheduleEvent = KalturaLiveStreamScheduleEvent;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveStreamScheduleEvent', KalturaLiveStreamScheduleEvent);
