// kaltura-node-typescript-client/api/types/KalturaConfMaps.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaConfMaps = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfMapsSourceLocation_1 = require("./KalturaConfMapsSourceLocation");
var KalturaConfMapsStatus_1 = require("./KalturaConfMapsStatus");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaConfMaps = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaConfMaps, _super);
    function KalturaConfMaps(data) {
        return _super.call(this, data) || this;
    }
    KalturaConfMaps.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaConfMaps' },
            name: { type: 's' },
            content: { type: 's' },
            rawData: { type: 's' },
            userId: { type: 's' },
            isEditable: { type: 'b', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            relatedHost: { type: 's' },
            version: { type: 'n', readOnly: true },
            sourceLocation: { type: 'es', subTypeConstructor: KalturaConfMapsSourceLocation_1.KalturaConfMapsSourceLocation, subType: 'KalturaConfMapsSourceLocation' },
            remarks: { type: 's' },
            status: { type: 'en', subTypeConstructor: KalturaConfMapsStatus_1.KalturaConfMapsStatus, subType: 'KalturaConfMapsStatus' },
            changeDescription: { type: 's' }
        });
        return result;
    };
    return KalturaConfMaps;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaConfMaps = KalturaConfMaps;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaConfMaps', KalturaConfMaps);
