// kaltura-node-typescript-client/api/types/KalturaZoomMeetingMetadata.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaZoomMeetingMetadata = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaRecordingType_1 = require("./KalturaRecordingType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaZoomMeetingMetadata = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaZoomMeetingMetadata, _super);
    function KalturaZoomMeetingMetadata(data) {
        return _super.call(this, data) || this;
    }
    KalturaZoomMeetingMetadata.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaZoomMeetingMetadata' },
            uuid: { type: 's' },
            meetingId: { type: 's' },
            accountId: { type: 's' },
            hostId: { type: 's' },
            topic: { type: 's' },
            meetingStartTime: { type: 's' },
            type: { type: 'en', subTypeConstructor: KalturaRecordingType_1.KalturaRecordingType, subType: 'KalturaRecordingType' }
        });
        return result;
    };
    return KalturaZoomMeetingMetadata;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaZoomMeetingMetadata = KalturaZoomMeetingMetadata;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaZoomMeetingMetadata', KalturaZoomMeetingMetadata);
