// kaltura-node-typescript-client/api/types/KalturaWidgetBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaWidgetBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaWidgetBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaWidgetBaseFilter, _super);
    function KalturaWidgetBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaWidgetBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaWidgetBaseFilter' },
            idEqual: { type: 's' },
            idIn: { type: 's' },
            sourceWidgetIdEqual: { type: 's' },
            rootWidgetIdEqual: { type: 's' },
            partnerIdEqual: { type: 'n' },
            entryIdEqual: { type: 's' },
            uiConfIdEqual: { type: 'n' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            partnerDataLike: { type: 's' }
        });
        return result;
    };
    return KalturaWidgetBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaWidgetBaseFilter = KalturaWidgetBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaWidgetBaseFilter', KalturaWidgetBaseFilter);
