// kaltura-node-typescript-client/api/types/KalturaThumbCuePointOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaThumbCuePointOrderBy = void 0;
var KalturaThumbCuePointOrderBy;
(function (KalturaThumbCuePointOrderBy) {
    KalturaThumbCuePointOrderBy["createdAtAsc"] = "+createdAt";
    KalturaThumbCuePointOrderBy["createdAtDesc"] = "-createdAt";
    KalturaThumbCuePointOrderBy["intIdAsc"] = "+intId";
    KalturaThumbCuePointOrderBy["intIdDesc"] = "-intId";
    KalturaThumbCuePointOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaThumbCuePointOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaThumbCuePointOrderBy["startTimeAsc"] = "+startTime";
    KalturaThumbCuePointOrderBy["startTimeDesc"] = "-startTime";
    KalturaThumbCuePointOrderBy["triggeredAtAsc"] = "+triggeredAt";
    KalturaThumbCuePointOrderBy["triggeredAtDesc"] = "-triggeredAt";
    KalturaThumbCuePointOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaThumbCuePointOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaThumbCuePointOrderBy = exports.KalturaThumbCuePointOrderBy || (exports.KalturaThumbCuePointOrderBy = {}));
