// kaltura-node-typescript-client/api/types/KalturaConvertCollectionFlavorData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaConvertCollectionFlavorData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaConvertCollectionFlavorData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaConvertCollectionFlavorData, _super);
    function KalturaConvertCollectionFlavorData(data) {
        return _super.call(this, data) || this;
    }
    KalturaConvertCollectionFlavorData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaConvertCollectionFlavorData' },
            flavorAssetId: { type: 's' },
            flavorParamsOutputId: { type: 'n' },
            readyBehavior: { type: 'n' },
            videoBitrate: { type: 'n' },
            audioBitrate: { type: 'n' },
            destFileSyncLocalPath: { type: 's' },
            destFileSyncRemoteUrl: { type: 's' }
        });
        return result;
    };
    return KalturaConvertCollectionFlavorData;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaConvertCollectionFlavorData = KalturaConvertCollectionFlavorData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaConvertCollectionFlavorData', KalturaConvertCollectionFlavorData);
