// kaltura-node-typescript-client/api/types/KalturaCielo24JobProviderData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCielo24JobProviderData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCielo24Priority_1 = require("./KalturaCielo24Priority");
var KalturaCielo24Fidelity_1 = require("./KalturaCielo24Fidelity");
var KalturaLanguage_1 = require("./KalturaLanguage");
var KalturaIntegrationJobProviderData_1 = require("./KalturaIntegrationJobProviderData");
var KalturaCielo24JobProviderData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCielo24JobProviderData, _super);
    function KalturaCielo24JobProviderData(data) {
        return _super.call(this, data) || this;
    }
    KalturaCielo24JobProviderData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCielo24JobProviderData' },
            entryId: { type: 's' },
            flavorAssetId: { type: 's' },
            captionAssetFormats: { type: 's' },
            priority: { type: 'es', subTypeConstructor: KalturaCielo24Priority_1.KalturaCielo24Priority, subType: 'KalturaCielo24Priority' },
            fidelity: { type: 'es', subTypeConstructor: KalturaCielo24Fidelity_1.KalturaCielo24Fidelity, subType: 'KalturaCielo24Fidelity' },
            username: { type: 's', readOnly: true },
            password: { type: 's', readOnly: true },
            baseUrl: { type: 's', readOnly: true },
            spokenLanguage: { type: 'es', subTypeConstructor: KalturaLanguage_1.KalturaLanguage, subType: 'KalturaLanguage' },
            replaceMediaContent: { type: 'b' },
            additionalParameters: { type: 's' }
        });
        return result;
    };
    return KalturaCielo24JobProviderData;
}(KalturaIntegrationJobProviderData_1.KalturaIntegrationJobProviderData));
exports.KalturaCielo24JobProviderData = KalturaCielo24JobProviderData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCielo24JobProviderData', KalturaCielo24JobProviderData);
