// kaltura-node-typescript-client/api/types/KalturaESearchGroupFieldName.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaESearchGroupFieldName = void 0;
var KalturaESearchGroupFieldName;
(function (KalturaESearchGroupFieldName) {
    KalturaESearchGroupFieldName["capabilities"] = "capabilities";
    KalturaESearchGroupFieldName["company"] = "company";
    KalturaESearchGroupFieldName["country"] = "country";
    KalturaESearchGroupFieldName["createdAt"] = "created_at";
    KalturaESearchGroupFieldName["email"] = "email";
    KalturaESearchGroupFieldName["externalId"] = "external_id";
    KalturaESearchGroupFieldName["firstName"] = "first_name";
    KalturaESearchGroupFieldName["fullName"] = "full_name";
    KalturaESearchGroupFieldName["groupIds"] = "group_ids";
    KalturaESearchGroupFieldName["isAdmin"] = "is_admin";
    KalturaESearchGroupFieldName["isHashed"] = "is_hashed";
    KalturaESearchGroupFieldName["lastName"] = "last_name";
    KalturaESearchGroupFieldName["loginEnabled"] = "login_enabled";
    KalturaESearchGroupFieldName["permissionNames"] = "permission_names";
    KalturaESearchGroupFieldName["roleIds"] = "role_ids";
    KalturaESearchGroupFieldName["screenName"] = "screen_name";
    KalturaESearchGroupFieldName["tags"] = "tags";
    KalturaESearchGroupFieldName["title"] = "title";
    KalturaESearchGroupFieldName["updatedAt"] = "updated_at";
    KalturaESearchGroupFieldName["userId"] = "user_id";
})(KalturaESearchGroupFieldName = exports.KalturaESearchGroupFieldName || (exports.KalturaESearchGroupFieldName = {}));
