// kaltura-node-typescript-client/api/types/KalturaCategoryEntryAdvancedFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCategoryEntryAdvancedFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCategoryEntryAdvancedOrderBy_1 = require("./KalturaCategoryEntryAdvancedOrderBy");
var KalturaSearchItem_1 = require("./KalturaSearchItem");
var KalturaCategoryEntryAdvancedFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCategoryEntryAdvancedFilter, _super);
    function KalturaCategoryEntryAdvancedFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaCategoryEntryAdvancedFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCategoryEntryAdvancedFilter' },
            categoriesMatchOr: { type: 's' },
            categoryEntryStatusIn: { type: 's' },
            orderBy: { type: 'es', subTypeConstructor: KalturaCategoryEntryAdvancedOrderBy_1.KalturaCategoryEntryAdvancedOrderBy, subType: 'KalturaCategoryEntryAdvancedOrderBy' },
            categoryIdEqual: { type: 'n' }
        });
        return result;
    };
    return KalturaCategoryEntryAdvancedFilter;
}(KalturaSearchItem_1.KalturaSearchItem));
exports.KalturaCategoryEntryAdvancedFilter = KalturaCategoryEntryAdvancedFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCategoryEntryAdvancedFilter', KalturaCategoryEntryAdvancedFilter);
