// kaltura-node-typescript-client/api/types/KalturaDeliveryProfileForensicWatermark.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDeliveryProfileForensicWatermark = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaKeyValue_1 = require("./KalturaKeyValue");
var KalturaDeliveryProfile_1 = require("./KalturaDeliveryProfile");
var KalturaDeliveryProfileForensicWatermark = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDeliveryProfileForensicWatermark, _super);
    function KalturaDeliveryProfileForensicWatermark(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.internalUrl === 'undefined')
            _this.internalUrl = [];
        return _this;
    }
    KalturaDeliveryProfileForensicWatermark.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDeliveryProfileForensicWatermark' },
            internalUrl: { type: 'a', subTypeConstructor: KalturaKeyValue_1.KalturaKeyValue, subType: 'KalturaKeyValue' },
            encryptionKey: { type: 's' },
            encryptionIv: { type: 's' },
            encryptionRegex: { type: 's' }
        });
        return result;
    };
    return KalturaDeliveryProfileForensicWatermark;
}(KalturaDeliveryProfile_1.KalturaDeliveryProfile));
exports.KalturaDeliveryProfileForensicWatermark = KalturaDeliveryProfileForensicWatermark;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDeliveryProfileForensicWatermark', KalturaDeliveryProfileForensicWatermark);
