// kaltura-node-typescript-client/api/types/KalturaLiveRestreamFeature.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveRestreamFeature = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaLiveFeature_1 = require("./KalturaLiveFeature");
var KalturaLiveRestreamFeature = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveRestreamFeature, _super);
    function KalturaLiveRestreamFeature(data) {
        return _super.call(this, data) || this;
    }
    KalturaLiveRestreamFeature.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveRestreamFeature' },
            primaryUrl: { type: 's' },
            secondaryUrl: { type: 's' },
            playbackUrl: { type: 's' },
            streamKey: { type: 's' }
        });
        return result;
    };
    return KalturaLiveRestreamFeature;
}(KalturaLiveFeature_1.KalturaLiveFeature));
exports.KalturaLiveRestreamFeature = KalturaLiveRestreamFeature;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveRestreamFeature', KalturaLiveRestreamFeature);
