// kaltura-node-typescript-client/api/types/KalturaReachProfileContentDeletionPolicy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaReachProfileContentDeletionPolicy = void 0;
var KalturaReachProfileContentDeletionPolicy;
(function (KalturaReachProfileContentDeletionPolicy) {
    KalturaReachProfileContentDeletionPolicy[KalturaReachProfileContentDeletionPolicy["deleteAfterMonth"] = 4] = "deleteAfterMonth";
    KalturaReachProfileContentDeletionPolicy[KalturaReachProfileContentDeletionPolicy["deleteAfterThreeMonths"] = 5] = "deleteAfterThreeMonths";
    KalturaReachProfileContentDeletionPolicy[KalturaReachProfileContentDeletionPolicy["deleteAfterWeek"] = 3] = "deleteAfterWeek";
    KalturaReachProfileContentDeletionPolicy[KalturaReachProfileContentDeletionPolicy["deleteOnceProcessed"] = 2] = "deleteOnceProcessed";
    KalturaReachProfileContentDeletionPolicy[KalturaReachProfileContentDeletionPolicy["doNothing"] = 1] = "doNothing";
})(KalturaReachProfileContentDeletionPolicy = exports.KalturaReachProfileContentDeletionPolicy || (exports.KalturaReachProfileContentDeletionPolicy = {}));
