// kaltura-node-typescript-client/api/types/KalturaConversionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaConversionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConversionProfileStatus_1 = require("./KalturaConversionProfileStatus");
var KalturaConversionProfileType_1 = require("./KalturaConversionProfileType");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaCropDimensions_1 = require("./KalturaCropDimensions");
var KalturaMediaParserType_1 = require("./KalturaMediaParserType");
var KalturaEntryReplacementOptions_1 = require("./KalturaEntryReplacementOptions");
var KalturaLanguage_1 = require("./KalturaLanguage");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaConversionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaConversionProfile, _super);
    function KalturaConversionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaConversionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaConversionProfile' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            status: { type: 'es', subTypeConstructor: KalturaConversionProfileStatus_1.KalturaConversionProfileStatus, subType: 'KalturaConversionProfileStatus' },
            type: { type: 'es', subTypeConstructor: KalturaConversionProfileType_1.KalturaConversionProfileType, subType: 'KalturaConversionProfileType' },
            name: { type: 's' },
            systemName: { type: 's' },
            tags: { type: 's' },
            description: { type: 's' },
            defaultEntryId: { type: 's' },
            createdAt: { type: 'd', readOnly: true },
            flavorParamsIds: { type: 's' },
            isDefault: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            isPartnerDefault: { type: 'b', readOnly: true },
            cropDimensions: { type: 'o', subTypeConstructor: KalturaCropDimensions_1.KalturaCropDimensions, subType: 'KalturaCropDimensions' },
            clipStart: { type: 'n' },
            clipDuration: { type: 'n' },
            xslTransformation: { type: 's' },
            storageProfileId: { type: 'n' },
            mediaParserType: { type: 'es', subTypeConstructor: KalturaMediaParserType_1.KalturaMediaParserType, subType: 'KalturaMediaParserType' },
            calculateComplexity: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            collectionTags: { type: 's' },
            conditionalProfiles: { type: 's' },
            detectGOP: { type: 'n' },
            mediaInfoXslTransformation: { type: 's' },
            defaultReplacementOptions: { type: 'o', subTypeConstructor: KalturaEntryReplacementOptions_1.KalturaEntryReplacementOptions, subType: 'KalturaEntryReplacementOptions' },
            defaultAudioLang: { type: 'es', subTypeConstructor: KalturaLanguage_1.KalturaLanguage, subType: 'KalturaLanguage' },
            deliveryTag: { type: 's' }
        });
        return result;
    };
    return KalturaConversionProfile;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaConversionProfile = KalturaConversionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaConversionProfile', KalturaConversionProfile);
