// kaltura-node-typescript-client/api/types/KalturaNotificationType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaNotificationType = void 0;
var KalturaNotificationType;
(function (KalturaNotificationType) {
    KalturaNotificationType[KalturaNotificationType["entryAdd"] = 1] = "entryAdd";
    KalturaNotificationType[KalturaNotificationType["entryBlock"] = 4] = "entryBlock";
    KalturaNotificationType[KalturaNotificationType["entryDelete"] = 3] = "entryDelete";
    KalturaNotificationType[KalturaNotificationType["entryUpdate"] = 5] = "entryUpdate";
    KalturaNotificationType[KalturaNotificationType["entryUpdateModeration"] = 7] = "entryUpdateModeration";
    KalturaNotificationType[KalturaNotificationType["entryUpdateThumbnail"] = 6] = "entryUpdateThumbnail";
    KalturaNotificationType[KalturaNotificationType["entrUpdatePermissions"] = 2] = "entrUpdatePermissions";
    KalturaNotificationType[KalturaNotificationType["userAdd"] = 21] = "userAdd";
    KalturaNotificationType[KalturaNotificationType["userBanned"] = 26] = "userBanned";
})(KalturaNotificationType = exports.KalturaNotificationType || (exports.KalturaNotificationType = {}));
