// kaltura-node-typescript-client/api/types/KalturaLiveStreamEntryOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveStreamEntryOrderBy = void 0;
var KalturaLiveStreamEntryOrderBy;
(function (KalturaLiveStreamEntryOrderBy) {
    KalturaLiveStreamEntryOrderBy["createdAtAsc"] = "+createdAt";
    KalturaLiveStreamEntryOrderBy["createdAtDesc"] = "-createdAt";
    KalturaLiveStreamEntryOrderBy["durationAsc"] = "+duration";
    KalturaLiveStreamEntryOrderBy["durationDesc"] = "-duration";
    KalturaLiveStreamEntryOrderBy["endDateAsc"] = "+endDate";
    KalturaLiveStreamEntryOrderBy["endDateDesc"] = "-endDate";
    KalturaLiveStreamEntryOrderBy["firstBroadcastAsc"] = "+firstBroadcast";
    KalturaLiveStreamEntryOrderBy["firstBroadcastDesc"] = "-firstBroadcast";
    KalturaLiveStreamEntryOrderBy["lastBroadcastAsc"] = "+lastBroadcast";
    KalturaLiveStreamEntryOrderBy["lastBroadcastDesc"] = "-lastBroadcast";
    KalturaLiveStreamEntryOrderBy["lastPlayedAtAsc"] = "+lastPlayedAt";
    KalturaLiveStreamEntryOrderBy["lastPlayedAtDesc"] = "-lastPlayedAt";
    KalturaLiveStreamEntryOrderBy["mediaTypeAsc"] = "+mediaType";
    KalturaLiveStreamEntryOrderBy["mediaTypeDesc"] = "-mediaType";
    KalturaLiveStreamEntryOrderBy["moderationCountAsc"] = "+moderationCount";
    KalturaLiveStreamEntryOrderBy["moderationCountDesc"] = "-moderationCount";
    KalturaLiveStreamEntryOrderBy["nameAsc"] = "+name";
    KalturaLiveStreamEntryOrderBy["nameDesc"] = "-name";
    KalturaLiveStreamEntryOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaLiveStreamEntryOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaLiveStreamEntryOrderBy["playsAsc"] = "+plays";
    KalturaLiveStreamEntryOrderBy["playsDesc"] = "-plays";
    KalturaLiveStreamEntryOrderBy["rankAsc"] = "+rank";
    KalturaLiveStreamEntryOrderBy["rankDesc"] = "-rank";
    KalturaLiveStreamEntryOrderBy["recentAsc"] = "+recent";
    KalturaLiveStreamEntryOrderBy["recentDesc"] = "-recent";
    KalturaLiveStreamEntryOrderBy["startDateAsc"] = "+startDate";
    KalturaLiveStreamEntryOrderBy["startDateDesc"] = "-startDate";
    KalturaLiveStreamEntryOrderBy["totalRankAsc"] = "+totalRank";
    KalturaLiveStreamEntryOrderBy["totalRankDesc"] = "-totalRank";
    KalturaLiveStreamEntryOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaLiveStreamEntryOrderBy["updatedAtDesc"] = "-updatedAt";
    KalturaLiveStreamEntryOrderBy["viewsAsc"] = "+views";
    KalturaLiveStreamEntryOrderBy["viewsDesc"] = "-views";
    KalturaLiveStreamEntryOrderBy["weightAsc"] = "+weight";
    KalturaLiveStreamEntryOrderBy["weightDesc"] = "-weight";
})(KalturaLiveStreamEntryOrderBy = exports.KalturaLiveStreamEntryOrderBy || (exports.KalturaLiveStreamEntryOrderBy = {}));
