// kaltura-node-typescript-client/api/types/KalturaUploadResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUploadResponse = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUploadErrorCode_1 = require("./KalturaUploadErrorCode");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaUploadResponse = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUploadResponse, _super);
    function KalturaUploadResponse(data) {
        return _super.call(this, data) || this;
    }
    KalturaUploadResponse.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUploadResponse' },
            uploadTokenId: { type: 's' },
            fileSize: { type: 'n' },
            errorCode: { type: 'en', subTypeConstructor: KalturaUploadErrorCode_1.KalturaUploadErrorCode, subType: 'KalturaUploadErrorCode' },
            errorDescription: { type: 's' }
        });
        return result;
    };
    return KalturaUploadResponse;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaUploadResponse = KalturaUploadResponse;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUploadResponse', KalturaUploadResponse);
