// kaltura-node-typescript-client/api/types/KalturaStorageExportJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaStorageExportJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaStorageJobData_1 = require("./KalturaStorageJobData");
var KalturaStorageExportJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaStorageExportJobData, _super);
    function KalturaStorageExportJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaStorageExportJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaStorageExportJobData' },
            force: { type: 'b' },
            createLink: { type: 'b' },
            assetId: { type: 's' },
            externalUrl: { type: 's' },
            port: { type: 'n' }
        });
        return result;
    };
    return KalturaStorageExportJobData;
}(KalturaStorageJobData_1.KalturaStorageJobData));
exports.KalturaStorageExportJobData = KalturaStorageExportJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaStorageExportJobData', KalturaStorageExportJobData);
