// kaltura-node-typescript-client/api/types/KalturaDistributionJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDistributionJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDistributionProfile_1 = require("./KalturaDistributionProfile");
var KalturaEntryDistribution_1 = require("./KalturaEntryDistribution");
var KalturaDistributionProviderType_1 = require("./KalturaDistributionProviderType");
var KalturaDistributionJobProviderData_1 = require("./KalturaDistributionJobProviderData");
var KalturaDistributionRemoteMediaFile_1 = require("./KalturaDistributionRemoteMediaFile");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaDistributionJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDistributionJobData, _super);
    function KalturaDistributionJobData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.mediaFiles === 'undefined')
            _this.mediaFiles = [];
        return _this;
    }
    KalturaDistributionJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDistributionJobData' },
            distributionProfileId: { type: 'n' },
            distributionProfile: { type: 'o', subTypeConstructor: KalturaDistributionProfile_1.KalturaDistributionProfile, subType: 'KalturaDistributionProfile' },
            entryDistributionId: { type: 'n' },
            entryDistribution: { type: 'o', subTypeConstructor: KalturaEntryDistribution_1.KalturaEntryDistribution, subType: 'KalturaEntryDistribution' },
            remoteId: { type: 's' },
            providerType: { type: 'es', subTypeConstructor: KalturaDistributionProviderType_1.KalturaDistributionProviderType, subType: 'KalturaDistributionProviderType' },
            providerData: { type: 'o', subTypeConstructor: KalturaDistributionJobProviderData_1.KalturaDistributionJobProviderData, subType: 'KalturaDistributionJobProviderData' },
            results: { type: 's' },
            sentData: { type: 's' },
            mediaFiles: { type: 'a', subTypeConstructor: KalturaDistributionRemoteMediaFile_1.KalturaDistributionRemoteMediaFile, subType: 'KalturaDistributionRemoteMediaFile' }
        });
        return result;
    };
    return KalturaDistributionJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaDistributionJobData = KalturaDistributionJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDistributionJobData', KalturaDistributionJobData);
