// kaltura-node-typescript-client/api/types/KalturaTvinciDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaTvinciDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaTvinciDistributionTag_1 = require("./KalturaTvinciDistributionTag");
var KalturaTvinciAssetsType_1 = require("./KalturaTvinciAssetsType");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaTvinciDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaTvinciDistributionProfile, _super);
    function KalturaTvinciDistributionProfile(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.tags === 'undefined')
            _this.tags = [];
        return _this;
    }
    KalturaTvinciDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaTvinciDistributionProfile' },
            ingestUrl: { type: 's' },
            username: { type: 's' },
            password: { type: 's' },
            tags: { type: 'a', subTypeConstructor: KalturaTvinciDistributionTag_1.KalturaTvinciDistributionTag, subType: 'KalturaTvinciDistributionTag' },
            xsltFile: { type: 's' },
            innerType: { type: 's' },
            assetsType: { type: 'en', subTypeConstructor: KalturaTvinciAssetsType_1.KalturaTvinciAssetsType, subType: 'KalturaTvinciAssetsType' }
        });
        return result;
    };
    return KalturaTvinciDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaTvinciDistributionProfile = KalturaTvinciDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaTvinciDistributionProfile', KalturaTvinciDistributionProfile);
