// kaltura-node-typescript-client/api/types/KalturaFlavorParamsOutput.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFlavorParamsOutput = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaFlavorParams_1 = require("./KalturaFlavorParams");
var KalturaFlavorParamsOutput = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaFlavorParamsOutput, _super);
    function KalturaFlavorParamsOutput(data) {
        return _super.call(this, data) || this;
    }
    KalturaFlavorParamsOutput.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaFlavorParamsOutput' },
            flavorParamsId: { type: 'n' },
            commandLinesStr: { type: 's' },
            flavorParamsVersion: { type: 's' },
            flavorAssetId: { type: 's' },
            flavorAssetVersion: { type: 's' },
            readyBehavior: { type: 'n' }
        });
        return result;
    };
    return KalturaFlavorParamsOutput;
}(KalturaFlavorParams_1.KalturaFlavorParams));
exports.KalturaFlavorParamsOutput = KalturaFlavorParamsOutput;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaFlavorParamsOutput', KalturaFlavorParamsOutput);
