// kaltura-node-typescript-client/api/types/KalturaMetadataObjectType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMetadataObjectType = void 0;
var KalturaMetadataObjectType;
(function (KalturaMetadataObjectType) {
    KalturaMetadataObjectType["adCuePoint"] = "adCuePointMetadata.AdCuePoint";
    KalturaMetadataObjectType["annotation"] = "annotationMetadata.Annotation";
    KalturaMetadataObjectType["answerCuePoint"] = "quiz.AnswerCuePoint";
    KalturaMetadataObjectType["category"] = "2";
    KalturaMetadataObjectType["codeCuePoint"] = "codeCuePointMetadata.CodeCuePoint";
    KalturaMetadataObjectType["dynamicObject"] = "5";
    KalturaMetadataObjectType["entry"] = "1";
    KalturaMetadataObjectType["partner"] = "4";
    KalturaMetadataObjectType["questionCuePoint"] = "quiz.QuestionCuePoint";
    KalturaMetadataObjectType["thumbCuePoint"] = "thumbCuePointMetadata.thumbCuePoint";
    KalturaMetadataObjectType["user"] = "3";
    KalturaMetadataObjectType["userEntry"] = "6";
})(KalturaMetadataObjectType = exports.KalturaMetadataObjectType || (exports.KalturaMetadataObjectType = {}));
