// kaltura-node-typescript-client/api/types/KalturaBaseSyndicationFeed.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBaseSyndicationFeed = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaSyndicationFeedStatus_1 = require("./KalturaSyndicationFeedStatus");
var KalturaSyndicationFeedType_1 = require("./KalturaSyndicationFeedType");
var KalturaSyndicationFeedEntriesOrderBy_1 = require("./KalturaSyndicationFeedEntriesOrderBy");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaBaseSyndicationFeed = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBaseSyndicationFeed, _super);
    function KalturaBaseSyndicationFeed(data) {
        return _super.call(this, data) || this;
    }
    KalturaBaseSyndicationFeed.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBaseSyndicationFeed' },
            id: { type: 's', readOnly: true },
            feedUrl: { type: 's', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            playlistId: { type: 's' },
            name: { type: 's' },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaSyndicationFeedStatus_1.KalturaSyndicationFeedStatus, subType: 'KalturaSyndicationFeedStatus' },
            type: { type: 'en', subTypeConstructor: KalturaSyndicationFeedType_1.KalturaSyndicationFeedType, subType: 'KalturaSyndicationFeedType' },
            landingPage: { type: 's' },
            createdAt: { type: 'd', readOnly: true },
            allowEmbed: { type: 'b' },
            playerUiconfId: { type: 'n' },
            flavorParamId: { type: 'n' },
            transcodeExistingContent: { type: 'b' },
            addToDefaultConversionProfile: { type: 'b' },
            categories: { type: 's' },
            storageId: { type: 'n' },
            entriesOrderBy: { type: 'es', subTypeConstructor: KalturaSyndicationFeedEntriesOrderBy_1.KalturaSyndicationFeedEntriesOrderBy, subType: 'KalturaSyndicationFeedEntriesOrderBy' },
            enforceEntitlement: { type: 'b' },
            privacyContext: { type: 's' },
            updatedAt: { type: 'd', readOnly: true },
            useCategoryEntries: { type: 'b' },
            feedContentTypeHeader: { type: 's' }
        });
        return result;
    };
    return KalturaBaseSyndicationFeed;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaBaseSyndicationFeed = KalturaBaseSyndicationFeed;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBaseSyndicationFeed', KalturaBaseSyndicationFeed);
