// kaltura-node-typescript-client/api/types/KalturaImageFlavorParamsOutput.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaImageFlavorParamsOutput = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaFlavorParamsOutput_1 = require("./KalturaFlavorParamsOutput");
var KalturaImageFlavorParamsOutput = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaImageFlavorParamsOutput, _super);
    function KalturaImageFlavorParamsOutput(data) {
        return _super.call(this, data) || this;
    }
    KalturaImageFlavorParamsOutput.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaImageFlavorParamsOutput' },
            densityWidth: { type: 'n' },
            densityHeight: { type: 'n' },
            sizeWidth: { type: 'n' },
            sizeHeight: { type: 'n' },
            depth: { type: 'n' }
        });
        return result;
    };
    return KalturaImageFlavorParamsOutput;
}(KalturaFlavorParamsOutput_1.KalturaFlavorParamsOutput));
exports.KalturaImageFlavorParamsOutput = KalturaImageFlavorParamsOutput;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaImageFlavorParamsOutput', KalturaImageFlavorParamsOutput);
