// kaltura-node-typescript-client/api/types/KalturaPlaylistCompareAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPlaylistCompareAttribute = void 0;
var KalturaPlaylistCompareAttribute;
(function (KalturaPlaylistCompareAttribute) {
    KalturaPlaylistCompareAttribute["accessControlId"] = "accessControlId";
    KalturaPlaylistCompareAttribute["createdAt"] = "createdAt";
    KalturaPlaylistCompareAttribute["endDate"] = "endDate";
    KalturaPlaylistCompareAttribute["moderationCount"] = "moderationCount";
    KalturaPlaylistCompareAttribute["moderationStatus"] = "moderationStatus";
    KalturaPlaylistCompareAttribute["partnerId"] = "partnerId";
    KalturaPlaylistCompareAttribute["partnerSortValue"] = "partnerSortValue";
    KalturaPlaylistCompareAttribute["rank"] = "rank";
    KalturaPlaylistCompareAttribute["replacementStatus"] = "replacementStatus";
    KalturaPlaylistCompareAttribute["startDate"] = "startDate";
    KalturaPlaylistCompareAttribute["status"] = "status";
    KalturaPlaylistCompareAttribute["totalRank"] = "totalRank";
    KalturaPlaylistCompareAttribute["type"] = "type";
    KalturaPlaylistCompareAttribute["updatedAt"] = "updatedAt";
})(KalturaPlaylistCompareAttribute = exports.KalturaPlaylistCompareAttribute || (exports.KalturaPlaylistCompareAttribute = {}));
