// kaltura-node-typescript-client/api/types/KalturaPlaybackSource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPlaybackSource = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDrmPlaybackPluginData_1 = require("./KalturaDrmPlaybackPluginData");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaPlaybackSource = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPlaybackSource, _super);
    function KalturaPlaybackSource(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.drm === 'undefined')
            _this.drm = [];
        return _this;
    }
    KalturaPlaybackSource.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPlaybackSource' },
            deliveryProfileId: { type: 's' },
            format: { type: 's' },
            protocols: { type: 's' },
            flavorIds: { type: 's' },
            url: { type: 's' },
            drm: { type: 'a', subTypeConstructor: KalturaDrmPlaybackPluginData_1.KalturaDrmPlaybackPluginData, subType: 'KalturaDrmPlaybackPluginData' }
        });
        return result;
    };
    return KalturaPlaybackSource;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaPlaybackSource = KalturaPlaybackSource;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPlaybackSource', KalturaPlaybackSource);
