// kaltura-node-typescript-client/api/types/KalturaCodeCuePoint.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCodeCuePoint = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCuePoint_1 = require("./KalturaCuePoint");
var KalturaCodeCuePoint = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCodeCuePoint, _super);
    function KalturaCodeCuePoint(data) {
        return _super.call(this, data) || this;
    }
    KalturaCodeCuePoint.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCodeCuePoint' },
            code: { type: 's' },
            description: { type: 's' },
            endTime: { type: 'n' },
            duration: { type: 'n', readOnly: true }
        });
        return result;
    };
    return KalturaCodeCuePoint;
}(KalturaCuePoint_1.KalturaCuePoint));
exports.KalturaCodeCuePoint = KalturaCodeCuePoint;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCodeCuePoint', KalturaCodeCuePoint);
