// kaltura-node-typescript-client/api/types/KalturaLiveChannelMatchAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveChannelMatchAttribute = void 0;
var KalturaLiveChannelMatchAttribute;
(function (KalturaLiveChannelMatchAttribute) {
    KalturaLiveChannelMatchAttribute["adminTags"] = "adminTags";
    KalturaLiveChannelMatchAttribute["categoriesIds"] = "categoriesIds";
    KalturaLiveChannelMatchAttribute["creatorId"] = "creatorId";
    KalturaLiveChannelMatchAttribute["description"] = "description";
    KalturaLiveChannelMatchAttribute["durationType"] = "durationType";
    KalturaLiveChannelMatchAttribute["flavorParamsIds"] = "flavorParamsIds";
    KalturaLiveChannelMatchAttribute["groupId"] = "groupId";
    KalturaLiveChannelMatchAttribute["id"] = "id";
    KalturaLiveChannelMatchAttribute["_name"] = "name";
    KalturaLiveChannelMatchAttribute["referenceId"] = "referenceId";
    KalturaLiveChannelMatchAttribute["replacedEntryId"] = "replacedEntryId";
    KalturaLiveChannelMatchAttribute["replacingEntryId"] = "replacingEntryId";
    KalturaLiveChannelMatchAttribute["searchText"] = "searchText";
    KalturaLiveChannelMatchAttribute["tags"] = "tags";
    KalturaLiveChannelMatchAttribute["userId"] = "userId";
})(KalturaLiveChannelMatchAttribute = exports.KalturaLiveChannelMatchAttribute || (exports.KalturaLiveChannelMatchAttribute = {}));
