// kaltura-node-typescript-client/api/types/KalturaActivitiBusinessProcessServer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaActivitiBusinessProcessServer = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaActivitiBusinessProcessServerProtocol_1 = require("./KalturaActivitiBusinessProcessServerProtocol");
var KalturaBusinessProcessServer_1 = require("./KalturaBusinessProcessServer");
var KalturaActivitiBusinessProcessServer = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaActivitiBusinessProcessServer, _super);
    function KalturaActivitiBusinessProcessServer(data) {
        return _super.call(this, data) || this;
    }
    KalturaActivitiBusinessProcessServer.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaActivitiBusinessProcessServer' },
            host: { type: 's' },
            port: { type: 'n' },
            protocol: { type: 'es', subTypeConstructor: KalturaActivitiBusinessProcessServerProtocol_1.KalturaActivitiBusinessProcessServerProtocol, subType: 'KalturaActivitiBusinessProcessServerProtocol' },
            username: { type: 's' },
            password: { type: 's' }
        });
        return result;
    };
    return KalturaActivitiBusinessProcessServer;
}(KalturaBusinessProcessServer_1.KalturaBusinessProcessServer));
exports.KalturaActivitiBusinessProcessServer = KalturaActivitiBusinessProcessServer;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaActivitiBusinessProcessServer', KalturaActivitiBusinessProcessServer);
