// kaltura-node-typescript-client/api/types/KalturaSshDropFolder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSshDropFolder = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaRemoteDropFolder_1 = require("./KalturaRemoteDropFolder");
var KalturaSshDropFolder = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaSshDropFolder, _super);
    function KalturaSshDropFolder(data) {
        return _super.call(this, data) || this;
    }
    KalturaSshDropFolder.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaSshDropFolder' },
            host: { type: 's' },
            port: { type: 'n' },
            username: { type: 's' },
            password: { type: 's' },
            privateKey: { type: 's' },
            publicKey: { type: 's' },
            passPhrase: { type: 's' }
        });
        return result;
    };
    return KalturaSshDropFolder;
}(KalturaRemoteDropFolder_1.KalturaRemoteDropFolder));
exports.KalturaSshDropFolder = KalturaSshDropFolder;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaSshDropFolder', KalturaSshDropFolder);
