// kaltura-node-typescript-client/api/types/KalturaESearchCuePointFieldName.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaESearchCuePointFieldName = void 0;
var KalturaESearchCuePointFieldName;
(function (KalturaESearchCuePointFieldName) {
    KalturaESearchCuePointFieldName["answers"] = "answers";
    KalturaESearchCuePointFieldName["endTime"] = "end_time";
    KalturaESearchCuePointFieldName["explanation"] = "explanation";
    KalturaESearchCuePointFieldName["hint"] = "hint";
    KalturaESearchCuePointFieldName["id"] = "id";
    KalturaESearchCuePointFieldName["_name"] = "name";
    KalturaESearchCuePointFieldName["question"] = "question";
    KalturaESearchCuePointFieldName["startTime"] = "start_time";
    KalturaESearchCuePointFieldName["subType"] = "sub_type";
    KalturaESearchCuePointFieldName["tags"] = "tags";
    KalturaESearchCuePointFieldName["text"] = "text";
    KalturaESearchCuePointFieldName["type"] = "type";
})(KalturaESearchCuePointFieldName = exports.KalturaESearchCuePointFieldName || (exports.KalturaESearchCuePointFieldName = {}));
