// kaltura-node-typescript-client/api/types/KalturaAccessControlBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAccessControlBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaAccessControlBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAccessControlBaseFilter, _super);
    function KalturaAccessControlBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaAccessControlBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAccessControlBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            systemNameEqual: { type: 's' },
            systemNameIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' }
        });
        return result;
    };
    return KalturaAccessControlBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaAccessControlBaseFilter = KalturaAccessControlBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAccessControlBaseFilter', KalturaAccessControlBaseFilter);
