// kaltura-node-typescript-client/api/types/KalturaGenericDistributionProviderBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaGenericDistributionProviderBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaGenericDistributionProviderStatus_1 = require("./KalturaGenericDistributionProviderStatus");
var KalturaDistributionProviderFilter_1 = require("./KalturaDistributionProviderFilter");
var KalturaGenericDistributionProviderBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaGenericDistributionProviderBaseFilter, _super);
    function KalturaGenericDistributionProviderBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaGenericDistributionProviderBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaGenericDistributionProviderBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            isDefaultEqual: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            isDefaultIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaGenericDistributionProviderStatus_1.KalturaGenericDistributionProviderStatus, subType: 'KalturaGenericDistributionProviderStatus' },
            statusIn: { type: 's' }
        });
        return result;
    };
    return KalturaGenericDistributionProviderBaseFilter;
}(KalturaDistributionProviderFilter_1.KalturaDistributionProviderFilter));
exports.KalturaGenericDistributionProviderBaseFilter = KalturaGenericDistributionProviderBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaGenericDistributionProviderBaseFilter', KalturaGenericDistributionProviderBaseFilter);
