// kaltura-node-typescript-client/api/types/KalturaMetadataProfileBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMetadataProfileBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaMetadataObjectType_1 = require("./KalturaMetadataObjectType");
var KalturaMetadataProfileStatus_1 = require("./KalturaMetadataProfileStatus");
var KalturaMetadataProfileCreateMode_1 = require("./KalturaMetadataProfileCreateMode");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaMetadataProfileBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMetadataProfileBaseFilter, _super);
    function KalturaMetadataProfileBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaMetadataProfileBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMetadataProfileBaseFilter' },
            idEqual: { type: 'n' },
            partnerIdEqual: { type: 'n' },
            metadataObjectTypeEqual: { type: 'es', subTypeConstructor: KalturaMetadataObjectType_1.KalturaMetadataObjectType, subType: 'KalturaMetadataObjectType' },
            metadataObjectTypeIn: { type: 's' },
            versionEqual: { type: 'n' },
            nameEqual: { type: 's' },
            systemNameEqual: { type: 's' },
            systemNameIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaMetadataProfileStatus_1.KalturaMetadataProfileStatus, subType: 'KalturaMetadataProfileStatus' },
            statusIn: { type: 's' },
            createModeEqual: { type: 'en', subTypeConstructor: KalturaMetadataProfileCreateMode_1.KalturaMetadataProfileCreateMode, subType: 'KalturaMetadataProfileCreateMode' },
            createModeNotEqual: { type: 'en', subTypeConstructor: KalturaMetadataProfileCreateMode_1.KalturaMetadataProfileCreateMode, subType: 'KalturaMetadataProfileCreateMode' },
            createModeIn: { type: 's' },
            createModeNotIn: { type: 's' }
        });
        return result;
    };
    return KalturaMetadataProfileBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaMetadataProfileBaseFilter = KalturaMetadataProfileBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMetadataProfileBaseFilter', KalturaMetadataProfileBaseFilter);
