// kaltura-node-typescript-client/api/types/KalturaLiveStreamDetails.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveStreamDetails = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaEntryServerNodeStatus_1 = require("./KalturaEntryServerNodeStatus");
var KalturaViewMode_1 = require("./KalturaViewMode");
var KalturaLiveStreamBroadcastStatus_1 = require("./KalturaLiveStreamBroadcastStatus");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaLiveStreamDetails = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveStreamDetails, _super);
    function KalturaLiveStreamDetails(data) {
        return _super.call(this, data) || this;
    }
    KalturaLiveStreamDetails.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveStreamDetails' },
            primaryStreamStatus: { type: 'en', subTypeConstructor: KalturaEntryServerNodeStatus_1.KalturaEntryServerNodeStatus, subType: 'KalturaEntryServerNodeStatus' },
            secondaryStreamStatus: { type: 'en', subTypeConstructor: KalturaEntryServerNodeStatus_1.KalturaEntryServerNodeStatus, subType: 'KalturaEntryServerNodeStatus' },
            viewMode: { type: 'en', subTypeConstructor: KalturaViewMode_1.KalturaViewMode, subType: 'KalturaViewMode' },
            wasBroadcast: { type: 'b' },
            broadcastStatus: { type: 'en', subTypeConstructor: KalturaLiveStreamBroadcastStatus_1.KalturaLiveStreamBroadcastStatus, subType: 'KalturaLiveStreamBroadcastStatus' }
        });
        return result;
    };
    return KalturaLiveStreamDetails;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaLiveStreamDetails = KalturaLiveStreamDetails;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveStreamDetails', KalturaLiveStreamDetails);
