// kaltura-node-typescript-client/api/types/KalturaLanguage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLanguage = void 0;
var KalturaLanguage;
(function (KalturaLanguage) {
    KalturaLanguage["aa"] = "Afar";
    KalturaLanguage["aal"] = "Afade";
    KalturaLanguage["ab"] = "Abkhazian";
    KalturaLanguage["abe"] = "Abnaki Western";
    KalturaLanguage["abq"] = "Abaza";
    KalturaLanguage["abu"] = "Abure";
    KalturaLanguage["ace"] = "Achinese";
    KalturaLanguage["acf"] = "Saint Lucian Creole French";
    KalturaLanguage["acn"] = "Achang";
    KalturaLanguage["act"] = "Achterhooks";
    KalturaLanguage["acv"] = "Achumawi";
    KalturaLanguage["adj"] = "Adioukrou";
    KalturaLanguage["adt"] = "Adynyamathanha";
    KalturaLanguage["ady"] = "Adyghe; Adygei";
    KalturaLanguage["af"] = "Afrikaans";
    KalturaLanguage["agq"] = "Aghem";
    KalturaLanguage["agr"] = "Aguaruna";
    KalturaLanguage["agu"] = "Aguacateco";
    KalturaLanguage["agx"] = "Aghul";
    KalturaLanguage["aii"] = "Assyrian Neo-Aramaic";
    KalturaLanguage["ain"] = "Ainu (Japan)";
    KalturaLanguage["akk"] = "Akkadian";
    KalturaLanguage["akl"] = "Aklanon";
    KalturaLanguage["aku"] = "Akum";
    KalturaLanguage["akz"] = "Alabama";
    KalturaLanguage["alc"] = "Qawasqar";
    KalturaLanguage["ale"] = "Aleut";
    KalturaLanguage["alg"] = "Algonquian languages";
    KalturaLanguage["aln"] = "Albanian (Gheg)";
    KalturaLanguage["alq"] = "Algonquin";
    KalturaLanguage["als"] = "Albanian (Tosk)";
    KalturaLanguage["alt"] = "Altai (Southern)";
    KalturaLanguage["am"] = "Amharic";
    KalturaLanguage["ame"] = "Yanesha";
    KalturaLanguage["amw"] = "Western Neo-Aramaic";
    KalturaLanguage["ang"] = "Old English";
    KalturaLanguage["apj"] = "Apache (Jicarilla)";
    KalturaLanguage["apw"] = "Apache (Western)";
    KalturaLanguage["ar"] = "Arabic";
    KalturaLanguage["arb"] = "Arabic (standard)";
    KalturaLanguage["arc"] = "Aramaic";
    KalturaLanguage["ari"] = "Arikara";
    KalturaLanguage["arn"] = "Araucanian";
    KalturaLanguage["arp"] = "Arapaho";
    KalturaLanguage["asb"] = "Assiniboine";
    KalturaLanguage["ast"] = "Asturian";
    KalturaLanguage["as"] = "Assamese";
    KalturaLanguage["ath"] = "Athapascan languages";
    KalturaLanguage["atj"] = "Atikamekw";
    KalturaLanguage["auc"] = "Waorani";
    KalturaLanguage["avk"] = "Kotava";
    KalturaLanguage["awa"] = "Awadhi";
    KalturaLanguage["ay"] = "Aymara";
    KalturaLanguage["az"] = "Azerbaijani";
    KalturaLanguage["ba"] = "Bashkir";
    KalturaLanguage["bal"] = "Baluchi";
    KalturaLanguage["ban"] = "Balinese";
    KalturaLanguage["bar"] = "Bavarian";
    KalturaLanguage["bas"] = "Basa (Cameroon)";
    KalturaLanguage["bcc"] = "Balochi Southern";
    KalturaLanguage["bcr"] = "Babine";
    KalturaLanguage["bdj"] = "Bai";
    KalturaLanguage["be"] = "Byelorussian (Belarusian)";
    KalturaLanguage["bea"] = "Beaver";
    KalturaLanguage["bej"] = "Beja";
    KalturaLanguage["bem"] = "Bemba (Zambia)";
    KalturaLanguage["bew"] = "Betawi";
    KalturaLanguage["bfq"] = "Badaga";
    KalturaLanguage["bft"] = "Balti";
    KalturaLanguage["bg"] = "Bulgarian";
    KalturaLanguage["bh"] = "Bihari";
    KalturaLanguage["bhb"] = "Bhili";
    KalturaLanguage["bho"] = "Bhojpuri";
    KalturaLanguage["bi"] = "Bislama";
    KalturaLanguage["bik"] = "Bikol";
    KalturaLanguage["bin"] = "Bini";
    KalturaLanguage["bla"] = "Siksika";
    KalturaLanguage["bn"] = "Bengali (Bangla)";
    KalturaLanguage["bo"] = "Tibetan";
    KalturaLanguage["bpy"] = "Bishnupriya Manipuri";
    KalturaLanguage["br"] = "Breton";
    KalturaLanguage["bua"] = "Buriat";
    KalturaLanguage["bug"] = "Buginese";
    KalturaLanguage["bT"] = "Arabic Tunisian Spoken";
    KalturaLanguage["ca"] = "Catalan";
    KalturaLanguage["caa"] = "Chor";
    KalturaLanguage["cad"] = "Caddo";
    KalturaLanguage["caf"] = "Carrier Southern";
    KalturaLanguage["cay"] = "Cayuga";
    KalturaLanguage["cdo"] = "Min Dong Chinese";
    KalturaLanguage["ceb"] = "Cebuano";
    KalturaLanguage["chc"] = "Catawba";
    KalturaLanguage["chg"] = "Chagatai";
    KalturaLanguage["chm"] = "Mari (Russia)";
    KalturaLanguage["chn"] = "Chinook jargon";
    KalturaLanguage["cho"] = "Choctaw";
    KalturaLanguage["chp"] = "Chipewyan";
    KalturaLanguage["chr"] = "Cherokee";
    KalturaLanguage["chy"] = "Cheyenne";
    KalturaLanguage["cic"] = "Chickasaw";
    KalturaLanguage["cim"] = "Cimbrian";
    KalturaLanguage["ciw"] = "Chippewa";
    KalturaLanguage["cjs"] = "Shor";
    KalturaLanguage["cjy"] = "Jinyu Chinese";
    KalturaLanguage["ckt"] = "Chukot";
    KalturaLanguage["clc"] = "Chilcotin";
    KalturaLanguage["cld"] = "Chaldean Neo-Aramaic";
    KalturaLanguage["clm"] = "Clallam Klallam";
    KalturaLanguage["cmn"] = "Mandarin Chinese";
    KalturaLanguage["co"] = "Corsican";
    KalturaLanguage["coc"] = "Cocopa";
    KalturaLanguage["coj"] = "Cochimi";
    KalturaLanguage["com"] = "Comanche";
    KalturaLanguage["coo"] = "Comox";
    KalturaLanguage["cop"] = "Coptic";
    KalturaLanguage["cr"] = "Cree";
    KalturaLanguage["crg"] = "Michif";
    KalturaLanguage["crh"] = "Crimean Tatar";
    KalturaLanguage["crx"] = "Carrier";
    KalturaLanguage["cs"] = "Czech";
    KalturaLanguage["csb"] = "Kashubian";
    KalturaLanguage["cup"] = "Cupeo";
    KalturaLanguage["cy"] = "Welsh";
    KalturaLanguage["czh"] = "Huizhou Chinese";
    KalturaLanguage["da"] = "Danish";
    KalturaLanguage["dak"] = "Dakota";
    KalturaLanguage["dar"] = "Dargwa";
    KalturaLanguage["ddo"] = "Tsez";
    KalturaLanguage["de"] = "German";
    KalturaLanguage["del"] = "Delaware";
    KalturaLanguage["dgr"] = "Dogrib";
    KalturaLanguage["dhv"] = "Dehu";
    KalturaLanguage["dih"] = "Kumiai";
    KalturaLanguage["din"] = "Dinka";
    KalturaLanguage["dje"] = "Zarma";
    KalturaLanguage["dlg"] = "Dolgan";
    KalturaLanguage["dng"] = "Dungan";
    KalturaLanguage["doh"] = "Dong";
    KalturaLanguage["doi"] = "Dogri (generic)";
    KalturaLanguage["dsb"] = "Sorbian Lower";
    KalturaLanguage["dua"] = "Duala";
    KalturaLanguage["dyu"] = "Dyula";
    KalturaLanguage["dz"] = "Bhutani";
    KalturaLanguage["eee"] = "E";
    KalturaLanguage["egl"] = "Emilian";
    KalturaLanguage["el"] = "Greek";
    KalturaLanguage["en"] = "English";
    KalturaLanguage["enm"] = "English Middle (1100-1500)";
    KalturaLanguage["enGb"] = "English (British)";
    KalturaLanguage["enUs"] = "English (American)";
    KalturaLanguage["eo"] = "Esperanto";
    KalturaLanguage["es"] = "Spanish";
    KalturaLanguage["esXl"] = "Spanish (Latin America)";
    KalturaLanguage["et"] = "Estonian";
    KalturaLanguage["eu"] = "Basque";
    KalturaLanguage["eve"] = "Even";
    KalturaLanguage["evn"] = "Evenki";
    KalturaLanguage["fa"] = "Farsi";
    KalturaLanguage["fan"] = "Fang (Equatorial Guinea)";
    KalturaLanguage["fax"] = "Fala";
    KalturaLanguage["fi"] = "Finnish";
    KalturaLanguage["fil"] = "Filipino";
    KalturaLanguage["fit"] = "Finnish (Tornedalen)";
    KalturaLanguage["fj"] = "Fiji";
    KalturaLanguage["fo"] = "Faeroese";
    KalturaLanguage["fon"] = "Fon";
    KalturaLanguage["fr"] = "French";
    KalturaLanguage["frk"] = "Frankish";
    KalturaLanguage["frp"] = "Franco-Prove";
    KalturaLanguage["frr"] = "Frisian Northern";
    KalturaLanguage["frCa"] = "French (Canada)";
    KalturaLanguage["fuc"] = "Pulaar";
    KalturaLanguage["fur"] = "Friulian";
    KalturaLanguage["fvr"] = "Fur";
    KalturaLanguage["fy"] = "Frisian";
    KalturaLanguage["ga"] = "Irish";
    KalturaLanguage["gaa"] = "Ga";
    KalturaLanguage["gag"] = "Gagauz";
    KalturaLanguage["gan"] = "Gan";
    KalturaLanguage["gbz"] = "Dari Zoroastrian";
    KalturaLanguage["gd"] = "Gaelic (Scottish)";
    KalturaLanguage["gdm"] = "Laal";
    KalturaLanguage["geh"] = "German Hutterite";
    KalturaLanguage["gez"] = "Geez";
    KalturaLanguage["gil"] = "Gilbertese";
    KalturaLanguage["git"] = "Gitxsan";
    KalturaLanguage["gl"] = "Galician";
    KalturaLanguage["gld"] = "Nanai";
    KalturaLanguage["gn"] = "Guarani";
    KalturaLanguage["gom"] = "Konkani Goan";
    KalturaLanguage["grc"] = "Greek Ancient (to 1453)";
    KalturaLanguage["gsw"] = "Swiss German Alemannic Alsatian";
    KalturaLanguage["gu"] = "Gujarati";
    KalturaLanguage["gv"] = "Gaelic (Manx)";
    KalturaLanguage["gwi"] = "Gwichin";
    KalturaLanguage["ha"] = "Hausa";
    KalturaLanguage["haa"] = "Han";
    KalturaLanguage["hai"] = "Haida";
    KalturaLanguage["hak"] = "Hakka";
    KalturaLanguage["has"] = "Haisla";
    KalturaLanguage["haw"] = "Hawaiian";
    KalturaLanguage["he"] = "Hebrew";
    KalturaLanguage["hei"] = "Heiltsuk";
    KalturaLanguage["hi"] = "Hindi";
    KalturaLanguage["hid"] = "Hidatsa";
    KalturaLanguage["hil"] = "Hiligaynon";
    KalturaLanguage["hkk"] = "Hokkien";
    KalturaLanguage["hmn"] = "Hmong";
    KalturaLanguage["hni"] = "Hani";
    KalturaLanguage["hnn"] = "Hainanese";
    KalturaLanguage["hop"] = "Hopi";
    KalturaLanguage["hr"] = "Croatian";
    KalturaLanguage["hsb"] = "Sorbian Upper";
    KalturaLanguage["hu"] = "Hungarian";
    KalturaLanguage["hur"] = "Halkomelem";
    KalturaLanguage["hy"] = "Armenian";
    KalturaLanguage["ia"] = "Interlingua";
    KalturaLanguage["id"] = "Indonesian";
    KalturaLanguage["ie"] = "Interlingue";
    KalturaLanguage["ik"] = "Inupiak";
    KalturaLanguage["ikt"] = "Inuinnaqtun";
    KalturaLanguage["ilo"] = "Iloko";
    KalturaLanguage["in"] = "Indonesian";
    KalturaLanguage["inh"] = "Ingush";
    KalturaLanguage["iro"] = "Iroquoian languages";
    KalturaLanguage["is"] = "Icelandic";
    KalturaLanguage["it"] = "Italian";
    KalturaLanguage["itl"] = "Itelmen";
    KalturaLanguage["iu"] = "Inuktitut";
    KalturaLanguage["iw"] = "Hebrew";
    KalturaLanguage["izh"] = "Ingrian";
    KalturaLanguage["ja"] = "Japanese";
    KalturaLanguage["jbo"] = "Lojban";
    KalturaLanguage["jct"] = "Judeo-Crimean Tatar";
    KalturaLanguage["jge"] = "Judeo-Georgian";
    KalturaLanguage["ji"] = "Yiddish";
    KalturaLanguage["jut"] = "Jutish";
    KalturaLanguage["jv"] = "Javanese";
    KalturaLanguage["ka"] = "Georgian";
    KalturaLanguage["kaa"] = "Karakalpak";
    KalturaLanguage["kab"] = "Kabyle";
    KalturaLanguage["kaj"] = "Jju";
    KalturaLanguage["kap"] = "Bezhta";
    KalturaLanguage["kaw"] = "Kawi";
    KalturaLanguage["kbd"] = "Kabardian";
    KalturaLanguage["kca"] = "Khanty";
    KalturaLanguage["kdr"] = "Karaim";
    KalturaLanguage["kea"] = "Kabuverdianu";
    KalturaLanguage["kfa"] = "Kodava";
    KalturaLanguage["kfr"] = "Kachchi";
    KalturaLanguage["kha"] = "Khasi";
    KalturaLanguage["kic"] = "Kickapoo";
    KalturaLanguage["kim"] = "Karagas";
    KalturaLanguage["kio"] = "Kiowa";
    KalturaLanguage["kjh"] = "Khakas";
    KalturaLanguage["kjv"] = "Kaikavian literary language (Kajkavian)";
    KalturaLanguage["kk"] = "Kazakh";
    KalturaLanguage["kkz"] = "Kaska";
    KalturaLanguage["kl"] = "Greenlandic";
    KalturaLanguage["klj"] = "Khalaj Turkic";
    KalturaLanguage["km"] = "Cambodian";
    KalturaLanguage["kn"] = "Kannada";
    KalturaLanguage["knn"] = "Konkani (specific)";
    KalturaLanguage["ko"] = "Korean";
    KalturaLanguage["koi"] = "Komi-Permyak";
    KalturaLanguage["kok"] = "Konkani (generic)";
    KalturaLanguage["kos"] = "Kosraean";
    KalturaLanguage["kpe"] = "Kpelle";
    KalturaLanguage["kpo"] = "Ikposo";
    KalturaLanguage["kpy"] = "Koryak";
    KalturaLanguage["krc"] = "Karachay-Balkar";
    KalturaLanguage["krl"] = "Karelian";
    KalturaLanguage["ks"] = "Kashmiri";
    KalturaLanguage["ksh"] = "Colognian";
    KalturaLanguage["ksk"] = "Kansa";
    KalturaLanguage["ku"] = "Kurdish";
    KalturaLanguage["kum"] = "Kumyk";
    KalturaLanguage["kut"] = "Kutenai";
    KalturaLanguage["kwk"] = "Kwakiutl";
    KalturaLanguage["kxm"] = "Khmer Northern";
    KalturaLanguage["ky"] = "Kirghiz";
    KalturaLanguage["la"] = "Latin";
    KalturaLanguage["lad"] = "Ladino";
    KalturaLanguage["lah"] = "Lahnda";
    KalturaLanguage["lb"] = "Luxembourgish (Letzeburgesch)";
    KalturaLanguage["lbe"] = "Lak";
    KalturaLanguage["lez"] = "Lezghian";
    KalturaLanguage["lhu"] = "Lahu";
    KalturaLanguage["li"] = "Limburgish ( Limburger)";
    KalturaLanguage["lif"] = "Limbu";
    KalturaLanguage["lij"] = "Ligurian";
    KalturaLanguage["lil"] = "Lillooet";
    KalturaLanguage["lki"] = "Laki";
    KalturaLanguage["lkt"] = "Lakota";
    KalturaLanguage["lld"] = "Ladin";
    KalturaLanguage["lmo"] = "Lombard";
    KalturaLanguage["ln"] = "Lingala";
    KalturaLanguage["lo"] = "Laothian";
    KalturaLanguage["lom"] = "Loma (Liberia)";
    KalturaLanguage["loz"] = "Lozi";
    KalturaLanguage["lt"] = "Lithuanian";
    KalturaLanguage["lua"] = "Luba-Lulua";
    KalturaLanguage["lud"] = "Ludian";
    KalturaLanguage["lun"] = "Lunda";
    KalturaLanguage["luo"] = "Luo (Kenya and Tanzania)";
    KalturaLanguage["luq"] = "Lucumi";
    KalturaLanguage["lut"] = "Lushootseed";
    KalturaLanguage["lv"] = "Latvian (Lettish)";
    KalturaLanguage["lzz"] = "Laz";
    KalturaLanguage["mad"] = "Madurese";
    KalturaLanguage["mag"] = "Magahi";
    KalturaLanguage["mai"] = "Maithili";
    KalturaLanguage["mas"] = "Masai";
    KalturaLanguage["mdf"] = "Moksha";
    KalturaLanguage["men"] = "Mende (Sierra Leone)";
    KalturaLanguage["mez"] = "Menominee";
    KalturaLanguage["mfe"] = "Morisyen";
    KalturaLanguage["mfy"] = "Mayo";
    KalturaLanguage["mg"] = "Malagasy";
    KalturaLanguage["mhq"] = "Mandan";
    KalturaLanguage["mi"] = "Maori";
    KalturaLanguage["mic"] = "Micmac";
    KalturaLanguage["mid"] = "Mandaic";
    KalturaLanguage["min"] = "Minangkabau";
    KalturaLanguage["mk"] = "Macedonian";
    KalturaLanguage["ml"] = "Malayalam";
    KalturaLanguage["mn"] = "Mongolian";
    KalturaLanguage["mnc"] = "Manchu";
    KalturaLanguage["mni"] = "Meitei";
    KalturaLanguage["mnp"] = "Min Bei Chinese";
    KalturaLanguage["mns"] = "Mansi";
    KalturaLanguage["mnw"] = "Mon";
    KalturaLanguage["mo"] = "Moldavian";
    KalturaLanguage["moe"] = "Montagnais";
    KalturaLanguage["moh"] = "Mohawk";
    KalturaLanguage["mos"] = "Mossi";
    KalturaLanguage["mr"] = "Marathi";
    KalturaLanguage["mrw"] = "Maranao";
    KalturaLanguage["ms"] = "Malay";
    KalturaLanguage["mt"] = "Maltese";
    KalturaLanguage["mtq"] = "Muong";
    KalturaLanguage["mu"] = "Multilingual";
    KalturaLanguage["mus"] = "Creek";
    KalturaLanguage["mwl"] = "Mirandese";
    KalturaLanguage["mwr"] = "Marwari";
    KalturaLanguage["mxi"] = "Mozarabic";
    KalturaLanguage["my"] = "Burmese";
    KalturaLanguage["myp"] = "Pirah";
    KalturaLanguage["myv"] = "Erzya";
    KalturaLanguage["mzb"] = "Tumzabt";
    KalturaLanguage["na"] = "Nauru";
    KalturaLanguage["nap"] = "Neapolitan";
    KalturaLanguage["naq"] = "Nama (Namibia)";
    KalturaLanguage["ncg"] = "Nisgaa";
    KalturaLanguage["nds"] = "Low German Low Saxon";
    KalturaLanguage["ne"] = "Nepali";
    KalturaLanguage["new"] = "Newari Nepal Bhasa";
    KalturaLanguage["nio"] = "Nganasan";
    KalturaLanguage["niu"] = "Niuean";
    KalturaLanguage["niv"] = "Gilyak Nivkh";
    KalturaLanguage["nl"] = "Dutch";
    KalturaLanguage["no"] = "Norwegian";
    KalturaLanguage["nod"] = "Thai (Northern)";
    KalturaLanguage["nog"] = "Nogai";
    KalturaLanguage["non"] = "Norse Old";
    KalturaLanguage["nov"] = "Novial";
    KalturaLanguage["nsk"] = "Naskapi";
    KalturaLanguage["nso"] = "Northern Sotho Pedi Sepedi";
    KalturaLanguage["nym"] = "Nyamwezi";
    KalturaLanguage["nyo"] = "Nyoro";
    KalturaLanguage["nys"] = "Nyungah";
    KalturaLanguage["oc"] = "Occitan";
    KalturaLanguage["oj"] = "Ojibwe, Ojibwa";
    KalturaLanguage["ojb"] = "Ojibwa Northwestern";
    KalturaLanguage["ojc"] = "Ojibwa Central";
    KalturaLanguage["ojg"] = "Ojibwa Eastern";
    KalturaLanguage["ojs"] = "Ojibwa Severn";
    KalturaLanguage["ojw"] = "Ojibwa Western";
    KalturaLanguage["oka"] = "Okanagan";
    KalturaLanguage["om"] = "Oromo (Afan, Galla)";
    KalturaLanguage["one"] = "Oneida";
    KalturaLanguage["ono"] = "Onondaga";
    KalturaLanguage["ood"] = "Tohono Oodham";
    KalturaLanguage["or"] = "Oriya";
    KalturaLanguage["ota"] = "Turkish Ottoman";
    KalturaLanguage["otw"] = "Ottawa";
    KalturaLanguage["pa"] = "Punjabi";
    KalturaLanguage["pag"] = "Pangasinan";
    KalturaLanguage["pam"] = "Pampanga";
    KalturaLanguage["pap"] = "Papiamento";
    KalturaLanguage["pau"] = "Palauan";
    KalturaLanguage["pcd"] = "Picard";
    KalturaLanguage["pdc"] = "German Pennsylvania";
    KalturaLanguage["pdt"] = "Plautdietsch";
    KalturaLanguage["pes"] = "Western Farsi";
    KalturaLanguage["pfl"] = "Pfaelzisch";
    KalturaLanguage["pih"] = "Pitcairn-Norfolk";
    KalturaLanguage["pl"] = "Polish";
    KalturaLanguage["pms"] = "Piedmontese";
    KalturaLanguage["pnt"] = "Pontic";
    KalturaLanguage["pot"] = "Potawatomi";
    KalturaLanguage["ppi"] = "Paipai";
    KalturaLanguage["pqm"] = "Malecite-Passamaquoddy";
    KalturaLanguage["prd"] = "Dari (Persian)";
    KalturaLanguage["prg"] = "Prussian";
    KalturaLanguage["prp"] = "Persian";
    KalturaLanguage["prs"] = "Persian (Dari)";
    KalturaLanguage["ps"] = "Pashto (Pushto)";
    KalturaLanguage["pt"] = "Portuguese";
    KalturaLanguage["ptBr"] = "Portuguese (Brazil)";
    KalturaLanguage["qtz"] = "Reserved for local use.";
    KalturaLanguage["qu"] = "Quechua";
    KalturaLanguage["quc"] = "Quich Central";
    KalturaLanguage["qxq"] = "Qashqai";
    KalturaLanguage["rap"] = "Rapanui";
    KalturaLanguage["rar"] = "Rarotongan";
    KalturaLanguage["rcf"] = "Runion Creole French";
    KalturaLanguage["rgn"] = "Romagnol";
    KalturaLanguage["rm"] = "Rhaeto-Romance";
    KalturaLanguage["rme"] = "Angloromani";
    KalturaLanguage["rmf"] = "Romani Kalo Finnish";
    KalturaLanguage["rmo"] = "Romani Sinte";
    KalturaLanguage["rn"] = "Kirundi (Rundi)";
    KalturaLanguage["ro"] = "Romanian";
    KalturaLanguage["rom"] = "Romany";
    KalturaLanguage["ru"] = "Russian";
    KalturaLanguage["rue"] = "Rusyn";
    KalturaLanguage["ruo"] = "Romanian Istro";
    KalturaLanguage["rup"] = "Aromanian";
    KalturaLanguage["ruq"] = "Romanian Megleno";
    KalturaLanguage["rw"] = "Kinyarwanda (Ruanda)";
    KalturaLanguage["ryu"] = "Okinawan Central";
    KalturaLanguage["sa"] = "Sanskrit";
    KalturaLanguage["sah"] = "Sakha";
    KalturaLanguage["sal"] = "Salishan languages";
    KalturaLanguage["sam"] = "Aramaic Samaritan";
    KalturaLanguage["sat"] = "Santali";
    KalturaLanguage["scn"] = "Sicilian";
    KalturaLanguage["sco"] = "Scots";
    KalturaLanguage["scs"] = "Slavey North";
    KalturaLanguage["sd"] = "Sindhi";
    KalturaLanguage["sdc"] = "Sardinian Sassarese";
    KalturaLanguage["sec"] = "Sechelt";
    KalturaLanguage["see"] = "Seneca";
    KalturaLanguage["sei"] = "Seri";
    KalturaLanguage["sek"] = "Sekani";
    KalturaLanguage["sel"] = "Selkup";
    KalturaLanguage["sg"] = "Sangro";
    KalturaLanguage["sh"] = "Serbo-Croatian";
    KalturaLanguage["shh"] = "Shoshoni";
    KalturaLanguage["shi"] = "Tachelhit";
    KalturaLanguage["shs"] = "Shuswap";
    KalturaLanguage["shy"] = "Tachawit";
    KalturaLanguage["si"] = "Sinhalese";
    KalturaLanguage["sid"] = "Sidamo";
    KalturaLanguage["sjd"] = "Sami Kildin";
    KalturaLanguage["sjw"] = "Shawnee";
    KalturaLanguage["sk"] = "Slovak";
    KalturaLanguage["sl"] = "Slovenian";
    KalturaLanguage["slr"] = "Salar";
    KalturaLanguage["sm"] = "Samoan";
    KalturaLanguage["sma"] = "Southern Sami";
    KalturaLanguage["smi"] = "Sami languages";
    KalturaLanguage["smj"] = "Lule Sami";
    KalturaLanguage["smn"] = "Inari Sami";
    KalturaLanguage["sms"] = "Skolt Sami";
    KalturaLanguage["sn"] = "Shona";
    KalturaLanguage["snk"] = "Soninke";
    KalturaLanguage["so"] = "Somali";
    KalturaLanguage["sq"] = "Albanian";
    KalturaLanguage["sr"] = "Serbian";
    KalturaLanguage["srm"] = "Saramaccan";
    KalturaLanguage["srn"] = "Sranan";
    KalturaLanguage["ss"] = "Siswati";
    KalturaLanguage["st"] = "Sesotho";
    KalturaLanguage["sto"] = "Stoney";
    KalturaLanguage["stq"] = "Saterland Frisian";
    KalturaLanguage["str"] = "Salish Straits";
    KalturaLanguage["su"] = "Sundanese";
    KalturaLanguage["sux"] = "Sumerian";
    KalturaLanguage["sv"] = "Swedish";
    KalturaLanguage["sva"] = "Svan";
    KalturaLanguage["sw"] = "Swahili (Kiswahili)";
    KalturaLanguage["swb"] = "Comorian";
    KalturaLanguage["swg"] = "Swabian";
    KalturaLanguage["swl"] = "Swedish Sign Language";
    KalturaLanguage["sxu"] = "Saxon Upper";
    KalturaLanguage["syr"] = "Syriac";
    KalturaLanguage["szl"] = "Silesian";
    KalturaLanguage["ta"] = "Tamil";
    KalturaLanguage["tab"] = "Tabassaran";
    KalturaLanguage["taq"] = "Tamasheq";
    KalturaLanguage["tar"] = "Tarahumara Central";
    KalturaLanguage["tbw"] = "Tagbanwa";
    KalturaLanguage["tce"] = "Southern Tutchone";
    KalturaLanguage["tcx"] = "Toda";
    KalturaLanguage["tcy"] = "Tulu";
    KalturaLanguage["tdd"] = "Tai Na";
    KalturaLanguage["te"] = "Telugu";
    KalturaLanguage["teo"] = "Teo Chew";
    KalturaLanguage["tet"] = "Tetum";
    KalturaLanguage["tg"] = "Tajik";
    KalturaLanguage["tgx"] = "Tagish";
    KalturaLanguage["th"] = "Thai";
    KalturaLanguage["thp"] = "Thompson";
    KalturaLanguage["tht"] = "Tahltan";
    KalturaLanguage["ti"] = "Tigrinya";
    KalturaLanguage["tig"] = "Tigre";
    KalturaLanguage["tk"] = "Turkmen";
    KalturaLanguage["tl"] = "Tagalog";
    KalturaLanguage["tlh"] = "Klingon tlhIngan-Hol";
    KalturaLanguage["tli"] = "Tlingit";
    KalturaLanguage["tly"] = "Talysh";
    KalturaLanguage["tn"] = "Setswana";
    KalturaLanguage["to"] = "Tonga";
    KalturaLanguage["tog"] = "Tonga (Nyasa)";
    KalturaLanguage["tpi"] = "Tok Pisin";
    KalturaLanguage["tpn"] = "Tupinamb";
    KalturaLanguage["tr"] = "Turkish";
    KalturaLanguage["trv"] = "Seediq";
    KalturaLanguage["ts"] = "Tsonga";
    KalturaLanguage["tsi"] = "Tsimshian";
    KalturaLanguage["tt"] = "Tatar";
    KalturaLanguage["ttq"] = "Tamajaq Tawallammat";
    KalturaLanguage["tts"] = "Thai Northeastern";
    KalturaLanguage["ttt"] = "Tat Muslim";
    KalturaLanguage["tum"] = "Tumbuka";
    KalturaLanguage["tus"] = "Tuscarora";
    KalturaLanguage["tuv"] = "Turkana";
    KalturaLanguage["tvl"] = "Tuvalu";
    KalturaLanguage["tw"] = "Twi";
    KalturaLanguage["tyv"] = "Tuvinian";
    KalturaLanguage["tzm"] = "Tamazight Central Atlas";
    KalturaLanguage["uby"] = "Ubykh";
    KalturaLanguage["udi"] = "Udi";
    KalturaLanguage["udm"] = "Udmurt";
    KalturaLanguage["ug"] = "Uighur";
    KalturaLanguage["uk"] = "Ukrainian";
    KalturaLanguage["un"] = "Undefined";
    KalturaLanguage["ur"] = "Urdu";
    KalturaLanguage["uum"] = "Urum";
    KalturaLanguage["uz"] = "Uzbek";
    KalturaLanguage["vec"] = "Venetian";
    KalturaLanguage["vep"] = "Veps";
    KalturaLanguage["vi"] = "Vietnamese";
    KalturaLanguage["vo"] = "Volapuk";
    KalturaLanguage["vor"] = "Voro";
    KalturaLanguage["vot"] = "Votic";
    KalturaLanguage["vro"] = "Vro";
    KalturaLanguage["war"] = "Waray (Philippines)";
    KalturaLanguage["wiy"] = "Wiyot";
    KalturaLanguage["wo"] = "Wolof";
    KalturaLanguage["wuu"] = "Wu Chinese";
    KalturaLanguage["wym"] = "Wymysorys";
    KalturaLanguage["xal"] = "Kalmyk Oirat";
    KalturaLanguage["xh"] = "Xhosa";
    KalturaLanguage["xmf"] = "Mingrelian";
    KalturaLanguage["xsl"] = "Slavey South";
    KalturaLanguage["xsv"] = "Sudovian";
    KalturaLanguage["yi"] = "Yiddish";
    KalturaLanguage["yo"] = "Yoruba";
    KalturaLanguage["yue"] = "Cantonese";
    KalturaLanguage["zai"] = "Zapotec Isthmus";
    KalturaLanguage["zh"] = "Chinese";
    KalturaLanguage["zhTw"] = "Taiwanese Mandarin";
    KalturaLanguage["zu"] = "Zulu";
    KalturaLanguage["zxx"] = "No linguistic content";
})(KalturaLanguage = exports.KalturaLanguage || (exports.KalturaLanguage = {}));
