// kaltura-node-typescript-client/api/types/KalturaBaseEntryOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBaseEntryOrderBy = void 0;
var KalturaBaseEntryOrderBy;
(function (KalturaBaseEntryOrderBy) {
    KalturaBaseEntryOrderBy["createdAtAsc"] = "+createdAt";
    KalturaBaseEntryOrderBy["createdAtDesc"] = "-createdAt";
    KalturaBaseEntryOrderBy["endDateAsc"] = "+endDate";
    KalturaBaseEntryOrderBy["endDateDesc"] = "-endDate";
    KalturaBaseEntryOrderBy["moderationCountAsc"] = "+moderationCount";
    KalturaBaseEntryOrderBy["moderationCountDesc"] = "-moderationCount";
    KalturaBaseEntryOrderBy["nameAsc"] = "+name";
    KalturaBaseEntryOrderBy["nameDesc"] = "-name";
    KalturaBaseEntryOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaBaseEntryOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaBaseEntryOrderBy["rankAsc"] = "+rank";
    KalturaBaseEntryOrderBy["rankDesc"] = "-rank";
    KalturaBaseEntryOrderBy["recentAsc"] = "+recent";
    KalturaBaseEntryOrderBy["recentDesc"] = "-recent";
    KalturaBaseEntryOrderBy["startDateAsc"] = "+startDate";
    KalturaBaseEntryOrderBy["startDateDesc"] = "-startDate";
    KalturaBaseEntryOrderBy["totalRankAsc"] = "+totalRank";
    KalturaBaseEntryOrderBy["totalRankDesc"] = "-totalRank";
    KalturaBaseEntryOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaBaseEntryOrderBy["updatedAtDesc"] = "-updatedAt";
    KalturaBaseEntryOrderBy["weightAsc"] = "+weight";
    KalturaBaseEntryOrderBy["weightDesc"] = "-weight";
})(KalturaBaseEntryOrderBy = exports.KalturaBaseEntryOrderBy || (exports.KalturaBaseEntryOrderBy = {}));
