// kaltura-node-typescript-client/api/types/KalturaPlaylist.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPlaylist = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaMediaEntryFilterForPlaylist_1 = require("./KalturaMediaEntryFilterForPlaylist");
var KalturaPlaylistType_1 = require("./KalturaPlaylistType");
var KalturaBaseEntry_1 = require("./KalturaBaseEntry");
var KalturaPlaylist = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPlaylist, _super);
    function KalturaPlaylist(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.filters === 'undefined')
            _this.filters = [];
        return _this;
    }
    KalturaPlaylist.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPlaylist' },
            playlistContent: { type: 's' },
            filters: { type: 'a', subTypeConstructor: KalturaMediaEntryFilterForPlaylist_1.KalturaMediaEntryFilterForPlaylist, subType: 'KalturaMediaEntryFilterForPlaylist' },
            totalResults: { type: 'n' },
            playlistType: { type: 'en', subTypeConstructor: KalturaPlaylistType_1.KalturaPlaylistType, subType: 'KalturaPlaylistType' },
            plays: { type: 'n', readOnly: true },
            views: { type: 'n', readOnly: true },
            duration: { type: 'n', readOnly: true },
            executeUrl: { type: 's', readOnly: true }
        });
        return result;
    };
    return KalturaPlaylist;
}(KalturaBaseEntry_1.KalturaBaseEntry));
exports.KalturaPlaylist = KalturaPlaylist;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPlaylist', KalturaPlaylist);
