// kaltura-node-typescript-client/api/types/KalturaLiveStreamScheduleEventOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveStreamScheduleEventOrderBy = void 0;
var KalturaLiveStreamScheduleEventOrderBy;
(function (KalturaLiveStreamScheduleEventOrderBy) {
    KalturaLiveStreamScheduleEventOrderBy["createdAtAsc"] = "+createdAt";
    KalturaLiveStreamScheduleEventOrderBy["createdAtDesc"] = "-createdAt";
    KalturaLiveStreamScheduleEventOrderBy["endDateAsc"] = "+endDate";
    KalturaLiveStreamScheduleEventOrderBy["endDateDesc"] = "-endDate";
    KalturaLiveStreamScheduleEventOrderBy["priorityAsc"] = "+priority";
    KalturaLiveStreamScheduleEventOrderBy["priorityDesc"] = "-priority";
    KalturaLiveStreamScheduleEventOrderBy["startDateAsc"] = "+startDate";
    KalturaLiveStreamScheduleEventOrderBy["startDateDesc"] = "-startDate";
    KalturaLiveStreamScheduleEventOrderBy["summaryAsc"] = "+summary";
    KalturaLiveStreamScheduleEventOrderBy["summaryDesc"] = "-summary";
    KalturaLiveStreamScheduleEventOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaLiveStreamScheduleEventOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaLiveStreamScheduleEventOrderBy = exports.KalturaLiveStreamScheduleEventOrderBy || (exports.KalturaLiveStreamScheduleEventOrderBy = {}));
