// kaltura-node-typescript-client/api/types/KalturaHuluDistributionJobProviderData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaHuluDistributionJobProviderData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCuePoint_1 = require("./KalturaCuePoint");
var KalturaString_1 = require("./KalturaString");
var KalturaConfigurableDistributionJobProviderData_1 = require("./KalturaConfigurableDistributionJobProviderData");
var KalturaHuluDistributionJobProviderData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaHuluDistributionJobProviderData, _super);
    function KalturaHuluDistributionJobProviderData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.cuePoints === 'undefined')
            _this.cuePoints = [];
        if (typeof _this.captionLocalPaths === 'undefined')
            _this.captionLocalPaths = [];
        return _this;
    }
    KalturaHuluDistributionJobProviderData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaHuluDistributionJobProviderData' },
            videoAssetFilePath: { type: 's' },
            thumbAssetFilePath: { type: 's' },
            cuePoints: { type: 'a', subTypeConstructor: KalturaCuePoint_1.KalturaCuePoint, subType: 'KalturaCuePoint' },
            fileBaseName: { type: 's' },
            captionLocalPaths: { type: 'a', subTypeConstructor: KalturaString_1.KalturaString, subType: 'KalturaString' }
        });
        return result;
    };
    return KalturaHuluDistributionJobProviderData;
}(KalturaConfigurableDistributionJobProviderData_1.KalturaConfigurableDistributionJobProviderData));
exports.KalturaHuluDistributionJobProviderData = KalturaHuluDistributionJobProviderData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaHuluDistributionJobProviderData', KalturaHuluDistributionJobProviderData);
