// kaltura-node-typescript-client/api/types/KalturaContainerFormat.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaContainerFormat = void 0;
var KalturaContainerFormat;
(function (KalturaContainerFormat) {
    KalturaContainerFormat["applehttp"] = "applehttp";
    KalturaContainerFormat["avi"] = "avi";
    KalturaContainerFormat["bif"] = "bif";
    KalturaContainerFormat["bmp"] = "bmp";
    KalturaContainerFormat["copy"] = "copy";
    KalturaContainerFormat["flv"] = "flv";
    KalturaContainerFormat["hls"] = "hls";
    KalturaContainerFormat["isma"] = "isma";
    KalturaContainerFormat["ismv"] = "ismv";
    KalturaContainerFormat["jpg"] = "jpg";
    KalturaContainerFormat["m2ts"] = "m2ts";
    KalturaContainerFormat["m4v"] = "m4v";
    KalturaContainerFormat["mkv"] = "mkv";
    KalturaContainerFormat["mov"] = "mov";
    KalturaContainerFormat["mp3"] = "mp3";
    KalturaContainerFormat["mp4"] = "mp4";
    KalturaContainerFormat["mpeg"] = "mpeg";
    KalturaContainerFormat["mpegts"] = "mpegts";
    KalturaContainerFormat["mxf"] = "mxf";
    KalturaContainerFormat["ogg"] = "ogg";
    KalturaContainerFormat["ogv"] = "ogv";
    KalturaContainerFormat["pdf"] = "pdf";
    KalturaContainerFormat["png"] = "png";
    KalturaContainerFormat["swf"] = "swf";
    KalturaContainerFormat["wav"] = "wav";
    KalturaContainerFormat["webm"] = "webm";
    KalturaContainerFormat["wma"] = "wma";
    KalturaContainerFormat["wmv"] = "wmv";
    KalturaContainerFormat["wvm"] = "wvm";
    KalturaContainerFormat["_3gp"] = "3gp";
})(KalturaContainerFormat = exports.KalturaContainerFormat || (exports.KalturaContainerFormat = {}));
