// kaltura-node-typescript-client/api/types/KalturaAnalyticsFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAnalyticsFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaReportFilter_1 = require("./KalturaReportFilter");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaAnalyticsFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAnalyticsFilter, _super);
    function KalturaAnalyticsFilter(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.filters === 'undefined')
            _this.filters = [];
        return _this;
    }
    KalturaAnalyticsFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAnalyticsFilter' },
            from_time: { type: 's' },
            to_time: { type: 's' },
            metrics: { type: 's' },
            utcOffset: { type: 'n' },
            dimensions: { type: 's' },
            filters: { type: 'a', subTypeConstructor: KalturaReportFilter_1.KalturaReportFilter, subType: 'KalturaReportFilter' },
            orderBy: { type: 's' }
        });
        return result;
    };
    return KalturaAnalyticsFilter;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaAnalyticsFilter = KalturaAnalyticsFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAnalyticsFilter', KalturaAnalyticsFilter);
