// kaltura-node-typescript-client/api/types/KalturaBaseEntryCompareAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBaseEntryCompareAttribute = void 0;
var KalturaBaseEntryCompareAttribute;
(function (KalturaBaseEntryCompareAttribute) {
    KalturaBaseEntryCompareAttribute["accessControlId"] = "accessControlId";
    KalturaBaseEntryCompareAttribute["createdAt"] = "createdAt";
    KalturaBaseEntryCompareAttribute["endDate"] = "endDate";
    KalturaBaseEntryCompareAttribute["moderationCount"] = "moderationCount";
    KalturaBaseEntryCompareAttribute["moderationStatus"] = "moderationStatus";
    KalturaBaseEntryCompareAttribute["partnerId"] = "partnerId";
    KalturaBaseEntryCompareAttribute["partnerSortValue"] = "partnerSortValue";
    KalturaBaseEntryCompareAttribute["rank"] = "rank";
    KalturaBaseEntryCompareAttribute["replacementStatus"] = "replacementStatus";
    KalturaBaseEntryCompareAttribute["startDate"] = "startDate";
    KalturaBaseEntryCompareAttribute["status"] = "status";
    KalturaBaseEntryCompareAttribute["totalRank"] = "totalRank";
    KalturaBaseEntryCompareAttribute["type"] = "type";
    KalturaBaseEntryCompareAttribute["updatedAt"] = "updatedAt";
})(KalturaBaseEntryCompareAttribute = exports.KalturaBaseEntryCompareAttribute || (exports.KalturaBaseEntryCompareAttribute = {}));
