// kaltura-node-typescript-client/api/types/KalturaHuluDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaHuluDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaString_1 = require("./KalturaString");
var KalturaDistributionProtocol_1 = require("./KalturaDistributionProtocol");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaHuluDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaHuluDistributionProfile, _super);
    function KalturaHuluDistributionProfile(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.seriesAdditionalCategories === 'undefined')
            _this.seriesAdditionalCategories = [];
        return _this;
    }
    KalturaHuluDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaHuluDistributionProfile' },
            sftpHost: { type: 's' },
            sftpLogin: { type: 's' },
            sftpPass: { type: 's' },
            seriesChannel: { type: 's' },
            seriesPrimaryCategory: { type: 's' },
            seriesAdditionalCategories: { type: 'a', subTypeConstructor: KalturaString_1.KalturaString, subType: 'KalturaString' },
            seasonNumber: { type: 's' },
            seasonSynopsis: { type: 's' },
            seasonTuneInInformation: { type: 's' },
            videoMediaType: { type: 's' },
            disableEpisodeNumberCustomValidation: { type: 'b' },
            protocol: { type: 'en', subTypeConstructor: KalturaDistributionProtocol_1.KalturaDistributionProtocol, subType: 'KalturaDistributionProtocol' },
            asperaHost: { type: 's' },
            asperaLogin: { type: 's' },
            asperaPass: { type: 's' },
            port: { type: 'n' },
            passphrase: { type: 's' },
            asperaPublicKey: { type: 's' },
            asperaPrivateKey: { type: 's' }
        });
        return result;
    };
    return KalturaHuluDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaHuluDistributionProfile = KalturaHuluDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaHuluDistributionProfile', KalturaHuluDistributionProfile);
