// kaltura-node-typescript-client/api/types/KalturaYouTubeDistributionJobProviderData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaYouTubeDistributionJobProviderData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionJobProviderData_1 = require("./KalturaConfigurableDistributionJobProviderData");
var KalturaYouTubeDistributionJobProviderData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaYouTubeDistributionJobProviderData, _super);
    function KalturaYouTubeDistributionJobProviderData(data) {
        return _super.call(this, data) || this;
    }
    KalturaYouTubeDistributionJobProviderData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaYouTubeDistributionJobProviderData' },
            videoAssetFilePath: { type: 's' },
            thumbAssetFilePath: { type: 's' },
            thumbAssetId: { type: 's' },
            captionAssetIds: { type: 's' },
            sftpDirectory: { type: 's' },
            sftpMetadataFilename: { type: 's' },
            currentPlaylists: { type: 's' },
            newPlaylists: { type: 's' },
            submitXml: { type: 's' },
            updateXml: { type: 's' },
            deleteXml: { type: 's' },
            googleClientId: { type: 's' },
            googleClientSecret: { type: 's' },
            googleTokenData: { type: 's' },
            captionsCsvMap: { type: 's' },
            submitCsvMap: { type: 's' },
            updateCsvMap: { type: 's' },
            deleteVideoIds: { type: 's' }
        });
        return result;
    };
    return KalturaYouTubeDistributionJobProviderData;
}(KalturaConfigurableDistributionJobProviderData_1.KalturaConfigurableDistributionJobProviderData));
exports.KalturaYouTubeDistributionJobProviderData = KalturaYouTubeDistributionJobProviderData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaYouTubeDistributionJobProviderData', KalturaYouTubeDistributionJobProviderData);
