// kaltura-node-typescript-client/api/types/KalturaVendorServiceFeature.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaVendorServiceFeature = void 0;
var KalturaVendorServiceFeature;
(function (KalturaVendorServiceFeature) {
    KalturaVendorServiceFeature[KalturaVendorServiceFeature["alignment"] = 3] = "alignment";
    KalturaVendorServiceFeature[KalturaVendorServiceFeature["audioDescription"] = 4] = "audioDescription";
    KalturaVendorServiceFeature[KalturaVendorServiceFeature["captions"] = 1] = "captions";
    KalturaVendorServiceFeature[KalturaVendorServiceFeature["chaptering"] = 5] = "chaptering";
    KalturaVendorServiceFeature[KalturaVendorServiceFeature["clips"] = 10] = "clips";
    KalturaVendorServiceFeature[KalturaVendorServiceFeature["dubbing"] = 7] = "dubbing";
    KalturaVendorServiceFeature[KalturaVendorServiceFeature["extendedAudioDescription"] = 9] = "extendedAudioDescription";
    KalturaVendorServiceFeature[KalturaVendorServiceFeature["intelligentTagging"] = 6] = "intelligentTagging";
    KalturaVendorServiceFeature[KalturaVendorServiceFeature["liveCaption"] = 8] = "liveCaption";
    KalturaVendorServiceFeature[KalturaVendorServiceFeature["liveTranslation"] = 11] = "liveTranslation";
    KalturaVendorServiceFeature[KalturaVendorServiceFeature["quiz"] = 12] = "quiz";
    KalturaVendorServiceFeature[KalturaVendorServiceFeature["translation"] = 2] = "translation";
})(KalturaVendorServiceFeature = exports.KalturaVendorServiceFeature || (exports.KalturaVendorServiceFeature = {}));
