// kaltura-node-typescript-client/api/types/KalturaThumbParamsOutputBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaThumbParamsOutputBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaThumbParamsFilter_1 = require("./KalturaThumbParamsFilter");
var KalturaThumbParamsOutputBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaThumbParamsOutputBaseFilter, _super);
    function KalturaThumbParamsOutputBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaThumbParamsOutputBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaThumbParamsOutputBaseFilter' },
            thumbParamsIdEqual: { type: 'n' },
            thumbParamsVersionEqual: { type: 's' },
            thumbAssetIdEqual: { type: 's' },
            thumbAssetVersionEqual: { type: 's' }
        });
        return result;
    };
    return KalturaThumbParamsOutputBaseFilter;
}(KalturaThumbParamsFilter_1.KalturaThumbParamsFilter));
exports.KalturaThumbParamsOutputBaseFilter = KalturaThumbParamsOutputBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaThumbParamsOutputBaseFilter', KalturaThumbParamsOutputBaseFilter);
