// kaltura-node-typescript-client/api/types/KalturaAnnotationBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAnnotationBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaCuePointFilter_1 = require("./KalturaCuePointFilter");
var KalturaAnnotationBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAnnotationBaseFilter, _super);
    function KalturaAnnotationBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaAnnotationBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAnnotationBaseFilter' },
            parentIdEqual: { type: 's' },
            parentIdIn: { type: 's' },
            textLike: { type: 's' },
            textMultiLikeOr: { type: 's' },
            textMultiLikeAnd: { type: 's' },
            endTimeGreaterThanOrEqual: { type: 'n' },
            endTimeLessThanOrEqual: { type: 'n' },
            durationGreaterThanOrEqual: { type: 'n' },
            durationLessThanOrEqual: { type: 'n' },
            isPublicEqual: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' }
        });
        return result;
    };
    return KalturaAnnotationBaseFilter;
}(KalturaCuePointFilter_1.KalturaCuePointFilter));
exports.KalturaAnnotationBaseFilter = KalturaAnnotationBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAnnotationBaseFilter', KalturaAnnotationBaseFilter);
