// kaltura-node-typescript-client/api/types/KalturaEntryScheduleEventBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEntryScheduleEventBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaScheduleEventFilter_1 = require("./KalturaScheduleEventFilter");
var KalturaEntryScheduleEventBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaEntryScheduleEventBaseFilter, _super);
    function KalturaEntryScheduleEventBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaEntryScheduleEventBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaEntryScheduleEventBaseFilter' },
            templateEntryIdEqual: { type: 's' },
            entryIdsLike: { type: 's' },
            entryIdsMultiLikeOr: { type: 's' },
            entryIdsMultiLikeAnd: { type: 's' },
            categoryIdsLike: { type: 's' },
            categoryIdsMultiLikeOr: { type: 's' },
            categoryIdsMultiLikeAnd: { type: 's' }
        });
        return result;
    };
    return KalturaEntryScheduleEventBaseFilter;
}(KalturaScheduleEventFilter_1.KalturaScheduleEventFilter));
exports.KalturaEntryScheduleEventBaseFilter = KalturaEntryScheduleEventBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaEntryScheduleEventBaseFilter', KalturaEntryScheduleEventBaseFilter);
