// kaltura-node-typescript-client/api/types/KalturaFtpDropFolder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFtpDropFolder = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaRemoteDropFolder_1 = require("./KalturaRemoteDropFolder");
var KalturaFtpDropFolder = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaFtpDropFolder, _super);
    function KalturaFtpDropFolder(data) {
        return _super.call(this, data) || this;
    }
    KalturaFtpDropFolder.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaFtpDropFolder' },
            host: { type: 's' },
            port: { type: 'n' },
            username: { type: 's' },
            password: { type: 's' }
        });
        return result;
    };
    return KalturaFtpDropFolder;
}(KalturaRemoteDropFolder_1.KalturaRemoteDropFolder));
exports.KalturaFtpDropFolder = KalturaFtpDropFolder;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaFtpDropFolder', KalturaFtpDropFolder);
