// kaltura-node-typescript-client/api/types/KalturaWebexDropFolder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaWebexDropFolder = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDropFolder_1 = require("./KalturaDropFolder");
var KalturaWebexDropFolder = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaWebexDropFolder, _super);
    function KalturaWebexDropFolder(data) {
        return _super.call(this, data) || this;
    }
    KalturaWebexDropFolder.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaWebexDropFolder' },
            webexUserId: { type: 's' },
            webexPassword: { type: 's' },
            webexSiteId: { type: 'n' },
            webexPartnerId: { type: 's' },
            webexServiceUrl: { type: 's' },
            webexHostIdMetadataFieldName: { type: 's' },
            deleteFromRecycleBin: { type: 'b' },
            webexServiceType: { type: 's' },
            webexSiteName: { type: 's' },
            deleteFromTimestamp: { type: 'd' }
        });
        return result;
    };
    return KalturaWebexDropFolder;
}(KalturaDropFolder_1.KalturaDropFolder));
exports.KalturaWebexDropFolder = KalturaWebexDropFolder;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaWebexDropFolder', KalturaWebexDropFolder);
