// kaltura-node-typescript-client/api/types/KalturaAdCuePoint.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAdCuePoint = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaAdProtocolType_1 = require("./KalturaAdProtocolType");
var KalturaAdType_1 = require("./KalturaAdType");
var KalturaCuePoint_1 = require("./KalturaCuePoint");
var KalturaAdCuePoint = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAdCuePoint, _super);
    function KalturaAdCuePoint(data) {
        return _super.call(this, data) || this;
    }
    KalturaAdCuePoint.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAdCuePoint' },
            protocolType: { type: 'es', subTypeConstructor: KalturaAdProtocolType_1.KalturaAdProtocolType, subType: 'KalturaAdProtocolType' },
            sourceUrl: { type: 's' },
            adType: { type: 'es', subTypeConstructor: KalturaAdType_1.KalturaAdType, subType: 'KalturaAdType' },
            title: { type: 's' },
            endTime: { type: 'n' },
            duration: { type: 'n' }
        });
        return result;
    };
    return KalturaAdCuePoint;
}(KalturaCuePoint_1.KalturaCuePoint));
exports.KalturaAdCuePoint = KalturaAdCuePoint;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAdCuePoint', KalturaAdCuePoint);
