// kaltura-node-typescript-client/api/types/KalturaUserRoleBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUserRoleBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUserRoleStatus_1 = require("./KalturaUserRoleStatus");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaUserRoleBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUserRoleBaseFilter, _super);
    function KalturaUserRoleBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaUserRoleBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUserRoleBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            nameEqual: { type: 's' },
            nameIn: { type: 's' },
            systemNameEqual: { type: 's' },
            systemNameIn: { type: 's' },
            descriptionLike: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaUserRoleStatus_1.KalturaUserRoleStatus, subType: 'KalturaUserRoleStatus' },
            statusIn: { type: 's' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            tagsMultiLikeOr: { type: 's' },
            tagsMultiLikeAnd: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' }
        });
        return result;
    };
    return KalturaUserRoleBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaUserRoleBaseFilter = KalturaUserRoleBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUserRoleBaseFilter', KalturaUserRoleBaseFilter);
