// kaltura-node-typescript-client/api/types/KalturaRecordScheduleEventOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaRecordScheduleEventOrderBy = void 0;
var KalturaRecordScheduleEventOrderBy;
(function (KalturaRecordScheduleEventOrderBy) {
    KalturaRecordScheduleEventOrderBy["createdAtAsc"] = "+createdAt";
    KalturaRecordScheduleEventOrderBy["createdAtDesc"] = "-createdAt";
    KalturaRecordScheduleEventOrderBy["endDateAsc"] = "+endDate";
    KalturaRecordScheduleEventOrderBy["endDateDesc"] = "-endDate";
    KalturaRecordScheduleEventOrderBy["priorityAsc"] = "+priority";
    KalturaRecordScheduleEventOrderBy["priorityDesc"] = "-priority";
    KalturaRecordScheduleEventOrderBy["startDateAsc"] = "+startDate";
    KalturaRecordScheduleEventOrderBy["startDateDesc"] = "-startDate";
    KalturaRecordScheduleEventOrderBy["summaryAsc"] = "+summary";
    KalturaRecordScheduleEventOrderBy["summaryDesc"] = "-summary";
    KalturaRecordScheduleEventOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaRecordScheduleEventOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaRecordScheduleEventOrderBy = exports.KalturaRecordScheduleEventOrderBy || (exports.KalturaRecordScheduleEventOrderBy = {}));
