// kaltura-node-typescript-client/api/types/KalturaDetachedResponseProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDetachedResponseProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaResponseProfileType_1 = require("./KalturaResponseProfileType");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaFilterPager_1 = require("./KalturaFilterPager");
var KalturaResponseProfileMapping_1 = require("./KalturaResponseProfileMapping");
var KalturaBaseResponseProfile_1 = require("./KalturaBaseResponseProfile");
var KalturaDetachedResponseProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDetachedResponseProfile, _super);
    function KalturaDetachedResponseProfile(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.relatedProfiles === 'undefined')
            _this.relatedProfiles = [];
        if (typeof _this.mappings === 'undefined')
            _this.mappings = [];
        return _this;
    }
    KalturaDetachedResponseProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDetachedResponseProfile' },
            name: { type: 's' },
            type: { type: 'en', subTypeConstructor: KalturaResponseProfileType_1.KalturaResponseProfileType, subType: 'KalturaResponseProfileType' },
            fields: { type: 's' },
            filter: { type: 'o', subTypeConstructor: KalturaRelatedFilter_1.KalturaRelatedFilter, subType: 'KalturaRelatedFilter' },
            pager: { type: 'o', subTypeConstructor: KalturaFilterPager_1.KalturaFilterPager, subType: 'KalturaFilterPager' },
            relatedProfiles: { type: 'a', subTypeConstructor: KalturaDetachedResponseProfile, subType: 'KalturaDetachedResponseProfile' },
            mappings: { type: 'a', subTypeConstructor: KalturaResponseProfileMapping_1.KalturaResponseProfileMapping, subType: 'KalturaResponseProfileMapping' }
        });
        return result;
    };
    return KalturaDetachedResponseProfile;
}(KalturaBaseResponseProfile_1.KalturaBaseResponseProfile));
exports.KalturaDetachedResponseProfile = KalturaDetachedResponseProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDetachedResponseProfile', KalturaDetachedResponseProfile);
