// kaltura-node-typescript-client/api/types/KalturaMediaEntryCompareAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMediaEntryCompareAttribute = void 0;
var KalturaMediaEntryCompareAttribute;
(function (KalturaMediaEntryCompareAttribute) {
    KalturaMediaEntryCompareAttribute["accessControlId"] = "accessControlId";
    KalturaMediaEntryCompareAttribute["createdAt"] = "createdAt";
    KalturaMediaEntryCompareAttribute["endDate"] = "endDate";
    KalturaMediaEntryCompareAttribute["lastPlayedAt"] = "lastPlayedAt";
    KalturaMediaEntryCompareAttribute["mediaDate"] = "mediaDate";
    KalturaMediaEntryCompareAttribute["mediaType"] = "mediaType";
    KalturaMediaEntryCompareAttribute["moderationCount"] = "moderationCount";
    KalturaMediaEntryCompareAttribute["moderationStatus"] = "moderationStatus";
    KalturaMediaEntryCompareAttribute["msDuration"] = "msDuration";
    KalturaMediaEntryCompareAttribute["partnerId"] = "partnerId";
    KalturaMediaEntryCompareAttribute["partnerSortValue"] = "partnerSortValue";
    KalturaMediaEntryCompareAttribute["plays"] = "plays";
    KalturaMediaEntryCompareAttribute["rank"] = "rank";
    KalturaMediaEntryCompareAttribute["replacementStatus"] = "replacementStatus";
    KalturaMediaEntryCompareAttribute["startDate"] = "startDate";
    KalturaMediaEntryCompareAttribute["status"] = "status";
    KalturaMediaEntryCompareAttribute["totalRank"] = "totalRank";
    KalturaMediaEntryCompareAttribute["type"] = "type";
    KalturaMediaEntryCompareAttribute["updatedAt"] = "updatedAt";
    KalturaMediaEntryCompareAttribute["views"] = "views";
})(KalturaMediaEntryCompareAttribute = exports.KalturaMediaEntryCompareAttribute || (exports.KalturaMediaEntryCompareAttribute = {}));
