// kaltura-node-typescript-client/api/types/KalturaGenericDistributionProfileAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaGenericDistributionProfileAction = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDistributionProtocol_1 = require("./KalturaDistributionProtocol");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaGenericDistributionProfileAction = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaGenericDistributionProfileAction, _super);
    function KalturaGenericDistributionProfileAction(data) {
        return _super.call(this, data) || this;
    }
    KalturaGenericDistributionProfileAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaGenericDistributionProfileAction' },
            protocol: { type: 'en', subTypeConstructor: KalturaDistributionProtocol_1.KalturaDistributionProtocol, subType: 'KalturaDistributionProtocol' },
            serverUrl: { type: 's' },
            serverPath: { type: 's' },
            username: { type: 's' },
            password: { type: 's' },
            ftpPassiveMode: { type: 'b' },
            httpFieldName: { type: 's' },
            httpFileName: { type: 's' }
        });
        return result;
    };
    return KalturaGenericDistributionProfileAction;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaGenericDistributionProfileAction = KalturaGenericDistributionProfileAction;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaGenericDistributionProfileAction', KalturaGenericDistributionProfileAction);
