// kaltura-node-typescript-client/api/types/KalturaEntryVendorTaskBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEntryVendorTaskBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaEntryVendorTaskStatus_1 = require("./KalturaEntryVendorTaskStatus");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaEntryVendorTaskBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaEntryVendorTaskBaseFilter, _super);
    function KalturaEntryVendorTaskBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaEntryVendorTaskBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaEntryVendorTaskBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            idNotIn: { type: 's' },
            vendorPartnerIdEqual: { type: 'n' },
            vendorPartnerIdIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            queueTimeGreaterThanOrEqual: { type: 'd' },
            queueTimeLessThanOrEqual: { type: 'd' },
            finishTimeGreaterThanOrEqual: { type: 'd' },
            finishTimeLessThanOrEqual: { type: 'd' },
            entryIdEqual: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaEntryVendorTaskStatus_1.KalturaEntryVendorTaskStatus, subType: 'KalturaEntryVendorTaskStatus' },
            statusIn: { type: 's' },
            reachProfileIdEqual: { type: 'n' },
            reachProfileIdIn: { type: 's' },
            catalogItemIdEqual: { type: 'n' },
            catalogItemIdIn: { type: 's' },
            userIdEqual: { type: 's' },
            contextEqual: { type: 's' },
            expectedFinishTimeGreaterThanOrEqual: { type: 'd' },
            expectedFinishTimeLessThanOrEqual: { type: 'd' }
        });
        return result;
    };
    return KalturaEntryVendorTaskBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaEntryVendorTaskBaseFilter = KalturaEntryVendorTaskBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaEntryVendorTaskBaseFilter', KalturaEntryVendorTaskBaseFilter);
