// kaltura-node-typescript-client/api/types/KalturaExternalMediaEntryMatchAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaExternalMediaEntryMatchAttribute = void 0;
var KalturaExternalMediaEntryMatchAttribute;
(function (KalturaExternalMediaEntryMatchAttribute) {
    KalturaExternalMediaEntryMatchAttribute["adminTags"] = "adminTags";
    KalturaExternalMediaEntryMatchAttribute["categoriesIds"] = "categoriesIds";
    KalturaExternalMediaEntryMatchAttribute["creatorId"] = "creatorId";
    KalturaExternalMediaEntryMatchAttribute["description"] = "description";
    KalturaExternalMediaEntryMatchAttribute["durationType"] = "durationType";
    KalturaExternalMediaEntryMatchAttribute["flavorParamsIds"] = "flavorParamsIds";
    KalturaExternalMediaEntryMatchAttribute["groupId"] = "groupId";
    KalturaExternalMediaEntryMatchAttribute["id"] = "id";
    KalturaExternalMediaEntryMatchAttribute["_name"] = "name";
    KalturaExternalMediaEntryMatchAttribute["referenceId"] = "referenceId";
    KalturaExternalMediaEntryMatchAttribute["replacedEntryId"] = "replacedEntryId";
    KalturaExternalMediaEntryMatchAttribute["replacingEntryId"] = "replacingEntryId";
    KalturaExternalMediaEntryMatchAttribute["searchText"] = "searchText";
    KalturaExternalMediaEntryMatchAttribute["tags"] = "tags";
    KalturaExternalMediaEntryMatchAttribute["userId"] = "userId";
})(KalturaExternalMediaEntryMatchAttribute = exports.KalturaExternalMediaEntryMatchAttribute || (exports.KalturaExternalMediaEntryMatchAttribute = {}));
