// kaltura-node-typescript-client/api/types/KalturaMappedObjectsCsvJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMappedObjectsCsvJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCsvAdditionalFieldInfo_1 = require("./KalturaCsvAdditionalFieldInfo");
var KalturaKeyValue_1 = require("./KalturaKeyValue");
var KalturaExportToCsvOptions_1 = require("./KalturaExportToCsvOptions");
var KalturaExportCsvJobData_1 = require("./KalturaExportCsvJobData");
var KalturaMappedObjectsCsvJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMappedObjectsCsvJobData, _super);
    function KalturaMappedObjectsCsvJobData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.additionalFields === 'undefined')
            _this.additionalFields = [];
        if (typeof _this.mappedFields === 'undefined')
            _this.mappedFields = [];
        return _this;
    }
    KalturaMappedObjectsCsvJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMappedObjectsCsvJobData' },
            metadataProfileId: { type: 'n' },
            additionalFields: { type: 'a', subTypeConstructor: KalturaCsvAdditionalFieldInfo_1.KalturaCsvAdditionalFieldInfo, subType: 'KalturaCsvAdditionalFieldInfo' },
            mappedFields: { type: 'a', subTypeConstructor: KalturaKeyValue_1.KalturaKeyValue, subType: 'KalturaKeyValue' },
            options: { type: 'o', subTypeConstructor: KalturaExportToCsvOptions_1.KalturaExportToCsvOptions, subType: 'KalturaExportToCsvOptions' }
        });
        return result;
    };
    return KalturaMappedObjectsCsvJobData;
}(KalturaExportCsvJobData_1.KalturaExportCsvJobData));
exports.KalturaMappedObjectsCsvJobData = KalturaMappedObjectsCsvJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMappedObjectsCsvJobData', KalturaMappedObjectsCsvJobData);
