// kaltura-node-typescript-client/api/types/KalturaResponseProfileCacheRecalculateOptions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaResponseProfileCacheRecalculateOptions = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaResponseProfileCacheRecalculateOptions = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaResponseProfileCacheRecalculateOptions, _super);
    function KalturaResponseProfileCacheRecalculateOptions(data) {
        return _super.call(this, data) || this;
    }
    KalturaResponseProfileCacheRecalculateOptions.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaResponseProfileCacheRecalculateOptions' },
            limit: { type: 'n' },
            cachedObjectType: { type: 's' },
            objectId: { type: 's' },
            startObjectKey: { type: 's' },
            endObjectKey: { type: 's' },
            jobCreatedAt: { type: 'd' },
            isFirstLoop: { type: 'b' }
        });
        return result;
    };
    return KalturaResponseProfileCacheRecalculateOptions;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaResponseProfileCacheRecalculateOptions = KalturaResponseProfileCacheRecalculateOptions;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaResponseProfileCacheRecalculateOptions', KalturaResponseProfileCacheRecalculateOptions);
