// kaltura-node-typescript-client/api/types/KalturaYahooSyndicationFeedCategories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaYahooSyndicationFeedCategories = void 0;
var KalturaYahooSyndicationFeedCategories;
(function (KalturaYahooSyndicationFeedCategories) {
    KalturaYahooSyndicationFeedCategories["action"] = "Action";
    KalturaYahooSyndicationFeedCategories["animals"] = "Animals";
    KalturaYahooSyndicationFeedCategories["artAndAnimation"] = "Art &amp; Animation";
    KalturaYahooSyndicationFeedCategories["commercials"] = "Commercials";
    KalturaYahooSyndicationFeedCategories["entertainmentAndTv"] = "Entertainment &amp; TV";
    KalturaYahooSyndicationFeedCategories["family"] = "Family";
    KalturaYahooSyndicationFeedCategories["food"] = "Food";
    KalturaYahooSyndicationFeedCategories["funnyVideos"] = "Funny Videos";
    KalturaYahooSyndicationFeedCategories["games"] = "Games";
    KalturaYahooSyndicationFeedCategories["healthAndBeauty"] = "Health &amp; Beauty";
    KalturaYahooSyndicationFeedCategories["howTo"] = "How-To";
    KalturaYahooSyndicationFeedCategories["moviesAndShorts"] = "Movies &amp; Shorts";
    KalturaYahooSyndicationFeedCategories["music"] = "Music";
    KalturaYahooSyndicationFeedCategories["newsAndPolitics"] = "News &amp; Politics";
    KalturaYahooSyndicationFeedCategories["peopleAndVlogs"] = "People &amp; Vlogs";
    KalturaYahooSyndicationFeedCategories["productsAndTech"] = "Products &amp; Tech.";
    KalturaYahooSyndicationFeedCategories["scienceAndEnvironment"] = "Science &amp; Environment";
    KalturaYahooSyndicationFeedCategories["sports"] = "Sports";
    KalturaYahooSyndicationFeedCategories["transportation"] = "Transportation";
    KalturaYahooSyndicationFeedCategories["travel"] = "Travel";
})(KalturaYahooSyndicationFeedCategories = exports.KalturaYahooSyndicationFeedCategories || (exports.KalturaYahooSyndicationFeedCategories = {}));
