// kaltura-node-typescript-client/api/types/KalturaDailymotionDistributionCaptionInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDailymotionDistributionCaptionInfo = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDailymotionDistributionCaptionAction_1 = require("./KalturaDailymotionDistributionCaptionAction");
var KalturaDailymotionDistributionCaptionFormat_1 = require("./KalturaDailymotionDistributionCaptionFormat");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaDailymotionDistributionCaptionInfo = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDailymotionDistributionCaptionInfo, _super);
    function KalturaDailymotionDistributionCaptionInfo(data) {
        return _super.call(this, data) || this;
    }
    KalturaDailymotionDistributionCaptionInfo.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDailymotionDistributionCaptionInfo' },
            language: { type: 's' },
            filePath: { type: 's' },
            remoteId: { type: 's' },
            action: { type: 'en', subTypeConstructor: KalturaDailymotionDistributionCaptionAction_1.KalturaDailymotionDistributionCaptionAction, subType: 'KalturaDailymotionDistributionCaptionAction' },
            version: { type: 's' },
            assetId: { type: 's' },
            format: { type: 'en', subTypeConstructor: KalturaDailymotionDistributionCaptionFormat_1.KalturaDailymotionDistributionCaptionFormat, subType: 'KalturaDailymotionDistributionCaptionFormat' }
        });
        return result;
    };
    return KalturaDailymotionDistributionCaptionInfo;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaDailymotionDistributionCaptionInfo = KalturaDailymotionDistributionCaptionInfo;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDailymotionDistributionCaptionInfo', KalturaDailymotionDistributionCaptionInfo);
