// kaltura-node-typescript-client/api/types/KalturaCropDimensions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCropDimensions = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaCropDimensions = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCropDimensions, _super);
    function KalturaCropDimensions(data) {
        return _super.call(this, data) || this;
    }
    KalturaCropDimensions.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCropDimensions' },
            left: { type: 'n' },
            top: { type: 'n' },
            width: { type: 'n' },
            height: { type: 'n' }
        });
        return result;
    };
    return KalturaCropDimensions;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaCropDimensions = KalturaCropDimensions;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCropDimensions', KalturaCropDimensions);
