// kaltura-node-typescript-client/api/types/KalturaDropFolderContentProcessorJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDropFolderContentProcessorJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDropFolderContentFileHandlerMatchPolicy_1 = require("./KalturaDropFolderContentFileHandlerMatchPolicy");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaDropFolderContentProcessorJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDropFolderContentProcessorJobData, _super);
    function KalturaDropFolderContentProcessorJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaDropFolderContentProcessorJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDropFolderContentProcessorJobData' },
            dropFolderId: { type: 'n' },
            dropFolderFileIds: { type: 's' },
            parsedSlug: { type: 's' },
            contentMatchPolicy: { type: 'en', subTypeConstructor: KalturaDropFolderContentFileHandlerMatchPolicy_1.KalturaDropFolderContentFileHandlerMatchPolicy, subType: 'KalturaDropFolderContentFileHandlerMatchPolicy' },
            conversionProfileId: { type: 'n' },
            parsedUserId: { type: 's' }
        });
        return result;
    };
    return KalturaDropFolderContentProcessorJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaDropFolderContentProcessorJobData = KalturaDropFolderContentProcessorJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDropFolderContentProcessorJobData', KalturaDropFolderContentProcessorJobData);
