// kaltura-node-typescript-client/api/types/KalturaIdeticDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaIdeticDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaIdeticDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaIdeticDistributionProfile, _super);
    function KalturaIdeticDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaIdeticDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaIdeticDistributionProfile' },
            ftpPath: { type: 's' },
            username: { type: 's' },
            password: { type: 's' },
            domain: { type: 's' }
        });
        return result;
    };
    return KalturaIdeticDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaIdeticDistributionProfile = KalturaIdeticDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaIdeticDistributionProfile', KalturaIdeticDistributionProfile);
