// kaltura-node-typescript-client/api/types/KalturaCatalogItemLanguage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCatalogItemLanguage = void 0;
var KalturaCatalogItemLanguage;
(function (KalturaCatalogItemLanguage) {
    KalturaCatalogItemLanguage["af"] = "Afrikaans";
    KalturaCatalogItemLanguage["ar"] = "Arabic";
    KalturaCatalogItemLanguage["autoDetect"] = "Auto Detect";
    KalturaCatalogItemLanguage["ba"] = "Bashkir";
    KalturaCatalogItemLanguage["be"] = "Byelorussian (Belarusian)";
    KalturaCatalogItemLanguage["bg"] = "Bulgarian";
    KalturaCatalogItemLanguage["ca"] = "Catalan";
    KalturaCatalogItemLanguage["cmn"] = "Mandarin Chinese";
    KalturaCatalogItemLanguage["cs"] = "Czech";
    KalturaCatalogItemLanguage["cy"] = "Welsh";
    KalturaCatalogItemLanguage["da"] = "Danish";
    KalturaCatalogItemLanguage["de"] = "German";
    KalturaCatalogItemLanguage["el"] = "Greek";
    KalturaCatalogItemLanguage["en"] = "English";
    KalturaCatalogItemLanguage["enGb"] = "English (British)";
    KalturaCatalogItemLanguage["enUs"] = "English (American)";
    KalturaCatalogItemLanguage["eo"] = "Esperanto";
    KalturaCatalogItemLanguage["es"] = "Spanish";
    KalturaCatalogItemLanguage["esXl"] = "Spanish (Latin America)";
    KalturaCatalogItemLanguage["et"] = "Estonian";
    KalturaCatalogItemLanguage["eu"] = "Basque";
    KalturaCatalogItemLanguage["fa"] = "Farsi";
    KalturaCatalogItemLanguage["fi"] = "Finnish";
    KalturaCatalogItemLanguage["fr"] = "French";
    KalturaCatalogItemLanguage["frCa"] = "French (Canada)";
    KalturaCatalogItemLanguage["ga"] = "Irish";
    KalturaCatalogItemLanguage["gd"] = "Gaelic (Scottish)";
    KalturaCatalogItemLanguage["gl"] = "Galician";
    KalturaCatalogItemLanguage["he"] = "Hebrew";
    KalturaCatalogItemLanguage["hi"] = "Hindi";
    KalturaCatalogItemLanguage["hr"] = "Croatian";
    KalturaCatalogItemLanguage["hu"] = "Hungarian";
    KalturaCatalogItemLanguage["ia"] = "Interlingua";
    KalturaCatalogItemLanguage["in"] = "Indonesian";
    KalturaCatalogItemLanguage["is"] = "Icelandic";
    KalturaCatalogItemLanguage["it"] = "Italian";
    KalturaCatalogItemLanguage["ja"] = "Japanese";
    KalturaCatalogItemLanguage["ko"] = "Korean";
    KalturaCatalogItemLanguage["lt"] = "Lithuanian";
    KalturaCatalogItemLanguage["lv"] = "Latvian (Lettish)";
    KalturaCatalogItemLanguage["ml"] = "Malayalam";
    KalturaCatalogItemLanguage["mn"] = "Mongolian";
    KalturaCatalogItemLanguage["mr"] = "Marathi";
    KalturaCatalogItemLanguage["ms"] = "Malay";
    KalturaCatalogItemLanguage["nl"] = "Dutch";
    KalturaCatalogItemLanguage["no"] = "Norwegian";
    KalturaCatalogItemLanguage["pl"] = "Polish";
    KalturaCatalogItemLanguage["pt"] = "Portuguese";
    KalturaCatalogItemLanguage["ptBr"] = "Portuguese (Brazil)";
    KalturaCatalogItemLanguage["ro"] = "Romanian";
    KalturaCatalogItemLanguage["ru"] = "Russian";
    KalturaCatalogItemLanguage["sk"] = "Slovak";
    KalturaCatalogItemLanguage["sl"] = "Slovenian";
    KalturaCatalogItemLanguage["sv"] = "Swedish";
    KalturaCatalogItemLanguage["ta"] = "Tamil";
    KalturaCatalogItemLanguage["te"] = "Telugu";
    KalturaCatalogItemLanguage["th"] = "Thai";
    KalturaCatalogItemLanguage["tr"] = "Turkish";
    KalturaCatalogItemLanguage["ug"] = "Uighur";
    KalturaCatalogItemLanguage["uk"] = "Ukrainian";
    KalturaCatalogItemLanguage["ur"] = "Urdu";
    KalturaCatalogItemLanguage["vi"] = "Vietnamese";
    KalturaCatalogItemLanguage["yue"] = "Cantonese";
    KalturaCatalogItemLanguage["zh"] = "Chinese";
    KalturaCatalogItemLanguage["zhTw"] = "Taiwanese Mandarin";
    KalturaCatalogItemLanguage["zu"] = "Zulu";
})(KalturaCatalogItemLanguage = exports.KalturaCatalogItemLanguage || (exports.KalturaCatalogItemLanguage = {}));
