// kaltura-node-typescript-client/api/types/KalturaStatsKmcEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaStatsKmcEvent = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaStatsKmcEventType_1 = require("./KalturaStatsKmcEventType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaStatsKmcEvent = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaStatsKmcEvent, _super);
    function KalturaStatsKmcEvent(data) {
        return _super.call(this, data) || this;
    }
    KalturaStatsKmcEvent.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaStatsKmcEvent' },
            clientVer: { type: 's' },
            kmcEventActionPath: { type: 's' },
            kmcEventType: { type: 'en', subTypeConstructor: KalturaStatsKmcEventType_1.KalturaStatsKmcEventType, subType: 'KalturaStatsKmcEventType' },
            eventTimestamp: { type: 'n' },
            sessionId: { type: 's' },
            partnerId: { type: 'n' },
            entryId: { type: 's' },
            widgetId: { type: 's' },
            uiconfId: { type: 'n' },
            userId: { type: 's' },
            userIp: { type: 's', readOnly: true }
        });
        return result;
    };
    return KalturaStatsKmcEvent;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaStatsKmcEvent = KalturaStatsKmcEvent;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaStatsKmcEvent', KalturaStatsKmcEvent);
