// kaltura-node-typescript-client/api/types/KalturaSessionCuePointBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSessionCuePointBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCuePointFilter_1 = require("./KalturaCuePointFilter");
var KalturaSessionCuePointBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaSessionCuePointBaseFilter, _super);
    function KalturaSessionCuePointBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaSessionCuePointBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaSessionCuePointBaseFilter' },
            endTimeGreaterThanOrEqual: { type: 'n' },
            endTimeLessThanOrEqual: { type: 'n' },
            durationGreaterThanOrEqual: { type: 'n' },
            durationLessThanOrEqual: { type: 'n' }
        });
        return result;
    };
    return KalturaSessionCuePointBaseFilter;
}(KalturaCuePointFilter_1.KalturaCuePointFilter));
exports.KalturaSessionCuePointBaseFilter = KalturaSessionCuePointBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaSessionCuePointBaseFilter', KalturaSessionCuePointBaseFilter);
