// kaltura-node-typescript-client/api/types/KalturaEntryScheduleEventOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEntryScheduleEventOrderBy = void 0;
var KalturaEntryScheduleEventOrderBy;
(function (KalturaEntryScheduleEventOrderBy) {
    KalturaEntryScheduleEventOrderBy["createdAtAsc"] = "+createdAt";
    KalturaEntryScheduleEventOrderBy["createdAtDesc"] = "-createdAt";
    KalturaEntryScheduleEventOrderBy["endDateAsc"] = "+endDate";
    KalturaEntryScheduleEventOrderBy["endDateDesc"] = "-endDate";
    KalturaEntryScheduleEventOrderBy["priorityAsc"] = "+priority";
    KalturaEntryScheduleEventOrderBy["priorityDesc"] = "-priority";
    KalturaEntryScheduleEventOrderBy["startDateAsc"] = "+startDate";
    KalturaEntryScheduleEventOrderBy["startDateDesc"] = "-startDate";
    KalturaEntryScheduleEventOrderBy["summaryAsc"] = "+summary";
    KalturaEntryScheduleEventOrderBy["summaryDesc"] = "-summary";
    KalturaEntryScheduleEventOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaEntryScheduleEventOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaEntryScheduleEventOrderBy = exports.KalturaEntryScheduleEventOrderBy || (exports.KalturaEntryScheduleEventOrderBy = {}));
