// kaltura-node-typescript-client/api/types/KalturaFileSyncDescriptor.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFileSyncDescriptor = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaFileSyncDescriptor = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaFileSyncDescriptor, _super);
    function KalturaFileSyncDescriptor(data) {
        return _super.call(this, data) || this;
    }
    KalturaFileSyncDescriptor.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaFileSyncDescriptor' },
            fileSyncLocalPath: { type: 's' },
            fileEncryptionKey: { type: 's' },
            fileSyncRemoteUrl: { type: 's' },
            fileSyncObjectSubType: { type: 'n' }
        });
        return result;
    };
    return KalturaFileSyncDescriptor;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaFileSyncDescriptor = KalturaFileSyncDescriptor;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaFileSyncDescriptor', KalturaFileSyncDescriptor);
