// kaltura-node-typescript-client/api/types/KalturaScheduleEventRecurrence.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaScheduleEventRecurrence = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaScheduleEventRecurrenceFrequency_1 = require("./KalturaScheduleEventRecurrenceFrequency");
var KalturaScheduleEventRecurrenceDay_1 = require("./KalturaScheduleEventRecurrenceDay");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaScheduleEventRecurrence = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaScheduleEventRecurrence, _super);
    function KalturaScheduleEventRecurrence(data) {
        return _super.call(this, data) || this;
    }
    KalturaScheduleEventRecurrence.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaScheduleEventRecurrence' },
            name: { type: 's' },
            frequency: { type: 'es', subTypeConstructor: KalturaScheduleEventRecurrenceFrequency_1.KalturaScheduleEventRecurrenceFrequency, subType: 'KalturaScheduleEventRecurrenceFrequency' },
            until: { type: 'd' },
            timeZone: { type: 's' },
            count: { type: 'n' },
            interval: { type: 'n' },
            bySecond: { type: 's' },
            byMinute: { type: 's' },
            byHour: { type: 's' },
            byDay: { type: 's' },
            byMonthDay: { type: 's' },
            byYearDay: { type: 's' },
            byWeekNumber: { type: 's' },
            byMonth: { type: 's' },
            byOffset: { type: 's' },
            weekStartDay: { type: 'es', subTypeConstructor: KalturaScheduleEventRecurrenceDay_1.KalturaScheduleEventRecurrenceDay, subType: 'KalturaScheduleEventRecurrenceDay' }
        });
        return result;
    };
    return KalturaScheduleEventRecurrence;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaScheduleEventRecurrence = KalturaScheduleEventRecurrence;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaScheduleEventRecurrence', KalturaScheduleEventRecurrence);
