// kaltura-node-typescript-client/api/types/KalturaSourceType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSourceType = void 0;
var KalturaSourceType;
(function (KalturaSourceType) {
    KalturaSourceType["akamaiLive"] = "29";
    KalturaSourceType["akamaiUniversalLive"] = "31";
    KalturaSourceType["clip"] = "35";
    KalturaSourceType["file"] = "1";
    KalturaSourceType["kalturaRecordedLive"] = "36";
    KalturaSourceType["lectureCapture"] = "37";
    KalturaSourceType["limelightLive"] = "limeLight.LIVE_STREAM";
    KalturaSourceType["liveChannel"] = "33";
    KalturaSourceType["liveStream"] = "32";
    KalturaSourceType["liveStreamOntextdataCaptions"] = "42";
    KalturaSourceType["manualLiveStream"] = "30";
    KalturaSourceType["recordedLive"] = "34";
    KalturaSourceType["searchProvider"] = "6";
    KalturaSourceType["url"] = "5";
    KalturaSourceType["velocixLive"] = "velocix.VELOCIX_LIVE";
    KalturaSourceType["webcam"] = "2";
})(KalturaSourceType = exports.KalturaSourceType || (exports.KalturaSourceType = {}));
