// kaltura-node-typescript-client/api/types/KalturaPermissionItemBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPermissionItemBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaPermissionItemType_1 = require("./KalturaPermissionItemType");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaPermissionItemBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPermissionItemBaseFilter, _super);
    function KalturaPermissionItemBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaPermissionItemBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPermissionItemBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            typeEqual: { type: 'es', subTypeConstructor: KalturaPermissionItemType_1.KalturaPermissionItemType, subType: 'KalturaPermissionItemType' },
            typeIn: { type: 's' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            tagsMultiLikeOr: { type: 's' },
            tagsMultiLikeAnd: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' }
        });
        return result;
    };
    return KalturaPermissionItemBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaPermissionItemBaseFilter = KalturaPermissionItemBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPermissionItemBaseFilter', KalturaPermissionItemBaseFilter);
