// kaltura-node-typescript-client/api/types/KalturaDrmPolicyBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDrmPolicyBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDrmProviderType_1 = require("./KalturaDrmProviderType");
var KalturaDrmPolicyStatus_1 = require("./KalturaDrmPolicyStatus");
var KalturaDrmLicenseScenario_1 = require("./KalturaDrmLicenseScenario");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaDrmPolicyBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDrmPolicyBaseFilter, _super);
    function KalturaDrmPolicyBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaDrmPolicyBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDrmPolicyBaseFilter' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            nameLike: { type: 's' },
            systemNameLike: { type: 's' },
            providerEqual: { type: 'es', subTypeConstructor: KalturaDrmProviderType_1.KalturaDrmProviderType, subType: 'KalturaDrmProviderType' },
            providerIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaDrmPolicyStatus_1.KalturaDrmPolicyStatus, subType: 'KalturaDrmPolicyStatus' },
            statusIn: { type: 's' },
            scenarioEqual: { type: 'es', subTypeConstructor: KalturaDrmLicenseScenario_1.KalturaDrmLicenseScenario, subType: 'KalturaDrmLicenseScenario' },
            scenarioIn: { type: 's' }
        });
        return result;
    };
    return KalturaDrmPolicyBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaDrmPolicyBaseFilter = KalturaDrmPolicyBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDrmPolicyBaseFilter', KalturaDrmPolicyBaseFilter);
