// kaltura-node-typescript-client/api/types/KalturaClipsVendorTaskData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaClipsVendorTaskData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaVendorTaskData_1 = require("./KalturaVendorTaskData");
var KalturaClipsVendorTaskData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaClipsVendorTaskData, _super);
    function KalturaClipsVendorTaskData(data) {
        return _super.call(this, data) || this;
    }
    KalturaClipsVendorTaskData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaClipsVendorTaskData' },
            clipsDuration: { type: 'n' },
            eventSessionContextId: { type: 's' },
            instruction: { type: 's' },
            clipsOutputJson: { type: 's' }
        });
        return result;
    };
    return KalturaClipsVendorTaskData;
}(KalturaVendorTaskData_1.KalturaVendorTaskData));
exports.KalturaClipsVendorTaskData = KalturaClipsVendorTaskData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaClipsVendorTaskData', KalturaClipsVendorTaskData);
