// kaltura-node-typescript-client/api/types/KalturaLiveToVodJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveToVodJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaLiveToVodJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveToVodJobData, _super);
    function KalturaLiveToVodJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaLiveToVodJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveToVodJobData' },
            vodEntryId: { type: 's' },
            liveEntryId: { type: 's' },
            totalVodDuration: { type: 'n' },
            lastSegmentDuration: { type: 'n' },
            amfArray: { type: 's' },
            lastCuePointSyncTime: { type: 'd' },
            lastSegmentDrift: { type: 'n' }
        });
        return result;
    };
    return KalturaLiveToVodJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaLiveToVodJobData = KalturaLiveToVodJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveToVodJobData', KalturaLiveToVodJobData);
