// kaltura-node-typescript-client/api/types/KalturaCatalogItemAdvancedFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCatalogItemAdvancedFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaVendorServiceType_1 = require("./KalturaVendorServiceType");
var KalturaVendorServiceFeature_1 = require("./KalturaVendorServiceFeature");
var KalturaVendorServiceTurnAroundTime_1 = require("./KalturaVendorServiceTurnAroundTime");
var KalturaCatalogItemLanguage_1 = require("./KalturaCatalogItemLanguage");
var KalturaSearchItem_1 = require("./KalturaSearchItem");
var KalturaCatalogItemAdvancedFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCatalogItemAdvancedFilter, _super);
    function KalturaCatalogItemAdvancedFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaCatalogItemAdvancedFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCatalogItemAdvancedFilter' },
            serviceTypeEqual: { type: 'en', subTypeConstructor: KalturaVendorServiceType_1.KalturaVendorServiceType, subType: 'KalturaVendorServiceType' },
            serviceTypeIn: { type: 's' },
            serviceFeatureEqual: { type: 'en', subTypeConstructor: KalturaVendorServiceFeature_1.KalturaVendorServiceFeature, subType: 'KalturaVendorServiceFeature' },
            serviceFeatureIn: { type: 's' },
            turnAroundTimeEqual: { type: 'en', subTypeConstructor: KalturaVendorServiceTurnAroundTime_1.KalturaVendorServiceTurnAroundTime, subType: 'KalturaVendorServiceTurnAroundTime' },
            turnAroundTimeIn: { type: 's' },
            sourceLanguageEqual: { type: 'es', subTypeConstructor: KalturaCatalogItemLanguage_1.KalturaCatalogItemLanguage, subType: 'KalturaCatalogItemLanguage' },
            targetLanguageEqual: { type: 'es', subTypeConstructor: KalturaCatalogItemLanguage_1.KalturaCatalogItemLanguage, subType: 'KalturaCatalogItemLanguage' }
        });
        return result;
    };
    return KalturaCatalogItemAdvancedFilter;
}(KalturaSearchItem_1.KalturaSearchItem));
exports.KalturaCatalogItemAdvancedFilter = KalturaCatalogItemAdvancedFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCatalogItemAdvancedFilter', KalturaCatalogItemAdvancedFilter);
