// kaltura-node-typescript-client/api/types/KalturaYouTubeDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaYouTubeDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaYouTubeDistributionFeedSpecVersion_1 = require("./KalturaYouTubeDistributionFeedSpecVersion");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaYouTubeDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaYouTubeDistributionProfile, _super);
    function KalturaYouTubeDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaYouTubeDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaYouTubeDistributionProfile' },
            feedSpecVersion: { type: 'es', subTypeConstructor: KalturaYouTubeDistributionFeedSpecVersion_1.KalturaYouTubeDistributionFeedSpecVersion, subType: 'KalturaYouTubeDistributionFeedSpecVersion' },
            username: { type: 's' },
            notificationEmail: { type: 's' },
            sftpHost: { type: 's' },
            sftpPort: { type: 'n' },
            sftpLogin: { type: 's' },
            sftpPublicKey: { type: 's' },
            sftpPrivateKey: { type: 's' },
            sftpBaseDir: { type: 's' },
            ownerName: { type: 's' },
            defaultCategory: { type: 's' },
            allowComments: { type: 's' },
            allowEmbedding: { type: 's' },
            allowRatings: { type: 's' },
            allowResponses: { type: 's' },
            commercialPolicy: { type: 's' },
            ugcPolicy: { type: 's' },
            target: { type: 's' },
            adServerPartnerId: { type: 's' },
            enableAdServer: { type: 'b' },
            allowPreRollAds: { type: 'b' },
            allowPostRollAds: { type: 'b' },
            strict: { type: 's' },
            overrideManualEdits: { type: 's' },
            urgentReference: { type: 's' },
            allowSyndication: { type: 's' },
            hideViewCount: { type: 's' },
            allowAdsenseForVideo: { type: 's' },
            allowInvideo: { type: 's' },
            allowMidRollAds: { type: 'b' },
            instreamStandard: { type: 's' },
            instreamTrueview: { type: 's' },
            claimType: { type: 's' },
            blockOutsideOwnership: { type: 's' },
            captionAutosync: { type: 's' },
            deleteReference: { type: 'b' },
            releaseClaims: { type: 'b' },
            apiAuthorizeUrl: { type: 's' },
            privacyStatus: { type: 's' },
            enableContentId: { type: 's' },
            thirdPartyAds: { type: 's' },
            productListingAds: { type: 's' },
            domainWhitelist: { type: 's' },
            notifySubscribers: { type: 's' }
        });
        return result;
    };
    return KalturaYouTubeDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaYouTubeDistributionProfile = KalturaYouTubeDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaYouTubeDistributionProfile', KalturaYouTubeDistributionProfile);
