// kaltura-node-typescript-client/api/types/KalturaLockFileSyncsResponse.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLockFileSyncsResponse = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaFileSync_1 = require("./KalturaFileSync");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaLockFileSyncsResponse = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLockFileSyncsResponse, _super);
    function KalturaLockFileSyncsResponse(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.fileSyncs === 'undefined')
            _this.fileSyncs = [];
        return _this;
    }
    KalturaLockFileSyncsResponse.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLockFileSyncsResponse' },
            fileSyncs: { type: 'a', subTypeConstructor: KalturaFileSync_1.KalturaFileSync, subType: 'KalturaFileSync' },
            limitReached: { type: 'b' },
            dcSecret: { type: 's' },
            baseUrl: { type: 's' }
        });
        return result;
    };
    return KalturaLockFileSyncsResponse;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaLockFileSyncsResponse = KalturaLockFileSyncsResponse;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLockFileSyncsResponse', KalturaLockFileSyncsResponse);
