// kaltura-node-typescript-client/api/types/KalturaBulkUploadBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBulkUploadBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBatchJobStatus_1 = require("./KalturaBatchJobStatus");
var KalturaBulkUploadObjectType_1 = require("./KalturaBulkUploadObjectType");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaBulkUploadBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBulkUploadBaseFilter, _super);
    function KalturaBulkUploadBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaBulkUploadBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBulkUploadBaseFilter' },
            uploadedOnGreaterThanOrEqual: { type: 'd' },
            uploadedOnLessThanOrEqual: { type: 'd' },
            uploadedOnEqual: { type: 'd' },
            statusIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaBatchJobStatus_1.KalturaBatchJobStatus, subType: 'KalturaBatchJobStatus' },
            bulkUploadObjectTypeEqual: { type: 'es', subTypeConstructor: KalturaBulkUploadObjectType_1.KalturaBulkUploadObjectType, subType: 'KalturaBulkUploadObjectType' },
            bulkUploadObjectTypeIn: { type: 's' }
        });
        return result;
    };
    return KalturaBulkUploadBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaBulkUploadBaseFilter = KalturaBulkUploadBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBulkUploadBaseFilter', KalturaBulkUploadBaseFilter);
