// kaltura-node-typescript-client/api/types/KalturaDistributionProfileBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDistributionProfileBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDistributionProfileStatus_1 = require("./KalturaDistributionProfileStatus");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaDistributionProfileBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDistributionProfileBaseFilter, _super);
    function KalturaDistributionProfileBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaDistributionProfileBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDistributionProfileBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaDistributionProfileStatus_1.KalturaDistributionProfileStatus, subType: 'KalturaDistributionProfileStatus' },
            statusIn: { type: 's' }
        });
        return result;
    };
    return KalturaDistributionProfileBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaDistributionProfileBaseFilter = KalturaDistributionProfileBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDistributionProfileBaseFilter', KalturaDistributionProfileBaseFilter);
