// kaltura-node-typescript-client/api/types/KalturaThumbParamsOutput.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaThumbParamsOutput = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaThumbParams_1 = require("./KalturaThumbParams");
var KalturaThumbParamsOutput = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaThumbParamsOutput, _super);
    function KalturaThumbParamsOutput(data) {
        return _super.call(this, data) || this;
    }
    KalturaThumbParamsOutput.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaThumbParamsOutput' },
            thumbParamsId: { type: 'n' },
            thumbParamsVersion: { type: 's' },
            thumbAssetId: { type: 's' },
            thumbAssetVersion: { type: 's' },
            rotate: { type: 'n' }
        });
        return result;
    };
    return KalturaThumbParamsOutput;
}(KalturaThumbParams_1.KalturaThumbParams));
exports.KalturaThumbParamsOutput = KalturaThumbParamsOutput;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaThumbParamsOutput', KalturaThumbParamsOutput);
