// kaltura-node-typescript-client/api/types/KalturaWidevineProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaWidevineProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDrmProfile_1 = require("./KalturaDrmProfile");
var KalturaWidevineProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaWidevineProfile, _super);
    function KalturaWidevineProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaWidevineProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaWidevineProfile' },
            key: { type: 's' },
            iv: { type: 's' },
            owner: { type: 's' },
            portal: { type: 's' },
            maxGop: { type: 'n' },
            regServerHost: { type: 's' }
        });
        return result;
    };
    return KalturaWidevineProfile;
}(KalturaDrmProfile_1.KalturaDrmProfile));
exports.KalturaWidevineProfile = KalturaWidevineProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaWidevineProfile', KalturaWidevineProfile);
