// kaltura-node-typescript-client/api/types/KalturaLiveStreamAdminEntryMatchAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveStreamAdminEntryMatchAttribute = void 0;
var KalturaLiveStreamAdminEntryMatchAttribute;
(function (KalturaLiveStreamAdminEntryMatchAttribute) {
    KalturaLiveStreamAdminEntryMatchAttribute["adminTags"] = "adminTags";
    KalturaLiveStreamAdminEntryMatchAttribute["categoriesIds"] = "categoriesIds";
    KalturaLiveStreamAdminEntryMatchAttribute["creatorId"] = "creatorId";
    KalturaLiveStreamAdminEntryMatchAttribute["description"] = "description";
    KalturaLiveStreamAdminEntryMatchAttribute["durationType"] = "durationType";
    KalturaLiveStreamAdminEntryMatchAttribute["flavorParamsIds"] = "flavorParamsIds";
    KalturaLiveStreamAdminEntryMatchAttribute["groupId"] = "groupId";
    KalturaLiveStreamAdminEntryMatchAttribute["id"] = "id";
    KalturaLiveStreamAdminEntryMatchAttribute["_name"] = "name";
    KalturaLiveStreamAdminEntryMatchAttribute["referenceId"] = "referenceId";
    KalturaLiveStreamAdminEntryMatchAttribute["replacedEntryId"] = "replacedEntryId";
    KalturaLiveStreamAdminEntryMatchAttribute["replacingEntryId"] = "replacingEntryId";
    KalturaLiveStreamAdminEntryMatchAttribute["searchText"] = "searchText";
    KalturaLiveStreamAdminEntryMatchAttribute["tags"] = "tags";
    KalturaLiveStreamAdminEntryMatchAttribute["userId"] = "userId";
})(KalturaLiveStreamAdminEntryMatchAttribute = exports.KalturaLiveStreamAdminEntryMatchAttribute || (exports.KalturaLiveStreamAdminEntryMatchAttribute = {}));
