// kaltura-node-typescript-client/api/types/KalturaMediaInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMediaInfo = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBitRateMode_1 = require("./KalturaBitRateMode");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaMediaInfo = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMediaInfo, _super);
    function KalturaMediaInfo(data) {
        return _super.call(this, data) || this;
    }
    KalturaMediaInfo.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMediaInfo' },
            id: { type: 'n', readOnly: true },
            flavorAssetId: { type: 's' },
            fileSize: { type: 'n' },
            containerFormat: { type: 's' },
            containerId: { type: 's' },
            containerProfile: { type: 's' },
            containerDuration: { type: 'n' },
            containerBitRate: { type: 'n' },
            videoFormat: { type: 's' },
            videoCodecId: { type: 's' },
            videoDuration: { type: 'n' },
            videoBitRate: { type: 'n' },
            videoBitRateMode: { type: 'en', subTypeConstructor: KalturaBitRateMode_1.KalturaBitRateMode, subType: 'KalturaBitRateMode' },
            videoWidth: { type: 'n' },
            videoHeight: { type: 'n' },
            videoFrameRate: { type: 'n' },
            videoDar: { type: 'n' },
            videoRotation: { type: 'n' },
            audioFormat: { type: 's' },
            audioCodecId: { type: 's' },
            audioDuration: { type: 'n' },
            audioBitRate: { type: 'n' },
            audioBitRateMode: { type: 'en', subTypeConstructor: KalturaBitRateMode_1.KalturaBitRateMode, subType: 'KalturaBitRateMode' },
            audioChannels: { type: 'n' },
            audioSamplingRate: { type: 'n' },
            audioResolution: { type: 'n' },
            writingLib: { type: 's' },
            rawData: { type: 's' },
            multiStreamInfo: { type: 's' },
            scanType: { type: 'n' },
            multiStream: { type: 's' },
            isFastStart: { type: 'n' },
            contentStreams: { type: 's' },
            complexityValue: { type: 'n' },
            maxGOP: { type: 'n' },
            matrixCoefficients: { type: 's' },
            colorTransfer: { type: 's' },
            colorPrimaries: { type: 's' },
            pixelFormat: { type: 's' },
            colorSpace: { type: 's' },
            chromaSubsampling: { type: 's' },
            bitsDepth: { type: 'n' }
        });
        return result;
    };
    return KalturaMediaInfo;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaMediaInfo = KalturaMediaInfo;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMediaInfo', KalturaMediaInfo);
