// kaltura-node-typescript-client/api/types/KalturaDropFolderFile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDropFolderFile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDropFolderFileStatus_1 = require("./KalturaDropFolderFileStatus");
var KalturaDropFolderType_1 = require("./KalturaDropFolderType");
var KalturaDropFolderFileErrorCode_1 = require("./KalturaDropFolderFileErrorCode");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaDropFolderFile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDropFolderFile, _super);
    function KalturaDropFolderFile(data) {
        return _super.call(this, data) || this;
    }
    KalturaDropFolderFile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDropFolderFile' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            dropFolderId: { type: 'n' },
            fileName: { type: 's' },
            fileSize: { type: 'n' },
            fileSizeLastSetAt: { type: 'd', readOnly: true },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaDropFolderFileStatus_1.KalturaDropFolderFileStatus, subType: 'KalturaDropFolderFileStatus' },
            type: { type: 'es', readOnly: true, subTypeConstructor: KalturaDropFolderType_1.KalturaDropFolderType, subType: 'KalturaDropFolderType' },
            parsedSlug: { type: 's' },
            parsedFlavor: { type: 's' },
            parsedUserId: { type: 's' },
            leadDropFolderFileId: { type: 'n' },
            deletedDropFolderFileId: { type: 'n' },
            entryId: { type: 's' },
            errorCode: { type: 'es', subTypeConstructor: KalturaDropFolderFileErrorCode_1.KalturaDropFolderFileErrorCode, subType: 'KalturaDropFolderFileErrorCode' },
            errorDescription: { type: 's' },
            lastModificationTime: { type: 's' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            uploadStartDetectedAt: { type: 'd' },
            uploadEndDetectedAt: { type: 'd' },
            importStartedAt: { type: 'd' },
            importEndedAt: { type: 'd' },
            batchJobId: { type: 'n', readOnly: true }
        });
        return result;
    };
    return KalturaDropFolderFile;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaDropFolderFile = KalturaDropFolderFile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDropFolderFile', KalturaDropFolderFile);
