// kaltura-node-typescript-client/api/types/KalturaDistributionProvider.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDistributionProvider = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDistributionProviderType_1 = require("./KalturaDistributionProviderType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaDistributionProvider = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDistributionProvider, _super);
    function KalturaDistributionProvider(data) {
        return _super.call(this, data) || this;
    }
    KalturaDistributionProvider.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDistributionProvider' },
            type: { type: 'es', readOnly: true, subTypeConstructor: KalturaDistributionProviderType_1.KalturaDistributionProviderType, subType: 'KalturaDistributionProviderType' },
            name: { type: 's' },
            scheduleUpdateEnabled: { type: 'b' },
            availabilityUpdateEnabled: { type: 'b' },
            deleteInsteadUpdate: { type: 'b' },
            intervalBeforeSunrise: { type: 'n' },
            intervalBeforeSunset: { type: 'n' },
            updateRequiredEntryFields: { type: 's' },
            updateRequiredMetadataXPaths: { type: 's' }
        });
        return result;
    };
    return KalturaDistributionProvider;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaDistributionProvider = KalturaDistributionProvider;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDistributionProvider', KalturaDistributionProvider);
