// kaltura-node-typescript-client/api/types/KalturaFlavorAssetBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFlavorAssetBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaFlavorAssetStatus_1 = require("./KalturaFlavorAssetStatus");
var KalturaAssetFilter_1 = require("./KalturaAssetFilter");
var KalturaFlavorAssetBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaFlavorAssetBaseFilter, _super);
    function KalturaFlavorAssetBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaFlavorAssetBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaFlavorAssetBaseFilter' },
            flavorParamsIdEqual: { type: 'n' },
            flavorParamsIdIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaFlavorAssetStatus_1.KalturaFlavorAssetStatus, subType: 'KalturaFlavorAssetStatus' },
            statusIn: { type: 's' },
            statusNotIn: { type: 's' }
        });
        return result;
    };
    return KalturaFlavorAssetBaseFilter;
}(KalturaAssetFilter_1.KalturaAssetFilter));
exports.KalturaFlavorAssetBaseFilter = KalturaFlavorAssetBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaFlavorAssetBaseFilter', KalturaFlavorAssetBaseFilter);
