// kaltura-node-typescript-client/api/types/KalturaAccessControlProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAccessControlProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaRule_1 = require("./KalturaRule");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaAccessControlProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAccessControlProfile, _super);
    function KalturaAccessControlProfile(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.rules === 'undefined')
            _this.rules = [];
        return _this;
    }
    KalturaAccessControlProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAccessControlProfile' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            name: { type: 's' },
            systemName: { type: 's' },
            description: { type: 's' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            isDefault: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            rules: { type: 'a', subTypeConstructor: KalturaRule_1.KalturaRule, subType: 'KalturaRule' }
        });
        return result;
    };
    return KalturaAccessControlProfile;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaAccessControlProfile = KalturaAccessControlProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAccessControlProfile', KalturaAccessControlProfile);
