// kaltura-node-typescript-client/api/types/KalturaDoubleClickDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDoubleClickDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaDoubleClickDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDoubleClickDistributionProfile, _super);
    function KalturaDoubleClickDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaDoubleClickDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDoubleClickDistributionProfile' },
            channelTitle: { type: 's' },
            channelLink: { type: 's' },
            channelDescription: { type: 's' },
            feedUrl: { type: 's', readOnly: true },
            cuePointsProvider: { type: 's' },
            itemsPerPage: { type: 's' },
            ignoreSchedulingInFeed: { type: 'b' }
        });
        return result;
    };
    return KalturaDoubleClickDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaDoubleClickDistributionProfile = KalturaDoubleClickDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDoubleClickDistributionProfile', KalturaDoubleClickDistributionProfile);
