// kaltura-node-typescript-client/api/types/KalturaQuestionCuePointOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaQuestionCuePointOrderBy = void 0;
var KalturaQuestionCuePointOrderBy;
(function (KalturaQuestionCuePointOrderBy) {
    KalturaQuestionCuePointOrderBy["createdAtAsc"] = "+createdAt";
    KalturaQuestionCuePointOrderBy["createdAtDesc"] = "-createdAt";
    KalturaQuestionCuePointOrderBy["intIdAsc"] = "+intId";
    KalturaQuestionCuePointOrderBy["intIdDesc"] = "-intId";
    KalturaQuestionCuePointOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaQuestionCuePointOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaQuestionCuePointOrderBy["startTimeAsc"] = "+startTime";
    KalturaQuestionCuePointOrderBy["startTimeDesc"] = "-startTime";
    KalturaQuestionCuePointOrderBy["triggeredAtAsc"] = "+triggeredAt";
    KalturaQuestionCuePointOrderBy["triggeredAtDesc"] = "-triggeredAt";
    KalturaQuestionCuePointOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaQuestionCuePointOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaQuestionCuePointOrderBy = exports.KalturaQuestionCuePointOrderBy || (exports.KalturaQuestionCuePointOrderBy = {}));
