// kaltura-node-typescript-client/api/types/KalturaFlavorAssetStatus.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFlavorAssetStatus = void 0;
var KalturaFlavorAssetStatus;
(function (KalturaFlavorAssetStatus) {
    KalturaFlavorAssetStatus[KalturaFlavorAssetStatus["converting"] = 1] = "converting";
    KalturaFlavorAssetStatus[KalturaFlavorAssetStatus["deleted"] = 3] = "deleted";
    KalturaFlavorAssetStatus[KalturaFlavorAssetStatus["error"] = -1] = "error";
    KalturaFlavorAssetStatus[KalturaFlavorAssetStatus["exporting"] = 9] = "exporting";
    KalturaFlavorAssetStatus[KalturaFlavorAssetStatus["importing"] = 7] = "importing";
    KalturaFlavorAssetStatus[KalturaFlavorAssetStatus["notApplicable"] = 4] = "notApplicable";
    KalturaFlavorAssetStatus[KalturaFlavorAssetStatus["queued"] = 0] = "queued";
    KalturaFlavorAssetStatus[KalturaFlavorAssetStatus["ready"] = 2] = "ready";
    KalturaFlavorAssetStatus[KalturaFlavorAssetStatus["temp"] = 5] = "temp";
    KalturaFlavorAssetStatus[KalturaFlavorAssetStatus["validating"] = 8] = "validating";
    KalturaFlavorAssetStatus[KalturaFlavorAssetStatus["waitForConvert"] = 6] = "waitForConvert";
})(KalturaFlavorAssetStatus = exports.KalturaFlavorAssetStatus || (exports.KalturaFlavorAssetStatus = {}));
