// kaltura-node-typescript-client/api/types/KalturaDocumentEntryBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDocumentEntryBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDocumentType_1 = require("./KalturaDocumentType");
var KalturaBaseEntryFilter_1 = require("./KalturaBaseEntryFilter");
var KalturaDocumentEntryBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDocumentEntryBaseFilter, _super);
    function KalturaDocumentEntryBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaDocumentEntryBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDocumentEntryBaseFilter' },
            documentTypeEqual: { type: 'en', subTypeConstructor: KalturaDocumentType_1.KalturaDocumentType, subType: 'KalturaDocumentType' },
            documentTypeIn: { type: 's' },
            assetParamsIdsMatchOr: { type: 's' },
            assetParamsIdsMatchAnd: { type: 's' }
        });
        return result;
    };
    return KalturaDocumentEntryBaseFilter;
}(KalturaBaseEntryFilter_1.KalturaBaseEntryFilter));
exports.KalturaDocumentEntryBaseFilter = KalturaDocumentEntryBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDocumentEntryBaseFilter', KalturaDocumentEntryBaseFilter);
