// kaltura-node-typescript-client/api/types/KalturaPlayReadyCopyEnablerType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPlayReadyCopyEnablerType = void 0;
var KalturaPlayReadyCopyEnablerType;
(function (KalturaPlayReadyCopyEnablerType) {
    KalturaPlayReadyCopyEnablerType["aacs"] = "C3CF56E0-7FF2-4491-809F-53E21D3ABF07";
    KalturaPlayReadyCopyEnablerType["clipboard"] = "6E76C588-C3A9-47ea-A875-546D5209FF38";
    KalturaPlayReadyCopyEnablerType["cprm"] = "CDD801AD-A577-48DB-950E-46D5F1592FAE";
    KalturaPlayReadyCopyEnablerType["css"] = "3CAF2814-A7AB-467C-B4DF-54ACC56C66DC";
    KalturaPlayReadyCopyEnablerType["device"] = "6848955D-516B-4EB0-90E8-8F6D5A77B85F";
    KalturaPlayReadyCopyEnablerType["helix"] = "CCB0B4E3-8B46-409e-A998-82556E3F5AF4";
    KalturaPlayReadyCopyEnablerType["orangeBookCd"] = "EC930B7D-1F2D-4682-A38B-8AB977721D0D";
    KalturaPlayReadyCopyEnablerType["pc"] = "CE480EDE-516B-40B3-90E1-D6CFC47630C5";
    KalturaPlayReadyCopyEnablerType["printer"] = "3CF2E054-F4D5-46cd-85A6-FCD152AD5FBE";
    KalturaPlayReadyCopyEnablerType["sdc"] = "79F78A0D-0B69-401e-8A90-8BEF30BCE192";
    KalturaPlayReadyCopyEnablerType["sdcLimited"] = "E6785609-64CC-4bfa-B82D-6B619733B746";
    KalturaPlayReadyCopyEnablerType["sdcPreview"] = "81BD9AD4-A720-4ea1-B510-5D4E6FFB6A4D";
})(KalturaPlayReadyCopyEnablerType = exports.KalturaPlayReadyCopyEnablerType || (exports.KalturaPlayReadyCopyEnablerType = {}));
