// kaltura-node-typescript-client/api/types/KalturaBusinessProcessServerBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBusinessProcessServerBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBusinessProcessServerStatus_1 = require("./KalturaBusinessProcessServerStatus");
var KalturaBusinessProcessProvider_1 = require("./KalturaBusinessProcessProvider");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaBusinessProcessServerBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBusinessProcessServerBaseFilter, _super);
    function KalturaBusinessProcessServerBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaBusinessProcessServerBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBusinessProcessServerBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            idNotIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            statusEqual: { type: 'es', subTypeConstructor: KalturaBusinessProcessServerStatus_1.KalturaBusinessProcessServerStatus, subType: 'KalturaBusinessProcessServerStatus' },
            statusNotEqual: { type: 'es', subTypeConstructor: KalturaBusinessProcessServerStatus_1.KalturaBusinessProcessServerStatus, subType: 'KalturaBusinessProcessServerStatus' },
            statusIn: { type: 's' },
            statusNotIn: { type: 's' },
            typeEqual: { type: 'es', subTypeConstructor: KalturaBusinessProcessProvider_1.KalturaBusinessProcessProvider, subType: 'KalturaBusinessProcessProvider' },
            typeIn: { type: 's' },
            dcEqual: { type: 'n' },
            dcEqOrNull: { type: 'n' }
        });
        return result;
    };
    return KalturaBusinessProcessServerBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaBusinessProcessServerBaseFilter = KalturaBusinessProcessServerBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBusinessProcessServerBaseFilter', KalturaBusinessProcessServerBaseFilter);
