// kaltura-node-typescript-client/api/types/KalturaDataEntryMatchAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDataEntryMatchAttribute = void 0;
var KalturaDataEntryMatchAttribute;
(function (KalturaDataEntryMatchAttribute) {
    KalturaDataEntryMatchAttribute["adminTags"] = "adminTags";
    KalturaDataEntryMatchAttribute["categoriesIds"] = "categoriesIds";
    KalturaDataEntryMatchAttribute["creatorId"] = "creatorId";
    KalturaDataEntryMatchAttribute["description"] = "description";
    KalturaDataEntryMatchAttribute["groupId"] = "groupId";
    KalturaDataEntryMatchAttribute["id"] = "id";
    KalturaDataEntryMatchAttribute["_name"] = "name";
    KalturaDataEntryMatchAttribute["referenceId"] = "referenceId";
    KalturaDataEntryMatchAttribute["replacedEntryId"] = "replacedEntryId";
    KalturaDataEntryMatchAttribute["replacingEntryId"] = "replacingEntryId";
    KalturaDataEntryMatchAttribute["searchText"] = "searchText";
    KalturaDataEntryMatchAttribute["tags"] = "tags";
    KalturaDataEntryMatchAttribute["userId"] = "userId";
})(KalturaDataEntryMatchAttribute = exports.KalturaDataEntryMatchAttribute || (exports.KalturaDataEntryMatchAttribute = {}));
