// kaltura-node-typescript-client/api/types/KalturaCategoryOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCategoryOrderBy = void 0;
var KalturaCategoryOrderBy;
(function (KalturaCategoryOrderBy) {
    KalturaCategoryOrderBy["createdAtAsc"] = "+createdAt";
    KalturaCategoryOrderBy["createdAtDesc"] = "-createdAt";
    KalturaCategoryOrderBy["depthAsc"] = "+depth";
    KalturaCategoryOrderBy["depthDesc"] = "-depth";
    KalturaCategoryOrderBy["directEntriesCountAsc"] = "+directEntriesCount";
    KalturaCategoryOrderBy["directEntriesCountDesc"] = "-directEntriesCount";
    KalturaCategoryOrderBy["directSubCategoriesCountAsc"] = "+directSubCategoriesCount";
    KalturaCategoryOrderBy["directSubCategoriesCountDesc"] = "-directSubCategoriesCount";
    KalturaCategoryOrderBy["entriesCountAsc"] = "+entriesCount";
    KalturaCategoryOrderBy["entriesCountDesc"] = "-entriesCount";
    KalturaCategoryOrderBy["fullNameAsc"] = "+fullName";
    KalturaCategoryOrderBy["fullNameDesc"] = "-fullName";
    KalturaCategoryOrderBy["membersCountAsc"] = "+membersCount";
    KalturaCategoryOrderBy["membersCountDesc"] = "-membersCount";
    KalturaCategoryOrderBy["nameAsc"] = "+name";
    KalturaCategoryOrderBy["nameDesc"] = "-name";
    KalturaCategoryOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaCategoryOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaCategoryOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaCategoryOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaCategoryOrderBy = exports.KalturaCategoryOrderBy || (exports.KalturaCategoryOrderBy = {}));
