// kaltura-node-typescript-client/api/types/KalturaMailJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMailJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaMailType_1 = require("./KalturaMailType");
var KalturaMailJobStatus_1 = require("./KalturaMailJobStatus");
var KalturaDynamicEmailContents_1 = require("./KalturaDynamicEmailContents");
var KalturaLanguageCode_1 = require("./KalturaLanguageCode");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaMailJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMailJobData, _super);
    function KalturaMailJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaMailJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMailJobData' },
            mailType: { type: 'es', subTypeConstructor: KalturaMailType_1.KalturaMailType, subType: 'KalturaMailType' },
            mailPriority: { type: 'n' },
            status: { type: 'en', subTypeConstructor: KalturaMailJobStatus_1.KalturaMailJobStatus, subType: 'KalturaMailJobStatus' },
            recipientName: { type: 's' },
            recipientEmail: { type: 's' },
            recipientId: { type: 'n' },
            fromName: { type: 's' },
            fromEmail: { type: 's' },
            bodyParams: { type: 's' },
            subjectParams: { type: 's' },
            dynamicEmailContents: { type: 'o', subTypeConstructor: KalturaDynamicEmailContents_1.KalturaDynamicEmailContents, subType: 'KalturaDynamicEmailContents' },
            templatePath: { type: 's' },
            language: { type: 'es', subTypeConstructor: KalturaLanguageCode_1.KalturaLanguageCode, subType: 'KalturaLanguageCode' },
            campaignId: { type: 'n' },
            minSendDate: { type: 'n' },
            isHtml: { type: 'b' },
            separator: { type: 's' }
        });
        return result;
    };
    return KalturaMailJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaMailJobData = KalturaMailJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMailJobData', KalturaMailJobData);
