// kaltura-node-typescript-client/api/types/KalturaAttUverseDistributionFile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAttUverseDistributionFile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaAssetType_1 = require("./KalturaAssetType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaAttUverseDistributionFile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAttUverseDistributionFile, _super);
    function KalturaAttUverseDistributionFile(data) {
        return _super.call(this, data) || this;
    }
    KalturaAttUverseDistributionFile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAttUverseDistributionFile' },
            remoteFilename: { type: 's' },
            localFilePath: { type: 's' },
            assetType: { type: 'es', subTypeConstructor: KalturaAssetType_1.KalturaAssetType, subType: 'KalturaAssetType' },
            assetId: { type: 's' }
        });
        return result;
    };
    return KalturaAttUverseDistributionFile;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaAttUverseDistributionFile = KalturaAttUverseDistributionFile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAttUverseDistributionFile', KalturaAttUverseDistributionFile);
