// kaltura-node-typescript-client/api/types/KalturaScheduleEventResourceBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaScheduleEventResourceBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaScheduleEventResourceBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaScheduleEventResourceBaseFilter, _super);
    function KalturaScheduleEventResourceBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaScheduleEventResourceBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaScheduleEventResourceBaseFilter' },
            eventIdEqual: { type: 'n' },
            eventIdIn: { type: 's' },
            resourceIdEqual: { type: 'n' },
            resourceIdIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' }
        });
        return result;
    };
    return KalturaScheduleEventResourceBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaScheduleEventResourceBaseFilter = KalturaScheduleEventResourceBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaScheduleEventResourceBaseFilter', KalturaScheduleEventResourceBaseFilter);
