// kaltura-node-typescript-client/api/types/KalturaCategoryUser.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCategoryUser = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCategoryUserPermissionLevel_1 = require("./KalturaCategoryUserPermissionLevel");
var KalturaCategoryUserStatus_1 = require("./KalturaCategoryUserStatus");
var KalturaUpdateMethodType_1 = require("./KalturaUpdateMethodType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaCategoryUser = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCategoryUser, _super);
    function KalturaCategoryUser(data) {
        return _super.call(this, data) || this;
    }
    KalturaCategoryUser.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCategoryUser' },
            categoryId: { type: 'n' },
            userId: { type: 's' },
            partnerId: { type: 'n', readOnly: true },
            permissionLevel: { type: 'en', subTypeConstructor: KalturaCategoryUserPermissionLevel_1.KalturaCategoryUserPermissionLevel, subType: 'KalturaCategoryUserPermissionLevel' },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaCategoryUserStatus_1.KalturaCategoryUserStatus, subType: 'KalturaCategoryUserStatus' },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            updateMethod: { type: 'en', subTypeConstructor: KalturaUpdateMethodType_1.KalturaUpdateMethodType, subType: 'KalturaUpdateMethodType' },
            categoryFullIds: { type: 's', readOnly: true },
            permissionNames: { type: 's' }
        });
        return result;
    };
    return KalturaCategoryUser;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaCategoryUser = KalturaCategoryUser;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCategoryUser', KalturaCategoryUser);
