// kaltura-node-typescript-client/api/types/KalturaMixEntryOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMixEntryOrderBy = void 0;
var KalturaMixEntryOrderBy;
(function (KalturaMixEntryOrderBy) {
    KalturaMixEntryOrderBy["createdAtAsc"] = "+createdAt";
    KalturaMixEntryOrderBy["createdAtDesc"] = "-createdAt";
    KalturaMixEntryOrderBy["durationAsc"] = "+duration";
    KalturaMixEntryOrderBy["durationDesc"] = "-duration";
    KalturaMixEntryOrderBy["endDateAsc"] = "+endDate";
    KalturaMixEntryOrderBy["endDateDesc"] = "-endDate";
    KalturaMixEntryOrderBy["lastPlayedAtAsc"] = "+lastPlayedAt";
    KalturaMixEntryOrderBy["lastPlayedAtDesc"] = "-lastPlayedAt";
    KalturaMixEntryOrderBy["moderationCountAsc"] = "+moderationCount";
    KalturaMixEntryOrderBy["moderationCountDesc"] = "-moderationCount";
    KalturaMixEntryOrderBy["nameAsc"] = "+name";
    KalturaMixEntryOrderBy["nameDesc"] = "-name";
    KalturaMixEntryOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaMixEntryOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaMixEntryOrderBy["playsAsc"] = "+plays";
    KalturaMixEntryOrderBy["playsDesc"] = "-plays";
    KalturaMixEntryOrderBy["rankAsc"] = "+rank";
    KalturaMixEntryOrderBy["rankDesc"] = "-rank";
    KalturaMixEntryOrderBy["recentAsc"] = "+recent";
    KalturaMixEntryOrderBy["recentDesc"] = "-recent";
    KalturaMixEntryOrderBy["startDateAsc"] = "+startDate";
    KalturaMixEntryOrderBy["startDateDesc"] = "-startDate";
    KalturaMixEntryOrderBy["totalRankAsc"] = "+totalRank";
    KalturaMixEntryOrderBy["totalRankDesc"] = "-totalRank";
    KalturaMixEntryOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaMixEntryOrderBy["updatedAtDesc"] = "-updatedAt";
    KalturaMixEntryOrderBy["viewsAsc"] = "+views";
    KalturaMixEntryOrderBy["viewsDesc"] = "-views";
    KalturaMixEntryOrderBy["weightAsc"] = "+weight";
    KalturaMixEntryOrderBy["weightDesc"] = "-weight";
})(KalturaMixEntryOrderBy = exports.KalturaMixEntryOrderBy || (exports.KalturaMixEntryOrderBy = {}));
