// kaltura-node-typescript-client/api/types/KalturaReportInputBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaReportInputBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaReportInputBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaReportInputBaseFilter, _super);
    function KalturaReportInputBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaReportInputBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaReportInputBaseFilter' },
            fromDate: { type: 'd' },
            toDate: { type: 'd' },
            fromDay: { type: 's' },
            toDay: { type: 's' }
        });
        return result;
    };
    return KalturaReportInputBaseFilter;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaReportInputBaseFilter = KalturaReportInputBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaReportInputBaseFilter', KalturaReportInputBaseFilter);
