// kaltura-node-typescript-client/api/types/KalturaQuiz.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaQuiz = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaKeyValue_1 = require("./KalturaKeyValue");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaScoreType_1 = require("./KalturaScoreType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaQuiz = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaQuiz, _super);
    function KalturaQuiz(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.uiAttributes === 'undefined')
            _this.uiAttributes = [];
        return _this;
    }
    KalturaQuiz.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaQuiz' },
            version: { type: 'n', readOnly: true },
            uiAttributes: { type: 'a', subTypeConstructor: KalturaKeyValue_1.KalturaKeyValue, subType: 'KalturaKeyValue' },
            showResultOnAnswer: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            showCorrectKeyOnAnswer: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            allowAnswerUpdate: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            showCorrectAfterSubmission: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            allowDownload: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            showGradeAfterSubmission: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            attemptsAllowed: { type: 'n' },
            scoreType: { type: 'en', subTypeConstructor: KalturaScoreType_1.KalturaScoreType, subType: 'KalturaScoreType' }
        });
        return result;
    };
    return KalturaQuiz;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaQuiz = KalturaQuiz;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaQuiz', KalturaQuiz);
