// kaltura-node-typescript-client/api/types/KalturaGroupUser.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaGroupUser = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaGroupUserStatus_1 = require("./KalturaGroupUserStatus");
var KalturaGroupUserCreationMode_1 = require("./KalturaGroupUserCreationMode");
var KalturaGroupUserRole_1 = require("./KalturaGroupUserRole");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaGroupUser = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaGroupUser, _super);
    function KalturaGroupUser(data) {
        return _super.call(this, data) || this;
    }
    KalturaGroupUser.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaGroupUser' },
            id: { type: 's', readOnly: true },
            userId: { type: 's' },
            groupId: { type: 's' },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaGroupUserStatus_1.KalturaGroupUserStatus, subType: 'KalturaGroupUserStatus' },
            partnerId: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            creationMode: { type: 'en', subTypeConstructor: KalturaGroupUserCreationMode_1.KalturaGroupUserCreationMode, subType: 'KalturaGroupUserCreationMode' },
            userRole: { type: 'en', subTypeConstructor: KalturaGroupUserRole_1.KalturaGroupUserRole, subType: 'KalturaGroupUserRole' }
        });
        return result;
    };
    return KalturaGroupUser;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaGroupUser = KalturaGroupUser;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaGroupUser', KalturaGroupUser);
