// kaltura-node-typescript-client/api/types/KalturaBulkUploadResultUser.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBulkUploadResultUser = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBulkUploadResult_1 = require("./KalturaBulkUploadResult");
var KalturaBulkUploadResultUser = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBulkUploadResultUser, _super);
    function KalturaBulkUploadResultUser(data) {
        return _super.call(this, data) || this;
    }
    KalturaBulkUploadResultUser.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBulkUploadResultUser' },
            userId: { type: 's' },
            screenName: { type: 's' },
            email: { type: 's' },
            description: { type: 's' },
            tags: { type: 's' },
            dateOfBirth: { type: 'n' },
            country: { type: 's' },
            state: { type: 's' },
            city: { type: 's' },
            zip: { type: 's' },
            gender: { type: 'n' },
            firstName: { type: 's' },
            lastName: { type: 's' },
            group: { type: 's' },
            externalId: { type: 's' },
            capabilities: { type: 's' }
        });
        return result;
    };
    return KalturaBulkUploadResultUser;
}(KalturaBulkUploadResult_1.KalturaBulkUploadResult));
exports.KalturaBulkUploadResultUser = KalturaBulkUploadResultUser;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBulkUploadResultUser', KalturaBulkUploadResultUser);
