// kaltura-node-typescript-client/api/types/KalturaEndUserReportInputFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEndUserReportInputFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaReportInputFilter_1 = require("./KalturaReportInputFilter");
var KalturaEndUserReportInputFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaEndUserReportInputFilter, _super);
    function KalturaEndUserReportInputFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaEndUserReportInputFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaEndUserReportInputFilter' },
            application: { type: 's' },
            userIds: { type: 's' },
            playbackContext: { type: 's' },
            ancestorPlaybackContext: { type: 's' }
        });
        return result;
    };
    return KalturaEndUserReportInputFilter;
}(KalturaReportInputFilter_1.KalturaReportInputFilter));
exports.KalturaEndUserReportInputFilter = KalturaEndUserReportInputFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaEndUserReportInputFilter', KalturaEndUserReportInputFilter);
