// kaltura-node-typescript-client/api/types/KalturaESearchEntryAbstractNestedItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaESearchEntryAbstractNestedItem = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaESearchItemType_1 = require("./KalturaESearchItemType");
var KalturaESearchRange_1 = require("./KalturaESearchRange");
var KalturaESearchEntryNestedBaseItem_1 = require("./KalturaESearchEntryNestedBaseItem");
var KalturaESearchEntryAbstractNestedItem = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaESearchEntryAbstractNestedItem, _super);
    function KalturaESearchEntryAbstractNestedItem(data) {
        return _super.call(this, data) || this;
    }
    KalturaESearchEntryAbstractNestedItem.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaESearchEntryAbstractNestedItem' },
            searchTerm: { type: 's' },
            itemType: { type: 'en', subTypeConstructor: KalturaESearchItemType_1.KalturaESearchItemType, subType: 'KalturaESearchItemType' },
            range: { type: 'o', subTypeConstructor: KalturaESearchRange_1.KalturaESearchRange, subType: 'KalturaESearchRange' },
            addHighlight: { type: 'b' }
        });
        return result;
    };
    return KalturaESearchEntryAbstractNestedItem;
}(KalturaESearchEntryNestedBaseItem_1.KalturaESearchEntryNestedBaseItem));
exports.KalturaESearchEntryAbstractNestedItem = KalturaESearchEntryAbstractNestedItem;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaESearchEntryAbstractNestedItem', KalturaESearchEntryAbstractNestedItem);
