// kaltura-node-typescript-client/api/types/KalturaAmazonS3StorageProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAmazonS3StorageProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaAmazonS3StorageProfileFilesPermissionLevel_1 = require("./KalturaAmazonS3StorageProfileFilesPermissionLevel");
var KalturaStorageProfile_1 = require("./KalturaStorageProfile");
var KalturaAmazonS3StorageProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAmazonS3StorageProfile, _super);
    function KalturaAmazonS3StorageProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaAmazonS3StorageProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAmazonS3StorageProfile' },
            filesPermissionInS3: { type: 'es', subTypeConstructor: KalturaAmazonS3StorageProfileFilesPermissionLevel_1.KalturaAmazonS3StorageProfileFilesPermissionLevel, subType: 'KalturaAmazonS3StorageProfileFilesPermissionLevel' },
            s3Region: { type: 's' },
            sseType: { type: 's' },
            sseKmsKeyId: { type: 's' },
            signatureType: { type: 's' },
            endPoint: { type: 's' }
        });
        return result;
    };
    return KalturaAmazonS3StorageProfile;
}(KalturaStorageProfile_1.KalturaStorageProfile));
exports.KalturaAmazonS3StorageProfile = KalturaAmazonS3StorageProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAmazonS3StorageProfile', KalturaAmazonS3StorageProfile);
