// kaltura-node-typescript-client/api/types/KalturaWidevineRepositorySyncJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaWidevineRepositorySyncJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaWidevineRepositorySyncMode_1 = require("./KalturaWidevineRepositorySyncMode");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaWidevineRepositorySyncJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaWidevineRepositorySyncJobData, _super);
    function KalturaWidevineRepositorySyncJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaWidevineRepositorySyncJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaWidevineRepositorySyncJobData' },
            syncMode: { type: 'en', subTypeConstructor: KalturaWidevineRepositorySyncMode_1.KalturaWidevineRepositorySyncMode, subType: 'KalturaWidevineRepositorySyncMode' },
            wvAssetIds: { type: 's' },
            modifiedAttributes: { type: 's' },
            monitorSyncCompletion: { type: 'n' }
        });
        return result;
    };
    return KalturaWidevineRepositorySyncJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaWidevineRepositorySyncJobData = KalturaWidevineRepositorySyncJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaWidevineRepositorySyncJobData', KalturaWidevineRepositorySyncJobData);
