// kaltura-node-typescript-client/api/types/KalturaMicrosoftTeamsIntegrationSetting.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMicrosoftTeamsIntegrationSetting = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaIntegrationSetting_1 = require("./KalturaIntegrationSetting");
var KalturaMicrosoftTeamsIntegrationSetting = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMicrosoftTeamsIntegrationSetting, _super);
    function KalturaMicrosoftTeamsIntegrationSetting(data) {
        return _super.call(this, data) || this;
    }
    KalturaMicrosoftTeamsIntegrationSetting.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMicrosoftTeamsIntegrationSetting' },
            clientSecret: { type: 's' },
            clientId: { type: 's' },
            userMetadataProfileId: { type: 'n' },
            scopes: { type: 's' },
            encryptionKey: { type: 's', readOnly: true }
        });
        return result;
    };
    return KalturaMicrosoftTeamsIntegrationSetting;
}(KalturaIntegrationSetting_1.KalturaIntegrationSetting));
exports.KalturaMicrosoftTeamsIntegrationSetting = KalturaMicrosoftTeamsIntegrationSetting;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMicrosoftTeamsIntegrationSetting', KalturaMicrosoftTeamsIntegrationSetting);
