// kaltura-node-typescript-client/api/types/KalturaModifyEntryObjectTask.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaModifyEntryObjectTask = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaKeyValue_1 = require("./KalturaKeyValue");
var KalturaObjectTask_1 = require("./KalturaObjectTask");
var KalturaModifyEntryObjectTask = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaModifyEntryObjectTask, _super);
    function KalturaModifyEntryObjectTask(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.inputMetadata === 'undefined')
            _this.inputMetadata = [];
        if (typeof _this.outputMetadata === 'undefined')
            _this.outputMetadata = [];
        return _this;
    }
    KalturaModifyEntryObjectTask.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaModifyEntryObjectTask' },
            inputMetadataProfileId: { type: 'n' },
            inputMetadata: { type: 'a', subTypeConstructor: KalturaKeyValue_1.KalturaKeyValue, subType: 'KalturaKeyValue' },
            outputMetadataProfileId: { type: 'n' },
            outputMetadata: { type: 'a', subTypeConstructor: KalturaKeyValue_1.KalturaKeyValue, subType: 'KalturaKeyValue' },
            inputUserId: { type: 's' },
            inputEntitledUsersEdit: { type: 's' },
            inputEntitledUsersPublish: { type: 's' },
            inputEntitledUsersView: { type: 's' },
            resetMediaRepurposingProcess: { type: 'b' }
        });
        return result;
    };
    return KalturaModifyEntryObjectTask;
}(KalturaObjectTask_1.KalturaObjectTask));
exports.KalturaModifyEntryObjectTask = KalturaModifyEntryObjectTask;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaModifyEntryObjectTask', KalturaModifyEntryObjectTask);
