// kaltura-node-typescript-client/api/types/KalturaBaseEntryCloneOptions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBaseEntryCloneOptions = void 0;
var KalturaBaseEntryCloneOptions;
(function (KalturaBaseEntryCloneOptions) {
    KalturaBaseEntryCloneOptions["accessControl"] = "4";
    KalturaBaseEntryCloneOptions["adCuePoints"] = "adCuePoint.AD_CUE_POINTS";
    KalturaBaseEntryCloneOptions["annotationCuePoints"] = "annotation.ANNOTATION_CUE_POINTS";
    KalturaBaseEntryCloneOptions["captions"] = "7";
    KalturaBaseEntryCloneOptions["categories"] = "2";
    KalturaBaseEntryCloneOptions["childEntries"] = "3";
    KalturaBaseEntryCloneOptions["codeCuePoints"] = "codeCuePoint.CODE_CUE_POINTS";
    KalturaBaseEntryCloneOptions["flavors"] = "6";
    KalturaBaseEntryCloneOptions["metadata"] = "5";
    KalturaBaseEntryCloneOptions["sessionCuePoints"] = "sessionCuePoint.SESSION_CUE_POINTS";
    KalturaBaseEntryCloneOptions["thumbCuePoints"] = "thumbCuePoint.THUMB_CUE_POINTS";
    KalturaBaseEntryCloneOptions["users"] = "1";
})(KalturaBaseEntryCloneOptions = exports.KalturaBaseEntryCloneOptions || (exports.KalturaBaseEntryCloneOptions = {}));
