// kaltura-node-typescript-client/api/types/KalturaITunesSyndicationFeedOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaITunesSyndicationFeedOrderBy = void 0;
var KalturaITunesSyndicationFeedOrderBy;
(function (KalturaITunesSyndicationFeedOrderBy) {
    KalturaITunesSyndicationFeedOrderBy["createdAtAsc"] = "+createdAt";
    KalturaITunesSyndicationFeedOrderBy["createdAtDesc"] = "-createdAt";
    KalturaITunesSyndicationFeedOrderBy["nameAsc"] = "+name";
    KalturaITunesSyndicationFeedOrderBy["nameDesc"] = "-name";
    KalturaITunesSyndicationFeedOrderBy["playlistIdAsc"] = "+playlistId";
    KalturaITunesSyndicationFeedOrderBy["playlistIdDesc"] = "-playlistId";
    KalturaITunesSyndicationFeedOrderBy["typeAsc"] = "+type";
    KalturaITunesSyndicationFeedOrderBy["typeDesc"] = "-type";
    KalturaITunesSyndicationFeedOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaITunesSyndicationFeedOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaITunesSyndicationFeedOrderBy = exports.KalturaITunesSyndicationFeedOrderBy || (exports.KalturaITunesSyndicationFeedOrderBy = {}));
