// kaltura-node-typescript-client/api/types/KalturaDocumentEntryOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDocumentEntryOrderBy = void 0;
var KalturaDocumentEntryOrderBy;
(function (KalturaDocumentEntryOrderBy) {
    KalturaDocumentEntryOrderBy["createdAtAsc"] = "+createdAt";
    KalturaDocumentEntryOrderBy["createdAtDesc"] = "-createdAt";
    KalturaDocumentEntryOrderBy["endDateAsc"] = "+endDate";
    KalturaDocumentEntryOrderBy["endDateDesc"] = "-endDate";
    KalturaDocumentEntryOrderBy["moderationCountAsc"] = "+moderationCount";
    KalturaDocumentEntryOrderBy["moderationCountDesc"] = "-moderationCount";
    KalturaDocumentEntryOrderBy["nameAsc"] = "+name";
    KalturaDocumentEntryOrderBy["nameDesc"] = "-name";
    KalturaDocumentEntryOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaDocumentEntryOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaDocumentEntryOrderBy["rankAsc"] = "+rank";
    KalturaDocumentEntryOrderBy["rankDesc"] = "-rank";
    KalturaDocumentEntryOrderBy["recentAsc"] = "+recent";
    KalturaDocumentEntryOrderBy["recentDesc"] = "-recent";
    KalturaDocumentEntryOrderBy["startDateAsc"] = "+startDate";
    KalturaDocumentEntryOrderBy["startDateDesc"] = "-startDate";
    KalturaDocumentEntryOrderBy["totalRankAsc"] = "+totalRank";
    KalturaDocumentEntryOrderBy["totalRankDesc"] = "-totalRank";
    KalturaDocumentEntryOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaDocumentEntryOrderBy["updatedAtDesc"] = "-updatedAt";
    KalturaDocumentEntryOrderBy["weightAsc"] = "+weight";
    KalturaDocumentEntryOrderBy["weightDesc"] = "-weight";
})(KalturaDocumentEntryOrderBy = exports.KalturaDocumentEntryOrderBy || (exports.KalturaDocumentEntryOrderBy = {}));
