// kaltura-node-typescript-client/api/types/KalturaMetadataFieldChangedCondition.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMetadataFieldChangedCondition = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaMatchCondition_1 = require("./KalturaMatchCondition");
var KalturaMetadataFieldChangedCondition = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMetadataFieldChangedCondition, _super);
    function KalturaMetadataFieldChangedCondition(data) {
        return _super.call(this, data) || this;
    }
    KalturaMetadataFieldChangedCondition.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMetadataFieldChangedCondition' },
            xPath: { type: 's' },
            profileId: { type: 'n' },
            profileSystemName: { type: 's' },
            versionA: { type: 's' },
            versionB: { type: 's' }
        });
        return result;
    };
    return KalturaMetadataFieldChangedCondition;
}(KalturaMatchCondition_1.KalturaMatchCondition));
exports.KalturaMetadataFieldChangedCondition = KalturaMetadataFieldChangedCondition;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMetadataFieldChangedCondition', KalturaMetadataFieldChangedCondition);
