// kaltura-node-typescript-client/api/types/KalturaWatchLaterUserEntryAdvancedFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaWatchLaterUserEntryAdvancedFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUserEntryExtendedStatus_1 = require("./KalturaUserEntryExtendedStatus");
var KalturaSearchItem_1 = require("./KalturaSearchItem");
var KalturaWatchLaterUserEntryAdvancedFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaWatchLaterUserEntryAdvancedFilter, _super);
    function KalturaWatchLaterUserEntryAdvancedFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaWatchLaterUserEntryAdvancedFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaWatchLaterUserEntryAdvancedFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            userIdEqual: { type: 's' },
            userIdIn: { type: 's' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            extendedStatusEqual: { type: 'es', subTypeConstructor: KalturaUserEntryExtendedStatus_1.KalturaUserEntryExtendedStatus, subType: 'KalturaUserEntryExtendedStatus' },
            extendedStatusIn: { type: 's' }
        });
        return result;
    };
    return KalturaWatchLaterUserEntryAdvancedFilter;
}(KalturaSearchItem_1.KalturaSearchItem));
exports.KalturaWatchLaterUserEntryAdvancedFilter = KalturaWatchLaterUserEntryAdvancedFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaWatchLaterUserEntryAdvancedFilter', KalturaWatchLaterUserEntryAdvancedFilter);
