// kaltura-node-typescript-client/api/types/KalturaBulkUploadResultVendorCatalogItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBulkUploadResultVendorCatalogItem = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaVendorServiceFeature_1 = require("./KalturaVendorServiceFeature");
var KalturaVendorServiceType_1 = require("./KalturaVendorServiceType");
var KalturaVendorServiceTurnAroundTime_1 = require("./KalturaVendorServiceTurnAroundTime");
var KalturaCatalogItemLanguage_1 = require("./KalturaCatalogItemLanguage");
var KalturaVendorCatalogItemOutputFormat_1 = require("./KalturaVendorCatalogItemOutputFormat");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaVendorCatalogItemPricing_1 = require("./KalturaVendorCatalogItemPricing");
var KalturaBulkUploadResult_1 = require("./KalturaBulkUploadResult");
var KalturaBulkUploadResultVendorCatalogItem = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBulkUploadResultVendorCatalogItem, _super);
    function KalturaBulkUploadResultVendorCatalogItem(data) {
        return _super.call(this, data) || this;
    }
    KalturaBulkUploadResultVendorCatalogItem.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBulkUploadResultVendorCatalogItem' },
            vendorCatalogItemId: { type: 'n' },
            vendorPartnerId: { type: 'n' },
            name: { type: 's' },
            systemName: { type: 's' },
            serviceFeature: { type: 'en', subTypeConstructor: KalturaVendorServiceFeature_1.KalturaVendorServiceFeature, subType: 'KalturaVendorServiceFeature' },
            serviceType: { type: 'en', subTypeConstructor: KalturaVendorServiceType_1.KalturaVendorServiceType, subType: 'KalturaVendorServiceType' },
            turnAroundTime: { type: 'en', subTypeConstructor: KalturaVendorServiceTurnAroundTime_1.KalturaVendorServiceTurnAroundTime, subType: 'KalturaVendorServiceTurnAroundTime' },
            sourceLanguage: { type: 'es', subTypeConstructor: KalturaCatalogItemLanguage_1.KalturaCatalogItemLanguage, subType: 'KalturaCatalogItemLanguage' },
            targetLanguage: { type: 'es', subTypeConstructor: KalturaCatalogItemLanguage_1.KalturaCatalogItemLanguage, subType: 'KalturaCatalogItemLanguage' },
            outputFormat: { type: 'en', subTypeConstructor: KalturaVendorCatalogItemOutputFormat_1.KalturaVendorCatalogItemOutputFormat, subType: 'KalturaVendorCatalogItemOutputFormat' },
            enableSpeakerId: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            fixedPriceAddons: { type: 'n' },
            pricing: { type: 'o', subTypeConstructor: KalturaVendorCatalogItemPricing_1.KalturaVendorCatalogItemPricing, subType: 'KalturaVendorCatalogItemPricing' },
            flavorParamsId: { type: 'n' },
            clearAudioFlavorParamsId: { type: 'n' }
        });
        return result;
    };
    return KalturaBulkUploadResultVendorCatalogItem;
}(KalturaBulkUploadResult_1.KalturaBulkUploadResult));
exports.KalturaBulkUploadResultVendorCatalogItem = KalturaBulkUploadResultVendorCatalogItem;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBulkUploadResultVendorCatalogItem', KalturaBulkUploadResultVendorCatalogItem);
