// kaltura-node-typescript-client/api/types/KalturaEventNotificationTemplateType.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEventNotificationTemplateType = void 0;
var KalturaEventNotificationTemplateType;
(function (KalturaEventNotificationTemplateType) {
    KalturaEventNotificationTemplateType["boolean"] = "booleanNotification.Boolean";
    KalturaEventNotificationTemplateType["bpmAbort"] = "businessProcessNotification.BusinessProcessAbort";
    KalturaEventNotificationTemplateType["bpmSignal"] = "businessProcessNotification.BusinessProcessSignal";
    KalturaEventNotificationTemplateType["bpmStart"] = "businessProcessNotification.BusinessProcessStart";
    KalturaEventNotificationTemplateType["email"] = "emailNotification.Email";
    KalturaEventNotificationTemplateType["http"] = "httpNotification.Http";
    KalturaEventNotificationTemplateType["kafka"] = "kafkaNotification.Kafka";
    KalturaEventNotificationTemplateType["push"] = "pushNotification.Push";
})(KalturaEventNotificationTemplateType = exports.KalturaEventNotificationTemplateType || (exports.KalturaEventNotificationTemplateType = {}));
