// kaltura-node-typescript-client/api/types/KalturaSchedulerStatus.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSchedulerStatus = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBatchJobType_1 = require("./KalturaBatchJobType");
var KalturaSchedulerStatusType_1 = require("./KalturaSchedulerStatusType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaSchedulerStatus = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaSchedulerStatus, _super);
    function KalturaSchedulerStatus(data) {
        return _super.call(this, data) || this;
    }
    KalturaSchedulerStatus.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaSchedulerStatus' },
            id: { type: 'n', readOnly: true },
            schedulerConfiguredId: { type: 'n' },
            workerConfiguredId: { type: 'n' },
            workerType: { type: 'es', subTypeConstructor: KalturaBatchJobType_1.KalturaBatchJobType, subType: 'KalturaBatchJobType' },
            type: { type: 'en', subTypeConstructor: KalturaSchedulerStatusType_1.KalturaSchedulerStatusType, subType: 'KalturaSchedulerStatusType' },
            value: { type: 'n' },
            schedulerId: { type: 'n', readOnly: true },
            workerId: { type: 'n', readOnly: true }
        });
        return result;
    };
    return KalturaSchedulerStatus;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaSchedulerStatus = KalturaSchedulerStatus;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaSchedulerStatus', KalturaSchedulerStatus);
