// kaltura-node-typescript-client/api/types/KalturaBulkUploadResultScheduleResource.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBulkUploadResultScheduleResource = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBulkUploadResult_1 = require("./KalturaBulkUploadResult");
var KalturaBulkUploadResultScheduleResource = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBulkUploadResultScheduleResource, _super);
    function KalturaBulkUploadResultScheduleResource(data) {
        return _super.call(this, data) || this;
    }
    KalturaBulkUploadResultScheduleResource.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBulkUploadResultScheduleResource' },
            resourceId: { type: 's' },
            name: { type: 's' },
            type: { type: 's' },
            systemName: { type: 's' },
            description: { type: 's' },
            tags: { type: 's' },
            parentType: { type: 's' },
            parentSystemName: { type: 's' }
        });
        return result;
    };
    return KalturaBulkUploadResultScheduleResource;
}(KalturaBulkUploadResult_1.KalturaBulkUploadResult));
exports.KalturaBulkUploadResultScheduleResource = KalturaBulkUploadResultScheduleResource;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBulkUploadResultScheduleResource', KalturaBulkUploadResultScheduleResource);
