// kaltura-node-typescript-client/api/types/KalturaCategoryFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCategoryFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCategoryBaseFilter_1 = require("./KalturaCategoryBaseFilter");
var KalturaCategoryFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCategoryFilter, _super);
    function KalturaCategoryFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaCategoryFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCategoryFilter' },
            freeText: { type: 's' },
            membersIn: { type: 's' },
            nameOrReferenceIdStartsWith: { type: 's' },
            managerEqual: { type: 's' },
            memberEqual: { type: 's' },
            fullNameStartsWithIn: { type: 's' },
            ancestorIdIn: { type: 's' },
            idOrInheritedParentIdIn: { type: 's' }
        });
        return result;
    };
    return KalturaCategoryFilter;
}(KalturaCategoryBaseFilter_1.KalturaCategoryBaseFilter));
exports.KalturaCategoryFilter = KalturaCategoryFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCategoryFilter', KalturaCategoryFilter);
