// kaltura-node-typescript-client/api/types/KalturaMixEntryCompareAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMixEntryCompareAttribute = void 0;
var KalturaMixEntryCompareAttribute;
(function (KalturaMixEntryCompareAttribute) {
    KalturaMixEntryCompareAttribute["accessControlId"] = "accessControlId";
    KalturaMixEntryCompareAttribute["createdAt"] = "createdAt";
    KalturaMixEntryCompareAttribute["endDate"] = "endDate";
    KalturaMixEntryCompareAttribute["lastPlayedAt"] = "lastPlayedAt";
    KalturaMixEntryCompareAttribute["moderationCount"] = "moderationCount";
    KalturaMixEntryCompareAttribute["moderationStatus"] = "moderationStatus";
    KalturaMixEntryCompareAttribute["msDuration"] = "msDuration";
    KalturaMixEntryCompareAttribute["partnerId"] = "partnerId";
    KalturaMixEntryCompareAttribute["partnerSortValue"] = "partnerSortValue";
    KalturaMixEntryCompareAttribute["plays"] = "plays";
    KalturaMixEntryCompareAttribute["rank"] = "rank";
    KalturaMixEntryCompareAttribute["replacementStatus"] = "replacementStatus";
    KalturaMixEntryCompareAttribute["startDate"] = "startDate";
    KalturaMixEntryCompareAttribute["status"] = "status";
    KalturaMixEntryCompareAttribute["totalRank"] = "totalRank";
    KalturaMixEntryCompareAttribute["type"] = "type";
    KalturaMixEntryCompareAttribute["updatedAt"] = "updatedAt";
    KalturaMixEntryCompareAttribute["views"] = "views";
})(KalturaMixEntryCompareAttribute = exports.KalturaMixEntryCompareAttribute || (exports.KalturaMixEntryCompareAttribute = {}));
