// kaltura-node-typescript-client/api/types/KalturaParseMultiLanguageCaptionAssetJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaParseMultiLanguageCaptionAssetJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaParseMultiLanguageCaptionAssetJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaParseMultiLanguageCaptionAssetJobData, _super);
    function KalturaParseMultiLanguageCaptionAssetJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaParseMultiLanguageCaptionAssetJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaParseMultiLanguageCaptionAssetJobData' },
            multiLanaguageCaptionAssetId: { type: 's' },
            entryId: { type: 's' },
            fileLocation: { type: 's' },
            fileEncryptionKey: { type: 's' }
        });
        return result;
    };
    return KalturaParseMultiLanguageCaptionAssetJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaParseMultiLanguageCaptionAssetJobData = KalturaParseMultiLanguageCaptionAssetJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaParseMultiLanguageCaptionAssetJobData', KalturaParseMultiLanguageCaptionAssetJobData);
