// kaltura-node-typescript-client/api/types/KalturaAnswerCuePointBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAnswerCuePointBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCuePointFilter_1 = require("./KalturaCuePointFilter");
var KalturaAnswerCuePointBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAnswerCuePointBaseFilter, _super);
    function KalturaAnswerCuePointBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaAnswerCuePointBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAnswerCuePointBaseFilter' },
            parentIdEqual: { type: 's' },
            parentIdIn: { type: 's' },
            quizUserEntryIdEqual: { type: 's' },
            quizUserEntryIdIn: { type: 's' }
        });
        return result;
    };
    return KalturaAnswerCuePointBaseFilter;
}(KalturaCuePointFilter_1.KalturaCuePointFilter));
exports.KalturaAnswerCuePointBaseFilter = KalturaAnswerCuePointBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAnswerCuePointBaseFilter', KalturaAnswerCuePointBaseFilter);
