// kaltura-node-typescript-client/api/types/KalturaLiveStreamEntryCompareAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveStreamEntryCompareAttribute = void 0;
var KalturaLiveStreamEntryCompareAttribute;
(function (KalturaLiveStreamEntryCompareAttribute) {
    KalturaLiveStreamEntryCompareAttribute["accessControlId"] = "accessControlId";
    KalturaLiveStreamEntryCompareAttribute["createdAt"] = "createdAt";
    KalturaLiveStreamEntryCompareAttribute["endDate"] = "endDate";
    KalturaLiveStreamEntryCompareAttribute["lastPlayedAt"] = "lastPlayedAt";
    KalturaLiveStreamEntryCompareAttribute["mediaDate"] = "mediaDate";
    KalturaLiveStreamEntryCompareAttribute["mediaType"] = "mediaType";
    KalturaLiveStreamEntryCompareAttribute["moderationCount"] = "moderationCount";
    KalturaLiveStreamEntryCompareAttribute["moderationStatus"] = "moderationStatus";
    KalturaLiveStreamEntryCompareAttribute["msDuration"] = "msDuration";
    KalturaLiveStreamEntryCompareAttribute["partnerId"] = "partnerId";
    KalturaLiveStreamEntryCompareAttribute["partnerSortValue"] = "partnerSortValue";
    KalturaLiveStreamEntryCompareAttribute["plays"] = "plays";
    KalturaLiveStreamEntryCompareAttribute["rank"] = "rank";
    KalturaLiveStreamEntryCompareAttribute["replacementStatus"] = "replacementStatus";
    KalturaLiveStreamEntryCompareAttribute["startDate"] = "startDate";
    KalturaLiveStreamEntryCompareAttribute["status"] = "status";
    KalturaLiveStreamEntryCompareAttribute["totalRank"] = "totalRank";
    KalturaLiveStreamEntryCompareAttribute["type"] = "type";
    KalturaLiveStreamEntryCompareAttribute["updatedAt"] = "updatedAt";
    KalturaLiveStreamEntryCompareAttribute["views"] = "views";
})(KalturaLiveStreamEntryCompareAttribute = exports.KalturaLiveStreamEntryCompareAttribute || (exports.KalturaLiveStreamEntryCompareAttribute = {}));
