// kaltura-node-typescript-client/api/types/KalturaBaseEntryMatchAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBaseEntryMatchAttribute = void 0;
var KalturaBaseEntryMatchAttribute;
(function (KalturaBaseEntryMatchAttribute) {
    KalturaBaseEntryMatchAttribute["adminTags"] = "adminTags";
    KalturaBaseEntryMatchAttribute["categoriesIds"] = "categoriesIds";
    KalturaBaseEntryMatchAttribute["creatorId"] = "creatorId";
    KalturaBaseEntryMatchAttribute["description"] = "description";
    KalturaBaseEntryMatchAttribute["groupId"] = "groupId";
    KalturaBaseEntryMatchAttribute["id"] = "id";
    KalturaBaseEntryMatchAttribute["_name"] = "name";
    KalturaBaseEntryMatchAttribute["referenceId"] = "referenceId";
    KalturaBaseEntryMatchAttribute["replacedEntryId"] = "replacedEntryId";
    KalturaBaseEntryMatchAttribute["replacingEntryId"] = "replacingEntryId";
    KalturaBaseEntryMatchAttribute["searchText"] = "searchText";
    KalturaBaseEntryMatchAttribute["tags"] = "tags";
    KalturaBaseEntryMatchAttribute["userId"] = "userId";
})(KalturaBaseEntryMatchAttribute = exports.KalturaBaseEntryMatchAttribute || (exports.KalturaBaseEntryMatchAttribute = {}));
