// kaltura-node-typescript-client/api/types/KalturaSearchResult.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSearchResult = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaLicenseType_1 = require("./KalturaLicenseType");
var KalturaSearch_1 = require("./KalturaSearch");
var KalturaSearchResult = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaSearchResult, _super);
    function KalturaSearchResult(data) {
        return _super.call(this, data) || this;
    }
    KalturaSearchResult.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaSearchResult' },
            id: { type: 's' },
            title: { type: 's' },
            thumbUrl: { type: 's' },
            description: { type: 's' },
            tags: { type: 's' },
            url: { type: 's' },
            sourceLink: { type: 's' },
            credit: { type: 's' },
            licenseType: { type: 'en', subTypeConstructor: KalturaLicenseType_1.KalturaLicenseType, subType: 'KalturaLicenseType' },
            flashPlaybackType: { type: 's' },
            fileExt: { type: 's' }
        });
        return result;
    };
    return KalturaSearchResult;
}(KalturaSearch_1.KalturaSearch));
exports.KalturaSearchResult = KalturaSearchResult;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaSearchResult', KalturaSearchResult);
