// kaltura-node-typescript-client/api/types/KalturaAnswerCuePoint.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAnswerCuePoint = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaString_1 = require("./KalturaString");
var KalturaCuePoint_1 = require("./KalturaCuePoint");
var KalturaAnswerCuePoint = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAnswerCuePoint, _super);
    function KalturaAnswerCuePoint(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.correctAnswerKeys === 'undefined')
            _this.correctAnswerKeys = [];
        return _this;
    }
    KalturaAnswerCuePoint.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAnswerCuePoint' },
            parentId: { type: 's' },
            quizUserEntryId: { type: 's' },
            answerKey: { type: 's' },
            openAnswer: { type: 's' },
            isCorrect: { type: 'en', readOnly: true, subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            correctAnswerKeys: { type: 'a', readOnly: true, subTypeConstructor: KalturaString_1.KalturaString, subType: 'KalturaString' },
            explanation: { type: 's', readOnly: true },
            feedback: { type: 's' }
        });
        return result;
    };
    return KalturaAnswerCuePoint;
}(KalturaCuePoint_1.KalturaCuePoint));
exports.KalturaAnswerCuePoint = KalturaAnswerCuePoint;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAnswerCuePoint', KalturaAnswerCuePoint);
