// kaltura-node-typescript-client/api/types/KalturaSessionCuePoint.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSessionCuePoint = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCuePoint_1 = require("./KalturaCuePoint");
var KalturaSessionCuePoint = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaSessionCuePoint, _super);
    function KalturaSessionCuePoint(data) {
        return _super.call(this, data) || this;
    }
    KalturaSessionCuePoint.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaSessionCuePoint' },
            name: { type: 's' },
            endTime: { type: 'n' },
            duration: { type: 'n', readOnly: true },
            sessionOwner: { type: 's' }
        });
        return result;
    };
    return KalturaSessionCuePoint;
}(KalturaCuePoint_1.KalturaCuePoint));
exports.KalturaSessionCuePoint = KalturaSessionCuePoint;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaSessionCuePoint', KalturaSessionCuePoint);
