// kaltura-node-typescript-client/api/types/KalturaBusinessProcessCase.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBusinessProcessCase = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaBusinessProcessCase = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBusinessProcessCase, _super);
    function KalturaBusinessProcessCase(data) {
        return _super.call(this, data) || this;
    }
    KalturaBusinessProcessCase.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBusinessProcessCase' },
            id: { type: 's' },
            businessProcessId: { type: 's' },
            businessProcessStartNotificationTemplateId: { type: 'n' },
            suspended: { type: 'b' },
            activityId: { type: 's' }
        });
        return result;
    };
    return KalturaBusinessProcessCase;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaBusinessProcessCase = KalturaBusinessProcessCase;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBusinessProcessCase', KalturaBusinessProcessCase);
