// kaltura-node-typescript-client/api/types/KalturaBaseEntryFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBaseEntryFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaBaseEntryBaseFilter_1 = require("./KalturaBaseEntryBaseFilter");
var KalturaBaseEntryFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBaseEntryFilter, _super);
    function KalturaBaseEntryFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaBaseEntryFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBaseEntryFilter' },
            freeText: { type: 's' },
            excludedFreeTextGroups: { type: 's' },
            descriptionLike: { type: 's' },
            isRoot: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            categoriesFullNameIn: { type: 's' },
            categoryAncestorIdIn: { type: 's' },
            redirectFromEntryId: { type: 's' },
            conversionProfileIdEqual: { type: 'n' }
        });
        return result;
    };
    return KalturaBaseEntryFilter;
}(KalturaBaseEntryBaseFilter_1.KalturaBaseEntryBaseFilter));
exports.KalturaBaseEntryFilter = KalturaBaseEntryFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBaseEntryFilter', KalturaBaseEntryFilter);
