// kaltura-node-typescript-client/api/types/KalturaESearchAbstractGroupItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaESearchAbstractGroupItem = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaESearchItemType_1 = require("./KalturaESearchItemType");
var KalturaESearchRange_1 = require("./KalturaESearchRange");
var KalturaESearchGroupBaseItem_1 = require("./KalturaESearchGroupBaseItem");
var KalturaESearchAbstractGroupItem = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaESearchAbstractGroupItem, _super);
    function KalturaESearchAbstractGroupItem(data) {
        return _super.call(this, data) || this;
    }
    KalturaESearchAbstractGroupItem.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaESearchAbstractGroupItem' },
            searchTerm: { type: 's' },
            itemType: { type: 'en', subTypeConstructor: KalturaESearchItemType_1.KalturaESearchItemType, subType: 'KalturaESearchItemType' },
            range: { type: 'o', subTypeConstructor: KalturaESearchRange_1.KalturaESearchRange, subType: 'KalturaESearchRange' },
            addHighlight: { type: 'b' }
        });
        return result;
    };
    return KalturaESearchAbstractGroupItem;
}(KalturaESearchGroupBaseItem_1.KalturaESearchGroupBaseItem));
exports.KalturaESearchAbstractGroupItem = KalturaESearchAbstractGroupItem;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaESearchAbstractGroupItem', KalturaESearchAbstractGroupItem);
