// kaltura-node-typescript-client/api/types/KalturaMediaEntryMatchAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMediaEntryMatchAttribute = void 0;
var KalturaMediaEntryMatchAttribute;
(function (KalturaMediaEntryMatchAttribute) {
    KalturaMediaEntryMatchAttribute["adminTags"] = "adminTags";
    KalturaMediaEntryMatchAttribute["categoriesIds"] = "categoriesIds";
    KalturaMediaEntryMatchAttribute["creatorId"] = "creatorId";
    KalturaMediaEntryMatchAttribute["description"] = "description";
    KalturaMediaEntryMatchAttribute["durationType"] = "durationType";
    KalturaMediaEntryMatchAttribute["flavorParamsIds"] = "flavorParamsIds";
    KalturaMediaEntryMatchAttribute["groupId"] = "groupId";
    KalturaMediaEntryMatchAttribute["id"] = "id";
    KalturaMediaEntryMatchAttribute["_name"] = "name";
    KalturaMediaEntryMatchAttribute["referenceId"] = "referenceId";
    KalturaMediaEntryMatchAttribute["replacedEntryId"] = "replacedEntryId";
    KalturaMediaEntryMatchAttribute["replacingEntryId"] = "replacingEntryId";
    KalturaMediaEntryMatchAttribute["searchText"] = "searchText";
    KalturaMediaEntryMatchAttribute["tags"] = "tags";
    KalturaMediaEntryMatchAttribute["userId"] = "userId";
})(KalturaMediaEntryMatchAttribute = exports.KalturaMediaEntryMatchAttribute || (exports.KalturaMediaEntryMatchAttribute = {}));
