// kaltura-node-typescript-client/api/types/KalturaFeedItemInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFeedItemInfo = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaFeedItemInfo = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaFeedItemInfo, _super);
    function KalturaFeedItemInfo(data) {
        return _super.call(this, data) || this;
    }
    KalturaFeedItemInfo.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaFeedItemInfo' },
            itemXPath: { type: 's' },
            itemPublishDateXPath: { type: 's' },
            itemUniqueIdentifierXPath: { type: 's' },
            itemContentFileSizeXPath: { type: 's' },
            itemContentUrlXPath: { type: 's' },
            itemContentBitrateXPath: { type: 's' },
            itemHashXPath: { type: 's' },
            itemContentXpath: { type: 's' },
            contentBitrateAttributeName: { type: 's' }
        });
        return result;
    };
    return KalturaFeedItemInfo;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaFeedItemInfo = KalturaFeedItemInfo;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaFeedItemInfo', KalturaFeedItemInfo);
