// kaltura-node-typescript-client/api/types/KalturaS3DropFolder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaS3DropFolder = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDropFolder_1 = require("./KalturaDropFolder");
var KalturaS3DropFolder = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaS3DropFolder, _super);
    function KalturaS3DropFolder(data) {
        return _super.call(this, data) || this;
    }
    KalturaS3DropFolder.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaS3DropFolder' },
            s3Host: { type: 's' },
            s3Region: { type: 's' },
            s3UserId: { type: 's' },
            s3Password: { type: 's' }
        });
        return result;
    };
    return KalturaS3DropFolder;
}(KalturaDropFolder_1.KalturaDropFolder));
exports.KalturaS3DropFolder = KalturaS3DropFolder;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaS3DropFolder', KalturaS3DropFolder);
