// kaltura-node-typescript-client/api/types/KalturaMetroPcsDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMetroPcsDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaMetroPcsDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMetroPcsDistributionProfile, _super);
    function KalturaMetroPcsDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaMetroPcsDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMetroPcsDistributionProfile' },
            ftpHost: { type: 's' },
            ftpLogin: { type: 's' },
            ftpPass: { type: 's' },
            ftpPath: { type: 's' },
            providerName: { type: 's' },
            providerId: { type: 's' },
            copyright: { type: 's' },
            entitlements: { type: 's' },
            rating: { type: 's' },
            itemType: { type: 's' }
        });
        return result;
    };
    return KalturaMetroPcsDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaMetroPcsDistributionProfile = KalturaMetroPcsDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMetroPcsDistributionProfile', KalturaMetroPcsDistributionProfile);
