// kaltura-node-typescript-client/api/types/KalturaEntryScheduleEvent.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEntryScheduleEvent = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaScheduleEvent_1 = require("./KalturaScheduleEvent");
var KalturaEntryScheduleEvent = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaEntryScheduleEvent, _super);
    function KalturaEntryScheduleEvent(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.blackoutConflicts === 'undefined')
            _this.blackoutConflicts = [];
        return _this;
    }
    KalturaEntryScheduleEvent.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaEntryScheduleEvent' },
            templateEntryId: { type: 's' },
            entryIds: { type: 's' },
            categoryIds: { type: 's' },
            blackoutConflicts: { type: 'a', readOnly: true, subTypeConstructor: KalturaScheduleEvent_1.KalturaScheduleEvent, subType: 'KalturaScheduleEvent' }
        });
        return result;
    };
    return KalturaEntryScheduleEvent;
}(KalturaScheduleEvent_1.KalturaScheduleEvent));
exports.KalturaEntryScheduleEvent = KalturaEntryScheduleEvent;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaEntryScheduleEvent', KalturaEntryScheduleEvent);
