// kaltura-node-typescript-client/api/types/KalturaESearchUserFieldName.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaESearchUserFieldName = void 0;
var KalturaESearchUserFieldName;
(function (KalturaESearchUserFieldName) {
    KalturaESearchUserFieldName["capabilities"] = "capabilities";
    KalturaESearchUserFieldName["company"] = "company";
    KalturaESearchUserFieldName["country"] = "country";
    KalturaESearchUserFieldName["createdAt"] = "created_at";
    KalturaESearchUserFieldName["email"] = "email";
    KalturaESearchUserFieldName["externalId"] = "external_id";
    KalturaESearchUserFieldName["firstName"] = "first_name";
    KalturaESearchUserFieldName["fullName"] = "full_name";
    KalturaESearchUserFieldName["groupIds"] = "group_ids";
    KalturaESearchUserFieldName["isAdmin"] = "is_admin";
    KalturaESearchUserFieldName["isHashed"] = "is_hashed";
    KalturaESearchUserFieldName["lastName"] = "last_name";
    KalturaESearchUserFieldName["loginEnabled"] = "login_enabled";
    KalturaESearchUserFieldName["permissionNames"] = "permission_names";
    KalturaESearchUserFieldName["roleIds"] = "role_ids";
    KalturaESearchUserFieldName["screenName"] = "screen_name";
    KalturaESearchUserFieldName["tags"] = "tags";
    KalturaESearchUserFieldName["title"] = "title";
    KalturaESearchUserFieldName["type"] = "user_type";
    KalturaESearchUserFieldName["updatedAt"] = "updated_at";
    KalturaESearchUserFieldName["userId"] = "user_id";
})(KalturaESearchUserFieldName = exports.KalturaESearchUserFieldName || (exports.KalturaESearchUserFieldName = {}));
