// kaltura-node-typescript-client/api/types/KalturaBatchHistoryData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBatchHistoryData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaBatchHistoryData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBatchHistoryData, _super);
    function KalturaBatchHistoryData(data) {
        return _super.call(this, data) || this;
    }
    KalturaBatchHistoryData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBatchHistoryData' },
            schedulerId: { type: 'n' },
            workerId: { type: 'n' },
            batchIndex: { type: 'n' },
            timeStamp: { type: 'n' },
            message: { type: 's' },
            errType: { type: 'n' },
            errNumber: { type: 'n' },
            hostName: { type: 's' },
            sessionId: { type: 's' },
            schedulerName: { type: 's' }
        });
        return result;
    };
    return KalturaBatchHistoryData;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaBatchHistoryData = KalturaBatchHistoryData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBatchHistoryData', KalturaBatchHistoryData);
