// kaltura-node-typescript-client/api/types/KalturaEmailIngestionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEmailIngestionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaEntryModerationStatus_1 = require("./KalturaEntryModerationStatus");
var KalturaEmailIngestionProfileStatus_1 = require("./KalturaEmailIngestionProfileStatus");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaEmailIngestionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaEmailIngestionProfile, _super);
    function KalturaEmailIngestionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaEmailIngestionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaEmailIngestionProfile' },
            id: { type: 'n', readOnly: true },
            name: { type: 's' },
            description: { type: 's' },
            emailAddress: { type: 's' },
            mailboxId: { type: 's' },
            partnerId: { type: 'n', readOnly: true },
            conversionProfile2Id: { type: 'n' },
            moderationStatus: { type: 'en', subTypeConstructor: KalturaEntryModerationStatus_1.KalturaEntryModerationStatus, subType: 'KalturaEntryModerationStatus' },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaEmailIngestionProfileStatus_1.KalturaEmailIngestionProfileStatus, subType: 'KalturaEmailIngestionProfileStatus' },
            createdAt: { type: 's', readOnly: true },
            defaultCategory: { type: 's' },
            defaultUserId: { type: 's' },
            defaultTags: { type: 's' },
            defaultAdminTags: { type: 's' },
            maxAttachmentSizeKbytes: { type: 'n' },
            maxAttachmentsPerMail: { type: 'n' }
        });
        return result;
    };
    return KalturaEmailIngestionProfile;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaEmailIngestionProfile = KalturaEmailIngestionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaEmailIngestionProfile', KalturaEmailIngestionProfile);
