// kaltura-node-typescript-client/api/types/KalturaCodeCuePointBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCodeCuePointBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCuePointFilter_1 = require("./KalturaCuePointFilter");
var KalturaCodeCuePointBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCodeCuePointBaseFilter, _super);
    function KalturaCodeCuePointBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaCodeCuePointBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCodeCuePointBaseFilter' },
            codeLike: { type: 's' },
            codeMultiLikeOr: { type: 's' },
            codeMultiLikeAnd: { type: 's' },
            codeEqual: { type: 's' },
            codeIn: { type: 's' },
            descriptionLike: { type: 's' },
            descriptionMultiLikeOr: { type: 's' },
            descriptionMultiLikeAnd: { type: 's' },
            endTimeGreaterThanOrEqual: { type: 'n' },
            endTimeLessThanOrEqual: { type: 'n' },
            durationGreaterThanOrEqual: { type: 'n' },
            durationLessThanOrEqual: { type: 'n' }
        });
        return result;
    };
    return KalturaCodeCuePointBaseFilter;
}(KalturaCuePointFilter_1.KalturaCuePointFilter));
exports.KalturaCodeCuePointBaseFilter = KalturaCodeCuePointBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCodeCuePointBaseFilter', KalturaCodeCuePointBaseFilter);
