// kaltura-node-typescript-client/api/types/KalturaBulkUploadJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBulkUploadJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBulkUploadObjectType_1 = require("./KalturaBulkUploadObjectType");
var KalturaBulkUploadObjectData_1 = require("./KalturaBulkUploadObjectData");
var KalturaBulkUploadType_1 = require("./KalturaBulkUploadType");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaBulkUploadJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBulkUploadJobData, _super);
    function KalturaBulkUploadJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaBulkUploadJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBulkUploadJobData' },
            userId: { type: 's', readOnly: true },
            uploadedBy: { type: 's', readOnly: true },
            conversionProfileId: { type: 'n', readOnly: true },
            resultsFileLocalPath: { type: 's', readOnly: true },
            resultsFileUrl: { type: 's', readOnly: true },
            numOfEntries: { type: 'n', readOnly: true },
            numOfObjects: { type: 'n', readOnly: true },
            filePath: { type: 's', readOnly: true },
            bulkUploadObjectType: { type: 'es', readOnly: true, subTypeConstructor: KalturaBulkUploadObjectType_1.KalturaBulkUploadObjectType, subType: 'KalturaBulkUploadObjectType' },
            fileName: { type: 's' },
            objectData: { type: 'o', readOnly: true, subTypeConstructor: KalturaBulkUploadObjectData_1.KalturaBulkUploadObjectData, subType: 'KalturaBulkUploadObjectData' },
            type: { type: 'es', readOnly: true, subTypeConstructor: KalturaBulkUploadType_1.KalturaBulkUploadType, subType: 'KalturaBulkUploadType' },
            emailRecipients: { type: 's' },
            numOfErrorObjects: { type: 'n' },
            privileges: { type: 's' }
        });
        return result;
    };
    return KalturaBulkUploadJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaBulkUploadJobData = KalturaBulkUploadJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBulkUploadJobData', KalturaBulkUploadJobData);
