// kaltura-node-typescript-client/api/types/KalturaESearchCuePointItemData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaESearchCuePointItemData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaString_1 = require("./KalturaString");
var KalturaESearchItemData_1 = require("./KalturaESearchItemData");
var KalturaESearchCuePointItemData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaESearchCuePointItemData, _super);
    function KalturaESearchCuePointItemData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.tags === 'undefined')
            _this.tags = [];
        if (typeof _this.answers === 'undefined')
            _this.answers = [];
        return _this;
    }
    KalturaESearchCuePointItemData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaESearchCuePointItemData' },
            cuePointType: { type: 's' },
            id: { type: 's' },
            name: { type: 's' },
            text: { type: 's' },
            tags: { type: 'a', subTypeConstructor: KalturaString_1.KalturaString, subType: 'KalturaString' },
            startTime: { type: 's' },
            endTime: { type: 's' },
            subType: { type: 's' },
            question: { type: 's' },
            answers: { type: 'a', subTypeConstructor: KalturaString_1.KalturaString, subType: 'KalturaString' },
            hint: { type: 's' },
            explanation: { type: 's' },
            assetId: { type: 's' }
        });
        return result;
    };
    return KalturaESearchCuePointItemData;
}(KalturaESearchItemData_1.KalturaESearchItemData));
exports.KalturaESearchCuePointItemData = KalturaESearchCuePointItemData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaESearchCuePointItemData', KalturaESearchCuePointItemData);
