// kaltura-node-typescript-client/api/types/KalturaAttachmentAsset.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAttachmentAsset = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaAttachmentType_1 = require("./KalturaAttachmentType");
var KalturaAttachmentAssetStatus_1 = require("./KalturaAttachmentAssetStatus");
var KalturaAsset_1 = require("./KalturaAsset");
var KalturaAttachmentAsset = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAttachmentAsset, _super);
    function KalturaAttachmentAsset(data) {
        return _super.call(this, data) || this;
    }
    KalturaAttachmentAsset.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAttachmentAsset' },
            filename: { type: 's' },
            title: { type: 's' },
            format: { type: 'es', subTypeConstructor: KalturaAttachmentType_1.KalturaAttachmentType, subType: 'KalturaAttachmentType' },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaAttachmentAssetStatus_1.KalturaAttachmentAssetStatus, subType: 'KalturaAttachmentAssetStatus' }
        });
        return result;
    };
    return KalturaAttachmentAsset;
}(KalturaAsset_1.KalturaAsset));
exports.KalturaAttachmentAsset = KalturaAttachmentAsset;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAttachmentAsset', KalturaAttachmentAsset);
