// kaltura-node-typescript-client/api/types/KalturaBulkUpload.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBulkUpload = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaBatchJobStatus_1 = require("./KalturaBatchJobStatus");
var KalturaBulkUploadType_1 = require("./KalturaBulkUploadType");
var KalturaBulkUploadResult_1 = require("./KalturaBulkUploadResult");
var KalturaBatchJobErrorTypes_1 = require("./KalturaBatchJobErrorTypes");
var KalturaBulkUploadObjectType_1 = require("./KalturaBulkUploadObjectType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaBulkUpload = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaBulkUpload, _super);
    function KalturaBulkUpload(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.results === 'undefined')
            _this.results = [];
        return _this;
    }
    KalturaBulkUpload.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaBulkUpload' },
            id: { type: 'n' },
            uploadedBy: { type: 's' },
            uploadedByUserId: { type: 's' },
            uploadedOn: { type: 'd' },
            numOfEntries: { type: 'n' },
            status: { type: 'en', subTypeConstructor: KalturaBatchJobStatus_1.KalturaBatchJobStatus, subType: 'KalturaBatchJobStatus' },
            logFileUrl: { type: 's' },
            csvFileUrl: { type: 's' },
            bulkFileUrl: { type: 's' },
            bulkUploadType: { type: 'es', subTypeConstructor: KalturaBulkUploadType_1.KalturaBulkUploadType, subType: 'KalturaBulkUploadType' },
            results: { type: 'a', subTypeConstructor: KalturaBulkUploadResult_1.KalturaBulkUploadResult, subType: 'KalturaBulkUploadResult' },
            error: { type: 's' },
            errorType: { type: 'en', subTypeConstructor: KalturaBatchJobErrorTypes_1.KalturaBatchJobErrorTypes, subType: 'KalturaBatchJobErrorTypes' },
            errorNumber: { type: 'n' },
            fileName: { type: 's' },
            description: { type: 's' },
            numOfObjects: { type: 'n' },
            bulkUploadObjectType: { type: 'es', subTypeConstructor: KalturaBulkUploadObjectType_1.KalturaBulkUploadObjectType, subType: 'KalturaBulkUploadObjectType' }
        });
        return result;
    };
    return KalturaBulkUpload;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaBulkUpload = KalturaBulkUpload;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaBulkUpload', KalturaBulkUpload);
