// kaltura-node-typescript-client/api/types/KalturaStorageProfileBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaStorageProfileBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaStorageProfileStatus_1 = require("./KalturaStorageProfileStatus");
var KalturaStorageProfileProtocol_1 = require("./KalturaStorageProfileProtocol");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaStorageProfileBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaStorageProfileBaseFilter, _super);
    function KalturaStorageProfileBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaStorageProfileBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaStorageProfileBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            systemNameEqual: { type: 's' },
            systemNameIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaStorageProfileStatus_1.KalturaStorageProfileStatus, subType: 'KalturaStorageProfileStatus' },
            statusIn: { type: 's' },
            protocolEqual: { type: 'es', subTypeConstructor: KalturaStorageProfileProtocol_1.KalturaStorageProfileProtocol, subType: 'KalturaStorageProfileProtocol' },
            protocolIn: { type: 's' }
        });
        return result;
    };
    return KalturaStorageProfileBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaStorageProfileBaseFilter = KalturaStorageProfileBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaStorageProfileBaseFilter', KalturaStorageProfileBaseFilter);
