// kaltura-node-typescript-client/api/types/KalturaHttpNotificationAuthenticationMethod.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaHttpNotificationAuthenticationMethod = void 0;
var KalturaHttpNotificationAuthenticationMethod;
(function (KalturaHttpNotificationAuthenticationMethod) {
    KalturaHttpNotificationAuthenticationMethod[KalturaHttpNotificationAuthenticationMethod["any"] = -17] = "any";
    KalturaHttpNotificationAuthenticationMethod[KalturaHttpNotificationAuthenticationMethod["anysafe"] = -18] = "anysafe";
    KalturaHttpNotificationAuthenticationMethod[KalturaHttpNotificationAuthenticationMethod["basic"] = 1] = "basic";
    KalturaHttpNotificationAuthenticationMethod[KalturaHttpNotificationAuthenticationMethod["digest"] = 2] = "digest";
    KalturaHttpNotificationAuthenticationMethod[KalturaHttpNotificationAuthenticationMethod["gssnegotiate"] = 4] = "gssnegotiate";
    KalturaHttpNotificationAuthenticationMethod[KalturaHttpNotificationAuthenticationMethod["ntlm"] = 8] = "ntlm";
})(KalturaHttpNotificationAuthenticationMethod = exports.KalturaHttpNotificationAuthenticationMethod || (exports.KalturaHttpNotificationAuthenticationMethod = {}));
