// kaltura-node-typescript-client/api/types/KalturaEntryServerNodeBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaEntryServerNodeBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaEntryServerNodeStatus_1 = require("./KalturaEntryServerNodeStatus");
var KalturaEntryServerNodeType_1 = require("./KalturaEntryServerNodeType");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaEntryServerNodeBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaEntryServerNodeBaseFilter, _super);
    function KalturaEntryServerNodeBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaEntryServerNodeBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaEntryServerNodeBaseFilter' },
            entryIdEqual: { type: 's' },
            entryIdIn: { type: 's' },
            serverNodeIdEqual: { type: 'n' },
            serverNodeIdIn: { type: 's' },
            serverNodeIdNotIn: { type: 's' },
            createdAtLessThanOrEqual: { type: 'd' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaEntryServerNodeStatus_1.KalturaEntryServerNodeStatus, subType: 'KalturaEntryServerNodeStatus' },
            statusIn: { type: 's' },
            serverTypeEqual: { type: 'es', subTypeConstructor: KalturaEntryServerNodeType_1.KalturaEntryServerNodeType, subType: 'KalturaEntryServerNodeType' },
            serverTypeIn: { type: 's' },
            serverTypeNotIn: { type: 's' }
        });
        return result;
    };
    return KalturaEntryServerNodeBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaEntryServerNodeBaseFilter = KalturaEntryServerNodeBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaEntryServerNodeBaseFilter', KalturaEntryServerNodeBaseFilter);
