// kaltura-node-typescript-client/api/types/KalturaLiveStreamBitrate.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveStreamBitrate = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaLiveStreamBitrate = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveStreamBitrate, _super);
    function KalturaLiveStreamBitrate(data) {
        return _super.call(this, data) || this;
    }
    KalturaLiveStreamBitrate.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveStreamBitrate' },
            bitrate: { type: 'n' },
            width: { type: 'n' },
            height: { type: 'n' },
            tags: { type: 's' }
        });
        return result;
    };
    return KalturaLiveStreamBitrate;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaLiveStreamBitrate = KalturaLiveStreamBitrate;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveStreamBitrate', KalturaLiveStreamBitrate);
