// kaltura-node-typescript-client/api/types/KalturaLiveChannelSegment.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveChannelSegment = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaLiveChannelSegmentType_1 = require("./KalturaLiveChannelSegmentType");
var KalturaLiveChannelSegmentStatus_1 = require("./KalturaLiveChannelSegmentStatus");
var KalturaLiveChannelSegmentTriggerType_1 = require("./KalturaLiveChannelSegmentTriggerType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaLiveChannelSegment = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveChannelSegment, _super);
    function KalturaLiveChannelSegment(data) {
        return _super.call(this, data) || this;
    }
    KalturaLiveChannelSegment.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveChannelSegment' },
            id: { type: 'n', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            name: { type: 's' },
            description: { type: 's' },
            tags: { type: 's' },
            type: { type: 'es', subTypeConstructor: KalturaLiveChannelSegmentType_1.KalturaLiveChannelSegmentType, subType: 'KalturaLiveChannelSegmentType' },
            status: { type: 'es', readOnly: true, subTypeConstructor: KalturaLiveChannelSegmentStatus_1.KalturaLiveChannelSegmentStatus, subType: 'KalturaLiveChannelSegmentStatus' },
            channelId: { type: 's' },
            entryId: { type: 's' },
            triggerType: { type: 'es', subTypeConstructor: KalturaLiveChannelSegmentTriggerType_1.KalturaLiveChannelSegmentTriggerType, subType: 'KalturaLiveChannelSegmentTriggerType' },
            triggerSegmentId: { type: 'n' },
            startTime: { type: 'n' },
            duration: { type: 'n' }
        });
        return result;
    };
    return KalturaLiveChannelSegment;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaLiveChannelSegment = KalturaLiveChannelSegment;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveChannelSegment', KalturaLiveChannelSegment);
