// kaltura-node-typescript-client/api/types/KalturaVerizonVcastDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaVerizonVcastDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaVerizonVcastDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaVerizonVcastDistributionProfile, _super);
    function KalturaVerizonVcastDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaVerizonVcastDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaVerizonVcastDistributionProfile' },
            ftpHost: { type: 's' },
            ftpLogin: { type: 's' },
            ftpPass: { type: 's' },
            providerName: { type: 's' },
            providerId: { type: 's' },
            entitlement: { type: 's' },
            priority: { type: 's' },
            allowStreaming: { type: 's' },
            streamingPriceCode: { type: 's' },
            allowDownload: { type: 's' },
            downloadPriceCode: { type: 's' }
        });
        return result;
    };
    return KalturaVerizonVcastDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaVerizonVcastDistributionProfile = KalturaVerizonVcastDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaVerizonVcastDistributionProfile', KalturaVerizonVcastDistributionProfile);
