// kaltura-node-typescript-client/api/types/KalturaCategoryUserAdvancedFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCategoryUserAdvancedFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaSearchItem_1 = require("./KalturaSearchItem");
var KalturaCategoryUserAdvancedFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCategoryUserAdvancedFilter, _super);
    function KalturaCategoryUserAdvancedFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaCategoryUserAdvancedFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCategoryUserAdvancedFilter' },
            memberIdEq: { type: 's' },
            memberIdIn: { type: 's' },
            memberPermissionsMatchOr: { type: 's' },
            memberPermissionsMatchAnd: { type: 's' }
        });
        return result;
    };
    return KalturaCategoryUserAdvancedFilter;
}(KalturaSearchItem_1.KalturaSearchItem));
exports.KalturaCategoryUserAdvancedFilter = KalturaCategoryUserAdvancedFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCategoryUserAdvancedFilter', KalturaCategoryUserAdvancedFilter);
