// kaltura-node-typescript-client/api/types/KalturaScheduleEventFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaScheduleEventFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaScheduleEventBaseFilter_1 = require("./KalturaScheduleEventBaseFilter");
var KalturaScheduleEventFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaScheduleEventFilter, _super);
    function KalturaScheduleEventFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaScheduleEventFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaScheduleEventFilter' },
            resourceIdsLike: { type: 's' },
            resourceIdsMultiLikeOr: { type: 's' },
            resourceIdsMultiLikeAnd: { type: 's' },
            parentResourceIdsLike: { type: 's' },
            parentResourceIdsMultiLikeOr: { type: 's' },
            parentResourceIdsMultiLikeAnd: { type: 's' },
            templateEntryCategoriesIdsMultiLikeAnd: { type: 's' },
            templateEntryCategoriesIdsMultiLikeOr: { type: 's' },
            resourceSystemNamesMultiLikeOr: { type: 's' },
            templateEntryCategoriesIdsLike: { type: 's' },
            resourceSystemNamesMultiLikeAnd: { type: 's' },
            resourceSystemNamesLike: { type: 's' },
            resourceIdEqual: { type: 's' }
        });
        return result;
    };
    return KalturaScheduleEventFilter;
}(KalturaScheduleEventBaseFilter_1.KalturaScheduleEventBaseFilter));
exports.KalturaScheduleEventFilter = KalturaScheduleEventFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaScheduleEventFilter', KalturaScheduleEventFilter);
