// kaltura-node-typescript-client/api/types/KalturaLiveStreamAdminEntryOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveStreamAdminEntryOrderBy = void 0;
var KalturaLiveStreamAdminEntryOrderBy;
(function (KalturaLiveStreamAdminEntryOrderBy) {
    KalturaLiveStreamAdminEntryOrderBy["createdAtAsc"] = "+createdAt";
    KalturaLiveStreamAdminEntryOrderBy["createdAtDesc"] = "-createdAt";
    KalturaLiveStreamAdminEntryOrderBy["durationAsc"] = "+duration";
    KalturaLiveStreamAdminEntryOrderBy["durationDesc"] = "-duration";
    KalturaLiveStreamAdminEntryOrderBy["endDateAsc"] = "+endDate";
    KalturaLiveStreamAdminEntryOrderBy["endDateDesc"] = "-endDate";
    KalturaLiveStreamAdminEntryOrderBy["firstBroadcastAsc"] = "+firstBroadcast";
    KalturaLiveStreamAdminEntryOrderBy["firstBroadcastDesc"] = "-firstBroadcast";
    KalturaLiveStreamAdminEntryOrderBy["lastBroadcastAsc"] = "+lastBroadcast";
    KalturaLiveStreamAdminEntryOrderBy["lastBroadcastDesc"] = "-lastBroadcast";
    KalturaLiveStreamAdminEntryOrderBy["lastPlayedAtAsc"] = "+lastPlayedAt";
    KalturaLiveStreamAdminEntryOrderBy["lastPlayedAtDesc"] = "-lastPlayedAt";
    KalturaLiveStreamAdminEntryOrderBy["mediaTypeAsc"] = "+mediaType";
    KalturaLiveStreamAdminEntryOrderBy["mediaTypeDesc"] = "-mediaType";
    KalturaLiveStreamAdminEntryOrderBy["moderationCountAsc"] = "+moderationCount";
    KalturaLiveStreamAdminEntryOrderBy["moderationCountDesc"] = "-moderationCount";
    KalturaLiveStreamAdminEntryOrderBy["nameAsc"] = "+name";
    KalturaLiveStreamAdminEntryOrderBy["nameDesc"] = "-name";
    KalturaLiveStreamAdminEntryOrderBy["partnerSortValueAsc"] = "+partnerSortValue";
    KalturaLiveStreamAdminEntryOrderBy["partnerSortValueDesc"] = "-partnerSortValue";
    KalturaLiveStreamAdminEntryOrderBy["playsAsc"] = "+plays";
    KalturaLiveStreamAdminEntryOrderBy["playsDesc"] = "-plays";
    KalturaLiveStreamAdminEntryOrderBy["rankAsc"] = "+rank";
    KalturaLiveStreamAdminEntryOrderBy["rankDesc"] = "-rank";
    KalturaLiveStreamAdminEntryOrderBy["recentAsc"] = "+recent";
    KalturaLiveStreamAdminEntryOrderBy["recentDesc"] = "-recent";
    KalturaLiveStreamAdminEntryOrderBy["startDateAsc"] = "+startDate";
    KalturaLiveStreamAdminEntryOrderBy["startDateDesc"] = "-startDate";
    KalturaLiveStreamAdminEntryOrderBy["totalRankAsc"] = "+totalRank";
    KalturaLiveStreamAdminEntryOrderBy["totalRankDesc"] = "-totalRank";
    KalturaLiveStreamAdminEntryOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaLiveStreamAdminEntryOrderBy["updatedAtDesc"] = "-updatedAt";
    KalturaLiveStreamAdminEntryOrderBy["viewsAsc"] = "+views";
    KalturaLiveStreamAdminEntryOrderBy["viewsDesc"] = "-views";
    KalturaLiveStreamAdminEntryOrderBy["weightAsc"] = "+weight";
    KalturaLiveStreamAdminEntryOrderBy["weightDesc"] = "-weight";
})(KalturaLiveStreamAdminEntryOrderBy = exports.KalturaLiveStreamAdminEntryOrderBy || (exports.KalturaLiveStreamAdminEntryOrderBy = {}));
