// kaltura-node-typescript-client/api/types/KalturaESearchAbstractCategoryItem.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaESearchAbstractCategoryItem = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaESearchItemType_1 = require("./KalturaESearchItemType");
var KalturaESearchRange_1 = require("./KalturaESearchRange");
var KalturaESearchCategoryBaseItem_1 = require("./KalturaESearchCategoryBaseItem");
var KalturaESearchAbstractCategoryItem = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaESearchAbstractCategoryItem, _super);
    function KalturaESearchAbstractCategoryItem(data) {
        return _super.call(this, data) || this;
    }
    KalturaESearchAbstractCategoryItem.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaESearchAbstractCategoryItem' },
            searchTerm: { type: 's' },
            itemType: { type: 'en', subTypeConstructor: KalturaESearchItemType_1.KalturaESearchItemType, subType: 'KalturaESearchItemType' },
            range: { type: 'o', subTypeConstructor: KalturaESearchRange_1.KalturaESearchRange, subType: 'KalturaESearchRange' },
            addHighlight: { type: 'b' }
        });
        return result;
    };
    return KalturaESearchAbstractCategoryItem;
}(KalturaESearchCategoryBaseItem_1.KalturaESearchCategoryBaseItem));
exports.KalturaESearchAbstractCategoryItem = KalturaESearchAbstractCategoryItem;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaESearchAbstractCategoryItem', KalturaESearchAbstractCategoryItem);
