// kaltura-node-typescript-client/api/types/KalturaLanguageCode.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLanguageCode = void 0;
var KalturaLanguageCode;
(function (KalturaLanguageCode) {
    KalturaLanguageCode["aa"] = "aa";
    KalturaLanguageCode["ab"] = "ab";
    KalturaLanguageCode["af"] = "af";
    KalturaLanguageCode["alg"] = "alg";
    KalturaLanguageCode["am"] = "am";
    KalturaLanguageCode["ar"] = "ar";
    KalturaLanguageCode["as"] = "as";
    KalturaLanguageCode["ath"] = "ath";
    KalturaLanguageCode["ay"] = "ay";
    KalturaLanguageCode["az"] = "az";
    KalturaLanguageCode["ba"] = "ba";
    KalturaLanguageCode["be"] = "be";
    KalturaLanguageCode["bg"] = "bg";
    KalturaLanguageCode["bh"] = "bh";
    KalturaLanguageCode["bi"] = "bi";
    KalturaLanguageCode["bla"] = "bla";
    KalturaLanguageCode["bn"] = "bn";
    KalturaLanguageCode["bo"] = "bo";
    KalturaLanguageCode["br"] = "br";
    KalturaLanguageCode["ca"] = "ca";
    KalturaLanguageCode["co"] = "co";
    KalturaLanguageCode["cr"] = "cr";
    KalturaLanguageCode["crg"] = "crg";
    KalturaLanguageCode["cs"] = "cs";
    KalturaLanguageCode["cy"] = "cy";
    KalturaLanguageCode["da"] = "da";
    KalturaLanguageCode["de"] = "de";
    KalturaLanguageCode["dz"] = "dz";
    KalturaLanguageCode["el"] = "el";
    KalturaLanguageCode["en"] = "en";
    KalturaLanguageCode["enGb"] = "en_gb";
    KalturaLanguageCode["enUs"] = "en_us";
    KalturaLanguageCode["eo"] = "eo";
    KalturaLanguageCode["es"] = "es";
    KalturaLanguageCode["esXl"] = "es_xl";
    KalturaLanguageCode["et"] = "et";
    KalturaLanguageCode["eu"] = "eu";
    KalturaLanguageCode["fa"] = "fa";
    KalturaLanguageCode["fi"] = "fi";
    KalturaLanguageCode["fj"] = "fj";
    KalturaLanguageCode["fo"] = "fo";
    KalturaLanguageCode["fr"] = "fr";
    KalturaLanguageCode["frCa"] = "fr_ca";
    KalturaLanguageCode["fy"] = "fy";
    KalturaLanguageCode["ga"] = "ga";
    KalturaLanguageCode["gd"] = "gd";
    KalturaLanguageCode["gl"] = "gl";
    KalturaLanguageCode["gn"] = "gn";
    KalturaLanguageCode["gu"] = "gu";
    KalturaLanguageCode["gv"] = "gv";
    KalturaLanguageCode["ha"] = "ha";
    KalturaLanguageCode["he"] = "he";
    KalturaLanguageCode["hi"] = "hi";
    KalturaLanguageCode["hr"] = "hr";
    KalturaLanguageCode["hu"] = "hu";
    KalturaLanguageCode["hy"] = "hy";
    KalturaLanguageCode["ia"] = "ia";
    KalturaLanguageCode["id"] = "id";
    KalturaLanguageCode["ie"] = "ie";
    KalturaLanguageCode["ik"] = "ik";
    KalturaLanguageCode["ikt"] = "ikt";
    KalturaLanguageCode["in"] = "in";
    KalturaLanguageCode["iro"] = "iro";
    KalturaLanguageCode["is"] = "is";
    KalturaLanguageCode["it"] = "it";
    KalturaLanguageCode["iu"] = "iu";
    KalturaLanguageCode["iw"] = "iw";
    KalturaLanguageCode["ja"] = "ja";
    KalturaLanguageCode["ji"] = "ji";
    KalturaLanguageCode["jv"] = "jv";
    KalturaLanguageCode["ka"] = "ka";
    KalturaLanguageCode["kk"] = "kk";
    KalturaLanguageCode["kl"] = "kl";
    KalturaLanguageCode["km"] = "km";
    KalturaLanguageCode["kn"] = "kn";
    KalturaLanguageCode["ko"] = "ko";
    KalturaLanguageCode["ks"] = "ks";
    KalturaLanguageCode["ku"] = "ku";
    KalturaLanguageCode["ky"] = "ky";
    KalturaLanguageCode["la"] = "la";
    KalturaLanguageCode["lb"] = "lb";
    KalturaLanguageCode["li"] = "li";
    KalturaLanguageCode["ln"] = "ln";
    KalturaLanguageCode["lo"] = "lo";
    KalturaLanguageCode["lt"] = "lt";
    KalturaLanguageCode["lv"] = "lv";
    KalturaLanguageCode["mg"] = "mg";
    KalturaLanguageCode["mi"] = "mi";
    KalturaLanguageCode["mic"] = "mic";
    KalturaLanguageCode["mk"] = "mk";
    KalturaLanguageCode["ml"] = "ml";
    KalturaLanguageCode["mn"] = "mn";
    KalturaLanguageCode["mo"] = "mo";
    KalturaLanguageCode["moe"] = "moe";
    KalturaLanguageCode["mr"] = "mr";
    KalturaLanguageCode["ms"] = "ms";
    KalturaLanguageCode["mt"] = "mt";
    KalturaLanguageCode["mu"] = "mu";
    KalturaLanguageCode["my"] = "my";
    KalturaLanguageCode["na"] = "na";
    KalturaLanguageCode["ne"] = "ne";
    KalturaLanguageCode["nl"] = "nl";
    KalturaLanguageCode["no"] = "no";
    KalturaLanguageCode["oc"] = "oc";
    KalturaLanguageCode["oj"] = "oj";
    KalturaLanguageCode["ojs"] = "ojs";
    KalturaLanguageCode["ojw"] = "ojw";
    KalturaLanguageCode["oka"] = "oka";
    KalturaLanguageCode["om"] = "om";
    KalturaLanguageCode["or"] = "or";
    KalturaLanguageCode["pa"] = "pa";
    KalturaLanguageCode["pl"] = "pl";
    KalturaLanguageCode["ps"] = "ps";
    KalturaLanguageCode["pt"] = "pt";
    KalturaLanguageCode["ptBr"] = "pt_br";
    KalturaLanguageCode["qu"] = "qu";
    KalturaLanguageCode["rm"] = "rm";
    KalturaLanguageCode["rn"] = "rn";
    KalturaLanguageCode["ro"] = "ro";
    KalturaLanguageCode["ru"] = "ru";
    KalturaLanguageCode["rw"] = "rw";
    KalturaLanguageCode["sa"] = "sa";
    KalturaLanguageCode["sal"] = "sal";
    KalturaLanguageCode["sd"] = "sd";
    KalturaLanguageCode["sg"] = "sg";
    KalturaLanguageCode["sh"] = "sh";
    KalturaLanguageCode["si"] = "si";
    KalturaLanguageCode["sk"] = "sk";
    KalturaLanguageCode["sl"] = "sl";
    KalturaLanguageCode["sm"] = "sm";
    KalturaLanguageCode["smi"] = "smi";
    KalturaLanguageCode["sn"] = "sn";
    KalturaLanguageCode["so"] = "so";
    KalturaLanguageCode["sq"] = "sq";
    KalturaLanguageCode["sr"] = "sr";
    KalturaLanguageCode["ss"] = "ss";
    KalturaLanguageCode["st"] = "st";
    KalturaLanguageCode["su"] = "su";
    KalturaLanguageCode["sv"] = "sv";
    KalturaLanguageCode["sw"] = "sw";
    KalturaLanguageCode["ta"] = "ta";
    KalturaLanguageCode["tce"] = "tce";
    KalturaLanguageCode["te"] = "te";
    KalturaLanguageCode["tg"] = "tg";
    KalturaLanguageCode["th"] = "th";
    KalturaLanguageCode["ti"] = "ti";
    KalturaLanguageCode["tk"] = "tk";
    KalturaLanguageCode["tl"] = "tl";
    KalturaLanguageCode["tn"] = "tn";
    KalturaLanguageCode["to"] = "to";
    KalturaLanguageCode["tr"] = "tr";
    KalturaLanguageCode["ts"] = "ts";
    KalturaLanguageCode["tt"] = "tt";
    KalturaLanguageCode["tw"] = "tw";
    KalturaLanguageCode["ug"] = "ug";
    KalturaLanguageCode["uk"] = "uk";
    KalturaLanguageCode["un"] = "un";
    KalturaLanguageCode["ur"] = "ur";
    KalturaLanguageCode["uz"] = "uz";
    KalturaLanguageCode["vi"] = "vi";
    KalturaLanguageCode["vo"] = "vo";
    KalturaLanguageCode["wo"] = "wo";
    KalturaLanguageCode["xh"] = "xh";
    KalturaLanguageCode["yi"] = "yi";
    KalturaLanguageCode["yo"] = "yo";
    KalturaLanguageCode["zh"] = "zh";
    KalturaLanguageCode["zhTw"] = "zh_tw";
    KalturaLanguageCode["zu"] = "zu";
    KalturaLanguageCode["zxx"] = "zxx";
})(KalturaLanguageCode = exports.KalturaLanguageCode || (exports.KalturaLanguageCode = {}));
