// kaltura-node-typescript-client/api/types/KalturaDocumentEntryCompareAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDocumentEntryCompareAttribute = void 0;
var KalturaDocumentEntryCompareAttribute;
(function (KalturaDocumentEntryCompareAttribute) {
    KalturaDocumentEntryCompareAttribute["accessControlId"] = "accessControlId";
    KalturaDocumentEntryCompareAttribute["createdAt"] = "createdAt";
    KalturaDocumentEntryCompareAttribute["endDate"] = "endDate";
    KalturaDocumentEntryCompareAttribute["moderationCount"] = "moderationCount";
    KalturaDocumentEntryCompareAttribute["moderationStatus"] = "moderationStatus";
    KalturaDocumentEntryCompareAttribute["partnerId"] = "partnerId";
    KalturaDocumentEntryCompareAttribute["partnerSortValue"] = "partnerSortValue";
    KalturaDocumentEntryCompareAttribute["rank"] = "rank";
    KalturaDocumentEntryCompareAttribute["replacementStatus"] = "replacementStatus";
    KalturaDocumentEntryCompareAttribute["startDate"] = "startDate";
    KalturaDocumentEntryCompareAttribute["status"] = "status";
    KalturaDocumentEntryCompareAttribute["totalRank"] = "totalRank";
    KalturaDocumentEntryCompareAttribute["type"] = "type";
    KalturaDocumentEntryCompareAttribute["updatedAt"] = "updatedAt";
})(KalturaDocumentEntryCompareAttribute = exports.KalturaDocumentEntryCompareAttribute || (exports.KalturaDocumentEntryCompareAttribute = {}));
