// kaltura-node-typescript-client/api/types/KalturaUploadToken.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUploadToken = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUploadTokenStatus_1 = require("./KalturaUploadTokenStatus");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaUploadToken = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUploadToken, _super);
    function KalturaUploadToken(data) {
        return _super.call(this, data) || this;
    }
    KalturaUploadToken.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUploadToken' },
            id: { type: 's', readOnly: true },
            partnerId: { type: 'n', readOnly: true },
            userId: { type: 's', readOnly: true },
            status: { type: 'en', readOnly: true, subTypeConstructor: KalturaUploadTokenStatus_1.KalturaUploadTokenStatus, subType: 'KalturaUploadTokenStatus' },
            fileName: { type: 's' },
            fileSize: { type: 'n' },
            uploadedFileSize: { type: 'n', readOnly: true },
            createdAt: { type: 'd', readOnly: true },
            updatedAt: { type: 'd', readOnly: true },
            uploadUrl: { type: 's', readOnly: true },
            autoFinalize: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            attachedObjectType: { type: 's', readOnly: true },
            attachedObjectId: { type: 's', readOnly: true }
        });
        return result;
    };
    return KalturaUploadToken;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaUploadToken = KalturaUploadToken;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUploadToken', KalturaUploadToken);
