// kaltura-node-typescript-client/api/types/KalturaMediaEntry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMediaEntry = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaMediaType_1 = require("./KalturaMediaType");
var KalturaSourceType_1 = require("./KalturaSourceType");
var KalturaSearchProviderType_1 = require("./KalturaSearchProviderType");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaStreamContainer_1 = require("./KalturaStreamContainer");
var KalturaPlayableEntry_1 = require("./KalturaPlayableEntry");
var KalturaMediaEntry = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMediaEntry, _super);
    function KalturaMediaEntry(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.streams === 'undefined')
            _this.streams = [];
        return _this;
    }
    KalturaMediaEntry.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMediaEntry' },
            mediaType: { type: 'en', subTypeConstructor: KalturaMediaType_1.KalturaMediaType, subType: 'KalturaMediaType' },
            conversionQuality: { type: 's' },
            sourceType: { type: 'es', subTypeConstructor: KalturaSourceType_1.KalturaSourceType, subType: 'KalturaSourceType' },
            sourceVersion: { type: 's' },
            searchProviderType: { type: 'en', subTypeConstructor: KalturaSearchProviderType_1.KalturaSearchProviderType, subType: 'KalturaSearchProviderType' },
            searchProviderId: { type: 's' },
            creditUserName: { type: 's' },
            creditUrl: { type: 's' },
            mediaDate: { type: 'd', readOnly: true },
            dataUrl: { type: 's', readOnly: true },
            flavorParamsIds: { type: 's', readOnly: true },
            isTrimDisabled: { type: 'en', readOnly: true, subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            streams: { type: 'a', subTypeConstructor: KalturaStreamContainer_1.KalturaStreamContainer, subType: 'KalturaStreamContainer' }
        });
        return result;
    };
    return KalturaMediaEntry;
}(KalturaPlayableEntry_1.KalturaPlayableEntry));
exports.KalturaMediaEntry = KalturaMediaEntry;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMediaEntry', KalturaMediaEntry);
