// kaltura-node-typescript-client/api/types/KalturaMicrosoftTeamsDropFolder.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMicrosoftTeamsDropFolder = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaRemoteDropFolder_1 = require("./KalturaRemoteDropFolder");
var KalturaMicrosoftTeamsDropFolder = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMicrosoftTeamsDropFolder, _super);
    function KalturaMicrosoftTeamsDropFolder(data) {
        return _super.call(this, data) || this;
    }
    KalturaMicrosoftTeamsDropFolder.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMicrosoftTeamsDropFolder' },
            integrationId: { type: 'n' },
            tenantId: { type: 's', readOnly: true },
            clientSecret: { type: 's', readOnly: true },
            clientId: { type: 's', readOnly: true }
        });
        return result;
    };
    return KalturaMicrosoftTeamsDropFolder;
}(KalturaRemoteDropFolder_1.KalturaRemoteDropFolder));
exports.KalturaMicrosoftTeamsDropFolder = KalturaMicrosoftTeamsDropFolder;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMicrosoftTeamsDropFolder', KalturaMicrosoftTeamsDropFolder);
