// kaltura-node-typescript-client/api/types/KalturaGroupUserBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaGroupUserBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaGroupUserStatus_1 = require("./KalturaGroupUserStatus");
var KalturaRelatedFilter_1 = require("./KalturaRelatedFilter");
var KalturaGroupUserBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaGroupUserBaseFilter, _super);
    function KalturaGroupUserBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaGroupUserBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaGroupUserBaseFilter' },
            userIdEqual: { type: 's' },
            userIdIn: { type: 's' },
            groupIdEqual: { type: 's' },
            groupIdIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaGroupUserStatus_1.KalturaGroupUserStatus, subType: 'KalturaGroupUserStatus' },
            statusIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' }
        });
        return result;
    };
    return KalturaGroupUserBaseFilter;
}(KalturaRelatedFilter_1.KalturaRelatedFilter));
exports.KalturaGroupUserBaseFilter = KalturaGroupUserBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaGroupUserBaseFilter', KalturaGroupUserBaseFilter);
