// kaltura-node-typescript-client/api/types/KalturaPushToNewsDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPushToNewsDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDistributionProtocol_1 = require("./KalturaDistributionProtocol");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaPushToNewsDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPushToNewsDistributionProfile, _super);
    function KalturaPushToNewsDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaPushToNewsDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPushToNewsDistributionProfile' },
            protocol: { type: 'en', subTypeConstructor: KalturaDistributionProtocol_1.KalturaDistributionProtocol, subType: 'KalturaDistributionProtocol' },
            host: { type: 's' },
            ips: { type: 's' },
            port: { type: 'n' },
            basePath: { type: 's' },
            username: { type: 's' },
            password: { type: 's' },
            namedItem: { type: 's' },
            certificateKey: { type: 's' },
            bodyXslt: { type: 's' },
            recentNewsTimeInterval: { type: 'n' }
        });
        return result;
    };
    return KalturaPushToNewsDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaPushToNewsDistributionProfile = KalturaPushToNewsDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPushToNewsDistributionProfile', KalturaPushToNewsDistributionProfile);
