// kaltura-node-typescript-client/api/types/KalturaGenericXsltSyndicationFeedOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaGenericXsltSyndicationFeedOrderBy = void 0;
var KalturaGenericXsltSyndicationFeedOrderBy;
(function (KalturaGenericXsltSyndicationFeedOrderBy) {
    KalturaGenericXsltSyndicationFeedOrderBy["createdAtAsc"] = "+createdAt";
    KalturaGenericXsltSyndicationFeedOrderBy["createdAtDesc"] = "-createdAt";
    KalturaGenericXsltSyndicationFeedOrderBy["nameAsc"] = "+name";
    KalturaGenericXsltSyndicationFeedOrderBy["nameDesc"] = "-name";
    KalturaGenericXsltSyndicationFeedOrderBy["playlistIdAsc"] = "+playlistId";
    KalturaGenericXsltSyndicationFeedOrderBy["playlistIdDesc"] = "-playlistId";
    KalturaGenericXsltSyndicationFeedOrderBy["typeAsc"] = "+type";
    KalturaGenericXsltSyndicationFeedOrderBy["typeDesc"] = "-type";
    KalturaGenericXsltSyndicationFeedOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaGenericXsltSyndicationFeedOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaGenericXsltSyndicationFeedOrderBy = exports.KalturaGenericXsltSyndicationFeedOrderBy || (exports.KalturaGenericXsltSyndicationFeedOrderBy = {}));
