// kaltura-node-typescript-client/api/types/KalturaFreewheelGenericDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFreewheelGenericDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaFreewheelGenericDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaFreewheelGenericDistributionProfile, _super);
    function KalturaFreewheelGenericDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaFreewheelGenericDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaFreewheelGenericDistributionProfile' },
            apikey: { type: 's' },
            email: { type: 's' },
            sftpPass: { type: 's' },
            sftpLogin: { type: 's' },
            contentOwner: { type: 's' },
            upstreamVideoId: { type: 's' },
            upstreamNetworkName: { type: 's' },
            upstreamNetworkId: { type: 's' },
            categoryId: { type: 's' },
            replaceGroup: { type: 'b' },
            replaceAirDates: { type: 'b' }
        });
        return result;
    };
    return KalturaFreewheelGenericDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaFreewheelGenericDistributionProfile = KalturaFreewheelGenericDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaFreewheelGenericDistributionProfile', KalturaFreewheelGenericDistributionProfile);
