// kaltura-node-typescript-client/api/types/KalturaMixEntryMatchAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMixEntryMatchAttribute = void 0;
var KalturaMixEntryMatchAttribute;
(function (KalturaMixEntryMatchAttribute) {
    KalturaMixEntryMatchAttribute["adminTags"] = "adminTags";
    KalturaMixEntryMatchAttribute["categoriesIds"] = "categoriesIds";
    KalturaMixEntryMatchAttribute["creatorId"] = "creatorId";
    KalturaMixEntryMatchAttribute["description"] = "description";
    KalturaMixEntryMatchAttribute["durationType"] = "durationType";
    KalturaMixEntryMatchAttribute["groupId"] = "groupId";
    KalturaMixEntryMatchAttribute["id"] = "id";
    KalturaMixEntryMatchAttribute["_name"] = "name";
    KalturaMixEntryMatchAttribute["referenceId"] = "referenceId";
    KalturaMixEntryMatchAttribute["replacedEntryId"] = "replacedEntryId";
    KalturaMixEntryMatchAttribute["replacingEntryId"] = "replacingEntryId";
    KalturaMixEntryMatchAttribute["searchText"] = "searchText";
    KalturaMixEntryMatchAttribute["tags"] = "tags";
    KalturaMixEntryMatchAttribute["userId"] = "userId";
})(KalturaMixEntryMatchAttribute = exports.KalturaMixEntryMatchAttribute || (exports.KalturaMixEntryMatchAttribute = {}));
