// kaltura-node-typescript-client/api/types/KalturaCaptionAssetItemFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCaptionAssetItemFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaLanguage_1 = require("./KalturaLanguage");
var KalturaCaptionAssetFilter_1 = require("./KalturaCaptionAssetFilter");
var KalturaCaptionAssetItemFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCaptionAssetItemFilter, _super);
    function KalturaCaptionAssetItemFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaCaptionAssetItemFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCaptionAssetItemFilter' },
            contentLike: { type: 's' },
            contentMultiLikeOr: { type: 's' },
            contentMultiLikeAnd: { type: 's' },
            partnerDescriptionLike: { type: 's' },
            partnerDescriptionMultiLikeOr: { type: 's' },
            partnerDescriptionMultiLikeAnd: { type: 's' },
            languageEqual: { type: 'es', subTypeConstructor: KalturaLanguage_1.KalturaLanguage, subType: 'KalturaLanguage' },
            languageIn: { type: 's' },
            labelEqual: { type: 's' },
            labelIn: { type: 's' },
            startTimeGreaterThanOrEqual: { type: 'n' },
            startTimeLessThanOrEqual: { type: 'n' },
            endTimeGreaterThanOrEqual: { type: 'n' },
            endTimeLessThanOrEqual: { type: 'n' }
        });
        return result;
    };
    return KalturaCaptionAssetItemFilter;
}(KalturaCaptionAssetFilter_1.KalturaCaptionAssetFilter));
exports.KalturaCaptionAssetItemFilter = KalturaCaptionAssetItemFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCaptionAssetItemFilter', KalturaCaptionAssetItemFilter);
