// kaltura-node-typescript-client/api/types/KalturaAccessControlScope.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAccessControlScope = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaAccessControlContextTypeHolder_1 = require("./KalturaAccessControlContextTypeHolder");
var KalturaKeyValue_1 = require("./KalturaKeyValue");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaAccessControlScope = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAccessControlScope, _super);
    function KalturaAccessControlScope(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.contexts === 'undefined')
            _this.contexts = [];
        if (typeof _this.hashes === 'undefined')
            _this.hashes = [];
        return _this;
    }
    KalturaAccessControlScope.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAccessControlScope' },
            referrer: { type: 's' },
            ip: { type: 's' },
            ks: { type: 's' },
            userAgent: { type: 's' },
            time: { type: 'd' },
            contexts: { type: 'a', subTypeConstructor: KalturaAccessControlContextTypeHolder_1.KalturaAccessControlContextTypeHolder, subType: 'KalturaAccessControlContextTypeHolder' },
            hashes: { type: 'a', subTypeConstructor: KalturaKeyValue_1.KalturaKeyValue, subType: 'KalturaKeyValue' }
        });
        return result;
    };
    return KalturaAccessControlScope;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaAccessControlScope = KalturaAccessControlScope;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAccessControlScope', KalturaAccessControlScope);
