// kaltura-node-typescript-client/api/types/KalturaUploadTokenBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUploadTokenBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUploadTokenStatus_1 = require("./KalturaUploadTokenStatus");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaUploadTokenBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUploadTokenBaseFilter, _super);
    function KalturaUploadTokenBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaUploadTokenBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUploadTokenBaseFilter' },
            idEqual: { type: 's' },
            idIn: { type: 's' },
            userIdEqual: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaUploadTokenStatus_1.KalturaUploadTokenStatus, subType: 'KalturaUploadTokenStatus' },
            statusIn: { type: 's' },
            fileNameEqual: { type: 's' },
            fileSizeEqual: { type: 'n' }
        });
        return result;
    };
    return KalturaUploadTokenBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaUploadTokenBaseFilter = KalturaUploadTokenBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUploadTokenBaseFilter', KalturaUploadTokenBaseFilter);
