// kaltura-node-typescript-client/api/types/KalturaLiveStreamParams.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveStreamParams = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaLiveStreamParams = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveStreamParams, _super);
    function KalturaLiveStreamParams(data) {
        return _super.call(this, data) || this;
    }
    KalturaLiveStreamParams.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveStreamParams' },
            bitrate: { type: 'n' },
            flavorId: { type: 's' },
            width: { type: 'n' },
            height: { type: 'n' },
            codec: { type: 's' },
            frameRate: { type: 'n' },
            keyFrameInterval: { type: 'n' },
            language: { type: 's' }
        });
        return result;
    };
    return KalturaLiveStreamParams;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaLiveStreamParams = KalturaLiveStreamParams;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveStreamParams', KalturaLiveStreamParams);
