// kaltura-node-typescript-client/api/types/KalturaUserBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaUserBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUserType_1 = require("./KalturaUserType");
var KalturaNullableBoolean_1 = require("./KalturaNullableBoolean");
var KalturaBaseUserFilter_1 = require("./KalturaBaseUserFilter");
var KalturaUserBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaUserBaseFilter, _super);
    function KalturaUserBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaUserBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaUserBaseFilter' },
            idEqual: { type: 's' },
            idIn: { type: 's' },
            typeEqual: { type: 'en', subTypeConstructor: KalturaUserType_1.KalturaUserType, subType: 'KalturaUserType' },
            typeIn: { type: 's' },
            isAdminEqual: { type: 'en', subTypeConstructor: KalturaNullableBoolean_1.KalturaNullableBoolean, subType: 'KalturaNullableBoolean' },
            firstNameStartsWith: { type: 's' },
            lastNameStartsWith: { type: 's' }
        });
        return result;
    };
    return KalturaUserBaseFilter;
}(KalturaBaseUserFilter_1.KalturaBaseUserFilter));
exports.KalturaUserBaseFilter = KalturaUserBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaUserBaseFilter', KalturaUserBaseFilter);
