// kaltura-node-typescript-client/api/types/KalturaVendorCaptionsCatalogItemBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaVendorCaptionsCatalogItemBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCatalogItemLanguage_1 = require("./KalturaCatalogItemLanguage");
var KalturaVendorCatalogItemOutputFormat_1 = require("./KalturaVendorCatalogItemOutputFormat");
var KalturaVendorCatalogItemFilter_1 = require("./KalturaVendorCatalogItemFilter");
var KalturaVendorCaptionsCatalogItemBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaVendorCaptionsCatalogItemBaseFilter, _super);
    function KalturaVendorCaptionsCatalogItemBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaVendorCaptionsCatalogItemBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaVendorCaptionsCatalogItemBaseFilter' },
            sourceLanguageEqual: { type: 'es', subTypeConstructor: KalturaCatalogItemLanguage_1.KalturaCatalogItemLanguage, subType: 'KalturaCatalogItemLanguage' },
            sourceLanguageIn: { type: 's' },
            outputFormatEqual: { type: 'en', subTypeConstructor: KalturaVendorCatalogItemOutputFormat_1.KalturaVendorCatalogItemOutputFormat, subType: 'KalturaVendorCatalogItemOutputFormat' },
            outputFormatIn: { type: 's' }
        });
        return result;
    };
    return KalturaVendorCaptionsCatalogItemBaseFilter;
}(KalturaVendorCatalogItemFilter_1.KalturaVendorCatalogItemFilter));
exports.KalturaVendorCaptionsCatalogItemBaseFilter = KalturaVendorCaptionsCatalogItemBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaVendorCaptionsCatalogItemBaseFilter', KalturaVendorCaptionsCatalogItemBaseFilter);
