// kaltura-node-typescript-client/api/types/KalturaAuditTrailFileSyncCreateInfo.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAuditTrailFileSyncCreateInfo = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaAuditTrailFileSyncType_1 = require("./KalturaAuditTrailFileSyncType");
var KalturaAuditTrailInfo_1 = require("./KalturaAuditTrailInfo");
var KalturaAuditTrailFileSyncCreateInfo = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAuditTrailFileSyncCreateInfo, _super);
    function KalturaAuditTrailFileSyncCreateInfo(data) {
        return _super.call(this, data) || this;
    }
    KalturaAuditTrailFileSyncCreateInfo.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAuditTrailFileSyncCreateInfo' },
            version: { type: 's' },
            objectSubType: { type: 'n' },
            dc: { type: 'n' },
            original: { type: 'b' },
            fileType: { type: 'en', subTypeConstructor: KalturaAuditTrailFileSyncType_1.KalturaAuditTrailFileSyncType, subType: 'KalturaAuditTrailFileSyncType' }
        });
        return result;
    };
    return KalturaAuditTrailFileSyncCreateInfo;
}(KalturaAuditTrailInfo_1.KalturaAuditTrailInfo));
exports.KalturaAuditTrailFileSyncCreateInfo = KalturaAuditTrailFileSyncCreateInfo;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAuditTrailFileSyncCreateInfo', KalturaAuditTrailFileSyncCreateInfo);
