// kaltura-node-typescript-client/api/types/KalturaSearch.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaSearch = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaSearchProviderType_1 = require("./KalturaSearchProviderType");
var KalturaMediaType_1 = require("./KalturaMediaType");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaSearch = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaSearch, _super);
    function KalturaSearch(data) {
        return _super.call(this, data) || this;
    }
    KalturaSearch.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaSearch' },
            keyWords: { type: 's' },
            searchSource: { type: 'en', subTypeConstructor: KalturaSearchProviderType_1.KalturaSearchProviderType, subType: 'KalturaSearchProviderType' },
            mediaType: { type: 'en', subTypeConstructor: KalturaMediaType_1.KalturaMediaType, subType: 'KalturaMediaType' },
            extraData: { type: 's' },
            authData: { type: 's' }
        });
        return result;
    };
    return KalturaSearch;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaSearch = KalturaSearch;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaSearch', KalturaSearch);
