// kaltura-node-typescript-client/api/types/KalturaCaptionAssetBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaCaptionAssetBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaCaptionType_1 = require("./KalturaCaptionType");
var KalturaCaptionAssetStatus_1 = require("./KalturaCaptionAssetStatus");
var KalturaAssetFilter_1 = require("./KalturaAssetFilter");
var KalturaCaptionAssetBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaCaptionAssetBaseFilter, _super);
    function KalturaCaptionAssetBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaCaptionAssetBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaCaptionAssetBaseFilter' },
            captionParamsIdEqual: { type: 'n' },
            captionParamsIdIn: { type: 's' },
            formatEqual: { type: 'es', subTypeConstructor: KalturaCaptionType_1.KalturaCaptionType, subType: 'KalturaCaptionType' },
            formatIn: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaCaptionAssetStatus_1.KalturaCaptionAssetStatus, subType: 'KalturaCaptionAssetStatus' },
            statusIn: { type: 's' },
            statusNotIn: { type: 's' }
        });
        return result;
    };
    return KalturaCaptionAssetBaseFilter;
}(KalturaAssetFilter_1.KalturaAssetFilter));
exports.KalturaCaptionAssetBaseFilter = KalturaCaptionAssetBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaCaptionAssetBaseFilter', KalturaCaptionAssetBaseFilter);
