// kaltura-node-typescript-client/api/types/KalturaImageFlavorParams.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaImageFlavorParams = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaFlavorParams_1 = require("./KalturaFlavorParams");
var KalturaImageFlavorParams = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaImageFlavorParams, _super);
    function KalturaImageFlavorParams(data) {
        return _super.call(this, data) || this;
    }
    KalturaImageFlavorParams.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaImageFlavorParams' },
            densityWidth: { type: 'n' },
            densityHeight: { type: 'n' },
            sizeWidth: { type: 'n' },
            sizeHeight: { type: 'n' },
            depth: { type: 'n' }
        });
        return result;
    };
    return KalturaImageFlavorParams;
}(KalturaFlavorParams_1.KalturaFlavorParams));
exports.KalturaImageFlavorParams = KalturaImageFlavorParams;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaImageFlavorParams', KalturaImageFlavorParams);
