// kaltura-node-typescript-client/api/types/KalturaConcatJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaConcatJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaString_1 = require("./KalturaString");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaConcatJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaConcatJobData, _super);
    function KalturaConcatJobData(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.srcFiles === 'undefined')
            _this.srcFiles = [];
        if (typeof _this.conversionCommands === 'undefined')
            _this.conversionCommands = [];
        return _this;
    }
    KalturaConcatJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaConcatJobData' },
            srcFiles: { type: 'a', subTypeConstructor: KalturaString_1.KalturaString, subType: 'KalturaString' },
            destFilePath: { type: 's' },
            flavorAssetId: { type: 's' },
            offset: { type: 'n' },
            duration: { type: 'n' },
            concatenatedDuration: { type: 'n' },
            shouldSort: { type: 'b' },
            conversionCommands: { type: 'a', subTypeConstructor: KalturaString_1.KalturaString, subType: 'KalturaString' },
            multiSource: { type: 'b' }
        });
        return result;
    };
    return KalturaConcatJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaConcatJobData = KalturaConcatJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaConcatJobData', KalturaConcatJobData);
