// kaltura-node-typescript-client/api/types/KalturaViewHistoryUserEntry.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaViewHistoryUserEntry = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaUserEntry_1 = require("./KalturaUserEntry");
var KalturaViewHistoryUserEntry = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaViewHistoryUserEntry, _super);
    function KalturaViewHistoryUserEntry(data) {
        return _super.call(this, data) || this;
    }
    KalturaViewHistoryUserEntry.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaViewHistoryUserEntry' },
            playbackContext: { type: 's' },
            lastTimeReached: { type: 'n' },
            lastUpdateTime: { type: 'd' },
            playlistLastEntryId: { type: 's' }
        });
        return result;
    };
    return KalturaViewHistoryUserEntry;
}(KalturaUserEntry_1.KalturaUserEntry));
exports.KalturaViewHistoryUserEntry = KalturaViewHistoryUserEntry;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaViewHistoryUserEntry', KalturaViewHistoryUserEntry);
