// kaltura-node-typescript-client/api/types/KalturaITunesSyndicationFeedCategories.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaITunesSyndicationFeedCategories = void 0;
var KalturaITunesSyndicationFeedCategories;
(function (KalturaITunesSyndicationFeedCategories) {
    KalturaITunesSyndicationFeedCategories["arts"] = "Arts";
    KalturaITunesSyndicationFeedCategories["artsDesign"] = "Arts/Design";
    KalturaITunesSyndicationFeedCategories["artsFashionBeauty"] = "Arts/Fashion &amp; Beauty";
    KalturaITunesSyndicationFeedCategories["artsFood"] = "Arts/Food";
    KalturaITunesSyndicationFeedCategories["artsLiterature"] = "Arts/Literature";
    KalturaITunesSyndicationFeedCategories["artsPerformingArts"] = "Arts/Performing Arts";
    KalturaITunesSyndicationFeedCategories["artsVisualArts"] = "Arts/Visual Arts";
    KalturaITunesSyndicationFeedCategories["business"] = "Business";
    KalturaITunesSyndicationFeedCategories["businessBusinessNews"] = "Business/Business News";
    KalturaITunesSyndicationFeedCategories["businessCareers"] = "Business/Careers";
    KalturaITunesSyndicationFeedCategories["businessInvesting"] = "Business/Investing";
    KalturaITunesSyndicationFeedCategories["businessManagementMarketing"] = "Business/Management &amp; Marketing";
    KalturaITunesSyndicationFeedCategories["businessShopping"] = "Business/Shopping";
    KalturaITunesSyndicationFeedCategories["comedy"] = "Comedy";
    KalturaITunesSyndicationFeedCategories["education"] = "Education";
    KalturaITunesSyndicationFeedCategories["educationHigherEducation"] = "Education/Higher Education";
    KalturaITunesSyndicationFeedCategories["educationK12"] = "Education/K-12";
    KalturaITunesSyndicationFeedCategories["educationLanguageCourses"] = "Education/Language Courses";
    KalturaITunesSyndicationFeedCategories["educationTechnology"] = "Education/Education Technology";
    KalturaITunesSyndicationFeedCategories["educationTraining"] = "Education/Training";
    KalturaITunesSyndicationFeedCategories["gamesHobbies"] = "Games &amp; Hobbies";
    KalturaITunesSyndicationFeedCategories["gamesHobbiesAutomotive"] = "Games &amp; Hobbies/Automotive";
    KalturaITunesSyndicationFeedCategories["gamesHobbiesAviation"] = "Games &amp; Hobbies/Aviation";
    KalturaITunesSyndicationFeedCategories["gamesHobbiesHobbies"] = "Games &amp; Hobbies/Hobbies";
    KalturaITunesSyndicationFeedCategories["gamesHobbiesOtherGames"] = "Games &amp; Hobbies/Other Games";
    KalturaITunesSyndicationFeedCategories["gamesHobbiesVideoGames"] = "Games &amp; Hobbies/Video Games";
    KalturaITunesSyndicationFeedCategories["governmentOrganizations"] = "Government &amp; Organizations";
    KalturaITunesSyndicationFeedCategories["governmentOrganizationsLocal"] = "Government &amp; Organizations/Local";
    KalturaITunesSyndicationFeedCategories["governmentOrganizationsNational"] = "Government &amp; Organizations/National";
    KalturaITunesSyndicationFeedCategories["governmentOrganizationsNonProfit"] = "Government &amp; Organizations/Non-Profit";
    KalturaITunesSyndicationFeedCategories["governmentOrganizationsRegional"] = "Government &amp; Organizations/Regional";
    KalturaITunesSyndicationFeedCategories["health"] = "Health";
    KalturaITunesSyndicationFeedCategories["healthAlternativeHealth"] = "Health/Alternative Health";
    KalturaITunesSyndicationFeedCategories["healthFitnessNutrition"] = "Health/Fitness &amp; Nutrition";
    KalturaITunesSyndicationFeedCategories["healthSelfHelp"] = "Health/Self-Help";
    KalturaITunesSyndicationFeedCategories["healthSexuality"] = "Health/Sexuality";
    KalturaITunesSyndicationFeedCategories["kidsFamily"] = "Kids &amp; Family";
    KalturaITunesSyndicationFeedCategories["music"] = "Music";
    KalturaITunesSyndicationFeedCategories["newsPolitics"] = "News &amp; Politics";
    KalturaITunesSyndicationFeedCategories["religionSpirituality"] = "Religion &amp; Spirituality";
    KalturaITunesSyndicationFeedCategories["religionSpiritualityBuddhism"] = "Religion &amp; Spirituality/Buddhism";
    KalturaITunesSyndicationFeedCategories["religionSpiritualityChristianity"] = "Religion &amp; Spirituality/Christianity";
    KalturaITunesSyndicationFeedCategories["religionSpiritualityHinduism"] = "Religion &amp; Spirituality/Hinduism";
    KalturaITunesSyndicationFeedCategories["religionSpiritualityIslam"] = "Religion &amp; Spirituality/Islam";
    KalturaITunesSyndicationFeedCategories["religionSpiritualityJudaism"] = "Religion &amp; Spirituality/Judaism";
    KalturaITunesSyndicationFeedCategories["religionSpiritualityOther"] = "Religion &amp; Spirituality/Other";
    KalturaITunesSyndicationFeedCategories["religionSpiritualitySpirituality"] = "Religion &amp; Spirituality/Spirituality";
    KalturaITunesSyndicationFeedCategories["scienceMedicine"] = "Science &amp; Medicine";
    KalturaITunesSyndicationFeedCategories["scienceMedicineMedicine"] = "Science &amp; Medicine/Medicine";
    KalturaITunesSyndicationFeedCategories["scienceMedicineNaturalSciences"] = "Science &amp; Medicine/Natural Sciences";
    KalturaITunesSyndicationFeedCategories["scienceMedicineSocialSciences"] = "Science &amp; Medicine/Social Sciences";
    KalturaITunesSyndicationFeedCategories["societyCulture"] = "Society &amp; Culture";
    KalturaITunesSyndicationFeedCategories["societyCultureHistory"] = "Society &amp; Culture/History";
    KalturaITunesSyndicationFeedCategories["societyCulturePersonalJournals"] = "Society &amp; Culture/Personal Journals";
    KalturaITunesSyndicationFeedCategories["societyCulturePhilosophy"] = "Society &amp; Culture/Philosophy";
    KalturaITunesSyndicationFeedCategories["societyCulturePlacesTravel"] = "Society &amp; Culture/Places &amp; Travel";
    KalturaITunesSyndicationFeedCategories["sportsRecreation"] = "Sports &amp; Recreation";
    KalturaITunesSyndicationFeedCategories["sportsRecreationAmateur"] = "Sports &amp; Recreation/Amateur";
    KalturaITunesSyndicationFeedCategories["sportsRecreationCollegeHighSchool"] = "Sports &amp; Recreation/College &amp; High School";
    KalturaITunesSyndicationFeedCategories["sportsRecreationOutdoor"] = "Sports &amp; Recreation/Outdoor";
    KalturaITunesSyndicationFeedCategories["sportsRecreationProfessional"] = "Sports &amp; Recreation/Professional";
    KalturaITunesSyndicationFeedCategories["technology"] = "Technology";
    KalturaITunesSyndicationFeedCategories["technologyGadgets"] = "Technology/Gadgets";
    KalturaITunesSyndicationFeedCategories["technologyPodcasting"] = "Technology/Podcasting";
    KalturaITunesSyndicationFeedCategories["technologySoftwareHowTo"] = "Technology/Software How-To";
    KalturaITunesSyndicationFeedCategories["technologyTechNews"] = "Technology/Tech News";
    KalturaITunesSyndicationFeedCategories["tvFilm"] = "TV &amp; Film";
})(KalturaITunesSyndicationFeedCategories = exports.KalturaITunesSyndicationFeedCategories || (exports.KalturaITunesSyndicationFeedCategories = {}));
