// kaltura-node-typescript-client/api/types/KalturaTagFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaTagFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaTaggedObjectType_1 = require("./KalturaTaggedObjectType");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaTagFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaTagFilter, _super);
    function KalturaTagFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaTagFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaTagFilter' },
            objectTypeEqual: { type: 'es', subTypeConstructor: KalturaTaggedObjectType_1.KalturaTaggedObjectType, subType: 'KalturaTaggedObjectType' },
            objectTypeIn: { type: 's' },
            tagEqual: { type: 's' },
            tagStartsWith: { type: 's' },
            instanceCountEqual: { type: 'n' },
            instanceCountIn: { type: 'n' }
        });
        return result;
    };
    return KalturaTagFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaTagFilter = KalturaTagFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaTagFilter', KalturaTagFilter);
