// kaltura-node-typescript-client/api/types/KalturaFtpDistributionFile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaFtpDistributionFile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaFtpDistributionFile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaFtpDistributionFile, _super);
    function KalturaFtpDistributionFile(data) {
        return _super.call(this, data) || this;
    }
    KalturaFtpDistributionFile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaFtpDistributionFile' },
            assetId: { type: 's' },
            filename: { type: 's' },
            contents: { type: 's' },
            localFilePath: { type: 's' },
            version: { type: 's' },
            hash: { type: 's' }
        });
        return result;
    };
    return KalturaFtpDistributionFile;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaFtpDistributionFile = KalturaFtpDistributionFile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaFtpDistributionFile', KalturaFtpDistributionFile);
