// kaltura-node-typescript-client/api/types/KalturaStreamContainer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaStreamContainer = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaStreamContainer = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaStreamContainer, _super);
    function KalturaStreamContainer(data) {
        return _super.call(this, data) || this;
    }
    KalturaStreamContainer.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaStreamContainer' },
            id: { type: 's' },
            type: { type: 's' },
            trackIndex: { type: 'n' },
            language: { type: 's' },
            channelIndex: { type: 'n' },
            label: { type: 's' },
            channelLayout: { type: 's' }
        });
        return result;
    };
    return KalturaStreamContainer;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaStreamContainer = KalturaStreamContainer;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaStreamContainer', KalturaStreamContainer);
