// kaltura-node-typescript-client/api/types/KalturaAttUverseDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAttUverseDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaAttUverseDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAttUverseDistributionProfile, _super);
    function KalturaAttUverseDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaAttUverseDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAttUverseDistributionProfile' },
            feedUrl: { type: 's', readOnly: true },
            ftpHost: { type: 's' },
            ftpUsername: { type: 's' },
            ftpPassword: { type: 's' },
            ftpPath: { type: 's' },
            channelTitle: { type: 's' },
            flavorAssetFilenameXslt: { type: 's' },
            thumbnailAssetFilenameXslt: { type: 's' },
            assetFilenameXslt: { type: 's' }
        });
        return result;
    };
    return KalturaAttUverseDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaAttUverseDistributionProfile = KalturaAttUverseDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAttUverseDistributionProfile', KalturaAttUverseDistributionProfile);
