// kaltura-node-typescript-client/api/types/KalturaPartnerUsage.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaPartnerUsage = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaPartnerUsage = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaPartnerUsage, _super);
    function KalturaPartnerUsage(data) {
        return _super.call(this, data) || this;
    }
    KalturaPartnerUsage.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaPartnerUsage' },
            hostingGB: { type: 'n', readOnly: true },
            Percent: { type: 'n', readOnly: true },
            packageBW: { type: 'n', readOnly: true },
            usageGB: { type: 'n', readOnly: true },
            reachedLimitDate: { type: 'n', readOnly: true },
            usageGraph: { type: 's', readOnly: true }
        });
        return result;
    };
    return KalturaPartnerUsage;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaPartnerUsage = KalturaPartnerUsage;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaPartnerUsage', KalturaPartnerUsage);
