// kaltura-node-typescript-client/api/types/KalturaMicrosoftTeamsDropFolderFile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMicrosoftTeamsDropFolderFile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDropFolderFile_1 = require("./KalturaDropFolderFile");
var KalturaMicrosoftTeamsDropFolderFile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMicrosoftTeamsDropFolderFile, _super);
    function KalturaMicrosoftTeamsDropFolderFile(data) {
        return _super.call(this, data) || this;
    }
    KalturaMicrosoftTeamsDropFolderFile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMicrosoftTeamsDropFolderFile' },
            remoteId: { type: 's' },
            ownerId: { type: 's' },
            additionalUserIds: { type: 's' },
            description: { type: 's' },
            targetCategoryIds: { type: 's' },
            name: { type: 's' },
            contentUrl: { type: 's' }
        });
        return result;
    };
    return KalturaMicrosoftTeamsDropFolderFile;
}(KalturaDropFolderFile_1.KalturaDropFolderFile));
exports.KalturaMicrosoftTeamsDropFolderFile = KalturaMicrosoftTeamsDropFolderFile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMicrosoftTeamsDropFolderFile', KalturaMicrosoftTeamsDropFolderFile);
