// kaltura-node-typescript-client/api/types/KalturaMsnDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMsnDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaMsnDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMsnDistributionProfile, _super);
    function KalturaMsnDistributionProfile(data) {
        return _super.call(this, data) || this;
    }
    KalturaMsnDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaMsnDistributionProfile' },
            username: { type: 's' },
            password: { type: 's' },
            domain: { type: 's' },
            csId: { type: 's' },
            source: { type: 's' },
            sourceFriendlyName: { type: 's' },
            pageGroup: { type: 's' },
            sourceFlavorParamsId: { type: 'n' },
            wmvFlavorParamsId: { type: 'n' },
            flvFlavorParamsId: { type: 'n' },
            slFlavorParamsId: { type: 'n' },
            slHdFlavorParamsId: { type: 'n' },
            msnvideoCat: { type: 's' },
            msnvideoTop: { type: 's' },
            msnvideoTopCat: { type: 's' }
        });
        return result;
    };
    return KalturaMsnDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaMsnDistributionProfile = KalturaMsnDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaMsnDistributionProfile', KalturaMsnDistributionProfile);
