// kaltura-node-typescript-client/api/types/KalturaDropFolderFileBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaDropFolderFileBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaDropFolderFileStatus_1 = require("./KalturaDropFolderFileStatus");
var KalturaDropFolderFileErrorCode_1 = require("./KalturaDropFolderFileErrorCode");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaDropFolderFileBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaDropFolderFileBaseFilter, _super);
    function KalturaDropFolderFileBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaDropFolderFileBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaDropFolderFileBaseFilter' },
            idEqual: { type: 'n' },
            idIn: { type: 's' },
            partnerIdEqual: { type: 'n' },
            partnerIdIn: { type: 's' },
            dropFolderIdEqual: { type: 'n' },
            dropFolderIdIn: { type: 's' },
            fileNameEqual: { type: 's' },
            fileNameIn: { type: 's' },
            fileNameLike: { type: 's' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaDropFolderFileStatus_1.KalturaDropFolderFileStatus, subType: 'KalturaDropFolderFileStatus' },
            statusIn: { type: 's' },
            statusNotIn: { type: 's' },
            parsedSlugEqual: { type: 's' },
            parsedSlugIn: { type: 's' },
            parsedSlugLike: { type: 's' },
            parsedFlavorEqual: { type: 's' },
            parsedFlavorIn: { type: 's' },
            parsedFlavorLike: { type: 's' },
            leadDropFolderFileIdEqual: { type: 'n' },
            deletedDropFolderFileIdEqual: { type: 'n' },
            entryIdEqual: { type: 's' },
            errorCodeEqual: { type: 'es', subTypeConstructor: KalturaDropFolderFileErrorCode_1.KalturaDropFolderFileErrorCode, subType: 'KalturaDropFolderFileErrorCode' },
            errorCodeIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' }
        });
        return result;
    };
    return KalturaDropFolderFileBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaDropFolderFileBaseFilter = KalturaDropFolderFileBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaDropFolderFileBaseFilter', KalturaDropFolderFileBaseFilter);
