// kaltura-node-typescript-client/api/types/KalturaLiveReportExportParams.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveReportExportParams = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var kaltura_object_base_1 = require("../kaltura-object-base");
var KalturaLiveReportExportParams = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaLiveReportExportParams, _super);
    function KalturaLiveReportExportParams(data) {
        return _super.call(this, data) || this;
    }
    KalturaLiveReportExportParams.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaLiveReportExportParams' },
            entryIds: { type: 's' },
            recpientEmail: { type: 's' },
            timeZoneOffset: { type: 'n' },
            applicationUrlTemplate: { type: 's' }
        });
        return result;
    };
    return KalturaLiveReportExportParams;
}(kaltura_object_base_1.KalturaObjectBase));
exports.KalturaLiveReportExportParams = KalturaLiveReportExportParams;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaLiveReportExportParams', KalturaLiveReportExportParams);
