// kaltura-node-typescript-client/api/types/KalturaComcastMrssDistributionProfile.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaComcastMrssDistributionProfile = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaKeyValue_1 = require("./KalturaKeyValue");
var KalturaConfigurableDistributionProfile_1 = require("./KalturaConfigurableDistributionProfile");
var KalturaComcastMrssDistributionProfile = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaComcastMrssDistributionProfile, _super);
    function KalturaComcastMrssDistributionProfile(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.cPlatformTvSeries === 'undefined')
            _this.cPlatformTvSeries = [];
        return _this;
    }
    KalturaComcastMrssDistributionProfile.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaComcastMrssDistributionProfile' },
            metadataProfileId: { type: 'n' },
            feedUrl: { type: 's', readOnly: true },
            feedTitle: { type: 's' },
            feedLink: { type: 's' },
            feedDescription: { type: 's' },
            feedLastBuildDate: { type: 's' },
            itemLink: { type: 's' },
            cPlatformTvSeries: { type: 'a', subTypeConstructor: KalturaKeyValue_1.KalturaKeyValue, subType: 'KalturaKeyValue' },
            cPlatformTvSeriesField: { type: 's' },
            shouldIncludeCuePoints: { type: 'b' },
            shouldIncludeCaptions: { type: 'b' },
            shouldAddThumbExtension: { type: 'b' }
        });
        return result;
    };
    return KalturaComcastMrssDistributionProfile;
}(KalturaConfigurableDistributionProfile_1.KalturaConfigurableDistributionProfile));
exports.KalturaComcastMrssDistributionProfile = KalturaComcastMrssDistributionProfile;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaComcastMrssDistributionProfile', KalturaComcastMrssDistributionProfile);
