// kaltura-node-typescript-client/api/types/KalturaBusinessProcessSignalNotificationTemplateOrderBy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaBusinessProcessSignalNotificationTemplateOrderBy = void 0;
var KalturaBusinessProcessSignalNotificationTemplateOrderBy;
(function (KalturaBusinessProcessSignalNotificationTemplateOrderBy) {
    KalturaBusinessProcessSignalNotificationTemplateOrderBy["createdAtAsc"] = "+createdAt";
    KalturaBusinessProcessSignalNotificationTemplateOrderBy["createdAtDesc"] = "-createdAt";
    KalturaBusinessProcessSignalNotificationTemplateOrderBy["idAsc"] = "+id";
    KalturaBusinessProcessSignalNotificationTemplateOrderBy["idDesc"] = "-id";
    KalturaBusinessProcessSignalNotificationTemplateOrderBy["updatedAtAsc"] = "+updatedAt";
    KalturaBusinessProcessSignalNotificationTemplateOrderBy["updatedAtDesc"] = "-updatedAt";
})(KalturaBusinessProcessSignalNotificationTemplateOrderBy = exports.KalturaBusinessProcessSignalNotificationTemplateOrderBy || (exports.KalturaBusinessProcessSignalNotificationTemplateOrderBy = {}));
