// kaltura-node-typescript-client/api/types/KalturaLiveEntryCompareAttribute.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaLiveEntryCompareAttribute = void 0;
var KalturaLiveEntryCompareAttribute;
(function (KalturaLiveEntryCompareAttribute) {
    KalturaLiveEntryCompareAttribute["accessControlId"] = "accessControlId";
    KalturaLiveEntryCompareAttribute["createdAt"] = "createdAt";
    KalturaLiveEntryCompareAttribute["endDate"] = "endDate";
    KalturaLiveEntryCompareAttribute["lastPlayedAt"] = "lastPlayedAt";
    KalturaLiveEntryCompareAttribute["mediaDate"] = "mediaDate";
    KalturaLiveEntryCompareAttribute["mediaType"] = "mediaType";
    KalturaLiveEntryCompareAttribute["moderationCount"] = "moderationCount";
    KalturaLiveEntryCompareAttribute["moderationStatus"] = "moderationStatus";
    KalturaLiveEntryCompareAttribute["msDuration"] = "msDuration";
    KalturaLiveEntryCompareAttribute["partnerId"] = "partnerId";
    KalturaLiveEntryCompareAttribute["partnerSortValue"] = "partnerSortValue";
    KalturaLiveEntryCompareAttribute["plays"] = "plays";
    KalturaLiveEntryCompareAttribute["rank"] = "rank";
    KalturaLiveEntryCompareAttribute["replacementStatus"] = "replacementStatus";
    KalturaLiveEntryCompareAttribute["startDate"] = "startDate";
    KalturaLiveEntryCompareAttribute["status"] = "status";
    KalturaLiveEntryCompareAttribute["totalRank"] = "totalRank";
    KalturaLiveEntryCompareAttribute["type"] = "type";
    KalturaLiveEntryCompareAttribute["updatedAt"] = "updatedAt";
    KalturaLiveEntryCompareAttribute["views"] = "views";
})(KalturaLiveEntryCompareAttribute = exports.KalturaLiveEntryCompareAttribute || (exports.KalturaLiveEntryCompareAttribute = {}));
