// kaltura-node-typescript-client/api/types/KalturaStorageJobData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaStorageJobData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaJobData_1 = require("./KalturaJobData");
var KalturaStorageJobData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaStorageJobData, _super);
    function KalturaStorageJobData(data) {
        return _super.call(this, data) || this;
    }
    KalturaStorageJobData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaStorageJobData' },
            serverUrl: { type: 's' },
            serverUsername: { type: 's' },
            serverPassword: { type: 's' },
            serverPrivateKey: { type: 's' },
            serverPublicKey: { type: 's' },
            serverPassPhrase: { type: 's' },
            ftpPassiveMode: { type: 'b' },
            srcFileSyncLocalPath: { type: 's' },
            srcFileEncryptionKey: { type: 's' },
            srcFileSyncId: { type: 's' },
            destFileSyncStoredPath: { type: 's' }
        });
        return result;
    };
    return KalturaStorageJobData;
}(KalturaJobData_1.KalturaJobData));
exports.KalturaStorageJobData = KalturaStorageJobData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaStorageJobData', KalturaStorageJobData);
