// kaltura-node-typescript-client/api/types/KalturaESearchMetadataItemData.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaESearchMetadataItemData = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaESearchItemData_1 = require("./KalturaESearchItemData");
var KalturaESearchMetadataItemData = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaESearchMetadataItemData, _super);
    function KalturaESearchMetadataItemData(data) {
        return _super.call(this, data) || this;
    }
    KalturaESearchMetadataItemData.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaESearchMetadataItemData' },
            xpath: { type: 's' },
            metadataProfileId: { type: 'n' },
            metadataFieldId: { type: 'n' },
            valueText: { type: 's' },
            valueInt: { type: 'n' }
        });
        return result;
    };
    return KalturaESearchMetadataItemData;
}(KalturaESearchItemData_1.KalturaESearchItemData));
exports.KalturaESearchMetadataItemData = KalturaESearchMetadataItemData;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaESearchMetadataItemData', KalturaESearchMetadataItemData);
