// kaltura-node-typescript-client/api/types/KalturaAppTokenBaseFilter.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaAppTokenBaseFilter = void 0;
var tslib_1 = require("tslib");
var kaltura_types_factory_1 = require("../kaltura-types-factory");
var KalturaAppTokenStatus_1 = require("./KalturaAppTokenStatus");
var KalturaFilter_1 = require("./KalturaFilter");
var KalturaAppTokenBaseFilter = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaAppTokenBaseFilter, _super);
    function KalturaAppTokenBaseFilter(data) {
        return _super.call(this, data) || this;
    }
    KalturaAppTokenBaseFilter.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            objectType: { type: 'c', default: 'KalturaAppTokenBaseFilter' },
            idEqual: { type: 's' },
            idIn: { type: 's' },
            createdAtGreaterThanOrEqual: { type: 'd' },
            createdAtLessThanOrEqual: { type: 'd' },
            updatedAtGreaterThanOrEqual: { type: 'd' },
            updatedAtLessThanOrEqual: { type: 'd' },
            statusEqual: { type: 'en', subTypeConstructor: KalturaAppTokenStatus_1.KalturaAppTokenStatus, subType: 'KalturaAppTokenStatus' },
            statusIn: { type: 's' },
            sessionUserIdEqual: { type: 's' }
        });
        return result;
    };
    return KalturaAppTokenBaseFilter;
}(KalturaFilter_1.KalturaFilter));
exports.KalturaAppTokenBaseFilter = KalturaAppTokenBaseFilter;
kaltura_types_factory_1.KalturaTypesFactory.registerType('KalturaAppTokenBaseFilter', KalturaAppTokenBaseFilter);
