// kaltura-node-typescript-client/api/types/LiveStreamAppendRecordingAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LiveStreamAppendRecordingAction = void 0;
var tslib_1 = require("tslib");
var KalturaLiveEntry_1 = require("./KalturaLiveEntry");
var KalturaEntryServerNodeType_1 = require("./KalturaEntryServerNodeType");
var KalturaDataCenterContentResource_1 = require("./KalturaDataCenterContentResource");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'liveStream' action 'appendRecording'.
 *
 * Usage: Append recorded video to live entry
 *
 * Server response type:         KalturaLiveEntry
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var LiveStreamAppendRecordingAction = /** @class */ (function (_super) {
    tslib_1.__extends(LiveStreamAppendRecordingAction, _super);
    function LiveStreamAppendRecordingAction(data) {
        var _this = _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaLiveEntry', responseConstructor: KalturaLiveEntry_1.KalturaLiveEntry }) || this;
        if (typeof _this.isLastChunk === 'undefined')
            _this.isLastChunk = false;
        return _this;
    }
    LiveStreamAppendRecordingAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'livestream' },
            action: { type: 'c', default: 'appendRecording' },
            entryId: { type: 's' },
            assetId: { type: 's' },
            mediaServerIndex: { type: 'es', subTypeConstructor: KalturaEntryServerNodeType_1.KalturaEntryServerNodeType, subType: 'KalturaEntryServerNodeType' },
            resource: { type: 'o', subTypeConstructor: KalturaDataCenterContentResource_1.KalturaDataCenterContentResource, subType: 'KalturaDataCenterContentResource' },
            duration: { type: 'n' },
            isLastChunk: { type: 'b' }
        });
        return result;
    };
    return LiveStreamAppendRecordingAction;
}(kaltura_request_1.KalturaRequest));
exports.LiveStreamAppendRecordingAction = LiveStreamAppendRecordingAction;
