// kaltura-node-typescript-client/api/types/LiveStreamAddLiveStreamPushPublishConfigurationAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LiveStreamAddLiveStreamPushPublishConfigurationAction = void 0;
var tslib_1 = require("tslib");
var KalturaLiveStreamEntry_1 = require("./KalturaLiveStreamEntry");
var KalturaPlaybackProtocol_1 = require("./KalturaPlaybackProtocol");
var KalturaLiveStreamConfiguration_1 = require("./KalturaLiveStreamConfiguration");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'liveStream' action 'addLiveStreamPushPublishConfiguration'.
 *
 * Usage: Add new pushPublish configuration to entry
 *
 * Server response type:         KalturaLiveStreamEntry
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var LiveStreamAddLiveStreamPushPublishConfigurationAction = /** @class */ (function (_super) {
    tslib_1.__extends(LiveStreamAddLiveStreamPushPublishConfigurationAction, _super);
    function LiveStreamAddLiveStreamPushPublishConfigurationAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaLiveStreamEntry', responseConstructor: KalturaLiveStreamEntry_1.KalturaLiveStreamEntry }) || this;
    }
    LiveStreamAddLiveStreamPushPublishConfigurationAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'livestream' },
            action: { type: 'c', default: 'addLiveStreamPushPublishConfiguration' },
            entryId: { type: 's' },
            protocol: { type: 'es', subTypeConstructor: KalturaPlaybackProtocol_1.KalturaPlaybackProtocol, subType: 'KalturaPlaybackProtocol' },
            url: { type: 's' },
            liveStreamConfiguration: { type: 'o', subTypeConstructor: KalturaLiveStreamConfiguration_1.KalturaLiveStreamConfiguration, subType: 'KalturaLiveStreamConfiguration' }
        });
        return result;
    };
    return LiveStreamAddLiveStreamPushPublishConfigurationAction;
}(kaltura_request_1.KalturaRequest));
exports.LiveStreamAddLiveStreamPushPublishConfigurationAction = LiveStreamAddLiveStreamPushPublishConfigurationAction;
