// kaltura-node-typescript-client/api/types/LiveChannelSetRecordedContentAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LiveChannelSetRecordedContentAction = void 0;
var tslib_1 = require("tslib");
var KalturaLiveEntry_1 = require("./KalturaLiveEntry");
var KalturaEntryServerNodeType_1 = require("./KalturaEntryServerNodeType");
var KalturaDataCenterContentResource_1 = require("./KalturaDataCenterContentResource");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'liveChannel' action 'setRecordedContent'.
 *
 * Usage: Set recorded video to live entry
 *
 * Server response type:         KalturaLiveEntry
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var LiveChannelSetRecordedContentAction = /** @class */ (function (_super) {
    tslib_1.__extends(LiveChannelSetRecordedContentAction, _super);
    function LiveChannelSetRecordedContentAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaLiveEntry', responseConstructor: KalturaLiveEntry_1.KalturaLiveEntry }) || this;
    }
    LiveChannelSetRecordedContentAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'livechannel' },
            action: { type: 'c', default: 'setRecordedContent' },
            entryId: { type: 's' },
            mediaServerIndex: { type: 'es', subTypeConstructor: KalturaEntryServerNodeType_1.KalturaEntryServerNodeType, subType: 'KalturaEntryServerNodeType' },
            resource: { type: 'o', subTypeConstructor: KalturaDataCenterContentResource_1.KalturaDataCenterContentResource, subType: 'KalturaDataCenterContentResource' },
            duration: { type: 'n' },
            recordedEntryId: { type: 's' },
            flavorParamsId: { type: 'n' }
        });
        return result;
    };
    return LiveChannelSetRecordedContentAction;
}(kaltura_request_1.KalturaRequest));
exports.LiveChannelSetRecordedContentAction = LiveChannelSetRecordedContentAction;
