// kaltura-node-typescript-client/api/types/LiveStreamRegisterMediaServerAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LiveStreamRegisterMediaServerAction = void 0;
var tslib_1 = require("tslib");
var KalturaLiveEntry_1 = require("./KalturaLiveEntry");
var KalturaEntryServerNodeType_1 = require("./KalturaEntryServerNodeType");
var KalturaEntryServerNodeStatus_1 = require("./KalturaEntryServerNodeStatus");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'liveStream' action 'registerMediaServer'.
 *
 * Usage: Register media server to live entry
 *
 * Server response type:         KalturaLiveEntry
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var LiveStreamRegisterMediaServerAction = /** @class */ (function (_super) {
    tslib_1.__extends(LiveStreamRegisterMediaServerAction, _super);
    function LiveStreamRegisterMediaServerAction(data) {
        var _this = _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaLiveEntry', responseConstructor: KalturaLiveEntry_1.KalturaLiveEntry }) || this;
        if (typeof _this.liveEntryStatus === 'undefined')
            _this.liveEntryStatus = 1;
        if (typeof _this.shouldCreateRecordedEntry === 'undefined')
            _this.shouldCreateRecordedEntry = true;
        return _this;
    }
    LiveStreamRegisterMediaServerAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'livestream' },
            action: { type: 'c', default: 'registerMediaServer' },
            entryId: { type: 's' },
            hostname: { type: 's' },
            mediaServerIndex: { type: 'es', subTypeConstructor: KalturaEntryServerNodeType_1.KalturaEntryServerNodeType, subType: 'KalturaEntryServerNodeType' },
            applicationName: { type: 's' },
            liveEntryStatus: { type: 'en', subTypeConstructor: KalturaEntryServerNodeStatus_1.KalturaEntryServerNodeStatus, subType: 'KalturaEntryServerNodeStatus' },
            shouldCreateRecordedEntry: { type: 'b' }
        });
        return result;
    };
    return LiveStreamRegisterMediaServerAction;
}(kaltura_request_1.KalturaRequest));
exports.LiveStreamRegisterMediaServerAction = LiveStreamRegisterMediaServerAction;
