// kaltura-node-typescript-client/api/types/LiveStreamCreatePeriodicSyncPointsAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LiveStreamCreatePeriodicSyncPointsAction = void 0;
var tslib_1 = require("tslib");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'liveStream' action 'createPeriodicSyncPoints'.
 *
 * Usage: Creates periodic metadata sync-point events on a live stream
 *
 * Server response type:         void
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var LiveStreamCreatePeriodicSyncPointsAction = /** @class */ (function (_super) {
    tslib_1.__extends(LiveStreamCreatePeriodicSyncPointsAction, _super);
    function LiveStreamCreatePeriodicSyncPointsAction(data) {
        return _super.call(this, data, { responseType: 'v', responseSubType: '', responseConstructor: null }) || this;
    }
    LiveStreamCreatePeriodicSyncPointsAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'livestream' },
            action: { type: 'c', default: 'createPeriodicSyncPoints' },
            entryId: { type: 's' },
            interval: { type: 'n' },
            duration: { type: 'n' }
        });
        return result;
    };
    return LiveStreamCreatePeriodicSyncPointsAction;
}(kaltura_request_1.KalturaRequest));
exports.LiveStreamCreatePeriodicSyncPointsAction = LiveStreamCreatePeriodicSyncPointsAction;
