// kaltura-node-typescript-client/api/types/LiveReportsGetReportAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LiveReportsGetReportAction = void 0;
var tslib_1 = require("tslib");
var KalturaLiveStatsListResponse_1 = require("./KalturaLiveStatsListResponse");
var KalturaLiveReportType_1 = require("./KalturaLiveReportType");
var KalturaLiveReportInputFilter_1 = require("./KalturaLiveReportInputFilter");
var KalturaFilterPager_1 = require("./KalturaFilterPager");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'liveReports' action 'getReport'.
 *
 *
 *
 * Server response type:         KalturaLiveStatsListResponse
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var LiveReportsGetReportAction = /** @class */ (function (_super) {
    tslib_1.__extends(LiveReportsGetReportAction, _super);
    function LiveReportsGetReportAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaLiveStatsListResponse', responseConstructor: KalturaLiveStatsListResponse_1.KalturaLiveStatsListResponse }) || this;
    }
    LiveReportsGetReportAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'livereports' },
            action: { type: 'c', default: 'getReport' },
            reportType: { type: 'es', subTypeConstructor: KalturaLiveReportType_1.KalturaLiveReportType, subType: 'KalturaLiveReportType' },
            filter: { type: 'o', subTypeConstructor: KalturaLiveReportInputFilter_1.KalturaLiveReportInputFilter, subType: 'KalturaLiveReportInputFilter' },
            pager: { type: 'o', subTypeConstructor: KalturaFilterPager_1.KalturaFilterPager, subType: 'KalturaFilterPager' }
        });
        return result;
    };
    return LiveReportsGetReportAction;
}(kaltura_request_1.KalturaRequest));
exports.LiveReportsGetReportAction = LiveReportsGetReportAction;
