// kaltura-node-typescript-client/api/types/LiveStreamUnregisterMediaServerAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LiveStreamUnregisterMediaServerAction = void 0;
var tslib_1 = require("tslib");
var KalturaLiveEntry_1 = require("./KalturaLiveEntry");
var KalturaEntryServerNodeType_1 = require("./KalturaEntryServerNodeType");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'liveStream' action 'unregisterMediaServer'.
 *
 * Usage: Unregister media server from live entry
 *
 * Server response type:         KalturaLiveEntry
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var LiveStreamUnregisterMediaServerAction = /** @class */ (function (_super) {
    tslib_1.__extends(LiveStreamUnregisterMediaServerAction, _super);
    function LiveStreamUnregisterMediaServerAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaLiveEntry', responseConstructor: KalturaLiveEntry_1.KalturaLiveEntry }) || this;
    }
    LiveStreamUnregisterMediaServerAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'livestream' },
            action: { type: 'c', default: 'unregisterMediaServer' },
            entryId: { type: 's' },
            hostname: { type: 's' },
            mediaServerIndex: { type: 'es', subTypeConstructor: KalturaEntryServerNodeType_1.KalturaEntryServerNodeType, subType: 'KalturaEntryServerNodeType' }
        });
        return result;
    };
    return LiveStreamUnregisterMediaServerAction;
}(kaltura_request_1.KalturaRequest));
exports.LiveStreamUnregisterMediaServerAction = LiveStreamUnregisterMediaServerAction;
