// kaltura-node-typescript-client/api/types/LiveReportsGetEventsAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LiveReportsGetEventsAction = void 0;
var tslib_1 = require("tslib");
var KalturaReportGraph_1 = require("./KalturaReportGraph");
var KalturaLiveReportType_1 = require("./KalturaLiveReportType");
var KalturaLiveReportInputFilter_1 = require("./KalturaLiveReportInputFilter");
var KalturaFilterPager_1 = require("./KalturaFilterPager");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'liveReports' action 'getEvents'.
 *
 *
 *
 * Server response type:         KalturaReportGraph[]
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var LiveReportsGetEventsAction = /** @class */ (function (_super) {
    tslib_1.__extends(LiveReportsGetEventsAction, _super);
    function LiveReportsGetEventsAction(data) {
        return _super.call(this, data, { responseType: 'a', responseSubType: 'KalturaReportGraph', responseConstructor: KalturaReportGraph_1.KalturaReportGraph }) || this;
    }
    LiveReportsGetEventsAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'livereports' },
            action: { type: 'c', default: 'getEvents' },
            reportType: { type: 'es', subTypeConstructor: KalturaLiveReportType_1.KalturaLiveReportType, subType: 'KalturaLiveReportType' },
            filter: { type: 'o', subTypeConstructor: KalturaLiveReportInputFilter_1.KalturaLiveReportInputFilter, subType: 'KalturaLiveReportInputFilter' },
            pager: { type: 'o', subTypeConstructor: KalturaFilterPager_1.KalturaFilterPager, subType: 'KalturaFilterPager' }
        });
        return result;
    };
    return LiveReportsGetEventsAction;
}(kaltura_request_1.KalturaRequest));
exports.LiveReportsGetEventsAction = LiveReportsGetEventsAction;
