// kaltura-node-typescript-client/api/types/LiveStreamAuthenticateAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.LiveStreamAuthenticateAction = void 0;
var tslib_1 = require("tslib");
var KalturaLiveStreamEntry_1 = require("./KalturaLiveStreamEntry");
var KalturaEntryServerNodeType_1 = require("./KalturaEntryServerNodeType");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'liveStream' action 'authenticate'.
 *
 * Usage: Authenticate live-stream entry against stream token and partner limitations
 *
 * Server response type:         KalturaLiveStreamEntry
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var LiveStreamAuthenticateAction = /** @class */ (function (_super) {
    tslib_1.__extends(LiveStreamAuthenticateAction, _super);
    function LiveStreamAuthenticateAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaLiveStreamEntry', responseConstructor: KalturaLiveStreamEntry_1.KalturaLiveStreamEntry }) || this;
    }
    LiveStreamAuthenticateAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'livestream' },
            action: { type: 'c', default: 'authenticate' },
            entryId: { type: 's' },
            token: { type: 's' },
            hostname: { type: 's' },
            mediaServerIndex: { type: 'es', subTypeConstructor: KalturaEntryServerNodeType_1.KalturaEntryServerNodeType, subType: 'KalturaEntryServerNodeType' },
            applicationName: { type: 's' }
        });
        return result;
    };
    return LiveStreamAuthenticateAction;
}(kaltura_request_1.KalturaRequest));
exports.LiveStreamAuthenticateAction = LiveStreamAuthenticateAction;
