// kaltura-node-typescript-client/api/types/MediaGetMrssAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaGetMrssAction = void 0;
var tslib_1 = require("tslib");
var KalturaExtendingItemMrssParameter_1 = require("./KalturaExtendingItemMrssParameter");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'media' action 'getMrss'.
 *
 * Usage: Get MRSS by entry id
 * XML will return as an escaped string
 *
 * Server response type:         string
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var MediaGetMrssAction = /** @class */ (function (_super) {
    tslib_1.__extends(MediaGetMrssAction, _super);
    function MediaGetMrssAction(data) {
        var _this = _super.call(this, data, { responseType: 's', responseSubType: '', responseConstructor: null }) || this;
        if (typeof _this.extendingItemsArray === 'undefined')
            _this.extendingItemsArray = [];
        return _this;
    }
    MediaGetMrssAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'media' },
            action: { type: 'c', default: 'getMrss' },
            entryId: { type: 's' },
            extendingItemsArray: { type: 'a', subTypeConstructor: KalturaExtendingItemMrssParameter_1.KalturaExtendingItemMrssParameter, subType: 'KalturaExtendingItemMrssParameter' },
            features: { type: 's' }
        });
        return result;
    };
    return MediaGetMrssAction;
}(kaltura_request_1.KalturaRequest));
exports.MediaGetMrssAction = MediaGetMrssAction;
