// kaltura-node-typescript-client/api/types/MediaAddFromBulkAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaAddFromBulkAction = void 0;
var tslib_1 = require("tslib");
var KalturaMediaEntry_1 = require("./KalturaMediaEntry");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'media' action 'addFromBulk'.
 *
 * Usage: Adds new media entry by importing an HTTP or FTP URL.
 * The entry will be queued for import and then for conversion.
 * This action should be exposed only to the batches
 *
 * Server response type:         KalturaMediaEntry
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var MediaAddFromBulkAction = /** @class */ (function (_super) {
    tslib_1.__extends(MediaAddFromBulkAction, _super);
    function MediaAddFromBulkAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaMediaEntry', responseConstructor: KalturaMediaEntry_1.KalturaMediaEntry }) || this;
    }
    MediaAddFromBulkAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'media' },
            action: { type: 'c', default: 'addFromBulk' },
            mediaEntry: { type: 'o', subTypeConstructor: KalturaMediaEntry_1.KalturaMediaEntry, subType: 'KalturaMediaEntry' },
            url: { type: 's' },
            bulkUploadId: { type: 'n' }
        });
        return result;
    };
    return MediaAddFromBulkAction;
}(kaltura_request_1.KalturaRequest));
exports.MediaAddFromBulkAction = MediaAddFromBulkAction;
