// kaltura-node-typescript-client/api/types/MediaUpdateContentAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaUpdateContentAction = void 0;
var tslib_1 = require("tslib");
var KalturaMediaEntry_1 = require("./KalturaMediaEntry");
var KalturaResource_1 = require("./KalturaResource");
var KalturaEntryReplacementOptions_1 = require("./KalturaEntryReplacementOptions");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'media' action 'updateContent'.
 *
 * Usage: Replace content associated with the media entry
 *
 * Server response type:         KalturaMediaEntry
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var MediaUpdateContentAction = /** @class */ (function (_super) {
    tslib_1.__extends(MediaUpdateContentAction, _super);
    function MediaUpdateContentAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaMediaEntry', responseConstructor: KalturaMediaEntry_1.KalturaMediaEntry }) || this;
    }
    MediaUpdateContentAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'media' },
            action: { type: 'c', default: 'updateContent' },
            entryId: { type: 's' },
            resource: { type: 'o', subTypeConstructor: KalturaResource_1.KalturaResource, subType: 'KalturaResource' },
            conversionProfileId: { type: 'n' },
            advancedOptions: { type: 'o', subTypeConstructor: KalturaEntryReplacementOptions_1.KalturaEntryReplacementOptions, subType: 'KalturaEntryReplacementOptions' }
        });
        return result;
    };
    return MediaUpdateContentAction;
}(kaltura_request_1.KalturaRequest));
exports.MediaUpdateContentAction = MediaUpdateContentAction;
