// kaltura-node-typescript-client/api/types/MediaConvertAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaConvertAction = void 0;
var tslib_1 = require("tslib");
var KalturaConversionAttribute_1 = require("./KalturaConversionAttribute");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'media' action 'convert'.
 *
 * Usage: Convert entry
 *
 * Server response type:         number
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var MediaConvertAction = /** @class */ (function (_super) {
    tslib_1.__extends(MediaConvertAction, _super);
    function MediaConvertAction(data) {
        var _this = _super.call(this, data, { responseType: 'n', responseSubType: '', responseConstructor: null }) || this;
        if (typeof _this.dynamicConversionAttributes === 'undefined')
            _this.dynamicConversionAttributes = [];
        return _this;
    }
    MediaConvertAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'media' },
            action: { type: 'c', default: 'convert' },
            entryId: { type: 's' },
            conversionProfileId: { type: 'n' },
            dynamicConversionAttributes: { type: 'a', subTypeConstructor: KalturaConversionAttribute_1.KalturaConversionAttribute, subType: 'KalturaConversionAttribute' }
        });
        return result;
    };
    return MediaConvertAction;
}(kaltura_request_1.KalturaRequest));
exports.MediaConvertAction = MediaConvertAction;
