// kaltura-node-typescript-client/api/types/MediaBulkUploadAddAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaBulkUploadAddAction = void 0;
var tslib_1 = require("tslib");
var KalturaBulkUpload_1 = require("./KalturaBulkUpload");
var KalturaBulkUploadJobData_1 = require("./KalturaBulkUploadJobData");
var KalturaBulkUploadEntryData_1 = require("./KalturaBulkUploadEntryData");
var kaltura_upload_request_1 = require("../kaltura-upload-request");
/**
 * Build request payload for service 'media' action 'bulkUploadAdd'.
 *
 * Usage: Add new bulk upload batch job
 * Conversion profile id can be specified in the API or in the CSV file, the one in the CSV file will be stronger.
 * If no conversion profile was specified, partner's default will be used
 *
 * Server response type:         KalturaBulkUpload
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaUploadRequest
 */
var MediaBulkUploadAddAction = /** @class */ (function (_super) {
    tslib_1.__extends(MediaBulkUploadAddAction, _super);
    function MediaBulkUploadAddAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaBulkUpload', responseConstructor: KalturaBulkUpload_1.KalturaBulkUpload }) || this;
    }
    MediaBulkUploadAddAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'media' },
            action: { type: 'c', default: 'bulkUploadAdd' },
            fileData: { type: 'f' },
            bulkUploadData: { type: 'o', subTypeConstructor: KalturaBulkUploadJobData_1.KalturaBulkUploadJobData, subType: 'KalturaBulkUploadJobData' },
            bulkUploadEntryData: { type: 'o', subTypeConstructor: KalturaBulkUploadEntryData_1.KalturaBulkUploadEntryData, subType: 'KalturaBulkUploadEntryData' }
        });
        return result;
    };
    return MediaBulkUploadAddAction;
}(kaltura_upload_request_1.KalturaUploadRequest));
exports.MediaBulkUploadAddAction = MediaBulkUploadAddAction;
