// kaltura-node-typescript-client/api/types/MediaUpdateThumbnailFromSourceEntryAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaUpdateThumbnailFromSourceEntryAction = void 0;
var tslib_1 = require("tslib");
var KalturaMediaEntry_1 = require("./KalturaMediaEntry");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'media' action 'updateThumbnailFromSourceEntry'.
 *
 * Usage: Update media entry thumbnail from a different entry by a specified time offset (In seconds)
 * If flavor params id not specified, source flavor will be used by default
 *
 * Server response type:         KalturaMediaEntry
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var MediaUpdateThumbnailFromSourceEntryAction = /** @class */ (function (_super) {
    tslib_1.__extends(MediaUpdateThumbnailFromSourceEntryAction, _super);
    function MediaUpdateThumbnailFromSourceEntryAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaMediaEntry', responseConstructor: KalturaMediaEntry_1.KalturaMediaEntry }) || this;
    }
    MediaUpdateThumbnailFromSourceEntryAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'media' },
            action: { type: 'c', default: 'updateThumbnailFromSourceEntry' },
            entryId: { type: 's' },
            sourceEntryId: { type: 's' },
            timeOffset: { type: 'n' },
            flavorParamsId: { type: 'n' }
        });
        return result;
    };
    return MediaUpdateThumbnailFromSourceEntryAction;
}(kaltura_request_1.KalturaRequest));
exports.MediaUpdateThumbnailFromSourceEntryAction = MediaUpdateThumbnailFromSourceEntryAction;
