// kaltura-node-typescript-client/api/types/MediaAddFromEntryAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MediaAddFromEntryAction = void 0;
var tslib_1 = require("tslib");
var KalturaMediaEntry_1 = require("./KalturaMediaEntry");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'media' action 'addFromEntry'.
 *
 * Usage: Copy entry into new entry
 *
 * Server response type:         KalturaMediaEntry
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var MediaAddFromEntryAction = /** @class */ (function (_super) {
    tslib_1.__extends(MediaAddFromEntryAction, _super);
    function MediaAddFromEntryAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaMediaEntry', responseConstructor: KalturaMediaEntry_1.KalturaMediaEntry }) || this;
    }
    MediaAddFromEntryAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'media' },
            action: { type: 'c', default: 'addFromEntry' },
            sourceEntryId: { type: 's' },
            mediaEntry: { type: 'o', subTypeConstructor: KalturaMediaEntry_1.KalturaMediaEntry, subType: 'KalturaMediaEntry' },
            sourceFlavorParamsId: { type: 'n' }
        });
        return result;
    };
    return MediaAddFromEntryAction;
}(kaltura_request_1.KalturaRequest));
exports.MediaAddFromEntryAction = MediaAddFromEntryAction;
