// kaltura-node-typescript-client/api/types/MetadataAddFromUrlAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetadataAddFromUrlAction = void 0;
var tslib_1 = require("tslib");
var KalturaMetadata_1 = require("./KalturaMetadata");
var KalturaMetadataObjectType_1 = require("./KalturaMetadataObjectType");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'metadata' action 'addFromUrl'.
 *
 * Usage: Allows you to add a metadata XML data from remote URL
 *
 * Server response type:         KalturaMetadata
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var MetadataAddFromUrlAction = /** @class */ (function (_super) {
    tslib_1.__extends(MetadataAddFromUrlAction, _super);
    function MetadataAddFromUrlAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaMetadata', responseConstructor: KalturaMetadata_1.KalturaMetadata }) || this;
    }
    MetadataAddFromUrlAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'metadata_metadata' },
            action: { type: 'c', default: 'addFromUrl' },
            metadataProfileId: { type: 'n' },
            objectType: { type: 'es', subTypeConstructor: KalturaMetadataObjectType_1.KalturaMetadataObjectType, subType: 'KalturaMetadataObjectType' },
            objectId: { type: 's' },
            url: { type: 's' }
        });
        return result;
    };
    return MetadataAddFromUrlAction;
}(kaltura_request_1.KalturaRequest));
exports.MetadataAddFromUrlAction = MetadataAddFromUrlAction;
