// kaltura-node-typescript-client/api/types/MetadataProfileUpdateAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetadataProfileUpdateAction = void 0;
var tslib_1 = require("tslib");
var KalturaMetadataProfile_1 = require("./KalturaMetadataProfile");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'metadataProfile' action 'update'.
 *
 * Usage: Update an existing metadata object
 *
 * Server response type:         KalturaMetadataProfile
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var MetadataProfileUpdateAction = /** @class */ (function (_super) {
    tslib_1.__extends(MetadataProfileUpdateAction, _super);
    function MetadataProfileUpdateAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaMetadataProfile', responseConstructor: KalturaMetadataProfile_1.KalturaMetadataProfile }) || this;
    }
    MetadataProfileUpdateAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'metadata_metadataprofile' },
            action: { type: 'c', default: 'update' },
            id: { type: 'n' },
            metadataProfile: { type: 'o', subTypeConstructor: KalturaMetadataProfile_1.KalturaMetadataProfile, subType: 'KalturaMetadataProfile' },
            xsdData: { type: 's' },
            viewsData: { type: 's' }
        });
        return result;
    };
    return MetadataProfileUpdateAction;
}(kaltura_request_1.KalturaRequest));
exports.MetadataProfileUpdateAction = MetadataProfileUpdateAction;
