// kaltura-node-typescript-client/api/types/MetadataAddFromFileAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetadataAddFromFileAction = void 0;
var tslib_1 = require("tslib");
var KalturaMetadata_1 = require("./KalturaMetadata");
var KalturaMetadataObjectType_1 = require("./KalturaMetadataObjectType");
var kaltura_upload_request_1 = require("../kaltura-upload-request");
/**
 * Build request payload for service 'metadata' action 'addFromFile'.
 *
 * Usage: Allows you to add a metadata object and metadata file associated with Kaltura object
 *
 * Server response type:         KalturaMetadata
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaUploadRequest
 */
var MetadataAddFromFileAction = /** @class */ (function (_super) {
    tslib_1.__extends(MetadataAddFromFileAction, _super);
    function MetadataAddFromFileAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaMetadata', responseConstructor: KalturaMetadata_1.KalturaMetadata }) || this;
    }
    MetadataAddFromFileAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'metadata_metadata' },
            action: { type: 'c', default: 'addFromFile' },
            metadataProfileId: { type: 'n' },
            objectType: { type: 'es', subTypeConstructor: KalturaMetadataObjectType_1.KalturaMetadataObjectType, subType: 'KalturaMetadataObjectType' },
            objectId: { type: 's' },
            xmlFile: { type: 'f' }
        });
        return result;
    };
    return MetadataAddFromFileAction;
}(kaltura_upload_request_1.KalturaUploadRequest));
exports.MetadataAddFromFileAction = MetadataAddFromFileAction;
