// kaltura-node-typescript-client/api/types/MetadataProfileAddFromFileAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetadataProfileAddFromFileAction = void 0;
var tslib_1 = require("tslib");
var KalturaMetadataProfile_1 = require("./KalturaMetadataProfile");
var kaltura_upload_request_1 = require("../kaltura-upload-request");
/**
 * Build request payload for service 'metadataProfile' action 'addFromFile'.
 *
 * Usage: Allows you to add a metadata profile object and metadata profile file associated with Kaltura object type
 *
 * Server response type:         KalturaMetadataProfile
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaUploadRequest
 */
var MetadataProfileAddFromFileAction = /** @class */ (function (_super) {
    tslib_1.__extends(MetadataProfileAddFromFileAction, _super);
    function MetadataProfileAddFromFileAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaMetadataProfile', responseConstructor: KalturaMetadataProfile_1.KalturaMetadataProfile }) || this;
    }
    MetadataProfileAddFromFileAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'metadata_metadataprofile' },
            action: { type: 'c', default: 'addFromFile' },
            metadataProfile: { type: 'o', subTypeConstructor: KalturaMetadataProfile_1.KalturaMetadataProfile, subType: 'KalturaMetadataProfile' },
            xsdFile: { type: 'f' },
            viewsFile: { type: 'f' }
        });
        return result;
    };
    return MetadataProfileAddFromFileAction;
}(kaltura_upload_request_1.KalturaUploadRequest));
exports.MetadataProfileAddFromFileAction = MetadataProfileAddFromFileAction;
