// kaltura-node-typescript-client/api/types/MetadataUpdateAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetadataUpdateAction = void 0;
var tslib_1 = require("tslib");
var KalturaMetadata_1 = require("./KalturaMetadata");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'metadata' action 'update'.
 *
 * Usage: Update an existing metadata object with new XML content
 *
 * Server response type:         KalturaMetadata
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var MetadataUpdateAction = /** @class */ (function (_super) {
    tslib_1.__extends(MetadataUpdateAction, _super);
    function MetadataUpdateAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaMetadata', responseConstructor: KalturaMetadata_1.KalturaMetadata }) || this;
    }
    MetadataUpdateAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'metadata_metadata' },
            action: { type: 'c', default: 'update' },
            id: { type: 'n' },
            xmlData: { type: 's' },
            version: { type: 'n' }
        });
        return result;
    };
    return MetadataUpdateAction;
}(kaltura_request_1.KalturaRequest));
exports.MetadataUpdateAction = MetadataUpdateAction;
