// kaltura-node-typescript-client/api/types/MetadataAddFromBulkAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MetadataAddFromBulkAction = void 0;
var tslib_1 = require("tslib");
var KalturaMetadata_1 = require("./KalturaMetadata");
var KalturaMetadataObjectType_1 = require("./KalturaMetadataObjectType");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'metadata' action 'addFromBulk'.
 *
 * Usage: Allows you to add a metadata XML data from remote URL.
 * Enables different permissions than addFromUrl action
 *
 * Server response type:         KalturaMetadata
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var MetadataAddFromBulkAction = /** @class */ (function (_super) {
    tslib_1.__extends(MetadataAddFromBulkAction, _super);
    function MetadataAddFromBulkAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaMetadata', responseConstructor: KalturaMetadata_1.KalturaMetadata }) || this;
    }
    MetadataAddFromBulkAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'metadata_metadata' },
            action: { type: 'c', default: 'addFromBulk' },
            metadataProfileId: { type: 'n' },
            objectType: { type: 'es', subTypeConstructor: KalturaMetadataObjectType_1.KalturaMetadataObjectType, subType: 'KalturaMetadataObjectType' },
            objectId: { type: 's' },
            url: { type: 's' }
        });
        return result;
    };
    return MetadataAddFromBulkAction;
}(kaltura_request_1.KalturaRequest));
exports.MetadataAddFromBulkAction = MetadataAddFromBulkAction;
