// kaltura-node-typescript-client/api/types/PartnerGetSecretsAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartnerGetSecretsAction = void 0;
var tslib_1 = require("tslib");
var KalturaPartner_1 = require("./KalturaPartner");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'partner' action 'getSecrets'.
 *
 * Usage: Retrieve partner secret and admin secret
 *
 * Server response type:         KalturaPartner
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var PartnerGetSecretsAction = /** @class */ (function (_super) {
    tslib_1.__extends(PartnerGetSecretsAction, _super);
    function PartnerGetSecretsAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaPartner', responseConstructor: KalturaPartner_1.KalturaPartner }) || this;
    }
    PartnerGetSecretsAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'partner' },
            action: { type: 'c', default: 'getSecrets' },
            partnerId: { type: 'n' },
            adminEmail: { type: 's' },
            cmsPassword: { type: 's' },
            otp: { type: 's' }
        });
        return result;
    };
    return PartnerGetSecretsAction;
}(kaltura_request_1.KalturaRequest));
exports.PartnerGetSecretsAction = PartnerGetSecretsAction;
