// kaltura-node-typescript-client/api/types/PartnerRegistrationValidationAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartnerRegistrationValidationAction = void 0;
var tslib_1 = require("tslib");
var KalturaPartner_1 = require("./KalturaPartner");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'partner' action 'registrationValidation'.
 *
 * Usage: Create a new Partner object
 *
 * Server response type:         boolean
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var PartnerRegistrationValidationAction = /** @class */ (function (_super) {
    tslib_1.__extends(PartnerRegistrationValidationAction, _super);
    function PartnerRegistrationValidationAction(data) {
        var _this = _super.call(this, data, { responseType: 'b', responseSubType: '', responseConstructor: null }) || this;
        if (typeof _this.silent === 'undefined')
            _this.silent = false;
        return _this;
    }
    PartnerRegistrationValidationAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'partner' },
            action: { type: 'c', default: 'registrationValidation' },
            partner: { type: 'o', subTypeConstructor: KalturaPartner_1.KalturaPartner, subType: 'KalturaPartner' },
            cmsPassword: { type: 's' },
            templatePartnerId: { type: 'n' },
            silent: { type: 'b' }
        });
        return result;
    };
    return PartnerRegistrationValidationAction;
}(kaltura_request_1.KalturaRequest));
exports.PartnerRegistrationValidationAction = PartnerRegistrationValidationAction;
