// kaltura-node-typescript-client/api/types/PartnerGetUsageAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartnerGetUsageAction = void 0;
var tslib_1 = require("tslib");
var KalturaPartnerUsage_1 = require("./KalturaPartnerUsage");
var KalturaReportInterval_1 = require("./KalturaReportInterval");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'partner' action 'getUsage'.
 *
 * Usage: Get usage statistics for a partner
 * Calculation is done according to partner's package
 * Additional data returned is a graph points of streaming usage in a time frame
 * The resolution can be "days" or "months"
 *
 * Server response type:         KalturaPartnerUsage
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var PartnerGetUsageAction = /** @class */ (function (_super) {
    tslib_1.__extends(PartnerGetUsageAction, _super);
    function PartnerGetUsageAction(data) {
        var _this = _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaPartnerUsage', responseConstructor: KalturaPartnerUsage_1.KalturaPartnerUsage }) || this;
        if (typeof _this.month === 'undefined')
            _this.month = 1;
        return _this;
    }
    PartnerGetUsageAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'partner' },
            action: { type: 'c', default: 'getUsage' },
            year: { type: 'n' },
            month: { type: 'n' },
            resolution: { type: 'es', subTypeConstructor: KalturaReportInterval_1.KalturaReportInterval, subType: 'KalturaReportInterval' }
        });
        return result;
    };
    return PartnerGetUsageAction;
}(kaltura_request_1.KalturaRequest));
exports.PartnerGetUsageAction = PartnerGetUsageAction;
