// kaltura-node-typescript-client/api/types/PartnerRegisterAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PartnerRegisterAction = void 0;
var tslib_1 = require("tslib");
var KalturaPartner_1 = require("./KalturaPartner");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'partner' action 'register'.
 *
 * Usage: Create a new Partner object
 *
 * Server response type:         KalturaPartner
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var PartnerRegisterAction = /** @class */ (function (_super) {
    tslib_1.__extends(PartnerRegisterAction, _super);
    function PartnerRegisterAction(data) {
        var _this = _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaPartner', responseConstructor: KalturaPartner_1.KalturaPartner }) || this;
        if (typeof _this.silent === 'undefined')
            _this.silent = false;
        return _this;
    }
    PartnerRegisterAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'partner' },
            action: { type: 'c', default: 'register' },
            partner: { type: 'o', subTypeConstructor: KalturaPartner_1.KalturaPartner, subType: 'KalturaPartner' },
            cmsPassword: { type: 's' },
            templatePartnerId: { type: 'n' },
            silent: { type: 'b' }
        });
        return result;
    };
    return PartnerRegisterAction;
}(kaltura_request_1.KalturaRequest));
exports.PartnerRegisterAction = PartnerRegisterAction;
