// kaltura-node-typescript-client/api/types/PexipListRoomsAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PexipListRoomsAction = void 0;
var tslib_1 = require("tslib");
var KalturaStringValue_1 = require("./KalturaStringValue");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'pexip' action 'listRooms'.
 *
 *
 *
 * Server response type:         KalturaStringValue[]
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var PexipListRoomsAction = /** @class */ (function (_super) {
    tslib_1.__extends(PexipListRoomsAction, _super);
    function PexipListRoomsAction(data) {
        var _this = _super.call(this, data, { responseType: 'a', responseSubType: 'KalturaStringValue', responseConstructor: KalturaStringValue_1.KalturaStringValue }) || this;
        if (typeof _this.pageSize === 'undefined')
            _this.pageSize = 500;
        if (typeof _this.activeOnly === 'undefined')
            _this.activeOnly = false;
        return _this;
    }
    PexipListRoomsAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'sip_pexip' },
            action: { type: 'c', default: 'listRooms' },
            offset: { type: 'n' },
            pageSize: { type: 'n' },
            activeOnly: { type: 'b' }
        });
        return result;
    };
    return PexipListRoomsAction;
}(kaltura_request_1.KalturaRequest));
exports.PexipListRoomsAction = PexipListRoomsAction;
