// kaltura-node-typescript-client/api/types/PexipGenerateSipUrlAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PexipGenerateSipUrlAction = void 0;
var tslib_1 = require("tslib");
var KalturaSipSourceType_1 = require("./KalturaSipSourceType");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'pexip' action 'generateSipUrl'.
 *
 *
 *
 * Server response type:         string
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var PexipGenerateSipUrlAction = /** @class */ (function (_super) {
    tslib_1.__extends(PexipGenerateSipUrlAction, _super);
    function PexipGenerateSipUrlAction(data) {
        var _this = _super.call(this, data, { responseType: 's', responseSubType: '', responseConstructor: null }) || this;
        if (typeof _this.regenerate === 'undefined')
            _this.regenerate = false;
        if (typeof _this.sourceType === 'undefined')
            _this.sourceType = 1;
        return _this;
    }
    PexipGenerateSipUrlAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'sip_pexip' },
            action: { type: 'c', default: 'generateSipUrl' },
            entryId: { type: 's' },
            regenerate: { type: 'b' },
            sourceType: { type: 'en', subTypeConstructor: KalturaSipSourceType_1.KalturaSipSourceType, subType: 'KalturaSipSourceType' }
        });
        return result;
    };
    return PexipGenerateSipUrlAction;
}(kaltura_request_1.KalturaRequest));
exports.PexipGenerateSipUrlAction = PexipGenerateSipUrlAction;
