// kaltura-node-typescript-client/api/types/PlaylistExecuteFromFiltersAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlaylistExecuteFromFiltersAction = void 0;
var tslib_1 = require("tslib");
var KalturaBaseEntry_1 = require("./KalturaBaseEntry");
var KalturaMediaEntryFilterForPlaylist_1 = require("./KalturaMediaEntryFilterForPlaylist");
var KalturaFilterPager_1 = require("./KalturaFilterPager");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'playlist' action 'executeFromFilters'.
 *
 * Usage: Retrieve playlist for playing purpose, based on media entry filters
 *
 * Server response type:         KalturaBaseEntry[]
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var PlaylistExecuteFromFiltersAction = /** @class */ (function (_super) {
    tslib_1.__extends(PlaylistExecuteFromFiltersAction, _super);
    function PlaylistExecuteFromFiltersAction(data) {
        var _this = _super.call(this, data, { responseType: 'a', responseSubType: 'KalturaBaseEntry', responseConstructor: KalturaBaseEntry_1.KalturaBaseEntry }) || this;
        if (typeof _this.filters === 'undefined')
            _this.filters = [];
        if (typeof _this.detailed === 'undefined')
            _this.detailed = "1";
        return _this;
    }
    PlaylistExecuteFromFiltersAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'playlist' },
            action: { type: 'c', default: 'executeFromFilters' },
            filters: { type: 'a', subTypeConstructor: KalturaMediaEntryFilterForPlaylist_1.KalturaMediaEntryFilterForPlaylist, subType: 'KalturaMediaEntryFilterForPlaylist' },
            totalResults: { type: 'n' },
            detailed: { type: 's' },
            pager: { type: 'o', subTypeConstructor: KalturaFilterPager_1.KalturaFilterPager, subType: 'KalturaFilterPager' }
        });
        return result;
    };
    return PlaylistExecuteFromFiltersAction;
}(kaltura_request_1.KalturaRequest));
exports.PlaylistExecuteFromFiltersAction = PlaylistExecuteFromFiltersAction;
