// kaltura-node-typescript-client/api/types/PlaylistExecuteFromContentAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlaylistExecuteFromContentAction = void 0;
var tslib_1 = require("tslib");
var KalturaBaseEntry_1 = require("./KalturaBaseEntry");
var KalturaPlaylistType_1 = require("./KalturaPlaylistType");
var KalturaFilterPager_1 = require("./KalturaFilterPager");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'playlist' action 'executeFromContent'.
 *
 * Usage: Retrieve playlist for playing purpose, based on content
 *
 * Server response type:         KalturaBaseEntry[]
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var PlaylistExecuteFromContentAction = /** @class */ (function (_super) {
    tslib_1.__extends(PlaylistExecuteFromContentAction, _super);
    function PlaylistExecuteFromContentAction(data) {
        return _super.call(this, data, { responseType: 'a', responseSubType: 'KalturaBaseEntry', responseConstructor: KalturaBaseEntry_1.KalturaBaseEntry }) || this;
    }
    PlaylistExecuteFromContentAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'playlist' },
            action: { type: 'c', default: 'executeFromContent' },
            playlistType: { type: 'en', subTypeConstructor: KalturaPlaylistType_1.KalturaPlaylistType, subType: 'KalturaPlaylistType' },
            playlistContent: { type: 's' },
            detailed: { type: 's' },
            pager: { type: 'o', subTypeConstructor: KalturaFilterPager_1.KalturaFilterPager, subType: 'KalturaFilterPager' }
        });
        return result;
    };
    return PlaylistExecuteFromContentAction;
}(kaltura_request_1.KalturaRequest));
exports.PlaylistExecuteFromContentAction = PlaylistExecuteFromContentAction;
