// kaltura-node-typescript-client/api/types/PlayReadyDrmGetLicenseDetailsAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlayReadyDrmGetLicenseDetailsAction = void 0;
var tslib_1 = require("tslib");
var KalturaPlayReadyLicenseDetails_1 = require("./KalturaPlayReadyLicenseDetails");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'playReadyDrm' action 'getLicenseDetails'.
 *
 * Usage: Get Play Ready policy and dates for license creation
 *
 * Server response type:         KalturaPlayReadyLicenseDetails
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var PlayReadyDrmGetLicenseDetailsAction = /** @class */ (function (_super) {
    tslib_1.__extends(PlayReadyDrmGetLicenseDetailsAction, _super);
    function PlayReadyDrmGetLicenseDetailsAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaPlayReadyLicenseDetails', responseConstructor: KalturaPlayReadyLicenseDetails_1.KalturaPlayReadyLicenseDetails }) || this;
    }
    PlayReadyDrmGetLicenseDetailsAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'playready_playreadydrm' },
            action: { type: 'c', default: 'getLicenseDetails' },
            keyId: { type: 's' },
            deviceId: { type: 's' },
            deviceType: { type: 'n' },
            entryId: { type: 's' },
            referrer: { type: 's' }
        });
        return result;
    };
    return PlayReadyDrmGetLicenseDetailsAction;
}(kaltura_request_1.KalturaRequest));
exports.PlayReadyDrmGetLicenseDetailsAction = PlayReadyDrmGetLicenseDetailsAction;
