// kaltura-node-typescript-client/api/types/PlaylistExecuteAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlaylistExecuteAction = void 0;
var tslib_1 = require("tslib");
var KalturaBaseEntry_1 = require("./KalturaBaseEntry");
var KalturaContext_1 = require("./KalturaContext");
var KalturaMediaEntryFilterForPlaylist_1 = require("./KalturaMediaEntryFilterForPlaylist");
var KalturaFilterPager_1 = require("./KalturaFilterPager");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'playlist' action 'execute'.
 *
 * Usage: Retrieve playlist for playing purpose
 *
 * Server response type:         KalturaBaseEntry[]
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var PlaylistExecuteAction = /** @class */ (function (_super) {
    tslib_1.__extends(PlaylistExecuteAction, _super);
    function PlaylistExecuteAction(data) {
        return _super.call(this, data, { responseType: 'a', responseSubType: 'KalturaBaseEntry', responseConstructor: KalturaBaseEntry_1.KalturaBaseEntry }) || this;
    }
    PlaylistExecuteAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'playlist' },
            action: { type: 'c', default: 'execute' },
            id: { type: 's' },
            detailed: { type: 's' },
            playlistContext: { type: 'o', subTypeConstructor: KalturaContext_1.KalturaContext, subType: 'KalturaContext' },
            filter: { type: 'o', subTypeConstructor: KalturaMediaEntryFilterForPlaylist_1.KalturaMediaEntryFilterForPlaylist, subType: 'KalturaMediaEntryFilterForPlaylist' },
            pager: { type: 'o', subTypeConstructor: KalturaFilterPager_1.KalturaFilterPager, subType: 'KalturaFilterPager' }
        });
        return result;
    };
    return PlaylistExecuteAction;
}(kaltura_request_1.KalturaRequest));
exports.PlaylistExecuteAction = PlaylistExecuteAction;
