// kaltura-node-typescript-client/api/types/PlaylistUpdateAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PlaylistUpdateAction = void 0;
var tslib_1 = require("tslib");
var KalturaPlaylist_1 = require("./KalturaPlaylist");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'playlist' action 'update'.
 *
 * Usage: Update existing playlist
 * Note - you cannot change playlist type. Updated playlist must be of the same type
 *
 * Server response type:         KalturaPlaylist
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var PlaylistUpdateAction = /** @class */ (function (_super) {
    tslib_1.__extends(PlaylistUpdateAction, _super);
    function PlaylistUpdateAction(data) {
        var _this = _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaPlaylist', responseConstructor: KalturaPlaylist_1.KalturaPlaylist }) || this;
        if (typeof _this.updateStats === 'undefined')
            _this.updateStats = false;
        return _this;
    }
    PlaylistUpdateAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'playlist' },
            action: { type: 'c', default: 'update' },
            id: { type: 's' },
            playlist: { type: 'o', subTypeConstructor: KalturaPlaylist_1.KalturaPlaylist, subType: 'KalturaPlaylist' },
            updateStats: { type: 'b' }
        });
        return result;
    };
    return PlaylistUpdateAction;
}(kaltura_request_1.KalturaRequest));
exports.PlaylistUpdateAction = PlaylistUpdateAction;
