// kaltura-node-typescript-client/api/types/PollVoteAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.PollVoteAction = void 0;
var tslib_1 = require("tslib");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'poll' action 'vote'.
 *
 * Usage: Vote Action
 *
 * Server response type:         string
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var PollVoteAction = /** @class */ (function (_super) {
    tslib_1.__extends(PollVoteAction, _super);
    function PollVoteAction(data) {
        return _super.call(this, data, { responseType: 's', responseSubType: '', responseConstructor: null }) || this;
    }
    PollVoteAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'poll_poll' },
            action: { type: 'c', default: 'vote' },
            pollId: { type: 's' },
            userId: { type: 's' },
            answerIds: { type: 's' }
        });
        return result;
    };
    return PollVoteAction;
}(kaltura_request_1.KalturaRequest));
exports.PollVoteAction = PollVoteAction;
