// kaltura-node-typescript-client/api/types/ReportGetUrlForReportAsCsvAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportGetUrlForReportAsCsvAction = void 0;
var tslib_1 = require("tslib");
var KalturaReportType_1 = require("./KalturaReportType");
var KalturaReportInputFilter_1 = require("./KalturaReportInputFilter");
var KalturaFilterPager_1 = require("./KalturaFilterPager");
var KalturaReportResponseOptions_1 = require("./KalturaReportResponseOptions");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'report' action 'getUrlForReportAsCsv'.
 *
 * Usage: will create a CSV file for the given report and return the URL to access it
 *
 * Server response type:         string
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var ReportGetUrlForReportAsCsvAction = /** @class */ (function (_super) {
    tslib_1.__extends(ReportGetUrlForReportAsCsvAction, _super);
    function ReportGetUrlForReportAsCsvAction(data) {
        return _super.call(this, data, { responseType: 's', responseSubType: '', responseConstructor: null }) || this;
    }
    ReportGetUrlForReportAsCsvAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'report' },
            action: { type: 'c', default: 'getUrlForReportAsCsv' },
            reportTitle: { type: 's' },
            reportText: { type: 's' },
            headers: { type: 's' },
            reportType: { type: 'es', subTypeConstructor: KalturaReportType_1.KalturaReportType, subType: 'KalturaReportType' },
            reportInputFilter: { type: 'o', subTypeConstructor: KalturaReportInputFilter_1.KalturaReportInputFilter, subType: 'KalturaReportInputFilter' },
            dimension: { type: 's' },
            pager: { type: 'o', subTypeConstructor: KalturaFilterPager_1.KalturaFilterPager, subType: 'KalturaFilterPager' },
            order: { type: 's' },
            objectIds: { type: 's' },
            responseOptions: { type: 'o', subTypeConstructor: KalturaReportResponseOptions_1.KalturaReportResponseOptions, subType: 'KalturaReportResponseOptions' }
        });
        return result;
    };
    return ReportGetUrlForReportAsCsvAction;
}(kaltura_request_1.KalturaRequest));
exports.ReportGetUrlForReportAsCsvAction = ReportGetUrlForReportAsCsvAction;
