// kaltura-node-typescript-client/api/types/ReportGetGraphsAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportGetGraphsAction = void 0;
var tslib_1 = require("tslib");
var KalturaReportGraph_1 = require("./KalturaReportGraph");
var KalturaReportType_1 = require("./KalturaReportType");
var KalturaReportInputFilter_1 = require("./KalturaReportInputFilter");
var KalturaReportResponseOptions_1 = require("./KalturaReportResponseOptions");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'report' action 'getGraphs'.
 *
 * Usage: report getGraphs action allows to get a graph data for a specific report
 *
 * Server response type:         KalturaReportGraph[]
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var ReportGetGraphsAction = /** @class */ (function (_super) {
    tslib_1.__extends(ReportGetGraphsAction, _super);
    function ReportGetGraphsAction(data) {
        return _super.call(this, data, { responseType: 'a', responseSubType: 'KalturaReportGraph', responseConstructor: KalturaReportGraph_1.KalturaReportGraph }) || this;
    }
    ReportGetGraphsAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'report' },
            action: { type: 'c', default: 'getGraphs' },
            reportType: { type: 'es', subTypeConstructor: KalturaReportType_1.KalturaReportType, subType: 'KalturaReportType' },
            reportInputFilter: { type: 'o', subTypeConstructor: KalturaReportInputFilter_1.KalturaReportInputFilter, subType: 'KalturaReportInputFilter' },
            dimension: { type: 's' },
            objectIds: { type: 's' },
            responseOptions: { type: 'o', subTypeConstructor: KalturaReportResponseOptions_1.KalturaReportResponseOptions, subType: 'KalturaReportResponseOptions' }
        });
        return result;
    };
    return ReportGetGraphsAction;
}(kaltura_request_1.KalturaRequest));
exports.ReportGetGraphsAction = ReportGetGraphsAction;
