// kaltura-node-typescript-client/api/types/ReportGetCsvFromStringParamsAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportGetCsvFromStringParamsAction = void 0;
var tslib_1 = require("tslib");
var kaltura_file_request_1 = require("../kaltura-file-request");
/**
 * Build request payload for service 'report' action 'getCsvFromStringParams'.
 *
 * Usage: Returns report CSV file executed by string params with the following convention: param1=value1;param2=value2
 * excludedFields can be supplied comma separated
 *
 * Server response type:         { url: string }
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaFileRequest
 */
var ReportGetCsvFromStringParamsAction = /** @class */ (function (_super) {
    tslib_1.__extends(ReportGetCsvFromStringParamsAction, _super);
    function ReportGetCsvFromStringParamsAction(data) {
        return _super.call(this, data) || this;
    }
    ReportGetCsvFromStringParamsAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'report' },
            action: { type: 'c', default: 'getCsvFromStringParams' },
            id: { type: 'n' },
            params: { type: 's' },
            excludedFields: { type: 's' }
        });
        return result;
    };
    return ReportGetCsvFromStringParamsAction;
}(kaltura_file_request_1.KalturaFileRequest));
exports.ReportGetCsvFromStringParamsAction = ReportGetCsvFromStringParamsAction;
