// kaltura-node-typescript-client/api/types/ReportGetBaseTotalAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportGetBaseTotalAction = void 0;
var tslib_1 = require("tslib");
var KalturaReportBaseTotal_1 = require("./KalturaReportBaseTotal");
var KalturaReportType_1 = require("./KalturaReportType");
var KalturaReportInputFilter_1 = require("./KalturaReportInputFilter");
var KalturaReportResponseOptions_1 = require("./KalturaReportResponseOptions");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'report' action 'getBaseTotal'.
 *
 * Usage: report getBaseTotal action allows to get the total base for storage reports
 *
 * Server response type:         KalturaReportBaseTotal[]
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var ReportGetBaseTotalAction = /** @class */ (function (_super) {
    tslib_1.__extends(ReportGetBaseTotalAction, _super);
    function ReportGetBaseTotalAction(data) {
        return _super.call(this, data, { responseType: 'a', responseSubType: 'KalturaReportBaseTotal', responseConstructor: KalturaReportBaseTotal_1.KalturaReportBaseTotal }) || this;
    }
    ReportGetBaseTotalAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'report' },
            action: { type: 'c', default: 'getBaseTotal' },
            reportType: { type: 'es', subTypeConstructor: KalturaReportType_1.KalturaReportType, subType: 'KalturaReportType' },
            reportInputFilter: { type: 'o', subTypeConstructor: KalturaReportInputFilter_1.KalturaReportInputFilter, subType: 'KalturaReportInputFilter' },
            objectIds: { type: 's' },
            responseOptions: { type: 'o', subTypeConstructor: KalturaReportResponseOptions_1.KalturaReportResponseOptions, subType: 'KalturaReportResponseOptions' }
        });
        return result;
    };
    return ReportGetBaseTotalAction;
}(kaltura_request_1.KalturaRequest));
exports.ReportGetBaseTotalAction = ReportGetBaseTotalAction;
