// kaltura-node-typescript-client/api/types/ReportGetTableAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportGetTableAction = void 0;
var tslib_1 = require("tslib");
var KalturaReportTable_1 = require("./KalturaReportTable");
var KalturaReportType_1 = require("./KalturaReportType");
var KalturaReportInputFilter_1 = require("./KalturaReportInputFilter");
var KalturaFilterPager_1 = require("./KalturaFilterPager");
var KalturaReportResponseOptions_1 = require("./KalturaReportResponseOptions");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'report' action 'getTable'.
 *
 * Usage: report getTable action allows to get a graph data for a specific report
 *
 * Server response type:         KalturaReportTable
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var ReportGetTableAction = /** @class */ (function (_super) {
    tslib_1.__extends(ReportGetTableAction, _super);
    function ReportGetTableAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaReportTable', responseConstructor: KalturaReportTable_1.KalturaReportTable }) || this;
    }
    ReportGetTableAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'report' },
            action: { type: 'c', default: 'getTable' },
            reportType: { type: 'es', subTypeConstructor: KalturaReportType_1.KalturaReportType, subType: 'KalturaReportType' },
            reportInputFilter: { type: 'o', subTypeConstructor: KalturaReportInputFilter_1.KalturaReportInputFilter, subType: 'KalturaReportInputFilter' },
            pager: { type: 'o', subTypeConstructor: KalturaFilterPager_1.KalturaFilterPager, subType: 'KalturaFilterPager' },
            order: { type: 's' },
            objectIds: { type: 's' },
            responseOptions: { type: 'o', subTypeConstructor: KalturaReportResponseOptions_1.KalturaReportResponseOptions, subType: 'KalturaReportResponseOptions' }
        });
        return result;
    };
    return ReportGetTableAction;
}(kaltura_request_1.KalturaRequest));
exports.ReportGetTableAction = ReportGetTableAction;
