// kaltura-node-typescript-client/api/types/ReportGetCsvAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportGetCsvAction = void 0;
var tslib_1 = require("tslib");
var KalturaKeyValue_1 = require("./KalturaKeyValue");
var kaltura_file_request_1 = require("../kaltura-file-request");
/**
 * Build request payload for service 'report' action 'getCsv'.
 *
 *
 *
 * Server response type:         { url: string }
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaFileRequest
 */
var ReportGetCsvAction = /** @class */ (function (_super) {
    tslib_1.__extends(ReportGetCsvAction, _super);
    function ReportGetCsvAction(data) {
        var _this = _super.call(this, data) || this;
        if (typeof _this.params === 'undefined')
            _this.params = [];
        return _this;
    }
    ReportGetCsvAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'report' },
            action: { type: 'c', default: 'getCsv' },
            id: { type: 'n' },
            params: { type: 'a', subTypeConstructor: KalturaKeyValue_1.KalturaKeyValue, subType: 'KalturaKeyValue' },
            excludedFields: { type: 's' }
        });
        return result;
    };
    return ReportGetCsvAction;
}(kaltura_file_request_1.KalturaFileRequest));
exports.ReportGetCsvAction = ReportGetCsvAction;
