// kaltura-node-typescript-client/api/types/ReportGetTotalAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ReportGetTotalAction = void 0;
var tslib_1 = require("tslib");
var KalturaReportTotal_1 = require("./KalturaReportTotal");
var KalturaReportType_1 = require("./KalturaReportType");
var KalturaReportInputFilter_1 = require("./KalturaReportInputFilter");
var KalturaReportResponseOptions_1 = require("./KalturaReportResponseOptions");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'report' action 'getTotal'.
 *
 * Usage: report getTotal action allows to get a graph data for a specific report
 *
 * Server response type:         KalturaReportTotal
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var ReportGetTotalAction = /** @class */ (function (_super) {
    tslib_1.__extends(ReportGetTotalAction, _super);
    function ReportGetTotalAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaReportTotal', responseConstructor: KalturaReportTotal_1.KalturaReportTotal }) || this;
    }
    ReportGetTotalAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'report' },
            action: { type: 'c', default: 'getTotal' },
            reportType: { type: 'es', subTypeConstructor: KalturaReportType_1.KalturaReportType, subType: 'KalturaReportType' },
            reportInputFilter: { type: 'o', subTypeConstructor: KalturaReportInputFilter_1.KalturaReportInputFilter, subType: 'KalturaReportInputFilter' },
            objectIds: { type: 's' },
            responseOptions: { type: 'o', subTypeConstructor: KalturaReportResponseOptions_1.KalturaReportResponseOptions, subType: 'KalturaReportResponseOptions' }
        });
        return result;
    };
    return ReportGetTotalAction;
}(kaltura_request_1.KalturaRequest));
exports.ReportGetTotalAction = ReportGetTotalAction;
