// kaltura-node-typescript-client/api/types/ScheduleEventResourceUpdateAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleEventResourceUpdateAction = void 0;
var tslib_1 = require("tslib");
var KalturaScheduleEventResource_1 = require("./KalturaScheduleEventResource");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'scheduleEventResource' action 'update'.
 *
 * Usage: Update an existing KalturaScheduleEventResource object
 *
 * Server response type:         KalturaScheduleEventResource
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var ScheduleEventResourceUpdateAction = /** @class */ (function (_super) {
    tslib_1.__extends(ScheduleEventResourceUpdateAction, _super);
    function ScheduleEventResourceUpdateAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaScheduleEventResource', responseConstructor: KalturaScheduleEventResource_1.KalturaScheduleEventResource }) || this;
    }
    ScheduleEventResourceUpdateAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'schedule_scheduleeventresource' },
            action: { type: 'c', default: 'update' },
            scheduleEventId: { type: 'n' },
            scheduleResourceId: { type: 'n' },
            scheduleEventResource: { type: 'o', subTypeConstructor: KalturaScheduleEventResource_1.KalturaScheduleEventResource, subType: 'KalturaScheduleEventResource' }
        });
        return result;
    };
    return ScheduleEventResourceUpdateAction;
}(kaltura_request_1.KalturaRequest));
exports.ScheduleEventResourceUpdateAction = ScheduleEventResourceUpdateAction;
