// kaltura-node-typescript-client/api/types/ScheduleEventUpdateLiveFeatureAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleEventUpdateLiveFeatureAction = void 0;
var tslib_1 = require("tslib");
var KalturaLiveStreamScheduleEvent_1 = require("./KalturaLiveStreamScheduleEvent");
var KalturaLiveFeature_1 = require("./KalturaLiveFeature");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'scheduleEvent' action 'updateLiveFeature'.
 *
 * Usage: Add feature to live event
 *
 * Server response type:         KalturaLiveStreamScheduleEvent
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var ScheduleEventUpdateLiveFeatureAction = /** @class */ (function (_super) {
    tslib_1.__extends(ScheduleEventUpdateLiveFeatureAction, _super);
    function ScheduleEventUpdateLiveFeatureAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaLiveStreamScheduleEvent', responseConstructor: KalturaLiveStreamScheduleEvent_1.KalturaLiveStreamScheduleEvent }) || this;
    }
    ScheduleEventUpdateLiveFeatureAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'schedule_scheduleevent' },
            action: { type: 'c', default: 'updateLiveFeature' },
            scheduledEventId: { type: 'n' },
            featureName: { type: 's' },
            liveFeature: { type: 'o', subTypeConstructor: KalturaLiveFeature_1.KalturaLiveFeature, subType: 'KalturaLiveFeature' }
        });
        return result;
    };
    return ScheduleEventUpdateLiveFeatureAction;
}(kaltura_request_1.KalturaRequest));
exports.ScheduleEventUpdateLiveFeatureAction = ScheduleEventUpdateLiveFeatureAction;
