// kaltura-node-typescript-client/api/types/ScheduleEventGetConflictsAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleEventGetConflictsAction = void 0;
var tslib_1 = require("tslib");
var KalturaScheduleEventListResponse_1 = require("./KalturaScheduleEventListResponse");
var KalturaScheduleEvent_1 = require("./KalturaScheduleEvent");
var KalturaScheduleEventConflictType_1 = require("./KalturaScheduleEventConflictType");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'scheduleEvent' action 'getConflicts'.
 *
 * Usage: List conflicting events for resourcesIds by event's dates
 *
 * Server response type:         KalturaScheduleEventListResponse
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var ScheduleEventGetConflictsAction = /** @class */ (function (_super) {
    tslib_1.__extends(ScheduleEventGetConflictsAction, _super);
    function ScheduleEventGetConflictsAction(data) {
        var _this = _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaScheduleEventListResponse', responseConstructor: KalturaScheduleEventListResponse_1.KalturaScheduleEventListResponse }) || this;
        if (typeof _this.scheduleEventConflictType === 'undefined')
            _this.scheduleEventConflictType = 1;
        return _this;
    }
    ScheduleEventGetConflictsAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'schedule_scheduleevent' },
            action: { type: 'c', default: 'getConflicts' },
            resourceIds: { type: 's' },
            scheduleEvent: { type: 'o', subTypeConstructor: KalturaScheduleEvent_1.KalturaScheduleEvent, subType: 'KalturaScheduleEvent' },
            scheduleEventIdToIgnore: { type: 's' },
            scheduleEventConflictType: { type: 'en', subTypeConstructor: KalturaScheduleEventConflictType_1.KalturaScheduleEventConflictType, subType: 'KalturaScheduleEventConflictType' }
        });
        return result;
    };
    return ScheduleEventGetConflictsAction;
}(kaltura_request_1.KalturaRequest));
exports.ScheduleEventGetConflictsAction = ScheduleEventGetConflictsAction;
