// kaltura-node-typescript-client/api/types/ScheduleEventResourceListAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ScheduleEventResourceListAction = void 0;
var tslib_1 = require("tslib");
var KalturaScheduleEventResourceListResponse_1 = require("./KalturaScheduleEventResourceListResponse");
var KalturaScheduleEventResourceFilter_1 = require("./KalturaScheduleEventResourceFilter");
var KalturaFilterPager_1 = require("./KalturaFilterPager");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'scheduleEventResource' action 'list'.
 *
 * Usage: List KalturaScheduleEventResource objects
 *
 * Server response type:         KalturaScheduleEventResourceListResponse
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var ScheduleEventResourceListAction = /** @class */ (function (_super) {
    tslib_1.__extends(ScheduleEventResourceListAction, _super);
    function ScheduleEventResourceListAction(data) {
        var _this = _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaScheduleEventResourceListResponse', responseConstructor: KalturaScheduleEventResourceListResponse_1.KalturaScheduleEventResourceListResponse }) || this;
        if (typeof _this.filterBlackoutConflicts === 'undefined')
            _this.filterBlackoutConflicts = true;
        return _this;
    }
    ScheduleEventResourceListAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'schedule_scheduleeventresource' },
            action: { type: 'c', default: 'list' },
            filter: { type: 'o', subTypeConstructor: KalturaScheduleEventResourceFilter_1.KalturaScheduleEventResourceFilter, subType: 'KalturaScheduleEventResourceFilter' },
            pager: { type: 'o', subTypeConstructor: KalturaFilterPager_1.KalturaFilterPager, subType: 'KalturaFilterPager' },
            filterBlackoutConflicts: { type: 'b' }
        });
        return result;
    };
    return ScheduleEventResourceListAction;
}(kaltura_request_1.KalturaRequest));
exports.ScheduleEventResourceListAction = ScheduleEventResourceListAction;
