// kaltura-node-typescript-client/api/types/SearchExternalLoginAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SearchExternalLoginAction = void 0;
var tslib_1 = require("tslib");
var KalturaSearchAuthData_1 = require("./KalturaSearchAuthData");
var KalturaSearchProviderType_1 = require("./KalturaSearchProviderType");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'search' action 'externalLogin'.
 *
 *
 *
 * Server response type:         KalturaSearchAuthData
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var SearchExternalLoginAction = /** @class */ (function (_super) {
    tslib_1.__extends(SearchExternalLoginAction, _super);
    function SearchExternalLoginAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaSearchAuthData', responseConstructor: KalturaSearchAuthData_1.KalturaSearchAuthData }) || this;
    }
    SearchExternalLoginAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'search' },
            action: { type: 'c', default: 'externalLogin' },
            searchSource: { type: 'en', subTypeConstructor: KalturaSearchProviderType_1.KalturaSearchProviderType, subType: 'KalturaSearchProviderType' },
            userName: { type: 's' },
            password: { type: 's' }
        });
        return result;
    };
    return SearchExternalLoginAction;
}(kaltura_request_1.KalturaRequest));
exports.SearchExternalLoginAction = SearchExternalLoginAction;
