// kaltura-node-typescript-client/api/types/ServerNodeReportStatusAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerNodeReportStatusAction = void 0;
var tslib_1 = require("tslib");
var KalturaServerNode_1 = require("./KalturaServerNode");
var KalturaServerNodeStatus_1 = require("./KalturaServerNodeStatus");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'serverNode' action 'reportStatus'.
 *
 * Usage: Update server node status
 *
 * Server response type:         KalturaServerNode
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var ServerNodeReportStatusAction = /** @class */ (function (_super) {
    tslib_1.__extends(ServerNodeReportStatusAction, _super);
    function ServerNodeReportStatusAction(data) {
        var _this = _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaServerNode', responseConstructor: KalturaServerNode_1.KalturaServerNode }) || this;
        if (typeof _this.serverNodeStatus === 'undefined')
            _this.serverNodeStatus = 1;
        return _this;
    }
    ServerNodeReportStatusAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'servernode' },
            action: { type: 'c', default: 'reportStatus' },
            hostName: { type: 's' },
            serverNode: { type: 'o', subTypeConstructor: KalturaServerNode_1.KalturaServerNode, subType: 'KalturaServerNode' },
            serverNodeStatus: { type: 'en', subTypeConstructor: KalturaServerNodeStatus_1.KalturaServerNodeStatus, subType: 'KalturaServerNodeStatus' }
        });
        return result;
    };
    return ServerNodeReportStatusAction;
}(kaltura_request_1.KalturaRequest));
exports.ServerNodeReportStatusAction = ServerNodeReportStatusAction;
