// kaltura-node-typescript-client/api/types/ServerNodeGetFullPathAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ServerNodeGetFullPathAction = void 0;
var tslib_1 = require("tslib");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'serverNode' action 'getFullPath'.
 *
 * Usage: Get the edge server node full path
 *
 * Server response type:         string
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var ServerNodeGetFullPathAction = /** @class */ (function (_super) {
    tslib_1.__extends(ServerNodeGetFullPathAction, _super);
    function ServerNodeGetFullPathAction(data) {
        var _this = _super.call(this, data, { responseType: 's', responseSubType: '', responseConstructor: null }) || this;
        if (typeof _this.protocol === 'undefined')
            _this.protocol = "http";
        return _this;
    }
    ServerNodeGetFullPathAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'servernode' },
            action: { type: 'c', default: 'getFullPath' },
            hostName: { type: 's' },
            protocol: { type: 's' },
            deliveryFormat: { type: 's' },
            deliveryType: { type: 's' }
        });
        return result;
    };
    return ServerNodeGetFullPathAction;
}(kaltura_request_1.KalturaRequest));
exports.ServerNodeGetFullPathAction = ServerNodeGetFullPathAction;
