// kaltura-node-typescript-client/api/types/SessionImpersonateByKsAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SessionImpersonateByKsAction = void 0;
var tslib_1 = require("tslib");
var KalturaSessionInfo_1 = require("./KalturaSessionInfo");
var KalturaSessionType_1 = require("./KalturaSessionType");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'session' action 'impersonateByKs'.
 *
 * Usage: Start an impersonated session with Kaltura's server.
 * The result KS info contains the session key that you should pass to all services that requires a ticket.
 * Type, expiry and privileges won't be changed if they're not set
 *
 * Server response type:         KalturaSessionInfo
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var SessionImpersonateByKsAction = /** @class */ (function (_super) {
    tslib_1.__extends(SessionImpersonateByKsAction, _super);
    function SessionImpersonateByKsAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaSessionInfo', responseConstructor: KalturaSessionInfo_1.KalturaSessionInfo }) || this;
    }
    SessionImpersonateByKsAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'session' },
            action: { type: 'c', default: 'impersonateByKs' },
            session: { type: 's' },
            type: { type: 'en', subTypeConstructor: KalturaSessionType_1.KalturaSessionType, subType: 'KalturaSessionType' },
            expiry: { type: 'n' },
            privileges: { type: 's' }
        });
        return result;
    };
    return SessionImpersonateByKsAction;
}(kaltura_request_1.KalturaRequest));
exports.SessionImpersonateByKsAction = SessionImpersonateByKsAction;
