// kaltura-node-typescript-client/api/types/SessionStartAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SessionStartAction = void 0;
var tslib_1 = require("tslib");
var KalturaSessionType_1 = require("./KalturaSessionType");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'session' action 'start'.
 *
 * Usage: Start a session with Kaltura's server.
 * The result KS is the session key that you should pass to all services that requires a ticket
 *
 * Server response type:         string
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var SessionStartAction = /** @class */ (function (_super) {
    tslib_1.__extends(SessionStartAction, _super);
    function SessionStartAction(data) {
        var _this = _super.call(this, data, { responseType: 's', responseSubType: '', responseConstructor: null }) || this;
        if (typeof _this.expiry === 'undefined')
            _this.expiry = 86400;
        return _this;
    }
    SessionStartAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'session' },
            action: { type: 'c', default: 'start' },
            secret: { type: 's' },
            userId: { type: 's' },
            type: { type: 'en', subTypeConstructor: KalturaSessionType_1.KalturaSessionType, subType: 'KalturaSessionType' },
            partnerId: { type: 'n' },
            expiry: { type: 'n' },
            privileges: { type: 's' }
        });
        return result;
    };
    return SessionStartAction;
}(kaltura_request_1.KalturaRequest));
exports.SessionStartAction = SessionStartAction;
