// kaltura-node-typescript-client/api/types/SharepointExtensionIsVersionSupportedAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SharepointExtensionIsVersionSupportedAction = void 0;
var tslib_1 = require("tslib");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'sharepointExtension' action 'isVersionSupported'.
 *
 * Usage: Is this Kaltura-Sharepoint-Server-Plugin supports minimum version of $major.$minor.$build (which is required by the extension)
 *
 * Server response type:         boolean
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var SharepointExtensionIsVersionSupportedAction = /** @class */ (function (_super) {
    tslib_1.__extends(SharepointExtensionIsVersionSupportedAction, _super);
    function SharepointExtensionIsVersionSupportedAction(data) {
        return _super.call(this, data, { responseType: 'b', responseSubType: '', responseConstructor: null }) || this;
    }
    SharepointExtensionIsVersionSupportedAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'kalturasharepointextension_sharepointextension' },
            action: { type: 'c', default: 'isVersionSupported' },
            serverMajor: { type: 'n' },
            serverMinor: { type: 'n' },
            serverBuild: { type: 'n' }
        });
        return result;
    };
    return SharepointExtensionIsVersionSupportedAction;
}(kaltura_request_1.KalturaRequest));
exports.SharepointExtensionIsVersionSupportedAction = SharepointExtensionIsVersionSupportedAction;
