// kaltura-node-typescript-client/api/types/SsoLoginAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SsoLoginAction = void 0;
var tslib_1 = require("tslib");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'sso' action 'login'.
 *
 * Usage: Login with SSO, getting redirect url according to application type and partner Id
 * or according to application type and domain
 *
 * Server response type:         string
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var SsoLoginAction = /** @class */ (function (_super) {
    tslib_1.__extends(SsoLoginAction, _super);
    function SsoLoginAction(data) {
        return _super.call(this, data, { responseType: 's', responseSubType: '', responseConstructor: null }) || this;
    }
    SsoLoginAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'sso_sso' },
            action: { type: 'c', default: 'login' },
            userId: { type: 's' },
            applicationType: { type: 's' },
            partnerId: { type: 'n' }
        });
        return result;
    };
    return SsoLoginAction;
}(kaltura_request_1.KalturaRequest));
exports.SsoLoginAction = SsoLoginAction;
