// kaltura-node-typescript-client/api/types/StorageProfileLockPendingFileSyncsAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.StorageProfileLockPendingFileSyncsAction = void 0;
var tslib_1 = require("tslib");
var KalturaLockFileSyncsResponse_1 = require("./KalturaLockFileSyncsResponse");
var KalturaFileSyncFilter_1 = require("./KalturaFileSyncFilter");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'storageProfile' action 'lockPendingFileSyncs'.
 *
 * Usage: storage profile lockPendingFileSyncs action locks file syncs for export by the file sync periodic worker
 *
 * Server response type:         KalturaLockFileSyncsResponse
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var StorageProfileLockPendingFileSyncsAction = /** @class */ (function (_super) {
    tslib_1.__extends(StorageProfileLockPendingFileSyncsAction, _super);
    function StorageProfileLockPendingFileSyncsAction(data) {
        var _this = _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaLockFileSyncsResponse', responseConstructor: KalturaLockFileSyncsResponse_1.KalturaLockFileSyncsResponse }) || this;
        if (typeof _this.maxSize === 'undefined')
            _this.maxSize = 9223372036854775807;
        return _this;
    }
    StorageProfileLockPendingFileSyncsAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'storageprofile' },
            action: { type: 'c', default: 'lockPendingFileSyncs' },
            filter: { type: 'o', subTypeConstructor: KalturaFileSyncFilter_1.KalturaFileSyncFilter, subType: 'KalturaFileSyncFilter' },
            workerId: { type: 'n' },
            storageProfileId: { type: 'n' },
            maxCount: { type: 'n' },
            maxSize: { type: 'n' }
        });
        return result;
    };
    return StorageProfileLockPendingFileSyncsAction;
}(kaltura_request_1.KalturaRequest));
exports.StorageProfileLockPendingFileSyncsAction = StorageProfileLockPendingFileSyncsAction;
