// kaltura-node-typescript-client/api/types/TagIndexCategoryEntryTagsAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TagIndexCategoryEntryTagsAction = void 0;
var tslib_1 = require("tslib");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'tag' action 'indexCategoryEntryTags'.
 *
 *
 *
 * Server response type:         void
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var TagIndexCategoryEntryTagsAction = /** @class */ (function (_super) {
    tslib_1.__extends(TagIndexCategoryEntryTagsAction, _super);
    function TagIndexCategoryEntryTagsAction(data) {
        return _super.call(this, data, { responseType: 'v', responseSubType: '', responseConstructor: null }) || this;
    }
    TagIndexCategoryEntryTagsAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'tagsearch_tag' },
            action: { type: 'c', default: 'indexCategoryEntryTags' },
            categoryId: { type: 'n' },
            pcToDecrement: { type: 's' },
            pcToIncrement: { type: 's' }
        });
        return result;
    };
    return TagIndexCategoryEntryTagsAction;
}(kaltura_request_1.KalturaRequest));
exports.TagIndexCategoryEntryTagsAction = TagIndexCategoryEntryTagsAction;
