// kaltura-node-typescript-client/api/types/ThumbAssetGetUrlAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThumbAssetGetUrlAction = void 0;
var tslib_1 = require("tslib");
var KalturaThumbParams_1 = require("./KalturaThumbParams");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'thumbAsset' action 'getUrl'.
 *
 * Usage: Get download URL for the asset
 *
 * Server response type:         string
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var ThumbAssetGetUrlAction = /** @class */ (function (_super) {
    tslib_1.__extends(ThumbAssetGetUrlAction, _super);
    function ThumbAssetGetUrlAction(data) {
        return _super.call(this, data, { responseType: 's', responseSubType: '', responseConstructor: null }) || this;
    }
    ThumbAssetGetUrlAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'thumbasset' },
            action: { type: 'c', default: 'getUrl' },
            id: { type: 's' },
            storageId: { type: 'n' },
            thumbParams: { type: 'o', subTypeConstructor: KalturaThumbParams_1.KalturaThumbParams, subType: 'KalturaThumbParams' }
        });
        return result;
    };
    return ThumbAssetGetUrlAction;
}(kaltura_request_1.KalturaRequest));
exports.ThumbAssetGetUrlAction = ThumbAssetGetUrlAction;
