// kaltura-node-typescript-client/api/types/ThumbAssetGenerateAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThumbAssetGenerateAction = void 0;
var tslib_1 = require("tslib");
var KalturaThumbAsset_1 = require("./KalturaThumbAsset");
var KalturaThumbParams_1 = require("./KalturaThumbParams");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'thumbAsset' action 'generate'.
 *
 *
 *
 * Server response type:         KalturaThumbAsset
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var ThumbAssetGenerateAction = /** @class */ (function (_super) {
    tslib_1.__extends(ThumbAssetGenerateAction, _super);
    function ThumbAssetGenerateAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaThumbAsset', responseConstructor: KalturaThumbAsset_1.KalturaThumbAsset }) || this;
    }
    ThumbAssetGenerateAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'thumbasset' },
            action: { type: 'c', default: 'generate' },
            entryId: { type: 's' },
            thumbParams: { type: 'o', subTypeConstructor: KalturaThumbParams_1.KalturaThumbParams, subType: 'KalturaThumbParams' },
            sourceAssetId: { type: 's' }
        });
        return result;
    };
    return ThumbAssetGenerateAction;
}(kaltura_request_1.KalturaRequest));
exports.ThumbAssetGenerateAction = ThumbAssetGenerateAction;
