// kaltura-node-typescript-client/api/types/ThumbAssetServeAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ThumbAssetServeAction = void 0;
var tslib_1 = require("tslib");
var KalturaThumbParams_1 = require("./KalturaThumbParams");
var KalturaThumbnailServeOptions_1 = require("./KalturaThumbnailServeOptions");
var kaltura_file_request_1 = require("../kaltura-file-request");
/**
 * Build request payload for service 'thumbAsset' action 'serve'.
 *
 * Usage: Serves thumbnail by its id
 *
 * Server response type:         { url: string }
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaFileRequest
 */
var ThumbAssetServeAction = /** @class */ (function (_super) {
    tslib_1.__extends(ThumbAssetServeAction, _super);
    function ThumbAssetServeAction(data) {
        return _super.call(this, data) || this;
    }
    ThumbAssetServeAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'thumbasset' },
            action: { type: 'c', default: 'serve' },
            thumbAssetId: { type: 's' },
            version: { type: 'n' },
            thumbParams: { type: 'o', subTypeConstructor: KalturaThumbParams_1.KalturaThumbParams, subType: 'KalturaThumbParams' },
            options: { type: 'o', subTypeConstructor: KalturaThumbnailServeOptions_1.KalturaThumbnailServeOptions, subType: 'KalturaThumbnailServeOptions' }
        });
        return result;
    };
    return ThumbAssetServeAction;
}(kaltura_file_request_1.KalturaFileRequest));
exports.ThumbAssetServeAction = ThumbAssetServeAction;
