// kaltura-node-typescript-client/api/types/UserScoreDeleteAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserScoreDeleteAction = void 0;
var tslib_1 = require("tslib");
var KalturaUserScorePropertiesResponse_1 = require("./KalturaUserScorePropertiesResponse");
var KalturaGameObjectType_1 = require("./KalturaGameObjectType");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'userScore' action 'delete'.
 *
 *
 *
 * Server response type:         KalturaUserScorePropertiesResponse
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var UserScoreDeleteAction = /** @class */ (function (_super) {
    tslib_1.__extends(UserScoreDeleteAction, _super);
    function UserScoreDeleteAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaUserScorePropertiesResponse', responseConstructor: KalturaUserScorePropertiesResponse_1.KalturaUserScorePropertiesResponse }) || this;
    }
    UserScoreDeleteAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'game_userscore' },
            action: { type: 'c', default: 'delete' },
            gameObjectId: { type: 's' },
            gameObjectType: { type: 'es', subTypeConstructor: KalturaGameObjectType_1.KalturaGameObjectType, subType: 'KalturaGameObjectType' },
            userId: { type: 's' }
        });
        return result;
    };
    return UserScoreDeleteAction;
}(kaltura_request_1.KalturaRequest));
exports.UserScoreDeleteAction = UserScoreDeleteAction;
