// kaltura-node-typescript-client/api/types/UserUpdateLoginDataAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserUpdateLoginDataAction = void 0;
var tslib_1 = require("tslib");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'user' action 'updateLoginData'.
 *
 * Usage: Updates a user's login data: email, password, name
 *
 * Server response type:         void
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var UserUpdateLoginDataAction = /** @class */ (function (_super) {
    tslib_1.__extends(UserUpdateLoginDataAction, _super);
    function UserUpdateLoginDataAction(data) {
        return _super.call(this, data, { responseType: 'v', responseSubType: '', responseConstructor: null }) || this;
    }
    UserUpdateLoginDataAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'user' },
            action: { type: 'c', default: 'updateLoginData' },
            oldLoginId: { type: 's' },
            password: { type: 's' },
            newLoginId: { type: 's' },
            newPassword: { type: 's' },
            newFirstName: { type: 's' },
            newLastName: { type: 's' },
            otp: { type: 's' }
        });
        return result;
    };
    return UserUpdateLoginDataAction;
}(kaltura_request_1.KalturaRequest));
exports.UserUpdateLoginDataAction = UserUpdateLoginDataAction;
