// kaltura-node-typescript-client/api/types/UserAddFromBulkUploadAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserAddFromBulkUploadAction = void 0;
var tslib_1 = require("tslib");
var KalturaBulkUpload_1 = require("./KalturaBulkUpload");
var KalturaBulkUploadJobData_1 = require("./KalturaBulkUploadJobData");
var KalturaBulkUploadUserData_1 = require("./KalturaBulkUploadUserData");
var kaltura_upload_request_1 = require("../kaltura-upload-request");
/**
 * Build request payload for service 'user' action 'addFromBulkUpload'.
 *
 *
 *
 * Server response type:         KalturaBulkUpload
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaUploadRequest
 */
var UserAddFromBulkUploadAction = /** @class */ (function (_super) {
    tslib_1.__extends(UserAddFromBulkUploadAction, _super);
    function UserAddFromBulkUploadAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaBulkUpload', responseConstructor: KalturaBulkUpload_1.KalturaBulkUpload }) || this;
    }
    UserAddFromBulkUploadAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'user' },
            action: { type: 'c', default: 'addFromBulkUpload' },
            fileData: { type: 'f' },
            bulkUploadData: { type: 'o', subTypeConstructor: KalturaBulkUploadJobData_1.KalturaBulkUploadJobData, subType: 'KalturaBulkUploadJobData' },
            bulkUploadUserData: { type: 'o', subTypeConstructor: KalturaBulkUploadUserData_1.KalturaBulkUploadUserData, subType: 'KalturaBulkUploadUserData' }
        });
        return result;
    };
    return UserAddFromBulkUploadAction;
}(kaltura_upload_request_1.KalturaUploadRequest));
exports.UserAddFromBulkUploadAction = UserAddFromBulkUploadAction;
