// kaltura-node-typescript-client/api/types/UserReplaceUserLoginDataAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserReplaceUserLoginDataAction = void 0;
var tslib_1 = require("tslib");
var KalturaUser_1 = require("./KalturaUser");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'user' action 'replaceUserLoginData'.
 *
 * Usage: Replace a user's existing login data to a new or an existing login data
 * to only be used when admin impersonates a partner
 *
 * Server response type:         KalturaUser
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var UserReplaceUserLoginDataAction = /** @class */ (function (_super) {
    tslib_1.__extends(UserReplaceUserLoginDataAction, _super);
    function UserReplaceUserLoginDataAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaUser', responseConstructor: KalturaUser_1.KalturaUser }) || this;
    }
    UserReplaceUserLoginDataAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'user' },
            action: { type: 'c', default: 'replaceUserLoginData' },
            userId: { type: 's' },
            newLoginId: { type: 's' },
            existingLoginId: { type: 's' }
        });
        return result;
    };
    return UserReplaceUserLoginDataAction;
}(kaltura_request_1.KalturaRequest));
exports.UserReplaceUserLoginDataAction = UserReplaceUserLoginDataAction;
