// kaltura-node-typescript-client/api/types/UserAppRoleUpdateAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserAppRoleUpdateAction = void 0;
var tslib_1 = require("tslib");
var KalturaUserAppRole_1 = require("./KalturaUserAppRole");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'userAppRole' action 'update'.
 *
 * Usage: Update an application role for a user
 *
 * Server response type:         KalturaUserAppRole
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var UserAppRoleUpdateAction = /** @class */ (function (_super) {
    tslib_1.__extends(UserAppRoleUpdateAction, _super);
    function UserAppRoleUpdateAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaUserAppRole', responseConstructor: KalturaUserAppRole_1.KalturaUserAppRole }) || this;
    }
    UserAppRoleUpdateAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'userapprole' },
            action: { type: 'c', default: 'update' },
            userId: { type: 's' },
            appGuid: { type: 's' },
            userAppRole: { type: 'o', subTypeConstructor: KalturaUserAppRole_1.KalturaUserAppRole, subType: 'KalturaUserAppRole' }
        });
        return result;
    };
    return UserAppRoleUpdateAction;
}(kaltura_request_1.KalturaRequest));
exports.UserAppRoleUpdateAction = UserAppRoleUpdateAction;
