// kaltura-node-typescript-client/api/types/UserEnableLoginAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserEnableLoginAction = void 0;
var tslib_1 = require("tslib");
var KalturaUser_1 = require("./KalturaUser");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'user' action 'enableLogin'.
 *
 * Usage: Enables a user to log into a partner account using an email address and a password
 *
 * Server response type:         KalturaUser
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var UserEnableLoginAction = /** @class */ (function (_super) {
    tslib_1.__extends(UserEnableLoginAction, _super);
    function UserEnableLoginAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaUser', responseConstructor: KalturaUser_1.KalturaUser }) || this;
    }
    UserEnableLoginAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'user' },
            action: { type: 'c', default: 'enableLogin' },
            userId: { type: 's' },
            loginId: { type: 's' },
            password: { type: 's' }
        });
        return result;
    };
    return UserEnableLoginAction;
}(kaltura_request_1.KalturaRequest));
exports.UserEnableLoginAction = UserEnableLoginAction;
