// kaltura-node-typescript-client/api/types/UserLoginByLoginIdAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserLoginByLoginIdAction = void 0;
var tslib_1 = require("tslib");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'user' action 'loginByLoginId'.
 *
 * Usage: Logs a user into a partner account with a user login ID and a user password
 *
 * Server response type:         string
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var UserLoginByLoginIdAction = /** @class */ (function (_super) {
    tslib_1.__extends(UserLoginByLoginIdAction, _super);
    function UserLoginByLoginIdAction(data) {
        var _this = _super.call(this, data, { responseType: 's', responseSubType: '', responseConstructor: null }) || this;
        if (typeof _this.expiry === 'undefined')
            _this.expiry = 86400;
        if (typeof _this.privileges === 'undefined')
            _this.privileges = "*";
        return _this;
    }
    UserLoginByLoginIdAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'user' },
            action: { type: 'c', default: 'loginByLoginId' },
            loginId: { type: 's' },
            password: { type: 's' },
            partnerId: { type: 'n' },
            expiry: { type: 'n' },
            privileges: { type: 's' },
            otp: { type: 's' }
        });
        return result;
    };
    return UserLoginByLoginIdAction;
}(kaltura_request_1.KalturaRequest));
exports.UserLoginByLoginIdAction = UserLoginByLoginIdAction;
