// kaltura-node-typescript-client/api/types/UserLoginAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserLoginAction = void 0;
var tslib_1 = require("tslib");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'user' action 'login'.
 *
 * Usage: Logs a user into a partner account with a partner ID, a partner user ID (puser), and a user password
 *
 * Server response type:         string
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var UserLoginAction = /** @class */ (function (_super) {
    tslib_1.__extends(UserLoginAction, _super);
    function UserLoginAction(data) {
        var _this = _super.call(this, data, { responseType: 's', responseSubType: '', responseConstructor: null }) || this;
        if (typeof _this.expiry === 'undefined')
            _this.expiry = 86400;
        if (typeof _this.privileges === 'undefined')
            _this.privileges = "*";
        return _this;
    }
    UserLoginAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'user' },
            action: { type: 'c', default: 'login' },
            partnerId: { type: 'n' },
            userId: { type: 's' },
            password: { type: 's' },
            expiry: { type: 'n' },
            privileges: { type: 's' }
        });
        return result;
    };
    return UserLoginAction;
}(kaltura_request_1.KalturaRequest));
exports.UserLoginAction = UserLoginAction;
