// kaltura-node-typescript-client/api/types/VarConsoleGetPartnerUsageAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VarConsoleGetPartnerUsageAction = void 0;
var tslib_1 = require("tslib");
var KalturaPartnerUsageListResponse_1 = require("./KalturaPartnerUsageListResponse");
var KalturaPartnerFilter_1 = require("./KalturaPartnerFilter");
var KalturaReportInputFilter_1 = require("./KalturaReportInputFilter");
var KalturaFilterPager_1 = require("./KalturaFilterPager");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'varConsole' action 'getPartnerUsage'.
 *
 * Usage: Function which calulates partner usage of a group of a VAR's sub-publishers
 *
 * Server response type:         KalturaPartnerUsageListResponse
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var VarConsoleGetPartnerUsageAction = /** @class */ (function (_super) {
    tslib_1.__extends(VarConsoleGetPartnerUsageAction, _super);
    function VarConsoleGetPartnerUsageAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaPartnerUsageListResponse', responseConstructor: KalturaPartnerUsageListResponse_1.KalturaPartnerUsageListResponse }) || this;
    }
    VarConsoleGetPartnerUsageAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'varconsole_varconsole' },
            action: { type: 'c', default: 'getPartnerUsage' },
            partnerFilter: { type: 'o', subTypeConstructor: KalturaPartnerFilter_1.KalturaPartnerFilter, subType: 'KalturaPartnerFilter' },
            usageFilter: { type: 'o', subTypeConstructor: KalturaReportInputFilter_1.KalturaReportInputFilter, subType: 'KalturaReportInputFilter' },
            pager: { type: 'o', subTypeConstructor: KalturaFilterPager_1.KalturaFilterPager, subType: 'KalturaFilterPager' }
        });
        return result;
    };
    return VarConsoleGetPartnerUsageAction;
}(kaltura_request_1.KalturaRequest));
exports.VarConsoleGetPartnerUsageAction = VarConsoleGetPartnerUsageAction;
