// kaltura-node-typescript-client/api/types/VendorCatalogItemAddFromBulkUploadAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VendorCatalogItemAddFromBulkUploadAction = void 0;
var tslib_1 = require("tslib");
var KalturaBulkUpload_1 = require("./KalturaBulkUpload");
var KalturaBulkUploadJobData_1 = require("./KalturaBulkUploadJobData");
var KalturaBulkUploadVendorCatalogItemData_1 = require("./KalturaBulkUploadVendorCatalogItemData");
var kaltura_upload_request_1 = require("../kaltura-upload-request");
/**
 * Build request payload for service 'vendorCatalogItem' action 'addFromBulkUpload'.
 *
 *
 *
 * Server response type:         KalturaBulkUpload
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaUploadRequest
 */
var VendorCatalogItemAddFromBulkUploadAction = /** @class */ (function (_super) {
    tslib_1.__extends(VendorCatalogItemAddFromBulkUploadAction, _super);
    function VendorCatalogItemAddFromBulkUploadAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaBulkUpload', responseConstructor: KalturaBulkUpload_1.KalturaBulkUpload }) || this;
    }
    VendorCatalogItemAddFromBulkUploadAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'reach_vendorcatalogitem' },
            action: { type: 'c', default: 'addFromBulkUpload' },
            fileData: { type: 'f' },
            bulkUploadData: { type: 'o', subTypeConstructor: KalturaBulkUploadJobData_1.KalturaBulkUploadJobData, subType: 'KalturaBulkUploadJobData' },
            bulkUploadVendorCatalogItemData: { type: 'o', subTypeConstructor: KalturaBulkUploadVendorCatalogItemData_1.KalturaBulkUploadVendorCatalogItemData, subType: 'KalturaBulkUploadVendorCatalogItemData' }
        });
        return result;
    };
    return VendorCatalogItemAddFromBulkUploadAction;
}(kaltura_upload_request_1.KalturaUploadRequest));
exports.VendorCatalogItemAddFromBulkUploadAction = VendorCatalogItemAddFromBulkUploadAction;
