// kaltura-node-typescript-client/api/types/VolatileInteractivityUpdateAction.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.VolatileInteractivityUpdateAction = void 0;
var tslib_1 = require("tslib");
var KalturaVolatileInteractivity_1 = require("./KalturaVolatileInteractivity");
var kaltura_request_1 = require("../kaltura-request");
/**
 * Build request payload for service 'volatileInteractivity' action 'update'.
 *
 * Usage: Update a volatile interactivity object
 *
 * Server response type:         KalturaVolatileInteractivity
 * Server failure response type: KalturaAPIException
 * @class
 * @extends KalturaRequest
 */
var VolatileInteractivityUpdateAction = /** @class */ (function (_super) {
    tslib_1.__extends(VolatileInteractivityUpdateAction, _super);
    function VolatileInteractivityUpdateAction(data) {
        return _super.call(this, data, { responseType: 'o', responseSubType: 'KalturaVolatileInteractivity', responseConstructor: KalturaVolatileInteractivity_1.KalturaVolatileInteractivity }) || this;
    }
    VolatileInteractivityUpdateAction.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { type: 'c', default: 'interactivity_volatileinteractivity' },
            action: { type: 'c', default: 'update' },
            entryId: { type: 's' },
            version: { type: 'n' },
            kalturaVolatileInteractivity: { type: 'o', subTypeConstructor: KalturaVolatileInteractivity_1.KalturaVolatileInteractivity, subType: 'KalturaVolatileInteractivity' }
        });
        return result;
    };
    return VolatileInteractivityUpdateAction;
}(kaltura_request_1.KalturaRequest));
exports.VolatileInteractivityUpdateAction = VolatileInteractivityUpdateAction;
