// kaltura-node-typescript-client/api/kaltura-multi-request.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.KalturaMultiRequest = void 0;
var tslib_1 = require("tslib");
var kaltura_request_base_1 = require("./kaltura-request-base");
var kaltura_multi_response_1 = require("./kaltura-multi-response");
var kaltura_api_exception_1 = require("./kaltura-api-exception");
var environment_1 = require("../environment");
var KalturaMultiRequest = /** @class */ (function (_super) {
    tslib_1.__extends(KalturaMultiRequest, _super);
    function KalturaMultiRequest() {
        var args = [];
        for (var _i = 0; _i < arguments.length; _i++) {
            args[_i] = arguments[_i];
        }
        var _this = _super.call(this, {}) || this;
        _this.requests = [];
        _this.requests = args;
        return _this;
    }
    KalturaMultiRequest.prototype.buildRequest = function (defaultRequestOptions) {
        var result = _super.prototype.toRequestObject.call(this);
        for (var i = 0, length_1 = this.requests.length; i < length_1; i++) {
            result[i] = this.requests[i].buildRequest(defaultRequestOptions);
        }
        return result;
    };
    KalturaMultiRequest.prototype._getMetadata = function () {
        var result = _super.prototype._getMetadata.call(this);
        Object.assign(result.properties, {
            service: { default: 'multirequest', type: 'c' }
        });
        return result;
    };
    KalturaMultiRequest.prototype._unwrapResponse = function (response) {
        if (environment_1.environment.response.nestedResponse) {
            if (response && response.hasOwnProperty('result')) {
                return response.result;
            }
            else if (response && response.hasOwnProperty('error')) {
                return response.error;
            }
        }
        return response;
    };
    KalturaMultiRequest.prototype.setCompletion = function (callback) {
        this.callback = callback;
        return this;
    };
    KalturaMultiRequest.prototype.handleResponse = function (responses) {
        var kalturaResponses = [];
        var unwrappedResponse = this._unwrapResponse(responses);
        var responseObject = null;
        if (!unwrappedResponse || !(unwrappedResponse instanceof Array) || unwrappedResponse.length !== this.requests.length) {
            var response = new kaltura_api_exception_1.KalturaAPIException("server response is invalid, expected array of ".concat(this.requests.length), 'client::response_type_error', null);
            for (var i = 0, len = this.requests.length; i < len; i++) {
                kalturaResponses.push(this.requests[i].handleResponse(response));
            }
        }
        else {
            for (var i = 0, len = this.requests.length; i < len; i++) {
                var serverResponse = unwrappedResponse[i];
                kalturaResponses.push(this.requests[i].handleResponse(serverResponse));
            }
            if (this.callback) {
                try {
                    this.callback(new kaltura_multi_response_1.KalturaMultiResponse(kalturaResponses));
                }
                catch (ex) {
                    // do nothing by design
                }
            }
        }
        return new kaltura_multi_response_1.KalturaMultiResponse(kalturaResponses);
    };
    return KalturaMultiRequest;
}(kaltura_request_base_1.KalturaRequestBase));
exports.KalturaMultiRequest = KalturaMultiRequest;
